library(tidyverse)
library(pheatmap)
library(dplyr)
library(RColorBrewer)
library(ggplot2)
library(tidyr)
library(Hmisc)
library(ggsci)
library(gridExtra)
library(readr)
library(seqinr)
library(VennDiagram)
library(phylotools)

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq")
Ath_symbols <- read.delim("~/GoogleDrive/My Drive/CRAG/RNAseq/Otros RNAseq/Woo et al (2016) Plant Physiol/Athaliana_447_Araport11.geneName.txt", header=FALSE)
colnames(Ath_symbols) <- c("ID", "Name")
Summary_Ath_FC <- read.table("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Otros RNAseq/Woo et al (2016) Plant Physiol/R/DESeq2_Woo_R_not_sig.txt",
                             sep = "\t", header = TRUE)
Summary_Ath_DEGs <- read.table("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Otros RNAseq/Woo et al (2016) Plant Physiol/R/Woo_R_Up&Down_filtered.txt",
                               sep = "\t", header = TRUE)
color3<-colorRampPalette(c("#1b9e77","#8ecdb6","#f7f7f7","#fba882","#d95f02"))(n=200)
############################################################################################
################################# ISOPRENOIDS FROM TOMATO TO BENTHAMIANA ###################

## First we load the Tomato Proteins and the file with all gene accesions of Isoprenoids that Manuel created
Tomato_4.0_fasta <- read.fasta("Otros RNAseq/Shinozaki et al (2018) Nat Com/ITAG4.0_proteins.fasta",
                               seqtype = "AA", as.string = TRUE)

Isoprenoids_Tomato <- read.table("Isoprenoids/Isoprenoid_Tomato_IDs.txt",
                                 sep = "\t", header = TRUE)
Tomato_4.0_description <- read_delim("Milan/RNAseq_Milan/06_GeneOntology/Tomato/2nd/Shinozaki/Slycopersicum_ITAG40.pep_description.txt", 
                                     "\t", escape_double = FALSE, comment = "#", 
                                     trim_ws = TRUE)

## We also load Pepper Proteins

Cannuum_ASM51225v2_fasta <- seqinr::read.fasta("Otros RNAseq/Gonzalez-Gordo (2019) Jour Exp Bot/Capsicum_annuum.ASM51225v2.pep.all.fa/Capsicum_annuum.ASM51225v2.pep.all.fasta",
                                       seqtype = "AA", as.string = TRUE)
Cannuum_ASM51225v2_description <- read_delim("Milan/RNAseq_Milan/06_GeneOntology/Pepper/Cannuum_ASM51225v2.pep.funct_ahrd_Annotation.txt", 
                                             "\t", escape_double = FALSE, comment = "#", col_names = FALSE,
                                             trim_ws = TRUE)
colnames(Cannuum_ASM51225v2_description) <- c("ID", "Description")

## We create a file to summarize the expression of Tomato genes in Yazdani experiment

Yazdani_Ripening_Breaker_DEG_FC <- read.table("Milan/RNAseq_Milan/DEGs_Transcript/Tomato/Yazdani/M82_B_vs_MG_filtered.txt",
                                              sep = "\t")
Yazdani_Ripening_Breaker_DEG_FC <- Yazdani_Ripening_Breaker_DEG_FC %>% select(log2FoldChange)
colnames(Yazdani_Ripening_Breaker_DEG_FC) <- "Ripening_Breaker_DEG"
Yazdani_Ripening_Breaker_All_FC <- read.table("Milan/RNAseq_Milan/DEGs_Transcript/Tomato/Yazdani/M82_B_vs_MG_All.txt", 
                                              sep = "\t")
Yazdani_Ripening_Breaker_All_FC <- Yazdani_Ripening_Breaker_All_FC %>% select(log2FoldChange)
colnames(Yazdani_Ripening_Breaker_All_FC) <- "Ripening_Breaker_All"
Yazdani_Ripening_Orange_DEG_FC <- read.table("Milan/RNAseq_Milan/DEGs_Transcript/Tomato/Yazdani/M82_or_vs_MG_filtered.txt",
                                             sep = "\t")
Yazdani_Ripening_Orange_DEG_FC <- Yazdani_Ripening_Orange_DEG_FC %>% select(log2FoldChange)
colnames(Yazdani_Ripening_Orange_DEG_FC) <- "Ripening_Orange_DEG"
Yazdani_Ripening_Orange_All_FC <- read.table("Milan/RNAseq_Milan/DEGs_Transcript/Tomato/Yazdani/M82_or_vs_MG_All.txt",
                                             sep = "\t")
Yazdani_Ripening_Orange_All_FC <- Yazdani_Ripening_Orange_All_FC %>% select(log2FoldChange)
colnames(Yazdani_Ripening_Orange_All_FC) <- "Ripening_Orange_All"
Yazdani_Ripening_Red_DEG_FC <- read.table("Milan/RNAseq_Milan/DEGs_Transcript/Tomato/Yazdani/M82_Red_vs_MG_filtered.txt",
                                          sep = "\t")
Yazdani_Ripening_Red_DEG_FC <- Yazdani_Ripening_Red_DEG_FC %>% select(log2FoldChange)
colnames(Yazdani_Ripening_Red_DEG_FC) <- "Ripening_Red_DEG"
Yazdani_Ripening_Red_All_FC <- read.table("Milan/RNAseq_Milan/DEGs_Transcript/Tomato/Yazdani/M82_Red_vs_MG_All.txt",
                                          sep = "\t")
Yazdani_Ripening_Red_All_FC <- Yazdani_Ripening_Red_All_FC %>% select(log2FoldChange)
colnames(Yazdani_Ripening_Red_All_FC) <- "Ripening_Red_All"
Yazdani_ORHis_MG_DEG_FC <- read.table("Milan/RNAseq_Milan/DEGs_Transcript/Tomato/Yazdani/MG_ORHis_vs_MG_M82_filtered.txt",
                                      sep = "\t")
Yazdani_ORHis_MG_DEG_FC <- Yazdani_ORHis_MG_DEG_FC %>% select(log2FoldChange)
colnames(Yazdani_ORHis_MG_DEG_FC) <- "ORHis_MG_DEG"
Yazdani_ORHis_MG_All_FC <- read.table("Milan/RNAseq_Milan/DEGs_Transcript/Tomato/Yazdani/MG_ORHis_vs_MG_M82_All.txt",
                                      sep = "\t")
Yazdani_ORHis_MG_All_FC <- Yazdani_ORHis_MG_All_FC %>% select(log2FoldChange)
colnames(Yazdani_ORHis_MG_All_FC) <- "ORHis_MG_All"
Yazdani_ORHis_Breaker_DEG_FC <- read.table("Milan/RNAseq_Milan/DEGs_Transcript/Tomato/Yazdani/B_ORHis_vs_B_M82_filtered.txt",
                                           sep = "\t")
Yazdani_ORHis_Breaker_DEG_FC <- Yazdani_ORHis_Breaker_DEG_FC %>% select(log2FoldChange)
colnames(Yazdani_ORHis_Breaker_DEG_FC) <- "ORHis_Breaker_DEG"
Yazdani_ORHis_Breaker_All_FC <- read.table("Milan/RNAseq_Milan/DEGs_Transcript/Tomato/Yazdani/B_ORHis_vs_B_M82_All.txt",
                                           sep = "\t")
Yazdani_ORHis_Breaker_All_FC <- Yazdani_ORHis_Breaker_All_FC %>% select(log2FoldChange)
colnames(Yazdani_ORHis_Breaker_All_FC) <- "ORHis_Breaker_All"

Summary_Yazdani_TPM_FC_Ann <- merge(Yazdani_Ripening_Breaker_All_FC, Yazdani_Ripening_Orange_All_FC, by = 0, all = TRUE)
colnames(Summary_Yazdani_TPM_FC_Ann)[1] <- "GeneID"
Summary_Yazdani_TPM_FC_Ann <- merge(Summary_Yazdani_TPM_FC_Ann, Yazdani_Ripening_Red_All_FC, by.x = "GeneID", by.y = 0, all = TRUE)
Summary_Yazdani_TPM_FC_Ann <- merge(Summary_Yazdani_TPM_FC_Ann, Yazdani_ORHis_MG_All_FC, by.x = "GeneID", by.y = 0, all = TRUE)
Summary_Yazdani_TPM_FC_Ann <- merge(Summary_Yazdani_TPM_FC_Ann, Yazdani_ORHis_Breaker_All_FC, by.x = "GeneID", by.y = 0, all = TRUE)
Summary_Yazdani_TPM_FC_Ann <- merge(Summary_Yazdani_TPM_FC_Ann, Yazdani_Ripening_Breaker_DEG_FC, by.x = "GeneID", by.y = 0, all = TRUE)
Summary_Yazdani_TPM_FC_Ann <- merge(Summary_Yazdani_TPM_FC_Ann, Yazdani_Ripening_Orange_DEG_FC, by.x = "GeneID", by.y = 0, all = TRUE)
Summary_Yazdani_TPM_FC_Ann <- merge(Summary_Yazdani_TPM_FC_Ann, Yazdani_Ripening_Red_DEG_FC, by.x = "GeneID", by.y = 0, all = TRUE)
Summary_Yazdani_TPM_FC_Ann <- merge(Summary_Yazdani_TPM_FC_Ann, Yazdani_ORHis_MG_DEG_FC, by.x = "GeneID", by.y = 0, all = TRUE)
Summary_Yazdani_TPM_FC_Ann <- merge(Summary_Yazdani_TPM_FC_Ann, Yazdani_ORHis_Breaker_DEG_FC, by.x = "GeneID", by.y = 0, all = TRUE)

Sumarry_Yazdani_TPM_All <- read.table("Summary_Yazdani_TPM_FC_AnnDEGs_Transcript/Tomato/Yazdani/TPM_Yazdani_all.txt",
                                      header = TRUE)
Summary_Yazdani_TPM_FC_Ann <- merge(Sumarry_Yazdani_TPM_All, Summary_Yazdani_TPM_FC_Ann, by.x = 1, by.y = 1, all.x = TRUE)
Summary_Yazdani_TPM_FC_Ann <- merge(Tomato_4.0_description, Summary_Yazdani_TPM_FC_Ann, by.x = 1, by.y = 1)
Summary_Yazdani_TPM_FC_Ann <- Summary_Yazdani_TPM_FC_Ann %>%
  select(1:11, 15:17, 12:14, 18:23, 45:53, 57:59, 54:56, 60:65, 24:32, 36:38, 33:35, 39:44, 66:75)

## We create a file to summarize the expression of Pepper genes in Pepper ripening experiment

Pepper_Ripening_Breaker_DEG_FC <- read.table("Milan/RNAseq_Milan/DEGs_Transcript/Pepper/Breaker_vs_Green_filtered.txt",
                                              sep = "\t")
Pepper_Ripening_Breaker_DEG_FC <- Pepper_Ripening_Breaker_DEG_FC %>% select(log2FoldChange)
colnames(Pepper_Ripening_Breaker_DEG_FC) <- "Ripening_Breaker_DEG"
Pepper_Ripening_Breaker_All_FC <- read.table("Milan/RNAseq_Milan/DEGs_Transcript/Pepper/Breaker_vs_Green_All.txt", 
                                              sep = "\t")
Pepper_Ripening_Breaker_All_FC <- Pepper_Ripening_Breaker_All_FC %>% select(log2FoldChange)
colnames(Pepper_Ripening_Breaker_All_FC) <- "Ripening_Breaker_All"
Pepper_Ripening_Red_DEG_FC <- read.table("Milan/RNAseq_Milan/DEGs_Transcript/Pepper/Red_vs_Green_filtered.txt",
                                             sep = "\t")
Pepper_Ripening_Red_DEG_FC <- Pepper_Ripening_Red_DEG_FC %>% select(log2FoldChange)
colnames(Pepper_Ripening_Red_DEG_FC) <- "Ripening_Red_DEG"
Pepper_Ripening_Red_All_FC <- read.table("Milan/RNAseq_Milan/DEGs_Transcript/Pepper/Red_vs_Green_All.txt",
                                          sep = "\t")
Pepper_Ripening_Red_All_FC <- Pepper_Ripening_Red_All_FC %>% select(log2FoldChange)
colnames(Pepper_Ripening_Red_All_FC) <- "Ripening_Red_All"


Summary_Pepper_TPM_FC_Ann <- merge(Pepper_Ripening_Breaker_All_FC, Pepper_Ripening_Red_All_FC, by = 0, all = TRUE)
colnames(Summary_Pepper_TPM_FC_Ann)[1] <- "GeneID"
Summary_Pepper_TPM_FC_Ann <- merge(Summary_Pepper_TPM_FC_Ann, Pepper_Ripening_Breaker_DEG_FC, by.x = "GeneID", by.y = 0, all = TRUE)
Summary_Pepper_TPM_FC_Ann <- merge(Summary_Pepper_TPM_FC_Ann, Pepper_Ripening_Red_DEG_FC, by.x = "GeneID", by.y = 0, all = TRUE)

Sumarry_Pepper_TPM_All <- read.table("Milan/RNAseq_Milan/DEGs_Transcript/Pepper/TPM_Pepper_All.txt",
                                      header = TRUE)
Summary_Pepper_TPM_FC_Ann <- merge(Sumarry_Pepper_TPM_All, Summary_Pepper_TPM_FC_Ann, by.x = 1, by.y = 1, all.x = TRUE)
Summary_Pepper_TPM_FC_Ann <- merge(Cannuum_ASM51225v2_description, Summary_Pepper_TPM_FC_Ann, by.x = 1, by.y = 1)


###########################################################################################
############################ CAROTENOIDS ##################################################

## We subset this genes, found their AA sequence and save the file
Tomato_Carotenoids_Name <- Isoprenoids_Tomato[Isoprenoids_Tomato$Pathway=="Carotenoid_biosynthesis",] %>%
  select(SOL)

Tomato_Carotenoids_Fasta <- Tomato_4.0_fasta[c(which(substr(names(Tomato_4.0_fasta), 1, 14) %in% Tomato_Carotenoids_Name$SOL))]
write.fasta(Tomato_Carotenoids_Fasta, as.string = TRUE, names = names(Tomato_Carotenoids_Fasta),
            file.out = "Milan/RNAseq_Milan/Funcional Groups/Tomato_Carotenoids.fasta")
write.table(names(Tomato_Carotenoids_Fasta), file="Milan/RNAseq_Milan/Funcional Groups/Tomato_Carotenoids.txt",
            quote = FALSE, sep = "\t", row.names = FALSE, col.names = FALSE)

## We load the file with the orthologs in benthamiana from OrthoVenn proccessed with Python
Tomato_Niben_Carotenoids <- read.csv("Milan/RNAseq_Milan/Funcional Groups/Niben_Carotenoids.csv", na.strings = "")
Tomato_Niben_Carotenoids[1] <- NULL
Tomato_Niben_Carotenoids <- Tomato_Niben_Carotenoids %>%
  gather(key="Tomato", value="Niben")
Tomato_Niben_Carotenoids <- na.omit(Tomato_Niben_Carotenoids)
Tomato_Niben_Carotenoids$SOL <- substr(Tomato_Niben_Carotenoids$Tomato, 1, 14)
Tomato_Niben_Carotenoids <- merge(Tomato_Niben_Carotenoids, Isoprenoids_Tomato, by="SOL")
Tomato_Niben_Carotenoids <- Tomato_Niben_Carotenoids[order(match(Tomato_Niben_Carotenoids[,1],Tomato_Carotenoids_Name[,1])),]

## We create a new df only with benthamiana accesions

Niben_Carotenoids <- Tomato_Niben_Carotenoids %>% 
  select(Niben, Name)
colnames(Niben_Carotenoids) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_Carotenoids <- merge(Niben_Carotenoids, TPM_All_FC_Annotated, by="Name")

## We select only the FC data from the different comparisons
Niben_Carotenoids <- Niben_Carotenoids %>% select(Name, 88:105)
Niben_Carotenoids <- Niben_Carotenoids[order(match(Niben_Carotenoids[,1], Tomato_Niben_Carotenoids[,3])),]

## We prepare the data for a heatmap and we map it
Niben_Carotenoids_Annotation <- merge(Niben_Carotenoids, Tomato_Niben_Carotenoids, by.x="Name", by.y = "Niben")
Niben_Carotenoids_Annotation <- Niben_Carotenoids_Annotation[order(match(Niben_Carotenoids_Annotation[,1], Tomato_Niben_Carotenoids[,3])),]
rownames(Niben_Carotenoids_Annotation) <- Niben_Carotenoids_Annotation$Name
Niben_Carotenoids_Annotation <- Niben_Carotenoids_Annotation %>% select(Name.y)
colnames(Niben_Carotenoids_Annotation) <- "Enzymes"

rownames(Niben_Carotenoids) <- Niben_Carotenoids$Name
Niben_Carotenoids[1] <- NULL

color<-colorRampPalette(c("#4DAC26","#B8E186", "#F7F7F7", "#F1B6DA", "#D01C8B"))(n=200)
color2<-colorRampPalette(c("#0571b0","#92c5de", "#F7F7F7", "#f4a582", "#ca0020"))(n=200)
color3 <-colorRampPalette(c("#fef0d9","#fdcc8a", "#fc8d59", "#e34a33", "#b30000"))(n=200)

Niben_Carotenoids <- Niben_Carotenoids %>%
  select(pcrtB6h_vs_GFP6h_filtered, pcrtB24h_vs_GFP24h_filtered, pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

min(Niben_Carotenoids, na.rm = TRUE)
max(Niben_Carotenoids, na.rm = TRUE)
breaks_Carotenoids=c(seq(-2,0, length.out = 100), seq(0.01,2,length.out = 100))
breaks_Stndr=c(seq(-4,0, length.out = 100), seq(0.01,4,length.out = 100))

color_Carotenoids <- colorRampPalette(c("#415a20","#9bac12","#e5c033","#c38144","#894539"))(n=length(unique(Niben_Carotenoids_Annotation$Enzymes)))
names(color_Carotenoids) <- unique(Niben_Carotenoids_Annotation$Enzymes)
color_Carotenoids <- list(Enzymes=color_Carotenoids)

pheatmap(Niben_Carotenoids, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         #breaks = breaks_Carotenoids,
         breaks = breaks_Stndr,
         legend_breaks = c(-4, -2, 0, 2, 4),
         legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         annotation_row=Niben_Carotenoids_Annotation, annotation_colors=color_Carotenoids, 
         annotation_names_row = TRUE, show_rownames = FALSE,  
         cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("6h", "24h", "22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h", "96h"),
         cex = 1,
         angle_col=45,
         main = "Carotenoids",
         annotation_legend = TRUE,
         #display_numbers = Nb_Sly_Ath_MVA_Sig,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)


## PSY candidates from BLAST and tree-cluster in MEGA X

Niben261_fasta <- read.fasta("Milan/RNAseq_Milan/Niben261_genome_complete01_manual_proteins.fasta",
                             seqtype = "AA", as.string = TRUE)
PSY_Niben <- c("Niben261Chr16g0522010.1", "Niben261Chr17g0776007.1", "Niben261Chr14g0831007.1", "Niben261Chr02g0115010.1", "Niben261Chr02g0115010.1", "Niben261Chr02g0433017.1")
PSY_Niben_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% PSY_Niben))]
write.fasta(PSY_Niben_fasta, as.string=TRUE, names= names(PSY_Niben_fasta),
            file.out = "Milan/RNAseq_Milan/Funcional Groups/Niben_PSY.fasta")

PSY_Tomato <- Isoprenoids_Tomato[Isoprenoids_Tomato$Name=="PSY1"| Isoprenoids_Tomato$Name=="PSY2"|Isoprenoids_Tomato$Name=="PSY3",] %>%
  select(SOL)
PSY_Tomato_fasta <- Tomato_4.0_fasta[c(which(substr(names(Tomato_4.0_fasta), 1, 14) %in% PSY_Tomato$SOL))]
write.fasta(PSY_Tomato_fasta, as.string = TRUE, names = names(PSY_Tomato_fasta),
            file.out = "Milan/RNAseq_Milan/Funcional Groups/Tomato_PSY.fasta")

Niben_Carotenoids_manual <- Tomato_Niben_Carotenoids %>% 
  select(Niben, Name)
colnames(Niben_Carotenoids_manual) <- c("Name", "Protein")

Niben_Carotenoids_manual <- Niben_Carotenoids_manual %>% 
  add_row("Name"="Niben261Chr17g0776007.1", "Protein"="PSY1", .before = 1) %>%
  add_row("Name"="Niben261Chr14g0831007.1", "Protein"="PSY1", .before = 2) %>%
  add_row("Name"="Niben261Chr02g0433017.1", "Protein"="PSY3", .before = 5)
  
Niben_Carotenoids_manual <- merge(Niben_Carotenoids_manual, TPM_All_FC_Annotated, by="Name")
Niben_Carotenoids_manual <- Niben_Carotenoids_manual %>% select(Name, 88:105)
Niben_Carotenoids_manual <- Niben_Carotenoids_manual[order(match(Niben_Carotenoids_manual[,1], Tomato_Niben_Carotenoids[,3])),]

rownames(Niben_Carotenoids_manual) <- Niben_Carotenoids_manual$Name
Niben_Carotenoids_manual[1] <- NULL

pheatmap(Niben_Carotenoids_manual,
         scale = "none",
         cluster_rows = FALSE,
         cluster_cols = FALSE)


###########################################################################################
############################ MEP pathway ##################################################

## We subset this genes, found their AA sequence and save the file
Tomato_MEP_Name <- Isoprenoids_Tomato[Isoprenoids_Tomato$Pathway=="MEP"|Isoprenoids_Tomato$Pathway=="IDI"|Isoprenoids_Tomato$Pathway=="GGPPS"|Isoprenoids_Tomato$Pathway=="Orange",] %>%
  select(SOL)

Tomato_MEP_Fasta <- Tomato_4.0_fasta[c(which(substr(names(Tomato_4.0_fasta), 1, 14) %in% Tomato_MEP_Name$SOL))]
write.fasta(Tomato_MEP_Fasta, as.string = TRUE, names = names(Tomato_MEP_Fasta),
            file.out = "Milan/RNAseq_Milan/Funcional Groups/Tomato_MEP.fasta")
write.table(names(Tomato_MEP_Fasta), file="Milan/RNAseq_Milan/Funcional Groups/Tomato_MEP.txt",
            quote = FALSE, sep = "\t", row.names = FALSE, col.names = FALSE)

## We load the file with the orthologs in benthamiana from OrthoVenn proccessed with Python
Tomato_Niben_MEP <- read.csv("Milan/RNAseq_Milan/Funcional Groups/Niben_MEP.csv", na.strings = "")
Tomato_Niben_MEP[1] <- NULL
Tomato_Niben_MEP <- Tomato_Niben_MEP %>%
  gather(key="Tomato", value="Niben")
Tomato_Niben_MEP <- na.omit(Tomato_Niben_MEP)
Tomato_Niben_MEP$SOL <- substr(Tomato_Niben_MEP$Tomato, 1, 14)
Tomato_Niben_MEP <- merge(Tomato_Niben_MEP, Isoprenoids_Tomato, by="SOL")
Tomato_Niben_MEP <- Tomato_Niben_MEP[order(match(Tomato_Niben_MEP[,1],Tomato_MEP_Name[,1])),]

## We create a new df only with benthamiana accesions

Niben_MEP <- Tomato_Niben_MEP %>% 
  select(Niben, Name)
colnames(Niben_MEP) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_MEP <- merge(Niben_MEP, TPM_All_FC_Annotated, by="Name")

## We select only the FC data from the different comparisons
Niben_MEP <- Niben_MEP %>% select(Name, 88:105)
Niben_MEP <- Niben_MEP[-c(13,27),]
Niben_MEP <- Niben_MEP[order(match(Niben_MEP[,1], Tomato_Niben_MEP[,3])),]


## We prepare the data for a heatmap and we map it

Niben_MEP_Annotation <- merge(Niben_MEP, Tomato_Niben_MEP, by.x="Name", by.y = "Niben")
Niben_MEP_Annotation <- Niben_MEP_Annotation[-c(13,27),]
Niben_MEP_Annotation <- Niben_MEP_Annotation[order(match(Niben_MEP_Annotation[,1], Tomato_Niben_MEP[,3])),]
rownames(Niben_MEP_Annotation) <- Niben_MEP_Annotation$Name
Niben_MEP_Annotation <- Niben_MEP_Annotation %>% select(Name.y)

rownames(Niben_MEP) <- Niben_MEP$Name
Niben_MEP[1] <- NULL

Niben_MEP <- Niben_MEP %>%
  select(pcrtB6h_vs_GFP6h_filtered, pcrtB24h_vs_GFP24h_filtered, pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

min(Niben_MEP, na.rm = TRUE)
max(Niben_MEP, na.rm = TRUE)
breaks_MEP=c(seq(-4,0, length.out = 100), seq(0.01,4,length.out = 100))

color_MEP <- list(
  Name.y = c("DXS1"="#ffd9ff", "DXS2"= "#ffb5f0", "DXS3a"="#f498d1", "DXR1"="#dc80b9", "DXR2"="#c46aa2", 
           "MCT"="#ac548c", "CMK"="#943e76", "MDS"="#7d2861", "HDS"="#670e4d", "HDR"="#4b0034",
           "IDI1"="#8da0cb", "IDI2"="#5c7098", "IDI3"="#2d4469",
           "GGDS1"="#92eed0", "GGDS2"="#67c3a6", "GGDS4"="#3c9a7e", "GGDS5"="#057259", "GGDS6"="#004a34","GGDS8"="#002511",
           "Or1"="#ed8056", "Or2"="#ad4a26")
)

pheatmap(Niben_MEP, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         #breaks = breaks_MEP,
         breaks = breaks_Stndr,
         legend_breaks = c(-4, -2, 0, 2, 4),
         legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         annotation_row=Niben_MEP_Annotation, annotation_colors=color_MEP, 
         annotation_names_row = TRUE, show_rownames = FALSE,  
         cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("6h", "24h", "22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h", "96h"),
         cex = 1,
         angle_col=45,
         main = "MEP",
         annotation_legend = TRUE,
         #display_numbers = Nb_Sly_Ath_MVA_Sig,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)


###########################################################################################
############################ CAROTENOID DEGRADATION ##################################################

## We subset this genes, found their AA sequence and save the file
Tomato_Degrad_Name <- Isoprenoids_Tomato[Isoprenoids_Tomato$Pathway=="Carotenoid_degradation",] %>%
  select(SOL)

Tomato_Degrad_Fasta <- Tomato_4.0_fasta[c(which(substr(names(Tomato_4.0_fasta), 1, 14) %in% Tomato_Degrad_Name$SOL))]
write.fasta(Tomato_Degrad_Fasta, as.string = TRUE, names = names(Tomato_Degrad_Fasta),
            file.out = "Milan/RNAseq_Milan/Funcional Groups/Tomato_Degrad.fasta")
write.table(names(Tomato_Degrad_Fasta), file="Milan/RNAseq_Milan/Funcional Groups/Tomato_Degrad.txt",
            quote = FALSE, sep = "\t", row.names = FALSE, col.names = FALSE)

## We load the file with the orthologs in benthamiana from OrthoVenn proccessed with Python
Tomato_Niben_Degrad <- read.csv("Milan/RNAseq_Milan/Funcional Groups/Niben_Degrad.csv", na.strings = "")
Tomato_Niben_Degrad[1] <- NULL
Tomato_Niben_Degrad <- Tomato_Niben_Degrad %>%
  gather(key="Tomato", value="Niben")
Tomato_Niben_Degrad <- na.omit(Tomato_Niben_Degrad)
Tomato_Niben_Degrad$SOL <- substr(Tomato_Niben_Degrad$Tomato, 1, 14)
Tomato_Niben_Degrad <- merge(Tomato_Niben_Degrad, Isoprenoids_Tomato, by="SOL")
Tomato_Niben_Degrad <- Tomato_Niben_Degrad[order(match(Tomato_Niben_Degrad[,1],Tomato_Degrad_Name[,1])),]

## We create a new df only with benthamiana accesions

Niben_Degrad <- Tomato_Niben_Degrad %>% 
  select(Niben, Name)
colnames(Niben_Degrad) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_Degrad <- merge(Niben_Degrad, TPM_All_FC_Annotated, by="Name")

## We select only the FC data from the different comparisons
Niben_Degrad <- Niben_Degrad %>% select(Name, 88:105)
Niben_Degrad <- Niben_Degrad[order(match(Niben_Degrad[,1], Tomato_Niben_Degrad[,3])),]

## We prepare the data for a heatmap and we map it
Niben_Degrad_Annotation <- merge(Niben_Degrad, Tomato_Niben_Degrad, by.x="Name", by.y = "Niben")
Niben_Degrad_Annotation <- Niben_Degrad_Annotation[order(match(Niben_Degrad_Annotation[,1], Tomato_Niben_Degrad[,3])),]
rownames(Niben_Degrad_Annotation) <- Niben_Degrad_Annotation$Name
Niben_Degrad_Annotation <- Niben_Degrad_Annotation %>% select(Name.y)
colnames(Niben_Degrad_Annotation) <- "Enzymes"

rownames(Niben_Degrad) <- Niben_Degrad$Name
Niben_Degrad[1] <- NULL

Niben_Degrad <- Niben_Degrad %>%
  select(pcrtB6h_vs_GFP6h_filtered, pcrtB24h_vs_GFP24h_filtered, pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

min(Niben_Degrad, na.rm = TRUE)
max(Niben_Degrad, na.rm = TRUE)
breaks_Degrad=c(seq(-2,0, length.out = 100), seq(0.01,3,length.out = 100))

color_Degrad <- colorRampPalette(c("#9bac12","#e5c033","#c38144","#894539"))(n=length(unique(Niben_Degrad_Annotation$Enzymes)))
names(color_Degrad) <- unique(Niben_Degrad_Annotation$Enzymes)
color_Degrad <- list(Enzymes=color_Degrad)

pheatmap(Niben_Degrad, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         #breaks = breaks_Degrad,
         breaks = breaks_Stndr,
         legend_breaks = c(-4, -2, 0, 2, 4),
         legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         annotation_row=Niben_Degrad_Annotation, annotation_colors=color_Degrad, 
         annotation_names_row = TRUE, show_rownames = FALSE,  
         cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("6h", "24h", "22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h", "96h"),
         cex = 1,
         angle_col=45,
         main = "Carotenoid Degradation",
         annotation_legend = TRUE,
         #display_numbers = Nb_Sly_Ath_MVA_Sig,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)


###########################################################################################
############################ ABA SYNTHESIS ##################################################

## We subset this genes, found their AA sequence and save the file
Tomato_ABA_Name <- Isoprenoids_Tomato[Isoprenoids_Tomato$Pathway=="ABA",] %>%
  select(SOL)

Tomato_ABA_Fasta <- Tomato_4.0_fasta[c(which(substr(names(Tomato_4.0_fasta), 1, 14) %in% Tomato_ABA_Name$SOL))]
write.fasta(Tomato_ABA_Fasta, as.string = TRUE, names = names(Tomato_ABA_Fasta),
            file.out = "Milan/RNAseq_Milan/Funcional Groups/Tomato_ABA.fasta")
write.table(names(Tomato_ABA_Fasta), file="Milan/RNAseq_Milan/Funcional Groups/Tomato_ABA.txt",
            quote = FALSE, sep = "\t", row.names = FALSE, col.names = FALSE)

## We load the file with the orthologs in benthamiana from OrthoVenn proccessed with Python
Tomato_Niben_ABA <- read.csv("Milan/RNAseq_Milan/Funcional Groups/Niben_ABA.csv", na.strings = "")
Tomato_Niben_ABA[1] <- NULL
Tomato_Niben_ABA <- Tomato_Niben_ABA %>%
  gather(key="Tomato", value="Niben")
Tomato_Niben_ABA <- na.omit(Tomato_Niben_ABA)
Tomato_Niben_ABA$SOL <- substr(Tomato_Niben_ABA$Tomato, 1, 14)
Tomato_Niben_ABA <- merge(Tomato_Niben_ABA, Isoprenoids_Tomato, by="SOL")
Tomato_Niben_ABA <- Tomato_Niben_ABA[-c(5,6,7),]
Tomato_Niben_ABA <- Tomato_Niben_ABA[order(match(Tomato_Niben_ABA[,1],Tomato_ABA_Name[,1])),]

## We create a new df only with benthamiana accesions

Niben_ABA <- Tomato_Niben_ABA %>% 
  select(Niben, Name)
colnames(Niben_ABA) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_ABA <- merge(Niben_ABA, TPM_All_FC_Annotated, by="Name")

## We select only the FC data from the different comparisons
Niben_ABA <- Niben_ABA %>% select(Name, 88:105)
Niben_ABA <- Niben_ABA[-c(23,25),]
Niben_ABA <- Niben_ABA[order(match(Niben_ABA[,1], Tomato_Niben_ABA[,3])),]

## We prepare the data for a heatmap and we map it
Niben_ABA_Annotation <- merge(Niben_ABA, Tomato_Niben_ABA, by.x="Name", by.y = "Niben")
Niben_ABA_Annotation <- Niben_ABA_Annotation[-c(23,25),]
Niben_ABA_Annotation <- Niben_ABA_Annotation[order(match(Niben_ABA_Annotation[,1], Tomato_Niben_ABA[,3])),]
rownames(Niben_ABA_Annotation) <- Niben_ABA_Annotation$Name
Niben_ABA_Annotation <- Niben_ABA_Annotation %>% select(Name.y)
colnames(Niben_ABA_Annotation) <- "Enzymes"

rownames(Niben_ABA) <- Niben_ABA$Name
Niben_ABA[1] <- NULL

Niben_ABA <- Niben_ABA %>%
  select(pcrtB6h_vs_GFP6h_filtered, pcrtB24h_vs_GFP24h_filtered, pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

min(Niben_ABA, na.rm = TRUE)
max(Niben_ABA, na.rm = TRUE)
breaks_ABA=c(seq(-3,0, length.out = 100), seq(0.01,6,length.out = 100))

color_ABA <- colorRampPalette(c("#415a20","#9bac12","#e5c033","#c38144","#894539"))(n=length(unique(Niben_ABA_Annotation$Enzymes)))
names(color_ABA) <- unique(Niben_ABA_Annotation$Enzymes)
color_ABA <- list(Enzymes=color_ABA)

pheatmap(Niben_ABA, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         #breaks = breaks_ABA,
         breaks = breaks_Stndr,
         legend_breaks = c(-4, -2, 0, 2, 4),
         legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         annotation_row=Niben_ABA_Annotation, annotation_colors=color_ABA, 
         annotation_names_row = TRUE, show_rownames = FALSE,  
         cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("6h", "24h", "22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h", "96h"),
         cex = 1,
         angle_col=45,
         main = "ABA",
         annotation_legend = TRUE,
         #display_numbers = Nb_Sly_Ath_MVA_Sig,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)


###########################################################################################
############################ TOCOPHEROLS SYNTHESIS ##################################################

## We subset this genes, found their AA sequence and save the file
Tomato_Tocopherols_Name <- Isoprenoids_Tomato[Isoprenoids_Tomato$Pathway=="Tocopherols",] %>%
  select(SOL)

Tomato_Tocopherols_Fasta <- Tomato_4.0_fasta[c(which(substr(names(Tomato_4.0_fasta), 1, 14) %in% Tomato_Tocopherols_Name$SOL))]
write.fasta(Tomato_Tocopherols_Fasta, as.string = TRUE, names = names(Tomato_Tocopherols_Fasta),
            file.out = "Milan/RNAseq_Milan/Funcional Groups/Tomato_Tocopherols.fasta")
write.table(names(Tomato_Tocopherols_Fasta), file="Milan/RNAseq_Milan/Funcional Groups/Tomato_Tocopherols.txt",
            quote = FALSE, sep = "\t", row.names = FALSE, col.names = FALSE)

## We load the file with the orthologs in benthamiana from OrthoVenn proccessed with Python
Tomato_Niben_Tocopherols <- read.csv("Milan/RNAseq_Milan/Funcional Groups/Niben_Tocopherols.csv", na.strings = "")
Tomato_Niben_Tocopherols[1] <- NULL
Tomato_Niben_Tocopherols <- Tomato_Niben_Tocopherols %>%
  gather(key="Tomato", value="Niben")
Tomato_Niben_Tocopherols <- na.omit(Tomato_Niben_Tocopherols)
Tomato_Niben_Tocopherols$SOL <- substr(Tomato_Niben_Tocopherols$Tomato, 1, 14)
Tomato_Niben_Tocopherols <- merge(Tomato_Niben_Tocopherols, Isoprenoids_Tomato, by="SOL")
Tomato_Niben_Tocopherols <- Tomato_Niben_Tocopherols[order(match(Tomato_Niben_Tocopherols[,1],Tomato_Tocopherols_Name[,1])),]

## We create a new df only with benthamiana accesions

Niben_Tocopherols <- Tomato_Niben_Tocopherols %>% 
  select(Niben, Name)
colnames(Niben_Tocopherols) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_Tocopherols <- merge(Niben_Tocopherols, TPM_All_FC_Annotated, by="Name")

## We select only the FC data from the different comparisons
Niben_Tocopherols <- Niben_Tocopherols %>% select(Name, 88:105)
Niben_Tocopherols <- Niben_Tocopherols[order(match(Niben_Tocopherols[,1], Tomato_Niben_Tocopherols[,3])),]

## We prepare the data for a heatmap and we map it
Niben_Tocopherols_Annotation <- merge(Niben_Tocopherols, Tomato_Niben_Tocopherols, by.x="Name", by.y = "Niben")
Niben_Tocopherols_Annotation <- Niben_Tocopherols_Annotation[order(match(Niben_Tocopherols_Annotation[,1], Tomato_Niben_Tocopherols[,3])),]
rownames(Niben_Tocopherols_Annotation) <- Niben_Tocopherols_Annotation$Name
Niben_Tocopherols_Annotation <- Niben_Tocopherols_Annotation %>% select(Name.y)
colnames(Niben_Tocopherols_Annotation) <- "Enzymes"

rownames(Niben_Tocopherols) <- Niben_Tocopherols$Name
Niben_Tocopherols[1] <- NULL

Niben_Tocopherols <- Niben_Tocopherols %>%
  select(pcrtB6h_vs_GFP6h_filtered, pcrtB24h_vs_GFP24h_filtered, pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

min(Niben_Tocopherols, na.rm = TRUE)
max(Niben_Tocopherols, na.rm = TRUE)
breaks_Tocopherols=c(seq(-1,0, length.out = 100), seq(0.01,3,length.out = 100))

color_Tocopherols <- colorRampPalette(c("#415a20","#9bac12","#e5c033","#c38144","#894539"))(n=length(unique(Niben_Tocopherols_Annotation$Enzymes)))
names(color_Tocopherols) <- unique(Niben_Tocopherols_Annotation$Enzymes)
color_Tocopherols <- list(Enzymes=color_Tocopherols)

pheatmap(Niben_Tocopherols, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         #breaks = breaks_Tocopherols,
         breaks = breaks_Stndr,
         legend_breaks = c(-4, -2, 0, 2, 4),
         legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         annotation_row=Niben_Tocopherols_Annotation, annotation_colors=color_Tocopherols, 
         annotation_names_row = TRUE, show_rownames = FALSE,  
         cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("6h", "24h", "22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h", "96h"),
         cex = 1,
         angle_col=45,
         main = "Tocopherols",
         annotation_legend = TRUE,
         #display_numbers = Nb_Sly_Ath_MVA_Sig,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)

###########################################################################################
############################ CHLOROPHYLLS SYNTHESIS ##################################################

## We subset this genes, found their AA sequence and save the file
Tomato_Chlorophylls_Name <- Isoprenoids_Tomato[Isoprenoids_Tomato$Pathway=="Tetrapyrrole_biosynthesis"|Isoprenoids_Tomato$Pathway=="Chlorophyll_biosynthesis",] %>%
  select(SOL)

Tomato_Chlorophylls_Fasta <- Tomato_4.0_fasta[c(which(substr(names(Tomato_4.0_fasta), 1, 14) %in% Tomato_Chlorophylls_Name$SOL))]
write.fasta(Tomato_Chlorophylls_Fasta, as.string = TRUE, names = names(Tomato_Chlorophylls_Fasta),
            file.out = "Milan/RNAseq_Milan/Funcional Groups/Tomato_Chlorophylls.fasta")
write.table(names(Tomato_Chlorophylls_Fasta), file="Milan/RNAseq_Milan/Funcional Groups/Tomato_Chlorophylls.txt",
            quote = FALSE, sep = "\t", row.names = FALSE, col.names = FALSE)

## We load the file with the orthologs in benthamiana from OrthoVenn proccessed with Python
Tomato_Niben_Chlorophylls <- read.csv("Milan/RNAseq_Milan/Funcional Groups/Niben_Chlorophylls.csv", na.strings = "")
Tomato_Niben_Chlorophylls[1] <- NULL
Tomato_Niben_Chlorophylls <- Tomato_Niben_Chlorophylls %>%
  gather(key="Tomato", value="Niben")
Tomato_Niben_Chlorophylls <- na.omit(Tomato_Niben_Chlorophylls)
Tomato_Niben_Chlorophylls$SOL <- substr(Tomato_Niben_Chlorophylls$Tomato, 1, 14)
Tomato_Niben_Chlorophylls <- merge(Tomato_Niben_Chlorophylls, Isoprenoids_Tomato, by="SOL")
Tomato_Niben_Chlorophylls <- Tomato_Niben_Chlorophylls[order(match(Tomato_Niben_Chlorophylls[,1],Tomato_Chlorophylls_Name[,1])),]

## We create a new df only with benthamiana accesions

Niben_Chlorophylls <- Tomato_Niben_Chlorophylls %>% 
  select(Niben, Name)
colnames(Niben_Chlorophylls) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_Chlorophylls <- merge(Niben_Chlorophylls, TPM_All_FC_Annotated, by="Name")

## We select only the FC data from the different comparisons
Niben_Chlorophylls <- Niben_Chlorophylls %>% select(Name, 88:105)
Niben_Chlorophylls <- Niben_Chlorophylls[order(match(Niben_Chlorophylls[,1], Tomato_Niben_Chlorophylls[,3])),]

## We prepare the data for a heatmap and we map it
Niben_Chlorophylls_Annotation <- merge(Niben_Chlorophylls, Tomato_Niben_Chlorophylls, by.x="Name", by.y = "Niben")
Niben_Chlorophylls_Annotation <- Niben_Chlorophylls_Annotation[order(match(Niben_Chlorophylls_Annotation[,1], Tomato_Niben_Chlorophylls[,3])),]
rownames(Niben_Chlorophylls_Annotation) <- Niben_Chlorophylls_Annotation$Name
Niben_Chlorophylls_Annotation <- Niben_Chlorophylls_Annotation %>% select(Name.y)
colnames(Niben_Chlorophylls_Annotation) <- "Enzymes"

rownames(Niben_Chlorophylls) <- Niben_Chlorophylls$Name
Niben_Chlorophylls[1] <- NULL

Niben_Chlorophylls <- Niben_Chlorophylls %>%
  select(pcrtB6h_vs_GFP6h_filtered, pcrtB24h_vs_GFP24h_filtered, pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

min(Niben_Chlorophylls, na.rm = TRUE)
max(Niben_Chlorophylls, na.rm = TRUE)
breaks_Chlorophylls=c(seq(-2,0, length.out = 100), seq(0.01,2,length.out = 100))

color_Chlorophylls <- colorRampPalette(c("#415a20","#9bac12","#e5c033","#c38144","#894539"))(n=length(unique(Niben_Chlorophylls_Annotation$Enzymes)))
names(color_Chlorophylls) <- unique(Niben_Chlorophylls_Annotation$Enzymes)
color_Chlorophylls <- list(Enzymes=color_Chlorophylls)

pheatmap(Niben_Chlorophylls, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         #breaks = breaks_Chlorophylls,
         breaks = breaks_Stndr,
         legend_breaks = c(-4, -2, 0, 2, 4),
         legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         annotation_row=Niben_Chlorophylls_Annotation, annotation_colors=color_Chlorophylls, 
         annotation_names_row = TRUE, show_rownames = FALSE,  
         cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("6h", "24h", "22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h", "96h"),
         cex = 1,
         angle_col=45,
         main = "Chlorophylls Biosynthesis",
         annotation_legend = TRUE,
         #display_numbers = Nb_Sly_Ath_MVA_Sig,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)


###########################################################################################
############################ CHLOROPHYLLS DEGRADATION ##################################################

## We subset this genes, found their AA sequence and save the file
Tomato_ChloroDeg_Name <- Isoprenoids_Tomato[Isoprenoids_Tomato$Pathway=="Chlorophyll_degradation",] %>%
  select(SOL)

Tomato_ChloroDeg_Fasta <- Tomato_4.0_fasta[c(which(substr(names(Tomato_4.0_fasta), 1, 14) %in% Tomato_ChloroDeg_Name$SOL))]
write.fasta(Tomato_ChloroDeg_Fasta, as.string = TRUE, names = names(Tomato_ChloroDeg_Fasta),
            file.out = "Milan/RNAseq_Milan/Funcional Groups/Tomato_ChloroDeg.fasta")
write.table(names(Tomato_ChloroDeg_Fasta), file="Milan/RNAseq_Milan/Funcional Groups/Tomato_ChloroDeg.txt",
            quote = FALSE, sep = "\t", row.names = FALSE, col.names = FALSE)

## We load the file with the orthologs in benthamiana from OrthoVenn proccessed with Python
Tomato_Niben_ChloroDeg <- read.csv("Milan/RNAseq_Milan/Funcional Groups/Niben_ChloroDeg.csv", na.strings = "")
Tomato_Niben_ChloroDeg[1] <- NULL
Tomato_Niben_ChloroDeg <- Tomato_Niben_ChloroDeg %>%
  gather(key="Tomato", value="Niben")
Tomato_Niben_ChloroDeg <- na.omit(Tomato_Niben_ChloroDeg)
Tomato_Niben_ChloroDeg$SOL <- substr(Tomato_Niben_ChloroDeg$Tomato, 1, 14)
Tomato_Niben_ChloroDeg <- merge(Tomato_Niben_ChloroDeg, Isoprenoids_Tomato, by="SOL")
Tomato_Niben_ChloroDeg <- Tomato_Niben_ChloroDeg[order(match(Tomato_Niben_ChloroDeg[,1],Tomato_ChloroDeg_Name[,1])),]

## We create a new df only with benthamiana accesions

Niben_ChloroDeg <- Tomato_Niben_ChloroDeg %>% 
  select(Niben, Name)
colnames(Niben_ChloroDeg) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_ChloroDeg <- merge(Niben_ChloroDeg, TPM_All_FC_Annotated, by="Name")

## We select only the FC data from the different comparisons
Niben_ChloroDeg <- Niben_ChloroDeg %>% select(Name, 88:105)
Niben_ChloroDeg <- Niben_ChloroDeg[order(match(Niben_ChloroDeg[,1], Tomato_Niben_ChloroDeg[,3])),]

## We prepare the data for a heatmap and we map it
Niben_ChloroDeg_Annotation <- merge(Niben_ChloroDeg, Tomato_Niben_ChloroDeg, by.x="Name", by.y = "Niben")
Niben_ChloroDeg_Annotation <- Niben_ChloroDeg_Annotation[order(match(Niben_ChloroDeg_Annotation[,1], Tomato_Niben_ChloroDeg[,3])),]
rownames(Niben_ChloroDeg_Annotation) <- Niben_ChloroDeg_Annotation$Name
Niben_ChloroDeg_Annotation <- Niben_ChloroDeg_Annotation %>% select(Name.y)
colnames(Niben_ChloroDeg_Annotation) <- "Enzymes"

rownames(Niben_ChloroDeg) <- Niben_ChloroDeg$Name
Niben_ChloroDeg[1] <- NULL

Niben_ChloroDeg <- Niben_ChloroDeg %>%
  select(pcrtB6h_vs_GFP6h_filtered, pcrtB24h_vs_GFP24h_filtered, pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

min(Niben_ChloroDeg, na.rm = TRUE)
max(Niben_ChloroDeg, na.rm = TRUE)
breaks_ChloroDeg=c(seq(-2,0, length.out = 100), seq(0.01,7,length.out = 100))

color_ChloroDeg <- colorRampPalette(c("#415a20","#9bac12","#e5c033","#c38144","#894539"))(n=length(unique(Niben_ChloroDeg_Annotation$Enzymes)))
names(color_ChloroDeg) <- unique(Niben_ChloroDeg_Annotation$Enzymes)
color_ChloroDeg <- list(Enzymes=color_ChloroDeg)

pheatmap(Niben_ChloroDeg, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         #breaks = breaks_ChloroDeg,
         breaks = breaks_Stndr,
         legend_breaks = c(-4, -2, 0, 2, 4),
         legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         annotation_row=Niben_ChloroDeg_Annotation, annotation_colors=color_ChloroDeg, 
         annotation_names_row = TRUE, show_rownames = FALSE,  
         cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("6h", "24h", "22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h", "96h"),
         cex = 1,
         angle_col=45,
         main = "Chlorophylls Degradation",
         annotation_legend = TRUE,
         #display_numbers = Nb_Sly_Ath_MVA_Sig,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)


###########################################################################################
############################ PHYLLOQUINONE & PLASTOQUINONE SYNTHESIS ##################################################

## We subset this genes, found their AA sequence and save the file
Tomato_PhylloPlasto_Name <- Isoprenoids_Tomato[Isoprenoids_Tomato$Pathway=="Phylloquinone_biosynthesis"|Isoprenoids_Tomato$Pathway=="Plastoquinone_biosynthesis",] %>%
  select(SOL)

Tomato_PhylloPlasto_Fasta <- Tomato_4.0_fasta[c(which(substr(names(Tomato_4.0_fasta), 1, 14) %in% Tomato_PhylloPlasto_Name$SOL))]
write.fasta(Tomato_PhylloPlasto_Fasta, as.string = TRUE, names = names(Tomato_PhylloPlasto_Fasta),
            file.out = "Milan/RNAseq_Milan/Funcional Groups/Tomato_PhylloPlasto.fasta")
write.table(names(Tomato_PhylloPlasto_Fasta), file="Milan/RNAseq_Milan/Funcional Groups/Tomato_PhylloPlasto.txt",
            quote = FALSE, sep = "\t", row.names = FALSE, col.names = FALSE)

## We load the file with the orthologs in benthamiana from OrthoVenn proccessed with Python
Tomato_Niben_PhylloPlasto <- read.csv("Milan/RNAseq_Milan/Funcional Groups/Niben_PhylloPlasto.csv", na.strings = "")
Tomato_Niben_PhylloPlasto[1] <- NULL
Tomato_Niben_PhylloPlasto <- Tomato_Niben_PhylloPlasto %>%
  gather(key="Tomato", value="Niben")
Tomato_Niben_PhylloPlasto <- na.omit(Tomato_Niben_PhylloPlasto)
Tomato_Niben_PhylloPlasto$SOL <- substr(Tomato_Niben_PhylloPlasto$Tomato, 1, 14)
Tomato_Niben_PhylloPlasto <- merge(Tomato_Niben_PhylloPlasto, Isoprenoids_Tomato, by="SOL")
Tomato_Niben_PhylloPlasto <- Tomato_Niben_PhylloPlasto[order(match(Tomato_Niben_PhylloPlasto[,1],Tomato_PhylloPlasto_Name[,1])),]

## We create a new df only with benthamiana accesions

Niben_PhylloPlasto <- Tomato_Niben_PhylloPlasto %>% 
  select(Niben, Name)
colnames(Niben_PhylloPlasto) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_PhylloPlasto <- merge(Niben_PhylloPlasto, TPM_All_FC_Annotated, by="Name")

## We select only the FC data from the different comparisons
Niben_PhylloPlasto <- Niben_PhylloPlasto %>% select(Name, 88:105)
Niben_PhylloPlasto <- Niben_PhylloPlasto[order(match(Niben_PhylloPlasto[,1], Tomato_Niben_PhylloPlasto[,3])),]

## We prepare the data for a heatmap and we map it
Niben_PhylloPlasto_Annotation <- merge(Niben_PhylloPlasto, Tomato_Niben_PhylloPlasto, by.x="Name", by.y = "Niben")
Niben_PhylloPlasto_Annotation <- Niben_PhylloPlasto_Annotation[order(match(Niben_PhylloPlasto_Annotation[,1], Tomato_Niben_PhylloPlasto[,3])),]
rownames(Niben_PhylloPlasto_Annotation) <- Niben_PhylloPlasto_Annotation$Name
Niben_PhylloPlasto_Annotation <- Niben_PhylloPlasto_Annotation %>% select(Name.y)
colnames(Niben_PhylloPlasto_Annotation) <- "Enzymes"

rownames(Niben_PhylloPlasto) <- Niben_PhylloPlasto$Name
Niben_PhylloPlasto[1] <- NULL

Niben_PhylloPlasto <- Niben_PhylloPlasto %>%
  select(pcrtB6h_vs_GFP6h_filtered, pcrtB24h_vs_GFP24h_filtered, pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

min(Niben_PhylloPlasto, na.rm = TRUE)
max(Niben_PhylloPlasto, na.rm = TRUE)
breaks_PhylloPlasto=c(seq(-2,0, length.out = 100), seq(0.01,2,length.out = 100))

color_PhylloPlasto <- colorRampPalette(c("#415a20","#9bac12","#e5c033","#c38144","#894539"))(n=length(unique(Niben_PhylloPlasto_Annotation$Enzymes)))
names(color_PhylloPlasto) <- unique(Niben_PhylloPlasto_Annotation$Enzymes)
color_PhylloPlasto <- list(Enzymes=color_PhylloPlasto)

pheatmap(Niben_PhylloPlasto, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         #breaks = breaks_PhylloPlasto,
         breaks = breaks_Stndr,
         legend_breaks = c(-4, -2, 0, 2, 4),
         legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         annotation_row=Niben_PhylloPlasto_Annotation, annotation_colors=color_PhylloPlasto, 
         annotation_names_row = TRUE, show_rownames = FALSE,  
         cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("6h", "24h", "22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h", "96h"),
         cex = 1,
         angle_col=45,
         main = "Phylloquinone & Plastoquinone Biosynthesis",
         annotation_legend = TRUE,
         #display_numbers = Nb_Sly_Ath_MVA_Sig,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)


###########################################################################################
############################ GIBBERELLIN SYNTHESIS ##################################################

## We subset this genes, found their AA sequence and save the file
Tomato_Gibb_Name <- Isoprenoids_Tomato[Isoprenoids_Tomato$Pathway=="Gibberellin_biosynthesis",] %>%
  select(SOL)

Tomato_Gibb_Fasta <- Tomato_4.0_fasta[c(which(substr(names(Tomato_4.0_fasta), 1, 14) %in% Tomato_Gibb_Name$SOL))]
write.fasta(Tomato_Gibb_Fasta, as.string = TRUE, names = names(Tomato_Gibb_Fasta),
            file.out = "Milan/RNAseq_Milan/Funcional Groups/Tomato_Gibb.fasta")
write.table(names(Tomato_Gibb_Fasta), file="Milan/RNAseq_Milan/Funcional Groups/Tomato_Gibb.txt",
            quote = FALSE, sep = "\t", row.names = FALSE, col.names = FALSE)

## We load the file with the orthologs in benthamiana from OrthoVenn proccessed with Python
Tomato_Niben_Gibb <- read.csv("Milan/RNAseq_Milan/Funcional Groups/Niben_Gibb.csv", na.strings = "")
Tomato_Niben_Gibb[1] <- NULL
Tomato_Niben_Gibb <- Tomato_Niben_Gibb %>%
  gather(key="Tomato", value="Niben")
Tomato_Niben_Gibb <- na.omit(Tomato_Niben_Gibb)
Tomato_Niben_Gibb$SOL <- substr(Tomato_Niben_Gibb$Tomato, 1, 14)
Tomato_Niben_Gibb <- merge(Tomato_Niben_Gibb, Isoprenoids_Tomato, by="SOL")
Tomato_Niben_Gibb <- Tomato_Niben_Gibb[-c(5,10,16,17),]
Tomato_Niben_Gibb <- Tomato_Niben_Gibb[order(match(Tomato_Niben_Gibb[,1],Tomato_Gibb_Name[,1])),]

## We create a new df only with benthamiana accesions

Niben_Gibb <- Tomato_Niben_Gibb %>% 
  select(Niben, Name)
colnames(Niben_Gibb) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_Gibb <- merge(Niben_Gibb, TPM_All_FC_Annotated, by="Name")

## We select only the FC data from the different comparisons
Niben_Gibb <- Niben_Gibb %>% select(Name, 88:105)
Niben_Gibb <- Niben_Gibb[-c(9),]
Niben_Gibb <- Niben_Gibb[order(match(Niben_Gibb[,1], Tomato_Niben_Gibb[,3])),]

## We prepare the data for a heatmap and we map it
Niben_Gibb_Annotation <- merge(Niben_Gibb, Tomato_Niben_Gibb, by.x="Name", by.y = "Niben")
Niben_Gibb_Annotation <- Niben_Gibb_Annotation[-c(9),]
Niben_Gibb_Annotation <- Niben_Gibb_Annotation[order(match(Niben_Gibb_Annotation[,1], Tomato_Niben_Gibb[,3])),]
rownames(Niben_Gibb_Annotation) <- Niben_Gibb_Annotation$Name
Niben_Gibb_Annotation <- Niben_Gibb_Annotation %>% select(Name.y)
colnames(Niben_Gibb_Annotation) <- "Enzymes"

rownames(Niben_Gibb) <- Niben_Gibb$Name
Niben_Gibb[1] <- NULL

Niben_Gibb <- Niben_Gibb %>%
  select(pcrtB6h_vs_GFP6h_filtered, pcrtB24h_vs_GFP24h_filtered, pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

min(Niben_Gibb, na.rm = TRUE)
max(Niben_Gibb, na.rm = TRUE)
breaks_Gibb=c(seq(-7,0, length.out = 100), seq(0.01,3,length.out = 100))

color_Gibb <- colorRampPalette(c("#415a20","#9bac12","#e5c033","#c38144","#894539"))(n=length(unique(Niben_Gibb_Annotation$Enzymes)))
names(color_Gibb) <- unique(Niben_Gibb_Annotation$Enzymes)
color_Gibb <- list(Enzymes=color_Gibb)

pheatmap(Niben_Gibb, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         #breaks = breaks_Gibb,
         breaks = breaks_Stndr,
         legend_breaks = c(-4, -2, 0, 2, 4),
         legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         annotation_row=Niben_Gibb_Annotation, annotation_colors=color_Gibb, 
         annotation_names_row = TRUE, show_rownames = FALSE,  
         cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("6h", "24h", "22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h", "96h"),
         cex = 1,
         angle_col=45,
         main = "Gibberellin Biosynthesis",
         annotation_legend = TRUE,
         #display_numbers = Nb_Sly_Ath_MVA_Sig,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)


###########################################################################################
############################ MVA SYNTHESIS ##################################################

## We subset this genes, found their AA sequence and save the file
Tomato_MVA_Name <- Isoprenoids_Tomato[Isoprenoids_Tomato$Pathway=="MVA_pathway"|Isoprenoids_Tomato$Pathway=="FPPS",] %>%
  select(SOL)

Tomato_MVA_Fasta <- Tomato_4.0_fasta[c(which(substr(names(Tomato_4.0_fasta), 1, 14) %in% Tomato_MVA_Name$SOL))]
write.fasta(Tomato_MVA_Fasta, as.string = TRUE, names = names(Tomato_MVA_Fasta),
            file.out = "Milan/RNAseq_Milan/Funcional Groups/Tomato_MVA.fasta")
write.table(names(Tomato_MVA_Fasta), file="Milan/RNAseq_Milan/Funcional Groups/Tomato_MVA.txt",
            quote = FALSE, sep = "\t", row.names = FALSE, col.names = FALSE)

## We load the file with the orthologs in benthamiana from OrthoVenn proccessed with Python
Tomato_Niben_MVA <- read.csv("Milan/RNAseq_Milan/Funcional Groups/Niben_MVA.csv", na.strings = "")
Tomato_Niben_MVA[1] <- NULL
Tomato_Niben_MVA <- Tomato_Niben_MVA %>%
  gather(key="Tomato", value="Niben")
Tomato_Niben_MVA <- na.omit(Tomato_Niben_MVA)
Tomato_Niben_MVA$SOL <- substr(Tomato_Niben_MVA$Tomato, 1, 14)
Tomato_Niben_MVA <- merge(Tomato_Niben_MVA, Isoprenoids_Tomato, by="SOL")
Tomato_Niben_MVA <- Tomato_Niben_MVA[-c(7),]
Tomato_Niben_MVA <- Tomato_Niben_MVA[order(match(Tomato_Niben_MVA[,1],Tomato_MVA_Name[,1])),]

## We create a new df only with benthamiana accesions

Niben_MVA <- Tomato_Niben_MVA %>% 
  select(Niben, Name)
colnames(Niben_MVA) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_MVA <- merge(Niben_MVA, TPM_All_FC_Annotated, by="Name")

## We select only the FC data from the different comparisons
Niben_MVA <- Niben_MVA %>% select(Name, 88:105)
Niben_MVA <- Niben_MVA[-c(14),]
Niben_MVA <- Niben_MVA[order(match(Niben_MVA[,1], Tomato_Niben_MVA[,3])),]

## We prepare the data for a heatmap and we map it
Niben_MVA_Annotation <- merge(Niben_MVA, Tomato_Niben_MVA, by.x="Name", by.y = "Niben")
Niben_MVA_Annotation <- Niben_MVA_Annotation[-c(14),]
Niben_MVA_Annotation <- Niben_MVA_Annotation[order(match(Niben_MVA_Annotation[,1], Tomato_Niben_MVA[,3])),]
rownames(Niben_MVA_Annotation) <- Niben_MVA_Annotation$Name
Niben_MVA_Annotation <- Niben_MVA_Annotation %>% select(Name.y)
colnames(Niben_MVA_Annotation) <- "Enzymes"

rownames(Niben_MVA) <- Niben_MVA$Name
Niben_MVA[1] <- NULL

Niben_MVA <- Niben_MVA %>%
  select(pcrtB6h_vs_GFP6h_filtered, pcrtB24h_vs_GFP24h_filtered, pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

min(Niben_MVA, na.rm = TRUE)
max(Niben_MVA, na.rm = TRUE)
breaks_MVA=c(seq(-3,0, length.out = 100), seq(0.01,2,length.out = 100))

color_MVA <- colorRampPalette(c("#415a20","#9bac12","#e5c033","#c38144","#894539"))(n=length(unique(Niben_MVA_Annotation$Enzymes)))
names(color_MVA) <- unique(Niben_MVA_Annotation$Enzymes)
color_MVA <- list(Enzymes=color_MVA)

pheatmap(Niben_MVA, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         #breaks = breaks_MVA,
         breaks = breaks_Stndr,
         legend_breaks = c(-4, -2, 0, 2, 4),
         legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         annotation_row=Niben_MVA_Annotation, annotation_colors=color_MVA, 
         annotation_names_row = TRUE, show_rownames = FALSE,  
         cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("6h", "24h", "22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h", "96h"),
         cex = 1,
         angle_col=45,
         main = "MVA",
         annotation_legend = TRUE,
         #display_numbers = Nb_Sly_Ath_MVA_Sig,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)



#########################################################################################################################################
####################################  iTAK  #############################################################################################
#########################################################################################################################################

Niben261_TFs_TRs <- read.table("Milan/RNAseq_Milan/Funcional Groups/iTAK/tf_classification.txt",
                         sep="\t")
colnames(Niben261_TFs_TRs) <- c("Name", "Family","Class", "Derivation")

Niben261_TFs <- Niben261_TFs_TRs[Niben261_TFs_TRs$Class=="TF",]
Niben261_TRs <- Niben261_TFs_TRs[Niben261_TFs_TRs$Class=="TR",]

Niben261_Pk <- read.table("Milan/RNAseq_Milan/Funcional Groups/iTAK/shiu_classification.txt",
                          sep = "\t")
colnames(Niben261_Pk) <- c("Name", "Family")

##########################################################################################################################################
################################## TRANSCRIPTION FACTORS #################################################################################

TFs_Families <- sort(unique(Niben261_TFs$Family))
Niben261_TFs_TC <- merge(Niben261_TFs, TPM_All_FC_Annotated, by="Name")
Niben261_TFs_TC <- Niben261_TFs_TC %>% dplyr::select(Name, 73, 75:88)#select(Name, 90:107)
rownames(Niben261_TFs_TC) <- Niben261_TFs_TC$Name
Niben261_TFs_TC[1] <- NULL
Niben261_TFs_TC$Count <- rowSums(!is.na(Niben261_TFs_TC))
Niben261_TFs_TC_Complete <- Niben261_TFs_TC[Niben261_TFs_TC$Count!=0,]
Niben261_TFs_TC_Complete <- Niben261_TFs_TC_Complete %>% select(-Count)

Niben261_TFs_TC_Complete_Annotation <- merge(Niben261_TFs_TC_Complete, Niben261_TFs, by.x = 0, by.y="Name")
Niben261_TFs_TC_Complete_Annotation <- Niben261_TFs_TC_Complete_Annotation %>%
  select(Row.names, Family)
Niben261_TFs_TC_Complete_Annotation <- Niben261_TFs_TC_Complete_Annotation[order(Niben261_TFs_TC_Complete_Annotation$Family),]
rownames(Niben261_TFs_TC_Complete_Annotation) <- Niben261_TFs_TC_Complete_Annotation$Row.names

Niben261_TFs_TC_Complete$Name <- rownames(Niben261_TFs_TC_Complete)
Niben261_TFs_TC_Complete <- Niben261_TFs_TC_Complete[order(match(Niben261_TFs_TC_Complete$Name, Niben261_TFs_TC_Complete_Annotation$Row.names)),]
Niben261_TFs_TC_Complete$Name <- NULL
Niben261_TFs_TC_Complete_Annotation$Row.names <- NULL

Niben261_TFs_TC_Complete <- Niben261_TFs_TC_Complete %>%
  select(pcrtB6h_vs_GFP6h_filtered, pcrtB24h_vs_GFP24h_filtered, pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

Niben261_TFs_TC_Complete <- Niben261_TFs_TC_Complete %>%
  select(pcrtB22h_vs_GFP22h_All, pcrtBAA_vs_GFPAA_All, pcrtB25h_vs_GFP25h_All, 
         pcrtBYY_vs_GFPYY_All, pcrtBBB_vs_GFPBB_All, pcrtB28h_vs_GFP28h_All, pcrtB34h_vs_GFP34h_All, pcrtBZZ_vs_GFPZZ_All, 
         pcrtB37h_vs_GFP37h_All, pcrtBCC_vs_GFPCC_All, pcrtB40h_vs_GFP40h_All, pcrtBDD_vs_GFPDD_All, pcrtB46h_vs_GFP46h_All, 
         pcrtBEE_vs_GFPEE_All, pcrtB56h_vs_GFP56h_All)


min(Niben261_TFs_TC_Complete, na.rm = TRUE)
max(Niben261_TFs_TC_Complete, na.rm = TRUE)
breaks_TFs_All=c(seq(-6,0, length.out = 100), seq(0.01,6,length.out = 100))

color_TFs_All <- colorRampPalette(c("#415a20","#9bac12","#e5c033","#c38144","#894539"))(n=length(unique(Niben261_TFs_TC_Complete_Annotation$Family)))
names(color_TFs_All) <- unique(Niben261_TFs_TC_Complete_Annotation$Family)
color_TFs_All <- list(Family=color_TFs_All)

Niben261_TFs_TC_Complete[is.na(Niben261_TFs_TC_Complete)] <- 0

pheatmap(Niben261_TFs_TC_Complete, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =TRUE,
         #breaks = breaks_TFs_1,
         #breaks = breaks_Stndr,
         legend_breaks = c(-4, -2, 0, 2, 4),
         legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "row",
         annotation_row=Niben261_TFs_TC_Complete_Annotation, annotation_colors=color_TFs_All, 
         annotation_names_row = TRUE, show_rownames = FALSE,  
         #cellwidth = 10, cellheight = 10,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c(#"6h", "24h", 
           "22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h"#,"96h"
           ),
         cex = 1,
         angle_col=45,
         main = "TFs All Families",
         annotation_legend = TRUE,
         #display_numbers = Nb_Sly_Ath_MVA_Sig,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)


###########################################################################################################################################
### "Alfin-like"   "AP2/ERF-AP2"  "AP2/ERF-ERF"  "AP2/ERF-RAV"

Niben261_TFs_1 <- Niben261_TFs[Niben261_TFs$Family=="Alfin-like"|Niben261_TFs$Family=="AP2/ERF-AP2"|Niben261_TFs$Family=="AP2/ERF-ERF"|Niben261_TFs$Family=="AP2/ERF-RAV",] %>%
  select(Name, Family)
Niben261_TFs_1 <- Niben261_TFs_1[order(Niben261_TFs_1$Family),]

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben261_TFs_1_FC <- merge(Niben261_TFs_1, TPM_All_FC_Annotated, by="Name")

## We select only the FC data from the different comparisons
Niben261_TFs_1_FC <- Niben261_TFs_1_FC %>% select(Name, 88:105)
Niben261_TFs_1_FC <- Niben261_TFs_1_FC[order(match(Niben261_TFs_1_FC$Name, Niben261_TFs_1$Name)),]

## We prepare the data for a heatmap and we map it
Niben261_TFs_1_Annotation <- Niben261_TFs_1
rownames(Niben261_TFs_1_Annotation) <- Niben261_TFs_1_Annotation$Name
Niben261_TFs_1_Annotation[1] <- NULL

rownames(Niben261_TFs_1_FC) <- Niben261_TFs_1_FC$Name
Niben261_TFs_1_FC[1] <- NULL

min(Niben261_TFs_1_FC, na.rm = TRUE)
max(Niben261_TFs_1_FC, na.rm = TRUE)
breaks_TFs_1=c(seq(-4,0, length.out = 100), seq(0.01,6,length.out = 100))

color_TFs_1 <- colorRampPalette(c("#415a20","#9bac12","#e5c033","#c38144","#894539"))(n=length(unique(Niben261_TFs_1_Annotation$Family)))
names(color_TFs_1) <- unique(Niben261_TFs_1_Annotation$Family)
color_TFs_1 <- list(Family=color_TFs_1)

pheatmap(Niben261_TFs_1_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         #breaks = breaks_TFs_1,
         breaks = breaks_Stndr,
         legend_breaks = c(-4, -2, 0, 2, 4),
         legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         annotation_row=Niben261_TFs_1_Annotation, annotation_colors=color_TFs_1, 
         annotation_names_row = TRUE, show_rownames = TRUE,  
         cellwidth = 10, cellheight = 10,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("6h", "24h", "22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h", "96h"),
         cex = 1,
         angle_col=45,
         main = "TFs Alfin-like, AP2/ERF-AP2, AP2/ERF-ERF, AP2/ERF-RAV",
         annotation_legend = TRUE,
         #display_numbers = Nb_Sly_Ath_MVA_Sig,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)


###########################################################################################################################################
## "B3" "B3-ARF" "BBR-BPC" "BES1" 

Niben261_TFs_2 <- Niben261_TFs[Niben261_TFs$Family=="B3"|Niben261_TFs$Family=="B3-ARF"|Niben261_TFs$Family=="BBR-BPC"|Niben261_TFs$Family=="BES1",] %>%
  select(Name, Family)
Niben261_TFs_2 <- Niben261_TFs_2[order(Niben261_TFs_2$Family),]

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben261_TFs_2_FC <- merge(Niben261_TFs_2, TPM_All_FC_Annotated, by="Name")

## We select only the FC data from the different comparisons
Niben261_TFs_2_FC <- Niben261_TFs_2_FC %>% select(Name, 88:105)
Niben261_TFs_2_FC <- Niben261_TFs_2_FC[order(match(Niben261_TFs_2_FC$Name, Niben261_TFs_2$Name)),]

## We prepare the data for a heatmap and we map it
Niben261_TFs_2_Annotation <- Niben261_TFs_2
rownames(Niben261_TFs_2_Annotation) <- Niben261_TFs_2_Annotation$Name
Niben261_TFs_2_Annotation[1] <- NULL

rownames(Niben261_TFs_2_FC) <- Niben261_TFs_2_FC$Name
Niben261_TFs_2_FC[1] <- NULL

min(Niben261_TFs_2_FC, na.rm = TRUE)
max(Niben261_TFs_2_FC, na.rm = TRUE)
breaks_TFs_2=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

color_TFs_2 <- colorRampPalette(c("#415a20","#9bac12","#e5c033","#c38144","#894539"))(n=length(unique(Niben261_TFs_2_Annotation$Family)))
names(color_TFs_2) <- unique(Niben261_TFs_2_Annotation$Family)
color_TFs_2 <- list(Family=color_TFs_2)

pheatmap(Niben261_TFs_2_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         #breaks = breaks_TFs_2,
         breaks = breaks_Stndr,
         legend_breaks = c(-4, -2, 0, 2, 4),
         legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         annotation_row=Niben261_TFs_2_Annotation, annotation_colors=color_TFs_2, 
         annotation_names_row = TRUE, show_rownames = TRUE,  
         cellwidth = 10, cellheight = 10,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("6h", "24h", "22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h", "96h"),
         cex = 1,
         angle_col=45,
         main = "TFs B3, B3-ARF, BBR_BPC, BES1",
         annotation_legend = TRUE,
         #display_numbers = Nb_Sly_Ath_MVA_Sig,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)


###########################################################################################################################################
## "bHLH" "BSD" "bZIP" 

Niben261_TFs_3 <- Niben261_TFs[Niben261_TFs$Family=="bHLH"|Niben261_TFs$Family=="BSD"|Niben261_TFs$Family=="bZIP",] %>%
  select(Name, Family)
Niben261_TFs_3 <- Niben261_TFs_3[order(Niben261_TFs_3$Family),]

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben261_TFs_3_FC <- merge(Niben261_TFs_3, TPM_All_FC_Annotated, by="Name")

## We select only the FC data from the different comparisons
Niben261_TFs_3_FC <- Niben261_TFs_3_FC %>% select(Name, 88:105)
Niben261_TFs_3_FC <- Niben261_TFs_3_FC[order(match(Niben261_TFs_3_FC$Name, Niben261_TFs_3$Name)),]

## We prepare the data for a heatmap and we map it
Niben261_TFs_3_Annotation <- Niben261_TFs_3
rownames(Niben261_TFs_3_Annotation) <- Niben261_TFs_3_Annotation$Name
Niben261_TFs_3_Annotation[1] <- NULL

rownames(Niben261_TFs_3_FC) <- Niben261_TFs_3_FC$Name
Niben261_TFs_3_FC[1] <- NULL

min(Niben261_TFs_3_FC, na.rm = TRUE)
max(Niben261_TFs_3_FC, na.rm = TRUE)
breaks_TFs_3=c(seq(-5,0, length.out = 100), seq(0.01,6,length.out = 100))

color_TFs_3 <- colorRampPalette(c("#415a20","#9bac12","#e5c033","#c38144","#894539"))(n=length(unique(Niben261_TFs_3_Annotation$Family)))
names(color_TFs_3) <- unique(Niben261_TFs_3_Annotation$Family)
color_TFs_3 <- list(Family=color_TFs_3)

pheatmap(Niben261_TFs_3_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         #breaks = breaks_TFs_3,
         breaks = breaks_Stndr,
         legend_breaks = c(-4, -2, 0, 2, 4),
         legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         annotation_row=Niben261_TFs_3_Annotation, annotation_colors=color_TFs_3, 
         annotation_names_row = TRUE, show_rownames = TRUE,  
         cellwidth = 10, cellheight = 10,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("6h", "22h", "24h", "25h", "22-25h", "25-28h", "25-28h", "28h", "34h", "37h", "34-37h","34-37-40h","40h","40-46h","46h", "56h", "46-56h","96h"),
         cex = 1,
         angle_col=45,
         main = "TFs bHLH BSD bZIP",
         annotation_legend = TRUE,
         #display_numbers = Nb_Sly_Ath_MVA_Sig,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)



##########################################################################################################################################
################################## TRANSCRIPTION REGULATORS #################################################################################

TRs_Families <- sort(unique(Niben261_TRs$Family))
Niben261_TRs_TC <- merge(Niben261_TRs, TPM_All_FC_Annotated, by="Name")
Niben261_TRs_TC <- Niben261_TRs_TC %>% select(Name, 90:107)
rownames(Niben261_TRs_TC) <- Niben261_TRs_TC$Name
Niben261_TRs_TC[1] <- NULL
Niben261_TRs_TC$Count <- rowSums(!is.na(Niben261_TRs_TC))
Niben261_TRs_TC_Complete <- Niben261_TRs_TC[Niben261_TRs_TC$Count!=0,]
Niben261_TRs_TC_Complete <- Niben261_TRs_TC_Complete %>% select(-Count)

Niben261_TRs_TC_Complete_Annotation <- merge(Niben261_TRs_TC_Complete, Niben261_TRs, by.x = 0, by.y="Name")
Niben261_TRs_TC_Complete_Annotation <- Niben261_TRs_TC_Complete_Annotation %>%
  select(Row.names, Family)
Niben261_TRs_TC_Complete_Annotation <- Niben261_TRs_TC_Complete_Annotation[order(Niben261_TRs_TC_Complete_Annotation$Family),]
rownames(Niben261_TRs_TC_Complete_Annotation) <- Niben261_TRs_TC_Complete_Annotation$Row.names

Niben261_TRs_TC_Complete$Name <- rownames(Niben261_TRs_TC_Complete)
Niben261_TRs_TC_Complete <- Niben261_TRs_TC_Complete[order(match(Niben261_TRs_TC_Complete$Name, Niben261_TRs_TC_Complete_Annotation$Row.names)),]
Niben261_TRs_TC_Complete$Name <- NULL
Niben261_TRs_TC_Complete_Annotation$Row.names <- NULL

Niben261_TRs_TC_Complete <- Niben261_TRs_TC_Complete %>%
  select(pcrtB6h_vs_GFP6h_filtered, pcrtB24h_vs_GFP24h_filtered, pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

min(Niben261_TRs_TC_Complete, na.rm = TRUE)
max(Niben261_TRs_TC_Complete, na.rm = TRUE)
breaks_TRs_All=c(seq(-4,0, length.out = 100), seq(0.01,6,length.out = 100))

color_TRs_All <- colorRampPalette(c("#415a20","#9bac12","#e5c033","#c38144","#894539"))(n=length(unique(Niben261_TRs_TC_Complete_Annotation$Family)))
names(color_TRs_All) <- unique(Niben261_TRs_TC_Complete_Annotation$Family)
color_TRs_All <- list(Family=color_TRs_All)

pheatmap(Niben261_TRs_TC_Complete, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         #breaks = breaks_TRs_All,
         breaks = breaks_Stndr,
         legend_breaks = c(-4, -2, 0, 2, 4),
         legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         annotation_row=Niben261_TRs_TC_Complete_Annotation, annotation_colors=color_TRs_All, 
         annotation_names_row = TRUE, show_rownames = FALSE,  
         #cellwidth = 10, cellheight = 10,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("6h", "24h", "22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h", "96h"),
         cex = 1,
         angle_col=45,
         main = "TRs All Families",
         annotation_legend = TRUE,
         #display_numbers = Nb_Sly_Ath_MVA_Sig,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)


##########################################################################################################################################
################################## PROTEIN KINASES #################################################################################

Pk_Families <- sort(unique(Niben261_Pk$Family))
Niben261_Pk_TC <- merge(Niben261_Pk, TPM_All_FC_Annotated, by ="Name")
Niben261_Pk_TC <- Niben261_Pk_TC %>% select(Name, 88:105)
rownames(Niben261_Pk_TC) <- Niben261_Pk_TC$Name
Niben261_Pk_TC[1] <- NULL
Niben261_Pk_TC$Count <- rowSums(!is.na(Niben261_Pk_TC))
Niben261_Pk_TC_Complete <- Niben261_Pk_TC[Niben261_Pk_TC$Count!=0,]
Niben261_Pk_TC_Complete <- Niben261_Pk_TC_Complete %>% select(-Count)

Niben261_Pk_TC_Complete_Annotation <- merge(Niben261_Pk_TC_Complete, Niben261_Pk, by.x = 0, by.y="Name")
Niben261_Pk_TC_Complete_Annotation <- Niben261_Pk_TC_Complete_Annotation %>%
  select(Row.names, Family)
Niben261_Pk_TC_Complete_Annotation <- Niben261_Pk_TC_Complete_Annotation[order(Niben261_Pk_TC_Complete_Annotation$Family),]
rownames(Niben261_Pk_TC_Complete_Annotation) <- Niben261_Pk_TC_Complete_Annotation$Row.names

Niben261_Pk_TC_Complete$Name <- rownames(Niben261_Pk_TC_Complete)
Niben261_Pk_TC_Complete <- Niben261_Pk_TC_Complete[order(match(Niben261_Pk_TC_Complete$Name, Niben261_Pk_TC_Complete_Annotation$Row.names)),]
Niben261_Pk_TC_Complete$Name <- NULL
Niben261_Pk_TC_Complete_Annotation$Row.names <- NULL

Niben261_Pk_TC_Complete <- Niben261_Pk_TC_Complete %>%
  select(pcrtB6h_vs_GFP6h_filtered, pcrtB24h_vs_GFP24h_filtered, pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

min(Niben261_Pk_TC_Complete, na.rm = TRUE)
max(Niben261_Pk_TC_Complete, na.rm = TRUE)
breaks_Pk_All=c(seq(-7,0, length.out = 100), seq(0.01,6,length.out = 100))

color_Pk_All <- colorRampPalette(c("#415a20","#9bac12","#e5c033","#c38144","#894539"))(n=length(unique(Niben261_Pk_TC_Complete_Annotation$Family)))
names(color_Pk_All) <- unique(Niben261_Pk_TC_Complete_Annotation$Family)
color_Pk_All <- list(Family=color_Pk_All)

pheatmap(Niben261_Pk_TC_Complete, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         #breaks = breaks_Pk_All,
         breaks = breaks_Stndr,
         legend_breaks = c(-4, -2, 0, 2, 4),
         legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         annotation_row=Niben261_Pk_TC_Complete_Annotation, annotation_colors=color_Pk_All, 
         annotation_names_row = TRUE, show_rownames = FALSE,  
         #cellwidth = 10, cellheight = 10,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("6h", "24h", "22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h", "96h"),
         cex = 1,
         angle_col=45,
         main = "Pk All Families",
         annotation_legend = TRUE,
         #display_numbers = Nb_Sly_Ath_MVA_Sig,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)



#########################################################################################################################################
####################################  CHAPEROME  #############################################################################################
#########################################################################################################################################
setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Chaperome_Ath <- read.table("../../Chaperome/Chaperome_Ath_list.txt",
                            sep = "\t", header = TRUE)
Chaperome_Ath$Genelocus <- toupper(Chaperome_Ath$Genelocus)

write.table(Chaperome_Ath$Genelocus, file= "Funcional Groups/Ath_Chaperome.txt",
            quote = FALSE, sep = "\t", row.names = FALSE, col.names = FALSE)


setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq")
## We load the file with the orthologs in benthamiana from OrthoVenn proccessed with Python
Ath_Niben_Chaperome <- read.csv("Milan/RNAseq_Milan/Funcional Groups/Niben_Chaperome.csv", na.strings = "")
Ath_Niben_Chaperome[1] <- NULL
Ath_Niben_Chaperome <- Ath_Niben_Chaperome %>%
  gather(key="Ath", value="Niben")
Ath_Niben_Chaperome <- na.omit(Ath_Niben_Chaperome)
Ath_Niben_Chaperome <- merge(Ath_Niben_Chaperome, Chaperome_Ath, by.x = "Ath", by.y = "Genelocus")
#Ath_Niben_Chaperome <- Ath_Niben_Chaperome[-c(5,6,7),]
Ath_Niben_Chaperome <- Ath_Niben_Chaperome[order(match(Ath_Niben_Chaperome[,1],Chaperome_Ath[,1])),]

## We create a new df only with benthamiana accesions

Niben_Chaperome <- Ath_Niben_Chaperome %>% 
  select(Niben, Protein.names)
colnames(Niben_Chaperome) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_Chaperome <- merge(Niben_Chaperome, TPM_All_FC_Annotated, by="Name")

## We select only the FC data from the different comparisons
Niben_Chaperome <- Niben_Chaperome %>% select(Name, 88:105)
Niben_Chaperome <- Niben_Chaperome[!duplicated(Niben_Chaperome$Name),]
Niben_Chaperome <- Niben_Chaperome[order(match(Niben_Chaperome[,1], Ath_Niben_Chaperome[,2])),]

## We prepare the data for a heatmap and we map it
Niben_Chaperome_Annotation <- merge(Niben_Chaperome, Ath_Niben_Chaperome, by.x="Name", by.y = "Niben")
Niben_Chaperome_Annotation <- Niben_Chaperome_Annotation[!duplicated(Niben_Chaperome_Annotation$Name),]
Niben_Chaperome_Annotation <- Niben_Chaperome_Annotation[order(match(Niben_Chaperome_Annotation[,1], Ath_Niben_Chaperome[,2])),]
#write.table(Niben_Chaperome_Annotation, "Milan/RNAseq_Milan/Funcional Groups/Chaperome_Niben261_OrthoVenn_DESeq2FC.txt", quote = FALSE, dec = ",", sep = "\t", row.names = FALSE)
rownames(Niben_Chaperome_Annotation) <- Niben_Chaperome_Annotation$Name
#Niben_Chaperome_Chloroplast <- Niben_Chaperome_Annotation %>% dplyr::filter(Presumed.subcellular.location=="Chloroplast")
#write.table(Niben_Chaperome_Chloroplast, "Milan/RNAseq_Milan/Funcional Groups/Chaperome_Chloroplast_Niben261_OrthoVenn_DESeq2FC.txt", quote = FALSE, dec = ",", sep = "\t", row.names = FALSE)
#Niben_Chaperome_Chloroplast_Annotation <- Niben_Chaperome_Chloroplast %>% select(Protein.names)
#Niben_Chaperome_Chloroplast <- Niben_Chaperome_Chloroplast %>% select(2:19)
Niben_Chaperome_Annotation <- Niben_Chaperome_Annotation %>% select(Protein.names)
colnames(Niben_Chaperome_Annotation) <- "Family"

rownames(Niben_Chaperome) <- Niben_Chaperome$Name
Niben_Chaperome[1] <- NULL

Niben_Chaperome <- Niben_Chaperome %>%
  select(pcrtB6h_vs_GFP6h_filtered, pcrtB24h_vs_GFP24h_filtered, pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

min(Niben_Chaperome, na.rm = TRUE)
max(Niben_Chaperome, na.rm = TRUE)
breaks_Chaperome=c(seq(-9,0, length.out = 100), seq(0.01,9,length.out = 100))

color_Chaperome <- colorRampPalette(c("#415a20","#9bac12","#e5c033","#c38144","#894539"))(n=length(unique(Niben_Chaperome_Annotation$Family)))
names(color_Chaperome) <- unique(Niben_Chaperome_Annotation$Family)
color_Chaperome <- list(Family=color_Chaperome)

pheatmap(Niben_Chaperome, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         #breaks = breaks_Chaperome,
         breaks = breaks_Stndr,
         legend_breaks = c(-4, -2, 0, 2, 4),
         legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         annotation_row=Niben_Chaperome_Annotation, annotation_colors=color_Chaperome, 
         annotation_names_row = TRUE, show_rownames = FALSE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("6h", "24h", "22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h", "96h"),
         cex = 1,
         angle_col=45,
         main = "Chaperome",
         annotation_legend = TRUE,
         #display_numbers = Nb_Sly_Ath_MVA_Sig,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)

##Chaperones in the Chloroplast
color_Chaperome_Chloroplast <- colorRampPalette(c("#415a20","#9bac12","#e5c033","#c38144","#894539"))(n=length(unique(Niben_Chaperome_Chloroplast_Annotation$Protein.names)))
names(color_Chaperome_Chloroplast) <- unique(Niben_Chaperome_Chloroplast_Annotation$Protein.names)
color_Chaperome_Chloroplast <- list(Protein.names=color_Chaperome_Chloroplast)


pheatmap(Niben_Chaperome_Chloroplast, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         #breaks = breaks_Chaperome,
         breaks = breaks_Stndr,
         legend_breaks = c(-4, -2, 0, 2, 4),
         legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         annotation_row=Niben_Chaperome_Chloroplast_Annotation, annotation_colors=color_Chaperome_Chloroplast, 
         annotation_names_row = TRUE, show_rownames = FALSE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("6h", "24h", "22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h", "96h"),
         cex = 1,
         angle_col=45,
         main = "Chaperome Chloroplast",
         annotation_legend = TRUE,
         #display_numbers = Nb_Sly_Ath_MVA_Sig,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)


#########################################################################################################################################
####################################  GENES INVOLVES IN CHROMOPLAST  #############################################################################################
#########################################################################################################################################
setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Chromoplast_Ath <- read.table("../../Chaperome/Genes of interest Arabidopsis.txt",
                            sep = "\t", header = TRUE)
Chromoplast_Ath$GeneID <- substr(Chromoplast_Ath$GeneID, 1, 9)
write.table(Chromoplast_Ath$GeneID, file= "Funcional Groups/Ath_Chromoplast.txt",
            quote = FALSE, sep = "\t", row.names = FALSE, col.names = FALSE)


setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq")
## We load the file with the orthologs in benthamiana from OrthoVenn proccessed with Python
Ath_Niben_Chromoplast <- read.csv("Milan/RNAseq_Milan/Funcional Groups/Niben_Chromoplast.csv", na.strings = "")
Ath_Niben_Chromoplast[1] <- NULL
Ath_Niben_Chromoplast <- Ath_Niben_Chromoplast %>%
  gather(key="Ath", value="Niben")
Ath_Niben_Chromoplast <- na.omit(Ath_Niben_Chromoplast)
Ath_Niben_Chromoplast <- merge(Ath_Niben_Chromoplast, Chromoplast_Ath, by.x = "Ath", by.y = "GeneID")
#Ath_Niben_Chromoplast <- Ath_Niben_Chromoplast[-c(5,6,7),]
Ath_Niben_Chromoplast <- Ath_Niben_Chromoplast[order(match(Ath_Niben_Chromoplast[,1],Chromoplast_Ath[,1])),]

## We create a new df only with benthamiana accesions

Niben_Chromoplast <- Ath_Niben_Chromoplast %>% 
  dplyr::select(Niben, Group)
colnames(Niben_Chromoplast) <- c("Name", "Family")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_Chromoplast <- merge(Niben_Chromoplast, TPM_All_FC_Annotated, by="Name")

## We select only the FC data from the different comparisons
Niben_Chromoplast <- Niben_Chromoplast %>% dplyr::select(Name, 88:105)
Niben_Chromoplast <- Niben_Chromoplast[!duplicated(Niben_Chromoplast$Name),]
Niben_Chromoplast <- Niben_Chromoplast[order(match(Niben_Chromoplast[,1], Ath_Niben_Chromoplast[,2])),]

## We prepare the data for a heatmap and we map it
Niben_Chromoplast_Annotation <- merge(Niben_Chromoplast, Ath_Niben_Chromoplast, by.x="Name", by.y = "Niben")
Niben_Chromoplast_Annotation <- Niben_Chromoplast_Annotation[!duplicated(Niben_Chromoplast_Annotation$Name),]
Niben_Chromoplast_Annotation <- Niben_Chromoplast_Annotation[order(match(Niben_Chromoplast_Annotation[,1], Ath_Niben_Chromoplast[,2])),]
rownames(Niben_Chromoplast_Annotation) <- Niben_Chromoplast_Annotation$Name
Niben_Chromoplast_Annotation <- Niben_Chromoplast_Annotation %>% dplyr::select(Group)
colnames(Niben_Chromoplast_Annotation) <- "Family"

rownames(Niben_Chromoplast) <- Niben_Chromoplast$Name
Niben_Chromoplast[1] <- NULL

Niben_Chromoplast <- Niben_Chromoplast %>%
  dplyr::select(pcrtB6h_vs_GFP6h_filtered, pcrtB24h_vs_GFP24h_filtered, pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

min(Niben_Chromoplast, na.rm = TRUE)
max(Niben_Chromoplast, na.rm = TRUE)
breaks_Chromoplast=c(seq(-7,0, length.out = 100), seq(0.01,9,length.out = 100))

color_Chromoplast <- colorRampPalette(c("#415a20","#9bac12","#e5c033","#c38144","#894539"))(n=length(unique(Niben_Chromoplast_Annotation$Family)))
names(color_Chromoplast) <- unique(Niben_Chromoplast_Annotation$Family)
color_Chromoplast <- list(Family=color_Chromoplast)

pheatmap(Niben_Chromoplast, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         #breaks = breaks_Chromoplast,
         breaks = breaks_Stndr,
         legend_breaks = c(-4, -2, 0, 2, 4),
         legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         annotation_row=Niben_Chromoplast_Annotation, annotation_colors=color_Chromoplast, 
         annotation_names_row = TRUE, show_rownames = FALSE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("6h", "24h", "22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h", "96h"),
         cex = 1,
         angle_col=45,
         main = "Chromoplast Genes",
         annotation_legend = TRUE,
         #display_numbers = Nb_Sly_Ath_MVA_Sig,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)


#########################################################################################################################################
####################################  PARTICULAR GENES ARABIDOPSIS  #############################################################################################
#########################################################################################################################################
setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")

Ath11_fasta <- read.fasta("../../Otros RNAseq/Woo et al (2016) Plant Physiol/Araport11_genes.201606.pep.fasta",
                             seqtype = "AA", as.string = TRUE)

##AT3G09640 (APX2)
AT3G09640_Arabidopsis <- read.table("Funcional Groups/AT3G09640_Arabidopsis.txt")
AT3G09640_Arabidopsis <- AT3G09640_Arabidopsis$V1
AT3G09640_Arabidopsis_fasta <- Ath11_fasta[c(which(substr(names(Ath11_fasta), 1,9) %in% AT3G09640_Arabidopsis))]
write.fasta(AT3G09640_Arabidopsis_fasta, as.string=TRUE, names= names(AT3G09640_Arabidopsis_fasta),
            file.out = "Funcional Groups/Ath_AT3G09640.fasta")

AT3G09640_Nicotiana <- read.table("Funcional Groups/AT3G09640_Niben261.txt")
AT3G09640_Nicotiana <- AT3G09640_Nicotiana$V1
AT3G09640_Nicotiana_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% AT3G09640_Nicotiana))]
write.fasta(AT3G09640_Nicotiana_fasta, as.string=TRUE, names= names(AT3G09640_Nicotiana_fasta),
            file.out = "Funcional Groups/Niben_AT3G09640.fasta")

AT3G09640_Tomato <- read.table("Funcional Groups/AT3G09640_Tomato.txt")
AT3G09640_Tomato <- AT3G09640_Tomato$V1
AT3G09640_Tomato_fasta <- Tomato_4.0_fasta[c(which(names(Tomato_4.0_fasta) %in% AT3G09640_Tomato))]
write.fasta(AT3G09640_Tomato_fasta, as.string=TRUE, names= names(AT3G09640_Tomato_fasta),
            file.out = "Funcional Groups/Tomato_AT3G09640.fasta")

# Based on the Filogenetic Tree
AT3G09640_Nicotiana_Tree <- c("Niben261Chr01g0007013.1", "Niben261Chr12g0257004.1")
APX2_Niben261 <- data.frame("Name" = AT3G09640_Nicotiana_Tree,
                            "Arabidopsis" = "AT3G09640")
APX2_Niben261 <- merge(APX2_Niben261, TPM_All_FC_Annotated, by="Name")
APX2_Niben261 <- APX2_Niben261 %>% select(1:3, 88:105)
write.table(APX2_Niben261, "Funcional Groups/APX2_Nicotiana_TimePoints_FC.txt", quote = FALSE, row.names = FALSE, sep = "\t", dec = ",")

##AT5G63980 (SAL1)
AT5G63980_Arabidopsis <- read.table("Funcional Groups/AT5G63980_Arabidopsis.txt")
AT5G63980_Arabidopsis <- AT5G63980_Arabidopsis$V1
AT5G63980_Arabidopsis_fasta <- Ath11_fasta[c(which(substr(names(Ath11_fasta), 1,9) %in% AT5G63980_Arabidopsis))]
write.fasta(AT5G63980_Arabidopsis_fasta, as.string=TRUE, names= names(AT5G63980_Arabidopsis_fasta),
            file.out = "Funcional Groups/Ath_AT5G63980.fasta")

AT5G63980_Nicotiana <- read.table("Funcional Groups/AT5G63980_Niben261.txt")
AT5G63980_Nicotiana <- AT5G63980_Nicotiana$V1
AT5G63980_Nicotiana_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% AT5G63980_Nicotiana))]
write.fasta(AT5G63980_Nicotiana_fasta, as.string=TRUE, names= names(AT5G63980_Nicotiana_fasta),
            file.out = "Funcional Groups/Niben_AT5G63980.fasta")

AT5G63980_Tomato <- read.table("Funcional Groups/AT5G63980_Tomato.txt")
AT5G63980_Tomato <- AT5G63980_Tomato$V1
AT5G63980_Tomato_fasta <- Tomato_4.0_fasta[c(which(names(Tomato_4.0_fasta) %in% AT5G63980_Tomato))]
write.fasta(AT5G63980_Tomato_fasta, as.string=TRUE, names= names(AT5G63980_Tomato_fasta),
            file.out = "Funcional Groups/Tomato_AT5G63980.fasta")

# Based on the Filogenetic Tree
AT5G63980_Nicotiana_Tree <- c("Niben261Chr14g0571005.1", "Niben261Chr15g0621017.1")
SAL1_Niben261 <- data.frame("Name" = AT5G63980_Nicotiana_Tree,
                            "Arabidopsis" = "AT5G63980")
SAL1_Niben261 <- merge(SAL1_Niben261, TPM_All_FC_Annotated, by="Name")
SAL1_Niben261 <- SAL1_Niben261 %>% select(1:3, 88:105)
write.table(SAL1_Niben261, "Funcional Groups/SAL1_Nicotiana_TimePoints_FC.txt", quote = FALSE, row.names = FALSE, sep = "\t", dec = ",")


#########################################################################################################################################
####################################   FILOGENETIC TREE   ##################################################################################
#########################################################################################################################################

normalize_df <- function(df) {
  y <- as.data.frame(t(apply(df, 1, function(x)(x-min(x))/(max(x)-min(x)))))
  return(y)
} 


which_Module_pwd30 <- function(df){
  Cornell_Modules_filt <- read.csv("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/WGCNA/Filtrated/geneInfo_pwd30.csv", row.names = 1)
  df_merged <- merge(df, Cornell_Modules_filt, by.x= "Name", by.y= "GeneID", all.x= TRUE)
  df_merged <- df_merged %>% select(Protein, moduleColor)
  df_merged[is.na(df_merged)] <- "None"
  df_merged <- df_merged[order(match(df_merged$Protein, df$Protein)),]
  df_merged$Protein <- factor(df_merged$Protein, levels = df$Protein)
  rownames(df_merged) <- df_merged$Protein
  df_merged$Protein <- NULL
  return(df_merged)
}

which_Module_pwd12 <- function(df){
  Cornell_Modules_filt <- read.csv("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/WGCNA/Filtrated/geneInfo_pwd12.csv", row.names = 1)
  df_merged <- merge(df, Cornell_Modules_filt, by.x= "Name", by.y= "GeneID", all.x= TRUE)
  df_merged <- df_merged %>% select(Protein, moduleColor)
  df_merged[is.na(df_merged)] <- "None"
  df_merged <- df_merged[order(match(df_merged$Protein, df$Protein)),]
  df_merged$Protein <- factor(df_merged$Protein, levels = df$Protein)
  rownames(df_merged) <- df_merged$Protein
  df_merged$Protein <- NULL
  return(df_merged)
}

which_Module_DESeq2 <- function(df){
  hubgenes_pwd12_DESeq2_All_expcrtB <- read.csv("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/WGCNA/Filtrated/DESeq2_All/Pwd12/geneInfo_DESeq2_All_pwd12_expcrtB.csv", header = TRUE)
  hubgenes_pwd12_DESeq2_All_Phytoene <- read.csv("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/WGCNA/Filtrated/DESeq2_All/Pwd12/geneInfo_DESeq2_All_pwd12_Phytoene.csv", header = TRUE)
  hubgenes_pwd12_DESeq2_All_Phytoene_2 <- hubgenes_pwd12_DESeq2_All_Phytoene %>% dplyr::select(GeneID, GS.Phytoene, p.GS.Phytoene)
  hubgenes_pwd12_DESeq2_All <- merge(hubgenes_pwd12_DESeq2_All_expcrtB, hubgenes_pwd12_DESeq2_All_Phytoene_2, by="GeneID")
  hubgenes_pwd12_DESeq2_All <- hubgenes_pwd12_DESeq2_All %>% dplyr::select(1:5, GS.Phytoene, p.GS.Phytoene, 6:49)
  df_merged <- merge(df, hubgenes_pwd12_DESeq2_All, by.x= "Name", by.y= "GeneID", all.x= TRUE)
  df_merged <- df_merged %>% select(Protein, moduleColor)
  df_merged[is.na(df_merged)] <- "None"
  df_merged$moduleColor <- factor(df_merged$moduleColor, 
                                  levels = c("grey", "tan", "cyan", "yellow", "salmon", "green", 
                                             "black", "red", "greenyellow", "pink", "grey60", "midnightblue",
                                             "lightyellow", "lightcyan",  "brown", "turquoise", 
                                             "blue", "lightgreen", "royalblue", "magenta", "purple", "None"))
  colnames(df_merged)[2] <- "moduleColorDESeq2"
  df_merged <- df_merged[order(match(df_merged$Protein, df$Protein)),]
  df_merged$Protein <- factor(df_merged$Protein, levels = df$Protein)
  rownames(df_merged) <- df_merged$Protein
  df_merged$Protein <- NULL
  return(df_merged)
}


module_colors <- list(
  moduleColor= c(black="black",
                 blue = "#71A9F7",
                 brown = "#BE8A60",
                 green = "#93C48B",  #"#87D68D",
                 greenyellow = "#D0E562",  #"#C5E99B",
                 grey = "grey",
                 magenta= "#f46197",
                 pink= "#ffc0cb",
                 purple= "#631D76",
                 red = "red",
                 turquoise= "turquoise",
                 yellow = "#FFEE88",
                 None = "#FFFFFF")
)

module_Chl_colors <- list(
  moduleColor= c(black="black",
                 blue = "#71A9F7",
                 brown = "#BE8A60",
                 green = "#93C48B",  #"#87D68D",
                 greenyellow = "#D0E562",  #"#C5E99B",
                 grey = "grey",
                 magenta= "#f46197",
                 pink= "#ffc0cb",
                 purple= "#631D76",
                 red = "red",
                 turquoise= "turquoise",
                 yellow = "#FFEE88",
                 None = "#FFFFFF"),
  Prediction= c(cTP= "#a1d76a",
                 mTP= "#e9a3c9",
                 noTP= "#f7f7f7")
)

module_Chl_colors_2 <- list(
  moduleColor= c(black="black",
                 blue = "#71A9F7",
                 brown = "#BE8A60",
                 cyan = "#00ffff",
                 darkred = "#8b0000",
                 green = "#008000",  #"#87D68D",
                 greenyellow = "#D0E562",  #"#C5E99B",
                 grey = "grey",
                 grey60 = "#b5b5b5",
                 lightcyan = "#e0ffff",
                 lightgreen = "#90ee90",
                 lightyellow = "#ffffc8",
                 magenta= "#f46197",
                 midnightblue = "#191970",
                 pink= "#ffc0cb",
                 purple= "#631D76",
                 red = "red",
                 royalblue = "#4169e1",
                 salmon = "#fa8072",
                 tan = "#d2b48c",
                 turquoise= "turquoise",
                 yellow = "#FFEE88",
                 None = "#FFFFFF"),
  Prediction= c(cTP= "#a1d76a",
                mTP= "#e9a3c9",
                noTP= "#f7f7f7")
)

module_Chl_colors_3 <- list(
  moduleColorDESeq2= c(black="black",
                       blue = "#71A9F7",
                       brown = "#BE8A60",
                       cyan = "#00ffff",
                       green = "#008000",  #"#87D68D",
                       greenyellow = "#D0E562",  #"#C5E99B",
                       grey = "grey",
                       grey60 = "#b5b5b5",
                       lightcyan = "#e0ffff",
                       lightgreen = "#90ee90",
                       lightyellow = "#ffffc8",
                       magenta= "#f46197",
                       midnightblue = "#191970",
                       pink= "#ffc0cb",
                       purple= "#631D76",
                       red = "red",
                       royalblue = "#4169e1",
                       salmon = "#fa8072",
                       tan = "#d2b48c",
                       turquoise= "turquoise",
                       yellow = "#FFEE88",
                       None = "#FFFFFF"),
  moduleColor= c(black="black",
                 blue = "#71A9F7",
                 brown = "#BE8A60",
                 cyan = "#00ffff",
                 darkred = "#8b0000",
                 green = "#008000",  #"#87D68D",
                 greenyellow = "#D0E562",  #"#C5E99B",
                 grey = "grey",
                 grey60 = "#b5b5b5",
                 lightcyan = "#e0ffff",
                 lightgreen = "#90ee90",
                 lightyellow = "#ffffc8",
                 magenta= "#f46197",
                 midnightblue = "#191970",
                 pink= "#ffc0cb",
                 purple= "#631D76",
                 red = "red",
                 royalblue = "#4169e1",
                 salmon = "#fa8072",
                 tan = "#d2b48c",
                 turquoise= "turquoise",
                 yellow = "#FFEE88",
                 None = "#FFFFFF"),
  Prediction= c(cTP= "#a1d76a",
                mTP= "#e9a3c9",
                noTP= "#f7f7f7")
)

module_Chl_colors_3$moduleColorDESeq2 <- module_Chl_colors_3$moduleColorDESeq2[order(match(names(module_Chl_colors_3$moduleColorDESeq2), levels(hubgenes_pwd12_DESeq2_All$moduleColor)))]

module_Chl_colors_4 <- list(
  moduleColorDESeq2= c(black="black",
                       blue = "#71A9F7",
                       brown = "#BE8A60",
                       cyan = "#00ffff",
                       green = "#008000",  #"#87D68D",
                       greenyellow = "#D0E562",  #"#C5E99B",
                       grey = "grey",
                       grey60 = "#b5b5b5",
                       lightcyan = "#e0ffff",
                       lightgreen = "#90ee90",
                       lightyellow = "#ffffc8",
                       magenta= "#f46197",
                       midnightblue = "#191970",
                       pink= "#ffc0cb",
                       purple= "#631D76",
                       red = "red",
                       royalblue = "#4169e1",
                       salmon = "#fa8072",
                       tan = "#d2b48c",
                       turquoise= "turquoise",
                       yellow = "#FFEE88",
                       None = "#FFFFFF"),
  moduleColor= c(black="black",
                 blue = "#71A9F7",
                 brown = "#BE8A60",
                 green = "#93C48B",  #"#87D68D",
                 greenyellow = "#D0E562",  #"#C5E99B",
                 grey = "grey",
                 magenta= "#f46197",
                 pink= "#ffc0cb",
                 purple= "#631D76",
                 red = "red",
                 turquoise= "turquoise",
                 yellow = "#FFEE88",
                 None = "#FFFFFF"),
  Prediction= c(cTP= "#a1d76a",
                mTP= "#e9a3c9",
                noTP= "#f7f7f7")
)

module_Chl_colors_4$moduleColorDESeq2 <- module_Chl_colors_4$moduleColorDESeq2[order(match(names(module_Chl_colors_4$moduleColorDESeq2), levels(hubgenes_pwd12_DESeq2_All$moduleColor)))]


heatmap_Niben <- function(groupname, datexpr= c("FC", "TPM"), RNAseq=c("Cornell", "96h"), pwd=c("12", "30")){
  Niben_Ann <- read.table(paste0("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan/Funcional Groups/FigTree/Niben_", groupname, ".txt"),
                                      sep = "\t", header = TRUE)
  colnames(Niben_Ann) <- c("Name", "Protein")
  Niben_Chloro <- read.delim(paste0("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan/Funcional Groups/Niben_", groupname, "_2_summary.targetp2"), 
                                         header=FALSE, comment.char="#")
  colnames(Niben_Chloro) <- c("ID",	"Prediction",	"noTP",	"SP",	"mTP",	"cTP",	"luTP",	"CS-Position")
  Niben_Chloro$Prediction <- factor(Niben_Chloro$Prediction, levels = c("cTP","mTP","noTP"))
  Niben_Chloro <- merge(Niben_Ann, Niben_Chloro, by.x = "Protein", by.y = "ID")
  Niben_Chloro <- Niben_Chloro[order(match(Niben_Chloro$Protein, Niben_Ann$Protein)),]
  rownames(Niben_Chloro) <- Niben_Chloro$Protein
  Niben_Chloro <- Niben_Chloro %>% select(Prediction)
  ## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information
  Niben_groupname <- merge(Niben_Ann, TPM_All_FC_Annotated, by="Name")
  ## We create the annotation for Modules
  if(pwd=="12"){
    Niben_Modules <- which_Module_pwd12(Niben_Ann)
  } else {
    Niben_Modules <- which_Module_pwd30(Niben_Ann)
  }
  Niben_Modules_DESeq2 <- which_Module_DESeq2(Niben_Ann)
  ## We merge this annotation of Modules with the annotation of Chloroplast TP prediction
  Niben_Modules <- merge(Niben_Modules_DESeq2,Niben_Modules, by = 0)
  Niben_Modules_Chl <- merge(Niben_Modules, Niben_Chloro, by.x = 1, by.y = 0)
  Niben_Modules_Chl <- Niben_Modules_Chl[order(match(Niben_Modules_Chl$Row.names, Niben_Ann$Protein)),]
  rownames(Niben_Modules_Chl) <- Niben_Modules_Chl$Row.names
  Niben_Modules_Chl$Row.names <- NULL
  ## We select only the FC data from the different comparisons
  Niben_TPM <- Niben_groupname %>% select(Protein, 4:51, 61:66)
  Niben_TPM <- Niben_TPM[!duplicated(Niben_TPM$Protein),]
  Niben_TPM <- Niben_TPM[order(match(Niben_TPM[,1], Niben_Ann[,2])),]
  ## We select only the FC data from the different comparisons
  Niben_FC <- Niben_groupname %>% select(Protein, 71, 73:87)
  Niben_FC <- Niben_FC[!duplicated(Niben_FC$Protein),]
  Niben_FC <- Niben_FC[order(match(Niben_FC[,1], Niben_Ann[,2])),]
  ## We select only the FC data from the different comparisons
  Niben_DEG <- Niben_groupname %>% select(Protein, 89, 91:105)
  Niben_DEG <- Niben_DEG[!duplicated(Niben_DEG$Protein),]
  Niben_DEG <- Niben_DEG[order(match(Niben_DEG[,1], Niben_Ann[,2])),]
  ## We create new columns with * to mark DEGs (one for each comparison)  
  Niben_DEG$pcrtB22h_vs_GFP22h_filtered <- ifelse(is.na(Niben_DEG$pcrtB22h_vs_GFP22h_filtered) == TRUE, " ", "*")
  Niben_DEG$pcrtB25h_vs_GFP25h_filtered <- ifelse(is.na(Niben_DEG$pcrtB25h_vs_GFP25h_filtered) == TRUE, " ", "*")
  Niben_DEG$pcrtBAA_vs_GFPAA_filtered <- ifelse(is.na(Niben_DEG$pcrtBAA_vs_GFPAA_filtered) == TRUE, " ", "*")
  Niben_DEG$pcrtBBB_vs_GFPBB_filtered <- ifelse(is.na(Niben_DEG$pcrtBBB_vs_GFPBB_filtered) == TRUE, " ", "*")
  Niben_DEG$pcrtBYY_vs_GFPYY_filtered <- ifelse(is.na(Niben_DEG$pcrtBYY_vs_GFPYY_filtered) == TRUE, " ", "*")
  Niben_DEG$pcrtB28h_vs_GFP28h_filtered <- ifelse(is.na(Niben_DEG$pcrtB28h_vs_GFP28h_filtered) == TRUE, " ", "*")
  Niben_DEG$pcrtB34h_vs_GFP34h_filtered <- ifelse(is.na(Niben_DEG$pcrtB34h_vs_GFP34h_filtered) == TRUE, " ", "*")
  Niben_DEG$pcrtB37h_vs_GFP37h_filtered <- ifelse(is.na(Niben_DEG$pcrtB37h_vs_GFP37h_filtered) == TRUE, " ", "*")
  Niben_DEG$pcrtBZZ_vs_GFPZZ_filtered <- ifelse(is.na(Niben_DEG$pcrtBZZ_vs_GFPZZ_filtered) == TRUE, " ", "*")
  Niben_DEG$pcrtBCC_vs_GFPCC_filtered <- ifelse(is.na(Niben_DEG$pcrtBCC_vs_GFPCC_filtered) == TRUE, " ", "*")
  Niben_DEG$pcrtB40h_vs_GFP40h_filtered <- ifelse(is.na(Niben_DEG$pcrtB40h_vs_GFP40h_filtered) == TRUE, " ", "*")
  Niben_DEG$pcrtBDD_vs_GFPDD_filtered <- ifelse(is.na(Niben_DEG$pcrtBDD_vs_GFPDD_filtered) == TRUE, " ", "*")
  Niben_DEG$pcrtB46h_vs_GFP46h_filtered <- ifelse(is.na(Niben_DEG$pcrtB46h_vs_GFP46h_filtered) == TRUE, " ", "*")
  Niben_DEG$pcrtB56h_vs_GFP56h_filtered <- ifelse(is.na(Niben_DEG$pcrtB56h_vs_GFP56h_filtered) == TRUE, " ", "*")
  Niben_DEG$pcrtBEE_vs_GFPEE_filtered <- ifelse(is.na(Niben_DEG$pcrtBEE_vs_GFPEE_filtered) == TRUE, " ", "*")
  Niben_DEG$pcrtB96h_vs_GFP96h_filtered <- ifelse(is.na(Niben_DEG$pcrtB96h_vs_GFP96h_filtered) == TRUE, " ", "*")
  row.names(Niben_DEG) <- Niben_DEG$Protein
  Niben_DEG[1] <- NULL
  ## We establish the order we want to show the columns of DEGs
  Niben_DEG <- Niben_DEG %>%
    select(pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
           pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
           pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
           pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)
  row.names(Niben_FC) <- Niben_FC$Protein
  Niben_FC[1] <- NULL
  ## We establish the order we want to show the columns of FC
  Niben_FC <- Niben_FC %>%
    select(pcrtB22h_vs_GFP22h_All, pcrtBAA_vs_GFPAA_All, pcrtB25h_vs_GFP25h_All, 
           pcrtBYY_vs_GFPYY_All, pcrtBBB_vs_GFPBB_All, pcrtB28h_vs_GFP28h_All, pcrtB34h_vs_GFP34h_All, pcrtBZZ_vs_GFPZZ_All, 
           pcrtB37h_vs_GFP37h_All, pcrtBCC_vs_GFPCC_All, pcrtB40h_vs_GFP40h_All, pcrtBDD_vs_GFPDD_All, pcrtB46h_vs_GFP46h_All, 
           pcrtBEE_vs_GFPEE_All, pcrtB56h_vs_GFP56h_All, pcrtB96h_vs_GFP96h_All)
  ## We create the breaks for the FC in the heatmap
  min(Niben_FC, na.rm = TRUE)
  max(Niben_FC, na.rm = TRUE)
  breaks_FC=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))
  rownames(Niben_TPM) <- Niben_TPM$Protein
  Niben_TPM[1] <- NULL
  ## We create the breaks for the TPM in the heatmap
  min(Niben_TPM, na.rm = TRUE)
  max(Niben_TPM, na.rm = TRUE)
  breaks_Niben_TPM=c(seq(0,100, length.out = 100))
  ## Show Cornell columns if we want to show them, or also 96h (default)
  if(RNAseq=="Cornell"){
    Niben_FC <- Niben_FC %>% select(1:15)
    Niben_DEG <- Niben_DEG %>% select(1:15)
    Niben_TPM <- Niben_TPM %>% select(1:48) 
  }
  ## We normalize the data of TPM to be showed in a range from 0 to 1
  Niben_TPM_norm <- normalize_df(Niben_TPM)
  ## This will be the heatmap created if we want to show FC data
  if(datexpr=="FC"){
    pheatmap(Niben_FC, 
             border_color = "grey60",legend= TRUE,
             color=color3, 
             cluster_rows =FALSE,
             breaks = breaks_Carotenoids_FC,
             #breaks = breaks_Stndr,
             legend_breaks = c(-3, -1.5, 0, 1.5, 3),
             legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
             #legend_breaks = c(0, 40, 80, 120, 160, 200),
             #legend_labels = c("0", "40", "80", "120", "160", ">200"),
             cluster_cols = FALSE,
             show_colnames=T, treeheight_col = 35, 
             fontsize=12, 
             scale = "none",
             #annotation_row=Niben_Carotenoids_Annotation, annotation_colors=color_Carotenoids, 
             #annotation_names_row = TRUE, 
             show_rownames = TRUE,  
             #cellwidth = 15, cellheight = 15,
             fontsize_row = 7,
             fontsize_col = 8,
             labels_col = if(RNAseq=="Cornell"){
               c("22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h")
             } else {
               c("22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h", "96h")
             },
             cex = 1,
             angle_col=45,
             #main = groupname,
             annotation_legend = FALSE,
             display_numbers = Niben_DEG,
             fontsize_number = 15,
             na_col = "darkgrey",
             #gaps_row = c(84, 104),
             annotation_row = Niben_Modules_Chl,
             annotation_colors = if(pwd=="30"){
               module_Chl_colors_4
             } else {
               module_Chl_colors_3
             },
             gaps_col = c(15)
    )
  } else { ## This will be the heatmap created if we want to show TPM data
    pheatmap(Niben_TPM_norm, 
             border_color = "grey60",legend= TRUE,
             #color=color3, 
             cluster_rows =FALSE,
             #`breaks = breaks_Niben_TPM,
             #breaks = breaks_Stndr,
             #legend_breaks = c(-4, -2, 0, 2, 4),
             #legend_labels = c("<-4", "-2", "0", "2", ">4"),
             #legend_breaks = c(0, 40, 80, 120, 160, 200),
             #legend_labels = c("0", "40", "80", "120", "160", ">200"),
             cluster_cols = FALSE,
             show_colnames=T, treeheight_col = 35, 
             fontsize=12, 
             scale = "none",
             #annotation_row=Niben_HSP20_Annotation, annotation_colors=color_HSP20, 
             #annotation_names_row = TRUE, 
             show_rownames = TRUE,  
             #cellwidth = 15, cellheight = 15,
             fontsize_row = 7,
             fontsize_col = 8,
             #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
             cex = 1,
             angle_col=45,
             #main = groupname,
             annotation_legend = FALSE,
             #display_numbers = Niben_HSP20_DEG,
             fontsize_number = 15,
             na_col = "darkgrey",
             annotation_row = Niben_Modules_Chl,
             annotation_colors = if(pwd=="30"){
               module_Chl_colors_4
             } else {
               module_Chl_colors_3
             },
             #gaps_row = c(22, 29, 49, 77),
             gaps_col = if(RNAseq=="Cornell"){
               c(24)
             } else {
               c(24, 48, 51)
             }
             )
  }
}

save_heatmaps <- function(groupname, height){
  pdf(file=paste0("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan/Funcional Groups/FigTree/", groupname, "/Heatmaps_Modules_Location/DESeq2/", groupname, "_Cornell_FC_pwd30_DESeq2.pdf"),
                  height= height, width= 5)
  heatmap_Niben(groupname, datexpr = "FC", RNAseq = "Cornell", pwd = "30")
  dev.off()
  pdf(file=paste0("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan/Funcional Groups/FigTree/", groupname, "/Heatmaps_Modules_Location/DESeq2/", groupname, "_Cornell_FC_pwd12_DESeq2.pdf"),
                  height= height, width= 5)
  heatmap_Niben(groupname, datexpr = "FC", RNAseq = "Cornell", pwd = "12")
  dev.off()
  pdf(file=paste0("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan/Funcional Groups/FigTree/", groupname, "/Heatmaps_Modules_Location/DESeq2/", groupname, "_Cornell&96h_FC_pwd30_DESeq2.pdf"),
      height= height, width= 5)
  heatmap_Niben(groupname, datexpr = "FC", RNAseq = "96h", pwd = "30")
  dev.off()
  pdf(file=paste0("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan/Funcional Groups/FigTree/", groupname, "/Heatmaps_Modules_Location/DESeq2/", groupname, "_Cornell&96h_FC_pwd12_DESeq2.pdf"),
      height= height, width= 5)
  heatmap_Niben(groupname, datexpr = "FC", RNAseq = "96h", pwd = "12")
  dev.off()
  pdf(file=paste0("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan/Funcional Groups/FigTree/", groupname, "/Heatmaps_Modules_Location/DESeq2/", groupname, "_Cornell_TPM_pwd30_DESeq2.pdf"),
      height= height, width= 8)
  heatmap_Niben(groupname, datexpr = "TPM", RNAseq = "Cornell", pwd = "30")
  dev.off()
  pdf(file=paste0("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan/Funcional Groups/FigTree/", groupname, "/Heatmaps_Modules_Location/DESeq2/", groupname, "_Cornell_TPM_pwd12_DESeq2.pdf"),
      height= height, width= 10)
  heatmap_Niben(groupname, datexpr = "TPM", RNAseq = "Cornell", pwd = "12")
  dev.off()
  pdf(file=paste0("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan/Funcional Groups/FigTree/", groupname, "/Heatmaps_Modules_Location/DESeq2/", groupname, "_Cornell&96h_TPM_pwd30_DESeq2.pdf"),
      height= height, width= 8)
  heatmap_Niben(groupname, datexpr = "TPM", RNAseq = "96h", pwd = "30")
  dev.off()
  pdf(file=paste0("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan/Funcional Groups/FigTree/", groupname, "/Heatmaps_Modules_Location/DESeq2/", groupname, "_Cornell&96h_TPM_pwd12_DESeq2.pdf"),
      height= height, width= 8)
  heatmap_Niben(groupname, datexpr = "TPM", RNAseq = "96h", pwd = "12")
  dev.off()
}

####################################  Autophagy  #############################################################################################
#########################################################################################################################################

Autophagy_Ath <- read.table("Funcional Groups/Ath_Autophagy.txt", header = TRUE)
Ath_Gene_Models <- read.table("../../Otros RNAseq/Woo et al (2016) Plant Physiol/TAIR10_representative_gene_models_CORRESPONDENCE.txt")
Autophagy_Ath <- merge(Autophagy_Ath, Ath_Gene_Models, by.x="ID", by.y=1)
Autophagy_Ath_fasta <- Ath11_fasta[c(which(names(Ath11_fasta) %in% Autophagy_Ath$V2))]
Autophagy_Ath_2 <- Autophagy_Ath %>% select(V2, Name)
write.fasta(Autophagy_Ath_fasta, as.string=TRUE, names= names(Autophagy_Ath_fasta),
            file.out = "Funcional Groups/Ath_Autophagy.fasta")
rename.fasta(infile = "Funcional Groups/Ath_Autophagy.fasta", Autophagy_Ath_2, outfile = "Funcional Groups/Ath_Autophagy_2.fasta")

Autophagy_Niben_Candidates <- read.table("Funcional Groups/Niben_Candidates_Autophagy.txt")
Autophagy_Niben_Candidates <- Autophagy_Niben_Candidates[Autophagy_Niben_Candidates$V3>=35,] %>% select(V2)
Autophagy_Niben_Candidates <- unique(Autophagy_Niben_Candidates)
Autophagy_Niben_Candidates_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Autophagy_Niben_Candidates$V2))]
write.fasta(Autophagy_Niben_Candidates_fasta, as.string=TRUE, names= names(Autophagy_Niben_Candidates_fasta),
            file.out = "Funcional Groups/Autophagy_Niben_Candidates_35.fasta")

Autophagy_Sly_Candidates <- read.table("Funcional Groups/Sly_Candidates_Autophagy.txt")
Autophagy_Sly_Candidates <- Autophagy_Sly_Candidates[Autophagy_Sly_Candidates$V3>=35,] %>% select(V2)
Autophagy_Sly_Candidates <- unique(Autophagy_Sly_Candidates)
Autophagy_Sly_Candidates_fasta <- Tomato_4.0_fasta[c(which(names(Tomato_4.0_fasta) %in% Autophagy_Sly_Candidates$V2))]
write.fasta(Autophagy_Sly_Candidates_fasta, as.string=TRUE, names= names(Autophagy_Sly_Candidates_fasta),
            file.out = "Funcional Groups/Autophagy_Sly_Candidates_35.fasta")

Autophagy_Cann_Candidates <- read.table("Funcional Groups/Cann_Candidates_Autophagy.txt")
Autophagy_Cann_Candidates <- Autophagy_Cann_Candidates[Autophagy_Cann_Candidates$V3>=35,] %>% select(V2)
Autophagy_Cann_Candidates <- unique(Autophagy_Cann_Candidates)
Autophagy_Cann_Candidates_fasta <- Cannuum_ASM51225v2_fasta[c(which(names(Cannuum_ASM51225v2_fasta) %in% Autophagy_Cann_Candidates$V2))]
write.fasta(Autophagy_Cann_Candidates_fasta, as.string=TRUE, names= names(Autophagy_Cann_Candidates_fasta),
            file.out = "Funcional Groups/Autophagy_Cann_Candidates_35.fasta")

Niben_Autophagy_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_Autophagy_Ann$Name))]
write.fasta(Niben_Autophagy_fasta, as.string=TRUE, names= names(Niben_Autophagy_fasta),
            file.out = "Funcional Groups/Niben_Autophagy.fasta")
Niben_Autophagy_Ann_2 <- Niben_Autophagy_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_Autophagy.fasta", Niben_Autophagy_Ann, outfile = "Funcional Groups/Niben_Autophagy_2.fasta")

save_heatmaps("Autophagy", height = 10)

#####Representing Heatmap for Niben

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Niben_Autophagy_Ann <- read.table("Funcional Groups/FigTree/Niben_Autophagy.txt",
                            sep = "\t", header = TRUE)

colnames(Niben_Autophagy_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_Autophagy <- merge(Niben_Autophagy_Ann, TPM_All_FC_Annotated, by="Name")

## We create the annotation for Modules

Niben_Autophagy_Modules <- which_Module_pwd30(Niben_Autophagy_Ann)

## We select only the FC data from the different comparisons
Niben_Autophagy_TPM <- Niben_Autophagy %>% select(Protein, 4:51, 61:66)
Niben_Autophagy_TPM <- Niben_Autophagy_TPM[!duplicated(Niben_Autophagy_TPM$Protein),]
Niben_Autophagy_TPM <- Niben_Autophagy_TPM[order(match(Niben_Autophagy_TPM[,1], Niben_Autophagy_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_Autophagy_FC <- Niben_Autophagy %>% select(Protein, 71, 73:87)
Niben_Autophagy_FC <- Niben_Autophagy_FC[!duplicated(Niben_Autophagy_FC$Protein),]
Niben_Autophagy_FC <- Niben_Autophagy_FC[order(match(Niben_Autophagy_FC[,1], Niben_Autophagy_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_Autophagy_DEG <- Niben_Autophagy %>% select(Protein, 89, 91:105)
Niben_Autophagy_DEG <- Niben_Autophagy_DEG[!duplicated(Niben_Autophagy_DEG$Protein),]
Niben_Autophagy_DEG <- Niben_Autophagy_DEG[order(match(Niben_Autophagy_DEG[,1], Niben_Autophagy_Ann[,2])),]

Niben_Autophagy_DEG$pcrtB22h_vs_GFP22h_filtered <- ifelse(is.na(Niben_Autophagy_DEG$pcrtB22h_vs_GFP22h_filtered) == TRUE, " ", "*")
Niben_Autophagy_DEG$pcrtB25h_vs_GFP25h_filtered <- ifelse(is.na(Niben_Autophagy_DEG$pcrtB25h_vs_GFP25h_filtered) == TRUE, " ", "*")
Niben_Autophagy_DEG$pcrtBAA_vs_GFPAA_filtered <- ifelse(is.na(Niben_Autophagy_DEG$pcrtBAA_vs_GFPAA_filtered) == TRUE, " ", "*")
Niben_Autophagy_DEG$pcrtBBB_vs_GFPBB_filtered <- ifelse(is.na(Niben_Autophagy_DEG$pcrtBBB_vs_GFPBB_filtered) == TRUE, " ", "*")
Niben_Autophagy_DEG$pcrtBYY_vs_GFPYY_filtered <- ifelse(is.na(Niben_Autophagy_DEG$pcrtBYY_vs_GFPYY_filtered) == TRUE, " ", "*")
Niben_Autophagy_DEG$pcrtB28h_vs_GFP28h_filtered <- ifelse(is.na(Niben_Autophagy_DEG$pcrtB28h_vs_GFP28h_filtered) == TRUE, " ", "*")
Niben_Autophagy_DEG$pcrtB34h_vs_GFP34h_filtered <- ifelse(is.na(Niben_Autophagy_DEG$pcrtB34h_vs_GFP34h_filtered) == TRUE, " ", "*")
Niben_Autophagy_DEG$pcrtB37h_vs_GFP37h_filtered <- ifelse(is.na(Niben_Autophagy_DEG$pcrtB37h_vs_GFP37h_filtered) == TRUE, " ", "*")
Niben_Autophagy_DEG$pcrtBZZ_vs_GFPZZ_filtered <- ifelse(is.na(Niben_Autophagy_DEG$pcrtBZZ_vs_GFPZZ_filtered) == TRUE, " ", "*")
Niben_Autophagy_DEG$pcrtBCC_vs_GFPCC_filtered <- ifelse(is.na(Niben_Autophagy_DEG$pcrtBCC_vs_GFPCC_filtered) == TRUE, " ", "*")
Niben_Autophagy_DEG$pcrtB40h_vs_GFP40h_filtered <- ifelse(is.na(Niben_Autophagy_DEG$pcrtB40h_vs_GFP40h_filtered) == TRUE, " ", "*")
Niben_Autophagy_DEG$pcrtBDD_vs_GFPDD_filtered <- ifelse(is.na(Niben_Autophagy_DEG$pcrtBDD_vs_GFPDD_filtered) == TRUE, " ", "*")
Niben_Autophagy_DEG$pcrtBZZ_vs_GFPZZ_filtered <- ifelse(is.na(Niben_Autophagy_DEG$pcrtBZZ_vs_GFPZZ_filtered) == TRUE, " ", "*")
Niben_Autophagy_DEG$pcrtB46h_vs_GFP46h_filtered <- ifelse(is.na(Niben_Autophagy_DEG$pcrtB46h_vs_GFP46h_filtered) == TRUE, " ", "*")
Niben_Autophagy_DEG$pcrtB56h_vs_GFP56h_filtered <- ifelse(is.na(Niben_Autophagy_DEG$pcrtB56h_vs_GFP56h_filtered) == TRUE, " ", "*")
Niben_Autophagy_DEG$pcrtBEE_vs_GFPEE_filtered <- ifelse(is.na(Niben_Autophagy_DEG$pcrtBEE_vs_GFPEE_filtered) == TRUE, " ", "*")
Niben_Autophagy_DEG$pcrtB96h_vs_GFP96h_filtered <- ifelse(is.na(Niben_Autophagy_DEG$pcrtB96h_vs_GFP96h_filtered) == TRUE, " ", "*")
row.names(Niben_Autophagy_DEG) <- Niben_Autophagy_DEG$Protein
Niben_Autophagy_DEG[1] <- NULL

Niben_Autophagy_DEG <- Niben_Autophagy_DEG %>%
  select(pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

row.names(Niben_Autophagy_FC) <- Niben_Autophagy_FC$Protein
Niben_Autophagy_FC[1] <- NULL

Niben_Autophagy_FC <- Niben_Autophagy_FC %>%
  select(pcrtB22h_vs_GFP22h_All, pcrtBAA_vs_GFPAA_All, pcrtB25h_vs_GFP25h_All, 
         pcrtBYY_vs_GFPYY_All, pcrtBBB_vs_GFPBB_All, pcrtB28h_vs_GFP28h_All, pcrtB34h_vs_GFP34h_All, pcrtBZZ_vs_GFPZZ_All, 
         pcrtB37h_vs_GFP37h_All, pcrtBCC_vs_GFPCC_All, pcrtB40h_vs_GFP40h_All, pcrtBDD_vs_GFPDD_All, pcrtB46h_vs_GFP46h_All, 
         pcrtBEE_vs_GFPEE_All, pcrtB56h_vs_GFP56h_All, pcrtB96h_vs_GFP96h_All)


min(Niben_Autophagy_FC, na.rm = TRUE)
max(Niben_Autophagy_FC, na.rm = TRUE)
breaks_Autophagy_FC=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Niben_Autophagy_FC %>% select(1:15), 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Autophagy_FC,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_Autophagy_Annotation, annotation_colors=color_Autophagy, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h", "96h"),
         cex = 1,
         angle_col=45,
         main = "Autophagy",
         annotation_legend = TRUE,
         display_numbers = Niben_Autophagy_DEG %>% select(1:15),
         fontsize_number = 15,
         na_col = "darkgrey",
         annotation_row = Niben_Autophagy_Modules,
         annotation_colors = module_colors
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)


rownames(Niben_Autophagy_TPM) <- Niben_Autophagy_TPM$Protein
Niben_Autophagy_TPM[1] <- NULL

min(Niben_Autophagy_TPM, na.rm = TRUE)
max(Niben_Autophagy_TPM, na.rm = TRUE)
breaks_Niben_Autophagy_TPM=c(seq(0,100, length.out = 200))

Niben_Autophagy_TPM_norm <- normalize_df(Niben_Autophagy_TPM)
Niben_Autophagy_TPM_norm_Cornell <- Niben_Autophagy_TPM %>%
  select(1:48) 
Niben_Autophagy_TPM_norm_Cornell <- normalize_df(Niben_Autophagy_TPM_norm_Cornell)

pheatmap(Niben_Autophagy_TPM_norm_Cornell, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Niben_Autophagy_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_Autophagy_Annotation, annotation_colors=color_Autophagy, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Autophagy",
         annotation_legend = TRUE,
         #display_numbers = Niben_Autophagy_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         annotation_row = Niben_Autophagy_Modules,
         annotation_colors = module_colors,
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(24)
)



#####Representing Heatmap for Sly

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Sly_Autophagy_Ann <- read.table("Funcional Groups/FigTree/Sly_Autophagy.txt",
                                  sep = "\t", header = TRUE)

colnames(Sly_Autophagy_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Sly_Autophagy <- merge(Sly_Autophagy_Ann, Summary_Yazdani_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Sly_Autophagy_TPM <- Sly_Autophagy %>% select(Protein, 4:66)
Sly_Autophagy_TPM <- Sly_Autophagy_TPM[!duplicated(Sly_Autophagy_TPM$Protein),]
Sly_Autophagy_TPM <- Sly_Autophagy_TPM[order(match(Sly_Autophagy_TPM[,1], Sly_Autophagy_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_Autophagy_FC <- Sly_Autophagy %>% select(Protein, 67:71)
Sly_Autophagy_FC <- Sly_Autophagy_FC[!duplicated(Sly_Autophagy_FC$Protein),]
Sly_Autophagy_FC <- Sly_Autophagy_FC[order(match(Sly_Autophagy_FC[,1], Sly_Autophagy_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_Autophagy_DEG <- Sly_Autophagy %>% select(Protein, 72:76)
Sly_Autophagy_DEG <- Sly_Autophagy_DEG[!duplicated(Sly_Autophagy_DEG$Protein),]
Sly_Autophagy_DEG <- Sly_Autophagy_DEG[order(match(Sly_Autophagy_DEG[,1], Sly_Autophagy_Ann[,2])),]

Sly_Autophagy_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Sly_Autophagy_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Sly_Autophagy_DEG$Ripening_Orange_DEG <- ifelse(is.na(Sly_Autophagy_DEG$Ripening_Orange_DEG) == TRUE, " ", "*")
Sly_Autophagy_DEG$Ripening_Red_DEG <- ifelse(is.na(Sly_Autophagy_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
Sly_Autophagy_DEG$ORHis_MG_DEG <- ifelse(is.na(Sly_Autophagy_DEG$ORHis_MG_DEG) == TRUE, " ", "*")
Sly_Autophagy_DEG$ORHis_Breaker_DEG <- ifelse(is.na(Sly_Autophagy_DEG$ORHis_Breaker_DEG) == TRUE, " ", "*")
row.names(Sly_Autophagy_DEG) <- Sly_Autophagy_DEG$Protein
Sly_Autophagy_DEG[1] <- NULL

rownames(Sly_Autophagy_FC) <- Sly_Autophagy_FC$Protein
Sly_Autophagy_FC[1] <- NULL

min(Sly_Autophagy_FC, na.rm = TRUE)
max(Sly_Autophagy_FC, na.rm = TRUE)
breaks_Sly_Autophagy=c(seq(-4,0, length.out = 100), seq(0.01,4,length.out = 100))

pheatmap(Sly_Autophagy_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Sly_Autophagy,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_Autophagy_Annotation, annotation_colors=color_Autophagy, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Autophagy",
         annotation_legend = TRUE,
         display_numbers = Sly_Autophagy_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = 3
)


rownames(Sly_Autophagy_TPM) <- Sly_Autophagy_TPM$Protein
Sly_Autophagy_TPM[1] <- NULL

min(Sly_Autophagy_TPM, na.rm = TRUE)
max(Sly_Autophagy_TPM, na.rm = TRUE)
breaks_Sly_Autophagy_TPM=c(seq(0,100, length.out = 200))

Sly_Autophagy_TPM_norm <- normalize_df(Sly_Autophagy_TPM)

pheatmap(Sly_Autophagy_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Sly_Autophagy_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_Autophagy_Annotation, annotation_colors=color_Autophagy, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Autophagy",
         annotation_legend = TRUE,
         #display_numbers = Sly_Autophagy_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(21,42)
)




#########################################################################################################################################
####################################  Luca  #############################################################################################
#########################################################################################################################################

Luca_Ath <- read.table("Funcional Groups/Ath_Luca.txt", header = TRUE)
Luca_Ath <- merge(Luca_Ath, Ath_Gene_Models, by.x="ID", by.y=1)
Luca_Ath_fasta <- Ath11_fasta[c(which(names(Ath11_fasta) %in% Luca_Ath$V2))]
Luca_Ath_2 <- Luca_Ath %>% select(V2, Name)
write.fasta(Luca_Ath_fasta, as.string=TRUE, names= names(Luca_Ath_fasta),
            file.out = "Funcional Groups/Ath_Luca.fasta")
rename.fasta(infile = "Funcional Groups/Ath_Luca.fasta", Luca_Ath_2, outfile = "Funcional Groups/Ath_Luca_2.fasta")

Luca_Niben_Candidates <- read.table("Funcional Groups/Niben_Candidates_Luca.txt")
Luca_Niben_Candidates <- Luca_Niben_Candidates[Luca_Niben_Candidates$V3>=35,] %>% select(V2)
Luca_Niben_Candidates <- unique(Luca_Niben_Candidates)
Luca_Niben_Candidates_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Luca_Niben_Candidates$V2))]
write.fasta(Luca_Niben_Candidates_fasta, as.string=TRUE, names= names(Luca_Niben_Candidates_fasta),
            file.out = "Funcional Groups/Luca_Niben_Candidates_35.fasta")

Luca_Sly_Candidates <- read.table("Funcional Groups/Sly_Candidates_Luca.txt")
Luca_Sly_Candidates <- Luca_Sly_Candidates[Luca_Sly_Candidates$V3>=35,] %>% select(V2)
Luca_Sly_Candidates <- unique(Luca_Sly_Candidates)
Luca_Sly_Candidates_fasta <- Tomato_4.0_fasta[c(which(names(Tomato_4.0_fasta) %in% Luca_Sly_Candidates$V2))]
write.fasta(Luca_Sly_Candidates_fasta, as.string=TRUE, names= names(Luca_Sly_Candidates_fasta),
            file.out = "Funcional Groups/Luca_Sly_Candidates_35.fasta")

## Representing Heatmap

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Niben_Luca_Ann <- read.table("Funcional Groups/FigTree/Niben_Luca.txt",
                                  sep = "\t", header = TRUE)

colnames(Niben_Luca_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_Luca <- merge(Niben_Luca_Ann, TPM_All_FC_Annotated, by="Name")

## We select only the FC data from the different comparisons
Niben_Luca <- Niben_Luca %>% select(Protein, 88:105)
Niben_Luca <- Niben_Luca[!duplicated(Niben_Luca$Protein),]
Niben_Luca <- Niben_Luca[order(match(Niben_Luca[,1], Niben_Luca_Ann[,2])),]

## We prepare the data for a heatmap and we map it
Niben_Luca_Annotation <- merge(Niben_Luca, Niben_Luca_Ann, by="Name")
Niben_Luca_Annotation <- Niben_Luca_Annotation[!duplicated(Niben_Luca_Annotation$Name),]
Niben_Luca_Annotation <- Niben_Luca_Annotation[order(match(Niben_Luca_Annotation[,1], Niben_Luca_Ann[,1])),]
#write.table(Niben_Luca_Annotation, "Milan/RNAseq_Milan/Funcional Groups/Luca_Niben261_OrthoVenn_DESeq2FC.txt", quote = FALSE, dec = ",", sep = "\t", row.names = FALSE)
rownames(Niben_Luca_Annotation) <- Niben_Luca_Annotation$Name
#Niben_Luca_Chloroplast <- Niben_Luca_Annotation %>% dplyr::filter(Presumed.subcellular.location=="Chloroplast")
#write.table(Niben_Luca_Chloroplast, "Milan/RNAseq_Milan/Funcional Groups/Luca_Chloroplast_Niben261_OrthoVenn_DESeq2FC.txt", quote = FALSE, dec = ",", sep = "\t", row.names = FALSE)
#Niben_Luca_Chloroplast_Annotation <- Niben_Luca_Chloroplast %>% select(Protein.names)
#Niben_Luca_Chloroplast <- Niben_Luca_Chloroplast %>% select(2:19)
Niben_Luca_Annotation <- Niben_Luca_Annotation %>% select(Protein)

rownames(Niben_Luca) <- Niben_Luca$Protein
Niben_Luca[1] <- NULL

Niben_Luca <- Niben_Luca %>%
  select(pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

min(Niben_Luca, na.rm = TRUE)
max(Niben_Luca, na.rm = TRUE)
breaks_Luca=c(seq(-2,0, length.out = 100), seq(0.01,2.2,length.out = 100))

color_Luca <- colorRampPalette(c("#415a20","#9bac12","#e5c033","#c38144","#894539"))(n=length(unique(Niben_Luca_Annotation$Protein)))
names(color_Luca) <- unique(Niben_Luca_Annotation$Protein)
color_Luca <- list(Protein=color_Luca)

pheatmap(Niben_Luca, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Luca,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_Luca_Annotation, annotation_colors=color_Luca, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h", "96h"),
         cex = 1,
         angle_col=45,
         main = "Luca",
         annotation_legend = TRUE,
         #display_numbers = Nb_Sly_Ath_MVA_Sig,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)



#########################################################################################################################################
####################################  VTE  #############################################################################################
#########################################################################################################################################

VTE_Ath <- read.table("Funcional Groups/Ath_VTE.txt", header = TRUE)
VTE_Ath <- merge(VTE_Ath, Ath_Gene_Models, by.x="ID", by.y=1)
VTE_Ath_fasta <- Ath11_fasta[c(which(names(Ath11_fasta) %in% VTE_Ath$V2))]
VTE_Ath_2 <- VTE_Ath %>% select(V2, Name)
write.fasta(VTE_Ath_fasta, as.string=TRUE, names= names(VTE_Ath_fasta),
            file.out = "Funcional Groups/Ath_VTE.fasta")
rename.fasta(infile = "Funcional Groups/Ath_VTE.fasta", VTE_Ath_2, outfile = "Funcional Groups/Ath_VTE_2.fasta")

VTE_Niben_Candidates <- read.table("Funcional Groups/Niben_Candidates_VTE.txt")
VTE_Niben_Candidates <- VTE_Niben_Candidates[VTE_Niben_Candidates$V3>=35,] %>% select(V2)
VTE_Niben_Candidates <- unique(VTE_Niben_Candidates)
VTE_Niben_Candidates_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% VTE_Niben_Candidates$V2))]
write.fasta(VTE_Niben_Candidates_fasta, as.string=TRUE, names= names(VTE_Niben_Candidates_fasta),
            file.out = "Funcional Groups/VTE_Niben_Candidates_35.fasta")

VTE_Sly_Candidates <- read.table("Funcional Groups/Sly_Candidates_VTE.txt")
VTE_Sly_Candidates <- VTE_Sly_Candidates[VTE_Sly_Candidates$V3>=35,] %>% select(V2)
VTE_Sly_Candidates <- unique(VTE_Sly_Candidates)
VTE_Sly_Candidates_fasta <- Tomato_4.0_fasta[c(which(names(Tomato_4.0_fasta) %in% VTE_Sly_Candidates$V2))]
write.fasta(VTE_Sly_Candidates_fasta, as.string=TRUE, names= names(VTE_Sly_Candidates_fasta),
            file.out = "Funcional Groups/VTE_Sly_Candidates_35.fasta")

Niben_VTE_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_VTE_Ann$Name))]
write.fasta(Niben_VTE_fasta, as.string=TRUE, names= names(Niben_VTE_fasta),
            file.out = "Funcional Groups/Niben_VTE.fasta")
Niben_VTE_Ann_2 <- Niben_VTE_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_VTE.fasta", Niben_VTE_Ann, outfile = "Funcional Groups/Niben_VTE_2.fasta")

save_heatmaps("VTE", height = 5)

## Representing Heatmap

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Niben_VTE_Ann <- read.table("Funcional Groups/FigTree/Niben_VTE.txt",
                              sep = "\t", header = TRUE)

colnames(Niben_VTE_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_VTE <- merge(Niben_VTE_Ann, TPM_All_FC_Annotated, by="Name")

## We create the annotation for Modules

Niben_VTE_Modules <- which_Module_pwd30(Niben_VTE_Ann)

## We select only the FC data from the different comparisons
Niben_VTE_TPM <- Niben_VTE %>% select(Protein, 4:51, 61:66)
Niben_VTE_TPM <- Niben_VTE_TPM[!duplicated(Niben_VTE_TPM$Protein),]
Niben_VTE_TPM <- Niben_VTE_TPM[order(match(Niben_VTE_TPM[,1], Niben_VTE_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_VTE_FC <- Niben_VTE %>% select(Protein, 71, 73:87)
Niben_VTE_FC <- Niben_VTE_FC[!duplicated(Niben_VTE_FC$Protein),]
Niben_VTE_FC <- Niben_VTE_FC[order(match(Niben_VTE_FC[,1], Niben_VTE_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_VTE_DEG <- Niben_VTE %>% select(Protein, 89, 91:105)
Niben_VTE_DEG <- Niben_VTE_DEG[!duplicated(Niben_VTE_DEG$Protein),]
Niben_VTE_DEG <- Niben_VTE_DEG[order(match(Niben_VTE_DEG[,1], Niben_VTE_Ann[,2])),]

Niben_VTE_DEG$pcrtB22h_vs_GFP22h_filtered <- ifelse(is.na(Niben_VTE_DEG$pcrtB22h_vs_GFP22h_filtered) == TRUE, " ", "*")
Niben_VTE_DEG$pcrtB25h_vs_GFP25h_filtered <- ifelse(is.na(Niben_VTE_DEG$pcrtB25h_vs_GFP25h_filtered) == TRUE, " ", "*")
Niben_VTE_DEG$pcrtBAA_vs_GFPAA_filtered <- ifelse(is.na(Niben_VTE_DEG$pcrtBAA_vs_GFPAA_filtered) == TRUE, " ", "*")
Niben_VTE_DEG$pcrtBBB_vs_GFPBB_filtered <- ifelse(is.na(Niben_VTE_DEG$pcrtBBB_vs_GFPBB_filtered) == TRUE, " ", "*")
Niben_VTE_DEG$pcrtBYY_vs_GFPYY_filtered <- ifelse(is.na(Niben_VTE_DEG$pcrtBYY_vs_GFPYY_filtered) == TRUE, " ", "*")
Niben_VTE_DEG$pcrtB28h_vs_GFP28h_filtered <- ifelse(is.na(Niben_VTE_DEG$pcrtB28h_vs_GFP28h_filtered) == TRUE, " ", "*")
Niben_VTE_DEG$pcrtB34h_vs_GFP34h_filtered <- ifelse(is.na(Niben_VTE_DEG$pcrtB34h_vs_GFP34h_filtered) == TRUE, " ", "*")
Niben_VTE_DEG$pcrtB37h_vs_GFP37h_filtered <- ifelse(is.na(Niben_VTE_DEG$pcrtB37h_vs_GFP37h_filtered) == TRUE, " ", "*")
Niben_VTE_DEG$pcrtBZZ_vs_GFPZZ_filtered <- ifelse(is.na(Niben_VTE_DEG$pcrtBZZ_vs_GFPZZ_filtered) == TRUE, " ", "*")
Niben_VTE_DEG$pcrtBCC_vs_GFPCC_filtered <- ifelse(is.na(Niben_VTE_DEG$pcrtBCC_vs_GFPCC_filtered) == TRUE, " ", "*")
Niben_VTE_DEG$pcrtB40h_vs_GFP40h_filtered <- ifelse(is.na(Niben_VTE_DEG$pcrtB40h_vs_GFP40h_filtered) == TRUE, " ", "*")
Niben_VTE_DEG$pcrtBDD_vs_GFPDD_filtered <- ifelse(is.na(Niben_VTE_DEG$pcrtBDD_vs_GFPDD_filtered) == TRUE, " ", "*")
Niben_VTE_DEG$pcrtB46h_vs_GFP46h_filtered <- ifelse(is.na(Niben_VTE_DEG$pcrtB46h_vs_GFP46h_filtered) == TRUE, " ", "*")
Niben_VTE_DEG$pcrtB56h_vs_GFP56h_filtered <- ifelse(is.na(Niben_VTE_DEG$pcrtB56h_vs_GFP56h_filtered) == TRUE, " ", "*")
Niben_VTE_DEG$pcrtBEE_vs_GFPEE_filtered <- ifelse(is.na(Niben_VTE_DEG$pcrtBEE_vs_GFPEE_filtered) == TRUE, " ", "*")
Niben_VTE_DEG$pcrtB96h_vs_GFP96h_filtered <- ifelse(is.na(Niben_VTE_DEG$pcrtB96h_vs_GFP96h_filtered) == TRUE, " ", "*")
row.names(Niben_VTE_DEG) <- Niben_VTE_DEG$Protein
Niben_VTE_DEG[1] <- NULL

Niben_VTE_DEG <- Niben_VTE_DEG %>%
  select(pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

row.names(Niben_VTE_FC) <- Niben_VTE_FC$Protein
Niben_VTE_FC[1] <- NULL

Niben_VTE_FC <- Niben_VTE_FC %>%
  select(pcrtB22h_vs_GFP22h_All, pcrtBAA_vs_GFPAA_All, pcrtB25h_vs_GFP25h_All, 
         pcrtBYY_vs_GFPYY_All, pcrtBBB_vs_GFPBB_All, pcrtB28h_vs_GFP28h_All, pcrtB34h_vs_GFP34h_All, pcrtBZZ_vs_GFPZZ_All, 
         pcrtB37h_vs_GFP37h_All, pcrtBCC_vs_GFPCC_All, pcrtB40h_vs_GFP40h_All, pcrtBDD_vs_GFPDD_All, pcrtB46h_vs_GFP46h_All, 
         pcrtBEE_vs_GFPEE_All, pcrtB56h_vs_GFP56h_All, pcrtB96h_vs_GFP96h_All)


min(Niben_VTE_FC, na.rm = TRUE)
max(Niben_VTE_FC, na.rm = TRUE)
breaks_VTE_FC=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Niben_VTE_FC %>% select(1:15), 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_VTE_FC,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_VTE_Annotation, annotation_colors=color_VTE, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h"), #, "96h"),
         cex = 1,
         angle_col=45,
         main = "VTE",
         annotation_legend = TRUE,
         display_numbers = Niben_VTE_DEG %>% select(1:15),
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(84, 104),
         annotation_row = Niben_VTE_Modules,
         annotation_colors = module_colors
         #gaps_col = c(15)
)


rownames(Niben_VTE_TPM) <- Niben_VTE_TPM$Protein
Niben_VTE_TPM[1] <- NULL

min(Niben_VTE_TPM, na.rm = TRUE)
max(Niben_VTE_TPM, na.rm = TRUE)
breaks_Niben_VTE_TPM=c(seq(0,100, length.out = 100))

Niben_VTE_TPM_norm <- normalize_df(Niben_VTE_TPM)
Niben_VTE_TPM_norm_Cornell <- Niben_VTE_TPM %>%
  select(1:48) 
Niben_VTE_TPM_norm_Cornell <- normalize_df(Niben_VTE_TPM_norm_Cornell)


pheatmap(Niben_VTE_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Niben_VTE_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_VTE_Annotation, annotation_colors=color_VTE, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "VTE",
         annotation_legend = TRUE,
         #display_numbers = Niben_VTE_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         annotation_row = Niben_VTE_Modules,
         annotation_colors = module_colors,
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(24, 48, 51)
)



#####Representing Heatmap for Sly

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Sly_VTE_Ann <- read.table("Funcional Groups/FigTree/Sly_VTE.txt",
                            sep = "\t", header = TRUE)

colnames(Sly_VTE_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Sly_VTE <- merge(Sly_VTE_Ann, Summary_Yazdani_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Sly_VTE_TPM <- Sly_VTE %>% select(Protein, 4:66)
Sly_VTE_TPM <- Sly_VTE_TPM[!duplicated(Sly_VTE_TPM$Protein),]
Sly_VTE_TPM <- Sly_VTE_TPM[order(match(Sly_VTE_TPM[,1], Sly_VTE_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_VTE_FC <- Sly_VTE %>% select(Protein, 67:71)
Sly_VTE_FC <- Sly_VTE_FC[!duplicated(Sly_VTE_FC$Protein),]
Sly_VTE_FC <- Sly_VTE_FC[order(match(Sly_VTE_FC[,1], Sly_VTE_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_VTE_DEG <- Sly_VTE %>% select(Protein, 72:76)
Sly_VTE_DEG <- Sly_VTE_DEG[!duplicated(Sly_VTE_DEG$Protein),]
Sly_VTE_DEG <- Sly_VTE_DEG[order(match(Sly_VTE_DEG[,1], Sly_VTE_Ann[,2])),]

Sly_VTE_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Sly_VTE_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Sly_VTE_DEG$Ripening_Orange_DEG <- ifelse(is.na(Sly_VTE_DEG$Ripening_Orange_DEG) == TRUE, " ", "*")
Sly_VTE_DEG$Ripening_Red_DEG <- ifelse(is.na(Sly_VTE_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
Sly_VTE_DEG$ORHis_MG_DEG <- ifelse(is.na(Sly_VTE_DEG$ORHis_MG_DEG) == TRUE, " ", "*")
Sly_VTE_DEG$ORHis_Breaker_DEG <- ifelse(is.na(Sly_VTE_DEG$ORHis_Breaker_DEG) == TRUE, " ", "*")
row.names(Sly_VTE_DEG) <- Sly_VTE_DEG$Protein
Sly_VTE_DEG[1] <- NULL

rownames(Sly_VTE_FC) <- Sly_VTE_FC$Protein
Sly_VTE_FC[1] <- NULL

min(Sly_VTE_FC, na.rm = TRUE)
max(Sly_VTE_FC, na.rm = TRUE)
breaks_Sly_VTE=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Sly_VTE_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Sly_VTE,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_VTE_Annotation, annotation_colors=color_VTE, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "VTE",
         annotation_legend = TRUE,
         display_numbers = Sly_VTE_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(39, 48),
         gaps_col = c(3)
)


rownames(Sly_VTE_TPM) <- Sly_VTE_TPM$Protein
Sly_VTE_TPM[1] <- NULL

min(Sly_VTE_TPM, na.rm = TRUE)
max(Sly_VTE_TPM, na.rm = TRUE)
breaks_Sly_VTE_TPM=c(seq(0,100, length.out = 200))

Sly_VTE_TPM_norm <- normalize_df(Sly_VTE_TPM)

pheatmap(Sly_VTE_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Sly_VTE_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_VTE_Annotation, annotation_colors=color_VTE, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "VTE",
         annotation_legend = TRUE,
         #display_numbers = Sly_VTE_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(21,42)
)


#####Representing Heatmap for Cann

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Cann_VTE_Ann <- read.table("Funcional Groups/FigTree/Cann_VTE.txt",
                             sep = "\t", header = TRUE)

colnames(Cann_VTE_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Cann_VTE <- merge(Cann_VTE_Ann, Summary_Pepper_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Cann_VTE_TPM <- Cann_VTE %>% select(Protein, 4:17)
Cann_VTE_TPM <- Cann_VTE_TPM[!duplicated(Cann_VTE_TPM$Protein),]
Cann_VTE_TPM <- Cann_VTE_TPM[order(match(Cann_VTE_TPM[,1], Cann_VTE_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_VTE_FC <- Cann_VTE %>% select(Protein, 18:19)
Cann_VTE_FC <- Cann_VTE_FC[!duplicated(Cann_VTE_FC$Protein),]
Cann_VTE_FC <- Cann_VTE_FC[order(match(Cann_VTE_FC[,1], Cann_VTE_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_VTE_DEG <- Cann_VTE %>% select(Protein, 20:21)
Cann_VTE_DEG <- Cann_VTE_DEG[!duplicated(Cann_VTE_DEG$Protein),]
Cann_VTE_DEG <- Cann_VTE_DEG[order(match(Cann_VTE_DEG[,1], Cann_VTE_Ann[,2])),]

Cann_VTE_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Cann_VTE_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Cann_VTE_DEG$Ripening_Red_DEG <- ifelse(is.na(Cann_VTE_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
row.names(Cann_VTE_DEG) <- Cann_VTE_DEG$Protein
Cann_VTE_DEG[1] <- NULL

rownames(Cann_VTE_FC) <- Cann_VTE_FC$Protein
Cann_VTE_FC[1] <- NULL

min(Cann_VTE_FC, na.rm = TRUE)
max(Cann_VTE_FC, na.rm = TRUE)
breaks_Cann_VTE=c(seq(-4.5,0, length.out = 100), seq(0.01,6,length.out = 100))

pheatmap(Cann_VTE_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Cann_VTE,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_VTE_Annotation, annotation_colors=color_VTE, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Red"),
         cex = 1,
         angle_col=45,
         main = "VTE",
         annotation_legend = TRUE,
         display_numbers = Cann_VTE_DEG,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)


rownames(Cann_VTE_TPM) <- Cann_VTE_TPM$Protein
Cann_VTE_TPM[1] <- NULL

min(Cann_VTE_TPM, na.rm = TRUE)
max(Cann_VTE_TPM, na.rm = TRUE)
breaks_Cann_VTE_TPM=c(seq(0,100, length.out = 200))

Cann_VTE_TPM_norm <- normalize_df(Cann_VTE_TPM)

pheatmap(Cann_VTE_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Cann_VTE_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_VTE_Annotation, annotation_colors=color_VTE, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "VTE",
         annotation_legend = TRUE,
         #display_numbers = Cann_VTE_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(4,9)
)



#########################################################################################################################################
####################################  FBN  #############################################################################################
#########################################################################################################################################

FBN_Ath <- read.table("Funcional Groups/Ath_FBN.txt", header = TRUE)
FBN_Ath <- merge(FBN_Ath, Ath_Gene_Models, by.x="ID", by.y=1)
FBN_Ath_fasta <- Ath11_fasta[c(which(names(Ath11_fasta) %in% FBN_Ath$V2))]
FBN_Ath_2 <- FBN_Ath %>% select(V2, Name)
write.fasta(FBN_Ath_fasta, as.string=TRUE, names= names(FBN_Ath_fasta),
            file.out = "Funcional Groups/Ath_FBN.fasta")
rename.fasta(infile = "Funcional Groups/Ath_FBN.fasta", FBN_Ath_2, outfile = "Funcional Groups/Ath_FBN_2.fasta")

FBN_Niben_Candidates <- read.table("Funcional Groups/Niben_Candidates_FBN.txt")
FBN_Niben_Candidates <- FBN_Niben_Candidates[FBN_Niben_Candidates$V3>=35,] %>% select(V2)
FBN_Niben_Candidates <- unique(FBN_Niben_Candidates)
FBN_Niben_Candidates_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% FBN_Niben_Candidates$V2))]
write.fasta(FBN_Niben_Candidates_fasta, as.string=TRUE, names= names(FBN_Niben_Candidates_fasta),
            file.out = "Funcional Groups/FBN_Niben_Candidates_35.fasta")

FBN_Sly_Candidates <- read.table("Funcional Groups/Sly_Candidates_FBN.txt")
FBN_Sly_Candidates <- FBN_Sly_Candidates[FBN_Sly_Candidates$V3>=35,] %>% select(V2)
FBN_Sly_Candidates <- unique(FBN_Sly_Candidates)
FBN_Sly_Candidates_fasta <- Tomato_4.0_fasta[c(which(names(Tomato_4.0_fasta) %in% FBN_Sly_Candidates$V2))]
write.fasta(FBN_Sly_Candidates_fasta, as.string=TRUE, names= names(FBN_Sly_Candidates_fasta),
            file.out = "Funcional Groups/FBN_Sly_Candidates_35.fasta")

Niben_FBN_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_FBN_Ann$Name))]
write.fasta(Niben_FBN_fasta, as.string=TRUE, names= names(Niben_FBN_fasta),
            file.out = "Funcional Groups/Niben_FBN.fasta")
Niben_FBN_Ann_2 <- Niben_FBN_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_FBN.fasta", Niben_FBN_Ann, outfile = "Funcional Groups/Niben_FBN_2.fasta")

save_heatmaps("FBN", height = 6)

## Representing Heatmap

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Niben_FBN_Ann <- read.table("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan\RNAseq_Milan\Funcional Groups/FigTree/Niben_FBN.txt",
                                    sep = "\t", header = TRUE)

colnames(Niben_FBN_Ann) <- c("Name", "Protein")

Niben_FBN_Chloro <- read.delim("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan/Funcional Groups/Niben_FBN_2_summary.targetp2", 
                                       header=FALSE, comment.char="#")
colnames(Niben_FBN_Chloro) <- c("ID",	"Prediction",	"noTP",	"SP",	"mTP",	"cTP",	"luTP",	"CS-Position")
Niben_FBN_Chloro$Prediction <- factor(Niben_FBN_Chloro$Prediction, levels = c("cTP","mTP","noTP"))
Niben_FBN_Chloro <- merge(Niben_FBN_Ann, Niben_FBN_Chloro, by.x = "Protein", by.y = "ID")

Niben_FBN_Chloro <- Niben_FBN_Chloro[order(match(Niben_FBN_Chloro$Protein, Niben_FBN_Ann$Protein)),]
rownames(Niben_FBN_Chloro) <- Niben_FBN_Chloro$Protein
Niben_FBN_Chloro <- Niben_FBN_Chloro %>% select(Prediction)


## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_FBN <- merge(Niben_FBN_Ann, TPM_All_FC_Annotated, by="Name")

## We create the annotation for Modules

Niben_FBN_Modules <- which_Module_pwd30(Niben_FBN_Ann)

## We merge this annotation of Modules with the annotation of Chloroplast TP prediction

Niben_FBN_Modules_Chl <- merge(Niben_FBN_Modules, Niben_FBN_Chloro, by = 0)
Niben_FBN_Modules_Chl <- Niben_FBN_Modules_Chl[order(match(Niben_FBN_Modules_Chl$Row.names, Niben_FBN_Ann$Protein)),]
rownames(Niben_FBN_Modules_Chl) <- Niben_FBN_Modules_Chl$Row.names
Niben_FBN_Modules_Chl$Row.names <- NULL

## We select only the FC data from the different comparisons
Niben_FBN_TPM <- Niben_FBN %>% select(Protein, 4:51, 61:66)
Niben_FBN_TPM <- Niben_FBN_TPM[!duplicated(Niben_FBN_TPM$Protein),]
Niben_FBN_TPM <- Niben_FBN_TPM[order(match(Niben_FBN_TPM[,1], Niben_FBN_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_FBN_FC <- Niben_FBN %>% select(Protein, 71, 73:87)
Niben_FBN_FC <- Niben_FBN_FC[!duplicated(Niben_FBN_FC$Protein),]
Niben_FBN_FC <- Niben_FBN_FC[order(match(Niben_FBN_FC[,1], Niben_FBN_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_FBN_DEG <- Niben_FBN %>% select(Protein, 89, 91:105)
Niben_FBN_DEG <- Niben_FBN_DEG[!duplicated(Niben_FBN_DEG$Protein),]
Niben_FBN_DEG <- Niben_FBN_DEG[order(match(Niben_FBN_DEG[,1], Niben_FBN_Ann[,2])),]

Niben_FBN_DEG$pcrtB22h_vs_GFP22h_filtered <- ifelse(is.na(Niben_FBN_DEG$pcrtB22h_vs_GFP22h_filtered) == TRUE, " ", "*")
Niben_FBN_DEG$pcrtB25h_vs_GFP25h_filtered <- ifelse(is.na(Niben_FBN_DEG$pcrtB25h_vs_GFP25h_filtered) == TRUE, " ", "*")
Niben_FBN_DEG$pcrtBAA_vs_GFPAA_filtered <- ifelse(is.na(Niben_FBN_DEG$pcrtBAA_vs_GFPAA_filtered) == TRUE, " ", "*")
Niben_FBN_DEG$pcrtBBB_vs_GFPBB_filtered <- ifelse(is.na(Niben_FBN_DEG$pcrtBBB_vs_GFPBB_filtered) == TRUE, " ", "*")
Niben_FBN_DEG$pcrtBYY_vs_GFPYY_filtered <- ifelse(is.na(Niben_FBN_DEG$pcrtBYY_vs_GFPYY_filtered) == TRUE, " ", "*")
Niben_FBN_DEG$pcrtB28h_vs_GFP28h_filtered <- ifelse(is.na(Niben_FBN_DEG$pcrtB28h_vs_GFP28h_filtered) == TRUE, " ", "*")
Niben_FBN_DEG$pcrtB34h_vs_GFP34h_filtered <- ifelse(is.na(Niben_FBN_DEG$pcrtB34h_vs_GFP34h_filtered) == TRUE, " ", "*")
Niben_FBN_DEG$pcrtB37h_vs_GFP37h_filtered <- ifelse(is.na(Niben_FBN_DEG$pcrtB37h_vs_GFP37h_filtered) == TRUE, " ", "*")
Niben_FBN_DEG$pcrtBZZ_vs_GFPZZ_filtered <- ifelse(is.na(Niben_FBN_DEG$pcrtBZZ_vs_GFPZZ_filtered) == TRUE, " ", "*")
Niben_FBN_DEG$pcrtBCC_vs_GFPCC_filtered <- ifelse(is.na(Niben_FBN_DEG$pcrtBCC_vs_GFPCC_filtered) == TRUE, " ", "*")
Niben_FBN_DEG$pcrtB40h_vs_GFP40h_filtered <- ifelse(is.na(Niben_FBN_DEG$pcrtB40h_vs_GFP40h_filtered) == TRUE, " ", "*")
Niben_FBN_DEG$pcrtBDD_vs_GFPDD_filtered <- ifelse(is.na(Niben_FBN_DEG$pcrtBDD_vs_GFPDD_filtered) == TRUE, " ", "*")
Niben_FBN_DEG$pcrtB46h_vs_GFP46h_filtered <- ifelse(is.na(Niben_FBN_DEG$pcrtB46h_vs_GFP46h_filtered) == TRUE, " ", "*")
Niben_FBN_DEG$pcrtB56h_vs_GFP56h_filtered <- ifelse(is.na(Niben_FBN_DEG$pcrtB56h_vs_GFP56h_filtered) == TRUE, " ", "*")
Niben_FBN_DEG$pcrtBEE_vs_GFPEE_filtered <- ifelse(is.na(Niben_FBN_DEG$pcrtBEE_vs_GFPEE_filtered) == TRUE, " ", "*")
Niben_FBN_DEG$pcrtB96h_vs_GFP96h_filtered <- ifelse(is.na(Niben_FBN_DEG$pcrtB96h_vs_GFP96h_filtered) == TRUE, " ", "*")
row.names(Niben_FBN_DEG) <- Niben_FBN_DEG$Protein
Niben_FBN_DEG[1] <- NULL

Niben_FBN_DEG <- Niben_FBN_DEG %>%
  select(pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

row.names(Niben_FBN_FC) <- Niben_FBN_FC$Protein
Niben_FBN_FC[1] <- NULL

Niben_FBN_FC <- Niben_FBN_FC %>%
  select(pcrtB22h_vs_GFP22h_All, pcrtBAA_vs_GFPAA_All, pcrtB25h_vs_GFP25h_All, 
         pcrtBYY_vs_GFPYY_All, pcrtBBB_vs_GFPBB_All, pcrtB28h_vs_GFP28h_All, pcrtB34h_vs_GFP34h_All, pcrtBZZ_vs_GFPZZ_All, 
         pcrtB37h_vs_GFP37h_All, pcrtBCC_vs_GFPCC_All, pcrtB40h_vs_GFP40h_All, pcrtBDD_vs_GFPDD_All, pcrtB46h_vs_GFP46h_All, 
         pcrtBEE_vs_GFPEE_All, pcrtB56h_vs_GFP56h_All, pcrtB96h_vs_GFP96h_All)


min(Niben_FBN_FC, na.rm = TRUE)
max(Niben_FBN_FC, na.rm = TRUE)
breaks_FBN_FC=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Niben_FBN_FC, # %>% select(1:15), 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_FBN_FC,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_FBN_Annotation, annotation_colors=color_FBN, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h", "96h"),
         cex = 1,
         angle_col=45,
         main = "FBN",
         annotation_legend = TRUE,
         display_numbers = Niben_FBN_DEG,# %>% select(1:15),
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(84, 104),
         annotation_row = Niben_FBN_Modules_Chl,
         annotation_colors = module_Chl_colors,
         gaps_col = c(15)
)


rownames(Niben_FBN_TPM) <- Niben_FBN_TPM$Protein
Niben_FBN_TPM[1] <- NULL

min(Niben_FBN_TPM, na.rm = TRUE)
max(Niben_FBN_TPM, na.rm = TRUE)
breaks_Niben_FBN_TPM=c(seq(0,100, length.out = 100))

Niben_FBN_TPM_norm <- normalize_df(Niben_FBN_TPM)
Niben_FBN_TPM_norm_Cornell <- Niben_FBN_TPM %>%
  select(1:48) 
Niben_FBN_TPM_norm_Cornell <- normalize_df(Niben_FBN_TPM_norm_Cornell)


pheatmap(Niben_FBN_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Niben_FBN_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_FBN_Annotation, annotation_colors=color_FBN, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "FBN",
         annotation_legend = TRUE,
         #display_numbers = Niben_FBN_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         annotation_row = Niben_FBN_Modules,
         annotation_colors = module_colors,
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(24, 48, 51)
)



#####Representing Heatmap for Sly

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Sly_FBN_Ann <- read.table("Funcional Groups/FigTree/Sly_FBN.txt",
                           sep = "\t", header = TRUE)

colnames(Sly_FBN_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Sly_FBN <- merge(Sly_FBN_Ann, Summary_Yazdani_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Sly_FBN_TPM <- Sly_FBN %>% select(Protein, 4:66)
Sly_FBN_TPM <- Sly_FBN_TPM[!duplicated(Sly_FBN_TPM$Protein),]
Sly_FBN_TPM <- Sly_FBN_TPM[order(match(Sly_FBN_TPM[,1], Sly_FBN_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_FBN_FC <- Sly_FBN %>% select(Protein, 67:71)
Sly_FBN_FC <- Sly_FBN_FC[!duplicated(Sly_FBN_FC$Protein),]
Sly_FBN_FC <- Sly_FBN_FC[order(match(Sly_FBN_FC[,1], Sly_FBN_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_FBN_DEG <- Sly_FBN %>% select(Protein, 72:76)
Sly_FBN_DEG <- Sly_FBN_DEG[!duplicated(Sly_FBN_DEG$Protein),]
Sly_FBN_DEG <- Sly_FBN_DEG[order(match(Sly_FBN_DEG[,1], Sly_FBN_Ann[,2])),]

Sly_FBN_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Sly_FBN_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Sly_FBN_DEG$Ripening_Orange_DEG <- ifelse(is.na(Sly_FBN_DEG$Ripening_Orange_DEG) == TRUE, " ", "*")
Sly_FBN_DEG$Ripening_Red_DEG <- ifelse(is.na(Sly_FBN_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
Sly_FBN_DEG$ORHis_MG_DEG <- ifelse(is.na(Sly_FBN_DEG$ORHis_MG_DEG) == TRUE, " ", "*")
Sly_FBN_DEG$ORHis_Breaker_DEG <- ifelse(is.na(Sly_FBN_DEG$ORHis_Breaker_DEG) == TRUE, " ", "*")
row.names(Sly_FBN_DEG) <- Sly_FBN_DEG$Protein
Sly_FBN_DEG[1] <- NULL

rownames(Sly_FBN_FC) <- Sly_FBN_FC$Protein
Sly_FBN_FC[1] <- NULL

min(Sly_FBN_FC, na.rm = TRUE)
max(Sly_FBN_FC, na.rm = TRUE)
breaks_Sly_FBN=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Sly_FBN_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Sly_FBN,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_FBN_Annotation, annotation_colors=color_FBN, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "FBN",
         annotation_legend = TRUE,
         display_numbers = Sly_FBN_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(39, 48),
         gaps_col = c(3)
)


rownames(Sly_FBN_TPM) <- Sly_FBN_TPM$Protein
Sly_FBN_TPM[1] <- NULL

min(Sly_FBN_TPM, na.rm = TRUE)
max(Sly_FBN_TPM, na.rm = TRUE)
breaks_Sly_FBN_TPM=c(seq(0,100, length.out = 200))

Sly_FBN_TPM_norm <- normalize_df(Sly_FBN_TPM)

pheatmap(Sly_FBN_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Sly_FBN_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_FBN_Annotation, annotation_colors=color_FBN, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "FBN",
         annotation_legend = TRUE,
         #display_numbers = Sly_FBN_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(21,42)
)


## Comparativa 96h_SlyYaz_Ath

Ath_FBN_Ann <- read.table("Funcional Groups/Ath_FBN.txt", header = TRUE)

Ath_FBN_FC <- merge(Ath_FBN_Ann, Summary_Ath_FC, by.x=1, by.y = 0, all.x = TRUE) %>%
  dplyr::select(Name, log2FoldChange)
Ath_FBN_FC <- Ath_FBN_FC[order(match(Ath_FBN_FC[,1], Ath_FBN_Ann[,2])),]

Ath_FBN_DEG <- merge(Ath_FBN_Ann, Summary_Ath_DEGs, by.x=1, by.y = 1, all.x = TRUE) %>%
  dplyr::select(Name, log2FC)
Ath_FBN_DEG <- Ath_FBN_DEG[order(match(Ath_FBN_DEG[,1], Ath_FBN_Ann[,2])),]

Ath_FBN_DEG$log2FC <- ifelse(is.na(Ath_FBN_DEG$log2FC) == TRUE, " ", "*")
row.names(Ath_FBN_DEG) <- Ath_FBN_DEG$Name
Ath_FBN_DEG[1] <- NULL

rownames(Ath_FBN_FC) <- Ath_FBN_FC$Name
Ath_FBN_FC[1] <- NULL

min(Ath_FBN_FC, na.rm = TRUE)
max(Ath_FBN_FC, na.rm = TRUE)
breaks_Ath_FBN=c(seq(-4.5,0, length.out = 100), seq(0.01,4.5,length.out = 100))

pheatmap(Ath_FBN_FC, 
         border_color = "grey60",legend= TRUE,
         color=color3, 
         cluster_rows =FALSE,
         breaks = breaks_Stndr,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_FBN_Annotation, annotation_colors=color_FBN, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = "Senescence",
         cex = 1,
         angle_col=45,
         main = "FBN",
         annotation_legend = TRUE,
         display_numbers = Ath_FBN_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         gaps_row = c(2, 3, 5, 6,7, 8, 10, 11, 12, 13)
         #gaps_col = c(2,5)
)

pheatmap(Niben_FBN_FC %>% select(16), 
         border_color = "grey60",legend= TRUE,
         color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_FBN_FC,
         breaks = breaks_Stndr,
         #legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         #legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_FBN_Annotation, annotation_colors=color_FBN, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = "96h",
         cex = 1,
         angle_col=45,
         main = "FBN",
         annotation_legend = TRUE,
         display_numbers = Niben_FBN_DEG %>% select(16),
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(84, 104),
         #annotation_row = Niben_FBN_Modules_Chl,
         #annotation_colors = module_Chl_colors_2,
         gaps_row = c(3, 6, 9, 11, 12, 15, 18, 20, 22, 23)
)


pheatmap(Sly_FBN_FC %>% select(2,3), 
         border_color = "grey60",legend= TRUE,
         color=color3, 
         cluster_rows =FALSE,
         breaks = breaks_Stndr,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         #legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_FBN_Annotation, annotation_colors=color_FBN, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Orange", "Ripening_Red"),
         cex = 1,
         angle_col=45,
         main = "FBN",
         annotation_legend = TRUE,
         display_numbers = Sly_FBN_DEG %>% select(2,3),
         fontsize_number = 15,
         na_col = "darkgrey",
         gaps_row = c(1, 3, 5, 6, 7, 8, 10,11, 12, 13)
)



#########################################################################################################################################
####################################  Carotenoids  #############################################################################################
#########################################################################################################################################

Carotenoids_Ath <- read.table("Funcional Groups/Ath_Carotenoids.txt", header = TRUE)
Carotenoids_Ath <- merge(Carotenoids_Ath, Ath_Gene_Models, by.x="ID", by.y = 1)
Carotenoids_Ath_fasta <- Ath11_fasta[c(which(names(Ath11_fasta) %in% Carotenoids_Ath$V2))]
Carotenoids_Ath_2 <- Carotenoids_Ath %>% select(V2, Name)
write.fasta(Carotenoids_Ath_fasta, as.string=TRUE, names= names(Carotenoids_Ath_fasta),
            file.out = "Funcional Groups/Ath_Carotenoids.fasta")
rename.fasta(infile = "Funcional Groups/Ath_Carotenoids.fasta", Carotenoids_Ath_2, outfile = "Funcional Groups/Ath_Carotenoids_2.fasta")

Carotenoids_Niben_Candidates <- read.table("Funcional Groups/Niben_Candidates_Carotenoids.txt")
Carotenoids_Niben_Candidates <- Carotenoids_Niben_Candidates[Carotenoids_Niben_Candidates$V3>=35,] %>% select(V2)
Carotenoids_Niben_Candidates <- unique(Carotenoids_Niben_Candidates)
Carotenoids_Niben_Candidates_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Carotenoids_Niben_Candidates$V2))]
write.fasta(Carotenoids_Niben_Candidates_fasta, as.string=TRUE, names= names(Carotenoids_Niben_Candidates_fasta),
            file.out = "Funcional Groups/Carotenoids_Niben_Candidates_35.fasta")

Carotenoids_Sly_Candidates <- read.table("Funcional Groups/Sly_Candidates_Carotenoids.txt")
Carotenoids_Sly_Candidates <- Carotenoids_Sly_Candidates[Carotenoids_Sly_Candidates$V3>=35,] %>% select(V2)
Carotenoids_Sly_Candidates <- unique(Carotenoids_Sly_Candidates)
Carotenoids_Sly_Candidates_fasta <- Tomato_4.0_fasta[c(which(names(Tomato_4.0_fasta) %in% Carotenoids_Sly_Candidates$V2))]
write.fasta(Carotenoids_Sly_Candidates_fasta, as.string=TRUE, names= names(Carotenoids_Sly_Candidates_fasta),
            file.out = "Funcional Groups/Carotenoids_Sly_Candidates_35.fasta")

Niben_Carotenoids_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_Carotenoids_Ann$Name))]
write.fasta(Niben_Carotenoids_fasta, as.string=TRUE, names= names(Niben_Carotenoids_fasta),
            file.out = "Funcional Groups/Niben_Carotenoids.fasta")
Niben_Carotenoids_Ann_2 <- Niben_Carotenoids_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_Carotenoids.fasta", Niben_Carotenoids_Ann, outfile = "Funcional Groups/Niben_Carotenoids_2.fasta")

## Representing Heatmap

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Niben_Carotenoids_Ann <- read.table("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan\RNAseq_Milan\Funcional Groups/FigTree/Niben_Carotenoids.txt",
                              sep = "\t", header = TRUE)

colnames(Niben_Carotenoids_Ann) <- c("Name", "Protein")

Niben_Carotenoids_Chloro <- read.delim("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan/Funcional Groups/Niben_Carotenoids_2_summary.targetp2", 
                                 header=FALSE, comment.char="#")
colnames(Niben_Carotenoids_Chloro) <- c("ID",	"Prediction",	"noTP",	"SP",	"mTP",	"cTP",	"luTP",	"CS-Position")
Niben_Carotenoids_Chloro$Prediction <- factor(Niben_Carotenoids_Chloro$Prediction, levels = c("cTP","mTP","noTP"))
Niben_Carotenoids_Chloro <- merge(Niben_Carotenoids_Ann, Niben_Carotenoids_Chloro, by.x = "Protein", by.y = "ID")

Niben_Carotenoids_Chloro <- Niben_Carotenoids_Chloro[order(match(Niben_Carotenoids_Chloro$Protein, Niben_Carotenoids_Ann$Protein)),]
rownames(Niben_Carotenoids_Chloro) <- Niben_Carotenoids_Chloro$Protein
Niben_Carotenoids_Chloro <- Niben_Carotenoids_Chloro %>% select(Prediction)


## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_Carotenoids <- merge(Niben_Carotenoids_Ann, TPM_All_FC_Annotated, by="Name")

## We create the annotation for Modules

Niben_Carotenoids_Modules <- which_Module_pwd30(Niben_Carotenoids_Ann)

## We merge this annotation of Modules with the annotation of Chloroplast TP prediction

Niben_Carotenoids_Modules_Chl <- merge(Niben_Carotenoids_Modules, Niben_Carotenoids_Chloro, by = 0)
Niben_Carotenoids_Modules_Chl <- Niben_Carotenoids_Modules_Chl[order(match(Niben_Carotenoids_Modules_Chl$Row.names, Niben_Carotenoids_Ann$Protein)),]
rownames(Niben_Carotenoids_Modules_Chl) <- Niben_Carotenoids_Modules_Chl$Row.names
Niben_Carotenoids_Modules_Chl$Row.names <- NULL

## We select only the FC data from the different comparisons
Niben_Carotenoids_TPM <- Niben_Carotenoids %>% select(Protein, 4:51, 61:66)
Niben_Carotenoids_TPM <- Niben_Carotenoids_TPM[!duplicated(Niben_Carotenoids_TPM$Protein),]
Niben_Carotenoids_TPM <- Niben_Carotenoids_TPM[order(match(Niben_Carotenoids_TPM[,1], Niben_Carotenoids_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_Carotenoids_FC <- Niben_Carotenoids %>% select(Protein, 71, 73:87)
Niben_Carotenoids_FC <- Niben_Carotenoids_FC[!duplicated(Niben_Carotenoids_FC$Protein),]
Niben_Carotenoids_FC <- Niben_Carotenoids_FC[order(match(Niben_Carotenoids_FC[,1], Niben_Carotenoids_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_Carotenoids_DEG <- Niben_Carotenoids %>% select(Protein, 89, 91:105)
Niben_Carotenoids_DEG <- Niben_Carotenoids_DEG[!duplicated(Niben_Carotenoids_DEG$Protein),]
Niben_Carotenoids_DEG <- Niben_Carotenoids_DEG[order(match(Niben_Carotenoids_DEG[,1], Niben_Carotenoids_Ann[,2])),]

Niben_Carotenoids_DEG$pcrtB22h_vs_GFP22h_filtered <- ifelse(is.na(Niben_Carotenoids_DEG$pcrtB22h_vs_GFP22h_filtered) == TRUE, " ", "*")
Niben_Carotenoids_DEG$pcrtB25h_vs_GFP25h_filtered <- ifelse(is.na(Niben_Carotenoids_DEG$pcrtB25h_vs_GFP25h_filtered) == TRUE, " ", "*")
Niben_Carotenoids_DEG$pcrtBAA_vs_GFPAA_filtered <- ifelse(is.na(Niben_Carotenoids_DEG$pcrtBAA_vs_GFPAA_filtered) == TRUE, " ", "*")
Niben_Carotenoids_DEG$pcrtBBB_vs_GFPBB_filtered <- ifelse(is.na(Niben_Carotenoids_DEG$pcrtBBB_vs_GFPBB_filtered) == TRUE, " ", "*")
Niben_Carotenoids_DEG$pcrtBYY_vs_GFPYY_filtered <- ifelse(is.na(Niben_Carotenoids_DEG$pcrtBYY_vs_GFPYY_filtered) == TRUE, " ", "*")
Niben_Carotenoids_DEG$pcrtB28h_vs_GFP28h_filtered <- ifelse(is.na(Niben_Carotenoids_DEG$pcrtB28h_vs_GFP28h_filtered) == TRUE, " ", "*")
Niben_Carotenoids_DEG$pcrtB34h_vs_GFP34h_filtered <- ifelse(is.na(Niben_Carotenoids_DEG$pcrtB34h_vs_GFP34h_filtered) == TRUE, " ", "*")
Niben_Carotenoids_DEG$pcrtB37h_vs_GFP37h_filtered <- ifelse(is.na(Niben_Carotenoids_DEG$pcrtB37h_vs_GFP37h_filtered) == TRUE, " ", "*")
Niben_Carotenoids_DEG$pcrtBZZ_vs_GFPZZ_filtered <- ifelse(is.na(Niben_Carotenoids_DEG$pcrtBZZ_vs_GFPZZ_filtered) == TRUE, " ", "*")
Niben_Carotenoids_DEG$pcrtBCC_vs_GFPCC_filtered <- ifelse(is.na(Niben_Carotenoids_DEG$pcrtBCC_vs_GFPCC_filtered) == TRUE, " ", "*")
Niben_Carotenoids_DEG$pcrtB40h_vs_GFP40h_filtered <- ifelse(is.na(Niben_Carotenoids_DEG$pcrtB40h_vs_GFP40h_filtered) == TRUE, " ", "*")
Niben_Carotenoids_DEG$pcrtBDD_vs_GFPDD_filtered <- ifelse(is.na(Niben_Carotenoids_DEG$pcrtBDD_vs_GFPDD_filtered) == TRUE, " ", "*")
Niben_Carotenoids_DEG$pcrtB46h_vs_GFP46h_filtered <- ifelse(is.na(Niben_Carotenoids_DEG$pcrtB46h_vs_GFP46h_filtered) == TRUE, " ", "*")
Niben_Carotenoids_DEG$pcrtB56h_vs_GFP56h_filtered <- ifelse(is.na(Niben_Carotenoids_DEG$pcrtB56h_vs_GFP56h_filtered) == TRUE, " ", "*")
Niben_Carotenoids_DEG$pcrtBEE_vs_GFPEE_filtered <- ifelse(is.na(Niben_Carotenoids_DEG$pcrtBEE_vs_GFPEE_filtered) == TRUE, " ", "*")
Niben_Carotenoids_DEG$pcrtB96h_vs_GFP96h_filtered <- ifelse(is.na(Niben_Carotenoids_DEG$pcrtB96h_vs_GFP96h_filtered) == TRUE, " ", "*")
row.names(Niben_Carotenoids_DEG) <- Niben_Carotenoids_DEG$Protein
Niben_Carotenoids_DEG[1] <- NULL

Niben_Carotenoids_DEG <- Niben_Carotenoids_DEG %>%
  select(pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

row.names(Niben_Carotenoids_FC) <- Niben_Carotenoids_FC$Protein
Niben_Carotenoids_FC[1] <- NULL

Niben_Carotenoids_FC <- Niben_Carotenoids_FC %>%
  select(pcrtB22h_vs_GFP22h_All, pcrtBAA_vs_GFPAA_All, pcrtB25h_vs_GFP25h_All, 
         pcrtBYY_vs_GFPYY_All, pcrtBBB_vs_GFPBB_All, pcrtB28h_vs_GFP28h_All, pcrtB34h_vs_GFP34h_All, pcrtBZZ_vs_GFPZZ_All, 
         pcrtB37h_vs_GFP37h_All, pcrtBCC_vs_GFPCC_All, pcrtB40h_vs_GFP40h_All, pcrtBDD_vs_GFPDD_All, pcrtB46h_vs_GFP46h_All, 
         pcrtBEE_vs_GFPEE_All, pcrtB56h_vs_GFP56h_All, pcrtB96h_vs_GFP96h_All)


min(Niben_Carotenoids_FC, na.rm = TRUE)
max(Niben_Carotenoids_FC, na.rm = TRUE)
breaks_Carotenoids_FC=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Niben_Carotenoids_FC, # %>% select(1:15), 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Carotenoids_FC,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_Carotenoids_Annotation, annotation_colors=color_Carotenoids, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h", "96h"),
         cex = 1,
         angle_col=45,
         main = "Carotenoids",
         annotation_legend = TRUE,
         display_numbers = Niben_Carotenoids_DEG,# %>% select(1:15),
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(84, 104),
         annotation_row = Niben_Carotenoids_Modules_Chl,
         annotation_colors = module_Chl_colors,
         gaps_col = c(15)
)


rownames(Niben_Carotenoids_TPM) <- Niben_Carotenoids_TPM$Protein
Niben_Carotenoids_TPM[1] <- NULL

min(Niben_Carotenoids_TPM, na.rm = TRUE)
max(Niben_Carotenoids_TPM, na.rm = TRUE)
breaks_Niben_Carotenoids_TPM=c(seq(0,100, length.out = 100))

Niben_Carotenoids_TPM_norm <- normalize_df(Niben_Carotenoids_TPM)
Niben_Carotenoids_TPM_norm_Cornell <- Niben_Carotenoids_TPM %>%
  select(1:48) 
Niben_Carotenoids_TPM_norm_Cornell <- normalize_df(Niben_Carotenoids_TPM_norm_Cornell)


pheatmap(Niben_Carotenoids_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Niben_Carotenoids_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_Carotenoids_Annotation, annotation_colors=color_Carotenoids, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Carotenoids",
         annotation_legend = TRUE,
         #display_numbers = Niben_Carotenoids_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         annotation_row = Niben_Carotenoids_Modules,
         annotation_colors = module_colors,
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(24, 48, 51)
)



#####Representing Heatmap for Sly

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Sly_Carotenoids_Ann <- read.table("Funcional Groups/FigTree/Sly_Carotenoids.txt",
                          sep = "\t", header = TRUE)

colnames(Sly_Carotenoids_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Sly_Carotenoids <- merge(Sly_Carotenoids_Ann, Summary_Yazdani_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Sly_Carotenoids_TPM <- Sly_Carotenoids %>% select(Protein, 4:66)
Sly_Carotenoids_TPM <- Sly_Carotenoids_TPM[!duplicated(Sly_Carotenoids_TPM$Protein),]
Sly_Carotenoids_TPM <- Sly_Carotenoids_TPM[order(match(Sly_Carotenoids_TPM[,1], Sly_Carotenoids_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_Carotenoids_FC <- Sly_Carotenoids %>% select(Protein, 67:71)
Sly_Carotenoids_FC <- Sly_Carotenoids_FC[!duplicated(Sly_Carotenoids_FC$Protein),]
Sly_Carotenoids_FC <- Sly_Carotenoids_FC[order(match(Sly_Carotenoids_FC[,1], Sly_Carotenoids_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_Carotenoids_DEG <- Sly_Carotenoids %>% select(Protein, 72:76)
Sly_Carotenoids_DEG <- Sly_Carotenoids_DEG[!duplicated(Sly_Carotenoids_DEG$Protein),]
Sly_Carotenoids_DEG <- Sly_Carotenoids_DEG[order(match(Sly_Carotenoids_DEG[,1], Sly_Carotenoids_Ann[,2])),]

Sly_Carotenoids_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Sly_Carotenoids_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Sly_Carotenoids_DEG$Ripening_Orange_DEG <- ifelse(is.na(Sly_Carotenoids_DEG$Ripening_Orange_DEG) == TRUE, " ", "*")
Sly_Carotenoids_DEG$Ripening_Red_DEG <- ifelse(is.na(Sly_Carotenoids_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
Sly_Carotenoids_DEG$ORHis_MG_DEG <- ifelse(is.na(Sly_Carotenoids_DEG$ORHis_MG_DEG) == TRUE, " ", "*")
Sly_Carotenoids_DEG$ORHis_Breaker_DEG <- ifelse(is.na(Sly_Carotenoids_DEG$ORHis_Breaker_DEG) == TRUE, " ", "*")
row.names(Sly_Carotenoids_DEG) <- Sly_Carotenoids_DEG$Protein
Sly_Carotenoids_DEG[1] <- NULL

rownames(Sly_Carotenoids_FC) <- Sly_Carotenoids_FC$Protein
Sly_Carotenoids_FC[1] <- NULL

min(Sly_Carotenoids_FC, na.rm = TRUE)
max(Sly_Carotenoids_FC, na.rm = TRUE)
breaks_Sly_Carotenoids=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Sly_Carotenoids_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Sly_Carotenoids,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_Carotenoids_Annotation, annotation_colors=color_Carotenoids, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Carotenoids",
         annotation_legend = TRUE,
         display_numbers = Sly_Carotenoids_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(39, 48),
         gaps_col = c(3)
)


rownames(Sly_Carotenoids_TPM) <- Sly_Carotenoids_TPM$Protein
Sly_Carotenoids_TPM[1] <- NULL

min(Sly_Carotenoids_TPM, na.rm = TRUE)
max(Sly_Carotenoids_TPM, na.rm = TRUE)
breaks_Sly_Carotenoids_TPM=c(seq(0,100, length.out = 200))

Sly_Carotenoids_TPM_norm <- normalize_df(Sly_Carotenoids_TPM)

pheatmap(Sly_Carotenoids_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Sly_Carotenoids_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_Carotenoids_Annotation, annotation_colors=color_Carotenoids, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Carotenoids",
         annotation_legend = TRUE,
         #display_numbers = Sly_Carotenoids_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(21,42)
)


#####Representing Heatmap for Cann

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Cann_Carotenoids_Ann <- read.table("Funcional Groups/FigTree/Cann_Carotenoids.txt",
                                  sep = "\t", header = TRUE)

colnames(Cann_Carotenoids_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Cann_Carotenoids <- merge(Cann_Carotenoids_Ann, Summary_Pepper_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Cann_Carotenoids_TPM <- Cann_Carotenoids %>% select(Protein, 4:17)
Cann_Carotenoids_TPM <- Cann_Carotenoids_TPM[!duplicated(Cann_Carotenoids_TPM$Protein),]
Cann_Carotenoids_TPM <- Cann_Carotenoids_TPM[order(match(Cann_Carotenoids_TPM[,1], Cann_Carotenoids_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_Carotenoids_FC <- Cann_Carotenoids %>% select(Protein, 18:19)
Cann_Carotenoids_FC <- Cann_Carotenoids_FC[!duplicated(Cann_Carotenoids_FC$Protein),]
Cann_Carotenoids_FC <- Cann_Carotenoids_FC[order(match(Cann_Carotenoids_FC[,1], Cann_Carotenoids_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_Carotenoids_DEG <- Cann_Carotenoids %>% select(Protein, 20:21)
Cann_Carotenoids_DEG <- Cann_Carotenoids_DEG[!duplicated(Cann_Carotenoids_DEG$Protein),]
Cann_Carotenoids_DEG <- Cann_Carotenoids_DEG[order(match(Cann_Carotenoids_DEG[,1], Cann_Carotenoids_Ann[,2])),]

Cann_Carotenoids_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Cann_Carotenoids_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Cann_Carotenoids_DEG$Ripening_Red_DEG <- ifelse(is.na(Cann_Carotenoids_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
row.names(Cann_Carotenoids_DEG) <- Cann_Carotenoids_DEG$Protein
Cann_Carotenoids_DEG[1] <- NULL

rownames(Cann_Carotenoids_FC) <- Cann_Carotenoids_FC$Protein
Cann_Carotenoids_FC[1] <- NULL

min(Cann_Carotenoids_FC, na.rm = TRUE)
max(Cann_Carotenoids_FC, na.rm = TRUE)
breaks_Cann_Carotenoids=c(seq(-4.5,0, length.out = 100), seq(0.01,6,length.out = 100))

pheatmap(Cann_Carotenoids_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Cann_Carotenoids,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_Carotenoids_Annotation, annotation_colors=color_Carotenoids, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Red"),
         cex = 1,
         angle_col=45,
         main = "Carotenoids",
         annotation_legend = TRUE,
         display_numbers = Cann_Carotenoids_DEG,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)


rownames(Cann_Carotenoids_TPM) <- Cann_Carotenoids_TPM$Protein
Cann_Carotenoids_TPM[1] <- NULL

min(Cann_Carotenoids_TPM, na.rm = TRUE)
max(Cann_Carotenoids_TPM, na.rm = TRUE)
breaks_Cann_Carotenoids_TPM=c(seq(0,100, length.out = 200))

Cann_Carotenoids_TPM_norm <- normalize_df(Cann_Carotenoids_TPM)

pheatmap(Cann_Carotenoids_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Cann_Carotenoids_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_Carotenoids_Annotation, annotation_colors=color_Carotenoids, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Carotenoids",
         annotation_legend = TRUE,
         #display_numbers = Cann_Carotenoids_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(4,9)
)


## Comparativa 96h_SlyYaz_Ath

Ath_Carotenoids_Ann <- read.table("Funcional Groups/Ath_Carotenoids.txt", header = TRUE)

Ath_Carotenoids_FC <- merge(Ath_Carotenoids_Ann, Summary_Ath_FC, by.x=1, by.y = 0) %>%
  dplyr::select(Name, log2FoldChange)
Ath_Carotenoids_FC <- Ath_Carotenoids_FC[order(match(Ath_Carotenoids_FC[,1], Ath_Carotenoids_Ann[,2])),]

Ath_Carotenoids_DEG <- merge(Ath_Carotenoids_Ann, Summary_Ath_DEGs, by.x=1, by.y = 1, all.x = TRUE) %>%
  dplyr::select(Name, log2FC)
Ath_Carotenoids_DEG <- Ath_Carotenoids_DEG[order(match(Ath_Carotenoids_DEG[,1], Ath_Carotenoids_Ann[,2])),]

Ath_Carotenoids_DEG$log2FC <- ifelse(is.na(Ath_Carotenoids_DEG$log2FC) == TRUE, " ", "*")
row.names(Ath_Carotenoids_DEG) <- Ath_Carotenoids_DEG$Name
Ath_Carotenoids_DEG[1] <- NULL

rownames(Ath_Carotenoids_FC) <- Ath_Carotenoids_FC$Name
Ath_Carotenoids_FC[1] <- NULL

min(Ath_Carotenoids_FC, na.rm = TRUE)
max(Ath_Carotenoids_FC, na.rm = TRUE)
breaks_Ath_Carotenoids=c(seq(-4.5,0, length.out = 100), seq(0.01,4.5,length.out = 100))

pheatmap(Ath_Carotenoids_FC, 
         border_color = "grey60",legend= TRUE,
         color=color3, 
         cluster_rows =FALSE,
         breaks = breaks_Stndr,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_Carotenoids_Annotation, annotation_colors=color_Carotenoids, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = "Senescence",
         cex = 1,
         angle_col=45,
         main = "Carotenoids",
         #annotation_legend = TRUE,
         display_numbers = Ath_Carotenoids_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         gaps_row = c(1, 2, 6, 7, 8, 10, 13, 14,15),
         #gaps_col = c(2,5)
)

pheatmap(Niben_Carotenoids_FC %>% select(16), 
         border_color = "grey60",legend= TRUE,
         color=color3, 
         cluster_rows =FALSE,
         breaks = breaks_Stndr,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         #legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_Carotenoids_Annotation, annotation_colors=color_Carotenoids, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = "96h",
         cex = 1,
         angle_col=45,
         main = "Carotenoids",
         annotation_legend = TRUE,
         display_numbers = Niben_Carotenoids_DEG %>% select(16),
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(84, 104),
         #annotation_row = Niben_Carotenoids_Modules_Chl,
         #annotation_colors = module_Chl_colors,
         gaps_row = c(6, 8, 17, 18, 20, 24, 30, 32, 34)
)

pheatmap(Sly_Carotenoids_FC %>% select(2,3), 
         border_color = "grey60",legend= TRUE,
         color=color3, 
         cluster_rows =FALSE,
         breaks = breaks_Stndr,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         #legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_Carotenoids_Annotation, annotation_colors=color_Carotenoids, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Orange", "Ripening_Red"),
         cex = 1,
         angle_col=45,
         main = "Carotenoids",
         annotation_legend = TRUE,
         display_numbers = Sly_Carotenoids_DEG %>% select(2,3),
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(39, 48),
         gaps_row = c(2, 3, 7, 10, 11, 13,17, 18, 19)
)


#########################################################################################################################################
####################################  MEP  #############################################################################################
#########################################################################################################################################

MEP_Ath <- read.table("Funcional Groups/Ath_MEP.txt", header = TRUE)
MEP_Ath <- merge(MEP_Ath, Ath_Gene_Models, by.x="ID", by.y=1)
MEP_Ath_fasta <- Ath11_fasta[c(which(names(Ath11_fasta) %in% MEP_Ath$V2))]
MEP_Ath_2 <- MEP_Ath %>% select(V2, Name)
write.fasta(MEP_Ath_fasta, as.string=TRUE, names= names(MEP_Ath_fasta),
            file.out = "Funcional Groups/Ath_MEP.fasta")
rename.fasta(infile = "Funcional Groups/Ath_MEP.fasta", MEP_Ath_2, outfile = "Funcional Groups/Ath_MEP_2.fasta")

MEP_Niben_Candidates <- read.table("Funcional Groups/Niben_Candidates_MEP.txt")
MEP_Niben_Candidates <- MEP_Niben_Candidates[MEP_Niben_Candidates$V3>=35,] %>% select(V2)
MEP_Niben_Candidates <- unique(MEP_Niben_Candidates)
MEP_Niben_Candidates_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% MEP_Niben_Candidates$V2))]
write.fasta(MEP_Niben_Candidates_fasta, as.string=TRUE, names= names(MEP_Niben_Candidates_fasta),
            file.out = "Funcional Groups/MEP_Niben_Candidates_35.fasta")

MEP_Sly_Candidates <- read.table("Funcional Groups/Sly_Candidates_MEP.txt")
MEP_Sly_Candidates <- MEP_Sly_Candidates[MEP_Sly_Candidates$V3>=35,] %>% select(V2)
MEP_Sly_Candidates <- unique(MEP_Sly_Candidates)
MEP_Sly_Candidates_fasta <- Tomato_4.0_fasta[c(which(names(Tomato_4.0_fasta) %in% MEP_Sly_Candidates$V2))]
write.fasta(MEP_Sly_Candidates_fasta, as.string=TRUE, names= names(MEP_Sly_Candidates_fasta),
            file.out = "Funcional Groups/MEP_Sly_Candidates_35.fasta")

Niben_MEP_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_MEP_Ann$Name))]
write.fasta(Niben_MEP_fasta, as.string=TRUE, names= names(Niben_MEP_fasta),
            file.out = "Funcional Groups/Niben_MEP.fasta")
Niben_MEP_Ann_2 <- Niben_MEP_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_MEP.fasta", Niben_MEP_Ann, outfile = "Funcional Groups/Niben_MEP_2.fasta")

save_heatmaps("MEP", height = 6)

## Representing Heatmap

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Niben_MEP_Ann <- read.table("Funcional Groups/FigTree/Niben_MEP.txt",
                                    sep = "\t", header = TRUE)

colnames(Niben_MEP_Ann) <- c("Name", "Protein")

Niben_MEP_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_MEP_Ann$Name))]
write.fasta(Niben_MEP_fasta, as.string=TRUE, names= names(Niben_MEP_fasta),
            file.out = "Funcional Groups/Niben_MEP.fasta")
Niben_MEP_Ann_2 <- Niben_MEP_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_MEP.fasta", Niben_MEP_Ann, outfile = "Funcional Groups/Niben_MEP_2.fasta")

Niben_MEP_Chloro <- read.delim("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan/Funcional Groups/Niben_MEP_2_summary.targetp2", 
                                       header=FALSE, comment.char="#")
colnames(Niben_MEP_Chloro) <- c("ID",	"Prediction",	"noTP",	"SP",	"mTP",	"cTP",	"luTP",	"CS-Position")
Niben_MEP_Chloro$Prediction <- factor(Niben_MEP_Chloro$Prediction, levels = c("cTP","mTP","noTP"))
Niben_MEP_Chloro <- merge(Niben_MEP_Ann, Niben_MEP_Chloro, by.x = "Protein", by.y = "ID")

Niben_MEP_Chloro <- Niben_MEP_Chloro[order(match(Niben_MEP_Chloro$Protein, Niben_MEP_Ann$Protein)),]
rownames(Niben_MEP_Chloro) <- Niben_MEP_Chloro$Protein
Niben_MEP_Chloro <- Niben_MEP_Chloro %>% select(Prediction)


## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_MEP <- merge(Niben_MEP_Ann, TPM_All_FC_Annotated, by="Name")

## We create the annotation for Modules

Niben_MEP_Modules <- which_Module_pwd12(Niben_MEP_Ann)

## We merge this annotation of Modules with the annotation of Chloroplast TP prediction

Niben_MEP_Modules_Chl <- merge(Niben_MEP_Modules, Niben_MEP_Chloro, by = 0)
Niben_MEP_Modules_Chl <- Niben_MEP_Modules_Chl[order(match(Niben_MEP_Modules_Chl$Row.names, Niben_MEP_Ann$Protein)),]
rownames(Niben_MEP_Modules_Chl) <- Niben_MEP_Modules_Chl$Row.names
Niben_MEP_Modules_Chl$Row.names <- NULL

## We select only the FC data from the different comparisons
Niben_MEP_TPM <- Niben_MEP %>% select(Protein, 4:51, 61:66)
Niben_MEP_TPM <- Niben_MEP_TPM[!duplicated(Niben_MEP_TPM$Protein),]
Niben_MEP_TPM <- Niben_MEP_TPM[order(match(Niben_MEP_TPM[,1], Niben_MEP_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_MEP_FC <- Niben_MEP %>% select(Protein, 71, 73:87)
Niben_MEP_FC <- Niben_MEP_FC[!duplicated(Niben_MEP_FC$Protein),]
Niben_MEP_FC <- Niben_MEP_FC[order(match(Niben_MEP_FC[,1], Niben_MEP_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_MEP_DEG <- Niben_MEP %>% select(Protein, 89, 91:105)
Niben_MEP_DEG <- Niben_MEP_DEG[!duplicated(Niben_MEP_DEG$Protein),]
Niben_MEP_DEG <- Niben_MEP_DEG[order(match(Niben_MEP_DEG[,1], Niben_MEP_Ann[,2])),]

Niben_MEP_DEG$pcrtB22h_vs_GFP22h_filtered <- ifelse(is.na(Niben_MEP_DEG$pcrtB22h_vs_GFP22h_filtered) == TRUE, " ", "*")
Niben_MEP_DEG$pcrtB25h_vs_GFP25h_filtered <- ifelse(is.na(Niben_MEP_DEG$pcrtB25h_vs_GFP25h_filtered) == TRUE, " ", "*")
Niben_MEP_DEG$pcrtBAA_vs_GFPAA_filtered <- ifelse(is.na(Niben_MEP_DEG$pcrtBAA_vs_GFPAA_filtered) == TRUE, " ", "*")
Niben_MEP_DEG$pcrtBBB_vs_GFPBB_filtered <- ifelse(is.na(Niben_MEP_DEG$pcrtBBB_vs_GFPBB_filtered) == TRUE, " ", "*")
Niben_MEP_DEG$pcrtBYY_vs_GFPYY_filtered <- ifelse(is.na(Niben_MEP_DEG$pcrtBYY_vs_GFPYY_filtered) == TRUE, " ", "*")
Niben_MEP_DEG$pcrtB28h_vs_GFP28h_filtered <- ifelse(is.na(Niben_MEP_DEG$pcrtB28h_vs_GFP28h_filtered) == TRUE, " ", "*")
Niben_MEP_DEG$pcrtB34h_vs_GFP34h_filtered <- ifelse(is.na(Niben_MEP_DEG$pcrtB34h_vs_GFP34h_filtered) == TRUE, " ", "*")
Niben_MEP_DEG$pcrtB37h_vs_GFP37h_filtered <- ifelse(is.na(Niben_MEP_DEG$pcrtB37h_vs_GFP37h_filtered) == TRUE, " ", "*")
Niben_MEP_DEG$pcrtBZZ_vs_GFPZZ_filtered <- ifelse(is.na(Niben_MEP_DEG$pcrtBZZ_vs_GFPZZ_filtered) == TRUE, " ", "*")
Niben_MEP_DEG$pcrtBCC_vs_GFPCC_filtered <- ifelse(is.na(Niben_MEP_DEG$pcrtBCC_vs_GFPCC_filtered) == TRUE, " ", "*")
Niben_MEP_DEG$pcrtB40h_vs_GFP40h_filtered <- ifelse(is.na(Niben_MEP_DEG$pcrtB40h_vs_GFP40h_filtered) == TRUE, " ", "*")
Niben_MEP_DEG$pcrtBDD_vs_GFPDD_filtered <- ifelse(is.na(Niben_MEP_DEG$pcrtBDD_vs_GFPDD_filtered) == TRUE, " ", "*")
Niben_MEP_DEG$pcrtB46h_vs_GFP46h_filtered <- ifelse(is.na(Niben_MEP_DEG$pcrtB46h_vs_GFP46h_filtered) == TRUE, " ", "*")
Niben_MEP_DEG$pcrtB56h_vs_GFP56h_filtered <- ifelse(is.na(Niben_MEP_DEG$pcrtB56h_vs_GFP56h_filtered) == TRUE, " ", "*")
Niben_MEP_DEG$pcrtBEE_vs_GFPEE_filtered <- ifelse(is.na(Niben_MEP_DEG$pcrtBEE_vs_GFPEE_filtered) == TRUE, " ", "*")
Niben_MEP_DEG$pcrtB96h_vs_GFP96h_filtered <- ifelse(is.na(Niben_MEP_DEG$pcrtB96h_vs_GFP96h_filtered) == TRUE, " ", "*")
row.names(Niben_MEP_DEG) <- Niben_MEP_DEG$Protein
Niben_MEP_DEG[1] <- NULL

Niben_MEP_DEG <- Niben_MEP_DEG %>%
  select(pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

row.names(Niben_MEP_FC) <- Niben_MEP_FC$Protein
Niben_MEP_FC[1] <- NULL

Niben_MEP_FC <- Niben_MEP_FC %>%
  select(pcrtB22h_vs_GFP22h_All, pcrtBAA_vs_GFPAA_All, pcrtB25h_vs_GFP25h_All, 
         pcrtBYY_vs_GFPYY_All, pcrtBBB_vs_GFPBB_All, pcrtB28h_vs_GFP28h_All, pcrtB34h_vs_GFP34h_All, pcrtBZZ_vs_GFPZZ_All, 
         pcrtB37h_vs_GFP37h_All, pcrtBCC_vs_GFPCC_All, pcrtB40h_vs_GFP40h_All, pcrtBDD_vs_GFPDD_All, pcrtB46h_vs_GFP46h_All, 
         pcrtBEE_vs_GFPEE_All, pcrtB56h_vs_GFP56h_All, pcrtB96h_vs_GFP96h_All)


min(Niben_MEP_FC, na.rm = TRUE)
max(Niben_MEP_FC, na.rm = TRUE)
breaks_MEP_FC=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Niben_MEP_FC %>% select(1:15), 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_MEP_FC,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_MEP_Annotation, annotation_colors=color_MEP, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h"),#, "96h"),
         cex = 1,
         angle_col=45,
         main = "MEP",
         annotation_legend = TRUE,
         display_numbers = Niben_MEP_DEG %>% select(1:15),
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(84, 104),
         annotation_row = Niben_MEP_Modules_Chl,
         annotation_colors = module_Chl_colors_2
         #gaps_col = c(15)
)


rownames(Niben_MEP_TPM) <- Niben_MEP_TPM$Protein
Niben_MEP_TPM[1] <- NULL

min(Niben_MEP_TPM, na.rm = TRUE)
max(Niben_MEP_TPM, na.rm = TRUE)
breaks_Niben_MEP_TPM=c(seq(0,100, length.out = 100))

Niben_MEP_TPM_norm <- normalize_df(Niben_MEP_TPM)
Niben_MEP_TPM_norm_Cornell <- Niben_MEP_TPM %>%
  select(1:48) 
Niben_MEP_TPM_norm_Cornell <- normalize_df(Niben_MEP_TPM_norm_Cornell)


pheatmap(Niben_MEP_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Niben_MEP_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_MEP_Annotation, annotation_colors=color_MEP, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "MEP",
         annotation_legend = TRUE,
         #display_numbers = Niben_MEP_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         annotation_row = Niben_MEP_Modules,
         annotation_colors = module_colors,
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(24, 48, 51)
)



#####Representing Heatmap for Sly

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Sly_MEP_Ann <- read.table("Funcional Groups/FigTree/Sly_MEP.txt",
                                  sep = "\t", header = TRUE)

colnames(Sly_MEP_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Sly_MEP <- merge(Sly_MEP_Ann, Summary_Yazdani_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Sly_MEP_TPM <- Sly_MEP %>% select(Protein, 4:66)
Sly_MEP_TPM <- Sly_MEP_TPM[!duplicated(Sly_MEP_TPM$Protein),]
Sly_MEP_TPM <- Sly_MEP_TPM[order(match(Sly_MEP_TPM[,1], Sly_MEP_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_MEP_FC <- Sly_MEP %>% select(Protein, 67:71)
Sly_MEP_FC <- Sly_MEP_FC[!duplicated(Sly_MEP_FC$Protein),]
Sly_MEP_FC <- Sly_MEP_FC[order(match(Sly_MEP_FC[,1], Sly_MEP_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_MEP_DEG <- Sly_MEP %>% select(Protein, 72:76)
Sly_MEP_DEG <- Sly_MEP_DEG[!duplicated(Sly_MEP_DEG$Protein),]
Sly_MEP_DEG <- Sly_MEP_DEG[order(match(Sly_MEP_DEG[,1], Sly_MEP_Ann[,2])),]

Sly_MEP_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Sly_MEP_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Sly_MEP_DEG$Ripening_Orange_DEG <- ifelse(is.na(Sly_MEP_DEG$Ripening_Orange_DEG) == TRUE, " ", "*")
Sly_MEP_DEG$Ripening_Red_DEG <- ifelse(is.na(Sly_MEP_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
Sly_MEP_DEG$ORHis_MG_DEG <- ifelse(is.na(Sly_MEP_DEG$ORHis_MG_DEG) == TRUE, " ", "*")
Sly_MEP_DEG$ORHis_Breaker_DEG <- ifelse(is.na(Sly_MEP_DEG$ORHis_Breaker_DEG) == TRUE, " ", "*")
row.names(Sly_MEP_DEG) <- Sly_MEP_DEG$Protein
Sly_MEP_DEG[1] <- NULL

rownames(Sly_MEP_FC) <- Sly_MEP_FC$Protein
Sly_MEP_FC[1] <- NULL

min(Sly_MEP_FC, na.rm = TRUE)
max(Sly_MEP_FC, na.rm = TRUE)
breaks_Sly_MEP=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Sly_MEP_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Sly_MEP,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_MEP_Annotation, annotation_colors=color_MEP, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "MEP",
         annotation_legend = TRUE,
         display_numbers = Sly_MEP_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(39, 48),
         gaps_col = c(3)
)


rownames(Sly_MEP_TPM) <- Sly_MEP_TPM$Protein
Sly_MEP_TPM[1] <- NULL

min(Sly_MEP_TPM, na.rm = TRUE)
max(Sly_MEP_TPM, na.rm = TRUE)
breaks_Sly_MEP_TPM=c(seq(0,100, length.out = 200))

Sly_MEP_TPM_norm <- normalize_df(Sly_MEP_TPM)

pheatmap(Sly_MEP_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Sly_MEP_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_MEP_Annotation, annotation_colors=color_MEP, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "MEP",
         annotation_legend = TRUE,
         #display_numbers = Sly_MEP_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(21,42)
)


#####Representing Heatmap for Cann

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Cann_MEP_Ann <- read.table("Funcional Groups/FigTree/Cann_MEP.txt",
                                   sep = "\t", header = TRUE)

colnames(Cann_MEP_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Cann_MEP <- merge(Cann_MEP_Ann, Summary_Pepper_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Cann_MEP_TPM <- Cann_MEP %>% select(Protein, 4:17)
Cann_MEP_TPM <- Cann_MEP_TPM[!duplicated(Cann_MEP_TPM$Protein),]
Cann_MEP_TPM <- Cann_MEP_TPM[order(match(Cann_MEP_TPM[,1], Cann_MEP_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_MEP_FC <- Cann_MEP %>% select(Protein, 18:19)
Cann_MEP_FC <- Cann_MEP_FC[!duplicated(Cann_MEP_FC$Protein),]
Cann_MEP_FC <- Cann_MEP_FC[order(match(Cann_MEP_FC[,1], Cann_MEP_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_MEP_DEG <- Cann_MEP %>% select(Protein, 20:21)
Cann_MEP_DEG <- Cann_MEP_DEG[!duplicated(Cann_MEP_DEG$Protein),]
Cann_MEP_DEG <- Cann_MEP_DEG[order(match(Cann_MEP_DEG[,1], Cann_MEP_Ann[,2])),]

Cann_MEP_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Cann_MEP_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Cann_MEP_DEG$Ripening_Red_DEG <- ifelse(is.na(Cann_MEP_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
row.names(Cann_MEP_DEG) <- Cann_MEP_DEG$Protein
Cann_MEP_DEG[1] <- NULL

rownames(Cann_MEP_FC) <- Cann_MEP_FC$Protein
Cann_MEP_FC[1] <- NULL

min(Cann_MEP_FC, na.rm = TRUE)
max(Cann_MEP_FC, na.rm = TRUE)
breaks_Cann_MEP=c(seq(-4.5,0, length.out = 100), seq(0.01,6,length.out = 100))

pheatmap(Cann_MEP_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Cann_MEP,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_MEP_Annotation, annotation_colors=color_MEP, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Red"),
         cex = 1,
         angle_col=45,
         main = "MEP",
         annotation_legend = TRUE,
         display_numbers = Cann_MEP_DEG,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)


rownames(Cann_MEP_TPM) <- Cann_MEP_TPM$Protein
Cann_MEP_TPM[1] <- NULL

min(Cann_MEP_TPM, na.rm = TRUE)
max(Cann_MEP_TPM, na.rm = TRUE)
breaks_Cann_MEP_TPM=c(seq(0,100, length.out = 200))

Cann_MEP_TPM_norm <- normalize_df(Cann_MEP_TPM)

pheatmap(Cann_MEP_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Cann_MEP_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_MEP_Annotation, annotation_colors=color_MEP, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "MEP",
         annotation_legend = TRUE,
         #display_numbers = Cann_MEP_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(4,9)
)


## Comparativa 96h_SlyYaz_Ath

Ath_MEP_Ann <- read.table("Funcional Groups/Ath_MEP.txt", header = TRUE)

Ath_MEP_FC <- merge(Ath_MEP_Ann, Summary_Ath_FC, by.x=1, by.y = 0) %>%
  dplyr::select(Name, log2FoldChange)
Ath_MEP_FC <- Ath_MEP_FC[order(match(Ath_MEP_FC[,1], Ath_MEP_Ann[,2])),]

Ath_MEP_DEG <- merge(Ath_MEP_Ann, Summary_Ath_DEGs, by.x=1, by.y = 1, all.x = TRUE) %>%
  dplyr::select(Name, log2FC)
Ath_MEP_DEG <- Ath_MEP_DEG[order(match(Ath_MEP_DEG[,1], Ath_MEP_Ann[,2])),]

Ath_MEP_DEG$log2FC <- ifelse(is.na(Ath_MEP_DEG$log2FC) == TRUE, " ", "*")
row.names(Ath_MEP_DEG) <- Ath_MEP_DEG$Name
Ath_MEP_DEG[1] <- NULL

rownames(Ath_MEP_FC) <- Ath_MEP_FC$Name
Ath_MEP_FC[1] <- NULL

min(Ath_MEP_FC, na.rm = TRUE)
max(Ath_MEP_FC, na.rm = TRUE)
breaks_Ath_MEP=c(seq(-4.5,0, length.out = 100), seq(0.01,4.5,length.out = 100))

pheatmap(Ath_MEP_FC, 
         border_color = "grey60",legend= TRUE,
         color=color3, 
         cluster_rows =FALSE,
         breaks = breaks_Stndr,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_MEP_Annotation, annotation_colors=color_MEP, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = "Senescence",
         cex = 1,
         angle_col=45,
         main = "MEP",
         annotation_legend = TRUE,
         display_numbers = Ath_MEP_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         gaps_row = c(1, 2, 3, 4, 5, 6, 7),
         #gaps_col = c(2,5)
)

pheatmap(Niben_MEP_FC %>% select(16), 
         border_color = "grey60",legend= TRUE,
         color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_MEP_FC,
         breaks = breaks_Stndr,
         #legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         #legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_MEP_Annotation, annotation_colors=color_MEP, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = "96h",
         cex = 1,
         angle_col=45,
         main = "MEP",
         annotation_legend = TRUE,
         display_numbers = Niben_MEP_DEG %>% select(16),
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(84, 104),
         #annotation_row = Niben_MEP_Modules_Chl,
         #annotation_colors = module_Chl_colors_2,
         gaps_row = c(10, 12,14,15,17,21)
)


pheatmap(Sly_MEP_FC %>% select(2,3), 
         border_color = "grey60",legend= TRUE,
         color=color3, 
         cluster_rows =FALSE,
         breaks = breaks_Stndr,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         #legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_MEP_Annotation, annotation_colors=color_MEP, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Orange", "Ripening_Red"),
         cex = 1,
         angle_col=45,
         main = "MEP",
         annotation_legend = TRUE,
         display_numbers = Sly_MEP_DEG %>% select(2,3),
         fontsize_number = 15,
         na_col = "darkgrey",
         gaps_row = c(3, 4, 5, 6, 7, 8)
)


#########################################################################################################################################
####################################  IDIGGPP  #############################################################################################
#########################################################################################################################################

IDIGGPP_Ath <- read.table("Funcional Groups/Ath_IDIGGPP.txt", header = TRUE)
IDIGGPP_Ath <- merge(IDIGGPP_Ath, Ath_Gene_Models, by.x="ID", by.y=1)
IDIGGPP_Ath_fasta <- Ath11_fasta[c(which(names(Ath11_fasta) %in% IDIGGPP_Ath$V2))]
IDIGGPP_Ath_2 <- IDIGGPP_Ath %>% select(V2, Name)
write.fasta(IDIGGPP_Ath_fasta, as.string=TRUE, names= names(IDIGGPP_Ath_fasta),
            file.out = "Funcional Groups/Ath_IDIGGPP.fasta")
rename.fasta(infile = "Funcional Groups/Ath_IDIGGPP.fasta", IDIGGPP_Ath_2, outfile = "Funcional Groups/Ath_IDIGGPP_2.fasta")

IDIGGPP_Niben_Candidates <- read.table("Funcional Groups/Niben_Candidates_IDIGGPP.txt")
IDIGGPP_Niben_Candidates <- IDIGGPP_Niben_Candidates[IDIGGPP_Niben_Candidates$V3>=35,] %>% select(V2)
IDIGGPP_Niben_Candidates <- unique(IDIGGPP_Niben_Candidates)
IDIGGPP_Niben_Candidates_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% IDIGGPP_Niben_Candidates$V2))]
write.fasta(IDIGGPP_Niben_Candidates_fasta, as.string=TRUE, names= names(IDIGGPP_Niben_Candidates_fasta),
            file.out = "Funcional Groups/IDIGGPP_Niben_Candidates_35.fasta")

IDIGGPP_Sly_Candidates <- read.table("Funcional Groups/Sly_Candidates_IDIGGPP.txt")
IDIGGPP_Sly_Candidates <- IDIGGPP_Sly_Candidates[IDIGGPP_Sly_Candidates$V3>=35,] %>% select(V2)
IDIGGPP_Sly_Candidates <- unique(IDIGGPP_Sly_Candidates)
IDIGGPP_Sly_Candidates_fasta <- Tomato_4.0_fasta[c(which(names(Tomato_4.0_fasta) %in% IDIGGPP_Sly_Candidates$V2))]
write.fasta(IDIGGPP_Sly_Candidates_fasta, as.string=TRUE, names= names(IDIGGPP_Sly_Candidates_fasta),
            file.out = "Funcional Groups/IDIGGPP_Sly_Candidates_35.fasta")

Niben_IDIGGPP_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_IDIGGPP_Ann$Name))]
write.fasta(Niben_IDIGGPP_fasta, as.string=TRUE, names= names(Niben_IDIGGPP_fasta),
            file.out = "Funcional Groups/Niben_IDIGGPP.fasta")
Niben_IDIGGPP_Ann_2 <- Niben_IDIGGPP_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_IDIGGPP.fasta", Niben_IDIGGPP_Ann, outfile = "Funcional Groups/Niben_IDIGGPP_2.fasta")

save_heatmaps("IDIGGPP", height = 5)


## Representing Heatmap

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Niben_IDIGGPP_Ann <- read.table("Funcional Groups/FigTree/Niben_IDIGGPP.txt",
                                    sep = "\t", header = TRUE)

colnames(Niben_IDIGGPP_Ann) <- c("Name", "Protein")

Niben_IDIGGPP_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_IDIGGPP_Ann$Name))]
write.fasta(Niben_IDIGGPP_fasta, as.string=TRUE, names= names(Niben_IDIGGPP_fasta),
            file.out = "Funcional Groups/Niben_IDIGGPP.fasta")
Niben_IDIGGPP_Ann_2 <- Niben_IDIGGPP_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_IDIGGPP.fasta", Niben_IDIGGPP_Ann, outfile = "Funcional Groups/Niben_IDIGGPP_2.fasta")

Niben_IDIGGPP_Chloro <- read.delim("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan/Funcional Groups/Niben_IDIGGPP_2_summary.targetp2", 
                                       header=FALSE, comment.char="#")
colnames(Niben_IDIGGPP_Chloro) <- c("ID",	"Prediction",	"noTP",	"SP",	"mTP",	"cTP",	"luTP",	"CS-Position")
Niben_IDIGGPP_Chloro$Prediction <- factor(Niben_IDIGGPP_Chloro$Prediction, levels = c("cTP","mTP","noTP"))
Niben_IDIGGPP_Chloro <- merge(Niben_IDIGGPP_Ann, Niben_IDIGGPP_Chloro, by.x = "Protein", by.y = "ID")

Niben_IDIGGPP_Chloro <- Niben_IDIGGPP_Chloro[order(match(Niben_IDIGGPP_Chloro$Protein, Niben_IDIGGPP_Ann$Protein)),]
rownames(Niben_IDIGGPP_Chloro) <- Niben_IDIGGPP_Chloro$Protein
Niben_IDIGGPP_Chloro <- Niben_IDIGGPP_Chloro %>% select(Prediction)


## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_IDIGGPP <- merge(Niben_IDIGGPP_Ann, TPM_All_FC_Annotated, by="Name")

## We create the annotation for Modules

Niben_IDIGGPP_Modules <- which_Module_pwd12(Niben_IDIGGPP_Ann)

## We merge this annotation of Modules with the annotation of Chloroplast TP prediction

Niben_IDIGGPP_Modules_Chl <- merge(Niben_IDIGGPP_Modules, Niben_IDIGGPP_Chloro, by = 0)
Niben_IDIGGPP_Modules_Chl <- Niben_IDIGGPP_Modules_Chl[order(match(Niben_IDIGGPP_Modules_Chl$Row.names, Niben_IDIGGPP_Ann$Protein)),]
rownames(Niben_IDIGGPP_Modules_Chl) <- Niben_IDIGGPP_Modules_Chl$Row.names
Niben_IDIGGPP_Modules_Chl$Row.names <- NULL

## We select only the FC data from the different comparisons
Niben_IDIGGPP_TPM <- Niben_IDIGGPP %>% select(Protein, 4:51, 61:66)
Niben_IDIGGPP_TPM <- Niben_IDIGGPP_TPM[!duplicated(Niben_IDIGGPP_TPM$Protein),]
Niben_IDIGGPP_TPM <- Niben_IDIGGPP_TPM[order(match(Niben_IDIGGPP_TPM[,1], Niben_IDIGGPP_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_IDIGGPP_FC <- Niben_IDIGGPP %>% select(Protein, 71, 73:87)
Niben_IDIGGPP_FC <- Niben_IDIGGPP_FC[!duplicated(Niben_IDIGGPP_FC$Protein),]
Niben_IDIGGPP_FC <- Niben_IDIGGPP_FC[order(match(Niben_IDIGGPP_FC[,1], Niben_IDIGGPP_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_IDIGGPP_DEG <- Niben_IDIGGPP %>% select(Protein, 89, 91:105)
Niben_IDIGGPP_DEG <- Niben_IDIGGPP_DEG[!duplicated(Niben_IDIGGPP_DEG$Protein),]
Niben_IDIGGPP_DEG <- Niben_IDIGGPP_DEG[order(match(Niben_IDIGGPP_DEG[,1], Niben_IDIGGPP_Ann[,2])),]

Niben_IDIGGPP_DEG$pcrtB22h_vs_GFP22h_filtered <- ifelse(is.na(Niben_IDIGGPP_DEG$pcrtB22h_vs_GFP22h_filtered) == TRUE, " ", "*")
Niben_IDIGGPP_DEG$pcrtB25h_vs_GFP25h_filtered <- ifelse(is.na(Niben_IDIGGPP_DEG$pcrtB25h_vs_GFP25h_filtered) == TRUE, " ", "*")
Niben_IDIGGPP_DEG$pcrtBAA_vs_GFPAA_filtered <- ifelse(is.na(Niben_IDIGGPP_DEG$pcrtBAA_vs_GFPAA_filtered) == TRUE, " ", "*")
Niben_IDIGGPP_DEG$pcrtBBB_vs_GFPBB_filtered <- ifelse(is.na(Niben_IDIGGPP_DEG$pcrtBBB_vs_GFPBB_filtered) == TRUE, " ", "*")
Niben_IDIGGPP_DEG$pcrtBYY_vs_GFPYY_filtered <- ifelse(is.na(Niben_IDIGGPP_DEG$pcrtBYY_vs_GFPYY_filtered) == TRUE, " ", "*")
Niben_IDIGGPP_DEG$pcrtB28h_vs_GFP28h_filtered <- ifelse(is.na(Niben_IDIGGPP_DEG$pcrtB28h_vs_GFP28h_filtered) == TRUE, " ", "*")
Niben_IDIGGPP_DEG$pcrtB34h_vs_GFP34h_filtered <- ifelse(is.na(Niben_IDIGGPP_DEG$pcrtB34h_vs_GFP34h_filtered) == TRUE, " ", "*")
Niben_IDIGGPP_DEG$pcrtB37h_vs_GFP37h_filtered <- ifelse(is.na(Niben_IDIGGPP_DEG$pcrtB37h_vs_GFP37h_filtered) == TRUE, " ", "*")
Niben_IDIGGPP_DEG$pcrtBZZ_vs_GFPZZ_filtered <- ifelse(is.na(Niben_IDIGGPP_DEG$pcrtBZZ_vs_GFPZZ_filtered) == TRUE, " ", "*")
Niben_IDIGGPP_DEG$pcrtBCC_vs_GFPCC_filtered <- ifelse(is.na(Niben_IDIGGPP_DEG$pcrtBCC_vs_GFPCC_filtered) == TRUE, " ", "*")
Niben_IDIGGPP_DEG$pcrtB40h_vs_GFP40h_filtered <- ifelse(is.na(Niben_IDIGGPP_DEG$pcrtB40h_vs_GFP40h_filtered) == TRUE, " ", "*")
Niben_IDIGGPP_DEG$pcrtBDD_vs_GFPDD_filtered <- ifelse(is.na(Niben_IDIGGPP_DEG$pcrtBDD_vs_GFPDD_filtered) == TRUE, " ", "*")
Niben_IDIGGPP_DEG$pcrtB46h_vs_GFP46h_filtered <- ifelse(is.na(Niben_IDIGGPP_DEG$pcrtB46h_vs_GFP46h_filtered) == TRUE, " ", "*")
Niben_IDIGGPP_DEG$pcrtB56h_vs_GFP56h_filtered <- ifelse(is.na(Niben_IDIGGPP_DEG$pcrtB56h_vs_GFP56h_filtered) == TRUE, " ", "*")
Niben_IDIGGPP_DEG$pcrtBEE_vs_GFPEE_filtered <- ifelse(is.na(Niben_IDIGGPP_DEG$pcrtBEE_vs_GFPEE_filtered) == TRUE, " ", "*")
Niben_IDIGGPP_DEG$pcrtB96h_vs_GFP96h_filtered <- ifelse(is.na(Niben_IDIGGPP_DEG$pcrtB96h_vs_GFP96h_filtered) == TRUE, " ", "*")
row.names(Niben_IDIGGPP_DEG) <- Niben_IDIGGPP_DEG$Protein
Niben_IDIGGPP_DEG[1] <- NULL

Niben_IDIGGPP_DEG <- Niben_IDIGGPP_DEG %>%
  select(pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

row.names(Niben_IDIGGPP_FC) <- Niben_IDIGGPP_FC$Protein
Niben_IDIGGPP_FC[1] <- NULL

Niben_IDIGGPP_FC <- Niben_IDIGGPP_FC %>%
  select(pcrtB22h_vs_GFP22h_All, pcrtBAA_vs_GFPAA_All, pcrtB25h_vs_GFP25h_All, 
         pcrtBYY_vs_GFPYY_All, pcrtBBB_vs_GFPBB_All, pcrtB28h_vs_GFP28h_All, pcrtB34h_vs_GFP34h_All, pcrtBZZ_vs_GFPZZ_All, 
         pcrtB37h_vs_GFP37h_All, pcrtBCC_vs_GFPCC_All, pcrtB40h_vs_GFP40h_All, pcrtBDD_vs_GFPDD_All, pcrtB46h_vs_GFP46h_All, 
         pcrtBEE_vs_GFPEE_All, pcrtB56h_vs_GFP56h_All, pcrtB96h_vs_GFP96h_All)


min(Niben_IDIGGPP_FC, na.rm = TRUE)
max(Niben_IDIGGPP_FC, na.rm = TRUE)
breaks_IDIGGPP_FC=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Niben_IDIGGPP_FC %>% select(1:15), 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_IDIGGPP_FC,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_IDIGGPP_Annotation, annotation_colors=color_IDIGGPP, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h"),#, "96h"),
         cex = 1,
         angle_col=45,
         main = "IDIGGPP",
         annotation_legend = TRUE,
         display_numbers = Niben_IDIGGPP_DEG %>% select(1:15),
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(84, 104),
         annotation_row = Niben_IDIGGPP_Modules_Chl,
         annotation_colors = module_Chl_colors_2
         #gaps_col = c(15)
)


rownames(Niben_IDIGGPP_TPM) <- Niben_IDIGGPP_TPM$Protein
Niben_IDIGGPP_TPM[1] <- NULL

min(Niben_IDIGGPP_TPM, na.rm = TRUE)
max(Niben_IDIGGPP_TPM, na.rm = TRUE)
breaks_Niben_IDIGGPP_TPM=c(seq(0,100, length.out = 100))

Niben_IDIGGPP_TPM_norm <- normalize_df(Niben_IDIGGPP_TPM)
Niben_IDIGGPP_TPM_norm_Cornell <- Niben_IDIGGPP_TPM %>%
  select(1:48) 
Niben_IDIGGPP_TPM_norm_Cornell <- normalize_df(Niben_IDIGGPP_TPM_norm_Cornell)


pheatmap(Niben_IDIGGPP_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Niben_IDIGGPP_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_IDIGGPP_Annotation, annotation_colors=color_IDIGGPP, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "IDIGGPP",
         annotation_legend = TRUE,
         #display_numbers = Niben_IDIGGPP_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         annotation_row = Niben_IDIGGPP_Modules,
         annotation_colors = module_colors,
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(24, 48, 51)
)



#####Representing Heatmap for Sly

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Sly_IDIGGPP_Ann <- read.table("Funcional Groups/FigTree/Sly_IDIGGPP.txt",
                          sep = "\t", header = TRUE)

colnames(Sly_IDIGGPP_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Sly_IDIGGPP <- merge(Sly_IDIGGPP_Ann, Summary_Yazdani_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Sly_IDIGGPP_TPM <- Sly_IDIGGPP %>% select(Protein, 4:66)
Sly_IDIGGPP_TPM <- Sly_IDIGGPP_TPM[!duplicated(Sly_IDIGGPP_TPM$Protein),]
Sly_IDIGGPP_TPM <- Sly_IDIGGPP_TPM[order(match(Sly_IDIGGPP_TPM[,1], Sly_IDIGGPP_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_IDIGGPP_FC <- Sly_IDIGGPP %>% select(Protein, 67:71)
Sly_IDIGGPP_FC <- Sly_IDIGGPP_FC[!duplicated(Sly_IDIGGPP_FC$Protein),]
Sly_IDIGGPP_FC <- Sly_IDIGGPP_FC[order(match(Sly_IDIGGPP_FC[,1], Sly_IDIGGPP_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_IDIGGPP_DEG <- Sly_IDIGGPP %>% select(Protein, 72:76)
Sly_IDIGGPP_DEG <- Sly_IDIGGPP_DEG[!duplicated(Sly_IDIGGPP_DEG$Protein),]
Sly_IDIGGPP_DEG <- Sly_IDIGGPP_DEG[order(match(Sly_IDIGGPP_DEG[,1], Sly_IDIGGPP_Ann[,2])),]

Sly_IDIGGPP_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Sly_IDIGGPP_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Sly_IDIGGPP_DEG$Ripening_Orange_DEG <- ifelse(is.na(Sly_IDIGGPP_DEG$Ripening_Orange_DEG) == TRUE, " ", "*")
Sly_IDIGGPP_DEG$Ripening_Red_DEG <- ifelse(is.na(Sly_IDIGGPP_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
Sly_IDIGGPP_DEG$ORHis_MG_DEG <- ifelse(is.na(Sly_IDIGGPP_DEG$ORHis_MG_DEG) == TRUE, " ", "*")
Sly_IDIGGPP_DEG$ORHis_Breaker_DEG <- ifelse(is.na(Sly_IDIGGPP_DEG$ORHis_Breaker_DEG) == TRUE, " ", "*")
row.names(Sly_IDIGGPP_DEG) <- Sly_IDIGGPP_DEG$Protein
Sly_IDIGGPP_DEG[1] <- NULL

rownames(Sly_IDIGGPP_FC) <- Sly_IDIGGPP_FC$Protein
Sly_IDIGGPP_FC[1] <- NULL

min(Sly_IDIGGPP_FC, na.rm = TRUE)
max(Sly_IDIGGPP_FC, na.rm = TRUE)
breaks_Sly_IDIGGPP=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Sly_IDIGGPP_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Sly_IDIGGPP,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_IDIGGPP_Annotation, annotation_colors=color_IDIGGPP, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "IDIGGPP",
         annotation_legend = TRUE,
         display_numbers = Sly_IDIGGPP_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(39, 48),
         gaps_col = c(3)
)


rownames(Sly_IDIGGPP_TPM) <- Sly_IDIGGPP_TPM$Protein
Sly_IDIGGPP_TPM[1] <- NULL

min(Sly_IDIGGPP_TPM, na.rm = TRUE)
max(Sly_IDIGGPP_TPM, na.rm = TRUE)
breaks_Sly_IDIGGPP_TPM=c(seq(0,100, length.out = 200))

Sly_IDIGGPP_TPM_norm <- normalize_df(Sly_IDIGGPP_TPM)

pheatmap(Sly_IDIGGPP_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Sly_IDIGGPP_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_IDIGGPP_Annotation, annotation_colors=color_IDIGGPP, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "IDIGGPP",
         annotation_legend = TRUE,
         #display_numbers = Sly_IDIGGPP_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(21,42)
)


#####Representing Heatmap for Cann

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Cann_IDIGGPP_Ann <- read.table("Funcional Groups/FigTree/Cann_IDIGGPP.txt",
                           sep = "\t", header = TRUE)

colnames(Cann_IDIGGPP_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Cann_IDIGGPP <- merge(Cann_IDIGGPP_Ann, Summary_Pepper_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Cann_IDIGGPP_TPM <- Cann_IDIGGPP %>% select(Protein, 4:17)
Cann_IDIGGPP_TPM <- Cann_IDIGGPP_TPM[!duplicated(Cann_IDIGGPP_TPM$Protein),]
Cann_IDIGGPP_TPM <- Cann_IDIGGPP_TPM[order(match(Cann_IDIGGPP_TPM[,1], Cann_IDIGGPP_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_IDIGGPP_FC <- Cann_IDIGGPP %>% select(Protein, 18:19)
Cann_IDIGGPP_FC <- Cann_IDIGGPP_FC[!duplicated(Cann_IDIGGPP_FC$Protein),]
Cann_IDIGGPP_FC <- Cann_IDIGGPP_FC[order(match(Cann_IDIGGPP_FC[,1], Cann_IDIGGPP_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_IDIGGPP_DEG <- Cann_IDIGGPP %>% select(Protein, 20:21)
Cann_IDIGGPP_DEG <- Cann_IDIGGPP_DEG[!duplicated(Cann_IDIGGPP_DEG$Protein),]
Cann_IDIGGPP_DEG <- Cann_IDIGGPP_DEG[order(match(Cann_IDIGGPP_DEG[,1], Cann_IDIGGPP_Ann[,2])),]

Cann_IDIGGPP_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Cann_IDIGGPP_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Cann_IDIGGPP_DEG$Ripening_Red_DEG <- ifelse(is.na(Cann_IDIGGPP_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
row.names(Cann_IDIGGPP_DEG) <- Cann_IDIGGPP_DEG$Protein
Cann_IDIGGPP_DEG[1] <- NULL

rownames(Cann_IDIGGPP_FC) <- Cann_IDIGGPP_FC$Protein
Cann_IDIGGPP_FC[1] <- NULL

min(Cann_IDIGGPP_FC, na.rm = TRUE)
max(Cann_IDIGGPP_FC, na.rm = TRUE)
breaks_Cann_IDIGGPP=c(seq(-4.5,0, length.out = 100), seq(0.01,6,length.out = 100))

pheatmap(Cann_IDIGGPP_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Cann_IDIGGPP,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_IDIGGPP_Annotation, annotation_colors=color_IDIGGPP, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Red"),
         cex = 1,
         angle_col=45,
         main = "IDIGGPP",
         annotation_legend = TRUE,
         display_numbers = Cann_IDIGGPP_DEG,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)


rownames(Cann_IDIGGPP_TPM) <- Cann_IDIGGPP_TPM$Protein
Cann_IDIGGPP_TPM[1] <- NULL

min(Cann_IDIGGPP_TPM, na.rm = TRUE)
max(Cann_IDIGGPP_TPM, na.rm = TRUE)
breaks_Cann_IDIGGPP_TPM=c(seq(0,100, length.out = 200))

Cann_IDIGGPP_TPM_norm <- normalize_df(Cann_IDIGGPP_TPM)

pheatmap(Cann_IDIGGPP_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Cann_IDIGGPP_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_IDIGGPP_Annotation, annotation_colors=color_IDIGGPP, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "IDIGGPP",
         annotation_legend = TRUE,
         #display_numbers = Cann_IDIGGPP_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(4,9)
)


## Comparativa 96h_SlyYaz_Ath

Ath_IDIGGPP_Ann <- read.table("Funcional Groups/Ath_IDIGGPP.txt", header = TRUE)

Ath_IDIGGPP_FC <- merge(Ath_IDIGGPP_Ann, Summary_Ath_FC, by.x=1, by.y = 0, all.x = TRUE) %>%
  dplyr::select(Name, log2FoldChange)
Ath_IDIGGPP_FC <- Ath_IDIGGPP_FC[order(match(Ath_IDIGGPP_FC[,1], Ath_IDIGGPP_Ann[,2])),]

Ath_IDIGGPP_DEG <- merge(Ath_IDIGGPP_Ann, Summary_Ath_DEGs, by.x=1, by.y = 1, all.x = TRUE) %>%
  dplyr::select(Name, log2FC)
Ath_IDIGGPP_DEG <- Ath_IDIGGPP_DEG[order(match(Ath_IDIGGPP_DEG[,1], Ath_IDIGGPP_Ann[,2])),]

Ath_IDIGGPP_DEG$log2FC <- ifelse(is.na(Ath_IDIGGPP_DEG$log2FC) == TRUE, " ", "*")
row.names(Ath_IDIGGPP_DEG) <- Ath_IDIGGPP_DEG$Name
Ath_IDIGGPP_DEG[1] <- NULL

rownames(Ath_IDIGGPP_FC) <- Ath_IDIGGPP_FC$Name
Ath_IDIGGPP_FC[1] <- NULL

min(Ath_IDIGGPP_FC, na.rm = TRUE)
max(Ath_IDIGGPP_FC, na.rm = TRUE)
breaks_Ath_IDIGGPP=c(seq(-4.5,0, length.out = 100), seq(0.01,4.5,length.out = 100))

pheatmap(Ath_IDIGGPP_FC, 
         border_color = "grey60",legend= TRUE,
         color=color3, 
         cluster_rows =FALSE,
         breaks = breaks_Stndr,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_IDIGGPP_Annotation, annotation_colors=color_IDIGGPP, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = "Senescence",
         cex = 1,
         angle_col=45,
         main = "IDIGGPP",
         annotation_legend = TRUE,
         display_numbers = Ath_IDIGGPP_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         gaps_row = c(2, 4)
         #gaps_col = c(2,5)
)

pheatmap(Niben_IDIGGPP_FC %>% select(16), 
         border_color = "grey60",legend= TRUE,
         color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_IDIGGPP_FC,
         breaks = breaks_Stndr,
         #legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         #legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_IDIGGPP_Annotation, annotation_colors=color_IDIGGPP, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = "96h",
         cex = 1,
         angle_col=45,
         main = "IDIGGPP",
         annotation_legend = TRUE,
         display_numbers = Niben_IDIGGPP_DEG %>% select(16),
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(84, 104),
         #annotation_row = Niben_IDIGGPP_Modules_Chl,
         #annotation_colors = module_Chl_colors_2,
         gaps_row = c(4, 8)
)


pheatmap(Sly_IDIGGPP_FC %>% select(2,3), 
         border_color = "grey60",legend= TRUE,
         color=color3, 
         cluster_rows =FALSE,
         breaks = breaks_Stndr,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         #legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_IDIGGPP_Annotation, annotation_colors=color_IDIGGPP, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Orange", "Ripening_Red"),
         cex = 1,
         angle_col=45,
         main = "IDIGGPP",
         annotation_legend = TRUE,
         display_numbers = Sly_IDIGGPP_DEG %>% select(2,3),
         fontsize_number = 15,
         na_col = "darkgrey",
         gaps_row = c(2, 5)
)


#########################################################################################################################################
####################################  MVA  #############################################################################################
#########################################################################################################################################

MVA_Ath <- read.table("Funcional Groups/Ath_MVA.txt", header = TRUE)
MVA_Ath <- merge(MVA_Ath, Ath_Gene_Models, by.x="ID", by.y=1)
MVA_Ath_fasta <- Ath11_fasta[c(which(names(Ath11_fasta) %in% MVA_Ath$V2))]
MVA_Ath_2 <- MVA_Ath %>% select(V2, Name)
write.fasta(MVA_Ath_fasta, as.string=TRUE, names= names(MVA_Ath_fasta),
            file.out = "Funcional Groups/Ath_MVA.fasta")
rename.fasta(infile = "Funcional Groups/Ath_MVA.fasta", MVA_Ath_2, outfile = "Funcional Groups/Ath_MVA_2.fasta")

MVA_Niben_Candidates <- read.table("Funcional Groups/Niben_Candidates_MVA.txt")
MVA_Niben_Candidates <- MVA_Niben_Candidates[MVA_Niben_Candidates$V3>=35,] %>% select(V2)
MVA_Niben_Candidates <- unique(MVA_Niben_Candidates)
MVA_Niben_Candidates_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% MVA_Niben_Candidates$V2))]
write.fasta(MVA_Niben_Candidates_fasta, as.string=TRUE, names= names(MVA_Niben_Candidates_fasta),
            file.out = "Funcional Groups/MVA_Niben_Candidates_35.fasta")

MVA_Sly_Candidates <- read.table("Funcional Groups/Sly_Candidates_MVA.txt")
MVA_Sly_Candidates <- MVA_Sly_Candidates[MVA_Sly_Candidates$V3>=35,] %>% select(V2)
MVA_Sly_Candidates <- unique(MVA_Sly_Candidates)
MVA_Sly_Candidates_fasta <- Tomato_4.0_fasta[c(which(names(Tomato_4.0_fasta) %in% MVA_Sly_Candidates$V2))]
write.fasta(MVA_Sly_Candidates_fasta, as.string=TRUE, names= names(MVA_Sly_Candidates_fasta),
            file.out = "Funcional Groups/MVA_Sly_Candidates_35.fasta")

Niben_MVA_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_MVA_Ann$Name))]
write.fasta(Niben_MVA_fasta, as.string=TRUE, names= names(Niben_MVA_fasta),
            file.out = "Funcional Groups/Niben_MVA.fasta")
Niben_MVA_Ann_2 <- Niben_MVA_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_MVA.fasta", Niben_MVA_Ann, outfile = "Funcional Groups/Niben_MVA_2.fasta")

save_heatmaps("MVA", height = 6)


## Representing Heatmap

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Niben_MVA_Ann <- read.table("Funcional Groups/FigTree/Niben_MVA.txt",
                                    sep = "\t", header = TRUE)

colnames(Niben_MVA_Ann) <- c("Name", "Protein")

Niben_MVA_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_MVA_Ann$Name))]
write.fasta(Niben_MVA_fasta, as.string=TRUE, names= names(Niben_MVA_fasta),
            file.out = "Funcional Groups/Niben_MVA.fasta")
Niben_MVA_Ann_2 <- Niben_MVA_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_MVA.fasta", Niben_MVA_Ann, outfile = "Funcional Groups/Niben_MVA_2.fasta")

Niben_MVA_Chloro <- read.delim("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan/Funcional Groups/Niben_MVA_2_summary.targetp2", 
                                       header=FALSE, comment.char="#")
colnames(Niben_MVA_Chloro) <- c("ID",	"Prediction",	"noTP",	"SP",	"mTP",	"cTP",	"luTP",	"CS-Position")
Niben_MVA_Chloro$Prediction <- factor(Niben_MVA_Chloro$Prediction, levels = c("cTP","mTP","noTP"))
Niben_MVA_Chloro <- merge(Niben_MVA_Ann, Niben_MVA_Chloro, by.x = "Protein", by.y = "ID")

Niben_MVA_Chloro <- Niben_MVA_Chloro[order(match(Niben_MVA_Chloro$Protein, Niben_MVA_Ann$Protein)),]
rownames(Niben_MVA_Chloro) <- Niben_MVA_Chloro$Protein
Niben_MVA_Chloro <- Niben_MVA_Chloro %>% select(Prediction)


## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_MVA <- merge(Niben_MVA_Ann, TPM_All_FC_Annotated, by="Name")

## We create the annotation for Modules

Niben_MVA_Modules <- which_Module_pwd12(Niben_MVA_Ann)

## We merge this annotation of Modules with the annotation of Chloroplast TP prediction

Niben_MVA_Modules_Chl <- merge(Niben_MVA_Modules, Niben_MVA_Chloro, by = 0)
Niben_MVA_Modules_Chl <- Niben_MVA_Modules_Chl[order(match(Niben_MVA_Modules_Chl$Row.names, Niben_MVA_Ann$Protein)),]
rownames(Niben_MVA_Modules_Chl) <- Niben_MVA_Modules_Chl$Row.names
Niben_MVA_Modules_Chl$Row.names <- NULL

## We select only the FC data from the different comparisons
Niben_MVA_TPM <- Niben_MVA %>% select(Protein, 4:51, 61:66)
Niben_MVA_TPM <- Niben_MVA_TPM[!duplicated(Niben_MVA_TPM$Protein),]
Niben_MVA_TPM <- Niben_MVA_TPM[order(match(Niben_MVA_TPM[,1], Niben_MVA_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_MVA_FC <- Niben_MVA %>% select(Protein, 71, 73:87)
Niben_MVA_FC <- Niben_MVA_FC[!duplicated(Niben_MVA_FC$Protein),]
Niben_MVA_FC <- Niben_MVA_FC[order(match(Niben_MVA_FC[,1], Niben_MVA_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_MVA_DEG <- Niben_MVA %>% select(Protein, 89, 91:105)
Niben_MVA_DEG <- Niben_MVA_DEG[!duplicated(Niben_MVA_DEG$Protein),]
Niben_MVA_DEG <- Niben_MVA_DEG[order(match(Niben_MVA_DEG[,1], Niben_MVA_Ann[,2])),]

Niben_MVA_DEG$pcrtB22h_vs_GFP22h_filtered <- ifelse(is.na(Niben_MVA_DEG$pcrtB22h_vs_GFP22h_filtered) == TRUE, " ", "*")
Niben_MVA_DEG$pcrtB25h_vs_GFP25h_filtered <- ifelse(is.na(Niben_MVA_DEG$pcrtB25h_vs_GFP25h_filtered) == TRUE, " ", "*")
Niben_MVA_DEG$pcrtBAA_vs_GFPAA_filtered <- ifelse(is.na(Niben_MVA_DEG$pcrtBAA_vs_GFPAA_filtered) == TRUE, " ", "*")
Niben_MVA_DEG$pcrtBBB_vs_GFPBB_filtered <- ifelse(is.na(Niben_MVA_DEG$pcrtBBB_vs_GFPBB_filtered) == TRUE, " ", "*")
Niben_MVA_DEG$pcrtBYY_vs_GFPYY_filtered <- ifelse(is.na(Niben_MVA_DEG$pcrtBYY_vs_GFPYY_filtered) == TRUE, " ", "*")
Niben_MVA_DEG$pcrtB28h_vs_GFP28h_filtered <- ifelse(is.na(Niben_MVA_DEG$pcrtB28h_vs_GFP28h_filtered) == TRUE, " ", "*")
Niben_MVA_DEG$pcrtB34h_vs_GFP34h_filtered <- ifelse(is.na(Niben_MVA_DEG$pcrtB34h_vs_GFP34h_filtered) == TRUE, " ", "*")
Niben_MVA_DEG$pcrtB37h_vs_GFP37h_filtered <- ifelse(is.na(Niben_MVA_DEG$pcrtB37h_vs_GFP37h_filtered) == TRUE, " ", "*")
Niben_MVA_DEG$pcrtBZZ_vs_GFPZZ_filtered <- ifelse(is.na(Niben_MVA_DEG$pcrtBZZ_vs_GFPZZ_filtered) == TRUE, " ", "*")
Niben_MVA_DEG$pcrtBCC_vs_GFPCC_filtered <- ifelse(is.na(Niben_MVA_DEG$pcrtBCC_vs_GFPCC_filtered) == TRUE, " ", "*")
Niben_MVA_DEG$pcrtB40h_vs_GFP40h_filtered <- ifelse(is.na(Niben_MVA_DEG$pcrtB40h_vs_GFP40h_filtered) == TRUE, " ", "*")
Niben_MVA_DEG$pcrtBDD_vs_GFPDD_filtered <- ifelse(is.na(Niben_MVA_DEG$pcrtBDD_vs_GFPDD_filtered) == TRUE, " ", "*")
Niben_MVA_DEG$pcrtB46h_vs_GFP46h_filtered <- ifelse(is.na(Niben_MVA_DEG$pcrtB46h_vs_GFP46h_filtered) == TRUE, " ", "*")
Niben_MVA_DEG$pcrtB56h_vs_GFP56h_filtered <- ifelse(is.na(Niben_MVA_DEG$pcrtB56h_vs_GFP56h_filtered) == TRUE, " ", "*")
Niben_MVA_DEG$pcrtBEE_vs_GFPEE_filtered <- ifelse(is.na(Niben_MVA_DEG$pcrtBEE_vs_GFPEE_filtered) == TRUE, " ", "*")
Niben_MVA_DEG$pcrtB96h_vs_GFP96h_filtered <- ifelse(is.na(Niben_MVA_DEG$pcrtB96h_vs_GFP96h_filtered) == TRUE, " ", "*")
row.names(Niben_MVA_DEG) <- Niben_MVA_DEG$Protein
Niben_MVA_DEG[1] <- NULL

Niben_MVA_DEG <- Niben_MVA_DEG %>%
  select(pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

row.names(Niben_MVA_FC) <- Niben_MVA_FC$Protein
Niben_MVA_FC[1] <- NULL

Niben_MVA_FC <- Niben_MVA_FC %>%
  select(pcrtB22h_vs_GFP22h_All, pcrtBAA_vs_GFPAA_All, pcrtB25h_vs_GFP25h_All, 
         pcrtBYY_vs_GFPYY_All, pcrtBBB_vs_GFPBB_All, pcrtB28h_vs_GFP28h_All, pcrtB34h_vs_GFP34h_All, pcrtBZZ_vs_GFPZZ_All, 
         pcrtB37h_vs_GFP37h_All, pcrtBCC_vs_GFPCC_All, pcrtB40h_vs_GFP40h_All, pcrtBDD_vs_GFPDD_All, pcrtB46h_vs_GFP46h_All, 
         pcrtBEE_vs_GFPEE_All, pcrtB56h_vs_GFP56h_All, pcrtB96h_vs_GFP96h_All)


min(Niben_MVA_FC, na.rm = TRUE)
max(Niben_MVA_FC, na.rm = TRUE)
breaks_MVA_FC=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Niben_MVA_FC %>% select(1:15), 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_MVA_FC,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_MVA_Annotation, annotation_colors=color_MVA, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h"),#, "96h"),
         cex = 1,
         angle_col=45,
         main = "MVA",
         annotation_legend = TRUE,
         display_numbers = Niben_MVA_DEG %>% select(1:15),
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(84, 104),
         annotation_row = Niben_MVA_Modules_Chl,
         annotation_colors = module_Chl_colors_2
         #gaps_col = c(15)
)


rownames(Niben_MVA_TPM) <- Niben_MVA_TPM$Protein
Niben_MVA_TPM[1] <- NULL

min(Niben_MVA_TPM, na.rm = TRUE)
max(Niben_MVA_TPM, na.rm = TRUE)
breaks_Niben_MVA_TPM=c(seq(0,100, length.out = 100))

Niben_MVA_TPM_norm <- normalize_df(Niben_MVA_TPM)
Niben_MVA_TPM_norm_Cornell <- Niben_MVA_TPM %>%
  select(1:48) 
Niben_MVA_TPM_norm_Cornell <- normalize_df(Niben_MVA_TPM_norm_Cornell)


pheatmap(Niben_MVA_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Niben_MVA_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_MVA_Annotation, annotation_colors=color_MVA, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "MVA",
         annotation_legend = TRUE,
         #display_numbers = Niben_MVA_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         annotation_row = Niben_MVA_Modules,
         annotation_colors = module_colors,
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(24, 48, 51)
)



#####Representing Heatmap for Sly

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Sly_MVA_Ann <- read.table("Funcional Groups/FigTree/Sly_MVA.txt",
                              sep = "\t", header = TRUE)

colnames(Sly_MVA_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Sly_MVA <- merge(Sly_MVA_Ann, Summary_Yazdani_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Sly_MVA_TPM <- Sly_MVA %>% select(Protein, 4:66)
Sly_MVA_TPM <- Sly_MVA_TPM[!duplicated(Sly_MVA_TPM$Protein),]
Sly_MVA_TPM <- Sly_MVA_TPM[order(match(Sly_MVA_TPM[,1], Sly_MVA_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_MVA_FC <- Sly_MVA %>% select(Protein, 67:71)
Sly_MVA_FC <- Sly_MVA_FC[!duplicated(Sly_MVA_FC$Protein),]
Sly_MVA_FC <- Sly_MVA_FC[order(match(Sly_MVA_FC[,1], Sly_MVA_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_MVA_DEG <- Sly_MVA %>% select(Protein, 72:76)
Sly_MVA_DEG <- Sly_MVA_DEG[!duplicated(Sly_MVA_DEG$Protein),]
Sly_MVA_DEG <- Sly_MVA_DEG[order(match(Sly_MVA_DEG[,1], Sly_MVA_Ann[,2])),]

Sly_MVA_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Sly_MVA_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Sly_MVA_DEG$Ripening_Orange_DEG <- ifelse(is.na(Sly_MVA_DEG$Ripening_Orange_DEG) == TRUE, " ", "*")
Sly_MVA_DEG$Ripening_Red_DEG <- ifelse(is.na(Sly_MVA_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
Sly_MVA_DEG$ORHis_MG_DEG <- ifelse(is.na(Sly_MVA_DEG$ORHis_MG_DEG) == TRUE, " ", "*")
Sly_MVA_DEG$ORHis_Breaker_DEG <- ifelse(is.na(Sly_MVA_DEG$ORHis_Breaker_DEG) == TRUE, " ", "*")
row.names(Sly_MVA_DEG) <- Sly_MVA_DEG$Protein
Sly_MVA_DEG[1] <- NULL

rownames(Sly_MVA_FC) <- Sly_MVA_FC$Protein
Sly_MVA_FC[1] <- NULL

min(Sly_MVA_FC, na.rm = TRUE)
max(Sly_MVA_FC, na.rm = TRUE)
breaks_Sly_MVA=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Sly_MVA_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Sly_MVA,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_MVA_Annotation, annotation_colors=color_MVA, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "MVA",
         annotation_legend = TRUE,
         display_numbers = Sly_MVA_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(39, 48),
         gaps_col = c(3)
)


rownames(Sly_MVA_TPM) <- Sly_MVA_TPM$Protein
Sly_MVA_TPM[1] <- NULL

min(Sly_MVA_TPM, na.rm = TRUE)
max(Sly_MVA_TPM, na.rm = TRUE)
breaks_Sly_MVA_TPM=c(seq(0,100, length.out = 200))

Sly_MVA_TPM_norm <- normalize_df(Sly_MVA_TPM)

pheatmap(Sly_MVA_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Sly_MVA_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_MVA_Annotation, annotation_colors=color_MVA, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "MVA",
         annotation_legend = TRUE,
         #display_numbers = Sly_MVA_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(21,42)
)


#####Representing Heatmap for Cann

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Cann_MVA_Ann <- read.table("Funcional Groups/FigTree/Cann_MVA.txt",
                               sep = "\t", header = TRUE)

colnames(Cann_MVA_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Cann_MVA <- merge(Cann_MVA_Ann, Summary_Pepper_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Cann_MVA_TPM <- Cann_MVA %>% select(Protein, 4:17)
Cann_MVA_TPM <- Cann_MVA_TPM[!duplicated(Cann_MVA_TPM$Protein),]
Cann_MVA_TPM <- Cann_MVA_TPM[order(match(Cann_MVA_TPM[,1], Cann_MVA_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_MVA_FC <- Cann_MVA %>% select(Protein, 18:19)
Cann_MVA_FC <- Cann_MVA_FC[!duplicated(Cann_MVA_FC$Protein),]
Cann_MVA_FC <- Cann_MVA_FC[order(match(Cann_MVA_FC[,1], Cann_MVA_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_MVA_DEG <- Cann_MVA %>% select(Protein, 20:21)
Cann_MVA_DEG <- Cann_MVA_DEG[!duplicated(Cann_MVA_DEG$Protein),]
Cann_MVA_DEG <- Cann_MVA_DEG[order(match(Cann_MVA_DEG[,1], Cann_MVA_Ann[,2])),]

Cann_MVA_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Cann_MVA_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Cann_MVA_DEG$Ripening_Red_DEG <- ifelse(is.na(Cann_MVA_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
row.names(Cann_MVA_DEG) <- Cann_MVA_DEG$Protein
Cann_MVA_DEG[1] <- NULL

rownames(Cann_MVA_FC) <- Cann_MVA_FC$Protein
Cann_MVA_FC[1] <- NULL

min(Cann_MVA_FC, na.rm = TRUE)
max(Cann_MVA_FC, na.rm = TRUE)
breaks_Cann_MVA=c(seq(-4.5,0, length.out = 100), seq(0.01,6,length.out = 100))

pheatmap(Cann_MVA_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Cann_MVA,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_MVA_Annotation, annotation_colors=color_MVA, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Red"),
         cex = 1,
         angle_col=45,
         main = "MVA",
         annotation_legend = TRUE,
         display_numbers = Cann_MVA_DEG,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)


rownames(Cann_MVA_TPM) <- Cann_MVA_TPM$Protein
Cann_MVA_TPM[1] <- NULL

min(Cann_MVA_TPM, na.rm = TRUE)
max(Cann_MVA_TPM, na.rm = TRUE)
breaks_Cann_MVA_TPM=c(seq(0,100, length.out = 200))

Cann_MVA_TPM_norm <- normalize_df(Cann_MVA_TPM)

pheatmap(Cann_MVA_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Cann_MVA_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_MVA_Annotation, annotation_colors=color_MVA, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "MVA",
         annotation_legend = TRUE,
         #display_numbers = Cann_MVA_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(4,9)
)



## Comparativa 96h_SlyYaz_Ath

Ath_MVA_Ann <- read.table("Funcional Groups/Ath_MVA.txt", header = TRUE)

Ath_MVA_FC <- merge(Ath_MVA_Ann, Summary_Ath_FC, by.x=1, by.y = 0, all.x = TRUE) %>%
  dplyr::select(Name, log2FoldChange)
Ath_MVA_FC <- Ath_MVA_FC[order(match(Ath_MVA_FC[,1], Ath_MVA_Ann[,2])),]

Ath_MVA_DEG <- merge(Ath_MVA_Ann, Summary_Ath_DEGs, by.x=1, by.y = 1, all.x = TRUE) %>%
  dplyr::select(Name, log2FC)
Ath_MVA_DEG <- Ath_MVA_DEG[order(match(Ath_MVA_DEG[,1], Ath_MVA_Ann[,2])),]

Ath_MVA_DEG$log2FC <- ifelse(is.na(Ath_MVA_DEG$log2FC) == TRUE, " ", "*")
row.names(Ath_MVA_DEG) <- Ath_MVA_DEG$Name
Ath_MVA_DEG[1] <- NULL

rownames(Ath_MVA_FC) <- Ath_MVA_FC$Name
Ath_MVA_FC[1] <- NULL

min(Ath_MVA_FC, na.rm = TRUE)
max(Ath_MVA_FC, na.rm = TRUE)
breaks_Ath_MVA=c(seq(-4.5,0, length.out = 100), seq(0.01,4.5,length.out = 100))

pheatmap(Ath_MVA_FC, 
         border_color = "grey60",legend= TRUE,
         color=color3, 
         cluster_rows =FALSE,
         breaks = breaks_Stndr,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_MVA_Annotation, annotation_colors=color_MVA, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = "Senescence",
         cex = 1,
         angle_col=45,
         main = "MVA",
         annotation_legend = TRUE,
         display_numbers = Ath_MVA_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         gaps_row = c(2, 3, 5, 6,7)
         #gaps_col = c(2,5)
)

pheatmap(Niben_MVA_FC %>% select(16), 
         border_color = "grey60",legend= TRUE,
         color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_MVA_FC,
         breaks = breaks_Stndr,
         #legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         #legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_MVA_Annotation, annotation_colors=color_MVA, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = "96h",
         cex = 1,
         angle_col=45,
         main = "MVA",
         annotation_legend = TRUE,
         display_numbers = Niben_MVA_DEG %>% select(16),
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(84, 104),
         #annotation_row = Niben_MVA_Modules_Chl,
         #annotation_colors = module_Chl_colors_2,
         gaps_row = c(5, 10, 15, 17, 19)
)


pheatmap(Sly_MVA_FC %>% select(2,3), 
         border_color = "grey60",legend= TRUE,
         color=color3, 
         cluster_rows =FALSE,
         breaks = breaks_Stndr,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         #legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_MVA_Annotation, annotation_colors=color_MVA, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Orange", "Ripening_Red"),
         cex = 1,
         angle_col=45,
         main = "MVA",
         annotation_legend = TRUE,
         display_numbers = Sly_MVA_DEG %>% select(2,3),
         fontsize_number = 15,
         na_col = "darkgrey",
         gaps_row = c(3, 7, 10, 11, 13)
)


#########################################################################################################################################
####################################  SP1  #############################################################################################
#########################################################################################################################################

SP1_Ath <- read.table("Funcional Groups/Ath_SP1.txt", header = TRUE)
SP1_Ath <- merge(SP1_Ath, Ath_Gene_Models, by.x="ID", by.y=1)
SP1_Ath_fasta <- Ath11_fasta[c(which(names(Ath11_fasta) %in% SP1_Ath$V2))]
SP1_Ath_2 <- SP1_Ath %>% select(V2, Name)
write.fasta(SP1_Ath_fasta, as.string=TRUE, names= names(SP1_Ath_fasta),
            file.out = "Funcional Groups/Ath_SP1.fasta")
rename.fasta(infile = "Funcional Groups/Ath_SP1.fasta", SP1_Ath_2, outfile = "Funcional Groups/Ath_SP1_2.fasta")

SP1_Niben_Candidates <- read.table("Funcional Groups/Niben_Candidates_SP1.txt")
SP1_Niben_Candidates <- SP1_Niben_Candidates[SP1_Niben_Candidates$V3>=35,] %>% select(V2)
SP1_Niben_Candidates <- unique(SP1_Niben_Candidates)
SP1_Niben_Candidates_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% SP1_Niben_Candidates$V2))]
write.fasta(SP1_Niben_Candidates_fasta, as.string=TRUE, names= names(SP1_Niben_Candidates_fasta),
            file.out = "Funcional Groups/SP1_Niben_Candidates_35.fasta")

SP1_Sly_Candidates <- read.table("Funcional Groups/Sly_Candidates_SP1.txt")
SP1_Sly_Candidates <- SP1_Sly_Candidates[SP1_Sly_Candidates$V3>=35,] %>% select(V2)
SP1_Sly_Candidates <- unique(SP1_Sly_Candidates)
SP1_Sly_Candidates_fasta <- Tomato_4.0_fasta[c(which(names(Tomato_4.0_fasta) %in% SP1_Sly_Candidates$V2))]
write.fasta(SP1_Sly_Candidates_fasta, as.string=TRUE, names= names(SP1_Sly_Candidates_fasta),
            file.out = "Funcional Groups/SP1_Sly_Candidates_35.fasta")

Niben_SP1_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_SP1_Ann$Name))]
write.fasta(Niben_SP1_fasta, as.string=TRUE, names= names(Niben_SP1_fasta),
            file.out = "Funcional Groups/Niben_SP1.fasta")
Niben_SP1_Ann_2 <- Niben_SP1_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_SP1.fasta", Niben_SP1_Ann, outfile = "Funcional Groups/Niben_SP1_2.fasta")

save_heatmaps("SP1", height = 6)


## Representing Heatmap

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
#Niben_SP1_Ann <- read.table("Funcional Groups/FigTree/Niben_SP1.txt",
#                                    sep = "\t", header = TRUE)
Niben_SP1_Ann <- data.frame(Name= c("Niben261Chr04g1586012.1", "Niben261Chr13g0593009.1"),
                            Protein= c("SP1a", "SP1b"))
colnames(Niben_SP1_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_SP1 <- merge(Niben_SP1_Ann, TPM_All_FC_Annotated, by="Name")

## We select only the FC data from the different comparisons
Niben_SP1 <- Niben_SP1 %>% select(Protein, 88:105)
Niben_SP1 <- Niben_SP1[!duplicated(Niben_SP1$Protein),]
Niben_SP1 <- Niben_SP1[order(match(Niben_SP1[,1], Niben_SP1_Ann[,2])),]

## We prepare the data for a heatmap and we map it
Niben_SP1_Annotation <- merge(Niben_SP1, Niben_SP1_Ann, by="Name")
Niben_SP1_Annotation <- Niben_SP1_Annotation[!duplicated(Niben_SP1_Annotation$Name),]
Niben_SP1_Annotation <- Niben_SP1_Annotation[order(match(Niben_SP1_Annotation[,1], Niben_SP1_Ann[,1])),]
#write.table(Niben_SP1_Annotation, "Milan/RNAseq_Milan/Funcional Groups/SP1_Niben261_OrthoVenn_DESeq2FC.txt", quote = FALSE, dec = ",", sep = "\t", row.names = FALSE)
rownames(Niben_SP1_Annotation) <- Niben_SP1_Annotation$Name
#Niben_SP1_Chloroplast <- Niben_SP1_Annotation %>% dplyr::filter(Presumed.subcellular.location=="Chloroplast")
#write.table(Niben_SP1_Chloroplast, "Milan/RNAseq_Milan/Funcional Groups/SP1_Chloroplast_Niben261_OrthoVenn_DESeq2FC.txt", quote = FALSE, dec = ",", sep = "\t", row.names = FALSE)
#Niben_SP1_Chloroplast_Annotation <- Niben_SP1_Chloroplast %>% select(Protein.names)
#Niben_SP1_Chloroplast <- Niben_SP1_Chloroplast %>% select(2:19)
Niben_SP1_Annotation <- Niben_SP1_Annotation %>% select(Protein)

rownames(Niben_SP1) <- Niben_SP1$Protein
Niben_SP1[1] <- NULL

Niben_SP1 <- Niben_SP1 %>%
  select(pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

min(Niben_SP1, na.rm = TRUE)
max(Niben_SP1, na.rm = TRUE)
breaks_SP1=c(seq(-1,0, length.out = 100), seq(0.01,1,length.out = 100))

color_SP1 <- colorRampPalette(c("#415a20","#9bac12","#e5c033","#c38144","#894539"))(n=length(unique(Niben_SP1_Annotation$Protein)))
names(color_SP1) <- unique(Niben_SP1_Annotation$Protein)
color_SP1 <- list(Protein=color_SP1)

pheatmap(Niben_SP1, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_SP1,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_SP1_Annotation, annotation_colors=color_SP1, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h", "96h"),
         cex = 1,
         angle_col=45,
         main = "SP1",
         annotation_legend = TRUE,
         #display_numbers = Nb_Sly_Ath_MVA_Sig,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)


#########################################################################################################################################
####################################  TOC33  #############################################################################################
#########################################################################################################################################

TOC33_Ath <- read.table("Funcional Groups/Ath_TOC33.txt", header = TRUE)
TOC33_Ath <- merge(TOC33_Ath, Ath_Gene_Models, by.x="ID", by.y=1)
TOC33_Ath_fasta <- Ath11_fasta[c(which(names(Ath11_fasta) %in% TOC33_Ath$V2))]
TOC33_Ath_2 <- TOC33_Ath %>% select(V2, Name)
write.fasta(TOC33_Ath_fasta, as.string=TRUE, names= names(TOC33_Ath_fasta),
            file.out = "Funcional Groups/Ath_TOC33.fasta")
rename.fasta(infile = "Funcional Groups/Ath_TOC33.fasta", TOC33_Ath_2, outfile = "Funcional Groups/Ath_TOC33_2.fasta")

TOC33_Niben_Candidates <- read.table("Funcional Groups/Niben_Candidates_TOC33.txt")
TOC33_Niben_Candidates <- TOC33_Niben_Candidates[TOC33_Niben_Candidates$V3>=35,] %>% select(V2)
TOC33_Niben_Candidates <- unique(TOC33_Niben_Candidates)
TOC33_Niben_Candidates_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% TOC33_Niben_Candidates$V2))]
write.fasta(TOC33_Niben_Candidates_fasta, as.string=TRUE, names= names(TOC33_Niben_Candidates_fasta),
            file.out = "Funcional Groups/TOC33_Niben_Candidates_35.fasta")

TOC33_Sly_Candidates <- read.table("Funcional Groups/Sly_Candidates_TOC33.txt")
TOC33_Sly_Candidates <- TOC33_Sly_Candidates[TOC33_Sly_Candidates$V3>=35,] %>% select(V2)
TOC33_Sly_Candidates <- unique(TOC33_Sly_Candidates)
TOC33_Sly_Candidates_fasta <- Tomato_4.0_fasta[c(which(names(Tomato_4.0_fasta) %in% TOC33_Sly_Candidates$V2))]
write.fasta(TOC33_Sly_Candidates_fasta, as.string=TRUE, names= names(TOC33_Sly_Candidates_fasta),
            file.out = "Funcional Groups/TOC33_Sly_Candidates_35.fasta")

## Representing Heatmap

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
#Niben_TOC33_Ann <- read.table("Funcional Groups/FigTree/Niben_TOC33.txt",
#                                    sep = "\t", header = TRUE)
Niben_TOC33_Ann <- data.frame(Name= c("Niben261Chr08g0566009.1", "Niben261Chr01g0946016.1", "Niben261Chr14g0751012.1", "Niben261Chr06g0969009.1", "Niben261Chr06g0969009.1"),
                            Protein= c("TOC33a", "TOC33b", "TOC33c", "TOC33d", "TOC33e"))
colnames(Niben_TOC33_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_TOC33 <- merge(Niben_TOC33_Ann, TPM_All_FC_Annotated, by="Name")

## We select only the FC data from the different comparisons
Niben_TOC33 <- Niben_TOC33 %>% select(Protein, 88:105)
Niben_TOC33 <- Niben_TOC33[!duplicated(Niben_TOC33$Protein),]
Niben_TOC33 <- Niben_TOC33[order(match(Niben_TOC33[,1], Niben_TOC33_Ann[,2])),]

## We prepare the data for a heatmap and we map it
Niben_TOC33_Annotation <- merge(Niben_TOC33, Niben_TOC33_Ann, by="Name")
Niben_TOC33_Annotation <- Niben_TOC33_Annotation[!duplicated(Niben_TOC33_Annotation$Name),]
Niben_TOC33_Annotation <- Niben_TOC33_Annotation[order(match(Niben_TOC33_Annotation[,1], Niben_TOC33_Ann[,1])),]
#write.table(Niben_TOC33_Annotation, "Milan/RNAseq_Milan/Funcional Groups/TOC33_Niben261_OrthoVenn_DESeq2FC.txt", quote = FALSE, dec = ",", sep = "\t", row.names = FALSE)
rownames(Niben_TOC33_Annotation) <- Niben_TOC33_Annotation$Name
#Niben_TOC33_Chloroplast <- Niben_TOC33_Annotation %>% dplyr::filter(Presumed.subcellular.location=="Chloroplast")
#write.table(Niben_TOC33_Chloroplast, "Milan/RNAseq_Milan/Funcional Groups/TOC33_Chloroplast_Niben261_OrthoVenn_DESeq2FC.txt", quote = FALSE, dec = ",", sep = "\t", row.names = FALSE)
#Niben_TOC33_Chloroplast_Annotation <- Niben_TOC33_Chloroplast %>% select(Protein.names)
#Niben_TOC33_Chloroplast <- Niben_TOC33_Chloroplast %>% select(2:19)
Niben_TOC33_Annotation <- Niben_TOC33_Annotation %>% select(Protein)

rownames(Niben_TOC33) <- Niben_TOC33$Protein
Niben_TOC33[1] <- NULL

Niben_TOC33 <- Niben_TOC33 %>%
  select(pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

min(Niben_TOC33, na.rm = TRUE)
max(Niben_TOC33, na.rm = TRUE)
breaks_TOC33=c(seq(-1,0, length.out = 100), seq(0.01,1,length.out = 100))

color_TOC33 <- colorRampPalette(c("#415a20","#9bac12","#e5c033","#c38144","#894539"))(n=length(unique(Niben_TOC33_Annotation$Protein)))
names(color_TOC33) <- unique(Niben_TOC33_Annotation$Protein)
color_TOC33 <- list(Protein=color_TOC33)

pheatmap(Niben_TOC33, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_TOC33,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_TOC33_Annotation, annotation_colors=color_TOC33, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h", "96h"),
         cex = 1,
         angle_col=45,
         main = "TOC33",
         annotation_legend = TRUE,
         #display_numbers = Nb_Sly_Ath_MVA_Sig,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)


#########################################################################################################################################
####################################  Degrading  #############################################################################################
#########################################################################################################################################

Degrading_Ath <- read.table("Funcional Groups/Ath_Degrading.txt", header = TRUE)
Degrading_Ath <- merge(Degrading_Ath, Ath_Gene_Models, by.x="ID", by.y=1)
Degrading_Ath_fasta <- Ath11_fasta[c(which(names(Ath11_fasta) %in% Degrading_Ath$V2))]
Degrading_Ath_2 <- Degrading_Ath %>% select(V2, Name)
write.fasta(Degrading_Ath_fasta, as.string=TRUE, names= names(Degrading_Ath_fasta),
            file.out = "Funcional Groups/Ath_Degrading.fasta")
rename.fasta(infile = "Funcional Groups/Ath_Degrading.fasta", Degrading_Ath_2, outfile = "Funcional Groups/Ath_Degrading_2.fasta")

Degrading_Niben_Candidates <- read.table("Funcional Groups/Niben_Candidates_Degrading.txt")
Degrading_Niben_Candidates <- Degrading_Niben_Candidates[Degrading_Niben_Candidates$V3>=35,] %>% select(V2)
Degrading_Niben_Candidates <- unique(Degrading_Niben_Candidates)
Degrading_Niben_Candidates_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Degrading_Niben_Candidates$V2))]
write.fasta(Degrading_Niben_Candidates_fasta, as.string=TRUE, names= names(Degrading_Niben_Candidates_fasta),
            file.out = "Funcional Groups/Degrading_Niben_Candidates_35.fasta")

Degrading_Sly_Candidates <- read.table("Funcional Groups/Sly_Candidates_Degrading.txt")
Degrading_Sly_Candidates <- Degrading_Sly_Candidates[Degrading_Sly_Candidates$V3>=35,] %>% select(V2)
Degrading_Sly_Candidates <- unique(Degrading_Sly_Candidates)
Degrading_Sly_Candidates_fasta <- Tomato_4.0_fasta[c(which(names(Tomato_4.0_fasta) %in% Degrading_Sly_Candidates$V2))]
write.fasta(Degrading_Sly_Candidates_fasta, as.string=TRUE, names= names(Degrading_Sly_Candidates_fasta),
            file.out = "Funcional Groups/Degrading_Sly_Candidates_35.fasta")

## Representing Heatmap

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
#Niben_Degrading_Ann <- read.table("Funcional Groups/FigTree/Niben_Degrading.txt",
#                                    sep = "\t", header = TRUE)
Niben_Degrading_Ann <- data.frame(Name= c("Niben261Chr08g0566009.1", "Niben261Chr01g0946016.1", "Niben261Chr14g0751012.1", "Niben261Chr06g0969009.1", "Niben261Chr06g0969009.1"),
                              Protein= c("Degradinga", "Degradingb", "Degradingc", "Degradingd", "Degradinge"))
colnames(Niben_Degrading_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_Degrading <- merge(Niben_Degrading_Ann, TPM_All_FC_Annotated, by="Name")

## We select only the FC data from the different comparisons
Niben_Degrading <- Niben_Degrading %>% select(Protein, 88:105)
Niben_Degrading <- Niben_Degrading[!duplicated(Niben_Degrading$Protein),]
Niben_Degrading <- Niben_Degrading[order(match(Niben_Degrading[,1], Niben_Degrading_Ann[,2])),]

## We prepare the data for a heatmap and we map it
Niben_Degrading_Annotation <- merge(Niben_Degrading, Niben_Degrading_Ann, by="Name")
Niben_Degrading_Annotation <- Niben_Degrading_Annotation[!duplicated(Niben_Degrading_Annotation$Name),]
Niben_Degrading_Annotation <- Niben_Degrading_Annotation[order(match(Niben_Degrading_Annotation[,1], Niben_Degrading_Ann[,1])),]
#write.table(Niben_Degrading_Annotation, "Milan/RNAseq_Milan/Funcional Groups/Degrading_Niben261_OrthoVenn_DESeq2FC.txt", quote = FALSE, dec = ",", sep = "\t", row.names = FALSE)
rownames(Niben_Degrading_Annotation) <- Niben_Degrading_Annotation$Name
#Niben_Degrading_Chloroplast <- Niben_Degrading_Annotation %>% dplyr::filter(Presumed.subcellular.location=="Chloroplast")
#write.table(Niben_Degrading_Chloroplast, "Milan/RNAseq_Milan/Funcional Groups/Degrading_Chloroplast_Niben261_OrthoVenn_DESeq2FC.txt", quote = FALSE, dec = ",", sep = "\t", row.names = FALSE)
#Niben_Degrading_Chloroplast_Annotation <- Niben_Degrading_Chloroplast %>% select(Protein.names)
#Niben_Degrading_Chloroplast <- Niben_Degrading_Chloroplast %>% select(2:19)
Niben_Degrading_Annotation <- Niben_Degrading_Annotation %>% select(Protein)

rownames(Niben_Degrading) <- Niben_Degrading$Protein
Niben_Degrading[1] <- NULL

Niben_Degrading <- Niben_Degrading %>%
  select(pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

min(Niben_Degrading, na.rm = TRUE)
max(Niben_Degrading, na.rm = TRUE)
breaks_Degrading=c(seq(-1,0, length.out = 100), seq(0.01,1,length.out = 100))

color_Degrading <- colorRampPalette(c("#415a20","#9bac12","#e5c033","#c38144","#894539"))(n=length(unique(Niben_Degrading_Annotation$Protein)))
names(color_Degrading) <- unique(Niben_Degrading_Annotation$Protein)
color_Degrading <- list(Protein=color_Degrading)

pheatmap(Niben_Degrading, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Degrading,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_Degrading_Annotation, annotation_colors=color_Degrading, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h", "96h"),
         cex = 1,
         angle_col=45,
         main = "Degrading",
         annotation_legend = TRUE,
         #display_numbers = Nb_Sly_Ath_MVA_Sig,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)


#########################################################################################################################################
####################################  ESCRT  #############################################################################################
#########################################################################################################################################

ESCRT_Ath <- read.table("Funcional Groups/Ath_ESCRT.txt", header = TRUE)
ESCRT_Ath <- merge(ESCRT_Ath, Ath_Gene_Models, by.x="ID", by.y=1)
ESCRT_Ath_fasta <- Ath11_fasta[c(which(names(Ath11_fasta) %in% ESCRT_Ath$V2))]
ESCRT_Ath_2 <- ESCRT_Ath %>% select(V2, Name)
write.fasta(ESCRT_Ath_fasta, as.string=TRUE, names= names(ESCRT_Ath_fasta),
            file.out = "Funcional Groups/Ath_ESCRT.fasta")
rename.fasta(infile = "Funcional Groups/Ath_ESCRT.fasta", ESCRT_Ath_2, outfile = "Funcional Groups/Ath_ESCRT_2.fasta")

ESCRT_Niben_Candidates <- read.table("Funcional Groups/Niben_Candidates_ESCRT.txt")
ESCRT_Niben_Candidates <- ESCRT_Niben_Candidates[ESCRT_Niben_Candidates$V3>=35,] %>% select(V2)
ESCRT_Niben_Candidates <- unique(ESCRT_Niben_Candidates)
ESCRT_Niben_Candidates_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% ESCRT_Niben_Candidates$V2))]
write.fasta(ESCRT_Niben_Candidates_fasta, as.string=TRUE, names= names(ESCRT_Niben_Candidates_fasta),
            file.out = "Funcional Groups/ESCRT_Niben_Candidates_35.fasta")

ESCRT_Sly_Candidates <- read.table("Funcional Groups/Sly_Candidates_ESCRT.txt")
ESCRT_Sly_Candidates <- ESCRT_Sly_Candidates[ESCRT_Sly_Candidates$V3>=35,] %>% select(V2)
ESCRT_Sly_Candidates <- unique(ESCRT_Sly_Candidates)
ESCRT_Sly_Candidates_fasta <- Tomato_4.0_fasta[c(which(names(Tomato_4.0_fasta) %in% ESCRT_Sly_Candidates$V2))]
write.fasta(ESCRT_Sly_Candidates_fasta, as.string=TRUE, names= names(ESCRT_Sly_Candidates_fasta),
            file.out = "Funcional Groups/ESCRT_Sly_Candidates_35.fasta")

Niben_ESCRT_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_ESCRT_Ann$Name))]
write.fasta(Niben_ESCRT_fasta, as.string=TRUE, names= names(Niben_ESCRT_fasta),
            file.out = "Funcional Groups/Niben_ESCRT.fasta")
Niben_ESCRT_Ann_2 <- Niben_ESCRT_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_ESCRT.fasta", Niben_ESCRT_Ann, outfile = "Funcional Groups/Niben_ESCRT_2.fasta")

save_heatmaps("ESCRT", height = 24)

## Representing Heatmap

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Niben_ESCRT_Ann <- read.table("Funcional Groups/FigTree/Niben_ESCRT.txt",
                            sep = "\t", header = TRUE)

colnames(Niben_ESCRT_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_ESCRT <- merge(Niben_ESCRT_Ann, TPM_All_FC_Annotated, by="Name")

## We create the annotation for Modules

Niben_ESCRT_Modules <- which_Module_pwd30(Niben_ESCRT_Ann)

## We select only the FC data from the different comparisons
Niben_ESCRT_TPM <- Niben_ESCRT %>% select(Protein, 4:51, 61:66)
Niben_ESCRT_TPM <- Niben_ESCRT_TPM[!duplicated(Niben_ESCRT_TPM$Protein),]
Niben_ESCRT_TPM <- Niben_ESCRT_TPM[order(match(Niben_ESCRT_TPM[,1], Niben_ESCRT_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_ESCRT_FC <- Niben_ESCRT %>% select(Protein, 71, 73:87)
Niben_ESCRT_FC <- Niben_ESCRT_FC[!duplicated(Niben_ESCRT_FC$Protein),]
Niben_ESCRT_FC <- Niben_ESCRT_FC[order(match(Niben_ESCRT_FC[,1], Niben_ESCRT_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_ESCRT_DEG <- Niben_ESCRT %>% select(Protein, 89, 91:105)
Niben_ESCRT_DEG <- Niben_ESCRT_DEG[!duplicated(Niben_ESCRT_DEG$Protein),]
Niben_ESCRT_DEG <- Niben_ESCRT_DEG[order(match(Niben_ESCRT_DEG[,1], Niben_ESCRT_Ann[,2])),]

Niben_ESCRT_DEG$pcrtB22h_vs_GFP22h_filtered <- ifelse(is.na(Niben_ESCRT_DEG$pcrtB22h_vs_GFP22h_filtered) == TRUE, " ", "*")
Niben_ESCRT_DEG$pcrtB25h_vs_GFP25h_filtered <- ifelse(is.na(Niben_ESCRT_DEG$pcrtB25h_vs_GFP25h_filtered) == TRUE, " ", "*")
Niben_ESCRT_DEG$pcrtBAA_vs_GFPAA_filtered <- ifelse(is.na(Niben_ESCRT_DEG$pcrtBAA_vs_GFPAA_filtered) == TRUE, " ", "*")
Niben_ESCRT_DEG$pcrtBBB_vs_GFPBB_filtered <- ifelse(is.na(Niben_ESCRT_DEG$pcrtBBB_vs_GFPBB_filtered) == TRUE, " ", "*")
Niben_ESCRT_DEG$pcrtBYY_vs_GFPYY_filtered <- ifelse(is.na(Niben_ESCRT_DEG$pcrtBYY_vs_GFPYY_filtered) == TRUE, " ", "*")
Niben_ESCRT_DEG$pcrtB28h_vs_GFP28h_filtered <- ifelse(is.na(Niben_ESCRT_DEG$pcrtB28h_vs_GFP28h_filtered) == TRUE, " ", "*")
Niben_ESCRT_DEG$pcrtB34h_vs_GFP34h_filtered <- ifelse(is.na(Niben_ESCRT_DEG$pcrtB34h_vs_GFP34h_filtered) == TRUE, " ", "*")
Niben_ESCRT_DEG$pcrtB37h_vs_GFP37h_filtered <- ifelse(is.na(Niben_ESCRT_DEG$pcrtB37h_vs_GFP37h_filtered) == TRUE, " ", "*")
Niben_ESCRT_DEG$pcrtBZZ_vs_GFPZZ_filtered <- ifelse(is.na(Niben_ESCRT_DEG$pcrtBZZ_vs_GFPZZ_filtered) == TRUE, " ", "*")
Niben_ESCRT_DEG$pcrtBCC_vs_GFPCC_filtered <- ifelse(is.na(Niben_ESCRT_DEG$pcrtBCC_vs_GFPCC_filtered) == TRUE, " ", "*")
Niben_ESCRT_DEG$pcrtB40h_vs_GFP40h_filtered <- ifelse(is.na(Niben_ESCRT_DEG$pcrtB40h_vs_GFP40h_filtered) == TRUE, " ", "*")
Niben_ESCRT_DEG$pcrtBDD_vs_GFPDD_filtered <- ifelse(is.na(Niben_ESCRT_DEG$pcrtBDD_vs_GFPDD_filtered) == TRUE, " ", "*")
Niben_ESCRT_DEG$pcrtB46h_vs_GFP46h_filtered <- ifelse(is.na(Niben_ESCRT_DEG$pcrtB46h_vs_GFP46h_filtered) == TRUE, " ", "*")
Niben_ESCRT_DEG$pcrtB56h_vs_GFP56h_filtered <- ifelse(is.na(Niben_ESCRT_DEG$pcrtB56h_vs_GFP56h_filtered) == TRUE, " ", "*")
Niben_ESCRT_DEG$pcrtBEE_vs_GFPEE_filtered <- ifelse(is.na(Niben_ESCRT_DEG$pcrtBEE_vs_GFPEE_filtered) == TRUE, " ", "*")
Niben_ESCRT_DEG$pcrtB96h_vs_GFP96h_filtered <- ifelse(is.na(Niben_ESCRT_DEG$pcrtB96h_vs_GFP96h_filtered) == TRUE, " ", "*")
row.names(Niben_ESCRT_DEG) <- Niben_ESCRT_DEG$Protein
Niben_ESCRT_DEG[1] <- NULL

Niben_ESCRT_DEG <- Niben_ESCRT_DEG %>%
  select(pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

row.names(Niben_ESCRT_FC) <- Niben_ESCRT_FC$Protein
Niben_ESCRT_FC[1] <- NULL

Niben_ESCRT_FC <- Niben_ESCRT_FC %>%
  select(pcrtB22h_vs_GFP22h_All, pcrtBAA_vs_GFPAA_All, pcrtB25h_vs_GFP25h_All, 
         pcrtBYY_vs_GFPYY_All, pcrtBBB_vs_GFPBB_All, pcrtB28h_vs_GFP28h_All, pcrtB34h_vs_GFP34h_All, pcrtBZZ_vs_GFPZZ_All, 
         pcrtB37h_vs_GFP37h_All, pcrtBCC_vs_GFPCC_All, pcrtB40h_vs_GFP40h_All, pcrtBDD_vs_GFPDD_All, pcrtB46h_vs_GFP46h_All, 
         pcrtBEE_vs_GFPEE_All, pcrtB56h_vs_GFP56h_All, pcrtB96h_vs_GFP96h_All)


min(Niben_ESCRT_FC, na.rm = TRUE)
max(Niben_ESCRT_FC, na.rm = TRUE)
breaks_ESCRT_FC=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Niben_ESCRT_FC %>% select(1:15), 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_ESCRT_FC,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_ESCRT_Annotation, annotation_colors=color_ESCRT, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h"),#, "96h"),
         cex = 1,
         angle_col=45,
         main = "ESCRT",
         annotation_legend = TRUE,
         display_numbers = Niben_ESCRT_DEG %>% select(1:15),
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(84, 104),
         annotation_row = Niben_ESCRT_Modules,
         annotation_colors = module_colors
         #gaps_col = c(15)
)


rownames(Niben_ESCRT_TPM) <- Niben_ESCRT_TPM$Protein
Niben_ESCRT_TPM[1] <- NULL

min(Niben_ESCRT_TPM, na.rm = TRUE)
max(Niben_ESCRT_TPM, na.rm = TRUE)
breaks_Niben_ESCRT_TPM=c(seq(0,100, length.out = 100))

Niben_ESCRT_TPM_norm <- normalize_df(Niben_ESCRT_TPM)
Niben_ESCRT_TPM_norm_Cornell <- Niben_ESCRT_TPM %>%
  select(1:48) 
Niben_ESCRT_TPM_norm_Cornell <- normalize_df(Niben_ESCRT_TPM_norm_Cornell)


pheatmap(Niben_ESCRT_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Niben_ESCRT_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_ESCRT_Annotation, annotation_colors=color_ESCRT, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "ESCRT",
         annotation_legend = TRUE,
         #display_numbers = Niben_ESCRT_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         annotation_row = Niben_ESCRT_Modules,
         annotation_colors = module_colors,
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(24, 48, 51)
)



#####Representing Heatmap for Sly

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Sly_ESCRT_Ann <- read.table("Funcional Groups/FigTree/Sly_ESCRT.txt",
                          sep = "\t", header = TRUE)

colnames(Sly_ESCRT_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Sly_ESCRT <- merge(Sly_ESCRT_Ann, Summary_Yazdani_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Sly_ESCRT_TPM <- Sly_ESCRT %>% select(Protein, 4:66)
Sly_ESCRT_TPM <- Sly_ESCRT_TPM[!duplicated(Sly_ESCRT_TPM$Protein),]
Sly_ESCRT_TPM <- Sly_ESCRT_TPM[order(match(Sly_ESCRT_TPM[,1], Sly_ESCRT_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_ESCRT_FC <- Sly_ESCRT %>% select(Protein, 67:71)
Sly_ESCRT_FC <- Sly_ESCRT_FC[!duplicated(Sly_ESCRT_FC$Protein),]
Sly_ESCRT_FC <- Sly_ESCRT_FC[order(match(Sly_ESCRT_FC[,1], Sly_ESCRT_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_ESCRT_DEG <- Sly_ESCRT %>% select(Protein, 72:76)
Sly_ESCRT_DEG <- Sly_ESCRT_DEG[!duplicated(Sly_ESCRT_DEG$Protein),]
Sly_ESCRT_DEG <- Sly_ESCRT_DEG[order(match(Sly_ESCRT_DEG[,1], Sly_ESCRT_Ann[,2])),]

Sly_ESCRT_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Sly_ESCRT_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Sly_ESCRT_DEG$Ripening_Orange_DEG <- ifelse(is.na(Sly_ESCRT_DEG$Ripening_Orange_DEG) == TRUE, " ", "*")
Sly_ESCRT_DEG$Ripening_Red_DEG <- ifelse(is.na(Sly_ESCRT_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
Sly_ESCRT_DEG$ORHis_MG_DEG <- ifelse(is.na(Sly_ESCRT_DEG$ORHis_MG_DEG) == TRUE, " ", "*")
Sly_ESCRT_DEG$ORHis_Breaker_DEG <- ifelse(is.na(Sly_ESCRT_DEG$ORHis_Breaker_DEG) == TRUE, " ", "*")
row.names(Sly_ESCRT_DEG) <- Sly_ESCRT_DEG$Protein
Sly_ESCRT_DEG[1] <- NULL

rownames(Sly_ESCRT_FC) <- Sly_ESCRT_FC$Protein
Sly_ESCRT_FC[1] <- NULL

min(Sly_ESCRT_FC, na.rm = TRUE)
max(Sly_ESCRT_FC, na.rm = TRUE)
breaks_Sly_ESCRT=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Sly_ESCRT_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Sly_ESCRT,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_ESCRT_Annotation, annotation_colors=color_ESCRT, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "ESCRT",
         annotation_legend = TRUE,
         display_numbers = Sly_ESCRT_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(39, 48),
         gaps_col = c(3)
)


rownames(Sly_ESCRT_TPM) <- Sly_ESCRT_TPM$Protein
Sly_ESCRT_TPM[1] <- NULL

min(Sly_ESCRT_TPM, na.rm = TRUE)
max(Sly_ESCRT_TPM, na.rm = TRUE)
breaks_Sly_ESCRT_TPM=c(seq(0,100, length.out = 200))

Sly_ESCRT_TPM_norm <- normalize_df(Sly_ESCRT_TPM)

pheatmap(Sly_ESCRT_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Sly_ESCRT_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_ESCRT_Annotation, annotation_colors=color_ESCRT, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "ESCRT",
         annotation_legend = TRUE,
         #display_numbers = Sly_ESCRT_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(21,42)
)


#####Representing Heatmap for Cann

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Cann_ESCRT_Ann <- read.table("Funcional Groups/FigTree/Cann_ESCRT.txt",
                           sep = "\t", header = TRUE)

colnames(Cann_ESCRT_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Cann_ESCRT <- merge(Cann_ESCRT_Ann, Summary_Pepper_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Cann_ESCRT_TPM <- Cann_ESCRT %>% select(Protein, 4:17)
Cann_ESCRT_TPM <- Cann_ESCRT_TPM[!duplicated(Cann_ESCRT_TPM$Protein),]
Cann_ESCRT_TPM <- Cann_ESCRT_TPM[order(match(Cann_ESCRT_TPM[,1], Cann_ESCRT_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_ESCRT_FC <- Cann_ESCRT %>% select(Protein, 18:19)
Cann_ESCRT_FC <- Cann_ESCRT_FC[!duplicated(Cann_ESCRT_FC$Protein),]
Cann_ESCRT_FC <- Cann_ESCRT_FC[order(match(Cann_ESCRT_FC[,1], Cann_ESCRT_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_ESCRT_DEG <- Cann_ESCRT %>% select(Protein, 20:21)
Cann_ESCRT_DEG <- Cann_ESCRT_DEG[!duplicated(Cann_ESCRT_DEG$Protein),]
Cann_ESCRT_DEG <- Cann_ESCRT_DEG[order(match(Cann_ESCRT_DEG[,1], Cann_ESCRT_Ann[,2])),]

Cann_ESCRT_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Cann_ESCRT_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Cann_ESCRT_DEG$Ripening_Red_DEG <- ifelse(is.na(Cann_ESCRT_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
row.names(Cann_ESCRT_DEG) <- Cann_ESCRT_DEG$Protein
Cann_ESCRT_DEG[1] <- NULL

rownames(Cann_ESCRT_FC) <- Cann_ESCRT_FC$Protein
Cann_ESCRT_FC[1] <- NULL

min(Cann_ESCRT_FC, na.rm = TRUE)
max(Cann_ESCRT_FC, na.rm = TRUE)
breaks_Cann_ESCRT=c(seq(-4.5,0, length.out = 100), seq(0.01,6,length.out = 100))

pheatmap(Cann_ESCRT_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Cann_ESCRT,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_ESCRT_Annotation, annotation_colors=color_ESCRT, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Red"),
         cex = 1,
         angle_col=45,
         main = "ESCRT",
         annotation_legend = TRUE,
         display_numbers = Cann_ESCRT_DEG,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)


rownames(Cann_ESCRT_TPM) <- Cann_ESCRT_TPM$Protein
Cann_ESCRT_TPM[1] <- NULL

min(Cann_ESCRT_TPM, na.rm = TRUE)
max(Cann_ESCRT_TPM, na.rm = TRUE)
breaks_Cann_ESCRT_TPM=c(seq(0,100, length.out = 200))

Cann_ESCRT_TPM_norm <- normalize_df(Cann_ESCRT_TPM)

pheatmap(Cann_ESCRT_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Cann_ESCRT_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_ESCRT_Annotation, annotation_colors=color_ESCRT, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "ESCRT",
         annotation_legend = TRUE,
         #display_numbers = Cann_ESCRT_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(4,9)
)


#########################################################################################################################################
####################################  TICTOC  #############################################################################################
#########################################################################################################################################

TICTOC_Ath <- read.table("Funcional Groups/Ath_TICTOC.txt", header = TRUE)
TICTOC_Ath <- merge(TICTOC_Ath, Ath_Gene_Models, by.x="ID", by.y=1)
TICTOC_Ath_fasta <- Ath11_fasta[c(which(names(Ath11_fasta) %in% TICTOC_Ath$V2))]
TICTOC_Ath_2 <- TICTOC_Ath %>% select(V2, Name)
write.fasta(TICTOC_Ath_fasta, as.string=TRUE, names= names(TICTOC_Ath_fasta),
            file.out = "Funcional Groups/Ath_TICTOC.fasta")
rename.fasta(infile = "Funcional Groups/Ath_TICTOC.fasta", TICTOC_Ath_2, outfile = "Funcional Groups/Ath_TICTOC_2.fasta")

TICTOC_Niben_Candidates <- read.table("Funcional Groups/Niben_Candidates_TICTOC.txt")
TICTOC_Niben_Candidates <- TICTOC_Niben_Candidates[TICTOC_Niben_Candidates$V3>=35,] %>% select(V2)
TICTOC_Niben_Candidates <- unique(TICTOC_Niben_Candidates)
TICTOC_Niben_Candidates_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% TICTOC_Niben_Candidates$V2))]
write.fasta(TICTOC_Niben_Candidates_fasta, as.string=TRUE, names= names(TICTOC_Niben_Candidates_fasta),
            file.out = "Funcional Groups/TICTOC_Niben_Candidates_35.fasta")

TICTOC_Sly_Candidates <- read.table("Funcional Groups/Sly_Candidates_TICTOC.txt")
TICTOC_Sly_Candidates <- TICTOC_Sly_Candidates[TICTOC_Sly_Candidates$V3>=35,] %>% select(V2)
TICTOC_Sly_Candidates <- unique(TICTOC_Sly_Candidates)
TICTOC_Sly_Candidates_fasta <- Tomato_4.0_fasta[c(which(names(Tomato_4.0_fasta) %in% TICTOC_Sly_Candidates$V2))]
write.fasta(TICTOC_Sly_Candidates_fasta, as.string=TRUE, names= names(TICTOC_Sly_Candidates_fasta),
            file.out = "Funcional Groups/TICTOC_Sly_Candidates_35.fasta")

Niben_TICTOC_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_TICTOC_Ann$Name))]
write.fasta(Niben_TICTOC_fasta, as.string=TRUE, names= names(Niben_TICTOC_fasta),
            file.out = "Funcional Groups/Niben_TICTOC.fasta")
Niben_TICTOC_Ann_2 <- Niben_TICTOC_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_TICTOC.fasta", Niben_TICTOC_Ann, outfile = "Funcional Groups/Niben_TICTOC_2.fasta")

save_heatmaps("TICTOC", height = 18)

## Representing Heatmap

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Niben_TICTOC_Ann <- read.table("Funcional Groups/FigTree/Niben_TICTOC.txt",
                              sep = "\t", header = TRUE)

colnames(Niben_TICTOC_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_TICTOC <- merge(Niben_TICTOC_Ann, TPM_All_FC_Annotated, by="Name")

## We create the annotation for Modules

Niben_TICTOC_Modules <- which_Module_pwd30(Niben_TICTOC_Ann)

## We select only the FC data from the different comparisons
Niben_TICTOC_TPM <- Niben_TICTOC %>% select(Protein, 4:51, 61:66)
Niben_TICTOC_TPM <- Niben_TICTOC_TPM[!duplicated(Niben_TICTOC_TPM$Protein),]
Niben_TICTOC_TPM <- Niben_TICTOC_TPM[order(match(Niben_TICTOC_TPM[,1], Niben_TICTOC_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_TICTOC_FC <- Niben_TICTOC %>% select(Protein, 71, 73:87)
Niben_TICTOC_FC <- Niben_TICTOC_FC[!duplicated(Niben_TICTOC_FC$Protein),]
Niben_TICTOC_FC <- Niben_TICTOC_FC[order(match(Niben_TICTOC_FC[,1], Niben_TICTOC_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_TICTOC_DEG <- Niben_TICTOC %>% select(Protein, 89, 91:105)
Niben_TICTOC_DEG <- Niben_TICTOC_DEG[!duplicated(Niben_TICTOC_DEG$Protein),]
Niben_TICTOC_DEG <- Niben_TICTOC_DEG[order(match(Niben_TICTOC_DEG[,1], Niben_TICTOC_Ann[,2])),]

Niben_TICTOC_DEG$pcrtB22h_vs_GFP22h_filtered <- ifelse(is.na(Niben_TICTOC_DEG$pcrtB22h_vs_GFP22h_filtered) == TRUE, " ", "*")
Niben_TICTOC_DEG$pcrtB25h_vs_GFP25h_filtered <- ifelse(is.na(Niben_TICTOC_DEG$pcrtB25h_vs_GFP25h_filtered) == TRUE, " ", "*")
Niben_TICTOC_DEG$pcrtBAA_vs_GFPAA_filtered <- ifelse(is.na(Niben_TICTOC_DEG$pcrtBAA_vs_GFPAA_filtered) == TRUE, " ", "*")
Niben_TICTOC_DEG$pcrtBBB_vs_GFPBB_filtered <- ifelse(is.na(Niben_TICTOC_DEG$pcrtBBB_vs_GFPBB_filtered) == TRUE, " ", "*")
Niben_TICTOC_DEG$pcrtBYY_vs_GFPYY_filtered <- ifelse(is.na(Niben_TICTOC_DEG$pcrtBYY_vs_GFPYY_filtered) == TRUE, " ", "*")
Niben_TICTOC_DEG$pcrtB28h_vs_GFP28h_filtered <- ifelse(is.na(Niben_TICTOC_DEG$pcrtB28h_vs_GFP28h_filtered) == TRUE, " ", "*")
Niben_TICTOC_DEG$pcrtB34h_vs_GFP34h_filtered <- ifelse(is.na(Niben_TICTOC_DEG$pcrtB34h_vs_GFP34h_filtered) == TRUE, " ", "*")
Niben_TICTOC_DEG$pcrtB37h_vs_GFP37h_filtered <- ifelse(is.na(Niben_TICTOC_DEG$pcrtB37h_vs_GFP37h_filtered) == TRUE, " ", "*")
Niben_TICTOC_DEG$pcrtBZZ_vs_GFPZZ_filtered <- ifelse(is.na(Niben_TICTOC_DEG$pcrtBZZ_vs_GFPZZ_filtered) == TRUE, " ", "*")
Niben_TICTOC_DEG$pcrtBCC_vs_GFPCC_filtered <- ifelse(is.na(Niben_TICTOC_DEG$pcrtBCC_vs_GFPCC_filtered) == TRUE, " ", "*")
Niben_TICTOC_DEG$pcrtB40h_vs_GFP40h_filtered <- ifelse(is.na(Niben_TICTOC_DEG$pcrtB40h_vs_GFP40h_filtered) == TRUE, " ", "*")
Niben_TICTOC_DEG$pcrtBDD_vs_GFPDD_filtered <- ifelse(is.na(Niben_TICTOC_DEG$pcrtBDD_vs_GFPDD_filtered) == TRUE, " ", "*")
Niben_TICTOC_DEG$pcrtB46h_vs_GFP46h_filtered <- ifelse(is.na(Niben_TICTOC_DEG$pcrtB46h_vs_GFP46h_filtered) == TRUE, " ", "*")
Niben_TICTOC_DEG$pcrtB56h_vs_GFP56h_filtered <- ifelse(is.na(Niben_TICTOC_DEG$pcrtB56h_vs_GFP56h_filtered) == TRUE, " ", "*")
Niben_TICTOC_DEG$pcrtBEE_vs_GFPEE_filtered <- ifelse(is.na(Niben_TICTOC_DEG$pcrtBEE_vs_GFPEE_filtered) == TRUE, " ", "*")
Niben_TICTOC_DEG$pcrtB96h_vs_GFP96h_filtered <- ifelse(is.na(Niben_TICTOC_DEG$pcrtB96h_vs_GFP96h_filtered) == TRUE, " ", "*")
row.names(Niben_TICTOC_DEG) <- Niben_TICTOC_DEG$Protein
Niben_TICTOC_DEG[1] <- NULL

Niben_TICTOC_DEG <- Niben_TICTOC_DEG %>%
  select(pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

row.names(Niben_TICTOC_FC) <- Niben_TICTOC_FC$Protein
Niben_TICTOC_FC[1] <- NULL

Niben_TICTOC_FC <- Niben_TICTOC_FC %>%
  select(pcrtB22h_vs_GFP22h_All, pcrtBAA_vs_GFPAA_All, pcrtB25h_vs_GFP25h_All, 
         pcrtBYY_vs_GFPYY_All, pcrtBBB_vs_GFPBB_All, pcrtB28h_vs_GFP28h_All, pcrtB34h_vs_GFP34h_All, pcrtBZZ_vs_GFPZZ_All, 
         pcrtB37h_vs_GFP37h_All, pcrtBCC_vs_GFPCC_All, pcrtB40h_vs_GFP40h_All, pcrtBDD_vs_GFPDD_All, pcrtB46h_vs_GFP46h_All, 
         pcrtBEE_vs_GFPEE_All, pcrtB56h_vs_GFP56h_All, pcrtB96h_vs_GFP96h_All)


min(Niben_TICTOC_FC, na.rm = TRUE)
max(Niben_TICTOC_FC, na.rm = TRUE)
breaks_TICTOC_FC=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Niben_TICTOC_FC, # %>% select(1:15), 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_TICTOC_FC,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_TICTOC_Annotation, annotation_colors=color_TICTOC, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h", "96h"),
         cex = 1,
         angle_col=45,
         main = "TICTOC",
         annotation_legend = TRUE,
         display_numbers = Niben_TICTOC_DEG, # %>% select(1:15),
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(84, 104),
         annotation_row = Niben_TICTOC_Modules,
         annotation_colors = module_colors
         #gaps_col = c(15)
)

Niben_TICTOC_FC_full <- Niben_TICTOC_FC[complete.cases(Niben_TICTOC_FC),]
Niben_TICTOC_DEG_full <- Niben_TICTOC_DEG[complete.cases(Niben_TICTOC_FC),]

pheatmap(Niben_TICTOC_FC_full %>% select(1:15), 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_TICTOC_FC,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_TICTOC_Annotation, annotation_colors=color_TICTOC, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h"), #, "96h"),
         cex = 1,
         angle_col=45,
         main = "TICTOC",
         annotation_legend = TRUE,
         display_numbers = Niben_TICTOC_DEG_full %>% select(1:15),
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(84, 104),
         annotation_row = Niben_TICTOC_Modules,
         annotation_colors = module_colors
         #gaps_col = c(15)
)


rownames(Niben_TICTOC_TPM) <- Niben_TICTOC_TPM$Protein
Niben_TICTOC_TPM[1] <- NULL

min(Niben_TICTOC_TPM, na.rm = TRUE)
max(Niben_TICTOC_TPM, na.rm = TRUE)
breaks_Niben_TICTOC_TPM=c(seq(0,100, length.out = 100))

Niben_TICTOC_TPM_norm <- normalize_df(Niben_TICTOC_TPM)
Niben_TICTOC_TPM_norm_Cornell <- Niben_TICTOC_TPM %>%
  select(1:48) 
Niben_TICTOC_TPM_norm_Cornell <- normalize_df(Niben_TICTOC_TPM_norm_Cornell)


pheatmap(Niben_TICTOC_TPM_norm_Cornell, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Niben_TICTOC_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_TICTOC_Annotation, annotation_colors=color_TICTOC, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "TICTOC",
         annotation_legend = TRUE,
         #display_numbers = Niben_TICTOC_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         annotation_row = Niben_TICTOC_Modules,
         annotation_colors = module_colors,
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(24)#, 48, 51)
)



#####Representing Heatmap for Sly

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Sly_TICTOC_Ann <- read.table("Funcional Groups/FigTree/Sly_TICTOC.txt",
                            sep = "\t", header = TRUE)

colnames(Sly_TICTOC_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Sly_TICTOC <- merge(Sly_TICTOC_Ann, Summary_Yazdani_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Sly_TICTOC_TPM <- Sly_TICTOC %>% select(Protein, 4:66)
Sly_TICTOC_TPM <- Sly_TICTOC_TPM[!duplicated(Sly_TICTOC_TPM$Protein),]
Sly_TICTOC_TPM <- Sly_TICTOC_TPM[order(match(Sly_TICTOC_TPM[,1], Sly_TICTOC_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_TICTOC_FC <- Sly_TICTOC %>% select(Protein, 67:71)
Sly_TICTOC_FC <- Sly_TICTOC_FC[!duplicated(Sly_TICTOC_FC$Protein),]
Sly_TICTOC_FC <- Sly_TICTOC_FC[order(match(Sly_TICTOC_FC[,1], Sly_TICTOC_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_TICTOC_DEG <- Sly_TICTOC %>% select(Protein, 72:76)
Sly_TICTOC_DEG <- Sly_TICTOC_DEG[!duplicated(Sly_TICTOC_DEG$Protein),]
Sly_TICTOC_DEG <- Sly_TICTOC_DEG[order(match(Sly_TICTOC_DEG[,1], Sly_TICTOC_Ann[,2])),]

Sly_TICTOC_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Sly_TICTOC_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Sly_TICTOC_DEG$Ripening_Orange_DEG <- ifelse(is.na(Sly_TICTOC_DEG$Ripening_Orange_DEG) == TRUE, " ", "*")
Sly_TICTOC_DEG$Ripening_Red_DEG <- ifelse(is.na(Sly_TICTOC_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
Sly_TICTOC_DEG$ORHis_MG_DEG <- ifelse(is.na(Sly_TICTOC_DEG$ORHis_MG_DEG) == TRUE, " ", "*")
Sly_TICTOC_DEG$ORHis_Breaker_DEG <- ifelse(is.na(Sly_TICTOC_DEG$ORHis_Breaker_DEG) == TRUE, " ", "*")
row.names(Sly_TICTOC_DEG) <- Sly_TICTOC_DEG$Protein
Sly_TICTOC_DEG[1] <- NULL

rownames(Sly_TICTOC_FC) <- Sly_TICTOC_FC$Protein
Sly_TICTOC_FC[1] <- NULL

min(Sly_TICTOC_FC, na.rm = TRUE)
max(Sly_TICTOC_FC, na.rm = TRUE)
breaks_Sly_TICTOC=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Sly_TICTOC_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Sly_TICTOC,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_TICTOC_Annotation, annotation_colors=color_TICTOC, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "TICTOC",
         annotation_legend = TRUE,
         display_numbers = Sly_TICTOC_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(39, 48),
         gaps_col = c(3)
)


rownames(Sly_TICTOC_TPM) <- Sly_TICTOC_TPM$Protein
Sly_TICTOC_TPM[1] <- NULL

min(Sly_TICTOC_TPM, na.rm = TRUE)
max(Sly_TICTOC_TPM, na.rm = TRUE)
breaks_Sly_TICTOC_TPM=c(seq(0,100, length.out = 200))

Sly_TICTOC_TPM_norm <- normalize_df(Sly_TICTOC_TPM)

pheatmap(Sly_TICTOC_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Sly_TICTOC_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_TICTOC_Annotation, annotation_colors=color_TICTOC, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "TICTOC",
         annotation_legend = TRUE,
         #display_numbers = Sly_TICTOC_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(21,42)
)


#####Representing Heatmap for Cann

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Cann_TICTOC_Ann <- read.table("Funcional Groups/FigTree/Cann_TICTOC.txt",
                             sep = "\t", header = TRUE)

colnames(Cann_TICTOC_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Cann_TICTOC <- merge(Cann_TICTOC_Ann, Summary_Pepper_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Cann_TICTOC_TPM <- Cann_TICTOC %>% select(Protein, 4:17)
Cann_TICTOC_TPM <- Cann_TICTOC_TPM[!duplicated(Cann_TICTOC_TPM$Protein),]
Cann_TICTOC_TPM <- Cann_TICTOC_TPM[order(match(Cann_TICTOC_TPM[,1], Cann_TICTOC_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_TICTOC_FC <- Cann_TICTOC %>% select(Protein, 18:19)
Cann_TICTOC_FC <- Cann_TICTOC_FC[!duplicated(Cann_TICTOC_FC$Protein),]
Cann_TICTOC_FC <- Cann_TICTOC_FC[order(match(Cann_TICTOC_FC[,1], Cann_TICTOC_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_TICTOC_DEG <- Cann_TICTOC %>% select(Protein, 20:21)
Cann_TICTOC_DEG <- Cann_TICTOC_DEG[!duplicated(Cann_TICTOC_DEG$Protein),]
Cann_TICTOC_DEG <- Cann_TICTOC_DEG[order(match(Cann_TICTOC_DEG[,1], Cann_TICTOC_Ann[,2])),]

Cann_TICTOC_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Cann_TICTOC_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Cann_TICTOC_DEG$Ripening_Red_DEG <- ifelse(is.na(Cann_TICTOC_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
row.names(Cann_TICTOC_DEG) <- Cann_TICTOC_DEG$Protein
Cann_TICTOC_DEG[1] <- NULL

rownames(Cann_TICTOC_FC) <- Cann_TICTOC_FC$Protein
Cann_TICTOC_FC[1] <- NULL

min(Cann_TICTOC_FC, na.rm = TRUE)
max(Cann_TICTOC_FC, na.rm = TRUE)
breaks_Cann_TICTOC=c(seq(-4.5,0, length.out = 100), seq(0.01,6,length.out = 100))

pheatmap(Cann_TICTOC_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Cann_TICTOC,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_TICTOC_Annotation, annotation_colors=color_TICTOC, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Red"),
         cex = 1,
         angle_col=45,
         main = "TICTOC",
         annotation_legend = TRUE,
         display_numbers = Cann_TICTOC_DEG,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)


rownames(Cann_TICTOC_TPM) <- Cann_TICTOC_TPM$Protein
Cann_TICTOC_TPM[1] <- NULL

min(Cann_TICTOC_TPM, na.rm = TRUE)
max(Cann_TICTOC_TPM, na.rm = TRUE)
breaks_Cann_TICTOC_TPM=c(seq(0,100, length.out = 200))

Cann_TICTOC_TPM_norm <- normalize_df(Cann_TICTOC_TPM)

pheatmap(Cann_TICTOC_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Cann_TICTOC_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_TICTOC_Annotation, annotation_colors=color_TICTOC, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "TICTOC",
         annotation_legend = TRUE,
         #display_numbers = Cann_TICTOC_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(4,9)
)

#########################################################################################################################################
####################################  ChloroBio  #############################################################################################
#########################################################################################################################################

ChloroBio_Ath <- read.table("Funcional Groups/Ath_ChloroBio.txt", header = TRUE)
ChloroBio_Ath <- merge(ChloroBio_Ath, Ath_Gene_Models, by.x="ID", by.y=1)
ChloroBio_Ath_fasta <- Ath11_fasta[c(which(names(Ath11_fasta) %in% ChloroBio_Ath$V2))]
ChloroBio_Ath_2 <- ChloroBio_Ath %>% select(V2, Name)
write.fasta(ChloroBio_Ath_fasta, as.string=TRUE, names= names(ChloroBio_Ath_fasta),
            file.out = "Funcional Groups/Ath_ChloroBio.fasta")
rename.fasta(infile = "Funcional Groups/Ath_ChloroBio.fasta", ChloroBio_Ath_2, outfile = "Funcional Groups/Ath_ChloroBio_2.fasta")

ChloroBio_Niben_Candidates <- read.table("Funcional Groups/Niben_Candidates_ChloroBio.txt")
ChloroBio_Niben_Candidates <- ChloroBio_Niben_Candidates[ChloroBio_Niben_Candidates$V3>=35,] %>% select(V2)
ChloroBio_Niben_Candidates <- unique(ChloroBio_Niben_Candidates)
ChloroBio_Niben_Candidates_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% ChloroBio_Niben_Candidates$V2))]
write.fasta(ChloroBio_Niben_Candidates_fasta, as.string=TRUE, names= names(ChloroBio_Niben_Candidates_fasta),
            file.out = "Funcional Groups/ChloroBio_Niben_Candidates_35.fasta")

ChloroBio_Sly_Candidates <- read.table("Funcional Groups/Sly_Candidates_ChloroBio.txt")
ChloroBio_Sly_Candidates <- ChloroBio_Sly_Candidates[ChloroBio_Sly_Candidates$V3>=35,] %>% select(V2)
ChloroBio_Sly_Candidates <- unique(ChloroBio_Sly_Candidates)
ChloroBio_Sly_Candidates_fasta <- Tomato_4.0_fasta[c(which(names(Tomato_4.0_fasta) %in% ChloroBio_Sly_Candidates$V2))]
write.fasta(ChloroBio_Sly_Candidates_fasta, as.string=TRUE, names= names(ChloroBio_Sly_Candidates_fasta),
            file.out = "Funcional Groups/ChloroBio_Sly_Candidates_35.fasta")

Niben_ChloroBio_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_ChloroBio_Ann$Name))]
write.fasta(Niben_ChloroBio_fasta, as.string=TRUE, names= names(Niben_ChloroBio_fasta),
            file.out = "Funcional Groups/Niben_ChloroBio.fasta")
Niben_ChloroBio_Ann_2 <- Niben_ChloroBio_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_ChloroBio.fasta", Niben_ChloroBio_Ann, outfile = "Funcional Groups/Niben_ChloroBio_2.fasta")

save_heatmaps("ChloroBio", height = 8)

## Representing Heatmap

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Niben_ChloroBio_Ann <- read.table("Funcional Groups/FigTree/Niben_ChloroBio.txt",
                            sep = "\t", header = TRUE)

colnames(Niben_ChloroBio_Ann) <- c("Name", "Protein")

Niben_ChloroBio_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_ChloroBio_Ann$Name))]
write.fasta(Niben_ChloroBio_fasta, as.string=TRUE, names= names(Niben_ChloroBio_fasta),
            file.out = "Funcional Groups/Niben_ChloroBio.fasta")
Niben_ChloroBio_Ann_2 <- Niben_ChloroBio_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_ChloroBio.fasta", Niben_ChloroBio_Ann, outfile = "Funcional Groups/Niben_ChloroBio_2.fasta")

Niben_ChloroBio_Chloro <- read.delim("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan/Funcional Groups/Niben_ChloroBio_2_summary.targetp2", 
                               header=FALSE, comment.char="#")
colnames(Niben_ChloroBio_Chloro) <- c("ID",	"Prediction",	"noTP",	"SP",	"mTP",	"cTP",	"luTP",	"CS-Position")
Niben_ChloroBio_Chloro$Prediction <- factor(Niben_ChloroBio_Chloro$Prediction, levels = c("cTP","mTP","noTP"))
Niben_ChloroBio_Chloro <- merge(Niben_ChloroBio_Ann, Niben_ChloroBio_Chloro, by.x = "Protein", by.y = "ID")

Niben_ChloroBio_Chloro <- Niben_ChloroBio_Chloro[order(match(Niben_ChloroBio_Chloro$Protein, Niben_ChloroBio_Ann$Protein)),]
rownames(Niben_ChloroBio_Chloro) <- Niben_ChloroBio_Chloro$Protein
Niben_ChloroBio_Chloro <- Niben_ChloroBio_Chloro %>% select(Prediction)


## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_ChloroBio <- merge(Niben_ChloroBio_Ann, TPM_All_FC_Annotated, by="Name")

## We create the annotation for Modules

Niben_ChloroBio_Modules <- which_Module_pwd12(Niben_ChloroBio_Ann)

## We merge this annotation of Modules with the annotation of Chloroplast TP prediction

Niben_ChloroBio_Modules_Chl <- merge(Niben_ChloroBio_Modules, Niben_ChloroBio_Chloro, by = 0)
Niben_ChloroBio_Modules_Chl <- Niben_ChloroBio_Modules_Chl[order(match(Niben_ChloroBio_Modules_Chl$Row.names, Niben_ChloroBio_Ann$Protein)),]
rownames(Niben_ChloroBio_Modules_Chl) <- Niben_ChloroBio_Modules_Chl$Row.names
Niben_ChloroBio_Modules_Chl$Row.names <- NULL

## We select only the FC data from the different comparisons
Niben_ChloroBio_TPM <- Niben_ChloroBio %>% select(Protein, 4:51, 61:66)
Niben_ChloroBio_TPM <- Niben_ChloroBio_TPM[!duplicated(Niben_ChloroBio_TPM$Protein),]
Niben_ChloroBio_TPM <- Niben_ChloroBio_TPM[order(match(Niben_ChloroBio_TPM[,1], Niben_ChloroBio_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_ChloroBio_FC <- Niben_ChloroBio %>% select(Protein, 71, 73:87)
Niben_ChloroBio_FC <- Niben_ChloroBio_FC[!duplicated(Niben_ChloroBio_FC$Protein),]
Niben_ChloroBio_FC <- Niben_ChloroBio_FC[order(match(Niben_ChloroBio_FC[,1], Niben_ChloroBio_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_ChloroBio_DEG <- Niben_ChloroBio %>% select(Protein, 89, 91:105)
Niben_ChloroBio_DEG <- Niben_ChloroBio_DEG[!duplicated(Niben_ChloroBio_DEG$Protein),]
Niben_ChloroBio_DEG <- Niben_ChloroBio_DEG[order(match(Niben_ChloroBio_DEG[,1], Niben_ChloroBio_Ann[,2])),]

Niben_ChloroBio_DEG$pcrtB22h_vs_GFP22h_filtered <- ifelse(is.na(Niben_ChloroBio_DEG$pcrtB22h_vs_GFP22h_filtered) == TRUE, " ", "*")
Niben_ChloroBio_DEG$pcrtB25h_vs_GFP25h_filtered <- ifelse(is.na(Niben_ChloroBio_DEG$pcrtB25h_vs_GFP25h_filtered) == TRUE, " ", "*")
Niben_ChloroBio_DEG$pcrtBAA_vs_GFPAA_filtered <- ifelse(is.na(Niben_ChloroBio_DEG$pcrtBAA_vs_GFPAA_filtered) == TRUE, " ", "*")
Niben_ChloroBio_DEG$pcrtBBB_vs_GFPBB_filtered <- ifelse(is.na(Niben_ChloroBio_DEG$pcrtBBB_vs_GFPBB_filtered) == TRUE, " ", "*")
Niben_ChloroBio_DEG$pcrtBYY_vs_GFPYY_filtered <- ifelse(is.na(Niben_ChloroBio_DEG$pcrtBYY_vs_GFPYY_filtered) == TRUE, " ", "*")
Niben_ChloroBio_DEG$pcrtB28h_vs_GFP28h_filtered <- ifelse(is.na(Niben_ChloroBio_DEG$pcrtB28h_vs_GFP28h_filtered) == TRUE, " ", "*")
Niben_ChloroBio_DEG$pcrtB34h_vs_GFP34h_filtered <- ifelse(is.na(Niben_ChloroBio_DEG$pcrtB34h_vs_GFP34h_filtered) == TRUE, " ", "*")
Niben_ChloroBio_DEG$pcrtB37h_vs_GFP37h_filtered <- ifelse(is.na(Niben_ChloroBio_DEG$pcrtB37h_vs_GFP37h_filtered) == TRUE, " ", "*")
Niben_ChloroBio_DEG$pcrtBZZ_vs_GFPZZ_filtered <- ifelse(is.na(Niben_ChloroBio_DEG$pcrtBZZ_vs_GFPZZ_filtered) == TRUE, " ", "*")
Niben_ChloroBio_DEG$pcrtBCC_vs_GFPCC_filtered <- ifelse(is.na(Niben_ChloroBio_DEG$pcrtBCC_vs_GFPCC_filtered) == TRUE, " ", "*")
Niben_ChloroBio_DEG$pcrtB40h_vs_GFP40h_filtered <- ifelse(is.na(Niben_ChloroBio_DEG$pcrtB40h_vs_GFP40h_filtered) == TRUE, " ", "*")
Niben_ChloroBio_DEG$pcrtBDD_vs_GFPDD_filtered <- ifelse(is.na(Niben_ChloroBio_DEG$pcrtBDD_vs_GFPDD_filtered) == TRUE, " ", "*")
Niben_ChloroBio_DEG$pcrtB46h_vs_GFP46h_filtered <- ifelse(is.na(Niben_ChloroBio_DEG$pcrtB46h_vs_GFP46h_filtered) == TRUE, " ", "*")
Niben_ChloroBio_DEG$pcrtB56h_vs_GFP56h_filtered <- ifelse(is.na(Niben_ChloroBio_DEG$pcrtB56h_vs_GFP56h_filtered) == TRUE, " ", "*")
Niben_ChloroBio_DEG$pcrtBEE_vs_GFPEE_filtered <- ifelse(is.na(Niben_ChloroBio_DEG$pcrtBEE_vs_GFPEE_filtered) == TRUE, " ", "*")
Niben_ChloroBio_DEG$pcrtB96h_vs_GFP96h_filtered <- ifelse(is.na(Niben_ChloroBio_DEG$pcrtB96h_vs_GFP96h_filtered) == TRUE, " ", "*")
row.names(Niben_ChloroBio_DEG) <- Niben_ChloroBio_DEG$Protein
Niben_ChloroBio_DEG[1] <- NULL

Niben_ChloroBio_DEG <- Niben_ChloroBio_DEG %>%
  select(pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

row.names(Niben_ChloroBio_FC) <- Niben_ChloroBio_FC$Protein
Niben_ChloroBio_FC[1] <- NULL

Niben_ChloroBio_FC <- Niben_ChloroBio_FC %>%
  select(pcrtB22h_vs_GFP22h_All, pcrtBAA_vs_GFPAA_All, pcrtB25h_vs_GFP25h_All, 
         pcrtBYY_vs_GFPYY_All, pcrtBBB_vs_GFPBB_All, pcrtB28h_vs_GFP28h_All, pcrtB34h_vs_GFP34h_All, pcrtBZZ_vs_GFPZZ_All, 
         pcrtB37h_vs_GFP37h_All, pcrtBCC_vs_GFPCC_All, pcrtB40h_vs_GFP40h_All, pcrtBDD_vs_GFPDD_All, pcrtB46h_vs_GFP46h_All, 
         pcrtBEE_vs_GFPEE_All, pcrtB56h_vs_GFP56h_All, pcrtB96h_vs_GFP96h_All)


min(Niben_ChloroBio_FC, na.rm = TRUE)
max(Niben_ChloroBio_FC, na.rm = TRUE)
breaks_ChloroBio_FC=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Niben_ChloroBio_FC %>% select(1:15), 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_ChloroBio_FC,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_ChloroBio_Annotation, annotation_colors=color_ChloroBio, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h"),#, "96h"),
         cex = 1,
         angle_col=45,
         main = "ChloroBio",
         annotation_legend = TRUE,
         display_numbers = Niben_ChloroBio_DEG %>% select(1:15),
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(84, 104),
         annotation_row = Niben_ChloroBio_Modules_Chl,
         annotation_colors = module_Chl_colors_2
         #gaps_col = c(15)
)


rownames(Niben_ChloroBio_TPM) <- Niben_ChloroBio_TPM$Protein
Niben_ChloroBio_TPM[1] <- NULL

min(Niben_ChloroBio_TPM, na.rm = TRUE)
max(Niben_ChloroBio_TPM, na.rm = TRUE)
breaks_Niben_ChloroBio_TPM=c(seq(0,100, length.out = 100))

Niben_ChloroBio_TPM_norm <- normalize_df(Niben_ChloroBio_TPM)
Niben_ChloroBio_TPM_norm_Cornell <- Niben_ChloroBio_TPM %>%
  select(1:48) 
Niben_ChloroBio_TPM_norm_Cornell <- normalize_df(Niben_ChloroBio_TPM_norm_Cornell)


pheatmap(Niben_ChloroBio_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Niben_ChloroBio_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_ChloroBio_Annotation, annotation_colors=color_ChloroBio, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "ChloroBio",
         annotation_legend = TRUE,
         #display_numbers = Niben_ChloroBio_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         annotation_row = Niben_ChloroBio_Modules,
         annotation_colors = module_colors,
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(24, 48, 51)
)




#####Representing Heatmap for Sly

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Sly_ChloroBio_Ann <- read.table("Funcional Groups/FigTree/Sly_ChloroBio.txt",
                             sep = "\t", header = TRUE)

colnames(Sly_ChloroBio_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Sly_ChloroBio <- merge(Sly_ChloroBio_Ann, Summary_Yazdani_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Sly_ChloroBio_TPM <- Sly_ChloroBio %>% select(Protein, 4:66)
Sly_ChloroBio_TPM <- Sly_ChloroBio_TPM[!duplicated(Sly_ChloroBio_TPM$Protein),]
Sly_ChloroBio_TPM <- Sly_ChloroBio_TPM[order(match(Sly_ChloroBio_TPM[,1], Sly_ChloroBio_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_ChloroBio_FC <- Sly_ChloroBio %>% select(Protein, 67:71)
Sly_ChloroBio_FC <- Sly_ChloroBio_FC[!duplicated(Sly_ChloroBio_FC$Protein),]
Sly_ChloroBio_FC <- Sly_ChloroBio_FC[order(match(Sly_ChloroBio_FC[,1], Sly_ChloroBio_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_ChloroBio_DEG <- Sly_ChloroBio %>% select(Protein, 72:76)
Sly_ChloroBio_DEG <- Sly_ChloroBio_DEG[!duplicated(Sly_ChloroBio_DEG$Protein),]
Sly_ChloroBio_DEG <- Sly_ChloroBio_DEG[order(match(Sly_ChloroBio_DEG[,1], Sly_ChloroBio_Ann[,2])),]

Sly_ChloroBio_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Sly_ChloroBio_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Sly_ChloroBio_DEG$Ripening_Orange_DEG <- ifelse(is.na(Sly_ChloroBio_DEG$Ripening_Orange_DEG) == TRUE, " ", "*")
Sly_ChloroBio_DEG$Ripening_Red_DEG <- ifelse(is.na(Sly_ChloroBio_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
Sly_ChloroBio_DEG$ORHis_MG_DEG <- ifelse(is.na(Sly_ChloroBio_DEG$ORHis_MG_DEG) == TRUE, " ", "*")
Sly_ChloroBio_DEG$ORHis_Breaker_DEG <- ifelse(is.na(Sly_ChloroBio_DEG$ORHis_Breaker_DEG) == TRUE, " ", "*")
row.names(Sly_ChloroBio_DEG) <- Sly_ChloroBio_DEG$Protein
Sly_ChloroBio_DEG[1] <- NULL

rownames(Sly_ChloroBio_FC) <- Sly_ChloroBio_FC$Protein
Sly_ChloroBio_FC[1] <- NULL

min(Sly_ChloroBio_FC, na.rm = TRUE)
max(Sly_ChloroBio_FC, na.rm = TRUE)
breaks_Sly_ChloroBio=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Sly_ChloroBio_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Sly_ChloroBio,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_ChloroBio_Annotation, annotation_colors=color_ChloroBio, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "ChloroBio",
         annotation_legend = TRUE,
         display_numbers = Sly_ChloroBio_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(39, 48),
         gaps_col = c(3)
)


rownames(Sly_ChloroBio_TPM) <- Sly_ChloroBio_TPM$Protein
Sly_ChloroBio_TPM[1] <- NULL

min(Sly_ChloroBio_TPM, na.rm = TRUE)
max(Sly_ChloroBio_TPM, na.rm = TRUE)
breaks_Sly_ChloroBio_TPM=c(seq(0,100, length.out = 200))

Sly_ChloroBio_TPM_norm <- normalize_df(Sly_ChloroBio_TPM)

pheatmap(Sly_ChloroBio_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Sly_ChloroBio_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_ChloroBio_Annotation, annotation_colors=color_ChloroBio, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "ChloroBio",
         annotation_legend = TRUE,
         #display_numbers = Sly_ChloroBio_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(21,42)
)


#####Representing Heatmap for Cann

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Cann_ChloroBio_Ann <- read.table("Funcional Groups/FigTree/Cann_ChloroBio.txt",
                              sep = "\t", header = TRUE)

colnames(Cann_ChloroBio_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Cann_ChloroBio <- merge(Cann_ChloroBio_Ann, Summary_Pepper_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Cann_ChloroBio_TPM <- Cann_ChloroBio %>% select(Protein, 4:17)
Cann_ChloroBio_TPM <- Cann_ChloroBio_TPM[!duplicated(Cann_ChloroBio_TPM$Protein),]
Cann_ChloroBio_TPM <- Cann_ChloroBio_TPM[order(match(Cann_ChloroBio_TPM[,1], Cann_ChloroBio_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_ChloroBio_FC <- Cann_ChloroBio %>% select(Protein, 18:19)
Cann_ChloroBio_FC <- Cann_ChloroBio_FC[!duplicated(Cann_ChloroBio_FC$Protein),]
Cann_ChloroBio_FC <- Cann_ChloroBio_FC[order(match(Cann_ChloroBio_FC[,1], Cann_ChloroBio_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_ChloroBio_DEG <- Cann_ChloroBio %>% select(Protein, 20:21)
Cann_ChloroBio_DEG <- Cann_ChloroBio_DEG[!duplicated(Cann_ChloroBio_DEG$Protein),]
Cann_ChloroBio_DEG <- Cann_ChloroBio_DEG[order(match(Cann_ChloroBio_DEG[,1], Cann_ChloroBio_Ann[,2])),]

Cann_ChloroBio_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Cann_ChloroBio_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Cann_ChloroBio_DEG$Ripening_Red_DEG <- ifelse(is.na(Cann_ChloroBio_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
row.names(Cann_ChloroBio_DEG) <- Cann_ChloroBio_DEG$Protein
Cann_ChloroBio_DEG[1] <- NULL

rownames(Cann_ChloroBio_FC) <- Cann_ChloroBio_FC$Protein
Cann_ChloroBio_FC[1] <- NULL

min(Cann_ChloroBio_FC, na.rm = TRUE)
max(Cann_ChloroBio_FC, na.rm = TRUE)
breaks_Cann_ChloroBio=c(seq(-4.5,0, length.out = 100), seq(0.01,6,length.out = 100))

pheatmap(Cann_ChloroBio_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Cann_ChloroBio,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_ChloroBio_Annotation, annotation_colors=color_ChloroBio, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Red"),
         cex = 1,
         angle_col=45,
         main = "ChloroBio",
         annotation_legend = TRUE,
         display_numbers = Cann_ChloroBio_DEG,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)


rownames(Cann_ChloroBio_TPM) <- Cann_ChloroBio_TPM$Protein
Cann_ChloroBio_TPM[1] <- NULL

min(Cann_ChloroBio_TPM, na.rm = TRUE)
max(Cann_ChloroBio_TPM, na.rm = TRUE)
breaks_Cann_ChloroBio_TPM=c(seq(0,100, length.out = 200))

Cann_ChloroBio_TPM_norm <- normalize_df(Cann_ChloroBio_TPM)

pheatmap(Cann_ChloroBio_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Cann_ChloroBio_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_ChloroBio_Annotation, annotation_colors=color_ChloroBio, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "ChloroBio",
         annotation_legend = TRUE,
         #display_numbers = Cann_ChloroBio_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(4,9)
)



## Comparativa 96h_SlyYaz_Ath

Ath_ChloroBio_Ann <- read.table("Funcional Groups/Ath_ChloroBio.txt", header = TRUE)

Ath_ChloroBio_FC <- merge(Ath_ChloroBio_Ann, Summary_Ath_FC, by.x=1, by.y = 0, all.x = TRUE) %>%
  dplyr::select(Name, log2FoldChange)
Ath_ChloroBio_FC <- Ath_ChloroBio_FC[order(match(Ath_ChloroBio_FC[,1], Ath_ChloroBio_Ann[,2])),]

Ath_ChloroBio_DEG <- merge(Ath_ChloroBio_Ann, Summary_Ath_DEGs, by.x=1, by.y = 1, all.x = TRUE) %>%
  dplyr::select(Name, log2FC)
Ath_ChloroBio_DEG <- Ath_ChloroBio_DEG[order(match(Ath_ChloroBio_DEG[,1], Ath_ChloroBio_Ann[,2])),]

Ath_ChloroBio_DEG$log2FC <- ifelse(is.na(Ath_ChloroBio_DEG$log2FC) == TRUE, " ", "*")
row.names(Ath_ChloroBio_DEG) <- Ath_ChloroBio_DEG$Name
Ath_ChloroBio_DEG[1] <- NULL

rownames(Ath_ChloroBio_FC) <- Ath_ChloroBio_FC$Name
Ath_ChloroBio_FC[1] <- NULL

min(Ath_ChloroBio_FC, na.rm = TRUE)
max(Ath_ChloroBio_FC, na.rm = TRUE)
breaks_Ath_ChloroBio=c(seq(-4.5,0, length.out = 100), seq(0.01,4.5,length.out = 100))

pheatmap(Ath_ChloroBio_FC, 
         border_color = "grey60",legend= TRUE,
         color=color3, 
         cluster_rows =FALSE,
         breaks = breaks_Stndr,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_ChloroBio_Annotation, annotation_colors=color_ChloroBio, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = "Senescence",
         cex = 1,
         angle_col=45,
         main = "ChloroBio",
         annotation_legend = TRUE,
         display_numbers = Ath_ChloroBio_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         gaps_row = c(3, 4, 5,7, 9, 10, 11, 13, 15, 17, 18, 19, 20, 21)
         #gaps_col = c(2,5)
)

pheatmap(Niben_ChloroBio_FC %>% select(16), 
         border_color = "grey60",legend= TRUE,
         color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_ChloroBio_FC,
         breaks = breaks_Stndr,
         #legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         #legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_ChloroBio_Annotation, annotation_colors=color_ChloroBio, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = "96h",
         cex = 1,
         angle_col=45,
         main = "ChloroBio",
         annotation_legend = TRUE,
         display_numbers = Niben_ChloroBio_DEG %>% select(16),
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(84, 104),
         #annotation_row = Niben_ChloroBio_Modules_Chl,
         #annotation_colors = module_Chl_colors_2,
         gaps_row = c(4, 8, 10, 13, 15, 21, 22, 28, 30, 34, 36, 39, 41, 43)
)


pheatmap(Sly_ChloroBio_FC %>% select(2,3), 
         border_color = "grey60",legend= TRUE,
         color=color3, 
         cluster_rows =FALSE,
         breaks = breaks_Stndr,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         #legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_ChloroBio_Annotation, annotation_colors=color_ChloroBio, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Orange", "Ripening_Red"),
         cex = 1,
         angle_col=45,
         main = "ChloroBio",
         annotation_legend = TRUE,
         display_numbers = Sly_ChloroBio_DEG %>% select(2,3),
         fontsize_number = 15,
         na_col = "darkgrey",
         gaps_row = c(3, 5, 6, 7, 8, 9, 10, 12, 13, 15, 16, 17, 18, 19)
)


#########################################################################################################################################
####################################  ChloroCyc  #############################################################################################
#########################################################################################################################################

ChloroCyc_Ath <- read.table("Funcional Groups/Ath_ChloroCyc.txt", header = TRUE)
ChloroCyc_Ath <- merge(ChloroCyc_Ath, Ath_Gene_Models, by.x="ID", by.y=1)
ChloroCyc_Ath_fasta <- Ath11_fasta[c(which(names(Ath11_fasta) %in% ChloroCyc_Ath$V2))]
ChloroCyc_Ath_2 <- ChloroCyc_Ath %>% select(V2, Name)
write.fasta(ChloroCyc_Ath_fasta, as.string=TRUE, names= names(ChloroCyc_Ath_fasta),
            file.out = "Funcional Groups/Ath_ChloroCyc.fasta")
rename.fasta(infile = "Funcional Groups/Ath_ChloroCyc.fasta", ChloroCyc_Ath_2, outfile = "Funcional Groups/Ath_ChloroCyc_2.fasta")

ChloroCyc_Niben_Candidates <- read.table("Funcional Groups/Niben_Candidates_ChloroCyc.txt")
ChloroCyc_Niben_Candidates <- ChloroCyc_Niben_Candidates[ChloroCyc_Niben_Candidates$V3>=35,] %>% select(V2)
ChloroCyc_Niben_Candidates <- unique(ChloroCyc_Niben_Candidates)
ChloroCyc_Niben_Candidates_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% ChloroCyc_Niben_Candidates$V2))]
write.fasta(ChloroCyc_Niben_Candidates_fasta, as.string=TRUE, names= names(ChloroCyc_Niben_Candidates_fasta),
            file.out = "Funcional Groups/ChloroCyc_Niben_Candidates_35.fasta")

ChloroCyc_Sly_Candidates <- read.table("Funcional Groups/Sly_Candidates_ChloroCyc.txt")
ChloroCyc_Sly_Candidates <- ChloroCyc_Sly_Candidates[ChloroCyc_Sly_Candidates$V3>=35,] %>% select(V2)
ChloroCyc_Sly_Candidates <- unique(ChloroCyc_Sly_Candidates)
ChloroCyc_Sly_Candidates_fasta <- Tomato_4.0_fasta[c(which(names(Tomato_4.0_fasta) %in% ChloroCyc_Sly_Candidates$V2))]
write.fasta(ChloroCyc_Sly_Candidates_fasta, as.string=TRUE, names= names(ChloroCyc_Sly_Candidates_fasta),
            file.out = "Funcional Groups/ChloroCyc_Sly_Candidates_35.fasta")

Niben_ChloroCyc_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_ChloroCyc_Ann$Name))]
write.fasta(Niben_ChloroCyc_fasta, as.string=TRUE, names= names(Niben_ChloroCyc_fasta),
            file.out = "Funcional Groups/Niben_ChloroCyc.fasta")
Niben_ChloroCyc_Ann_2 <- Niben_ChloroCyc_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_ChloroCyc.fasta", Niben_ChloroCyc_Ann, outfile = "Funcional Groups/Niben_ChloroCyc_2.fasta")

save_heatmaps("ChloroCyc", height = 4)

## Representing Heatmap

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Niben_ChloroCyc_Ann <- read.table("Funcional Groups/FigTree/Niben_ChloroCyc.txt",
                                  sep = "\t", header = TRUE)

colnames(Niben_ChloroCyc_Ann) <- c("Name", "Protein")

Niben_ChloroCyc_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_ChloroCyc_Ann$Name))]
write.fasta(Niben_ChloroCyc_fasta, as.string=TRUE, names= names(Niben_ChloroCyc_fasta),
            file.out = "Funcional Groups/Niben_ChloroCyc.fasta")
Niben_ChloroCyc_Ann_2 <- Niben_ChloroCyc_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_ChloroCyc.fasta", Niben_ChloroCyc_Ann, outfile = "Funcional Groups/Niben_ChloroCyc_2.fasta")

Niben_ChloroCyc_Chloro <- read.delim("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan/Funcional Groups/Niben_ChloroCyc_2_summary.targetp2", 
                                     header=FALSE, comment.char="#")
colnames(Niben_ChloroCyc_Chloro) <- c("ID",	"Prediction",	"noTP",	"SP",	"mTP",	"cTP",	"luTP",	"CS-Position")
Niben_ChloroCyc_Chloro$Prediction <- factor(Niben_ChloroCyc_Chloro$Prediction, levels = c("cTP","mTP","noTP"))
Niben_ChloroCyc_Chloro <- merge(Niben_ChloroCyc_Ann, Niben_ChloroCyc_Chloro, by.x = "Protein", by.y = "ID")

Niben_ChloroCyc_Chloro <- Niben_ChloroCyc_Chloro[order(match(Niben_ChloroCyc_Chloro$Protein, Niben_ChloroCyc_Ann$Protein)),]
rownames(Niben_ChloroCyc_Chloro) <- Niben_ChloroCyc_Chloro$Protein
Niben_ChloroCyc_Chloro <- Niben_ChloroCyc_Chloro %>% select(Prediction)


## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_ChloroCyc <- merge(Niben_ChloroCyc_Ann, TPM_All_FC_Annotated, by="Name")

## We create the annotation for Modules

Niben_ChloroCyc_Modules <- which_Module_pwd12(Niben_ChloroCyc_Ann)

## We merge this annotation of Modules with the annotation of Chloroplast TP prediction

Niben_ChloroCyc_Modules_Chl <- merge(Niben_ChloroCyc_Modules, Niben_ChloroCyc_Chloro, by = 0)
Niben_ChloroCyc_Modules_Chl <- Niben_ChloroCyc_Modules_Chl[order(match(Niben_ChloroCyc_Modules_Chl$Row.names, Niben_ChloroCyc_Ann$Protein)),]
rownames(Niben_ChloroCyc_Modules_Chl) <- Niben_ChloroCyc_Modules_Chl$Row.names
Niben_ChloroCyc_Modules_Chl$Row.names <- NULL

## We select only the FC data from the different comparisons
Niben_ChloroCyc_TPM <- Niben_ChloroCyc %>% select(Protein, 4:51, 61:66)
Niben_ChloroCyc_TPM <- Niben_ChloroCyc_TPM[!duplicated(Niben_ChloroCyc_TPM$Protein),]
Niben_ChloroCyc_TPM <- Niben_ChloroCyc_TPM[order(match(Niben_ChloroCyc_TPM[,1], Niben_ChloroCyc_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_ChloroCyc_FC <- Niben_ChloroCyc %>% select(Protein, 71, 73:87)
Niben_ChloroCyc_FC <- Niben_ChloroCyc_FC[!duplicated(Niben_ChloroCyc_FC$Protein),]
Niben_ChloroCyc_FC <- Niben_ChloroCyc_FC[order(match(Niben_ChloroCyc_FC[,1], Niben_ChloroCyc_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_ChloroCyc_DEG <- Niben_ChloroCyc %>% select(Protein, 89, 91:105)
Niben_ChloroCyc_DEG <- Niben_ChloroCyc_DEG[!duplicated(Niben_ChloroCyc_DEG$Protein),]
Niben_ChloroCyc_DEG <- Niben_ChloroCyc_DEG[order(match(Niben_ChloroCyc_DEG[,1], Niben_ChloroCyc_Ann[,2])),]

Niben_ChloroCyc_DEG$pcrtB22h_vs_GFP22h_filtered <- ifelse(is.na(Niben_ChloroCyc_DEG$pcrtB22h_vs_GFP22h_filtered) == TRUE, " ", "*")
Niben_ChloroCyc_DEG$pcrtB25h_vs_GFP25h_filtered <- ifelse(is.na(Niben_ChloroCyc_DEG$pcrtB25h_vs_GFP25h_filtered) == TRUE, " ", "*")
Niben_ChloroCyc_DEG$pcrtBAA_vs_GFPAA_filtered <- ifelse(is.na(Niben_ChloroCyc_DEG$pcrtBAA_vs_GFPAA_filtered) == TRUE, " ", "*")
Niben_ChloroCyc_DEG$pcrtBBB_vs_GFPBB_filtered <- ifelse(is.na(Niben_ChloroCyc_DEG$pcrtBBB_vs_GFPBB_filtered) == TRUE, " ", "*")
Niben_ChloroCyc_DEG$pcrtBYY_vs_GFPYY_filtered <- ifelse(is.na(Niben_ChloroCyc_DEG$pcrtBYY_vs_GFPYY_filtered) == TRUE, " ", "*")
Niben_ChloroCyc_DEG$pcrtB28h_vs_GFP28h_filtered <- ifelse(is.na(Niben_ChloroCyc_DEG$pcrtB28h_vs_GFP28h_filtered) == TRUE, " ", "*")
Niben_ChloroCyc_DEG$pcrtB34h_vs_GFP34h_filtered <- ifelse(is.na(Niben_ChloroCyc_DEG$pcrtB34h_vs_GFP34h_filtered) == TRUE, " ", "*")
Niben_ChloroCyc_DEG$pcrtB37h_vs_GFP37h_filtered <- ifelse(is.na(Niben_ChloroCyc_DEG$pcrtB37h_vs_GFP37h_filtered) == TRUE, " ", "*")
Niben_ChloroCyc_DEG$pcrtBZZ_vs_GFPZZ_filtered <- ifelse(is.na(Niben_ChloroCyc_DEG$pcrtBZZ_vs_GFPZZ_filtered) == TRUE, " ", "*")
Niben_ChloroCyc_DEG$pcrtBCC_vs_GFPCC_filtered <- ifelse(is.na(Niben_ChloroCyc_DEG$pcrtBCC_vs_GFPCC_filtered) == TRUE, " ", "*")
Niben_ChloroCyc_DEG$pcrtB40h_vs_GFP40h_filtered <- ifelse(is.na(Niben_ChloroCyc_DEG$pcrtB40h_vs_GFP40h_filtered) == TRUE, " ", "*")
Niben_ChloroCyc_DEG$pcrtBDD_vs_GFPDD_filtered <- ifelse(is.na(Niben_ChloroCyc_DEG$pcrtBDD_vs_GFPDD_filtered) == TRUE, " ", "*")
Niben_ChloroCyc_DEG$pcrtB46h_vs_GFP46h_filtered <- ifelse(is.na(Niben_ChloroCyc_DEG$pcrtB46h_vs_GFP46h_filtered) == TRUE, " ", "*")
Niben_ChloroCyc_DEG$pcrtB56h_vs_GFP56h_filtered <- ifelse(is.na(Niben_ChloroCyc_DEG$pcrtB56h_vs_GFP56h_filtered) == TRUE, " ", "*")
Niben_ChloroCyc_DEG$pcrtBEE_vs_GFPEE_filtered <- ifelse(is.na(Niben_ChloroCyc_DEG$pcrtBEE_vs_GFPEE_filtered) == TRUE, " ", "*")
Niben_ChloroCyc_DEG$pcrtB96h_vs_GFP96h_filtered <- ifelse(is.na(Niben_ChloroCyc_DEG$pcrtB96h_vs_GFP96h_filtered) == TRUE, " ", "*")
row.names(Niben_ChloroCyc_DEG) <- Niben_ChloroCyc_DEG$Protein
Niben_ChloroCyc_DEG[1] <- NULL

Niben_ChloroCyc_DEG <- Niben_ChloroCyc_DEG %>%
  select(pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

row.names(Niben_ChloroCyc_FC) <- Niben_ChloroCyc_FC$Protein
Niben_ChloroCyc_FC[1] <- NULL

Niben_ChloroCyc_FC <- Niben_ChloroCyc_FC %>%
  select(pcrtB22h_vs_GFP22h_All, pcrtBAA_vs_GFPAA_All, pcrtB25h_vs_GFP25h_All, 
         pcrtBYY_vs_GFPYY_All, pcrtBBB_vs_GFPBB_All, pcrtB28h_vs_GFP28h_All, pcrtB34h_vs_GFP34h_All, pcrtBZZ_vs_GFPZZ_All, 
         pcrtB37h_vs_GFP37h_All, pcrtBCC_vs_GFPCC_All, pcrtB40h_vs_GFP40h_All, pcrtBDD_vs_GFPDD_All, pcrtB46h_vs_GFP46h_All, 
         pcrtBEE_vs_GFPEE_All, pcrtB56h_vs_GFP56h_All, pcrtB96h_vs_GFP96h_All)


min(Niben_ChloroCyc_FC, na.rm = TRUE)
max(Niben_ChloroCyc_FC, na.rm = TRUE)
breaks_ChloroCyc_FC=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Niben_ChloroCyc_FC %>% select(1:15), 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_ChloroCyc_FC,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_ChloroCyc_Annotation, annotation_colors=color_ChloroCyc, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h"),#, "96h"),
         cex = 1,
         angle_col=45,
         main = "ChloroCyc",
         annotation_legend = TRUE,
         display_numbers = Niben_ChloroCyc_DEG %>% select(1:15),
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(84, 104),
         annotation_row = Niben_ChloroCyc_Modules_Chl,
         annotation_colors = module_Chl_colors_2
         #gaps_col = c(15)
)


rownames(Niben_ChloroCyc_TPM) <- Niben_ChloroCyc_TPM$Protein
Niben_ChloroCyc_TPM[1] <- NULL

min(Niben_ChloroCyc_TPM, na.rm = TRUE)
max(Niben_ChloroCyc_TPM, na.rm = TRUE)
breaks_Niben_ChloroCyc_TPM=c(seq(0,100, length.out = 100))

Niben_ChloroCyc_TPM_norm <- normalize_df(Niben_ChloroCyc_TPM)
Niben_ChloroCyc_TPM_norm_Cornell <- Niben_ChloroCyc_TPM %>%
  select(1:48) 
Niben_ChloroCyc_TPM_norm_Cornell <- normalize_df(Niben_ChloroCyc_TPM_norm_Cornell)


pheatmap(Niben_ChloroCyc_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Niben_ChloroCyc_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_ChloroCyc_Annotation, annotation_colors=color_ChloroCyc, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "ChloroCyc",
         annotation_legend = TRUE,
         #display_numbers = Niben_ChloroCyc_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         annotation_row = Niben_ChloroCyc_Modules,
         annotation_colors = module_colors,
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(24, 48, 51)
)


#####Representing Heatmap for Sly

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Sly_ChloroCyc_Ann <- read.table("Funcional Groups/FigTree/Sly_ChloroCyc.txt",
                                sep = "\t", header = TRUE)

colnames(Sly_ChloroCyc_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Sly_ChloroCyc <- merge(Sly_ChloroCyc_Ann, Summary_Yazdani_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Sly_ChloroCyc_TPM <- Sly_ChloroCyc %>% select(Protein, 4:66)
Sly_ChloroCyc_TPM <- Sly_ChloroCyc_TPM[!duplicated(Sly_ChloroCyc_TPM$Protein),]
Sly_ChloroCyc_TPM <- Sly_ChloroCyc_TPM[order(match(Sly_ChloroCyc_TPM[,1], Sly_ChloroCyc_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_ChloroCyc_FC <- Sly_ChloroCyc %>% select(Protein, 67:71)
Sly_ChloroCyc_FC <- Sly_ChloroCyc_FC[!duplicated(Sly_ChloroCyc_FC$Protein),]
Sly_ChloroCyc_FC <- Sly_ChloroCyc_FC[order(match(Sly_ChloroCyc_FC[,1], Sly_ChloroCyc_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_ChloroCyc_DEG <- Sly_ChloroCyc %>% select(Protein, 72:76)
Sly_ChloroCyc_DEG <- Sly_ChloroCyc_DEG[!duplicated(Sly_ChloroCyc_DEG$Protein),]
Sly_ChloroCyc_DEG <- Sly_ChloroCyc_DEG[order(match(Sly_ChloroCyc_DEG[,1], Sly_ChloroCyc_Ann[,2])),]

Sly_ChloroCyc_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Sly_ChloroCyc_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Sly_ChloroCyc_DEG$Ripening_Orange_DEG <- ifelse(is.na(Sly_ChloroCyc_DEG$Ripening_Orange_DEG) == TRUE, " ", "*")
Sly_ChloroCyc_DEG$Ripening_Red_DEG <- ifelse(is.na(Sly_ChloroCyc_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
Sly_ChloroCyc_DEG$ORHis_MG_DEG <- ifelse(is.na(Sly_ChloroCyc_DEG$ORHis_MG_DEG) == TRUE, " ", "*")
Sly_ChloroCyc_DEG$ORHis_Breaker_DEG <- ifelse(is.na(Sly_ChloroCyc_DEG$ORHis_Breaker_DEG) == TRUE, " ", "*")
row.names(Sly_ChloroCyc_DEG) <- Sly_ChloroCyc_DEG$Protein
Sly_ChloroCyc_DEG[1] <- NULL

rownames(Sly_ChloroCyc_FC) <- Sly_ChloroCyc_FC$Protein
Sly_ChloroCyc_FC[1] <- NULL

min(Sly_ChloroCyc_FC, na.rm = TRUE)
max(Sly_ChloroCyc_FC, na.rm = TRUE)
breaks_Sly_ChloroCyc=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Sly_ChloroCyc_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Sly_ChloroCyc,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_ChloroCyc_Annotation, annotation_colors=color_ChloroCyc, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "ChloroCyc",
         annotation_legend = TRUE,
         display_numbers = Sly_ChloroCyc_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(39, 48),
         gaps_col = c(3)
)


rownames(Sly_ChloroCyc_TPM) <- Sly_ChloroCyc_TPM$Protein
Sly_ChloroCyc_TPM[1] <- NULL

min(Sly_ChloroCyc_TPM, na.rm = TRUE)
max(Sly_ChloroCyc_TPM, na.rm = TRUE)
breaks_Sly_ChloroCyc_TPM=c(seq(0,100, length.out = 200))

Sly_ChloroCyc_TPM_norm <- normalize_df(Sly_ChloroCyc_TPM)

pheatmap(Sly_ChloroCyc_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Sly_ChloroCyc_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_ChloroCyc_Annotation, annotation_colors=color_ChloroCyc, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "ChloroCyc",
         annotation_legend = TRUE,
         #display_numbers = Sly_ChloroCyc_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(21,42)
)


#####Representing Heatmap for Cann

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Cann_ChloroCyc_Ann <- read.table("Funcional Groups/FigTree/Cann_ChloroCyc.txt",
                                 sep = "\t", header = TRUE)

colnames(Cann_ChloroCyc_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Cann_ChloroCyc <- merge(Cann_ChloroCyc_Ann, Summary_Pepper_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Cann_ChloroCyc_TPM <- Cann_ChloroCyc %>% select(Protein, 4:17)
Cann_ChloroCyc_TPM <- Cann_ChloroCyc_TPM[!duplicated(Cann_ChloroCyc_TPM$Protein),]
Cann_ChloroCyc_TPM <- Cann_ChloroCyc_TPM[order(match(Cann_ChloroCyc_TPM[,1], Cann_ChloroCyc_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_ChloroCyc_FC <- Cann_ChloroCyc %>% select(Protein, 18:19)
Cann_ChloroCyc_FC <- Cann_ChloroCyc_FC[!duplicated(Cann_ChloroCyc_FC$Protein),]
Cann_ChloroCyc_FC <- Cann_ChloroCyc_FC[order(match(Cann_ChloroCyc_FC[,1], Cann_ChloroCyc_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_ChloroCyc_DEG <- Cann_ChloroCyc %>% select(Protein, 20:21)
Cann_ChloroCyc_DEG <- Cann_ChloroCyc_DEG[!duplicated(Cann_ChloroCyc_DEG$Protein),]
Cann_ChloroCyc_DEG <- Cann_ChloroCyc_DEG[order(match(Cann_ChloroCyc_DEG[,1], Cann_ChloroCyc_Ann[,2])),]

Cann_ChloroCyc_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Cann_ChloroCyc_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Cann_ChloroCyc_DEG$Ripening_Red_DEG <- ifelse(is.na(Cann_ChloroCyc_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
row.names(Cann_ChloroCyc_DEG) <- Cann_ChloroCyc_DEG$Protein
Cann_ChloroCyc_DEG[1] <- NULL

rownames(Cann_ChloroCyc_FC) <- Cann_ChloroCyc_FC$Protein
Cann_ChloroCyc_FC[1] <- NULL

min(Cann_ChloroCyc_FC, na.rm = TRUE)
max(Cann_ChloroCyc_FC, na.rm = TRUE)
breaks_Cann_ChloroCyc=c(seq(-4.5,0, length.out = 100), seq(0.01,6,length.out = 100))

pheatmap(Cann_ChloroCyc_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Cann_ChloroCyc,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_ChloroCyc_Annotation, annotation_colors=color_ChloroCyc, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Red"),
         cex = 1,
         angle_col=45,
         main = "ChloroCyc",
         annotation_legend = TRUE,
         display_numbers = Cann_ChloroCyc_DEG,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)


rownames(Cann_ChloroCyc_TPM) <- Cann_ChloroCyc_TPM$Protein
Cann_ChloroCyc_TPM[1] <- NULL

min(Cann_ChloroCyc_TPM, na.rm = TRUE)
max(Cann_ChloroCyc_TPM, na.rm = TRUE)
breaks_Cann_ChloroCyc_TPM=c(seq(0,100, length.out = 200))

Cann_ChloroCyc_TPM_norm <- normalize_df(Cann_ChloroCyc_TPM)

pheatmap(Cann_ChloroCyc_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Cann_ChloroCyc_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_ChloroCyc_Annotation, annotation_colors=color_ChloroCyc, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "ChloroCyc",
         annotation_legend = TRUE,
         #display_numbers = Cann_ChloroCyc_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(4,9)
)


## Comparativa 96h_SlyYaz_Ath

Ath_ChloroCyc_Ann <- read.table("Funcional Groups/Ath_ChloroCyc.txt", header = TRUE)

Ath_ChloroCyc_FC <- merge(Ath_ChloroCyc_Ann, Summary_Ath_FC, by.x=1, by.y = 0, all.x = TRUE) %>%
  dplyr::select(Name, log2FoldChange)
Ath_ChloroCyc_FC <- Ath_ChloroCyc_FC[order(match(Ath_ChloroCyc_FC[,1], Ath_ChloroCyc_Ann[,2])),]

Ath_ChloroCyc_DEG <- merge(Ath_ChloroCyc_Ann, Summary_Ath_DEGs, by.x=1, by.y = 1, all.x = TRUE) %>%
  dplyr::select(Name, log2FC)
Ath_ChloroCyc_DEG <- Ath_ChloroCyc_DEG[order(match(Ath_ChloroCyc_DEG[,1], Ath_ChloroCyc_Ann[,2])),]

Ath_ChloroCyc_DEG$log2FC <- ifelse(is.na(Ath_ChloroCyc_DEG$log2FC) == TRUE, " ", "*")
row.names(Ath_ChloroCyc_DEG) <- Ath_ChloroCyc_DEG$Name
Ath_ChloroCyc_DEG[1] <- NULL

rownames(Ath_ChloroCyc_FC) <- Ath_ChloroCyc_FC$Name
Ath_ChloroCyc_FC[1] <- NULL

min(Ath_ChloroCyc_FC, na.rm = TRUE)
max(Ath_ChloroCyc_FC, na.rm = TRUE)
breaks_Ath_ChloroCyc=c(seq(-4.5,0, length.out = 100), seq(0.01,4.5,length.out = 100))

pheatmap(Ath_ChloroCyc_FC, 
         border_color = "grey60",legend= TRUE,
         color=color3, 
         cluster_rows =FALSE,
         breaks = breaks_Stndr,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_ChloroCyc_Annotation, annotation_colors=color_ChloroCyc, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = "Senescence",
         cex = 1,
         angle_col=45,
         main = "ChloroCyc",
         annotation_legend = TRUE,
         display_numbers = Ath_ChloroCyc_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         gaps_row = c(1,2, 4)
         #gaps_col = c(2,5)
)

pheatmap(Niben_ChloroCyc_FC %>% select(16), 
         border_color = "grey60",legend= TRUE,
         color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_ChloroCyc_FC,
         breaks = breaks_Stndr,
         #legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         #legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_ChloroCyc_Annotation, annotation_colors=color_ChloroCyc, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = "96h",
         cex = 1,
         angle_col=45,
         main = "ChloroCyc",
         annotation_legend = TRUE,
         display_numbers = Niben_ChloroCyc_DEG %>% select(16),
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(84, 104),
         #annotation_row = Niben_ChloroCyc_Modules_Chl,
         #annotation_colors = module_Chl_colors_2,
         gaps_row = c(5, 7, 11)
)


pheatmap(Sly_ChloroCyc_FC %>% select(2,3), 
         border_color = "grey60",legend= TRUE,
         color=color3, 
         cluster_rows =FALSE,
         breaks = breaks_Stndr,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         #legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_ChloroCyc_Annotation, annotation_colors=color_ChloroCyc, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Orange", "Ripening_Red"),
         cex = 1,
         angle_col=45,
         main = "ChloroCyc",
         annotation_legend = TRUE,
         display_numbers = Sly_ChloroCyc_DEG %>% select(2,3),
         fontsize_number = 15,
         na_col = "darkgrey",
         gaps_row = c(3, 4, 6)
)

#########################################################################################################################################
####################################  ChloroDeg  #############################################################################################
#########################################################################################################################################

ChloroDeg_Ath <- read.table("Funcional Groups/Ath_ChloroDeg.txt", header = TRUE)
ChloroDeg_Ath <- merge(ChloroDeg_Ath, Ath_Gene_Models, by.x="ID", by.y=1)
ChloroDeg_Ath_fasta <- Ath11_fasta[c(which(names(Ath11_fasta) %in% ChloroDeg_Ath$V2))]
ChloroDeg_Ath_2 <- ChloroDeg_Ath %>% select(V2, Name)
write.fasta(ChloroDeg_Ath_fasta, as.string=TRUE, names= names(ChloroDeg_Ath_fasta),
            file.out = "Funcional Groups/Ath_ChloroDeg.fasta")
rename.fasta(infile = "Funcional Groups/Ath_ChloroDeg.fasta", ChloroDeg_Ath_2, outfile = "Funcional Groups/Ath_ChloroDeg_2.fasta")

ChloroDeg_Niben_Candidates <- read.table("Funcional Groups/Niben_Candidates_ChloroDeg.txt")
ChloroDeg_Niben_Candidates <- ChloroDeg_Niben_Candidates[ChloroDeg_Niben_Candidates$V3>=35,] %>% select(V2)
ChloroDeg_Niben_Candidates <- unique(ChloroDeg_Niben_Candidates)
ChloroDeg_Niben_Candidates_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% ChloroDeg_Niben_Candidates$V2))]
write.fasta(ChloroDeg_Niben_Candidates_fasta, as.string=TRUE, names= names(ChloroDeg_Niben_Candidates_fasta),
            file.out = "Funcional Groups/ChloroDeg_Niben_Candidates_35.fasta")

ChloroDeg_Sly_Candidates <- read.table("Funcional Groups/Sly_Candidates_ChloroDeg.txt")
ChloroDeg_Sly_Candidates <- ChloroDeg_Sly_Candidates[ChloroDeg_Sly_Candidates$V3>=35,] %>% select(V2)
ChloroDeg_Sly_Candidates <- unique(ChloroDeg_Sly_Candidates)
ChloroDeg_Sly_Candidates_fasta <- Tomato_4.0_fasta[c(which(names(Tomato_4.0_fasta) %in% ChloroDeg_Sly_Candidates$V2))]
write.fasta(ChloroDeg_Sly_Candidates_fasta, as.string=TRUE, names= names(ChloroDeg_Sly_Candidates_fasta),
            file.out = "Funcional Groups/ChloroDeg_Sly_Candidates_35.fasta")

Niben_ChloroDeg_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_ChloroDeg_Ann$Name))]
write.fasta(Niben_ChloroDeg_fasta, as.string=TRUE, names= names(Niben_ChloroDeg_fasta),
            file.out = "Funcional Groups/Niben_ChloroDeg.fasta")
Niben_ChloroDeg_Ann_2 <- Niben_ChloroDeg_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_ChloroDeg.fasta", Niben_ChloroDeg_Ann, outfile = "Funcional Groups/Niben_ChloroDeg_2.fasta")

save_heatmaps("ChloroDeg", height = 4)

## Representing Heatmap

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Niben_ChloroDeg_Ann <- read.table("Funcional Groups/FigTree/Niben_ChloroDeg.txt",
                                  sep = "\t", header = TRUE)

colnames(Niben_ChloroDeg_Ann) <- c("Name", "Protein")

Niben_ChloroDeg_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_ChloroDeg_Ann$Name))]
write.fasta(Niben_ChloroDeg_fasta, as.string=TRUE, names= names(Niben_ChloroDeg_fasta),
            file.out = "Funcional Groups/Niben_ChloroDeg.fasta")
Niben_ChloroDeg_Ann_2 <- Niben_ChloroDeg_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_ChloroDeg.fasta", Niben_ChloroDeg_Ann, outfile = "Funcional Groups/Niben_ChloroDeg_2.fasta")

Niben_ChloroDeg_Chloro <- read.delim("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan/Funcional Groups/Niben_ChloroDeg_2_summary.targetp2", 
                                     header=FALSE, comment.char="#")
colnames(Niben_ChloroDeg_Chloro) <- c("ID",	"Prediction",	"noTP",	"SP",	"mTP",	"cTP",	"luTP",	"CS-Position")
Niben_ChloroDeg_Chloro$Prediction <- factor(Niben_ChloroDeg_Chloro$Prediction, levels = c("cTP","mTP","noTP"))
Niben_ChloroDeg_Chloro <- merge(Niben_ChloroDeg_Ann, Niben_ChloroDeg_Chloro, by.x = "Protein", by.y = "ID")

Niben_ChloroDeg_Chloro <- Niben_ChloroDeg_Chloro[order(match(Niben_ChloroDeg_Chloro$Protein, Niben_ChloroDeg_Ann$Protein)),]
rownames(Niben_ChloroDeg_Chloro) <- Niben_ChloroDeg_Chloro$Protein
Niben_ChloroDeg_Chloro <- Niben_ChloroDeg_Chloro %>% select(Prediction)


## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_ChloroDeg <- merge(Niben_ChloroDeg_Ann, TPM_All_FC_Annotated, by="Name")

## We create the annotation for Modules

Niben_ChloroDeg_Modules <- which_Module_pwd12(Niben_ChloroDeg_Ann)

## We merge this annotation of Modules with the annotation of Chloroplast TP prediction

Niben_ChloroDeg_Modules_Chl <- merge(Niben_ChloroDeg_Modules, Niben_ChloroDeg_Chloro, by = 0)
Niben_ChloroDeg_Modules_Chl <- Niben_ChloroDeg_Modules_Chl[order(match(Niben_ChloroDeg_Modules_Chl$Row.names, Niben_ChloroDeg_Ann$Protein)),]
rownames(Niben_ChloroDeg_Modules_Chl) <- Niben_ChloroDeg_Modules_Chl$Row.names
Niben_ChloroDeg_Modules_Chl$Row.names <- NULL

## We select only the FC data from the different comparisons
Niben_ChloroDeg_TPM <- Niben_ChloroDeg %>% select(Protein, 4:51, 61:66)
Niben_ChloroDeg_TPM <- Niben_ChloroDeg_TPM[!duplicated(Niben_ChloroDeg_TPM$Protein),]
Niben_ChloroDeg_TPM <- Niben_ChloroDeg_TPM[order(match(Niben_ChloroDeg_TPM[,1], Niben_ChloroDeg_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_ChloroDeg_FC <- Niben_ChloroDeg %>% select(Protein, 71, 73:87)
Niben_ChloroDeg_FC <- Niben_ChloroDeg_FC[!duplicated(Niben_ChloroDeg_FC$Protein),]
Niben_ChloroDeg_FC <- Niben_ChloroDeg_FC[order(match(Niben_ChloroDeg_FC[,1], Niben_ChloroDeg_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_ChloroDeg_DEG <- Niben_ChloroDeg %>% select(Protein, 89, 91:105)
Niben_ChloroDeg_DEG <- Niben_ChloroDeg_DEG[!duplicated(Niben_ChloroDeg_DEG$Protein),]
Niben_ChloroDeg_DEG <- Niben_ChloroDeg_DEG[order(match(Niben_ChloroDeg_DEG[,1], Niben_ChloroDeg_Ann[,2])),]

Niben_ChloroDeg_DEG$pcrtB22h_vs_GFP22h_filtered <- ifelse(is.na(Niben_ChloroDeg_DEG$pcrtB22h_vs_GFP22h_filtered) == TRUE, " ", "*")
Niben_ChloroDeg_DEG$pcrtB25h_vs_GFP25h_filtered <- ifelse(is.na(Niben_ChloroDeg_DEG$pcrtB25h_vs_GFP25h_filtered) == TRUE, " ", "*")
Niben_ChloroDeg_DEG$pcrtBAA_vs_GFPAA_filtered <- ifelse(is.na(Niben_ChloroDeg_DEG$pcrtBAA_vs_GFPAA_filtered) == TRUE, " ", "*")
Niben_ChloroDeg_DEG$pcrtBBB_vs_GFPBB_filtered <- ifelse(is.na(Niben_ChloroDeg_DEG$pcrtBBB_vs_GFPBB_filtered) == TRUE, " ", "*")
Niben_ChloroDeg_DEG$pcrtBYY_vs_GFPYY_filtered <- ifelse(is.na(Niben_ChloroDeg_DEG$pcrtBYY_vs_GFPYY_filtered) == TRUE, " ", "*")
Niben_ChloroDeg_DEG$pcrtB28h_vs_GFP28h_filtered <- ifelse(is.na(Niben_ChloroDeg_DEG$pcrtB28h_vs_GFP28h_filtered) == TRUE, " ", "*")
Niben_ChloroDeg_DEG$pcrtB34h_vs_GFP34h_filtered <- ifelse(is.na(Niben_ChloroDeg_DEG$pcrtB34h_vs_GFP34h_filtered) == TRUE, " ", "*")
Niben_ChloroDeg_DEG$pcrtB37h_vs_GFP37h_filtered <- ifelse(is.na(Niben_ChloroDeg_DEG$pcrtB37h_vs_GFP37h_filtered) == TRUE, " ", "*")
Niben_ChloroDeg_DEG$pcrtBZZ_vs_GFPZZ_filtered <- ifelse(is.na(Niben_ChloroDeg_DEG$pcrtBZZ_vs_GFPZZ_filtered) == TRUE, " ", "*")
Niben_ChloroDeg_DEG$pcrtBCC_vs_GFPCC_filtered <- ifelse(is.na(Niben_ChloroDeg_DEG$pcrtBCC_vs_GFPCC_filtered) == TRUE, " ", "*")
Niben_ChloroDeg_DEG$pcrtB40h_vs_GFP40h_filtered <- ifelse(is.na(Niben_ChloroDeg_DEG$pcrtB40h_vs_GFP40h_filtered) == TRUE, " ", "*")
Niben_ChloroDeg_DEG$pcrtBDD_vs_GFPDD_filtered <- ifelse(is.na(Niben_ChloroDeg_DEG$pcrtBDD_vs_GFPDD_filtered) == TRUE, " ", "*")
Niben_ChloroDeg_DEG$pcrtB46h_vs_GFP46h_filtered <- ifelse(is.na(Niben_ChloroDeg_DEG$pcrtB46h_vs_GFP46h_filtered) == TRUE, " ", "*")
Niben_ChloroDeg_DEG$pcrtB56h_vs_GFP56h_filtered <- ifelse(is.na(Niben_ChloroDeg_DEG$pcrtB56h_vs_GFP56h_filtered) == TRUE, " ", "*")
Niben_ChloroDeg_DEG$pcrtBEE_vs_GFPEE_filtered <- ifelse(is.na(Niben_ChloroDeg_DEG$pcrtBEE_vs_GFPEE_filtered) == TRUE, " ", "*")
Niben_ChloroDeg_DEG$pcrtB96h_vs_GFP96h_filtered <- ifelse(is.na(Niben_ChloroDeg_DEG$pcrtB96h_vs_GFP96h_filtered) == TRUE, " ", "*")
row.names(Niben_ChloroDeg_DEG) <- Niben_ChloroDeg_DEG$Protein
Niben_ChloroDeg_DEG[1] <- NULL

Niben_ChloroDeg_DEG <- Niben_ChloroDeg_DEG %>%
  select(pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

row.names(Niben_ChloroDeg_FC) <- Niben_ChloroDeg_FC$Protein
Niben_ChloroDeg_FC[1] <- NULL

Niben_ChloroDeg_FC <- Niben_ChloroDeg_FC %>%
  select(pcrtB22h_vs_GFP22h_All, pcrtBAA_vs_GFPAA_All, pcrtB25h_vs_GFP25h_All, 
         pcrtBYY_vs_GFPYY_All, pcrtBBB_vs_GFPBB_All, pcrtB28h_vs_GFP28h_All, pcrtB34h_vs_GFP34h_All, pcrtBZZ_vs_GFPZZ_All, 
         pcrtB37h_vs_GFP37h_All, pcrtBCC_vs_GFPCC_All, pcrtB40h_vs_GFP40h_All, pcrtBDD_vs_GFPDD_All, pcrtB46h_vs_GFP46h_All, 
         pcrtBEE_vs_GFPEE_All, pcrtB56h_vs_GFP56h_All, pcrtB96h_vs_GFP96h_All)


min(Niben_ChloroDeg_FC, na.rm = TRUE)
max(Niben_ChloroDeg_FC, na.rm = TRUE)
breaks_ChloroDeg_FC=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Niben_ChloroDeg_FC %>% select(1:15), 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_ChloroDeg_FC,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_ChloroDeg_Annotation, annotation_colors=color_ChloroDeg, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h"),#, "96h"),
         cex = 1,
         angle_col=45,
         main = "ChloroDeg",
         annotation_legend = TRUE,
         display_numbers = Niben_ChloroDeg_DEG %>% select(1:15),
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(84, 104),
         annotation_row = Niben_ChloroDeg_Modules_Chl,
         annotation_colors = module_Chl_colors_2
         #gaps_col = c(15)
)


rownames(Niben_ChloroDeg_TPM) <- Niben_ChloroDeg_TPM$Protein
Niben_ChloroDeg_TPM[1] <- NULL

min(Niben_ChloroDeg_TPM, na.rm = TRUE)
max(Niben_ChloroDeg_TPM, na.rm = TRUE)
breaks_Niben_ChloroDeg_TPM=c(seq(0,100, length.out = 100))

Niben_ChloroDeg_TPM_norm <- normalize_df(Niben_ChloroDeg_TPM)
Niben_ChloroDeg_TPM_norm_Cornell <- Niben_ChloroDeg_TPM %>%
  select(1:48) 
Niben_ChloroDeg_TPM_norm_Cornell <- normalize_df(Niben_ChloroDeg_TPM_norm_Cornell)


pheatmap(Niben_ChloroDeg_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Niben_ChloroDeg_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_ChloroDeg_Annotation, annotation_colors=color_ChloroDeg, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "ChloroDeg",
         annotation_legend = TRUE,
         #display_numbers = Niben_ChloroDeg_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         annotation_row = Niben_ChloroDeg_Modules,
         annotation_colors = module_colors,
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(24, 48, 51)
)



#####Representing Heatmap for Sly

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Sly_ChloroDeg_Ann <- read.table("Funcional Groups/FigTree/Sly_ChloroDeg.txt",
                                sep = "\t", header = TRUE)

colnames(Sly_ChloroDeg_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Sly_ChloroDeg <- merge(Sly_ChloroDeg_Ann, Summary_Yazdani_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Sly_ChloroDeg_TPM <- Sly_ChloroDeg %>% select(Protein, 4:66)
Sly_ChloroDeg_TPM <- Sly_ChloroDeg_TPM[!duplicated(Sly_ChloroDeg_TPM$Protein),]
Sly_ChloroDeg_TPM <- Sly_ChloroDeg_TPM[order(match(Sly_ChloroDeg_TPM[,1], Sly_ChloroDeg_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_ChloroDeg_FC <- Sly_ChloroDeg %>% select(Protein, 67:71)
Sly_ChloroDeg_FC <- Sly_ChloroDeg_FC[!duplicated(Sly_ChloroDeg_FC$Protein),]
Sly_ChloroDeg_FC <- Sly_ChloroDeg_FC[order(match(Sly_ChloroDeg_FC[,1], Sly_ChloroDeg_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_ChloroDeg_DEG <- Sly_ChloroDeg %>% select(Protein, 72:76)
Sly_ChloroDeg_DEG <- Sly_ChloroDeg_DEG[!duplicated(Sly_ChloroDeg_DEG$Protein),]
Sly_ChloroDeg_DEG <- Sly_ChloroDeg_DEG[order(match(Sly_ChloroDeg_DEG[,1], Sly_ChloroDeg_Ann[,2])),]

Sly_ChloroDeg_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Sly_ChloroDeg_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Sly_ChloroDeg_DEG$Ripening_Orange_DEG <- ifelse(is.na(Sly_ChloroDeg_DEG$Ripening_Orange_DEG) == TRUE, " ", "*")
Sly_ChloroDeg_DEG$Ripening_Red_DEG <- ifelse(is.na(Sly_ChloroDeg_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
Sly_ChloroDeg_DEG$ORHis_MG_DEG <- ifelse(is.na(Sly_ChloroDeg_DEG$ORHis_MG_DEG) == TRUE, " ", "*")
Sly_ChloroDeg_DEG$ORHis_Breaker_DEG <- ifelse(is.na(Sly_ChloroDeg_DEG$ORHis_Breaker_DEG) == TRUE, " ", "*")
row.names(Sly_ChloroDeg_DEG) <- Sly_ChloroDeg_DEG$Protein
Sly_ChloroDeg_DEG[1] <- NULL

rownames(Sly_ChloroDeg_FC) <- Sly_ChloroDeg_FC$Protein
Sly_ChloroDeg_FC[1] <- NULL

min(Sly_ChloroDeg_FC, na.rm = TRUE)
max(Sly_ChloroDeg_FC, na.rm = TRUE)
breaks_Sly_ChloroDeg=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Sly_ChloroDeg_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Sly_ChloroDeg,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_ChloroDeg_Annotation, annotation_colors=color_ChloroDeg, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "ChloroDeg",
         annotation_legend = TRUE,
         display_numbers = Sly_ChloroDeg_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(39, 48),
         gaps_col = c(3)
)


rownames(Sly_ChloroDeg_TPM) <- Sly_ChloroDeg_TPM$Protein
Sly_ChloroDeg_TPM[1] <- NULL

min(Sly_ChloroDeg_TPM, na.rm = TRUE)
max(Sly_ChloroDeg_TPM, na.rm = TRUE)
breaks_Sly_ChloroDeg_TPM=c(seq(0,100, length.out = 200))

Sly_ChloroDeg_TPM_norm <- normalize_df(Sly_ChloroDeg_TPM)

pheatmap(Sly_ChloroDeg_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Sly_ChloroDeg_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_ChloroDeg_Annotation, annotation_colors=color_ChloroDeg, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "ChloroDeg",
         annotation_legend = TRUE,
         #display_numbers = Sly_ChloroDeg_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(21,42)
)


#####Representing Heatmap for Cann

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Cann_ChloroDeg_Ann <- read.table("Funcional Groups/FigTree/Cann_ChloroDeg.txt",
                                 sep = "\t", header = TRUE)

colnames(Cann_ChloroDeg_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Cann_ChloroDeg <- merge(Cann_ChloroDeg_Ann, Summary_Pepper_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Cann_ChloroDeg_TPM <- Cann_ChloroDeg %>% select(Protein, 4:17)
Cann_ChloroDeg_TPM <- Cann_ChloroDeg_TPM[!duplicated(Cann_ChloroDeg_TPM$Protein),]
Cann_ChloroDeg_TPM <- Cann_ChloroDeg_TPM[order(match(Cann_ChloroDeg_TPM[,1], Cann_ChloroDeg_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_ChloroDeg_FC <- Cann_ChloroDeg %>% select(Protein, 18:19)
Cann_ChloroDeg_FC <- Cann_ChloroDeg_FC[!duplicated(Cann_ChloroDeg_FC$Protein),]
Cann_ChloroDeg_FC <- Cann_ChloroDeg_FC[order(match(Cann_ChloroDeg_FC[,1], Cann_ChloroDeg_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_ChloroDeg_DEG <- Cann_ChloroDeg %>% select(Protein, 20:21)
Cann_ChloroDeg_DEG <- Cann_ChloroDeg_DEG[!duplicated(Cann_ChloroDeg_DEG$Protein),]
Cann_ChloroDeg_DEG <- Cann_ChloroDeg_DEG[order(match(Cann_ChloroDeg_DEG[,1], Cann_ChloroDeg_Ann[,2])),]

Cann_ChloroDeg_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Cann_ChloroDeg_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Cann_ChloroDeg_DEG$Ripening_Red_DEG <- ifelse(is.na(Cann_ChloroDeg_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
row.names(Cann_ChloroDeg_DEG) <- Cann_ChloroDeg_DEG$Protein
Cann_ChloroDeg_DEG[1] <- NULL

rownames(Cann_ChloroDeg_FC) <- Cann_ChloroDeg_FC$Protein
Cann_ChloroDeg_FC[1] <- NULL

min(Cann_ChloroDeg_FC, na.rm = TRUE)
max(Cann_ChloroDeg_FC, na.rm = TRUE)
breaks_Cann_ChloroDeg=c(seq(-4.5,0, length.out = 100), seq(0.01,6,length.out = 100))

pheatmap(Cann_ChloroDeg_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Cann_ChloroDeg,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_ChloroDeg_Annotation, annotation_colors=color_ChloroDeg, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Red"),
         cex = 1,
         angle_col=45,
         main = "ChloroDeg",
         annotation_legend = TRUE,
         display_numbers = Cann_ChloroDeg_DEG,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)


rownames(Cann_ChloroDeg_TPM) <- Cann_ChloroDeg_TPM$Protein
Cann_ChloroDeg_TPM[1] <- NULL

min(Cann_ChloroDeg_TPM, na.rm = TRUE)
max(Cann_ChloroDeg_TPM, na.rm = TRUE)
breaks_Cann_ChloroDeg_TPM=c(seq(0,100, length.out = 200))

Cann_ChloroDeg_TPM_norm <- normalize_df(Cann_ChloroDeg_TPM)

pheatmap(Cann_ChloroDeg_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Cann_ChloroDeg_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_ChloroDeg_Annotation, annotation_colors=color_ChloroDeg, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "ChloroDeg",
         annotation_legend = TRUE,
         #display_numbers = Cann_ChloroDeg_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(4,9)
)


## Comparativa 96h_SlyYaz_Ath

Ath_ChloroDeg_Ann <- read.table("Funcional Groups/Ath_ChloroDeg.txt", header = TRUE)

Ath_ChloroDeg_FC <- merge(Ath_ChloroDeg_Ann, Summary_Ath_FC, by.x=1, by.y = 0, all.x = TRUE) %>%
  dplyr::select(Name, log2FoldChange)
Ath_ChloroDeg_FC <- Ath_ChloroDeg_FC[order(match(Ath_ChloroDeg_FC[,1], Ath_ChloroDeg_Ann[,2])),]

Ath_ChloroDeg_DEG <- merge(Ath_ChloroDeg_Ann, Summary_Ath_DEGs, by.x=1, by.y = 1, all.x = TRUE) %>%
  dplyr::select(Name, log2FC)
Ath_ChloroDeg_DEG <- Ath_ChloroDeg_DEG[order(match(Ath_ChloroDeg_DEG[,1], Ath_ChloroDeg_Ann[,2])),]

Ath_ChloroDeg_DEG$log2FC <- ifelse(is.na(Ath_ChloroDeg_DEG$log2FC) == TRUE, " ", "*")
row.names(Ath_ChloroDeg_DEG) <- Ath_ChloroDeg_DEG$Name
Ath_ChloroDeg_DEG[1] <- NULL

rownames(Ath_ChloroDeg_FC) <- Ath_ChloroDeg_FC$Name
Ath_ChloroDeg_FC[1] <- NULL

min(Ath_ChloroDeg_FC, na.rm = TRUE)
max(Ath_ChloroDeg_FC, na.rm = TRUE)
breaks_Ath_ChloroDeg=c(seq(-4.5,0, length.out = 100), seq(0.01,4.5,length.out = 100))

pheatmap(Ath_ChloroDeg_FC, 
         border_color = "grey60",legend= TRUE,
         color=color3, 
         cluster_rows =FALSE,
         breaks = breaks_Stndr,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_ChloroDeg_Annotation, annotation_colors=color_ChloroDeg, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = "Senescence",
         cex = 1,
         angle_col=45,
         main = "ChloroDeg",
         annotation_legend = TRUE,
         display_numbers = Ath_ChloroDeg_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         gaps_row = c(2, 3, 4)
         #gaps_col = c(2,5)
)

pheatmap(Niben_ChloroDeg_FC %>% select(16), 
         border_color = "grey60",legend= TRUE,
         color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_ChloroDeg_FC,
         breaks = breaks_Stndr,
         #legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         #legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_ChloroDeg_Annotation, annotation_colors=color_ChloroDeg, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = "96h",
         cex = 1,
         angle_col=45,
         main = "ChloroDeg",
         annotation_legend = TRUE,
         display_numbers = Niben_ChloroDeg_DEG %>% select(16),
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(84, 104),
         #annotation_row = Niben_ChloroDeg_Modules_Chl,
         #annotation_colors = module_Chl_colors_2,
         gaps_row = c(4, 8, 11)
)


pheatmap(Sly_ChloroDeg_FC %>% select(2,3), 
         border_color = "grey60",legend= TRUE,
         color=color3, 
         cluster_rows =FALSE,
         breaks = breaks_Stndr,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         #legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_ChloroDeg_Annotation, annotation_colors=color_ChloroDeg, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Orange", "Ripening_Red"),
         cex = 1,
         angle_col=45,
         main = "ChloroDeg",
         annotation_legend = TRUE,
         display_numbers = Sly_ChloroDeg_DEG %>% select(2,3),
         fontsize_number = 15,
         na_col = "darkgrey",
         gaps_row = c(4, 6, 8)
)

#########################################################################################################################################
####################################  Gibb  #############################################################################################
#########################################################################################################################################

Gibb_Ath <- read.table("Funcional Groups/Ath_Gibb.txt", header = TRUE)
Gibb_Ath <- merge(Gibb_Ath, Ath_Gene_Models, by.x="ID", by.y=1)
Gibb_Ath_fasta <- Ath11_fasta[c(which(names(Ath11_fasta) %in% Gibb_Ath$V2))]
Gibb_Ath_2 <- Gibb_Ath %>% select(V2, Name)
write.fasta(Gibb_Ath_fasta, as.string=TRUE, names= names(Gibb_Ath_fasta),
            file.out = "Funcional Groups/Ath_Gibb.fasta")
rename.fasta(infile = "Funcional Groups/Ath_Gibb.fasta", Gibb_Ath_2, outfile = "Funcional Groups/Ath_Gibb_2.fasta")

Gibb_Niben_Candidates <- read.table("Funcional Groups/Niben_Candidates_Gibb.txt")
Gibb_Niben_Candidates <- Gibb_Niben_Candidates[Gibb_Niben_Candidates$V3>=35,] %>% select(V2)
Gibb_Niben_Candidates <- unique(Gibb_Niben_Candidates)
Gibb_Niben_Candidates_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Gibb_Niben_Candidates$V2))]
write.fasta(Gibb_Niben_Candidates_fasta, as.string=TRUE, names= names(Gibb_Niben_Candidates_fasta),
            file.out = "Funcional Groups/Gibb_Niben_Candidates_35.fasta")

Gibb_Sly_Candidates <- read.table("Funcional Groups/Sly_Candidates_Gibb.txt")
Gibb_Sly_Candidates <- Gibb_Sly_Candidates[Gibb_Sly_Candidates$V3>=35,] %>% select(V2)
Gibb_Sly_Candidates <- unique(Gibb_Sly_Candidates)
Gibb_Sly_Candidates_fasta <- Tomato_4.0_fasta[c(which(names(Tomato_4.0_fasta) %in% Gibb_Sly_Candidates$V2))]
write.fasta(Gibb_Sly_Candidates_fasta, as.string=TRUE, names= names(Gibb_Sly_Candidates_fasta),
            file.out = "Funcional Groups/Gibb_Sly_Candidates_35.fasta")

Niben_Gibb_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_Gibb_Ann$Name))]
write.fasta(Niben_Gibb_fasta, as.string=TRUE, names= names(Niben_Gibb_fasta),
            file.out = "Funcional Groups/Niben_Gibb.fasta")
Niben_Gibb_Ann_2 <- Niben_Gibb_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_Gibb.fasta", Niben_Gibb_Ann, outfile = "Funcional Groups/Niben_Gibb_2.fasta")

save_heatmaps("Gibb", height = 8)

## Representing Heatmap

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Niben_Gibb_Ann <- read.table("Funcional Groups/FigTree/Niben_Gibb.txt",
                                  sep = "\t", header = TRUE)

colnames(Niben_Gibb_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_Gibb <- merge(Niben_Gibb_Ann, TPM_All_FC_Annotated, by="Name")

## We create the annotation for Modules

Niben_Gibb_Modules <- which_Module_pwd30(Niben_Gibb_Ann)

## We select only the FC data from the different comparisons
Niben_Gibb_TPM <- Niben_Gibb %>% select(Protein, 4:51, 61:66)
Niben_Gibb_TPM <- Niben_Gibb_TPM[!duplicated(Niben_Gibb_TPM$Protein),]
Niben_Gibb_TPM <- Niben_Gibb_TPM[order(match(Niben_Gibb_TPM[,1], Niben_Gibb_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_Gibb_FC <- Niben_Gibb %>% select(Protein, 71, 73:87)
Niben_Gibb_FC <- Niben_Gibb_FC[!duplicated(Niben_Gibb_FC$Protein),]
Niben_Gibb_FC <- Niben_Gibb_FC[order(match(Niben_Gibb_FC[,1], Niben_Gibb_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_Gibb_DEG <- Niben_Gibb %>% select(Protein, 89, 91:105)
Niben_Gibb_DEG <- Niben_Gibb_DEG[!duplicated(Niben_Gibb_DEG$Protein),]
Niben_Gibb_DEG <- Niben_Gibb_DEG[order(match(Niben_Gibb_DEG[,1], Niben_Gibb_Ann[,2])),]

Niben_Gibb_DEG$pcrtB22h_vs_GFP22h_filtered <- ifelse(is.na(Niben_Gibb_DEG$pcrtB22h_vs_GFP22h_filtered) == TRUE, " ", "*")
Niben_Gibb_DEG$pcrtB25h_vs_GFP25h_filtered <- ifelse(is.na(Niben_Gibb_DEG$pcrtB25h_vs_GFP25h_filtered) == TRUE, " ", "*")
Niben_Gibb_DEG$pcrtBAA_vs_GFPAA_filtered <- ifelse(is.na(Niben_Gibb_DEG$pcrtBAA_vs_GFPAA_filtered) == TRUE, " ", "*")
Niben_Gibb_DEG$pcrtBBB_vs_GFPBB_filtered <- ifelse(is.na(Niben_Gibb_DEG$pcrtBBB_vs_GFPBB_filtered) == TRUE, " ", "*")
Niben_Gibb_DEG$pcrtBYY_vs_GFPYY_filtered <- ifelse(is.na(Niben_Gibb_DEG$pcrtBYY_vs_GFPYY_filtered) == TRUE, " ", "*")
Niben_Gibb_DEG$pcrtB28h_vs_GFP28h_filtered <- ifelse(is.na(Niben_Gibb_DEG$pcrtB28h_vs_GFP28h_filtered) == TRUE, " ", "*")
Niben_Gibb_DEG$pcrtB34h_vs_GFP34h_filtered <- ifelse(is.na(Niben_Gibb_DEG$pcrtB34h_vs_GFP34h_filtered) == TRUE, " ", "*")
Niben_Gibb_DEG$pcrtB37h_vs_GFP37h_filtered <- ifelse(is.na(Niben_Gibb_DEG$pcrtB37h_vs_GFP37h_filtered) == TRUE, " ", "*")
Niben_Gibb_DEG$pcrtBZZ_vs_GFPZZ_filtered <- ifelse(is.na(Niben_Gibb_DEG$pcrtBZZ_vs_GFPZZ_filtered) == TRUE, " ", "*")
Niben_Gibb_DEG$pcrtBCC_vs_GFPCC_filtered <- ifelse(is.na(Niben_Gibb_DEG$pcrtBCC_vs_GFPCC_filtered) == TRUE, " ", "*")
Niben_Gibb_DEG$pcrtB40h_vs_GFP40h_filtered <- ifelse(is.na(Niben_Gibb_DEG$pcrtB40h_vs_GFP40h_filtered) == TRUE, " ", "*")
Niben_Gibb_DEG$pcrtBDD_vs_GFPDD_filtered <- ifelse(is.na(Niben_Gibb_DEG$pcrtBDD_vs_GFPDD_filtered) == TRUE, " ", "*")
Niben_Gibb_DEG$pcrtB46h_vs_GFP46h_filtered <- ifelse(is.na(Niben_Gibb_DEG$pcrtB46h_vs_GFP46h_filtered) == TRUE, " ", "*")
Niben_Gibb_DEG$pcrtB56h_vs_GFP56h_filtered <- ifelse(is.na(Niben_Gibb_DEG$pcrtB56h_vs_GFP56h_filtered) == TRUE, " ", "*")
Niben_Gibb_DEG$pcrtBEE_vs_GFPEE_filtered <- ifelse(is.na(Niben_Gibb_DEG$pcrtBEE_vs_GFPEE_filtered) == TRUE, " ", "*")
Niben_Gibb_DEG$pcrtB96h_vs_GFP96h_filtered <- ifelse(is.na(Niben_Gibb_DEG$pcrtB96h_vs_GFP96h_filtered) == TRUE, " ", "*")
row.names(Niben_Gibb_DEG) <- Niben_Gibb_DEG$Protein
Niben_Gibb_DEG[1] <- NULL

Niben_Gibb_DEG <- Niben_Gibb_DEG %>%
  select(pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

row.names(Niben_Gibb_FC) <- Niben_Gibb_FC$Protein
Niben_Gibb_FC[1] <- NULL

Niben_Gibb_FC <- Niben_Gibb_FC %>%
  select(pcrtB22h_vs_GFP22h_All, pcrtBAA_vs_GFPAA_All, pcrtB25h_vs_GFP25h_All, 
         pcrtBYY_vs_GFPYY_All, pcrtBBB_vs_GFPBB_All, pcrtB28h_vs_GFP28h_All, pcrtB34h_vs_GFP34h_All, pcrtBZZ_vs_GFPZZ_All, 
         pcrtB37h_vs_GFP37h_All, pcrtBCC_vs_GFPCC_All, pcrtB40h_vs_GFP40h_All, pcrtBDD_vs_GFPDD_All, pcrtB46h_vs_GFP46h_All, 
         pcrtBEE_vs_GFPEE_All, pcrtB56h_vs_GFP56h_All, pcrtB96h_vs_GFP96h_All)


min(Niben_Gibb_FC, na.rm = TRUE)
max(Niben_Gibb_FC, na.rm = TRUE)
breaks_Gibb_FC=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Niben_Gibb_FC, # %>% select(1:15), 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Gibb_FC,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_Gibb_Annotation, annotation_colors=color_Gibb, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h", "96h"),
         cex = 1,
         angle_col=45,
         main = "Gibb",
         annotation_legend = TRUE,
         display_numbers = Niben_Gibb_DEG, # %>% select(1:15),
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(84, 104),
         annotation_row = Niben_Gibb_Modules,
         annotation_colors = module_colors
         #gaps_col = c(15)
)


rownames(Niben_Gibb_TPM) <- Niben_Gibb_TPM$Protein
Niben_Gibb_TPM[1] <- NULL

min(Niben_Gibb_TPM, na.rm = TRUE)
max(Niben_Gibb_TPM, na.rm = TRUE)
breaks_Niben_Gibb_TPM=c(seq(0,100, length.out = 100))

Niben_Gibb_TPM_norm <- normalize_df(Niben_Gibb_TPM)
Niben_Gibb_TPM_norm_Cornell <- Niben_Gibb_TPM %>%
  select(1:48) 
Niben_Gibb_TPM_norm_Cornell <- normalize_df(Niben_Gibb_TPM_norm_Cornell)


pheatmap(Niben_Gibb_TPM_norm_Cornell, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Niben_Gibb_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_Gibb_Annotation, annotation_colors=color_Gibb, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Gibb",
         annotation_legend = TRUE,
         #display_numbers = Niben_Gibb_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         annotation_row = Niben_Gibb_Modules,
         annotation_colors = module_colors,
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(24) #, 48, 51)
)



#####Representing Heatmap for Sly

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Sly_Gibb_Ann <- read.table("Funcional Groups/FigTree/Sly_Gibb.txt",
                                sep = "\t", header = TRUE)

colnames(Sly_Gibb_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Sly_Gibb <- merge(Sly_Gibb_Ann, Summary_Yazdani_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Sly_Gibb_TPM <- Sly_Gibb %>% select(Protein, 4:66)
Sly_Gibb_TPM <- Sly_Gibb_TPM[!duplicated(Sly_Gibb_TPM$Protein),]
Sly_Gibb_TPM <- Sly_Gibb_TPM[order(match(Sly_Gibb_TPM[,1], Sly_Gibb_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_Gibb_FC <- Sly_Gibb %>% select(Protein, 67:71)
Sly_Gibb_FC <- Sly_Gibb_FC[!duplicated(Sly_Gibb_FC$Protein),]
Sly_Gibb_FC <- Sly_Gibb_FC[order(match(Sly_Gibb_FC[,1], Sly_Gibb_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_Gibb_DEG <- Sly_Gibb %>% select(Protein, 72:76)
Sly_Gibb_DEG <- Sly_Gibb_DEG[!duplicated(Sly_Gibb_DEG$Protein),]
Sly_Gibb_DEG <- Sly_Gibb_DEG[order(match(Sly_Gibb_DEG[,1], Sly_Gibb_Ann[,2])),]

Sly_Gibb_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Sly_Gibb_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Sly_Gibb_DEG$Ripening_Orange_DEG <- ifelse(is.na(Sly_Gibb_DEG$Ripening_Orange_DEG) == TRUE, " ", "*")
Sly_Gibb_DEG$Ripening_Red_DEG <- ifelse(is.na(Sly_Gibb_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
Sly_Gibb_DEG$ORHis_MG_DEG <- ifelse(is.na(Sly_Gibb_DEG$ORHis_MG_DEG) == TRUE, " ", "*")
Sly_Gibb_DEG$ORHis_Breaker_DEG <- ifelse(is.na(Sly_Gibb_DEG$ORHis_Breaker_DEG) == TRUE, " ", "*")
row.names(Sly_Gibb_DEG) <- Sly_Gibb_DEG$Protein
Sly_Gibb_DEG[1] <- NULL

rownames(Sly_Gibb_FC) <- Sly_Gibb_FC$Protein
Sly_Gibb_FC[1] <- NULL

min(Sly_Gibb_FC, na.rm = TRUE)
max(Sly_Gibb_FC, na.rm = TRUE)
breaks_Sly_Gibb=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Sly_Gibb_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Sly_Gibb,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_Gibb_Annotation, annotation_colors=color_Gibb, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Gibb",
         annotation_legend = TRUE,
         display_numbers = Sly_Gibb_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(39, 48),
         gaps_col = c(3)
)


rownames(Sly_Gibb_TPM) <- Sly_Gibb_TPM$Protein
Sly_Gibb_TPM[1] <- NULL

min(Sly_Gibb_TPM, na.rm = TRUE)
max(Sly_Gibb_TPM, na.rm = TRUE)
breaks_Sly_Gibb_TPM=c(seq(0,100, length.out = 200))

Sly_Gibb_TPM_norm <- normalize_df(Sly_Gibb_TPM)

pheatmap(Sly_Gibb_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Sly_Gibb_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_Gibb_Annotation, annotation_colors=color_Gibb, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Gibb",
         annotation_legend = TRUE,
         #display_numbers = Sly_Gibb_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(21,42)
)


#####Representing Heatmap for Cann

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Cann_Gibb_Ann <- read.table("Funcional Groups/FigTree/Cann_Gibb.txt",
                                 sep = "\t", header = TRUE)

colnames(Cann_Gibb_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Cann_Gibb <- merge(Cann_Gibb_Ann, Summary_Pepper_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Cann_Gibb_TPM <- Cann_Gibb %>% select(Protein, 4:17)
Cann_Gibb_TPM <- Cann_Gibb_TPM[!duplicated(Cann_Gibb_TPM$Protein),]
Cann_Gibb_TPM <- Cann_Gibb_TPM[order(match(Cann_Gibb_TPM[,1], Cann_Gibb_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_Gibb_FC <- Cann_Gibb %>% select(Protein, 18:19)
Cann_Gibb_FC <- Cann_Gibb_FC[!duplicated(Cann_Gibb_FC$Protein),]
Cann_Gibb_FC <- Cann_Gibb_FC[order(match(Cann_Gibb_FC[,1], Cann_Gibb_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_Gibb_DEG <- Cann_Gibb %>% select(Protein, 20:21)
Cann_Gibb_DEG <- Cann_Gibb_DEG[!duplicated(Cann_Gibb_DEG$Protein),]
Cann_Gibb_DEG <- Cann_Gibb_DEG[order(match(Cann_Gibb_DEG[,1], Cann_Gibb_Ann[,2])),]

Cann_Gibb_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Cann_Gibb_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Cann_Gibb_DEG$Ripening_Red_DEG <- ifelse(is.na(Cann_Gibb_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
row.names(Cann_Gibb_DEG) <- Cann_Gibb_DEG$Protein
Cann_Gibb_DEG[1] <- NULL

rownames(Cann_Gibb_FC) <- Cann_Gibb_FC$Protein
Cann_Gibb_FC[1] <- NULL

min(Cann_Gibb_FC, na.rm = TRUE)
max(Cann_Gibb_FC, na.rm = TRUE)
breaks_Cann_Gibb=c(seq(-4.5,0, length.out = 100), seq(0.01,6,length.out = 100))

pheatmap(Cann_Gibb_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Cann_Gibb,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_Gibb_Annotation, annotation_colors=color_Gibb, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Red"),
         cex = 1,
         angle_col=45,
         main = "Gibb",
         annotation_legend = TRUE,
         display_numbers = Cann_Gibb_DEG,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)


rownames(Cann_Gibb_TPM) <- Cann_Gibb_TPM$Protein
Cann_Gibb_TPM[1] <- NULL

min(Cann_Gibb_TPM, na.rm = TRUE)
max(Cann_Gibb_TPM, na.rm = TRUE)
breaks_Cann_Gibb_TPM=c(seq(0,100, length.out = 200))

Cann_Gibb_TPM_norm <- normalize_df(Cann_Gibb_TPM)

pheatmap(Cann_Gibb_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Cann_Gibb_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_Gibb_Annotation, annotation_colors=color_Gibb, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Gibb",
         annotation_legend = TRUE,
         #display_numbers = Cann_Gibb_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(4,9)
)


########################################################################################################################################
####################################  DELLA  #############################################################################################
#########################################################################################################################################

DELLA_Ath <- read.table("Funcional Groups/Ath_DELLA.txt", header = TRUE)
DELLA_Ath <- merge(DELLA_Ath, Ath_Gene_Models, by.x="ID", by.y=1)
DELLA_Ath_fasta <- Ath11_fasta[c(which(names(Ath11_fasta) %in% DELLA_Ath$V2))]
DELLA_Ath_2 <- DELLA_Ath %>% select(V2, Name)
write.fasta(DELLA_Ath_fasta, as.string=TRUE, names= names(DELLA_Ath_fasta),
            file.out = "Funcional Groups/Ath_DELLA.fasta")
rename.fasta(infile = "Funcional Groups/Ath_DELLA.fasta", DELLA_Ath_2, outfile = "Funcional Groups/Ath_DELLA_2.fasta")

DELLA_Niben_Candidates <- read.table("Funcional Groups/Niben_Candidates_DELLA.txt")
DELLA_Niben_Candidates <- DELLA_Niben_Candidates[DELLA_Niben_Candidates$V3>=35,] %>% select(V2)
DELLA_Niben_Candidates <- unique(DELLA_Niben_Candidates)
DELLA_Niben_Candidates_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% DELLA_Niben_Candidates$V2))]
write.fasta(DELLA_Niben_Candidates_fasta, as.string=TRUE, names= names(DELLA_Niben_Candidates_fasta),
            file.out = "Funcional Groups/DELLA_Niben_Candidates_35.fasta")

DELLA_Sly_Candidates <- read.table("Funcional Groups/Sly_Candidates_DELLA.txt")
DELLA_Sly_Candidates <- DELLA_Sly_Candidates[DELLA_Sly_Candidates$V3>=35,] %>% select(V2)
DELLA_Sly_Candidates <- unique(DELLA_Sly_Candidates)
DELLA_Sly_Candidates_fasta <- Tomato_4.0_fasta[c(which(names(Tomato_4.0_fasta) %in% DELLA_Sly_Candidates$V2))]
write.fasta(DELLA_Sly_Candidates_fasta, as.string=TRUE, names= names(DELLA_Sly_Candidates_fasta),
            file.out = "Funcional Groups/DELLA_Sly_Candidates_35.fasta")

Niben_DELLA_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_DELLA_Ann$Name))]
write.fasta(Niben_DELLA_fasta, as.string=TRUE, names= names(Niben_DELLA_fasta),
            file.out = "Funcional Groups/Niben_DELLA.fasta")
Niben_DELLA_Ann_2 <- Niben_DELLA_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_DELLA.fasta", Niben_DELLA_Ann, outfile = "Funcional Groups/Niben_DELLA_2.fasta")

save_heatmaps("DELLA", height = 3)

## Representing Heatmap

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Niben_DELLA_Ann <- read.table("Funcional Groups/FigTree/Niben_DELLA.txt",
                             sep = "\t", header = TRUE)

colnames(Niben_DELLA_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_DELLA <- merge(Niben_DELLA_Ann, TPM_All_FC_Annotated, by="Name")

## We create the annotation for Modules

Niben_DELLA_Modules <- which_Module_pwd30(Niben_DELLA_Ann)

## We select only the FC data from the different comparisons
Niben_DELLA_TPM <- Niben_DELLA %>% select(Protein, 4:51, 61:66)
Niben_DELLA_TPM <- Niben_DELLA_TPM[!duplicated(Niben_DELLA_TPM$Protein),]
Niben_DELLA_TPM <- Niben_DELLA_TPM[order(match(Niben_DELLA_TPM[,1], Niben_DELLA_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_DELLA_FC <- Niben_DELLA %>% select(Protein, 71, 73:87)
Niben_DELLA_FC <- Niben_DELLA_FC[!duplicated(Niben_DELLA_FC$Protein),]
Niben_DELLA_FC <- Niben_DELLA_FC[order(match(Niben_DELLA_FC[,1], Niben_DELLA_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_DELLA_DEG <- Niben_DELLA %>% select(Protein, 89, 91:105)
Niben_DELLA_DEG <- Niben_DELLA_DEG[!duplicated(Niben_DELLA_DEG$Protein),]
Niben_DELLA_DEG <- Niben_DELLA_DEG[order(match(Niben_DELLA_DEG[,1], Niben_DELLA_Ann[,2])),]

Niben_DELLA_DEG$pcrtB22h_vs_GFP22h_filtered <- ifelse(is.na(Niben_DELLA_DEG$pcrtB22h_vs_GFP22h_filtered) == TRUE, " ", "*")
Niben_DELLA_DEG$pcrtB25h_vs_GFP25h_filtered <- ifelse(is.na(Niben_DELLA_DEG$pcrtB25h_vs_GFP25h_filtered) == TRUE, " ", "*")
Niben_DELLA_DEG$pcrtBAA_vs_GFPAA_filtered <- ifelse(is.na(Niben_DELLA_DEG$pcrtBAA_vs_GFPAA_filtered) == TRUE, " ", "*")
Niben_DELLA_DEG$pcrtBBB_vs_GFPBB_filtered <- ifelse(is.na(Niben_DELLA_DEG$pcrtBBB_vs_GFPBB_filtered) == TRUE, " ", "*")
Niben_DELLA_DEG$pcrtBYY_vs_GFPYY_filtered <- ifelse(is.na(Niben_DELLA_DEG$pcrtBYY_vs_GFPYY_filtered) == TRUE, " ", "*")
Niben_DELLA_DEG$pcrtB28h_vs_GFP28h_filtered <- ifelse(is.na(Niben_DELLA_DEG$pcrtB28h_vs_GFP28h_filtered) == TRUE, " ", "*")
Niben_DELLA_DEG$pcrtB34h_vs_GFP34h_filtered <- ifelse(is.na(Niben_DELLA_DEG$pcrtB34h_vs_GFP34h_filtered) == TRUE, " ", "*")
Niben_DELLA_DEG$pcrtB37h_vs_GFP37h_filtered <- ifelse(is.na(Niben_DELLA_DEG$pcrtB37h_vs_GFP37h_filtered) == TRUE, " ", "*")
Niben_DELLA_DEG$pcrtBZZ_vs_GFPZZ_filtered <- ifelse(is.na(Niben_DELLA_DEG$pcrtBZZ_vs_GFPZZ_filtered) == TRUE, " ", "*")
Niben_DELLA_DEG$pcrtBCC_vs_GFPCC_filtered <- ifelse(is.na(Niben_DELLA_DEG$pcrtBCC_vs_GFPCC_filtered) == TRUE, " ", "*")
Niben_DELLA_DEG$pcrtB40h_vs_GFP40h_filtered <- ifelse(is.na(Niben_DELLA_DEG$pcrtB40h_vs_GFP40h_filtered) == TRUE, " ", "*")
Niben_DELLA_DEG$pcrtBDD_vs_GFPDD_filtered <- ifelse(is.na(Niben_DELLA_DEG$pcrtBDD_vs_GFPDD_filtered) == TRUE, " ", "*")
Niben_DELLA_DEG$pcrtB46h_vs_GFP46h_filtered <- ifelse(is.na(Niben_DELLA_DEG$pcrtB46h_vs_GFP46h_filtered) == TRUE, " ", "*")
Niben_DELLA_DEG$pcrtB56h_vs_GFP56h_filtered <- ifelse(is.na(Niben_DELLA_DEG$pcrtB56h_vs_GFP56h_filtered) == TRUE, " ", "*")
Niben_DELLA_DEG$pcrtBEE_vs_GFPEE_filtered <- ifelse(is.na(Niben_DELLA_DEG$pcrtBEE_vs_GFPEE_filtered) == TRUE, " ", "*")
Niben_DELLA_DEG$pcrtB96h_vs_GFP96h_filtered <- ifelse(is.na(Niben_DELLA_DEG$pcrtB96h_vs_GFP96h_filtered) == TRUE, " ", "*")
row.names(Niben_DELLA_DEG) <- Niben_DELLA_DEG$Protein
Niben_DELLA_DEG[1] <- NULL

Niben_DELLA_DEG <- Niben_DELLA_DEG %>%
  select(pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

row.names(Niben_DELLA_FC) <- Niben_DELLA_FC$Protein
Niben_DELLA_FC[1] <- NULL

Niben_DELLA_FC <- Niben_DELLA_FC %>%
  select(pcrtB22h_vs_GFP22h_All, pcrtBAA_vs_GFPAA_All, pcrtB25h_vs_GFP25h_All, 
         pcrtBYY_vs_GFPYY_All, pcrtBBB_vs_GFPBB_All, pcrtB28h_vs_GFP28h_All, pcrtB34h_vs_GFP34h_All, pcrtBZZ_vs_GFPZZ_All, 
         pcrtB37h_vs_GFP37h_All, pcrtBCC_vs_GFPCC_All, pcrtB40h_vs_GFP40h_All, pcrtBDD_vs_GFPDD_All, pcrtB46h_vs_GFP46h_All, 
         pcrtBEE_vs_GFPEE_All, pcrtB56h_vs_GFP56h_All, pcrtB96h_vs_GFP96h_All)


min(Niben_DELLA_FC, na.rm = TRUE)
max(Niben_DELLA_FC, na.rm = TRUE)
breaks_DELLA_FC=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Niben_DELLA_FC %>% select(1:15), 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_DELLA_FC,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_DELLA_Annotation, annotation_colors=color_DELLA, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h"), #, "96h"),
         cex = 1,
         angle_col=45,
         main = "DELLA",
         annotation_legend = TRUE,
         display_numbers = Niben_DELLA_DEG %>% select(1:15),
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(84, 104),
         annotation_row = Niben_DELLA_Modules,
         annotation_colors = module_colors
         #gaps_col = c(15)
)


rownames(Niben_DELLA_TPM) <- Niben_DELLA_TPM$Protein
Niben_DELLA_TPM[1] <- NULL

min(Niben_DELLA_TPM, na.rm = TRUE)
max(Niben_DELLA_TPM, na.rm = TRUE)
breaks_Niben_DELLA_TPM=c(seq(0,100, length.out = 100))

Niben_DELLA_TPM_norm <- normalize_df(Niben_DELLA_TPM)
Niben_DELLA_TPM_norm_Cornell <- Niben_DELLA_TPM %>%
  select(1:48) 
Niben_DELLA_TPM_norm_Cornell <- normalize_df(Niben_DELLA_TPM_norm_Cornell)


pheatmap(Niben_DELLA_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Niben_DELLA_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_DELLA_Annotation, annotation_colors=color_DELLA, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "DELLA",
         annotation_legend = TRUE,
         #display_numbers = Niben_DELLA_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         annotation_row = Niben_DELLA_Modules,
         annotation_colors = module_colors,
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(24, 48, 51)
)



#####Representing Heatmap for Sly

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Sly_DELLA_Ann <- read.table("Funcional Groups/FigTree/Sly_DELLA.txt",
                           sep = "\t", header = TRUE)

colnames(Sly_DELLA_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Sly_DELLA <- merge(Sly_DELLA_Ann, Summary_Yazdani_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Sly_DELLA_TPM <- Sly_DELLA %>% select(Protein, 4:66)
Sly_DELLA_TPM <- Sly_DELLA_TPM[!duplicated(Sly_DELLA_TPM$Protein),]
Sly_DELLA_TPM <- Sly_DELLA_TPM[order(match(Sly_DELLA_TPM[,1], Sly_DELLA_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_DELLA_FC <- Sly_DELLA %>% select(Protein, 67:71)
Sly_DELLA_FC <- Sly_DELLA_FC[!duplicated(Sly_DELLA_FC$Protein),]
Sly_DELLA_FC <- Sly_DELLA_FC[order(match(Sly_DELLA_FC[,1], Sly_DELLA_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_DELLA_DEG <- Sly_DELLA %>% select(Protein, 72:76)
Sly_DELLA_DEG <- Sly_DELLA_DEG[!duplicated(Sly_DELLA_DEG$Protein),]
Sly_DELLA_DEG <- Sly_DELLA_DEG[order(match(Sly_DELLA_DEG[,1], Sly_DELLA_Ann[,2])),]

Sly_DELLA_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Sly_DELLA_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Sly_DELLA_DEG$Ripening_Orange_DEG <- ifelse(is.na(Sly_DELLA_DEG$Ripening_Orange_DEG) == TRUE, " ", "*")
Sly_DELLA_DEG$Ripening_Red_DEG <- ifelse(is.na(Sly_DELLA_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
Sly_DELLA_DEG$ORHis_MG_DEG <- ifelse(is.na(Sly_DELLA_DEG$ORHis_MG_DEG) == TRUE, " ", "*")
Sly_DELLA_DEG$ORHis_Breaker_DEG <- ifelse(is.na(Sly_DELLA_DEG$ORHis_Breaker_DEG) == TRUE, " ", "*")
row.names(Sly_DELLA_DEG) <- Sly_DELLA_DEG$Protein
Sly_DELLA_DEG[1] <- NULL

rownames(Sly_DELLA_FC) <- Sly_DELLA_FC$Protein
Sly_DELLA_FC[1] <- NULL

min(Sly_DELLA_FC, na.rm = TRUE)
max(Sly_DELLA_FC, na.rm = TRUE)
breaks_Sly_DELLA=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Sly_DELLA_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Sly_DELLA,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_DELLA_Annotation, annotation_colors=color_DELLA, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "DELLA",
         annotation_legend = TRUE,
         display_numbers = Sly_DELLA_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(39, 48),
         gaps_col = c(3)
)


rownames(Sly_DELLA_TPM) <- Sly_DELLA_TPM$Protein
Sly_DELLA_TPM[1] <- NULL

min(Sly_DELLA_TPM, na.rm = TRUE)
max(Sly_DELLA_TPM, na.rm = TRUE)
breaks_Sly_DELLA_TPM=c(seq(0,100, length.out = 200))

Sly_DELLA_TPM_norm <- normalize_df(Sly_DELLA_TPM)

pheatmap(Sly_DELLA_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Sly_DELLA_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_DELLA_Annotation, annotation_colors=color_DELLA, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "DELLA",
         annotation_legend = TRUE,
         #display_numbers = Sly_DELLA_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(21,42)
)


#####Representing Heatmap for Cann

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Cann_DELLA_Ann <- read.table("Funcional Groups/FigTree/Cann_DELLA.txt",
                            sep = "\t", header = TRUE)

colnames(Cann_DELLA_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Cann_DELLA <- merge(Cann_DELLA_Ann, Summary_Pepper_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Cann_DELLA_TPM <- Cann_DELLA %>% select(Protein, 4:17)
Cann_DELLA_TPM <- Cann_DELLA_TPM[!duplicated(Cann_DELLA_TPM$Protein),]
Cann_DELLA_TPM <- Cann_DELLA_TPM[order(match(Cann_DELLA_TPM[,1], Cann_DELLA_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_DELLA_FC <- Cann_DELLA %>% select(Protein, 18:19)
Cann_DELLA_FC <- Cann_DELLA_FC[!duplicated(Cann_DELLA_FC$Protein),]
Cann_DELLA_FC <- Cann_DELLA_FC[order(match(Cann_DELLA_FC[,1], Cann_DELLA_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_DELLA_DEG <- Cann_DELLA %>% select(Protein, 20:21)
Cann_DELLA_DEG <- Cann_DELLA_DEG[!duplicated(Cann_DELLA_DEG$Protein),]
Cann_DELLA_DEG <- Cann_DELLA_DEG[order(match(Cann_DELLA_DEG[,1], Cann_DELLA_Ann[,2])),]

Cann_DELLA_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Cann_DELLA_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Cann_DELLA_DEG$Ripening_Red_DEG <- ifelse(is.na(Cann_DELLA_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
row.names(Cann_DELLA_DEG) <- Cann_DELLA_DEG$Protein
Cann_DELLA_DEG[1] <- NULL

rownames(Cann_DELLA_FC) <- Cann_DELLA_FC$Protein
Cann_DELLA_FC[1] <- NULL

min(Cann_DELLA_FC, na.rm = TRUE)
max(Cann_DELLA_FC, na.rm = TRUE)
breaks_Cann_DELLA=c(seq(-4.5,0, length.out = 100), seq(0.01,6,length.out = 100))

pheatmap(Cann_DELLA_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Cann_DELLA,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_DELLA_Annotation, annotation_colors=color_DELLA, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Red"),
         cex = 1,
         angle_col=45,
         main = "DELLA",
         annotation_legend = TRUE,
         display_numbers = Cann_DELLA_DEG,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)


rownames(Cann_DELLA_TPM) <- Cann_DELLA_TPM$Protein
Cann_DELLA_TPM[1] <- NULL

min(Cann_DELLA_TPM, na.rm = TRUE)
max(Cann_DELLA_TPM, na.rm = TRUE)
breaks_Cann_DELLA_TPM=c(seq(0,100, length.out = 200))

Cann_DELLA_TPM_norm <- normalize_df(Cann_DELLA_TPM)

pheatmap(Cann_DELLA_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Cann_DELLA_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_DELLA_Annotation, annotation_colors=color_DELLA, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "DELLA",
         annotation_legend = TRUE,
         #display_numbers = Cann_DELLA_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(4,9)
)


#########################################################################################################################################
####################################  Jasmo  #############################################################################################
#########################################################################################################################################

Jasmo_Ath <- read.table("Funcional Groups/Ath_Jasmo.txt", header = TRUE)
Jasmo_Ath <- merge(Jasmo_Ath, Ath_Gene_Models, by.x="ID", by.y=1)
Jasmo_Ath_fasta <- Ath11_fasta[c(which(names(Ath11_fasta) %in% Jasmo_Ath$V2))]
Jasmo_Ath_2 <- Jasmo_Ath %>% select(V2, Name)
write.fasta(Jasmo_Ath_fasta, as.string=TRUE, names= names(Jasmo_Ath_fasta),
            file.out = "Funcional Groups/Ath_Jasmo.fasta")
rename.fasta(infile = "Funcional Groups/Ath_Jasmo.fasta", Jasmo_Ath_2, outfile = "Funcional Groups/Ath_Jasmo_2.fasta")

Jasmo_Niben_Candidates <- read.table("Funcional Groups/Niben_Candidates_Jasmo.txt")
Jasmo_Niben_Candidates <- Jasmo_Niben_Candidates[Jasmo_Niben_Candidates$V3>=35,] %>% select(V2)
Jasmo_Niben_Candidates <- unique(Jasmo_Niben_Candidates)
Jasmo_Niben_Candidates_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Jasmo_Niben_Candidates$V2))]
write.fasta(Jasmo_Niben_Candidates_fasta, as.string=TRUE, names= names(Jasmo_Niben_Candidates_fasta),
            file.out = "Funcional Groups/Jasmo_Niben_Candidates_35.fasta")

Jasmo_Sly_Candidates <- read.table("Funcional Groups/Sly_Candidates_Jasmo.txt")
Jasmo_Sly_Candidates <- Jasmo_Sly_Candidates[Jasmo_Sly_Candidates$V3>=35,] %>% select(V2)
Jasmo_Sly_Candidates <- unique(Jasmo_Sly_Candidates)
Jasmo_Sly_Candidates_fasta <- Tomato_4.0_fasta[c(which(names(Tomato_4.0_fasta) %in% Jasmo_Sly_Candidates$V2))]
write.fasta(Jasmo_Sly_Candidates_fasta, as.string=TRUE, names= names(Jasmo_Sly_Candidates_fasta),
            file.out = "Funcional Groups/Jasmo_Sly_Candidates_35.fasta")

Niben_Jasmo_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_Jasmo_Ann$Name))]
write.fasta(Niben_Jasmo_fasta, as.string=TRUE, names= names(Niben_Jasmo_fasta),
            file.out = "Funcional Groups/Niben_Jasmo.fasta")
Niben_Jasmo_Ann_2 <- Niben_Jasmo_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_Jasmo.fasta", Niben_Jasmo_Ann, outfile = "Funcional Groups/Niben_Jasmo_2.fasta")

save_heatmaps("Jasmo", height = 10)

## Representing Heatmap

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Niben_Jasmo_Ann <- read.table("Funcional Groups/FigTree/Niben_Jasmo.txt",
                              sep = "\t", header = TRUE)

colnames(Niben_Jasmo_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_Jasmo <- merge(Niben_Jasmo_Ann, TPM_All_FC_Annotated, by="Name")

## We create the annotation for Modules

Niben_Jasmo_Modules <- which_Module_pwd30(Niben_Jasmo_Ann)

## We select only the FC data from the different comparisons
Niben_Jasmo_TPM <- Niben_Jasmo %>% select(Protein, 4:51, 61:66)
Niben_Jasmo_TPM <- Niben_Jasmo_TPM[!duplicated(Niben_Jasmo_TPM$Protein),]
Niben_Jasmo_TPM <- Niben_Jasmo_TPM[order(match(Niben_Jasmo_TPM[,1], Niben_Jasmo_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_Jasmo_FC <- Niben_Jasmo %>% select(Protein, 71, 73:87)
Niben_Jasmo_FC <- Niben_Jasmo_FC[!duplicated(Niben_Jasmo_FC$Protein),]
Niben_Jasmo_FC <- Niben_Jasmo_FC[order(match(Niben_Jasmo_FC[,1], Niben_Jasmo_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_Jasmo_DEG <- Niben_Jasmo %>% select(Protein, 89, 91:105)
Niben_Jasmo_DEG <- Niben_Jasmo_DEG[!duplicated(Niben_Jasmo_DEG$Protein),]
Niben_Jasmo_DEG <- Niben_Jasmo_DEG[order(match(Niben_Jasmo_DEG[,1], Niben_Jasmo_Ann[,2])),]

Niben_Jasmo_DEG$pcrtB22h_vs_GFP22h_filtered <- ifelse(is.na(Niben_Jasmo_DEG$pcrtB22h_vs_GFP22h_filtered) == TRUE, " ", "*")
Niben_Jasmo_DEG$pcrtB25h_vs_GFP25h_filtered <- ifelse(is.na(Niben_Jasmo_DEG$pcrtB25h_vs_GFP25h_filtered) == TRUE, " ", "*")
Niben_Jasmo_DEG$pcrtBAA_vs_GFPAA_filtered <- ifelse(is.na(Niben_Jasmo_DEG$pcrtBAA_vs_GFPAA_filtered) == TRUE, " ", "*")
Niben_Jasmo_DEG$pcrtBBB_vs_GFPBB_filtered <- ifelse(is.na(Niben_Jasmo_DEG$pcrtBBB_vs_GFPBB_filtered) == TRUE, " ", "*")
Niben_Jasmo_DEG$pcrtBYY_vs_GFPYY_filtered <- ifelse(is.na(Niben_Jasmo_DEG$pcrtBYY_vs_GFPYY_filtered) == TRUE, " ", "*")
Niben_Jasmo_DEG$pcrtB28h_vs_GFP28h_filtered <- ifelse(is.na(Niben_Jasmo_DEG$pcrtB28h_vs_GFP28h_filtered) == TRUE, " ", "*")
Niben_Jasmo_DEG$pcrtB34h_vs_GFP34h_filtered <- ifelse(is.na(Niben_Jasmo_DEG$pcrtB34h_vs_GFP34h_filtered) == TRUE, " ", "*")
Niben_Jasmo_DEG$pcrtB37h_vs_GFP37h_filtered <- ifelse(is.na(Niben_Jasmo_DEG$pcrtB37h_vs_GFP37h_filtered) == TRUE, " ", "*")
Niben_Jasmo_DEG$pcrtBZZ_vs_GFPZZ_filtered <- ifelse(is.na(Niben_Jasmo_DEG$pcrtBZZ_vs_GFPZZ_filtered) == TRUE, " ", "*")
Niben_Jasmo_DEG$pcrtBCC_vs_GFPCC_filtered <- ifelse(is.na(Niben_Jasmo_DEG$pcrtBCC_vs_GFPCC_filtered) == TRUE, " ", "*")
Niben_Jasmo_DEG$pcrtB40h_vs_GFP40h_filtered <- ifelse(is.na(Niben_Jasmo_DEG$pcrtB40h_vs_GFP40h_filtered) == TRUE, " ", "*")
Niben_Jasmo_DEG$pcrtBDD_vs_GFPDD_filtered <- ifelse(is.na(Niben_Jasmo_DEG$pcrtBDD_vs_GFPDD_filtered) == TRUE, " ", "*")
Niben_Jasmo_DEG$pcrtB46h_vs_GFP46h_filtered <- ifelse(is.na(Niben_Jasmo_DEG$pcrtB46h_vs_GFP46h_filtered) == TRUE, " ", "*")
Niben_Jasmo_DEG$pcrtB56h_vs_GFP56h_filtered <- ifelse(is.na(Niben_Jasmo_DEG$pcrtB56h_vs_GFP56h_filtered) == TRUE, " ", "*")
Niben_Jasmo_DEG$pcrtBEE_vs_GFPEE_filtered <- ifelse(is.na(Niben_Jasmo_DEG$pcrtBEE_vs_GFPEE_filtered) == TRUE, " ", "*")
Niben_Jasmo_DEG$pcrtB96h_vs_GFP96h_filtered <- ifelse(is.na(Niben_Jasmo_DEG$pcrtB96h_vs_GFP96h_filtered) == TRUE, " ", "*")
row.names(Niben_Jasmo_DEG) <- Niben_Jasmo_DEG$Protein
Niben_Jasmo_DEG[1] <- NULL

Niben_Jasmo_DEG <- Niben_Jasmo_DEG %>%
  select(pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

row.names(Niben_Jasmo_FC) <- Niben_Jasmo_FC$Protein
Niben_Jasmo_FC[1] <- NULL

Niben_Jasmo_FC <- Niben_Jasmo_FC %>%
  select(pcrtB22h_vs_GFP22h_All, pcrtBAA_vs_GFPAA_All, pcrtB25h_vs_GFP25h_All, 
         pcrtBYY_vs_GFPYY_All, pcrtBBB_vs_GFPBB_All, pcrtB28h_vs_GFP28h_All, pcrtB34h_vs_GFP34h_All, pcrtBZZ_vs_GFPZZ_All, 
         pcrtB37h_vs_GFP37h_All, pcrtBCC_vs_GFPCC_All, pcrtB40h_vs_GFP40h_All, pcrtBDD_vs_GFPDD_All, pcrtB46h_vs_GFP46h_All, 
         pcrtBEE_vs_GFPEE_All, pcrtB56h_vs_GFP56h_All, pcrtB96h_vs_GFP96h_All)


min(Niben_Jasmo_FC, na.rm = TRUE)
max(Niben_Jasmo_FC, na.rm = TRUE)
breaks_Jasmo_FC=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Niben_Jasmo_FC, # %>% select(1:15), 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Jasmo_FC,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_Jasmo_Annotation, annotation_colors=color_Jasmo, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h", "96h"),
         cex = 1,
         angle_col=45,
         main = "Jasmo",
         annotation_legend = TRUE,
         display_numbers = Niben_Jasmo_DEG, # %>% select(1:15),
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(84, 104),
         annotation_row = Niben_Jasmo_Modules,
         annotation_colors = module_colors
         #gaps_col = c(15)
)


rownames(Niben_Jasmo_TPM) <- Niben_Jasmo_TPM$Protein
Niben_Jasmo_TPM[1] <- NULL

min(Niben_Jasmo_TPM, na.rm = TRUE)
max(Niben_Jasmo_TPM, na.rm = TRUE)
breaks_Niben_Jasmo_TPM=c(seq(0,100, length.out = 100))

Niben_Jasmo_TPM_norm <- normalize_df(Niben_Jasmo_TPM)
Niben_Jasmo_TPM_norm_Cornell <- Niben_Jasmo_TPM %>%
  select(1:48) 
Niben_Jasmo_TPM_norm_Cornell <- normalize_df(Niben_Jasmo_TPM_norm_Cornell)


pheatmap(Niben_Jasmo_TPM_norm_Cornell, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Niben_Jasmo_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_Jasmo_Annotation, annotation_colors=color_Jasmo, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Jasmo",
         annotation_legend = TRUE,
         #display_numbers = Niben_Jasmo_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         annotation_row = Niben_Jasmo_Modules,
         annotation_colors = module_colors,
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(24)#, 48, 51)
)



#####Representing Heatmap for Sly

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Sly_Jasmo_Ann <- read.table("Funcional Groups/FigTree/Sly_Jasmo.txt",
                            sep = "\t", header = TRUE)

colnames(Sly_Jasmo_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Sly_Jasmo <- merge(Sly_Jasmo_Ann, Summary_Yazdani_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Sly_Jasmo_TPM <- Sly_Jasmo %>% select(Protein, 4:66)
Sly_Jasmo_TPM <- Sly_Jasmo_TPM[!duplicated(Sly_Jasmo_TPM$Protein),]
Sly_Jasmo_TPM <- Sly_Jasmo_TPM[order(match(Sly_Jasmo_TPM[,1], Sly_Jasmo_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_Jasmo_FC <- Sly_Jasmo %>% select(Protein, 67:71)
Sly_Jasmo_FC <- Sly_Jasmo_FC[!duplicated(Sly_Jasmo_FC$Protein),]
Sly_Jasmo_FC <- Sly_Jasmo_FC[order(match(Sly_Jasmo_FC[,1], Sly_Jasmo_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_Jasmo_DEG <- Sly_Jasmo %>% select(Protein, 72:76)
Sly_Jasmo_DEG <- Sly_Jasmo_DEG[!duplicated(Sly_Jasmo_DEG$Protein),]
Sly_Jasmo_DEG <- Sly_Jasmo_DEG[order(match(Sly_Jasmo_DEG[,1], Sly_Jasmo_Ann[,2])),]

Sly_Jasmo_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Sly_Jasmo_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Sly_Jasmo_DEG$Ripening_Orange_DEG <- ifelse(is.na(Sly_Jasmo_DEG$Ripening_Orange_DEG) == TRUE, " ", "*")
Sly_Jasmo_DEG$Ripening_Red_DEG <- ifelse(is.na(Sly_Jasmo_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
Sly_Jasmo_DEG$ORHis_MG_DEG <- ifelse(is.na(Sly_Jasmo_DEG$ORHis_MG_DEG) == TRUE, " ", "*")
Sly_Jasmo_DEG$ORHis_Breaker_DEG <- ifelse(is.na(Sly_Jasmo_DEG$ORHis_Breaker_DEG) == TRUE, " ", "*")
row.names(Sly_Jasmo_DEG) <- Sly_Jasmo_DEG$Protein
Sly_Jasmo_DEG[1] <- NULL

rownames(Sly_Jasmo_FC) <- Sly_Jasmo_FC$Protein
Sly_Jasmo_FC[1] <- NULL

min(Sly_Jasmo_FC, na.rm = TRUE)
max(Sly_Jasmo_FC, na.rm = TRUE)
breaks_Sly_Jasmo=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Sly_Jasmo_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Sly_Jasmo,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_Jasmo_Annotation, annotation_colors=color_Jasmo, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Jasmo",
         annotation_legend = TRUE,
         display_numbers = Sly_Jasmo_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(39, 48),
         gaps_col = c(3)
)


rownames(Sly_Jasmo_TPM) <- Sly_Jasmo_TPM$Protein
Sly_Jasmo_TPM[1] <- NULL

min(Sly_Jasmo_TPM, na.rm = TRUE)
max(Sly_Jasmo_TPM, na.rm = TRUE)
breaks_Sly_Jasmo_TPM=c(seq(0,100, length.out = 200))

Sly_Jasmo_TPM_norm <- normalize_df(Sly_Jasmo_TPM)

pheatmap(Sly_Jasmo_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Sly_Jasmo_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_Jasmo_Annotation, annotation_colors=color_Jasmo, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Jasmo",
         annotation_legend = TRUE,
         #display_numbers = Sly_Jasmo_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(21,42)
)


#####Representing Heatmap for Cann

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Cann_Jasmo_Ann <- read.table("Funcional Groups/FigTree/Cann_Jasmo.txt",
                             sep = "\t", header = TRUE)

colnames(Cann_Jasmo_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Cann_Jasmo <- merge(Cann_Jasmo_Ann, Summary_Pepper_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Cann_Jasmo_TPM <- Cann_Jasmo %>% select(Protein, 4:17)
Cann_Jasmo_TPM <- Cann_Jasmo_TPM[!duplicated(Cann_Jasmo_TPM$Protein),]
Cann_Jasmo_TPM <- Cann_Jasmo_TPM[order(match(Cann_Jasmo_TPM[,1], Cann_Jasmo_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_Jasmo_FC <- Cann_Jasmo %>% select(Protein, 18:19)
Cann_Jasmo_FC <- Cann_Jasmo_FC[!duplicated(Cann_Jasmo_FC$Protein),]
Cann_Jasmo_FC <- Cann_Jasmo_FC[order(match(Cann_Jasmo_FC[,1], Cann_Jasmo_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_Jasmo_DEG <- Cann_Jasmo %>% select(Protein, 20:21)
Cann_Jasmo_DEG <- Cann_Jasmo_DEG[!duplicated(Cann_Jasmo_DEG$Protein),]
Cann_Jasmo_DEG <- Cann_Jasmo_DEG[order(match(Cann_Jasmo_DEG[,1], Cann_Jasmo_Ann[,2])),]

Cann_Jasmo_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Cann_Jasmo_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Cann_Jasmo_DEG$Ripening_Red_DEG <- ifelse(is.na(Cann_Jasmo_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
row.names(Cann_Jasmo_DEG) <- Cann_Jasmo_DEG$Protein
Cann_Jasmo_DEG[1] <- NULL

rownames(Cann_Jasmo_FC) <- Cann_Jasmo_FC$Protein
Cann_Jasmo_FC[1] <- NULL

min(Cann_Jasmo_FC, na.rm = TRUE)
max(Cann_Jasmo_FC, na.rm = TRUE)
breaks_Cann_Jasmo=c(seq(-4.5,0, length.out = 100), seq(0.01,6,length.out = 100))

pheatmap(Cann_Jasmo_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Cann_Jasmo,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_Jasmo_Annotation, annotation_colors=color_Jasmo, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Red"),
         cex = 1,
         angle_col=45,
         main = "Jasmo",
         annotation_legend = TRUE,
         display_numbers = Cann_Jasmo_DEG,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)


rownames(Cann_Jasmo_TPM) <- Cann_Jasmo_TPM$Protein
Cann_Jasmo_TPM[1] <- NULL

min(Cann_Jasmo_TPM, na.rm = TRUE)
max(Cann_Jasmo_TPM, na.rm = TRUE)
breaks_Cann_Jasmo_TPM=c(seq(0,100, length.out = 200))

Cann_Jasmo_TPM_norm <- normalize_df(Cann_Jasmo_TPM)

pheatmap(Cann_Jasmo_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Cann_Jasmo_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_Jasmo_Annotation, annotation_colors=color_Jasmo, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Jasmo",
         annotation_legend = TRUE,
         #display_numbers = Cann_Jasmo_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(4,9)
)


#########################################################################################################################################
####################################  ABA  #############################################################################################
#########################################################################################################################################

ABA_Ath <- read.table("Funcional Groups/Ath_ABA.txt", header = TRUE)
ABA_Ath <- merge(ABA_Ath, Ath_Gene_Models, by.x="ID", by.y=1)
ABA_Ath_fasta <- Ath11_fasta[c(which(names(Ath11_fasta) %in% ABA_Ath$V2))]
ABA_Ath_2 <- ABA_Ath %>% select(V2, Name)
write.fasta(ABA_Ath_fasta, as.string=TRUE, names= names(ABA_Ath_fasta),
            file.out = "Funcional Groups/Ath_ABA.fasta")
rename.fasta(infile = "Funcional Groups/Ath_ABA.fasta", ABA_Ath_2, outfile = "Funcional Groups/Ath_ABA_2.fasta")

ABA_Niben_Candidates <- read.table("Funcional Groups/Niben_Candidates_ABA.txt")
ABA_Niben_Candidates <- ABA_Niben_Candidates[ABA_Niben_Candidates$V3>=35,] %>% select(V2)
ABA_Niben_Candidates <- unique(ABA_Niben_Candidates)
ABA_Niben_Candidates_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% ABA_Niben_Candidates$V2))]
write.fasta(ABA_Niben_Candidates_fasta, as.string=TRUE, names= names(ABA_Niben_Candidates_fasta),
            file.out = "Funcional Groups/ABA_Niben_Candidates_35.fasta")

ABA_Sly_Candidates <- read.table("Funcional Groups/Sly_Candidates_ABA.txt")
ABA_Sly_Candidates <- ABA_Sly_Candidates[ABA_Sly_Candidates$V3>=35,] %>% select(V2)
ABA_Sly_Candidates <- unique(ABA_Sly_Candidates)
ABA_Sly_Candidates_fasta <- Tomato_4.0_fasta[c(which(names(Tomato_4.0_fasta) %in% ABA_Sly_Candidates$V2))]
write.fasta(ABA_Sly_Candidates_fasta, as.string=TRUE, names= names(ABA_Sly_Candidates_fasta),
            file.out = "Funcional Groups/ABA_Sly_Candidates_35.fasta")

Niben_ABA_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_ABA_Ann$Name))]
write.fasta(Niben_ABA_fasta, as.string=TRUE, names= names(Niben_ABA_fasta),
            file.out = "Funcional Groups/Niben_ABA.fasta")
Niben_ABA_Ann_2 <- Niben_ABA_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_ABA.fasta", Niben_ABA_Ann, outfile = "Funcional Groups/Niben_ABA_2.fasta")

save_heatmaps("ABA", height = 8)

## Representing Heatmap

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Niben_ABA_Ann <- read.table("Funcional Groups/FigTree/Niben_ABA.txt",
                              sep = "\t", header = TRUE)

colnames(Niben_ABA_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_ABA <- merge(Niben_ABA_Ann, TPM_All_FC_Annotated, by="Name")

## We create the annotation for Modules

Niben_ABA_Modules <- which_Module_pwd30(Niben_ABA_Ann)

## We select only the FC data from the different comparisons
Niben_ABA_TPM <- Niben_ABA %>% select(Protein, 4:51, 61:66)
Niben_ABA_TPM <- Niben_ABA_TPM[!duplicated(Niben_ABA_TPM$Protein),]
Niben_ABA_TPM <- Niben_ABA_TPM[order(match(Niben_ABA_TPM[,1], Niben_ABA_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_ABA_FC <- Niben_ABA %>% select(Protein, 71, 73:87)
Niben_ABA_FC <- Niben_ABA_FC[!duplicated(Niben_ABA_FC$Protein),]
Niben_ABA_FC <- Niben_ABA_FC[order(match(Niben_ABA_FC[,1], Niben_ABA_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_ABA_DEG <- Niben_ABA %>% select(Protein, 89, 91:105)
Niben_ABA_DEG <- Niben_ABA_DEG[!duplicated(Niben_ABA_DEG$Protein),]
Niben_ABA_DEG <- Niben_ABA_DEG[order(match(Niben_ABA_DEG[,1], Niben_ABA_Ann[,2])),]

Niben_ABA_DEG$pcrtB22h_vs_GFP22h_filtered <- ifelse(is.na(Niben_ABA_DEG$pcrtB22h_vs_GFP22h_filtered) == TRUE, " ", "*")
Niben_ABA_DEG$pcrtB25h_vs_GFP25h_filtered <- ifelse(is.na(Niben_ABA_DEG$pcrtB25h_vs_GFP25h_filtered) == TRUE, " ", "*")
Niben_ABA_DEG$pcrtBAA_vs_GFPAA_filtered <- ifelse(is.na(Niben_ABA_DEG$pcrtBAA_vs_GFPAA_filtered) == TRUE, " ", "*")
Niben_ABA_DEG$pcrtBBB_vs_GFPBB_filtered <- ifelse(is.na(Niben_ABA_DEG$pcrtBBB_vs_GFPBB_filtered) == TRUE, " ", "*")
Niben_ABA_DEG$pcrtBYY_vs_GFPYY_filtered <- ifelse(is.na(Niben_ABA_DEG$pcrtBYY_vs_GFPYY_filtered) == TRUE, " ", "*")
Niben_ABA_DEG$pcrtB28h_vs_GFP28h_filtered <- ifelse(is.na(Niben_ABA_DEG$pcrtB28h_vs_GFP28h_filtered) == TRUE, " ", "*")
Niben_ABA_DEG$pcrtB34h_vs_GFP34h_filtered <- ifelse(is.na(Niben_ABA_DEG$pcrtB34h_vs_GFP34h_filtered) == TRUE, " ", "*")
Niben_ABA_DEG$pcrtB37h_vs_GFP37h_filtered <- ifelse(is.na(Niben_ABA_DEG$pcrtB37h_vs_GFP37h_filtered) == TRUE, " ", "*")
Niben_ABA_DEG$pcrtBZZ_vs_GFPZZ_filtered <- ifelse(is.na(Niben_ABA_DEG$pcrtBZZ_vs_GFPZZ_filtered) == TRUE, " ", "*")
Niben_ABA_DEG$pcrtBCC_vs_GFPCC_filtered <- ifelse(is.na(Niben_ABA_DEG$pcrtBCC_vs_GFPCC_filtered) == TRUE, " ", "*")
Niben_ABA_DEG$pcrtB40h_vs_GFP40h_filtered <- ifelse(is.na(Niben_ABA_DEG$pcrtB40h_vs_GFP40h_filtered) == TRUE, " ", "*")
Niben_ABA_DEG$pcrtBDD_vs_GFPDD_filtered <- ifelse(is.na(Niben_ABA_DEG$pcrtBDD_vs_GFPDD_filtered) == TRUE, " ", "*")
Niben_ABA_DEG$pcrtB46h_vs_GFP46h_filtered <- ifelse(is.na(Niben_ABA_DEG$pcrtB46h_vs_GFP46h_filtered) == TRUE, " ", "*")
Niben_ABA_DEG$pcrtB56h_vs_GFP56h_filtered <- ifelse(is.na(Niben_ABA_DEG$pcrtB56h_vs_GFP56h_filtered) == TRUE, " ", "*")
Niben_ABA_DEG$pcrtBEE_vs_GFPEE_filtered <- ifelse(is.na(Niben_ABA_DEG$pcrtBEE_vs_GFPEE_filtered) == TRUE, " ", "*")
Niben_ABA_DEG$pcrtB96h_vs_GFP96h_filtered <- ifelse(is.na(Niben_ABA_DEG$pcrtB96h_vs_GFP96h_filtered) == TRUE, " ", "*")
row.names(Niben_ABA_DEG) <- Niben_ABA_DEG$Protein
Niben_ABA_DEG[1] <- NULL

Niben_ABA_DEG <- Niben_ABA_DEG %>%
  select(pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

row.names(Niben_ABA_FC) <- Niben_ABA_FC$Protein
Niben_ABA_FC[1] <- NULL

Niben_ABA_FC <- Niben_ABA_FC %>%
  select(pcrtB22h_vs_GFP22h_All, pcrtBAA_vs_GFPAA_All, pcrtB25h_vs_GFP25h_All, 
         pcrtBYY_vs_GFPYY_All, pcrtBBB_vs_GFPBB_All, pcrtB28h_vs_GFP28h_All, pcrtB34h_vs_GFP34h_All, pcrtBZZ_vs_GFPZZ_All, 
         pcrtB37h_vs_GFP37h_All, pcrtBCC_vs_GFPCC_All, pcrtB40h_vs_GFP40h_All, pcrtBDD_vs_GFPDD_All, pcrtB46h_vs_GFP46h_All, 
         pcrtBEE_vs_GFPEE_All, pcrtB56h_vs_GFP56h_All, pcrtB96h_vs_GFP96h_All)


min(Niben_ABA_FC, na.rm = TRUE)
max(Niben_ABA_FC, na.rm = TRUE)
breaks_ABA_FC=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Niben_ABA_FC %>% select(1:15), 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_ABA_FC,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_ABA_Annotation, annotation_colors=color_ABA, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h"), #, "96h"),
         cex = 1,
         angle_col=45,
         main = "ABA",
         annotation_legend = TRUE,
         display_numbers = Niben_ABA_DEG %>% select(1:15),
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(84, 104),
         annotation_row = Niben_ABA_Modules,
         annotation_colors = module_colors
         #gaps_col = c(15)
)


rownames(Niben_ABA_TPM) <- Niben_ABA_TPM$Protein
Niben_ABA_TPM[1] <- NULL

min(Niben_ABA_TPM, na.rm = TRUE)
max(Niben_ABA_TPM, na.rm = TRUE)
breaks_Niben_ABA_TPM=c(seq(0,100, length.out = 100))

Niben_ABA_TPM_norm <- normalize_df(Niben_ABA_TPM)
Niben_ABA_TPM_norm_Cornell <- Niben_ABA_TPM %>%
  select(1:48) 
Niben_ABA_TPM_norm_Cornell <- normalize_df(Niben_ABA_TPM_norm_Cornell)


pheatmap(Niben_ABA_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Niben_ABA_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_ABA_Annotation, annotation_colors=color_ABA, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "ABA",
         annotation_legend = TRUE,
         #display_numbers = Niben_ABA_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         annotation_row = Niben_ABA_Modules,
         annotation_colors = module_colors,
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(24, 48, 51)
)



#####Representing Heatmap for Sly

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Sly_ABA_Ann <- read.table("Funcional Groups/FigTree/Sly_ABA.txt",
                            sep = "\t", header = TRUE)

colnames(Sly_ABA_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Sly_ABA <- merge(Sly_ABA_Ann, Summary_Yazdani_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Sly_ABA_TPM <- Sly_ABA %>% select(Protein, 4:66)
Sly_ABA_TPM <- Sly_ABA_TPM[!duplicated(Sly_ABA_TPM$Protein),]
Sly_ABA_TPM <- Sly_ABA_TPM[order(match(Sly_ABA_TPM[,1], Sly_ABA_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_ABA_FC <- Sly_ABA %>% select(Protein, 67:71)
Sly_ABA_FC <- Sly_ABA_FC[!duplicated(Sly_ABA_FC$Protein),]
Sly_ABA_FC <- Sly_ABA_FC[order(match(Sly_ABA_FC[,1], Sly_ABA_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_ABA_DEG <- Sly_ABA %>% select(Protein, 72:76)
Sly_ABA_DEG <- Sly_ABA_DEG[!duplicated(Sly_ABA_DEG$Protein),]
Sly_ABA_DEG <- Sly_ABA_DEG[order(match(Sly_ABA_DEG[,1], Sly_ABA_Ann[,2])),]

Sly_ABA_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Sly_ABA_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Sly_ABA_DEG$Ripening_Orange_DEG <- ifelse(is.na(Sly_ABA_DEG$Ripening_Orange_DEG) == TRUE, " ", "*")
Sly_ABA_DEG$Ripening_Red_DEG <- ifelse(is.na(Sly_ABA_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
Sly_ABA_DEG$ORHis_MG_DEG <- ifelse(is.na(Sly_ABA_DEG$ORHis_MG_DEG) == TRUE, " ", "*")
Sly_ABA_DEG$ORHis_Breaker_DEG <- ifelse(is.na(Sly_ABA_DEG$ORHis_Breaker_DEG) == TRUE, " ", "*")
row.names(Sly_ABA_DEG) <- Sly_ABA_DEG$Protein
Sly_ABA_DEG[1] <- NULL

rownames(Sly_ABA_FC) <- Sly_ABA_FC$Protein
Sly_ABA_FC[1] <- NULL

min(Sly_ABA_FC, na.rm = TRUE)
max(Sly_ABA_FC, na.rm = TRUE)
breaks_Sly_ABA=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Sly_ABA_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Sly_ABA,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_ABA_Annotation, annotation_colors=color_ABA, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "ABA",
         annotation_legend = TRUE,
         display_numbers = Sly_ABA_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(39, 48),
         gaps_col = c(3)
)


rownames(Sly_ABA_TPM) <- Sly_ABA_TPM$Protein
Sly_ABA_TPM[1] <- NULL

min(Sly_ABA_TPM, na.rm = TRUE)
max(Sly_ABA_TPM, na.rm = TRUE)
breaks_Sly_ABA_TPM=c(seq(0,100, length.out = 200))

Sly_ABA_TPM_norm <- normalize_df(Sly_ABA_TPM)

pheatmap(Sly_ABA_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Sly_ABA_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_ABA_Annotation, annotation_colors=color_ABA, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "ABA",
         annotation_legend = TRUE,
         #display_numbers = Sly_ABA_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(21,42)
)


#####Representing Heatmap for Cann

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Cann_ABA_Ann <- read.table("Funcional Groups/FigTree/Cann_ABA.txt",
                             sep = "\t", header = TRUE)

colnames(Cann_ABA_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Cann_ABA <- merge(Cann_ABA_Ann, Summary_Pepper_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Cann_ABA_TPM <- Cann_ABA %>% select(Protein, 4:17)
Cann_ABA_TPM <- Cann_ABA_TPM[!duplicated(Cann_ABA_TPM$Protein),]
Cann_ABA_TPM <- Cann_ABA_TPM[order(match(Cann_ABA_TPM[,1], Cann_ABA_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_ABA_FC <- Cann_ABA %>% select(Protein, 18:19)
Cann_ABA_FC <- Cann_ABA_FC[!duplicated(Cann_ABA_FC$Protein),]
Cann_ABA_FC <- Cann_ABA_FC[order(match(Cann_ABA_FC[,1], Cann_ABA_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_ABA_DEG <- Cann_ABA %>% select(Protein, 20:21)
Cann_ABA_DEG <- Cann_ABA_DEG[!duplicated(Cann_ABA_DEG$Protein),]
Cann_ABA_DEG <- Cann_ABA_DEG[order(match(Cann_ABA_DEG[,1], Cann_ABA_Ann[,2])),]

Cann_ABA_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Cann_ABA_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Cann_ABA_DEG$Ripening_Red_DEG <- ifelse(is.na(Cann_ABA_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
row.names(Cann_ABA_DEG) <- Cann_ABA_DEG$Protein
Cann_ABA_DEG[1] <- NULL

rownames(Cann_ABA_FC) <- Cann_ABA_FC$Protein
Cann_ABA_FC[1] <- NULL

min(Cann_ABA_FC, na.rm = TRUE)
max(Cann_ABA_FC, na.rm = TRUE)
breaks_Cann_ABA=c(seq(-4.5,0, length.out = 100), seq(0.01,6,length.out = 100))

pheatmap(Cann_ABA_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Cann_ABA,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_ABA_Annotation, annotation_colors=color_ABA, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Red"),
         cex = 1,
         angle_col=45,
         main = "ABA",
         annotation_legend = TRUE,
         display_numbers = Cann_ABA_DEG,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)


rownames(Cann_ABA_TPM) <- Cann_ABA_TPM$Protein
Cann_ABA_TPM[1] <- NULL

min(Cann_ABA_TPM, na.rm = TRUE)
max(Cann_ABA_TPM, na.rm = TRUE)
breaks_Cann_ABA_TPM=c(seq(0,100, length.out = 200))

Cann_ABA_TPM_norm <- normalize_df(Cann_ABA_TPM)

pheatmap(Cann_ABA_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Cann_ABA_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_ABA_Annotation, annotation_colors=color_ABA, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "ABA",
         annotation_legend = TRUE,
         #display_numbers = Cann_ABA_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(4,9)
)


#########################################################################################################################################
####################################  Auxin  #############################################################################################
#########################################################################################################################################

Auxin_Ath <- read.table("Funcional Groups/Ath_Auxin.txt", header = TRUE)
Auxin_Ath <- merge(Auxin_Ath, Ath_Gene_Models, by.x="ID", by.y=1)
Auxin_Ath_fasta <- Ath11_fasta[c(which(names(Ath11_fasta) %in% Auxin_Ath$V2))]
Auxin_Ath_2 <- Auxin_Ath %>% select(V2, Name)
write.fasta(Auxin_Ath_fasta, as.string=TRUE, names= names(Auxin_Ath_fasta),
            file.out = "Funcional Groups/Ath_Auxin.fasta")
rename.fasta(infile = "Funcional Groups/Ath_Auxin.fasta", Auxin_Ath_2, outfile = "Funcional Groups/Ath_Auxin_2.fasta")

Auxin_Niben_Candidates <- read.table("Funcional Groups/Niben_Candidates_Auxin.txt")
Auxin_Niben_Candidates <- Auxin_Niben_Candidates[Auxin_Niben_Candidates$V3>=35,] %>% select(V2)
Auxin_Niben_Candidates <- unique(Auxin_Niben_Candidates)
Auxin_Niben_Candidates_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Auxin_Niben_Candidates$V2))]
write.fasta(Auxin_Niben_Candidates_fasta, as.string=TRUE, names= names(Auxin_Niben_Candidates_fasta),
            file.out = "Funcional Groups/Auxin_Niben_Candidates_35.fasta")

Auxin_Sly_Candidates <- read.table("Funcional Groups/Sly_Candidates_Auxin.txt")
Auxin_Sly_Candidates <- Auxin_Sly_Candidates[Auxin_Sly_Candidates$V3>=35,] %>% select(V2)
Auxin_Sly_Candidates <- unique(Auxin_Sly_Candidates)
Auxin_Sly_Candidates_fasta <- Tomato_4.0_fasta[c(which(names(Tomato_4.0_fasta) %in% Auxin_Sly_Candidates$V2))]
write.fasta(Auxin_Sly_Candidates_fasta, as.string=TRUE, names= names(Auxin_Sly_Candidates_fasta),
            file.out = "Funcional Groups/Auxin_Sly_Candidates_35.fasta")

Niben_Auxin_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_Auxin_Ann$Name))]
write.fasta(Niben_Auxin_fasta, as.string=TRUE, names= names(Niben_Auxin_fasta),
            file.out = "Funcional Groups/Niben_Auxin.fasta")
Niben_Auxin_Ann_2 <- Niben_Auxin_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_Auxin.fasta", Niben_Auxin_Ann, outfile = "Funcional Groups/Niben_Auxin_2.fasta")

save_heatmaps("Auxin", height = 10)

## Representing Heatmap

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Niben_Auxin_Ann <- read.table("Funcional Groups/FigTree/Niben_Auxin.txt",
                            sep = "\t", header = TRUE)

colnames(Niben_Auxin_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_Auxin <- merge(Niben_Auxin_Ann, TPM_All_FC_Annotated, by="Name")

## We create the annotation for Modules

Niben_Auxin_Modules <- which_Module_pwd30(Niben_Auxin_Ann)

## We select only the FC data from the different comparisons
Niben_Auxin_TPM <- Niben_Auxin %>% select(Protein, 4:51, 61:66)
Niben_Auxin_TPM <- Niben_Auxin_TPM[!duplicated(Niben_Auxin_TPM$Protein),]
Niben_Auxin_TPM <- Niben_Auxin_TPM[order(match(Niben_Auxin_TPM[,1], Niben_Auxin_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_Auxin_FC <- Niben_Auxin %>% select(Protein, 71, 73:87)
Niben_Auxin_FC <- Niben_Auxin_FC[!duplicated(Niben_Auxin_FC$Protein),]
Niben_Auxin_FC <- Niben_Auxin_FC[order(match(Niben_Auxin_FC[,1], Niben_Auxin_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_Auxin_DEG <- Niben_Auxin %>% select(Protein, 89, 91:105)
Niben_Auxin_DEG <- Niben_Auxin_DEG[!duplicated(Niben_Auxin_DEG$Protein),]
Niben_Auxin_DEG <- Niben_Auxin_DEG[order(match(Niben_Auxin_DEG[,1], Niben_Auxin_Ann[,2])),]

Niben_Auxin_DEG$pcrtB22h_vs_GFP22h_filtered <- ifelse(is.na(Niben_Auxin_DEG$pcrtB22h_vs_GFP22h_filtered) == TRUE, " ", "*")
Niben_Auxin_DEG$pcrtB25h_vs_GFP25h_filtered <- ifelse(is.na(Niben_Auxin_DEG$pcrtB25h_vs_GFP25h_filtered) == TRUE, " ", "*")
Niben_Auxin_DEG$pcrtBAA_vs_GFPAA_filtered <- ifelse(is.na(Niben_Auxin_DEG$pcrtBAA_vs_GFPAA_filtered) == TRUE, " ", "*")
Niben_Auxin_DEG$pcrtBBB_vs_GFPBB_filtered <- ifelse(is.na(Niben_Auxin_DEG$pcrtBBB_vs_GFPBB_filtered) == TRUE, " ", "*")
Niben_Auxin_DEG$pcrtBYY_vs_GFPYY_filtered <- ifelse(is.na(Niben_Auxin_DEG$pcrtBYY_vs_GFPYY_filtered) == TRUE, " ", "*")
Niben_Auxin_DEG$pcrtB28h_vs_GFP28h_filtered <- ifelse(is.na(Niben_Auxin_DEG$pcrtB28h_vs_GFP28h_filtered) == TRUE, " ", "*")
Niben_Auxin_DEG$pcrtB34h_vs_GFP34h_filtered <- ifelse(is.na(Niben_Auxin_DEG$pcrtB34h_vs_GFP34h_filtered) == TRUE, " ", "*")
Niben_Auxin_DEG$pcrtB37h_vs_GFP37h_filtered <- ifelse(is.na(Niben_Auxin_DEG$pcrtB37h_vs_GFP37h_filtered) == TRUE, " ", "*")
Niben_Auxin_DEG$pcrtBZZ_vs_GFPZZ_filtered <- ifelse(is.na(Niben_Auxin_DEG$pcrtBZZ_vs_GFPZZ_filtered) == TRUE, " ", "*")
Niben_Auxin_DEG$pcrtBCC_vs_GFPCC_filtered <- ifelse(is.na(Niben_Auxin_DEG$pcrtBCC_vs_GFPCC_filtered) == TRUE, " ", "*")
Niben_Auxin_DEG$pcrtB40h_vs_GFP40h_filtered <- ifelse(is.na(Niben_Auxin_DEG$pcrtB40h_vs_GFP40h_filtered) == TRUE, " ", "*")
Niben_Auxin_DEG$pcrtBDD_vs_GFPDD_filtered <- ifelse(is.na(Niben_Auxin_DEG$pcrtBDD_vs_GFPDD_filtered) == TRUE, " ", "*")
Niben_Auxin_DEG$pcrtB46h_vs_GFP46h_filtered <- ifelse(is.na(Niben_Auxin_DEG$pcrtB46h_vs_GFP46h_filtered) == TRUE, " ", "*")
Niben_Auxin_DEG$pcrtB56h_vs_GFP56h_filtered <- ifelse(is.na(Niben_Auxin_DEG$pcrtB56h_vs_GFP56h_filtered) == TRUE, " ", "*")
Niben_Auxin_DEG$pcrtBEE_vs_GFPEE_filtered <- ifelse(is.na(Niben_Auxin_DEG$pcrtBEE_vs_GFPEE_filtered) == TRUE, " ", "*")
Niben_Auxin_DEG$pcrtB96h_vs_GFP96h_filtered <- ifelse(is.na(Niben_Auxin_DEG$pcrtB96h_vs_GFP96h_filtered) == TRUE, " ", "*")
row.names(Niben_Auxin_DEG) <- Niben_Auxin_DEG$Protein
Niben_Auxin_DEG[1] <- NULL

Niben_Auxin_DEG <- Niben_Auxin_DEG %>%
  select(pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

row.names(Niben_Auxin_FC) <- Niben_Auxin_FC$Protein
Niben_Auxin_FC[1] <- NULL

Niben_Auxin_FC <- Niben_Auxin_FC %>%
  select(pcrtB22h_vs_GFP22h_All, pcrtBAA_vs_GFPAA_All, pcrtB25h_vs_GFP25h_All, 
         pcrtBYY_vs_GFPYY_All, pcrtBBB_vs_GFPBB_All, pcrtB28h_vs_GFP28h_All, pcrtB34h_vs_GFP34h_All, pcrtBZZ_vs_GFPZZ_All, 
         pcrtB37h_vs_GFP37h_All, pcrtBCC_vs_GFPCC_All, pcrtB40h_vs_GFP40h_All, pcrtBDD_vs_GFPDD_All, pcrtB46h_vs_GFP46h_All, 
         pcrtBEE_vs_GFPEE_All, pcrtB56h_vs_GFP56h_All, pcrtB96h_vs_GFP96h_All)


min(Niben_Auxin_FC, na.rm = TRUE)
max(Niben_Auxin_FC, na.rm = TRUE)
breaks_Auxin_FC=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Niben_Auxin_FC, # %>% select(1:15), 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Auxin_FC,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_Auxin_Annotation, annotation_colors=color_Auxin, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h", "96h"),
         cex = 1,
         angle_col=45,
         main = "Auxin",
         annotation_legend = TRUE,
         display_numbers = Niben_Auxin_DEG, # %>% select(1:15),
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(84, 104),
         annotation_row = Niben_Auxin_Modules,
         annotation_colors = module_colors
         #gaps_col = c(15)
)


rownames(Niben_Auxin_TPM) <- Niben_Auxin_TPM$Protein
Niben_Auxin_TPM[1] <- NULL

min(Niben_Auxin_TPM, na.rm = TRUE)
max(Niben_Auxin_TPM, na.rm = TRUE)
breaks_Niben_Auxin_TPM=c(seq(0,100, length.out = 100))

Niben_Auxin_TPM_norm <- normalize_df(Niben_Auxin_TPM)
Niben_Auxin_TPM_norm_Cornell <- Niben_Auxin_TPM %>%
  select(1:48) 
Niben_Auxin_TPM_norm_Cornell <- normalize_df(Niben_Auxin_TPM_norm_Cornell)


pheatmap(Niben_Auxin_TPM_norm_Cornell, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Niben_Auxin_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_Auxin_Annotation, annotation_colors=color_Auxin, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Auxin",
         annotation_legend = TRUE,
         #display_numbers = Niben_Auxin_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         annotation_row = Niben_Auxin_Modules,
         annotation_colors = module_colors,
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(24)#, 48, 51)
)



#####Representing Heatmap for Sly

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Sly_Auxin_Ann <- read.table("Funcional Groups/FigTree/Sly_Auxin.txt",
                          sep = "\t", header = TRUE)

colnames(Sly_Auxin_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Sly_Auxin <- merge(Sly_Auxin_Ann, Summary_Yazdani_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Sly_Auxin_TPM <- Sly_Auxin %>% select(Protein, 4:66)
Sly_Auxin_TPM <- Sly_Auxin_TPM[!duplicated(Sly_Auxin_TPM$Protein),]
Sly_Auxin_TPM <- Sly_Auxin_TPM[order(match(Sly_Auxin_TPM[,1], Sly_Auxin_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_Auxin_FC <- Sly_Auxin %>% select(Protein, 67:71)
Sly_Auxin_FC <- Sly_Auxin_FC[!duplicated(Sly_Auxin_FC$Protein),]
Sly_Auxin_FC <- Sly_Auxin_FC[order(match(Sly_Auxin_FC[,1], Sly_Auxin_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_Auxin_DEG <- Sly_Auxin %>% select(Protein, 72:76)
Sly_Auxin_DEG <- Sly_Auxin_DEG[!duplicated(Sly_Auxin_DEG$Protein),]
Sly_Auxin_DEG <- Sly_Auxin_DEG[order(match(Sly_Auxin_DEG[,1], Sly_Auxin_Ann[,2])),]

Sly_Auxin_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Sly_Auxin_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Sly_Auxin_DEG$Ripening_Orange_DEG <- ifelse(is.na(Sly_Auxin_DEG$Ripening_Orange_DEG) == TRUE, " ", "*")
Sly_Auxin_DEG$Ripening_Red_DEG <- ifelse(is.na(Sly_Auxin_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
Sly_Auxin_DEG$ORHis_MG_DEG <- ifelse(is.na(Sly_Auxin_DEG$ORHis_MG_DEG) == TRUE, " ", "*")
Sly_Auxin_DEG$ORHis_Breaker_DEG <- ifelse(is.na(Sly_Auxin_DEG$ORHis_Breaker_DEG) == TRUE, " ", "*")
row.names(Sly_Auxin_DEG) <- Sly_Auxin_DEG$Protein
Sly_Auxin_DEG[1] <- NULL

rownames(Sly_Auxin_FC) <- Sly_Auxin_FC$Protein
Sly_Auxin_FC[1] <- NULL

min(Sly_Auxin_FC, na.rm = TRUE)
max(Sly_Auxin_FC, na.rm = TRUE)
breaks_Sly_Auxin=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Sly_Auxin_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Sly_Auxin,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_Auxin_Annotation, annotation_colors=color_Auxin, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Auxin",
         annotation_legend = TRUE,
         display_numbers = Sly_Auxin_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(39, 48),
         gaps_col = c(3)
)


rownames(Sly_Auxin_TPM) <- Sly_Auxin_TPM$Protein
Sly_Auxin_TPM[1] <- NULL

min(Sly_Auxin_TPM, na.rm = TRUE)
max(Sly_Auxin_TPM, na.rm = TRUE)
breaks_Sly_Auxin_TPM=c(seq(0,100, length.out = 200))

Sly_Auxin_TPM_norm <- normalize_df(Sly_Auxin_TPM)

pheatmap(Sly_Auxin_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Sly_Auxin_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_Auxin_Annotation, annotation_colors=color_Auxin, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Auxin",
         annotation_legend = TRUE,
         #display_numbers = Sly_Auxin_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(21,42)
)


#####Representing Heatmap for Cann

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Cann_Auxin_Ann <- read.table("Funcional Groups/FigTree/Cann_Auxin.txt",
                           sep = "\t", header = TRUE)

colnames(Cann_Auxin_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Cann_Auxin <- merge(Cann_Auxin_Ann, Summary_Pepper_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Cann_Auxin_TPM <- Cann_Auxin %>% select(Protein, 4:17)
Cann_Auxin_TPM <- Cann_Auxin_TPM[!duplicated(Cann_Auxin_TPM$Protein),]
Cann_Auxin_TPM <- Cann_Auxin_TPM[order(match(Cann_Auxin_TPM[,1], Cann_Auxin_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_Auxin_FC <- Cann_Auxin %>% select(Protein, 18:19)
Cann_Auxin_FC <- Cann_Auxin_FC[!duplicated(Cann_Auxin_FC$Protein),]
Cann_Auxin_FC <- Cann_Auxin_FC[order(match(Cann_Auxin_FC[,1], Cann_Auxin_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_Auxin_DEG <- Cann_Auxin %>% select(Protein, 20:21)
Cann_Auxin_DEG <- Cann_Auxin_DEG[!duplicated(Cann_Auxin_DEG$Protein),]
Cann_Auxin_DEG <- Cann_Auxin_DEG[order(match(Cann_Auxin_DEG[,1], Cann_Auxin_Ann[,2])),]

Cann_Auxin_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Cann_Auxin_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Cann_Auxin_DEG$Ripening_Red_DEG <- ifelse(is.na(Cann_Auxin_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
row.names(Cann_Auxin_DEG) <- Cann_Auxin_DEG$Protein
Cann_Auxin_DEG[1] <- NULL

rownames(Cann_Auxin_FC) <- Cann_Auxin_FC$Protein
Cann_Auxin_FC[1] <- NULL

min(Cann_Auxin_FC, na.rm = TRUE)
max(Cann_Auxin_FC, na.rm = TRUE)
breaks_Cann_Auxin=c(seq(-4.5,0, length.out = 100), seq(0.01,6,length.out = 100))

pheatmap(Cann_Auxin_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Cann_Auxin,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_Auxin_Annotation, annotation_colors=color_Auxin, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Red"),
         cex = 1,
         angle_col=45,
         main = "Auxin",
         annotation_legend = TRUE,
         display_numbers = Cann_Auxin_DEG,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)


rownames(Cann_Auxin_TPM) <- Cann_Auxin_TPM$Protein
Cann_Auxin_TPM[1] <- NULL

min(Cann_Auxin_TPM, na.rm = TRUE)
max(Cann_Auxin_TPM, na.rm = TRUE)
breaks_Cann_Auxin_TPM=c(seq(0,100, length.out = 200))

Cann_Auxin_TPM_norm <- normalize_df(Cann_Auxin_TPM)

pheatmap(Cann_Auxin_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Cann_Auxin_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_Auxin_Annotation, annotation_colors=color_Auxin, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Auxin",
         annotation_legend = TRUE,
         #display_numbers = Cann_Auxin_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(4,9)
)


#########################################################################################################################################
####################################  Brass  #############################################################################################
#########################################################################################################################################

Brass_Ath <- read.table("Funcional Groups/Ath_Brass.txt", header = TRUE)
Brass_Ath <- merge(Brass_Ath, Ath_Gene_Models, by.x="ID", by.y=1)
Brass_Ath_fasta <- Ath11_fasta[c(which(names(Ath11_fasta) %in% Brass_Ath$V2))]
Brass_Ath_2 <- Brass_Ath %>% select(V2, Name)
write.fasta(Brass_Ath_fasta, as.string=TRUE, names= names(Brass_Ath_fasta),
            file.out = "Funcional Groups/Ath_Brass.fasta")
rename.fasta(infile = "Funcional Groups/Ath_Brass.fasta", Brass_Ath_2, outfile = "Funcional Groups/Ath_Brass_2.fasta")

Brass_Niben_Candidates <- read.table("Funcional Groups/Niben_Candidates_Brass.txt")
Brass_Niben_Candidates <- Brass_Niben_Candidates[Brass_Niben_Candidates$V3>=35,] %>% select(V2)
Brass_Niben_Candidates <- unique(Brass_Niben_Candidates)
Brass_Niben_Candidates_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Brass_Niben_Candidates$V2))]
write.fasta(Brass_Niben_Candidates_fasta, as.string=TRUE, names= names(Brass_Niben_Candidates_fasta),
            file.out = "Funcional Groups/Brass_Niben_Candidates_35.fasta")

Brass_Sly_Candidates <- read.table("Funcional Groups/Sly_Candidates_Brass.txt")
Brass_Sly_Candidates <- Brass_Sly_Candidates[Brass_Sly_Candidates$V3>=35,] %>% select(V2)
Brass_Sly_Candidates <- unique(Brass_Sly_Candidates)
Brass_Sly_Candidates_fasta <- Tomato_4.0_fasta[c(which(names(Tomato_4.0_fasta) %in% Brass_Sly_Candidates$V2))]
write.fasta(Brass_Sly_Candidates_fasta, as.string=TRUE, names= names(Brass_Sly_Candidates_fasta),
            file.out = "Funcional Groups/Brass_Sly_Candidates_35.fasta")

Niben_Brass_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_Brass_Ann$Name))]
write.fasta(Niben_Brass_fasta, as.string=TRUE, names= names(Niben_Brass_fasta),
            file.out = "Funcional Groups/Niben_Brass.fasta")
Niben_Brass_Ann_2 <- Niben_Brass_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_Brass.fasta", Niben_Brass_Ann, outfile = "Funcional Groups/Niben_Brass_2.fasta")

save_heatmaps("Brass", height = 6)

## Representing Heatmap

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Niben_Brass_Ann <- read.table("Funcional Groups/FigTree/Niben_Brass.txt",
                              sep = "\t", header = TRUE)

colnames(Niben_Brass_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_Brass <- merge(Niben_Brass_Ann, TPM_All_FC_Annotated, by="Name")

## We create the annotation for Modules

Niben_Brass_Modules <- which_Module_pwd30(Niben_Brass_Ann)

## We select only the FC data from the different comparisons
Niben_Brass_TPM <- Niben_Brass %>% select(Protein, 4:51, 61:66)
Niben_Brass_TPM <- Niben_Brass_TPM[!duplicated(Niben_Brass_TPM$Protein),]
Niben_Brass_TPM <- Niben_Brass_TPM[order(match(Niben_Brass_TPM[,1], Niben_Brass_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_Brass_FC <- Niben_Brass %>% select(Protein, 71, 73:87)
Niben_Brass_FC <- Niben_Brass_FC[!duplicated(Niben_Brass_FC$Protein),]
Niben_Brass_FC <- Niben_Brass_FC[order(match(Niben_Brass_FC[,1], Niben_Brass_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_Brass_DEG <- Niben_Brass %>% select(Protein, 89, 91:105)
Niben_Brass_DEG <- Niben_Brass_DEG[!duplicated(Niben_Brass_DEG$Protein),]
Niben_Brass_DEG <- Niben_Brass_DEG[order(match(Niben_Brass_DEG[,1], Niben_Brass_Ann[,2])),]

Niben_Brass_DEG$pcrtB22h_vs_GFP22h_filtered <- ifelse(is.na(Niben_Brass_DEG$pcrtB22h_vs_GFP22h_filtered) == TRUE, " ", "*")
Niben_Brass_DEG$pcrtB25h_vs_GFP25h_filtered <- ifelse(is.na(Niben_Brass_DEG$pcrtB25h_vs_GFP25h_filtered) == TRUE, " ", "*")
Niben_Brass_DEG$pcrtBAA_vs_GFPAA_filtered <- ifelse(is.na(Niben_Brass_DEG$pcrtBAA_vs_GFPAA_filtered) == TRUE, " ", "*")
Niben_Brass_DEG$pcrtBBB_vs_GFPBB_filtered <- ifelse(is.na(Niben_Brass_DEG$pcrtBBB_vs_GFPBB_filtered) == TRUE, " ", "*")
Niben_Brass_DEG$pcrtBYY_vs_GFPYY_filtered <- ifelse(is.na(Niben_Brass_DEG$pcrtBYY_vs_GFPYY_filtered) == TRUE, " ", "*")
Niben_Brass_DEG$pcrtB28h_vs_GFP28h_filtered <- ifelse(is.na(Niben_Brass_DEG$pcrtB28h_vs_GFP28h_filtered) == TRUE, " ", "*")
Niben_Brass_DEG$pcrtB34h_vs_GFP34h_filtered <- ifelse(is.na(Niben_Brass_DEG$pcrtB34h_vs_GFP34h_filtered) == TRUE, " ", "*")
Niben_Brass_DEG$pcrtB37h_vs_GFP37h_filtered <- ifelse(is.na(Niben_Brass_DEG$pcrtB37h_vs_GFP37h_filtered) == TRUE, " ", "*")
Niben_Brass_DEG$pcrtBZZ_vs_GFPZZ_filtered <- ifelse(is.na(Niben_Brass_DEG$pcrtBZZ_vs_GFPZZ_filtered) == TRUE, " ", "*")
Niben_Brass_DEG$pcrtBCC_vs_GFPCC_filtered <- ifelse(is.na(Niben_Brass_DEG$pcrtBCC_vs_GFPCC_filtered) == TRUE, " ", "*")
Niben_Brass_DEG$pcrtB40h_vs_GFP40h_filtered <- ifelse(is.na(Niben_Brass_DEG$pcrtB40h_vs_GFP40h_filtered) == TRUE, " ", "*")
Niben_Brass_DEG$pcrtBDD_vs_GFPDD_filtered <- ifelse(is.na(Niben_Brass_DEG$pcrtBDD_vs_GFPDD_filtered) == TRUE, " ", "*")
Niben_Brass_DEG$pcrtB46h_vs_GFP46h_filtered <- ifelse(is.na(Niben_Brass_DEG$pcrtB46h_vs_GFP46h_filtered) == TRUE, " ", "*")
Niben_Brass_DEG$pcrtB56h_vs_GFP56h_filtered <- ifelse(is.na(Niben_Brass_DEG$pcrtB56h_vs_GFP56h_filtered) == TRUE, " ", "*")
Niben_Brass_DEG$pcrtBEE_vs_GFPEE_filtered <- ifelse(is.na(Niben_Brass_DEG$pcrtBEE_vs_GFPEE_filtered) == TRUE, " ", "*")
Niben_Brass_DEG$pcrtB96h_vs_GFP96h_filtered <- ifelse(is.na(Niben_Brass_DEG$pcrtB96h_vs_GFP96h_filtered) == TRUE, " ", "*")
row.names(Niben_Brass_DEG) <- Niben_Brass_DEG$Protein
Niben_Brass_DEG[1] <- NULL

Niben_Brass_DEG <- Niben_Brass_DEG %>%
  select(pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

row.names(Niben_Brass_FC) <- Niben_Brass_FC$Protein
Niben_Brass_FC[1] <- NULL

Niben_Brass_FC <- Niben_Brass_FC %>%
  select(pcrtB22h_vs_GFP22h_All, pcrtBAA_vs_GFPAA_All, pcrtB25h_vs_GFP25h_All, 
         pcrtBYY_vs_GFPYY_All, pcrtBBB_vs_GFPBB_All, pcrtB28h_vs_GFP28h_All, pcrtB34h_vs_GFP34h_All, pcrtBZZ_vs_GFPZZ_All, 
         pcrtB37h_vs_GFP37h_All, pcrtBCC_vs_GFPCC_All, pcrtB40h_vs_GFP40h_All, pcrtBDD_vs_GFPDD_All, pcrtB46h_vs_GFP46h_All, 
         pcrtBEE_vs_GFPEE_All, pcrtB56h_vs_GFP56h_All, pcrtB96h_vs_GFP96h_All)


min(Niben_Brass_FC, na.rm = TRUE)
max(Niben_Brass_FC, na.rm = TRUE)
breaks_Brass_FC=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Niben_Brass_FC %>% select(1:15), 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Brass_FC,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_Brass_Annotation, annotation_colors=color_Brass, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h"), #, "96h"),
         cex = 1,
         angle_col=45,
         main = "Brass",
         annotation_legend = TRUE,
         display_numbers = Niben_Brass_DEG %>% select(1:15),
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(84, 104),
         annotation_row = Niben_Brass_Modules,
         annotation_colors = module_colors
         #gaps_col = c(15)
)


rownames(Niben_Brass_TPM) <- Niben_Brass_TPM$Protein
Niben_Brass_TPM[1] <- NULL

min(Niben_Brass_TPM, na.rm = TRUE)
max(Niben_Brass_TPM, na.rm = TRUE)
breaks_Niben_Brass_TPM=c(seq(0,100, length.out = 100))

Niben_Brass_TPM_norm <- normalize_df(Niben_Brass_TPM)
Niben_Brass_TPM_norm_Cornell <- Niben_Brass_TPM %>%
  select(1:48) 
Niben_Brass_TPM_norm_Cornell <- normalize_df(Niben_Brass_TPM_norm_Cornell)


pheatmap(Niben_Brass_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Niben_Brass_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_Brass_Annotation, annotation_colors=color_Brass, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Brass",
         annotation_legend = TRUE,
         #display_numbers = Niben_Brass_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         annotation_row = Niben_Brass_Modules,
         annotation_colors = module_colors,
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(24, 48, 51)
)



#####Representing Heatmap for Sly

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Sly_Brass_Ann <- read.table("Funcional Groups/FigTree/Sly_Brass.txt",
                            sep = "\t", header = TRUE)

colnames(Sly_Brass_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Sly_Brass <- merge(Sly_Brass_Ann, Summary_Yazdani_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Sly_Brass_TPM <- Sly_Brass %>% select(Protein, 4:66)
Sly_Brass_TPM <- Sly_Brass_TPM[!duplicated(Sly_Brass_TPM$Protein),]
Sly_Brass_TPM <- Sly_Brass_TPM[order(match(Sly_Brass_TPM[,1], Sly_Brass_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_Brass_FC <- Sly_Brass %>% select(Protein, 67:71)
Sly_Brass_FC <- Sly_Brass_FC[!duplicated(Sly_Brass_FC$Protein),]
Sly_Brass_FC <- Sly_Brass_FC[order(match(Sly_Brass_FC[,1], Sly_Brass_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_Brass_DEG <- Sly_Brass %>% select(Protein, 72:76)
Sly_Brass_DEG <- Sly_Brass_DEG[!duplicated(Sly_Brass_DEG$Protein),]
Sly_Brass_DEG <- Sly_Brass_DEG[order(match(Sly_Brass_DEG[,1], Sly_Brass_Ann[,2])),]

Sly_Brass_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Sly_Brass_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Sly_Brass_DEG$Ripening_Orange_DEG <- ifelse(is.na(Sly_Brass_DEG$Ripening_Orange_DEG) == TRUE, " ", "*")
Sly_Brass_DEG$Ripening_Red_DEG <- ifelse(is.na(Sly_Brass_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
Sly_Brass_DEG$ORHis_MG_DEG <- ifelse(is.na(Sly_Brass_DEG$ORHis_MG_DEG) == TRUE, " ", "*")
Sly_Brass_DEG$ORHis_Breaker_DEG <- ifelse(is.na(Sly_Brass_DEG$ORHis_Breaker_DEG) == TRUE, " ", "*")
row.names(Sly_Brass_DEG) <- Sly_Brass_DEG$Protein
Sly_Brass_DEG[1] <- NULL

rownames(Sly_Brass_FC) <- Sly_Brass_FC$Protein
Sly_Brass_FC[1] <- NULL

min(Sly_Brass_FC, na.rm = TRUE)
max(Sly_Brass_FC, na.rm = TRUE)
breaks_Sly_Brass=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Sly_Brass_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Sly_Brass,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_Brass_Annotation, annotation_colors=color_Brass, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Brass",
         annotation_legend = TRUE,
         display_numbers = Sly_Brass_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(39, 48),
         gaps_col = c(3)
)


rownames(Sly_Brass_TPM) <- Sly_Brass_TPM$Protein
Sly_Brass_TPM[1] <- NULL

min(Sly_Brass_TPM, na.rm = TRUE)
max(Sly_Brass_TPM, na.rm = TRUE)
breaks_Sly_Brass_TPM=c(seq(0,100, length.out = 200))

Sly_Brass_TPM_norm <- normalize_df(Sly_Brass_TPM)

pheatmap(Sly_Brass_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Sly_Brass_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_Brass_Annotation, annotation_colors=color_Brass, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Brass",
         annotation_legend = TRUE,
         #display_numbers = Sly_Brass_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(21,42)
)


#####Representing Heatmap for Cann

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Cann_Brass_Ann <- read.table("Funcional Groups/FigTree/Cann_Brass.txt",
                             sep = "\t", header = TRUE)

colnames(Cann_Brass_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Cann_Brass <- merge(Cann_Brass_Ann, Summary_Pepper_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Cann_Brass_TPM <- Cann_Brass %>% select(Protein, 4:17)
Cann_Brass_TPM <- Cann_Brass_TPM[!duplicated(Cann_Brass_TPM$Protein),]
Cann_Brass_TPM <- Cann_Brass_TPM[order(match(Cann_Brass_TPM[,1], Cann_Brass_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_Brass_FC <- Cann_Brass %>% select(Protein, 18:19)
Cann_Brass_FC <- Cann_Brass_FC[!duplicated(Cann_Brass_FC$Protein),]
Cann_Brass_FC <- Cann_Brass_FC[order(match(Cann_Brass_FC[,1], Cann_Brass_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_Brass_DEG <- Cann_Brass %>% select(Protein, 20:21)
Cann_Brass_DEG <- Cann_Brass_DEG[!duplicated(Cann_Brass_DEG$Protein),]
Cann_Brass_DEG <- Cann_Brass_DEG[order(match(Cann_Brass_DEG[,1], Cann_Brass_Ann[,2])),]

Cann_Brass_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Cann_Brass_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Cann_Brass_DEG$Ripening_Red_DEG <- ifelse(is.na(Cann_Brass_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
row.names(Cann_Brass_DEG) <- Cann_Brass_DEG$Protein
Cann_Brass_DEG[1] <- NULL

rownames(Cann_Brass_FC) <- Cann_Brass_FC$Protein
Cann_Brass_FC[1] <- NULL

min(Cann_Brass_FC, na.rm = TRUE)
max(Cann_Brass_FC, na.rm = TRUE)
breaks_Cann_Brass=c(seq(-4.5,0, length.out = 100), seq(0.01,6,length.out = 100))

pheatmap(Cann_Brass_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Cann_Brass,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_Brass_Annotation, annotation_colors=color_Brass, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Red"),
         cex = 1,
         angle_col=45,
         main = "Brass",
         annotation_legend = TRUE,
         display_numbers = Cann_Brass_DEG,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)


rownames(Cann_Brass_TPM) <- Cann_Brass_TPM$Protein
Cann_Brass_TPM[1] <- NULL

min(Cann_Brass_TPM, na.rm = TRUE)
max(Cann_Brass_TPM, na.rm = TRUE)
breaks_Cann_Brass_TPM=c(seq(0,100, length.out = 200))

Cann_Brass_TPM_norm <- normalize_df(Cann_Brass_TPM)

pheatmap(Cann_Brass_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Cann_Brass_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_Brass_Annotation, annotation_colors=color_Brass, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Brass",
         annotation_legend = TRUE,
         #display_numbers = Cann_Brass_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(4,9)
)


#########################################################################################################################################
####################################  Cytok  #############################################################################################
#########################################################################################################################################

Cytok_Ath <- read.table("Funcional Groups/Ath_Cytok.txt", header = TRUE)
Cytok_Ath <- merge(Cytok_Ath, Ath_Gene_Models, by.x="ID", by.y=1)
Cytok_Ath_fasta <- Ath11_fasta[c(which(names(Ath11_fasta) %in% Cytok_Ath$V2))]
Cytok_Ath_2 <- Cytok_Ath %>% select(V2, Name)
write.fasta(Cytok_Ath_fasta, as.string=TRUE, names= names(Cytok_Ath_fasta),
            file.out = "Funcional Groups/Ath_Cytok.fasta")
rename.fasta(infile = "Funcional Groups/Ath_Cytok.fasta", Cytok_Ath_2, outfile = "Funcional Groups/Ath_Cytok_2.fasta")

Cytok_Niben_Candidates <- read.table("Funcional Groups/Niben_Candidates_Cytok.txt")
Cytok_Niben_Candidates <- Cytok_Niben_Candidates[Cytok_Niben_Candidates$V3>=35,] %>% select(V2)
Cytok_Niben_Candidates <- unique(Cytok_Niben_Candidates)
Cytok_Niben_Candidates_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Cytok_Niben_Candidates$V2))]
write.fasta(Cytok_Niben_Candidates_fasta, as.string=TRUE, names= names(Cytok_Niben_Candidates_fasta),
            file.out = "Funcional Groups/Cytok_Niben_Candidates_35.fasta")

Cytok_Sly_Candidates <- read.table("Funcional Groups/Sly_Candidates_Cytok.txt")
Cytok_Sly_Candidates <- Cytok_Sly_Candidates[Cytok_Sly_Candidates$V3>=35,] %>% select(V2)
Cytok_Sly_Candidates <- unique(Cytok_Sly_Candidates)
Cytok_Sly_Candidates_fasta <- Tomato_4.0_fasta[c(which(names(Tomato_4.0_fasta) %in% Cytok_Sly_Candidates$V2))]
write.fasta(Cytok_Sly_Candidates_fasta, as.string=TRUE, names= names(Cytok_Sly_Candidates_fasta),
            file.out = "Funcional Groups/Cytok_Sly_Candidates_35.fasta")

Niben_Cytok_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_Cytok_Ann$Name))]
write.fasta(Niben_Cytok_fasta, as.string=TRUE, names= names(Niben_Cytok_fasta),
            file.out = "Funcional Groups/Niben_Cytok.fasta")
Niben_Cytok_Ann_2 <- Niben_Cytok_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_Cytok.fasta", Niben_Cytok_Ann, outfile = "Funcional Groups/Niben_Cytok_2.fasta")

save_heatmaps("Cytok", height = 10)

## Representing Heatmap

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Niben_Cytok_Ann <- read.table("Funcional Groups/FigTree/Niben_Cytok.txt",
                              sep = "\t", header = TRUE)

colnames(Niben_Cytok_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_Cytok <- merge(Niben_Cytok_Ann, TPM_All_FC_Annotated, by="Name")

## We create the annotation for Modules

Niben_Cytok_Modules <- which_Module_pwd30(Niben_Cytok_Ann)

## We select only the FC data from the different comparisons
Niben_Cytok_TPM <- Niben_Cytok %>% select(Protein, 4:51, 61:66)
Niben_Cytok_TPM <- Niben_Cytok_TPM[!duplicated(Niben_Cytok_TPM$Protein),]
Niben_Cytok_TPM <- Niben_Cytok_TPM[order(match(Niben_Cytok_TPM[,1], Niben_Cytok_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_Cytok_FC <- Niben_Cytok %>% select(Protein, 71, 73:87)
Niben_Cytok_FC <- Niben_Cytok_FC[!duplicated(Niben_Cytok_FC$Protein),]
Niben_Cytok_FC <- Niben_Cytok_FC[order(match(Niben_Cytok_FC[,1], Niben_Cytok_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_Cytok_DEG <- Niben_Cytok %>% select(Protein, 89, 91:105)
Niben_Cytok_DEG <- Niben_Cytok_DEG[!duplicated(Niben_Cytok_DEG$Protein),]
Niben_Cytok_DEG <- Niben_Cytok_DEG[order(match(Niben_Cytok_DEG[,1], Niben_Cytok_Ann[,2])),]

Niben_Cytok_DEG$pcrtB22h_vs_GFP22h_filtered <- ifelse(is.na(Niben_Cytok_DEG$pcrtB22h_vs_GFP22h_filtered) == TRUE, " ", "*")
Niben_Cytok_DEG$pcrtB25h_vs_GFP25h_filtered <- ifelse(is.na(Niben_Cytok_DEG$pcrtB25h_vs_GFP25h_filtered) == TRUE, " ", "*")
Niben_Cytok_DEG$pcrtBAA_vs_GFPAA_filtered <- ifelse(is.na(Niben_Cytok_DEG$pcrtBAA_vs_GFPAA_filtered) == TRUE, " ", "*")
Niben_Cytok_DEG$pcrtBBB_vs_GFPBB_filtered <- ifelse(is.na(Niben_Cytok_DEG$pcrtBBB_vs_GFPBB_filtered) == TRUE, " ", "*")
Niben_Cytok_DEG$pcrtBYY_vs_GFPYY_filtered <- ifelse(is.na(Niben_Cytok_DEG$pcrtBYY_vs_GFPYY_filtered) == TRUE, " ", "*")
Niben_Cytok_DEG$pcrtB28h_vs_GFP28h_filtered <- ifelse(is.na(Niben_Cytok_DEG$pcrtB28h_vs_GFP28h_filtered) == TRUE, " ", "*")
Niben_Cytok_DEG$pcrtB34h_vs_GFP34h_filtered <- ifelse(is.na(Niben_Cytok_DEG$pcrtB34h_vs_GFP34h_filtered) == TRUE, " ", "*")
Niben_Cytok_DEG$pcrtB37h_vs_GFP37h_filtered <- ifelse(is.na(Niben_Cytok_DEG$pcrtB37h_vs_GFP37h_filtered) == TRUE, " ", "*")
Niben_Cytok_DEG$pcrtBZZ_vs_GFPZZ_filtered <- ifelse(is.na(Niben_Cytok_DEG$pcrtBZZ_vs_GFPZZ_filtered) == TRUE, " ", "*")
Niben_Cytok_DEG$pcrtBCC_vs_GFPCC_filtered <- ifelse(is.na(Niben_Cytok_DEG$pcrtBCC_vs_GFPCC_filtered) == TRUE, " ", "*")
Niben_Cytok_DEG$pcrtB40h_vs_GFP40h_filtered <- ifelse(is.na(Niben_Cytok_DEG$pcrtB40h_vs_GFP40h_filtered) == TRUE, " ", "*")
Niben_Cytok_DEG$pcrtBDD_vs_GFPDD_filtered <- ifelse(is.na(Niben_Cytok_DEG$pcrtBDD_vs_GFPDD_filtered) == TRUE, " ", "*")
Niben_Cytok_DEG$pcrtB46h_vs_GFP46h_filtered <- ifelse(is.na(Niben_Cytok_DEG$pcrtB46h_vs_GFP46h_filtered) == TRUE, " ", "*")
Niben_Cytok_DEG$pcrtB56h_vs_GFP56h_filtered <- ifelse(is.na(Niben_Cytok_DEG$pcrtB56h_vs_GFP56h_filtered) == TRUE, " ", "*")
Niben_Cytok_DEG$pcrtBEE_vs_GFPEE_filtered <- ifelse(is.na(Niben_Cytok_DEG$pcrtBEE_vs_GFPEE_filtered) == TRUE, " ", "*")
Niben_Cytok_DEG$pcrtB96h_vs_GFP96h_filtered <- ifelse(is.na(Niben_Cytok_DEG$pcrtB96h_vs_GFP96h_filtered) == TRUE, " ", "*")
row.names(Niben_Cytok_DEG) <- Niben_Cytok_DEG$Protein
Niben_Cytok_DEG[1] <- NULL

Niben_Cytok_DEG <- Niben_Cytok_DEG %>%
  select(pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

row.names(Niben_Cytok_FC) <- Niben_Cytok_FC$Protein
Niben_Cytok_FC[1] <- NULL

Niben_Cytok_FC <- Niben_Cytok_FC %>%
  select(pcrtB22h_vs_GFP22h_All, pcrtBAA_vs_GFPAA_All, pcrtB25h_vs_GFP25h_All, 
         pcrtBYY_vs_GFPYY_All, pcrtBBB_vs_GFPBB_All, pcrtB28h_vs_GFP28h_All, pcrtB34h_vs_GFP34h_All, pcrtBZZ_vs_GFPZZ_All, 
         pcrtB37h_vs_GFP37h_All, pcrtBCC_vs_GFPCC_All, pcrtB40h_vs_GFP40h_All, pcrtBDD_vs_GFPDD_All, pcrtB46h_vs_GFP46h_All, 
         pcrtBEE_vs_GFPEE_All, pcrtB56h_vs_GFP56h_All, pcrtB96h_vs_GFP96h_All)


min(Niben_Cytok_FC, na.rm = TRUE)
max(Niben_Cytok_FC, na.rm = TRUE)
breaks_Cytok_FC=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Niben_Cytok_FC, # %>% select(1:15), 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Cytok_FC,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_Cytok_Annotation, annotation_colors=color_Cytok, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h", "96h"),
         cex = 1,
         angle_col=45,
         main = "Cytok",
         annotation_legend = TRUE,
         display_numbers = Niben_Cytok_DEG, # %>% select(1:15),
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(84, 104),
         annotation_row = Niben_Cytok_Modules,
         annotation_colors = module_colors
         #gaps_col = c(15)
)


rownames(Niben_Cytok_TPM) <- Niben_Cytok_TPM$Protein
Niben_Cytok_TPM[1] <- NULL

min(Niben_Cytok_TPM, na.rm = TRUE)
max(Niben_Cytok_TPM, na.rm = TRUE)
breaks_Niben_Cytok_TPM=c(seq(0,100, length.out = 100))

Niben_Cytok_TPM_norm <- normalize_df(Niben_Cytok_TPM)
Niben_Cytok_TPM_norm_Cornell <- Niben_Cytok_TPM %>%
  select(1:48) 
Niben_Cytok_TPM_norm_Cornell <- normalize_df(Niben_Cytok_TPM_norm_Cornell)


pheatmap(Niben_Cytok_TPM_norm_Cornell, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Niben_Cytok_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_Cytok_Annotation, annotation_colors=color_Cytok, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Cytok",
         annotation_legend = TRUE,
         #display_numbers = Niben_Cytok_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         annotation_row = Niben_Cytok_Modules,
         annotation_colors = module_colors,
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(24)#, 48, 51)
)



#####Representing Heatmap for Sly

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Sly_Cytok_Ann <- read.table("Funcional Groups/FigTree/Sly_Cytok.txt",
                            sep = "\t", header = TRUE)

colnames(Sly_Cytok_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Sly_Cytok <- merge(Sly_Cytok_Ann, Summary_Yazdani_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Sly_Cytok_TPM <- Sly_Cytok %>% select(Protein, 4:66)
Sly_Cytok_TPM <- Sly_Cytok_TPM[!duplicated(Sly_Cytok_TPM$Protein),]
Sly_Cytok_TPM <- Sly_Cytok_TPM[order(match(Sly_Cytok_TPM[,1], Sly_Cytok_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_Cytok_FC <- Sly_Cytok %>% select(Protein, 67:71)
Sly_Cytok_FC <- Sly_Cytok_FC[!duplicated(Sly_Cytok_FC$Protein),]
Sly_Cytok_FC <- Sly_Cytok_FC[order(match(Sly_Cytok_FC[,1], Sly_Cytok_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_Cytok_DEG <- Sly_Cytok %>% select(Protein, 72:76)
Sly_Cytok_DEG <- Sly_Cytok_DEG[!duplicated(Sly_Cytok_DEG$Protein),]
Sly_Cytok_DEG <- Sly_Cytok_DEG[order(match(Sly_Cytok_DEG[,1], Sly_Cytok_Ann[,2])),]

Sly_Cytok_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Sly_Cytok_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Sly_Cytok_DEG$Ripening_Orange_DEG <- ifelse(is.na(Sly_Cytok_DEG$Ripening_Orange_DEG) == TRUE, " ", "*")
Sly_Cytok_DEG$Ripening_Red_DEG <- ifelse(is.na(Sly_Cytok_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
Sly_Cytok_DEG$ORHis_MG_DEG <- ifelse(is.na(Sly_Cytok_DEG$ORHis_MG_DEG) == TRUE, " ", "*")
Sly_Cytok_DEG$ORHis_Breaker_DEG <- ifelse(is.na(Sly_Cytok_DEG$ORHis_Breaker_DEG) == TRUE, " ", "*")
row.names(Sly_Cytok_DEG) <- Sly_Cytok_DEG$Protein
Sly_Cytok_DEG[1] <- NULL

rownames(Sly_Cytok_FC) <- Sly_Cytok_FC$Protein
Sly_Cytok_FC[1] <- NULL

min(Sly_Cytok_FC, na.rm = TRUE)
max(Sly_Cytok_FC, na.rm = TRUE)
breaks_Sly_Cytok=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Sly_Cytok_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Sly_Cytok,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_Cytok_Annotation, annotation_colors=color_Cytok, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Cytok",
         annotation_legend = TRUE,
         display_numbers = Sly_Cytok_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(39, 48),
         gaps_col = c(3)
)


rownames(Sly_Cytok_TPM) <- Sly_Cytok_TPM$Protein
Sly_Cytok_TPM[1] <- NULL

min(Sly_Cytok_TPM, na.rm = TRUE)
max(Sly_Cytok_TPM, na.rm = TRUE)
breaks_Sly_Cytok_TPM=c(seq(0,100, length.out = 200))

Sly_Cytok_TPM_norm <- normalize_df(Sly_Cytok_TPM)

pheatmap(Sly_Cytok_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Sly_Cytok_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_Cytok_Annotation, annotation_colors=color_Cytok, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Cytok",
         annotation_legend = TRUE,
         #display_numbers = Sly_Cytok_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(21,42)
)


#####Representing Heatmap for Cann

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Cann_Cytok_Ann <- read.table("Funcional Groups/FigTree/Cann_Cytok.txt",
                             sep = "\t", header = TRUE)

colnames(Cann_Cytok_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Cann_Cytok <- merge(Cann_Cytok_Ann, Summary_Pepper_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Cann_Cytok_TPM <- Cann_Cytok %>% select(Protein, 4:17)
Cann_Cytok_TPM <- Cann_Cytok_TPM[!duplicated(Cann_Cytok_TPM$Protein),]
Cann_Cytok_TPM <- Cann_Cytok_TPM[order(match(Cann_Cytok_TPM[,1], Cann_Cytok_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_Cytok_FC <- Cann_Cytok %>% select(Protein, 18:19)
Cann_Cytok_FC <- Cann_Cytok_FC[!duplicated(Cann_Cytok_FC$Protein),]
Cann_Cytok_FC <- Cann_Cytok_FC[order(match(Cann_Cytok_FC[,1], Cann_Cytok_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_Cytok_DEG <- Cann_Cytok %>% select(Protein, 20:21)
Cann_Cytok_DEG <- Cann_Cytok_DEG[!duplicated(Cann_Cytok_DEG$Protein),]
Cann_Cytok_DEG <- Cann_Cytok_DEG[order(match(Cann_Cytok_DEG[,1], Cann_Cytok_Ann[,2])),]

Cann_Cytok_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Cann_Cytok_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Cann_Cytok_DEG$Ripening_Red_DEG <- ifelse(is.na(Cann_Cytok_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
row.names(Cann_Cytok_DEG) <- Cann_Cytok_DEG$Protein
Cann_Cytok_DEG[1] <- NULL

rownames(Cann_Cytok_FC) <- Cann_Cytok_FC$Protein
Cann_Cytok_FC[1] <- NULL

min(Cann_Cytok_FC, na.rm = TRUE)
max(Cann_Cytok_FC, na.rm = TRUE)
breaks_Cann_Cytok=c(seq(-4.5,0, length.out = 100), seq(0.01,6,length.out = 100))

pheatmap(Cann_Cytok_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Cann_Cytok,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_Cytok_Annotation, annotation_colors=color_Cytok, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Red"),
         cex = 1,
         angle_col=45,
         main = "Cytok",
         annotation_legend = TRUE,
         display_numbers = Cann_Cytok_DEG,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)


rownames(Cann_Cytok_TPM) <- Cann_Cytok_TPM$Protein
Cann_Cytok_TPM[1] <- NULL

min(Cann_Cytok_TPM, na.rm = TRUE)
max(Cann_Cytok_TPM, na.rm = TRUE)
breaks_Cann_Cytok_TPM=c(seq(0,100, length.out = 200))

Cann_Cytok_TPM_norm <- normalize_df(Cann_Cytok_TPM)

pheatmap(Cann_Cytok_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Cann_Cytok_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_Cytok_Annotation, annotation_colors=color_Cytok, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Cytok",
         annotation_legend = TRUE,
         #display_numbers = Cann_Cytok_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(4,9)
)

#########################################################################################################################################
####################################  Ethylene  #############################################################################################
#########################################################################################################################################

Ethylene_Ath <- read.table("Funcional Groups/Ath_Ethylene.txt", header = TRUE)
Ethylene_Ath <- merge(Ethylene_Ath, Ath_Gene_Models, by.x="ID", by.y=1)
Ethylene_Ath_fasta <- Ath11_fasta[c(which(names(Ath11_fasta) %in% Ethylene_Ath$V2))]
Ethylene_Ath_2 <- Ethylene_Ath %>% select(V2, Name)
write.fasta(Ethylene_Ath_fasta, as.string=TRUE, names= names(Ethylene_Ath_fasta),
            file.out = "Funcional Groups/Ath_Ethylene.fasta")
rename.fasta(infile = "Funcional Groups/Ath_Ethylene.fasta", Ethylene_Ath_2, outfile = "Funcional Groups/Ath_Ethylene_2.fasta")

Ethylene_Niben_Candidates <- read.table("Funcional Groups/Niben_Candidates_Ethylene.txt")
Ethylene_Niben_Candidates <- Ethylene_Niben_Candidates[Ethylene_Niben_Candidates$V3>=35,] %>% select(V2)
Ethylene_Niben_Candidates <- unique(Ethylene_Niben_Candidates)
Ethylene_Niben_Candidates_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Ethylene_Niben_Candidates$V2))]
write.fasta(Ethylene_Niben_Candidates_fasta, as.string=TRUE, names= names(Ethylene_Niben_Candidates_fasta),
            file.out = "Funcional Groups/Ethylene_Niben_Candidates_35.fasta")

Ethylene_Sly_Candidates <- read.table("Funcional Groups/Sly_Candidates_Ethylene.txt")
Ethylene_Sly_Candidates <- Ethylene_Sly_Candidates[Ethylene_Sly_Candidates$V3>=35,] %>% select(V2)
Ethylene_Sly_Candidates <- unique(Ethylene_Sly_Candidates)
Ethylene_Sly_Candidates_fasta <- Tomato_4.0_fasta[c(which(names(Tomato_4.0_fasta) %in% Ethylene_Sly_Candidates$V2))]
write.fasta(Ethylene_Sly_Candidates_fasta, as.string=TRUE, names= names(Ethylene_Sly_Candidates_fasta),
            file.out = "Funcional Groups/Ethylene_Sly_Candidates_35.fasta")

Niben_Ethylene_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_Ethylene_Ann$Name))]
write.fasta(Niben_Ethylene_fasta, as.string=TRUE, names= names(Niben_Ethylene_fasta),
            file.out = "Funcional Groups/Niben_Ethylene.fasta")
Niben_Ethylene_Ann_2 <- Niben_Ethylene_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_Ethylene.fasta", Niben_Ethylene_Ann, outfile = "Funcional Groups/Niben_Ethylene_2.fasta")

save_heatmaps("Ethylene", height = 6)

## Representing Heatmap

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Niben_Ethylene_Ann <- read.table("Funcional Groups/FigTree/Niben_Ethylene.txt",
                              sep = "\t", header = TRUE)

colnames(Niben_Ethylene_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_Ethylene <- merge(Niben_Ethylene_Ann, TPM_All_FC_Annotated, by="Name")

## We create the annotation for Modules

Niben_Ethylene_Modules <- which_Module_pwd30(Niben_Ethylene_Ann)

## We select only the FC data from the different comparisons
Niben_Ethylene_TPM <- Niben_Ethylene %>% select(Protein, 4:51, 61:66)
Niben_Ethylene_TPM <- Niben_Ethylene_TPM[!duplicated(Niben_Ethylene_TPM$Protein),]
Niben_Ethylene_TPM <- Niben_Ethylene_TPM[order(match(Niben_Ethylene_TPM[,1], Niben_Ethylene_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_Ethylene_FC <- Niben_Ethylene %>% select(Protein, 71, 73:87)
Niben_Ethylene_FC <- Niben_Ethylene_FC[!duplicated(Niben_Ethylene_FC$Protein),]
Niben_Ethylene_FC <- Niben_Ethylene_FC[order(match(Niben_Ethylene_FC[,1], Niben_Ethylene_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_Ethylene_DEG <- Niben_Ethylene %>% select(Protein, 89, 91:105)
Niben_Ethylene_DEG <- Niben_Ethylene_DEG[!duplicated(Niben_Ethylene_DEG$Protein),]
Niben_Ethylene_DEG <- Niben_Ethylene_DEG[order(match(Niben_Ethylene_DEG[,1], Niben_Ethylene_Ann[,2])),]

Niben_Ethylene_DEG$pcrtB22h_vs_GFP22h_filtered <- ifelse(is.na(Niben_Ethylene_DEG$pcrtB22h_vs_GFP22h_filtered) == TRUE, " ", "*")
Niben_Ethylene_DEG$pcrtB25h_vs_GFP25h_filtered <- ifelse(is.na(Niben_Ethylene_DEG$pcrtB25h_vs_GFP25h_filtered) == TRUE, " ", "*")
Niben_Ethylene_DEG$pcrtBAA_vs_GFPAA_filtered <- ifelse(is.na(Niben_Ethylene_DEG$pcrtBAA_vs_GFPAA_filtered) == TRUE, " ", "*")
Niben_Ethylene_DEG$pcrtBBB_vs_GFPBB_filtered <- ifelse(is.na(Niben_Ethylene_DEG$pcrtBBB_vs_GFPBB_filtered) == TRUE, " ", "*")
Niben_Ethylene_DEG$pcrtBYY_vs_GFPYY_filtered <- ifelse(is.na(Niben_Ethylene_DEG$pcrtBYY_vs_GFPYY_filtered) == TRUE, " ", "*")
Niben_Ethylene_DEG$pcrtB28h_vs_GFP28h_filtered <- ifelse(is.na(Niben_Ethylene_DEG$pcrtB28h_vs_GFP28h_filtered) == TRUE, " ", "*")
Niben_Ethylene_DEG$pcrtB34h_vs_GFP34h_filtered <- ifelse(is.na(Niben_Ethylene_DEG$pcrtB34h_vs_GFP34h_filtered) == TRUE, " ", "*")
Niben_Ethylene_DEG$pcrtB37h_vs_GFP37h_filtered <- ifelse(is.na(Niben_Ethylene_DEG$pcrtB37h_vs_GFP37h_filtered) == TRUE, " ", "*")
Niben_Ethylene_DEG$pcrtBZZ_vs_GFPZZ_filtered <- ifelse(is.na(Niben_Ethylene_DEG$pcrtBZZ_vs_GFPZZ_filtered) == TRUE, " ", "*")
Niben_Ethylene_DEG$pcrtBCC_vs_GFPCC_filtered <- ifelse(is.na(Niben_Ethylene_DEG$pcrtBCC_vs_GFPCC_filtered) == TRUE, " ", "*")
Niben_Ethylene_DEG$pcrtB40h_vs_GFP40h_filtered <- ifelse(is.na(Niben_Ethylene_DEG$pcrtB40h_vs_GFP40h_filtered) == TRUE, " ", "*")
Niben_Ethylene_DEG$pcrtBDD_vs_GFPDD_filtered <- ifelse(is.na(Niben_Ethylene_DEG$pcrtBDD_vs_GFPDD_filtered) == TRUE, " ", "*")
Niben_Ethylene_DEG$pcrtB46h_vs_GFP46h_filtered <- ifelse(is.na(Niben_Ethylene_DEG$pcrtB46h_vs_GFP46h_filtered) == TRUE, " ", "*")
Niben_Ethylene_DEG$pcrtB56h_vs_GFP56h_filtered <- ifelse(is.na(Niben_Ethylene_DEG$pcrtB56h_vs_GFP56h_filtered) == TRUE, " ", "*")
Niben_Ethylene_DEG$pcrtBEE_vs_GFPEE_filtered <- ifelse(is.na(Niben_Ethylene_DEG$pcrtBEE_vs_GFPEE_filtered) == TRUE, " ", "*")
Niben_Ethylene_DEG$pcrtB96h_vs_GFP96h_filtered <- ifelse(is.na(Niben_Ethylene_DEG$pcrtB96h_vs_GFP96h_filtered) == TRUE, " ", "*")
row.names(Niben_Ethylene_DEG) <- Niben_Ethylene_DEG$Protein
Niben_Ethylene_DEG[1] <- NULL

Niben_Ethylene_DEG <- Niben_Ethylene_DEG %>%
  select(pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

row.names(Niben_Ethylene_FC) <- Niben_Ethylene_FC$Protein
Niben_Ethylene_FC[1] <- NULL

Niben_Ethylene_FC <- Niben_Ethylene_FC %>%
  select(pcrtB22h_vs_GFP22h_All, pcrtBAA_vs_GFPAA_All, pcrtB25h_vs_GFP25h_All, 
         pcrtBYY_vs_GFPYY_All, pcrtBBB_vs_GFPBB_All, pcrtB28h_vs_GFP28h_All, pcrtB34h_vs_GFP34h_All, pcrtBZZ_vs_GFPZZ_All, 
         pcrtB37h_vs_GFP37h_All, pcrtBCC_vs_GFPCC_All, pcrtB40h_vs_GFP40h_All, pcrtBDD_vs_GFPDD_All, pcrtB46h_vs_GFP46h_All, 
         pcrtBEE_vs_GFPEE_All, pcrtB56h_vs_GFP56h_All, pcrtB96h_vs_GFP96h_All)


min(Niben_Ethylene_FC, na.rm = TRUE)
max(Niben_Ethylene_FC, na.rm = TRUE)
breaks_Ethylene_FC=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Niben_Ethylene_FC %>% select(1:15), 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Ethylene_FC,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_Ethylene_Annotation, annotation_colors=color_Ethylene, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h"), #, "96h"),
         cex = 1,
         angle_col=45,
         main = "Ethylene",
         annotation_legend = TRUE,
         display_numbers = Niben_Ethylene_DEG %>% select(1:15),
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(84, 104),
         annotation_row = Niben_Ethylene_Modules,
         annotation_colors = module_colors
         #gaps_col = c(15)
)


rownames(Niben_Ethylene_TPM) <- Niben_Ethylene_TPM$Protein
Niben_Ethylene_TPM[1] <- NULL

min(Niben_Ethylene_TPM, na.rm = TRUE)
max(Niben_Ethylene_TPM, na.rm = TRUE)
breaks_Niben_Ethylene_TPM=c(seq(0,100, length.out = 100))

Niben_Ethylene_TPM_norm <- normalize_df(Niben_Ethylene_TPM)
Niben_Ethylene_TPM_norm_Cornell <- Niben_Ethylene_TPM %>%
  select(1:48) 
Niben_Ethylene_TPM_norm_Cornell <- normalize_df(Niben_Ethylene_TPM_norm_Cornell)


pheatmap(Niben_Ethylene_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Niben_Ethylene_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_Ethylene_Annotation, annotation_colors=color_Ethylene, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Ethylene",
         annotation_legend = TRUE,
         #display_numbers = Niben_Ethylene_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         annotation_row = Niben_Ethylene_Modules,
         annotation_colors = module_colors,
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(24, 48, 51)
)



#####Representing Heatmap for Sly

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Sly_Ethylene_Ann <- read.table("Funcional Groups/FigTree/Sly_Ethylene.txt",
                            sep = "\t", header = TRUE)

colnames(Sly_Ethylene_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Sly_Ethylene <- merge(Sly_Ethylene_Ann, Summary_Yazdani_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Sly_Ethylene_TPM <- Sly_Ethylene %>% select(Protein, 4:66)
Sly_Ethylene_TPM <- Sly_Ethylene_TPM[!duplicated(Sly_Ethylene_TPM$Protein),]
Sly_Ethylene_TPM <- Sly_Ethylene_TPM[order(match(Sly_Ethylene_TPM[,1], Sly_Ethylene_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_Ethylene_FC <- Sly_Ethylene %>% select(Protein, 67:71)
Sly_Ethylene_FC <- Sly_Ethylene_FC[!duplicated(Sly_Ethylene_FC$Protein),]
Sly_Ethylene_FC <- Sly_Ethylene_FC[order(match(Sly_Ethylene_FC[,1], Sly_Ethylene_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_Ethylene_DEG <- Sly_Ethylene %>% select(Protein, 72:76)
Sly_Ethylene_DEG <- Sly_Ethylene_DEG[!duplicated(Sly_Ethylene_DEG$Protein),]
Sly_Ethylene_DEG <- Sly_Ethylene_DEG[order(match(Sly_Ethylene_DEG[,1], Sly_Ethylene_Ann[,2])),]

Sly_Ethylene_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Sly_Ethylene_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Sly_Ethylene_DEG$Ripening_Orange_DEG <- ifelse(is.na(Sly_Ethylene_DEG$Ripening_Orange_DEG) == TRUE, " ", "*")
Sly_Ethylene_DEG$Ripening_Red_DEG <- ifelse(is.na(Sly_Ethylene_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
Sly_Ethylene_DEG$ORHis_MG_DEG <- ifelse(is.na(Sly_Ethylene_DEG$ORHis_MG_DEG) == TRUE, " ", "*")
Sly_Ethylene_DEG$ORHis_Breaker_DEG <- ifelse(is.na(Sly_Ethylene_DEG$ORHis_Breaker_DEG) == TRUE, " ", "*")
row.names(Sly_Ethylene_DEG) <- Sly_Ethylene_DEG$Protein
Sly_Ethylene_DEG[1] <- NULL

rownames(Sly_Ethylene_FC) <- Sly_Ethylene_FC$Protein
Sly_Ethylene_FC[1] <- NULL

min(Sly_Ethylene_FC, na.rm = TRUE)
max(Sly_Ethylene_FC, na.rm = TRUE)
breaks_Sly_Ethylene=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Sly_Ethylene_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Sly_Ethylene,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_Ethylene_Annotation, annotation_colors=color_Ethylene, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Ethylene",
         annotation_legend = TRUE,
         display_numbers = Sly_Ethylene_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(39, 48),
         gaps_col = c(3)
)


rownames(Sly_Ethylene_TPM) <- Sly_Ethylene_TPM$Protein
Sly_Ethylene_TPM[1] <- NULL

min(Sly_Ethylene_TPM, na.rm = TRUE)
max(Sly_Ethylene_TPM, na.rm = TRUE)
breaks_Sly_Ethylene_TPM=c(seq(0,100, length.out = 200))

Sly_Ethylene_TPM_norm <- normalize_df(Sly_Ethylene_TPM)

pheatmap(Sly_Ethylene_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Sly_Ethylene_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_Ethylene_Annotation, annotation_colors=color_Ethylene, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Ethylene",
         annotation_legend = TRUE,
         #display_numbers = Sly_Ethylene_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(21,42)
)


#####Representing Heatmap for Cann

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Cann_Ethylene_Ann <- read.table("Funcional Groups/FigTree/Cann_Ethylene.txt",
                             sep = "\t", header = TRUE)

colnames(Cann_Ethylene_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Cann_Ethylene <- merge(Cann_Ethylene_Ann, Summary_Pepper_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Cann_Ethylene_TPM <- Cann_Ethylene %>% select(Protein, 4:17)
Cann_Ethylene_TPM <- Cann_Ethylene_TPM[!duplicated(Cann_Ethylene_TPM$Protein),]
Cann_Ethylene_TPM <- Cann_Ethylene_TPM[order(match(Cann_Ethylene_TPM[,1], Cann_Ethylene_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_Ethylene_FC <- Cann_Ethylene %>% select(Protein, 18:19)
Cann_Ethylene_FC <- Cann_Ethylene_FC[!duplicated(Cann_Ethylene_FC$Protein),]
Cann_Ethylene_FC <- Cann_Ethylene_FC[order(match(Cann_Ethylene_FC[,1], Cann_Ethylene_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_Ethylene_DEG <- Cann_Ethylene %>% select(Protein, 20:21)
Cann_Ethylene_DEG <- Cann_Ethylene_DEG[!duplicated(Cann_Ethylene_DEG$Protein),]
Cann_Ethylene_DEG <- Cann_Ethylene_DEG[order(match(Cann_Ethylene_DEG[,1], Cann_Ethylene_Ann[,2])),]

Cann_Ethylene_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Cann_Ethylene_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Cann_Ethylene_DEG$Ripening_Red_DEG <- ifelse(is.na(Cann_Ethylene_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
row.names(Cann_Ethylene_DEG) <- Cann_Ethylene_DEG$Protein
Cann_Ethylene_DEG[1] <- NULL

rownames(Cann_Ethylene_FC) <- Cann_Ethylene_FC$Protein
Cann_Ethylene_FC[1] <- NULL

min(Cann_Ethylene_FC, na.rm = TRUE)
max(Cann_Ethylene_FC, na.rm = TRUE)
breaks_Cann_Ethylene=c(seq(-4.5,0, length.out = 100), seq(0.01,6,length.out = 100))

pheatmap(Cann_Ethylene_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Cann_Ethylene,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_Ethylene_Annotation, annotation_colors=color_Ethylene, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Red"),
         cex = 1,
         angle_col=45,
         main = "Ethylene",
         annotation_legend = TRUE,
         display_numbers = Cann_Ethylene_DEG,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)


rownames(Cann_Ethylene_TPM) <- Cann_Ethylene_TPM$Protein
Cann_Ethylene_TPM[1] <- NULL

min(Cann_Ethylene_TPM, na.rm = TRUE)
max(Cann_Ethylene_TPM, na.rm = TRUE)
breaks_Cann_Ethylene_TPM=c(seq(0,100, length.out = 200))

Cann_Ethylene_TPM_norm <- normalize_df(Cann_Ethylene_TPM)

pheatmap(Cann_Ethylene_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Cann_Ethylene_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_Ethylene_Annotation, annotation_colors=color_Ethylene, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Ethylene",
         annotation_legend = TRUE,
         #display_numbers = Cann_Ethylene_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(4,9)
)


#########################################################################################################################################
####################################  Salicylic  #############################################################################################
#########################################################################################################################################

Salicylic_Ath <- read.table("Funcional Groups/Ath_Salicylic.txt", header = TRUE)
Salicylic_Ath <- merge(Salicylic_Ath, Ath_Gene_Models, by.x="ID", by.y=1)
Salicylic_Ath_fasta <- Ath11_fasta[c(which(names(Ath11_fasta) %in% Salicylic_Ath$V2))]
Salicylic_Ath_2 <- Salicylic_Ath %>% select(V2, Name)
write.fasta(Salicylic_Ath_fasta, as.string=TRUE, names= names(Salicylic_Ath_fasta),
            file.out = "Funcional Groups/Ath_Salicylic.fasta")
rename.fasta(infile = "Funcional Groups/Ath_Salicylic.fasta", Salicylic_Ath_2, outfile = "Funcional Groups/Ath_Salicylic_2.fasta")

Salicylic_Niben_Candidates <- read.table("Funcional Groups/Niben_Candidates_Salicylic.txt")
Salicylic_Niben_Candidates <- Salicylic_Niben_Candidates[Salicylic_Niben_Candidates$V3>=35,] %>% select(V2)
Salicylic_Niben_Candidates <- unique(Salicylic_Niben_Candidates)
Salicylic_Niben_Candidates_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Salicylic_Niben_Candidates$V2))]
write.fasta(Salicylic_Niben_Candidates_fasta, as.string=TRUE, names= names(Salicylic_Niben_Candidates_fasta),
            file.out = "Funcional Groups/Salicylic_Niben_Candidates_35.fasta")

Salicylic_Sly_Candidates <- read.table("Funcional Groups/Sly_Candidates_Salicylic.txt")
Salicylic_Sly_Candidates <- Salicylic_Sly_Candidates[Salicylic_Sly_Candidates$V3>=35,] %>% select(V2)
Salicylic_Sly_Candidates <- unique(Salicylic_Sly_Candidates)
Salicylic_Sly_Candidates_fasta <- Tomato_4.0_fasta[c(which(names(Tomato_4.0_fasta) %in% Salicylic_Sly_Candidates$V2))]
write.fasta(Salicylic_Sly_Candidates_fasta, as.string=TRUE, names= names(Salicylic_Sly_Candidates_fasta),
            file.out = "Funcional Groups/Salicylic_Sly_Candidates_35.fasta")

Salicylic_Cann_Candidates <- read.table("Funcional Groups/Cann_Candidates_Salicylic.txt")
Salicylic_Cann_Candidates <- Salicylic_Cann_Candidates[Salicylic_Cann_Candidates$V3>=35,] %>% select(V2)
Salicylic_Cann_Candidates <- unique(Salicylic_Cann_Candidates)
Salicylic_Cann_Candidates_fasta <- Cannuum_ASM51225v2_fasta[c(which(names(Cannuum_ASM51225v2_fasta) %in% Salicylic_Cann_Candidates$V2))]
write.fasta(Salicylic_Cann_Candidates_fasta, as.string=TRUE, names= names(Salicylic_Cann_Candidates_fasta),
            file.out = "Funcional Groups/Salicylic_Cann_Candidates_35.fasta")

Niben_Salicylic_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_Salicylic_Ann$Name))]
write.fasta(Niben_Salicylic_fasta, as.string=TRUE, names= names(Niben_Salicylic_fasta),
            file.out = "Funcional Groups/Niben_Salicylic.fasta")
Niben_Salicylic_Ann_2 <- Niben_Salicylic_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_Salicylic.fasta", Niben_Salicylic_Ann, outfile = "Funcional Groups/Niben_Salicylic_2.fasta")

save_heatmaps("Salicylic", height = 7)

## Representing Heatmap

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Niben_Salicylic_Ann <- read.table("Funcional Groups/FigTree/Niben_Salicylic.txt",
                                 sep = "\t", header = TRUE)

colnames(Niben_Salicylic_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_Salicylic <- merge(Niben_Salicylic_Ann, TPM_All_FC_Annotated, by="Name")

## We create the annotation for Modules

Niben_Salicylic_Modules <- which_Module_pwd30(Niben_Salicylic_Ann)

## We select only the FC data from the different comparisons
Niben_Salicylic_TPM <- Niben_Salicylic %>% select(Protein, 4:51, 61:66)
Niben_Salicylic_TPM <- Niben_Salicylic_TPM[!duplicated(Niben_Salicylic_TPM$Protein),]
Niben_Salicylic_TPM <- Niben_Salicylic_TPM[order(match(Niben_Salicylic_TPM[,1], Niben_Salicylic_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_Salicylic_FC <- Niben_Salicylic %>% select(Protein, 71, 73:87)
Niben_Salicylic_FC <- Niben_Salicylic_FC[!duplicated(Niben_Salicylic_FC$Protein),]
Niben_Salicylic_FC <- Niben_Salicylic_FC[order(match(Niben_Salicylic_FC[,1], Niben_Salicylic_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_Salicylic_DEG <- Niben_Salicylic %>% select(Protein, 89, 91:105)
Niben_Salicylic_DEG <- Niben_Salicylic_DEG[!duplicated(Niben_Salicylic_DEG$Protein),]
Niben_Salicylic_DEG <- Niben_Salicylic_DEG[order(match(Niben_Salicylic_DEG[,1], Niben_Salicylic_Ann[,2])),]

Niben_Salicylic_DEG$pcrtB22h_vs_GFP22h_filtered <- ifelse(is.na(Niben_Salicylic_DEG$pcrtB22h_vs_GFP22h_filtered) == TRUE, " ", "*")
Niben_Salicylic_DEG$pcrtB25h_vs_GFP25h_filtered <- ifelse(is.na(Niben_Salicylic_DEG$pcrtB25h_vs_GFP25h_filtered) == TRUE, " ", "*")
Niben_Salicylic_DEG$pcrtBAA_vs_GFPAA_filtered <- ifelse(is.na(Niben_Salicylic_DEG$pcrtBAA_vs_GFPAA_filtered) == TRUE, " ", "*")
Niben_Salicylic_DEG$pcrtBBB_vs_GFPBB_filtered <- ifelse(is.na(Niben_Salicylic_DEG$pcrtBBB_vs_GFPBB_filtered) == TRUE, " ", "*")
Niben_Salicylic_DEG$pcrtBYY_vs_GFPYY_filtered <- ifelse(is.na(Niben_Salicylic_DEG$pcrtBYY_vs_GFPYY_filtered) == TRUE, " ", "*")
Niben_Salicylic_DEG$pcrtB28h_vs_GFP28h_filtered <- ifelse(is.na(Niben_Salicylic_DEG$pcrtB28h_vs_GFP28h_filtered) == TRUE, " ", "*")
Niben_Salicylic_DEG$pcrtB34h_vs_GFP34h_filtered <- ifelse(is.na(Niben_Salicylic_DEG$pcrtB34h_vs_GFP34h_filtered) == TRUE, " ", "*")
Niben_Salicylic_DEG$pcrtB37h_vs_GFP37h_filtered <- ifelse(is.na(Niben_Salicylic_DEG$pcrtB37h_vs_GFP37h_filtered) == TRUE, " ", "*")
Niben_Salicylic_DEG$pcrtBZZ_vs_GFPZZ_filtered <- ifelse(is.na(Niben_Salicylic_DEG$pcrtBZZ_vs_GFPZZ_filtered) == TRUE, " ", "*")
Niben_Salicylic_DEG$pcrtBCC_vs_GFPCC_filtered <- ifelse(is.na(Niben_Salicylic_DEG$pcrtBCC_vs_GFPCC_filtered) == TRUE, " ", "*")
Niben_Salicylic_DEG$pcrtB40h_vs_GFP40h_filtered <- ifelse(is.na(Niben_Salicylic_DEG$pcrtB40h_vs_GFP40h_filtered) == TRUE, " ", "*")
Niben_Salicylic_DEG$pcrtBDD_vs_GFPDD_filtered <- ifelse(is.na(Niben_Salicylic_DEG$pcrtBDD_vs_GFPDD_filtered) == TRUE, " ", "*")
Niben_Salicylic_DEG$pcrtB46h_vs_GFP46h_filtered <- ifelse(is.na(Niben_Salicylic_DEG$pcrtB46h_vs_GFP46h_filtered) == TRUE, " ", "*")
Niben_Salicylic_DEG$pcrtB56h_vs_GFP56h_filtered <- ifelse(is.na(Niben_Salicylic_DEG$pcrtB56h_vs_GFP56h_filtered) == TRUE, " ", "*")
Niben_Salicylic_DEG$pcrtBEE_vs_GFPEE_filtered <- ifelse(is.na(Niben_Salicylic_DEG$pcrtBEE_vs_GFPEE_filtered) == TRUE, " ", "*")
Niben_Salicylic_DEG$pcrtB96h_vs_GFP96h_filtered <- ifelse(is.na(Niben_Salicylic_DEG$pcrtB96h_vs_GFP96h_filtered) == TRUE, " ", "*")
row.names(Niben_Salicylic_DEG) <- Niben_Salicylic_DEG$Protein
Niben_Salicylic_DEG[1] <- NULL

Niben_Salicylic_DEG <- Niben_Salicylic_DEG %>%
  select(pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

row.names(Niben_Salicylic_FC) <- Niben_Salicylic_FC$Protein
Niben_Salicylic_FC[1] <- NULL

Niben_Salicylic_FC <- Niben_Salicylic_FC %>%
  select(pcrtB22h_vs_GFP22h_All, pcrtBAA_vs_GFPAA_All, pcrtB25h_vs_GFP25h_All, 
         pcrtBYY_vs_GFPYY_All, pcrtBBB_vs_GFPBB_All, pcrtB28h_vs_GFP28h_All, pcrtB34h_vs_GFP34h_All, pcrtBZZ_vs_GFPZZ_All, 
         pcrtB37h_vs_GFP37h_All, pcrtBCC_vs_GFPCC_All, pcrtB40h_vs_GFP40h_All, pcrtBDD_vs_GFPDD_All, pcrtB46h_vs_GFP46h_All, 
         pcrtBEE_vs_GFPEE_All, pcrtB56h_vs_GFP56h_All, pcrtB96h_vs_GFP96h_All)


min(Niben_Salicylic_FC, na.rm = TRUE)
max(Niben_Salicylic_FC, na.rm = TRUE)
breaks_Salicylic_FC=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Niben_Salicylic_FC, # %>% select(1:15), 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Salicylic_FC,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_Salicylic_Annotation, annotation_colors=color_Salicylic, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h", "96h"),
         cex = 1,
         angle_col=45,
         main = "Salicylic",
         annotation_legend = TRUE,
         display_numbers = Niben_Salicylic_DEG, # %>% select(1:15),
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(84, 104),
         annotation_row = Niben_Salicylic_Modules,
         annotation_colors = module_colors
         #gaps_col = c(15)
)


rownames(Niben_Salicylic_TPM) <- Niben_Salicylic_TPM$Protein
Niben_Salicylic_TPM[1] <- NULL

min(Niben_Salicylic_TPM, na.rm = TRUE)
max(Niben_Salicylic_TPM, na.rm = TRUE)
breaks_Niben_Salicylic_TPM=c(seq(0,100, length.out = 100))

Niben_Salicylic_TPM_norm <- normalize_df(Niben_Salicylic_TPM)
Niben_Salicylic_TPM_norm_Cornell <- Niben_Salicylic_TPM %>%
  select(1:48) 
Niben_Salicylic_TPM_norm_Cornell <- normalize_df(Niben_Salicylic_TPM_norm_Cornell)


pheatmap(Niben_Salicylic_TPM_norm_Cornell, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Niben_Salicylic_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_Salicylic_Annotation, annotation_colors=color_Salicylic, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Salicylic",
         annotation_legend = TRUE,
         #display_numbers = Niben_Salicylic_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         annotation_row = Niben_Salicylic_Modules,
         annotation_colors = module_colors,
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(24)#, 48, 51)
)



#####Representing Heatmap for Sly

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Sly_Salicylic_Ann <- read.table("Funcional Groups/FigTree/Sly_Salicylic.txt",
                               sep = "\t", header = TRUE)

colnames(Sly_Salicylic_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Sly_Salicylic <- merge(Sly_Salicylic_Ann, Summary_Yazdani_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Sly_Salicylic_TPM <- Sly_Salicylic %>% select(Protein, 4:66)
Sly_Salicylic_TPM <- Sly_Salicylic_TPM[!duplicated(Sly_Salicylic_TPM$Protein),]
Sly_Salicylic_TPM <- Sly_Salicylic_TPM[order(match(Sly_Salicylic_TPM[,1], Sly_Salicylic_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_Salicylic_FC <- Sly_Salicylic %>% select(Protein, 67:71)
Sly_Salicylic_FC <- Sly_Salicylic_FC[!duplicated(Sly_Salicylic_FC$Protein),]
Sly_Salicylic_FC <- Sly_Salicylic_FC[order(match(Sly_Salicylic_FC[,1], Sly_Salicylic_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_Salicylic_DEG <- Sly_Salicylic %>% select(Protein, 72:76)
Sly_Salicylic_DEG <- Sly_Salicylic_DEG[!duplicated(Sly_Salicylic_DEG$Protein),]
Sly_Salicylic_DEG <- Sly_Salicylic_DEG[order(match(Sly_Salicylic_DEG[,1], Sly_Salicylic_Ann[,2])),]

Sly_Salicylic_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Sly_Salicylic_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Sly_Salicylic_DEG$Ripening_Orange_DEG <- ifelse(is.na(Sly_Salicylic_DEG$Ripening_Orange_DEG) == TRUE, " ", "*")
Sly_Salicylic_DEG$Ripening_Red_DEG <- ifelse(is.na(Sly_Salicylic_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
Sly_Salicylic_DEG$ORHis_MG_DEG <- ifelse(is.na(Sly_Salicylic_DEG$ORHis_MG_DEG) == TRUE, " ", "*")
Sly_Salicylic_DEG$ORHis_Breaker_DEG <- ifelse(is.na(Sly_Salicylic_DEG$ORHis_Breaker_DEG) == TRUE, " ", "*")
row.names(Sly_Salicylic_DEG) <- Sly_Salicylic_DEG$Protein
Sly_Salicylic_DEG[1] <- NULL

rownames(Sly_Salicylic_FC) <- Sly_Salicylic_FC$Protein
Sly_Salicylic_FC[1] <- NULL

min(Sly_Salicylic_FC, na.rm = TRUE)
max(Sly_Salicylic_FC, na.rm = TRUE)
breaks_Sly_Salicylic=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Sly_Salicylic_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Sly_Salicylic,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_Salicylic_Annotation, annotation_colors=color_Salicylic, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Salicylic",
         annotation_legend = TRUE,
         display_numbers = Sly_Salicylic_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(39, 48),
         gaps_col = c(3)
)


rownames(Sly_Salicylic_TPM) <- Sly_Salicylic_TPM$Protein
Sly_Salicylic_TPM[1] <- NULL

min(Sly_Salicylic_TPM, na.rm = TRUE)
max(Sly_Salicylic_TPM, na.rm = TRUE)
breaks_Sly_Salicylic_TPM=c(seq(0,100, length.out = 200))

Sly_Salicylic_TPM_norm <- normalize_df(Sly_Salicylic_TPM)

pheatmap(Sly_Salicylic_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Sly_Salicylic_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_Salicylic_Annotation, annotation_colors=color_Salicylic, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Salicylic",
         annotation_legend = TRUE,
         #display_numbers = Sly_Salicylic_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(21,42)
)


#####Representing Heatmap for Cann

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Cann_Salicylic_Ann <- read.table("Funcional Groups/FigTree/Cann_Salicylic.txt",
                                sep = "\t", header = TRUE)

colnames(Cann_Salicylic_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Cann_Salicylic <- merge(Cann_Salicylic_Ann, Summary_Pepper_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Cann_Salicylic_TPM <- Cann_Salicylic %>% select(Protein, 4:17)
Cann_Salicylic_TPM <- Cann_Salicylic_TPM[!duplicated(Cann_Salicylic_TPM$Protein),]
Cann_Salicylic_TPM <- Cann_Salicylic_TPM[order(match(Cann_Salicylic_TPM[,1], Cann_Salicylic_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_Salicylic_FC <- Cann_Salicylic %>% select(Protein, 18:19)
Cann_Salicylic_FC <- Cann_Salicylic_FC[!duplicated(Cann_Salicylic_FC$Protein),]
Cann_Salicylic_FC <- Cann_Salicylic_FC[order(match(Cann_Salicylic_FC[,1], Cann_Salicylic_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_Salicylic_DEG <- Cann_Salicylic %>% select(Protein, 20:21)
Cann_Salicylic_DEG <- Cann_Salicylic_DEG[!duplicated(Cann_Salicylic_DEG$Protein),]
Cann_Salicylic_DEG <- Cann_Salicylic_DEG[order(match(Cann_Salicylic_DEG[,1], Cann_Salicylic_Ann[,2])),]

Cann_Salicylic_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Cann_Salicylic_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Cann_Salicylic_DEG$Ripening_Red_DEG <- ifelse(is.na(Cann_Salicylic_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
row.names(Cann_Salicylic_DEG) <- Cann_Salicylic_DEG$Protein
Cann_Salicylic_DEG[1] <- NULL

rownames(Cann_Salicylic_FC) <- Cann_Salicylic_FC$Protein
Cann_Salicylic_FC[1] <- NULL

min(Cann_Salicylic_FC, na.rm = TRUE)
max(Cann_Salicylic_FC, na.rm = TRUE)
breaks_Cann_Salicylic=c(seq(-4.5,0, length.out = 100), seq(0.01,6,length.out = 100))

pheatmap(Cann_Salicylic_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Cann_Salicylic,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_Salicylic_Annotation, annotation_colors=color_Salicylic, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Red"),
         cex = 1,
         angle_col=45,
         main = "Salicylic",
         annotation_legend = TRUE,
         display_numbers = Cann_Salicylic_DEG,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)


rownames(Cann_Salicylic_TPM) <- Cann_Salicylic_TPM$Protein
Cann_Salicylic_TPM[1] <- NULL

min(Cann_Salicylic_TPM, na.rm = TRUE)
max(Cann_Salicylic_TPM, na.rm = TRUE)
breaks_Cann_Salicylic_TPM=c(seq(0,100, length.out = 200))

Cann_Salicylic_TPM_norm <- normalize_df(Cann_Salicylic_TPM)

pheatmap(Cann_Salicylic_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Cann_Salicylic_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_Salicylic_Annotation, annotation_colors=color_Salicylic, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Salicylic",
         annotation_legend = TRUE,
         #display_numbers = Cann_Salicylic_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(4,9)
)


#########################################################################################################################################
####################################  Strigo  #############################################################################################
#########################################################################################################################################

Strigo_Ath <- read.table("Funcional Groups/Ath_Strigo.txt", header = TRUE)
Strigo_Ath <- merge(Strigo_Ath, Ath_Gene_Models, by.x="ID", by.y=1)
Strigo_Ath_fasta <- Ath11_fasta[c(which(names(Ath11_fasta) %in% Strigo_Ath$V2))]
Strigo_Ath_2 <- Strigo_Ath %>% select(V2, Name)
write.fasta(Strigo_Ath_fasta, as.string=TRUE, names= names(Strigo_Ath_fasta),
            file.out = "Funcional Groups/Ath_Strigo.fasta")
rename.fasta(infile = "Funcional Groups/Ath_Strigo.fasta", Strigo_Ath_2, outfile = "Funcional Groups/Ath_Strigo_2.fasta")

Strigo_Niben_Candidates <- read.table("Funcional Groups/Niben_Candidates_Strigo.txt")
Strigo_Niben_Candidates <- Strigo_Niben_Candidates[Strigo_Niben_Candidates$V3>=35,] %>% select(V2)
Strigo_Niben_Candidates <- unique(Strigo_Niben_Candidates)
Strigo_Niben_Candidates_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Strigo_Niben_Candidates$V2))]
write.fasta(Strigo_Niben_Candidates_fasta, as.string=TRUE, names= names(Strigo_Niben_Candidates_fasta),
            file.out = "Funcional Groups/Strigo_Niben_Candidates_35.fasta")

Strigo_Sly_Candidates <- read.table("Funcional Groups/Sly_Candidates_Strigo.txt")
Strigo_Sly_Candidates <- Strigo_Sly_Candidates[Strigo_Sly_Candidates$V3>=35,] %>% select(V2)
Strigo_Sly_Candidates <- unique(Strigo_Sly_Candidates)
Strigo_Sly_Candidates_fasta <- Tomato_4.0_fasta[c(which(names(Tomato_4.0_fasta) %in% Strigo_Sly_Candidates$V2))]
write.fasta(Strigo_Sly_Candidates_fasta, as.string=TRUE, names= names(Strigo_Sly_Candidates_fasta),
            file.out = "Funcional Groups/Strigo_Sly_Candidates_35.fasta")

Strigo_Cann_Candidates <- read.table("Funcional Groups/Cann_Candidates_Strigo.txt")
Strigo_Cann_Candidates <- Strigo_Cann_Candidates[Strigo_Cann_Candidates$V3>=35,] %>% select(V2)
Strigo_Cann_Candidates <- unique(Strigo_Cann_Candidates)
Strigo_Cann_Candidates_fasta <- Cannuum_ASM51225v2_fasta[c(which(names(Cannuum_ASM51225v2_fasta) %in% Strigo_Cann_Candidates$V2))]
write.fasta(Strigo_Cann_Candidates_fasta, as.string=TRUE, names= names(Strigo_Cann_Candidates_fasta),
            file.out = "Funcional Groups/Strigo_Cann_Candidates_35.fasta")

Niben_Strigo_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_Strigo_Ann$Name))]
write.fasta(Niben_Strigo_fasta, as.string=TRUE, names= names(Niben_Strigo_fasta),
            file.out = "Funcional Groups/Niben_Strigo.fasta")
Niben_Strigo_Ann_2 <- Niben_Strigo_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_Strigo.fasta", Niben_Strigo_Ann, outfile = "Funcional Groups/Niben_Strigo_2.fasta")

save_heatmaps("Strigo", height = 3)

## Representing Heatmap

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Niben_Strigo_Ann <- read.table("Funcional Groups/FigTree/Niben_Strigo.txt",
                                  sep = "\t", header = TRUE)

colnames(Niben_Strigo_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_Strigo <- merge(Niben_Strigo_Ann, TPM_All_FC_Annotated, by="Name")

## We create the annotation for Modules

Niben_Strigo_Modules <- which_Module_pwd30(Niben_Strigo_Ann)

## We select only the FC data from the different comparisons
Niben_Strigo_TPM <- Niben_Strigo %>% select(Protein, 4:51, 61:66)
Niben_Strigo_TPM <- Niben_Strigo_TPM[!duplicated(Niben_Strigo_TPM$Protein),]
Niben_Strigo_TPM <- Niben_Strigo_TPM[order(match(Niben_Strigo_TPM[,1], Niben_Strigo_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_Strigo_FC <- Niben_Strigo %>% select(Protein, 71, 73:87)
Niben_Strigo_FC <- Niben_Strigo_FC[!duplicated(Niben_Strigo_FC$Protein),]
Niben_Strigo_FC <- Niben_Strigo_FC[order(match(Niben_Strigo_FC[,1], Niben_Strigo_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_Strigo_DEG <- Niben_Strigo %>% select(Protein, 89, 91:105)
Niben_Strigo_DEG <- Niben_Strigo_DEG[!duplicated(Niben_Strigo_DEG$Protein),]
Niben_Strigo_DEG <- Niben_Strigo_DEG[order(match(Niben_Strigo_DEG[,1], Niben_Strigo_Ann[,2])),]

Niben_Strigo_DEG$pcrtB22h_vs_GFP22h_filtered <- ifelse(is.na(Niben_Strigo_DEG$pcrtB22h_vs_GFP22h_filtered) == TRUE, " ", "*")
Niben_Strigo_DEG$pcrtB25h_vs_GFP25h_filtered <- ifelse(is.na(Niben_Strigo_DEG$pcrtB25h_vs_GFP25h_filtered) == TRUE, " ", "*")
Niben_Strigo_DEG$pcrtBAA_vs_GFPAA_filtered <- ifelse(is.na(Niben_Strigo_DEG$pcrtBAA_vs_GFPAA_filtered) == TRUE, " ", "*")
Niben_Strigo_DEG$pcrtBBB_vs_GFPBB_filtered <- ifelse(is.na(Niben_Strigo_DEG$pcrtBBB_vs_GFPBB_filtered) == TRUE, " ", "*")
Niben_Strigo_DEG$pcrtBYY_vs_GFPYY_filtered <- ifelse(is.na(Niben_Strigo_DEG$pcrtBYY_vs_GFPYY_filtered) == TRUE, " ", "*")
Niben_Strigo_DEG$pcrtB28h_vs_GFP28h_filtered <- ifelse(is.na(Niben_Strigo_DEG$pcrtB28h_vs_GFP28h_filtered) == TRUE, " ", "*")
Niben_Strigo_DEG$pcrtB34h_vs_GFP34h_filtered <- ifelse(is.na(Niben_Strigo_DEG$pcrtB34h_vs_GFP34h_filtered) == TRUE, " ", "*")
Niben_Strigo_DEG$pcrtB37h_vs_GFP37h_filtered <- ifelse(is.na(Niben_Strigo_DEG$pcrtB37h_vs_GFP37h_filtered) == TRUE, " ", "*")
Niben_Strigo_DEG$pcrtBZZ_vs_GFPZZ_filtered <- ifelse(is.na(Niben_Strigo_DEG$pcrtBZZ_vs_GFPZZ_filtered) == TRUE, " ", "*")
Niben_Strigo_DEG$pcrtBCC_vs_GFPCC_filtered <- ifelse(is.na(Niben_Strigo_DEG$pcrtBCC_vs_GFPCC_filtered) == TRUE, " ", "*")
Niben_Strigo_DEG$pcrtB40h_vs_GFP40h_filtered <- ifelse(is.na(Niben_Strigo_DEG$pcrtB40h_vs_GFP40h_filtered) == TRUE, " ", "*")
Niben_Strigo_DEG$pcrtBDD_vs_GFPDD_filtered <- ifelse(is.na(Niben_Strigo_DEG$pcrtBDD_vs_GFPDD_filtered) == TRUE, " ", "*")
Niben_Strigo_DEG$pcrtB46h_vs_GFP46h_filtered <- ifelse(is.na(Niben_Strigo_DEG$pcrtB46h_vs_GFP46h_filtered) == TRUE, " ", "*")
Niben_Strigo_DEG$pcrtB56h_vs_GFP56h_filtered <- ifelse(is.na(Niben_Strigo_DEG$pcrtB56h_vs_GFP56h_filtered) == TRUE, " ", "*")
Niben_Strigo_DEG$pcrtBEE_vs_GFPEE_filtered <- ifelse(is.na(Niben_Strigo_DEG$pcrtBEE_vs_GFPEE_filtered) == TRUE, " ", "*")
Niben_Strigo_DEG$pcrtB96h_vs_GFP96h_filtered <- ifelse(is.na(Niben_Strigo_DEG$pcrtB96h_vs_GFP96h_filtered) == TRUE, " ", "*")
row.names(Niben_Strigo_DEG) <- Niben_Strigo_DEG$Protein
Niben_Strigo_DEG[1] <- NULL

Niben_Strigo_DEG <- Niben_Strigo_DEG %>%
  select(pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

row.names(Niben_Strigo_FC) <- Niben_Strigo_FC$Protein
Niben_Strigo_FC[1] <- NULL

Niben_Strigo_FC <- Niben_Strigo_FC %>%
  select(pcrtB22h_vs_GFP22h_All, pcrtBAA_vs_GFPAA_All, pcrtB25h_vs_GFP25h_All, 
         pcrtBYY_vs_GFPYY_All, pcrtBBB_vs_GFPBB_All, pcrtB28h_vs_GFP28h_All, pcrtB34h_vs_GFP34h_All, pcrtBZZ_vs_GFPZZ_All, 
         pcrtB37h_vs_GFP37h_All, pcrtBCC_vs_GFPCC_All, pcrtB40h_vs_GFP40h_All, pcrtBDD_vs_GFPDD_All, pcrtB46h_vs_GFP46h_All, 
         pcrtBEE_vs_GFPEE_All, pcrtB56h_vs_GFP56h_All, pcrtB96h_vs_GFP96h_All)


min(Niben_Strigo_FC, na.rm = TRUE)
max(Niben_Strigo_FC, na.rm = TRUE)
breaks_Strigo_FC=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Niben_Strigo_FC, # %>% select(1:15), 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Strigo_FC,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_Strigo_Annotation, annotation_colors=color_Strigo, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h", "96h"),
         cex = 1,
         angle_col=45,
         main = "Strigo",
         annotation_legend = TRUE,
         display_numbers = Niben_Strigo_DEG, # %>% select(1:15),
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(84, 104),
         annotation_row = Niben_Strigo_Modules,
         annotation_colors = module_colors
         #gaps_col = c(15)
)


rownames(Niben_Strigo_TPM) <- Niben_Strigo_TPM$Protein
Niben_Strigo_TPM[1] <- NULL

min(Niben_Strigo_TPM, na.rm = TRUE)
max(Niben_Strigo_TPM, na.rm = TRUE)
breaks_Niben_Strigo_TPM=c(seq(0,100, length.out = 100))

Niben_Strigo_TPM_norm <- normalize_df(Niben_Strigo_TPM)
Niben_Strigo_TPM_norm_Cornell <- Niben_Strigo_TPM %>%
  select(1:48) 
Niben_Strigo_TPM_norm_Cornell <- normalize_df(Niben_Strigo_TPM_norm_Cornell)


pheatmap(Niben_Strigo_TPM_norm_Cornell, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Niben_Strigo_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_Strigo_Annotation, annotation_colors=color_Strigo, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Strigo",
         annotation_legend = TRUE,
         #display_numbers = Niben_Strigo_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         annotation_row = Niben_Strigo_Modules,
         annotation_colors = module_colors,
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(24)#, 48, 51)
)



#####Representing Heatmap for Sly

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Sly_Strigo_Ann <- read.table("Funcional Groups/FigTree/Sly_Strigo.txt",
                                sep = "\t", header = TRUE)

colnames(Sly_Strigo_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Sly_Strigo <- merge(Sly_Strigo_Ann, Summary_Yazdani_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Sly_Strigo_TPM <- Sly_Strigo %>% select(Protein, 4:66)
Sly_Strigo_TPM <- Sly_Strigo_TPM[!duplicated(Sly_Strigo_TPM$Protein),]
Sly_Strigo_TPM <- Sly_Strigo_TPM[order(match(Sly_Strigo_TPM[,1], Sly_Strigo_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_Strigo_FC <- Sly_Strigo %>% select(Protein, 67:71)
Sly_Strigo_FC <- Sly_Strigo_FC[!duplicated(Sly_Strigo_FC$Protein),]
Sly_Strigo_FC <- Sly_Strigo_FC[order(match(Sly_Strigo_FC[,1], Sly_Strigo_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_Strigo_DEG <- Sly_Strigo %>% select(Protein, 72:76)
Sly_Strigo_DEG <- Sly_Strigo_DEG[!duplicated(Sly_Strigo_DEG$Protein),]
Sly_Strigo_DEG <- Sly_Strigo_DEG[order(match(Sly_Strigo_DEG[,1], Sly_Strigo_Ann[,2])),]

Sly_Strigo_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Sly_Strigo_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Sly_Strigo_DEG$Ripening_Orange_DEG <- ifelse(is.na(Sly_Strigo_DEG$Ripening_Orange_DEG) == TRUE, " ", "*")
Sly_Strigo_DEG$Ripening_Red_DEG <- ifelse(is.na(Sly_Strigo_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
Sly_Strigo_DEG$ORHis_MG_DEG <- ifelse(is.na(Sly_Strigo_DEG$ORHis_MG_DEG) == TRUE, " ", "*")
Sly_Strigo_DEG$ORHis_Breaker_DEG <- ifelse(is.na(Sly_Strigo_DEG$ORHis_Breaker_DEG) == TRUE, " ", "*")
row.names(Sly_Strigo_DEG) <- Sly_Strigo_DEG$Protein
Sly_Strigo_DEG[1] <- NULL

rownames(Sly_Strigo_FC) <- Sly_Strigo_FC$Protein
Sly_Strigo_FC[1] <- NULL

min(Sly_Strigo_FC, na.rm = TRUE)
max(Sly_Strigo_FC, na.rm = TRUE)
breaks_Sly_Strigo=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Sly_Strigo_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Sly_Strigo,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_Strigo_Annotation, annotation_colors=color_Strigo, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Strigo",
         annotation_legend = TRUE,
         display_numbers = Sly_Strigo_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(39, 48),
         gaps_col = c(3)
)


rownames(Sly_Strigo_TPM) <- Sly_Strigo_TPM$Protein
Sly_Strigo_TPM[1] <- NULL

min(Sly_Strigo_TPM, na.rm = TRUE)
max(Sly_Strigo_TPM, na.rm = TRUE)
breaks_Sly_Strigo_TPM=c(seq(0,100, length.out = 200))

Sly_Strigo_TPM_norm <- normalize_df(Sly_Strigo_TPM)

pheatmap(Sly_Strigo_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Sly_Strigo_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_Strigo_Annotation, annotation_colors=color_Strigo, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Strigo",
         annotation_legend = TRUE,
         #display_numbers = Sly_Strigo_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(21,42)
)


#####Representing Heatmap for Cann

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Cann_Strigo_Ann <- read.table("Funcional Groups/FigTree/Cann_Strigo.txt",
                                 sep = "\t", header = TRUE)

colnames(Cann_Strigo_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Cann_Strigo <- merge(Cann_Strigo_Ann, Summary_Pepper_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Cann_Strigo_TPM <- Cann_Strigo %>% select(Protein, 4:17)
Cann_Strigo_TPM <- Cann_Strigo_TPM[!duplicated(Cann_Strigo_TPM$Protein),]
Cann_Strigo_TPM <- Cann_Strigo_TPM[order(match(Cann_Strigo_TPM[,1], Cann_Strigo_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_Strigo_FC <- Cann_Strigo %>% select(Protein, 18:19)
Cann_Strigo_FC <- Cann_Strigo_FC[!duplicated(Cann_Strigo_FC$Protein),]
Cann_Strigo_FC <- Cann_Strigo_FC[order(match(Cann_Strigo_FC[,1], Cann_Strigo_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_Strigo_DEG <- Cann_Strigo %>% select(Protein, 20:21)
Cann_Strigo_DEG <- Cann_Strigo_DEG[!duplicated(Cann_Strigo_DEG$Protein),]
Cann_Strigo_DEG <- Cann_Strigo_DEG[order(match(Cann_Strigo_DEG[,1], Cann_Strigo_Ann[,2])),]

Cann_Strigo_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Cann_Strigo_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Cann_Strigo_DEG$Ripening_Red_DEG <- ifelse(is.na(Cann_Strigo_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
row.names(Cann_Strigo_DEG) <- Cann_Strigo_DEG$Protein
Cann_Strigo_DEG[1] <- NULL

rownames(Cann_Strigo_FC) <- Cann_Strigo_FC$Protein
Cann_Strigo_FC[1] <- NULL

min(Cann_Strigo_FC, na.rm = TRUE)
max(Cann_Strigo_FC, na.rm = TRUE)
breaks_Cann_Strigo=c(seq(-4.5,0, length.out = 100), seq(0.01,6,length.out = 100))

pheatmap(Cann_Strigo_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Cann_Strigo,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_Strigo_Annotation, annotation_colors=color_Strigo, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Red"),
         cex = 1,
         angle_col=45,
         main = "Strigo",
         annotation_legend = TRUE,
         display_numbers = Cann_Strigo_DEG,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)


rownames(Cann_Strigo_TPM) <- Cann_Strigo_TPM$Protein
Cann_Strigo_TPM[1] <- NULL

min(Cann_Strigo_TPM, na.rm = TRUE)
max(Cann_Strigo_TPM, na.rm = TRUE)
breaks_Cann_Strigo_TPM=c(seq(0,100, length.out = 200))

Cann_Strigo_TPM_norm <- normalize_df(Cann_Strigo_TPM)

pheatmap(Cann_Strigo_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Cann_Strigo_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_Strigo_Annotation, annotation_colors=color_Strigo, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Strigo",
         annotation_legend = TRUE,
         #display_numbers = Cann_Strigo_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(4,9)
)


#########################################################################################################################################
####################################  PIFs  #############################################################################################
#########################################################################################################################################

PIFs_Ath <- read.table("Funcional Groups/Ath_PIFs.txt", header = TRUE)
PIFs_Ath <- merge(PIFs_Ath, Ath_Gene_Models, by.x="ID", by.y=1)
PIFs_Ath_fasta <- Ath11_fasta[c(which(names(Ath11_fasta) %in% PIFs_Ath$V2))]
PIFs_Ath_2 <- PIFs_Ath %>% select(V2, Name)
write.fasta(PIFs_Ath_fasta, as.string=TRUE, names= names(PIFs_Ath_fasta),
            file.out = "Funcional Groups/Ath_PIFs.fasta")
rename.fasta(infile = "Funcional Groups/Ath_PIFs.fasta", PIFs_Ath_2, outfile = "Funcional Groups/Ath_PIFs_2.fasta")

PIFs_Niben_Candidates <- read.table("Funcional Groups/Niben_Candidates_PIFs.txt")
PIFs_Niben_Candidates <- PIFs_Niben_Candidates[PIFs_Niben_Candidates$V3>=35,] %>% select(V2)
PIFs_Niben_Candidates <- unique(PIFs_Niben_Candidates)
PIFs_Niben_Candidates_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% PIFs_Niben_Candidates$V2))]
write.fasta(PIFs_Niben_Candidates_fasta, as.string=TRUE, names= names(PIFs_Niben_Candidates_fasta),
            file.out = "Funcional Groups/PIFs_Niben_Candidates_35.fasta")

PIFs_Sly_Candidates <- read.table("Funcional Groups/Sly_Candidates_PIFs.txt")
PIFs_Sly_Candidates <- PIFs_Sly_Candidates[PIFs_Sly_Candidates$V3>=35,] %>% select(V2)
PIFs_Sly_Candidates <- unique(PIFs_Sly_Candidates)
PIFs_Sly_Candidates_fasta <- Tomato_4.0_fasta[c(which(names(Tomato_4.0_fasta) %in% PIFs_Sly_Candidates$V2))]
write.fasta(PIFs_Sly_Candidates_fasta, as.string=TRUE, names= names(PIFs_Sly_Candidates_fasta),
            file.out = "Funcional Groups/PIFs_Sly_Candidates_35.fasta")

PIFs_Cann_Candidates <- read.table("Funcional Groups/Cann_Candidates_PIFs.txt")
PIFs_Cann_Candidates <- PIFs_Cann_Candidates[PIFs_Cann_Candidates$V3>=35,] %>% select(V2)
PIFs_Cann_Candidates <- unique(PIFs_Cann_Candidates)
PIFs_Cann_Candidates_fasta <- Cannuum_ASM51225v2_fasta[c(which(names(Cannuum_ASM51225v2_fasta) %in% PIFs_Cann_Candidates$V2))]
write.fasta(PIFs_Cann_Candidates_fasta, as.string=TRUE, names= names(PIFs_Cann_Candidates_fasta),
            file.out = "Funcional Groups/PIFs_Cann_Candidates_35.fasta")

Niben_PIFs_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_PIFs_Ann$Name))]
write.fasta(Niben_PIFs_fasta, as.string=TRUE, names= names(Niben_PIFs_fasta),
            file.out = "Funcional Groups/Niben_PIFs.fasta")
Niben_PIFs_Ann_2 <- Niben_PIFs_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_PIFs.fasta", Niben_PIFs_Ann, outfile = "Funcional Groups/Niben_PIFs_2.fasta")

save_heatmaps("PIFs", height = 5)

## Representing Heatmap

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Niben_PIFs_Ann <- read.table("Funcional Groups/FigTree/Niben_PIFs.txt",
                               sep = "\t", header = TRUE)

colnames(Niben_PIFs_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_PIFs <- merge(Niben_PIFs_Ann, TPM_All_FC_Annotated, by="Name")

## We create the annotation for Modules

Niben_PIFs_Modules <- which_Module_pwd30(Niben_PIFs_Ann)

## We select only the FC data from the different comparisons
Niben_PIFs_TPM <- Niben_PIFs %>% select(Protein, 4:51, 61:66)
Niben_PIFs_TPM <- Niben_PIFs_TPM[!duplicated(Niben_PIFs_TPM$Protein),]
Niben_PIFs_TPM <- Niben_PIFs_TPM[order(match(Niben_PIFs_TPM[,1], Niben_PIFs_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_PIFs_FC <- Niben_PIFs %>% select(Protein, 71, 73:87)
Niben_PIFs_FC <- Niben_PIFs_FC[!duplicated(Niben_PIFs_FC$Protein),]
Niben_PIFs_FC <- Niben_PIFs_FC[order(match(Niben_PIFs_FC[,1], Niben_PIFs_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_PIFs_DEG <- Niben_PIFs %>% select(Protein, 89, 91:105)
Niben_PIFs_DEG <- Niben_PIFs_DEG[!duplicated(Niben_PIFs_DEG$Protein),]
Niben_PIFs_DEG <- Niben_PIFs_DEG[order(match(Niben_PIFs_DEG[,1], Niben_PIFs_Ann[,2])),]

Niben_PIFs_DEG$pcrtB22h_vs_GFP22h_filtered <- ifelse(is.na(Niben_PIFs_DEG$pcrtB22h_vs_GFP22h_filtered) == TRUE, " ", "*")
Niben_PIFs_DEG$pcrtB25h_vs_GFP25h_filtered <- ifelse(is.na(Niben_PIFs_DEG$pcrtB25h_vs_GFP25h_filtered) == TRUE, " ", "*")
Niben_PIFs_DEG$pcrtBAA_vs_GFPAA_filtered <- ifelse(is.na(Niben_PIFs_DEG$pcrtBAA_vs_GFPAA_filtered) == TRUE, " ", "*")
Niben_PIFs_DEG$pcrtBBB_vs_GFPBB_filtered <- ifelse(is.na(Niben_PIFs_DEG$pcrtBBB_vs_GFPBB_filtered) == TRUE, " ", "*")
Niben_PIFs_DEG$pcrtBYY_vs_GFPYY_filtered <- ifelse(is.na(Niben_PIFs_DEG$pcrtBYY_vs_GFPYY_filtered) == TRUE, " ", "*")
Niben_PIFs_DEG$pcrtB28h_vs_GFP28h_filtered <- ifelse(is.na(Niben_PIFs_DEG$pcrtB28h_vs_GFP28h_filtered) == TRUE, " ", "*")
Niben_PIFs_DEG$pcrtB34h_vs_GFP34h_filtered <- ifelse(is.na(Niben_PIFs_DEG$pcrtB34h_vs_GFP34h_filtered) == TRUE, " ", "*")
Niben_PIFs_DEG$pcrtB37h_vs_GFP37h_filtered <- ifelse(is.na(Niben_PIFs_DEG$pcrtB37h_vs_GFP37h_filtered) == TRUE, " ", "*")
Niben_PIFs_DEG$pcrtBZZ_vs_GFPZZ_filtered <- ifelse(is.na(Niben_PIFs_DEG$pcrtBZZ_vs_GFPZZ_filtered) == TRUE, " ", "*")
Niben_PIFs_DEG$pcrtBCC_vs_GFPCC_filtered <- ifelse(is.na(Niben_PIFs_DEG$pcrtBCC_vs_GFPCC_filtered) == TRUE, " ", "*")
Niben_PIFs_DEG$pcrtB40h_vs_GFP40h_filtered <- ifelse(is.na(Niben_PIFs_DEG$pcrtB40h_vs_GFP40h_filtered) == TRUE, " ", "*")
Niben_PIFs_DEG$pcrtBDD_vs_GFPDD_filtered <- ifelse(is.na(Niben_PIFs_DEG$pcrtBDD_vs_GFPDD_filtered) == TRUE, " ", "*")
Niben_PIFs_DEG$pcrtB46h_vs_GFP46h_filtered <- ifelse(is.na(Niben_PIFs_DEG$pcrtB46h_vs_GFP46h_filtered) == TRUE, " ", "*")
Niben_PIFs_DEG$pcrtB56h_vs_GFP56h_filtered <- ifelse(is.na(Niben_PIFs_DEG$pcrtB56h_vs_GFP56h_filtered) == TRUE, " ", "*")
Niben_PIFs_DEG$pcrtBEE_vs_GFPEE_filtered <- ifelse(is.na(Niben_PIFs_DEG$pcrtBEE_vs_GFPEE_filtered) == TRUE, " ", "*")
Niben_PIFs_DEG$pcrtB96h_vs_GFP96h_filtered <- ifelse(is.na(Niben_PIFs_DEG$pcrtB96h_vs_GFP96h_filtered) == TRUE, " ", "*")
row.names(Niben_PIFs_DEG) <- Niben_PIFs_DEG$Protein
Niben_PIFs_DEG[1] <- NULL

Niben_PIFs_DEG <- Niben_PIFs_DEG %>%
  select(pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

row.names(Niben_PIFs_FC) <- Niben_PIFs_FC$Protein
Niben_PIFs_FC[1] <- NULL

Niben_PIFs_FC <- Niben_PIFs_FC %>%
  select(pcrtB22h_vs_GFP22h_All, pcrtBAA_vs_GFPAA_All, pcrtB25h_vs_GFP25h_All, 
         pcrtBYY_vs_GFPYY_All, pcrtBBB_vs_GFPBB_All, pcrtB28h_vs_GFP28h_All, pcrtB34h_vs_GFP34h_All, pcrtBZZ_vs_GFPZZ_All, 
         pcrtB37h_vs_GFP37h_All, pcrtBCC_vs_GFPCC_All, pcrtB40h_vs_GFP40h_All, pcrtBDD_vs_GFPDD_All, pcrtB46h_vs_GFP46h_All, 
         pcrtBEE_vs_GFPEE_All, pcrtB56h_vs_GFP56h_All, pcrtB96h_vs_GFP96h_All)


min(Niben_PIFs_FC, na.rm = TRUE)
max(Niben_PIFs_FC, na.rm = TRUE)
breaks_PIFs_FC=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Niben_PIFs_FC %>% select(1:15), 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_PIFs_FC,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_PIFs_Annotation, annotation_colors=color_PIFs, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h"),#, "96h"),
         cex = 1,
         angle_col=45,
         main = "PIFs",
         annotation_legend = TRUE,
         display_numbers = Niben_PIFs_DEG %>% select(1:15),
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(84, 104),
         annotation_row = Niben_PIFs_Modules,
         annotation_colors = module_colors
         #gaps_col = c(15)
)


rownames(Niben_PIFs_TPM) <- Niben_PIFs_TPM$Protein
Niben_PIFs_TPM[1] <- NULL

min(Niben_PIFs_TPM, na.rm = TRUE)
max(Niben_PIFs_TPM, na.rm = TRUE)
breaks_Niben_PIFs_TPM=c(seq(0,100, length.out = 100))

Niben_PIFs_TPM_norm <- normalize_df(Niben_PIFs_TPM)
Niben_PIFs_TPM_norm_Cornell <- Niben_PIFs_TPM %>%
  select(1:48) 
Niben_PIFs_TPM_norm_Cornell <- normalize_df(Niben_PIFs_TPM_norm_Cornell)


pheatmap(Niben_PIFs_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Niben_PIFs_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_PIFs_Annotation, annotation_colors=color_PIFs, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "PIFs",
         annotation_legend = TRUE,
         #display_numbers = Niben_PIFs_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         annotation_row = Niben_PIFs_Modules,
         annotation_colors = module_colors,
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(24, 48, 51)
)



#####Representing Heatmap for Sly

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Sly_PIFs_Ann <- read.table("Funcional Groups/FigTree/Sly_PIFs.txt",
                             sep = "\t", header = TRUE)

colnames(Sly_PIFs_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Sly_PIFs <- merge(Sly_PIFs_Ann, Summary_Yazdani_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Sly_PIFs_TPM <- Sly_PIFs %>% select(Protein, 4:66)
Sly_PIFs_TPM <- Sly_PIFs_TPM[!duplicated(Sly_PIFs_TPM$Protein),]
Sly_PIFs_TPM <- Sly_PIFs_TPM[order(match(Sly_PIFs_TPM[,1], Sly_PIFs_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_PIFs_FC <- Sly_PIFs %>% select(Protein, 67:71)
Sly_PIFs_FC <- Sly_PIFs_FC[!duplicated(Sly_PIFs_FC$Protein),]
Sly_PIFs_FC <- Sly_PIFs_FC[order(match(Sly_PIFs_FC[,1], Sly_PIFs_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_PIFs_DEG <- Sly_PIFs %>% select(Protein, 72:76)
Sly_PIFs_DEG <- Sly_PIFs_DEG[!duplicated(Sly_PIFs_DEG$Protein),]
Sly_PIFs_DEG <- Sly_PIFs_DEG[order(match(Sly_PIFs_DEG[,1], Sly_PIFs_Ann[,2])),]

Sly_PIFs_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Sly_PIFs_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Sly_PIFs_DEG$Ripening_Orange_DEG <- ifelse(is.na(Sly_PIFs_DEG$Ripening_Orange_DEG) == TRUE, " ", "*")
Sly_PIFs_DEG$Ripening_Red_DEG <- ifelse(is.na(Sly_PIFs_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
Sly_PIFs_DEG$ORHis_MG_DEG <- ifelse(is.na(Sly_PIFs_DEG$ORHis_MG_DEG) == TRUE, " ", "*")
Sly_PIFs_DEG$ORHis_Breaker_DEG <- ifelse(is.na(Sly_PIFs_DEG$ORHis_Breaker_DEG) == TRUE, " ", "*")
row.names(Sly_PIFs_DEG) <- Sly_PIFs_DEG$Protein
Sly_PIFs_DEG[1] <- NULL

rownames(Sly_PIFs_FC) <- Sly_PIFs_FC$Protein
Sly_PIFs_FC[1] <- NULL

min(Sly_PIFs_FC, na.rm = TRUE)
max(Sly_PIFs_FC, na.rm = TRUE)
breaks_Sly_PIFs=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Sly_PIFs_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Sly_PIFs,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_PIFs_Annotation, annotation_colors=color_PIFs, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "PIFs",
         annotation_legend = TRUE,
         display_numbers = Sly_PIFs_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(39, 48),
         gaps_col = c(3)
)


rownames(Sly_PIFs_TPM) <- Sly_PIFs_TPM$Protein
Sly_PIFs_TPM[1] <- NULL

min(Sly_PIFs_TPM, na.rm = TRUE)
max(Sly_PIFs_TPM, na.rm = TRUE)
breaks_Sly_PIFs_TPM=c(seq(0,100, length.out = 200))

Sly_PIFs_TPM_norm <- normalize_df(Sly_PIFs_TPM)

pheatmap(Sly_PIFs_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Sly_PIFs_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_PIFs_Annotation, annotation_colors=color_PIFs, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "PIFs",
         annotation_legend = TRUE,
         #display_numbers = Sly_PIFs_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(21,42)
)


#####Representing Heatmap for Cann

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Cann_PIFs_Ann <- read.table("Funcional Groups/FigTree/Cann_PIFs.txt",
                              sep = "\t", header = TRUE)

colnames(Cann_PIFs_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Cann_PIFs <- merge(Cann_PIFs_Ann, Summary_Pepper_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Cann_PIFs_TPM <- Cann_PIFs %>% select(Protein, 4:17)
Cann_PIFs_TPM <- Cann_PIFs_TPM[!duplicated(Cann_PIFs_TPM$Protein),]
Cann_PIFs_TPM <- Cann_PIFs_TPM[order(match(Cann_PIFs_TPM[,1], Cann_PIFs_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_PIFs_FC <- Cann_PIFs %>% select(Protein, 18:19)
Cann_PIFs_FC <- Cann_PIFs_FC[!duplicated(Cann_PIFs_FC$Protein),]
Cann_PIFs_FC <- Cann_PIFs_FC[order(match(Cann_PIFs_FC[,1], Cann_PIFs_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_PIFs_DEG <- Cann_PIFs %>% select(Protein, 20:21)
Cann_PIFs_DEG <- Cann_PIFs_DEG[!duplicated(Cann_PIFs_DEG$Protein),]
Cann_PIFs_DEG <- Cann_PIFs_DEG[order(match(Cann_PIFs_DEG[,1], Cann_PIFs_Ann[,2])),]

Cann_PIFs_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Cann_PIFs_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Cann_PIFs_DEG$Ripening_Red_DEG <- ifelse(is.na(Cann_PIFs_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
row.names(Cann_PIFs_DEG) <- Cann_PIFs_DEG$Protein
Cann_PIFs_DEG[1] <- NULL

rownames(Cann_PIFs_FC) <- Cann_PIFs_FC$Protein
Cann_PIFs_FC[1] <- NULL

min(Cann_PIFs_FC, na.rm = TRUE)
max(Cann_PIFs_FC, na.rm = TRUE)
breaks_Cann_PIFs=c(seq(-4.5,0, length.out = 100), seq(0.01,6,length.out = 100))

pheatmap(Cann_PIFs_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Cann_PIFs,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_PIFs_Annotation, annotation_colors=color_PIFs, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Red"),
         cex = 1,
         angle_col=45,
         main = "PIFs",
         annotation_legend = TRUE,
         display_numbers = Cann_PIFs_DEG,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)


rownames(Cann_PIFs_TPM) <- Cann_PIFs_TPM$Protein
Cann_PIFs_TPM[1] <- NULL

min(Cann_PIFs_TPM, na.rm = TRUE)
max(Cann_PIFs_TPM, na.rm = TRUE)
breaks_Cann_PIFs_TPM=c(seq(0,100, length.out = 200))

Cann_PIFs_TPM_norm <- normalize_df(Cann_PIFs_TPM)

pheatmap(Cann_PIFs_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Cann_PIFs_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_PIFs_Annotation, annotation_colors=color_PIFs, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "PIFs",
         annotation_legend = TRUE,
         #display_numbers = Cann_PIFs_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(4,9)
)


#########################################################################################################################################
####################################  Clock  #############################################################################################
#########################################################################################################################################

Clock_Ath <- read.table("Funcional Groups/Ath_Clock.txt", header = TRUE)
Clock_Ath <- merge(Clock_Ath, Ath_Gene_Models, by.x="ID", by.y=1)
Clock_Ath_fasta <- Ath11_fasta[c(which(names(Ath11_fasta) %in% Clock_Ath$V2))]
Clock_Ath_2 <- Clock_Ath %>% select(V2, Name)
write.fasta(Clock_Ath_fasta, as.string=TRUE, names= names(Clock_Ath_fasta),
            file.out = "Funcional Groups/Ath_Clock.fasta")
rename.fasta(infile = "Funcional Groups/Ath_Clock.fasta", Clock_Ath_2, outfile = "Funcional Groups/Ath_Clock_2.fasta")

Clock_Niben_Candidates <- read.table("Funcional Groups/Niben_Candidates_Clock.txt")
Clock_Niben_Candidates <- Clock_Niben_Candidates[Clock_Niben_Candidates$V3>=35,] %>% select(V2)
Clock_Niben_Candidates <- unique(Clock_Niben_Candidates)
Clock_Niben_Candidates_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Clock_Niben_Candidates$V2))]
write.fasta(Clock_Niben_Candidates_fasta, as.string=TRUE, names= names(Clock_Niben_Candidates_fasta),
            file.out = "Funcional Groups/Clock_Niben_Candidates_35.fasta")

Clock_Sly_Candidates <- read.table("Funcional Groups/Sly_Candidates_Clock.txt")
Clock_Sly_Candidates <- Clock_Sly_Candidates[Clock_Sly_Candidates$V3>=35,] %>% select(V2)
Clock_Sly_Candidates <- unique(Clock_Sly_Candidates)
Clock_Sly_Candidates_fasta <- Tomato_4.0_fasta[c(which(names(Tomato_4.0_fasta) %in% Clock_Sly_Candidates$V2))]
write.fasta(Clock_Sly_Candidates_fasta, as.string=TRUE, names= names(Clock_Sly_Candidates_fasta),
            file.out = "Funcional Groups/Clock_Sly_Candidates_35.fasta")

Clock_Cann_Candidates <- read.table("Funcional Groups/Cann_Candidates_Clock.txt")
Clock_Cann_Candidates <- Clock_Cann_Candidates[Clock_Cann_Candidates$V3>=35,] %>% select(V2)
Clock_Cann_Candidates <- unique(Clock_Cann_Candidates)
Clock_Cann_Candidates_fasta <- Cannuum_ASM51225v2_fasta[c(which(names(Cannuum_ASM51225v2_fasta) %in% Clock_Cann_Candidates$V2))]
write.fasta(Clock_Cann_Candidates_fasta, as.string=TRUE, names= names(Clock_Cann_Candidates_fasta),
            file.out = "Funcional Groups/Clock_Cann_Candidates_35.fasta")

Niben_Clock_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_Clock_Ann$Name))]
write.fasta(Niben_Clock_fasta, as.string=TRUE, names= names(Niben_Clock_fasta),
            file.out = "Funcional Groups/Niben_Clock.fasta")
Niben_Clock_Ann_2 <- Niben_Clock_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_Clock.fasta", Niben_Clock_Ann, outfile = "Funcional Groups/Niben_Clock_2.fasta")

save_heatmaps("Clock", height = 9)

## Representing Heatmap

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Niben_Clock_Ann <- read.table("Funcional Groups/FigTree/Niben_Clock.txt",
                             sep = "\t", header = TRUE)

colnames(Niben_Clock_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_Clock <- merge(Niben_Clock_Ann, TPM_All_FC_Annotated, by="Name")

## We select only the FC data from the different comparisons
Niben_Clock_TPM <- Niben_Clock %>% select(Protein, 4:51, 61:66)
Niben_Clock_TPM <- Niben_Clock_TPM[!duplicated(Niben_Clock_TPM$Protein),]
Niben_Clock_TPM <- Niben_Clock_TPM[order(match(Niben_Clock_TPM[,1], Niben_Clock_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_Clock_FC <- Niben_Clock %>% select(Protein, 71, 73:87)
Niben_Clock_FC <- Niben_Clock_FC[!duplicated(Niben_Clock_FC$Protein),]
Niben_Clock_FC <- Niben_Clock_FC[order(match(Niben_Clock_FC[,1], Niben_Clock_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_Clock_DEG <- Niben_Clock %>% select(Protein, 89, 91:105)
Niben_Clock_DEG <- Niben_Clock_DEG[!duplicated(Niben_Clock_DEG$Protein),]
Niben_Clock_DEG <- Niben_Clock_DEG[order(match(Niben_Clock_DEG[,1], Niben_Clock_Ann[,2])),]

Niben_Clock_DEG$pcrtB22h_vs_GFP22h_filtered <- ifelse(is.na(Niben_Clock_DEG$pcrtB22h_vs_GFP22h_filtered) == TRUE, " ", "*")
Niben_Clock_DEG$pcrtB25h_vs_GFP25h_filtered <- ifelse(is.na(Niben_Clock_DEG$pcrtB25h_vs_GFP25h_filtered) == TRUE, " ", "*")
Niben_Clock_DEG$pcrtBAA_vs_GFPAA_filtered <- ifelse(is.na(Niben_Clock_DEG$pcrtBAA_vs_GFPAA_filtered) == TRUE, " ", "*")
Niben_Clock_DEG$pcrtBBB_vs_GFPBB_filtered <- ifelse(is.na(Niben_Clock_DEG$pcrtBBB_vs_GFPBB_filtered) == TRUE, " ", "*")
Niben_Clock_DEG$pcrtBYY_vs_GFPYY_filtered <- ifelse(is.na(Niben_Clock_DEG$pcrtBYY_vs_GFPYY_filtered) == TRUE, " ", "*")
Niben_Clock_DEG$pcrtB28h_vs_GFP28h_filtered <- ifelse(is.na(Niben_Clock_DEG$pcrtB28h_vs_GFP28h_filtered) == TRUE, " ", "*")
Niben_Clock_DEG$pcrtB34h_vs_GFP34h_filtered <- ifelse(is.na(Niben_Clock_DEG$pcrtB34h_vs_GFP34h_filtered) == TRUE, " ", "*")
Niben_Clock_DEG$pcrtB37h_vs_GFP37h_filtered <- ifelse(is.na(Niben_Clock_DEG$pcrtB37h_vs_GFP37h_filtered) == TRUE, " ", "*")
Niben_Clock_DEG$pcrtBZZ_vs_GFPZZ_filtered <- ifelse(is.na(Niben_Clock_DEG$pcrtBZZ_vs_GFPZZ_filtered) == TRUE, " ", "*")
Niben_Clock_DEG$pcrtBCC_vs_GFPCC_filtered <- ifelse(is.na(Niben_Clock_DEG$pcrtBCC_vs_GFPCC_filtered) == TRUE, " ", "*")
Niben_Clock_DEG$pcrtB40h_vs_GFP40h_filtered <- ifelse(is.na(Niben_Clock_DEG$pcrtB40h_vs_GFP40h_filtered) == TRUE, " ", "*")
Niben_Clock_DEG$pcrtBDD_vs_GFPDD_filtered <- ifelse(is.na(Niben_Clock_DEG$pcrtBDD_vs_GFPDD_filtered) == TRUE, " ", "*")
Niben_Clock_DEG$pcrtB46h_vs_GFP46h_filtered <- ifelse(is.na(Niben_Clock_DEG$pcrtB46h_vs_GFP46h_filtered) == TRUE, " ", "*")
Niben_Clock_DEG$pcrtB56h_vs_GFP56h_filtered <- ifelse(is.na(Niben_Clock_DEG$pcrtB56h_vs_GFP56h_filtered) == TRUE, " ", "*")
Niben_Clock_DEG$pcrtBEE_vs_GFPEE_filtered <- ifelse(is.na(Niben_Clock_DEG$pcrtBEE_vs_GFPEE_filtered) == TRUE, " ", "*")
Niben_Clock_DEG$pcrtB96h_vs_GFP96h_filtered <- ifelse(is.na(Niben_Clock_DEG$pcrtB96h_vs_GFP96h_filtered) == TRUE, " ", "*")
row.names(Niben_Clock_DEG) <- Niben_Clock_DEG$Protein
Niben_Clock_DEG[1] <- NULL

Niben_Clock_DEG <- Niben_Clock_DEG %>%
  select(pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

row.names(Niben_Clock_FC) <- Niben_Clock_FC$Protein
Niben_Clock_FC[1] <- NULL

Niben_Clock_FC <- Niben_Clock_FC %>%
  select(pcrtB22h_vs_GFP22h_All, pcrtBAA_vs_GFPAA_All, pcrtB25h_vs_GFP25h_All, 
         pcrtBYY_vs_GFPYY_All, pcrtBBB_vs_GFPBB_All, pcrtB28h_vs_GFP28h_All, pcrtB34h_vs_GFP34h_All, pcrtBZZ_vs_GFPZZ_All, 
         pcrtB37h_vs_GFP37h_All, pcrtBCC_vs_GFPCC_All, pcrtB40h_vs_GFP40h_All, pcrtBDD_vs_GFPDD_All, pcrtB46h_vs_GFP46h_All, 
         pcrtBEE_vs_GFPEE_All, pcrtB56h_vs_GFP56h_All, pcrtB96h_vs_GFP96h_All)


min(Niben_Clock_FC, na.rm = TRUE)
max(Niben_Clock_FC, na.rm = TRUE)
breaks_Clock_FC=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Niben_Clock_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Clock_FC,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_Clock_Annotation, annotation_colors=color_Clock, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h", "96h"),
         cex = 1,
         angle_col=45,
         main = "Clock",
         annotation_legend = TRUE,
         display_numbers = Niben_Clock_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(84, 104),
         gaps_col = c(15)
)

theme_Publication <- function(base_size=14, base_family="helvetica") {
  library(grid)
  library(ggthemes)
  (theme_foundation(base_size=base_size, base_family=base_family)
    + theme(plot.title = element_text(face = "bold",
                                      size = rel(1.2), hjust = 0.5),
            text = element_text(),
            panel.background = element_rect(colour = NA),
            plot.background = element_rect(colour = NA),
            panel.border = element_rect(colour = NA),
            axis.title = element_text(face = "bold",size = rel(1)),
            axis.title.y = element_text(angle=90,vjust =2),
            axis.title.x = element_text(vjust = -0.2),
            axis.text = element_text(), 
            axis.line = element_line(colour="black"),
            axis.ticks = element_line(),
            panel.grid.major = element_line(colour="#f0f0f0"),
            panel.grid.minor = element_blank(),
            legend.key = element_rect(colour = NA),
            legend.position = "bottom",
            legend.direction = "horizontal",
            legend.key.size= unit(0.2, "cm"),
            legend.margin = unit(0, "cm"),
            legend.title = element_text(face="italic"),
            plot.margin=unit(c(10,5,5,5),"mm"),
            strip.background=element_rect(colour="#f0f0f0",fill="#f0f0f0"),
            strip.text = element_text(face="bold")
    ))
  
}
windowsFonts("helvetica"= windowsFont("helvetica"))

Niben_Clock_TPM_adapted <- Niben_Clock_TPM[1:28,]
Niben_Clock_TPM_adapted <- Niben_Clock_TPM_adapted %>% select(1:48)
Niben_Clock_TPM_adapted$Gene <- rownames(Niben_Clock_TPM_adapted)
Niben_Clock_TPM_adapted <- Niben_Clock_TPM_adapted %>% gather(key= "Samples", value= "values", -Gene)
Niben_Clock_TPM_adapted$Gene <- factor(Niben_Clock_TPM_adapted$Gene, levels = c("NbTOC1a", "NbTOC1b", "NbTOC1c", "NbTOC1d",
                                                                                "NbLUX", "NbELF4", "NbELF4_L1", "NbELF4_L2",
                                                                                "NbGI1a", "NbGI1b", "NbGI2a", "NbGI2b", "NbGI2c",
                                                                                "NbPPR5a", "NbPPR5b", "NbPPR7a", "NbPPR7b",
                                                                                "NbPPR9a", "NbPPR9b", "NbELF3_L1a", "NbELF3_L1b",
                                                                                "NbELF3_L2a", "NbELF3_L2b", "NbRVE4a", "NbRVE4b",
                                                                                "NbRVE6a", "NbRVE6b","NbRVE6c"))
Niben_Clock_TPM_adapted$Construct <- str_sub(Niben_Clock_TPM_adapted$Samples, 1, -5)
Niben_Clock_TPM_adapted$Construct <- as.factor(Niben_Clock_TPM_adapted$Construct)
Niben_Clock_TPM_adapted$TimePoint <- str_sub(Niben_Clock_TPM_adapted$Samples, -4, -2)
Niben_Clock_TPM_adapted$TimePoint <- as.factor(Niben_Clock_TPM_adapted$TimePoint)
Niben_Clock_TPM_adapted$Replicate <- str_sub(Niben_Clock_TPM_adapted$Samples, -1)
Niben_Clock_TPM_adapted$Replicate <- as.factor(Niben_Clock_TPM_adapted$Replicate)

ggplot(Niben_Clock_TPM_adapted, aes(x = TimePoint, y = values, group = Construct))+
  geom_point(shape = 20, size = 3, aes(col = Construct), alpha = 0.5)+
  #stat_summary(fun.data = mean_sdl, geom = "errorbar",fun.args = list(mult = 1), width = 0.4)+
  geom_line(data = Niben_Clock_TPM_adapted %>%
              select(Gene, Construct, TimePoint, values) %>%
              group_by(Gene, Construct, TimePoint)%>%
              summarise(AVG = mean(values)),
            aes(x = TimePoint, y = AVG, col = Construct),
            size = 1)+
  geom_crossbar(data = Niben_Clock_TPM_adapted %>%
                  select(Gene, Construct, TimePoint, values) %>%
                  group_by(Gene, Construct, TimePoint)%>%
                  summarise(AVG = mean(values)),
                aes(x = TimePoint, y = AVG, ymin = AVG, ymax = AVG, col = Construct))+
  scale_fill_manual(values = c("GFP"= "#b2df8a", 
                               "pcrtB" = "#fdbf6f"))+
  scale_color_manual(values = c("GFP"= "#33a02c", 
                                "pcrtB" = "#ff7f00"))+
  scale_y_continuous("TPM")+
  labs(title = "Relative Expression", x = "Time Point")+
  facet_wrap(~Gene, scales = "free_y", ncol = 2)+
  theme_Publication()+
  theme(legend.position = "top")



rownames(Niben_Clock_TPM) <- Niben_Clock_TPM$Protein
Niben_Clock_TPM[1] <- NULL

min(Niben_Clock_TPM, na.rm = TRUE)
max(Niben_Clock_TPM, na.rm = TRUE)
breaks_Niben_Clock_TPM=c(seq(0,20, length.out = 100))

pheatmap(Niben_Clock_TPM, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         breaks = breaks_Niben_Clock_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_Clock_Annotation, annotation_colors=color_Clock, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Clock",
         annotation_legend = TRUE,
         #display_numbers = Niben_Clock_DEG,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)



#####Representing Heatmap for Sly

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Sly_Clock_Ann <- read.table("Funcional Groups/FigTree/Sly_Clock.txt",
                           sep = "\t", header = TRUE)

colnames(Sly_Clock_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Sly_Clock <- merge(Sly_Clock_Ann, Summary_Yazdani_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Sly_Clock_TPM <- Sly_Clock %>% select(Protein, 4:66)
Sly_Clock_TPM <- Sly_Clock_TPM[!duplicated(Sly_Clock_TPM$Protein),]
Sly_Clock_TPM <- Sly_Clock_TPM[order(match(Sly_Clock_TPM[,1], Sly_Clock_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_Clock_FC <- Sly_Clock %>% select(Protein, 67:71)
Sly_Clock_FC <- Sly_Clock_FC[!duplicated(Sly_Clock_FC$Protein),]
Sly_Clock_FC <- Sly_Clock_FC[order(match(Sly_Clock_FC[,1], Sly_Clock_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_Clock_DEG <- Sly_Clock %>% select(Protein, 72:76)
Sly_Clock_DEG <- Sly_Clock_DEG[!duplicated(Sly_Clock_DEG$Protein),]
Sly_Clock_DEG <- Sly_Clock_DEG[order(match(Sly_Clock_DEG[,1], Sly_Clock_Ann[,2])),]

Sly_Clock_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Sly_Clock_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Sly_Clock_DEG$Ripening_Orange_DEG <- ifelse(is.na(Sly_Clock_DEG$Ripening_Orange_DEG) == TRUE, " ", "*")
Sly_Clock_DEG$Ripening_Red_DEG <- ifelse(is.na(Sly_Clock_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
Sly_Clock_DEG$ORHis_MG_DEG <- ifelse(is.na(Sly_Clock_DEG$ORHis_MG_DEG) == TRUE, " ", "*")
Sly_Clock_DEG$ORHis_Breaker_DEG <- ifelse(is.na(Sly_Clock_DEG$ORHis_Breaker_DEG) == TRUE, " ", "*")
row.names(Sly_Clock_DEG) <- Sly_Clock_DEG$Protein
Sly_Clock_DEG[1] <- NULL

rownames(Sly_Clock_FC) <- Sly_Clock_FC$Protein
Sly_Clock_FC[1] <- NULL

min(Sly_Clock_FC, na.rm = TRUE)
max(Sly_Clock_FC, na.rm = TRUE)
breaks_Sly_Clock=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Sly_Clock_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Sly_Clock,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_Clock_Annotation, annotation_colors=color_Clock, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Clock",
         annotation_legend = TRUE,
         display_numbers = Sly_Clock_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(39, 48),
         gaps_col = c(3)
)


rownames(Sly_Clock_TPM) <- Sly_Clock_TPM$Protein
Sly_Clock_TPM[1] <- NULL

min(Sly_Clock_TPM, na.rm = TRUE)
max(Sly_Clock_TPM, na.rm = TRUE)
breaks_Sly_Clock_TPM=c(seq(0,50, length.out = 100))

pheatmap(Sly_Clock_TPM, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         breaks = breaks_Sly_Clock_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_Clock_Annotation, annotation_colors=color_Clock, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Clock",
         annotation_legend = TRUE,
         #display_numbers = Sly_Clock_DEG,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)


#####Representing Heatmap for Cann

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Cann_Clock_Ann <- read.table("Funcional Groups/FigTree/Cann_Clock.txt",
                            sep = "\t", header = TRUE)

colnames(Cann_Clock_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Cann_Clock <- merge(Cann_Clock_Ann, Summary_Pepper_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Cann_Clock_TPM <- Cann_Clock %>% select(Protein, 4:17)
Cann_Clock_TPM <- Cann_Clock_TPM[!duplicated(Cann_Clock_TPM$Protein),]
Cann_Clock_TPM <- Cann_Clock_TPM[order(match(Cann_Clock_TPM[,1], Cann_Clock_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_Clock_FC <- Cann_Clock %>% select(Protein, 18:19)
Cann_Clock_FC <- Cann_Clock_FC[!duplicated(Cann_Clock_FC$Protein),]
Cann_Clock_FC <- Cann_Clock_FC[order(match(Cann_Clock_FC[,1], Cann_Clock_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_Clock_DEG <- Cann_Clock %>% select(Protein, 20:21)
Cann_Clock_DEG <- Cann_Clock_DEG[!duplicated(Cann_Clock_DEG$Protein),]
Cann_Clock_DEG <- Cann_Clock_DEG[order(match(Cann_Clock_DEG[,1], Cann_Clock_Ann[,2])),]

Cann_Clock_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Cann_Clock_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Cann_Clock_DEG$Ripening_Red_DEG <- ifelse(is.na(Cann_Clock_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
row.names(Cann_Clock_DEG) <- Cann_Clock_DEG$Protein
Cann_Clock_DEG[1] <- NULL

rownames(Cann_Clock_FC) <- Cann_Clock_FC$Protein
Cann_Clock_FC[1] <- NULL

min(Cann_Clock_FC, na.rm = TRUE)
max(Cann_Clock_FC, na.rm = TRUE)
breaks_Cann_Clock=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Cann_Clock_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Cann_Clock,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_Clock_Annotation, annotation_colors=color_Clock, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Red"),
         cex = 1,
         angle_col=45,
         main = "Clock",
         annotation_legend = TRUE,
         display_numbers = Cann_Clock_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(45, 56),
         #gaps_col = c(2,5)
)


rownames(Cann_Clock_TPM) <- Cann_Clock_TPM$Protein
Cann_Clock_TPM[1] <- NULL

min(Cann_Clock_TPM, na.rm = TRUE)
max(Cann_Clock_TPM, na.rm = TRUE)
breaks_Cann_Clock_TPM=c(seq(0,100, length.out = 200))

pheatmap(Cann_Clock_TPM, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Cann_Clock_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "row",
         #annotation_row=Cann_Clock_Annotation, annotation_colors=color_Clock, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Clock",
         annotation_legend = TRUE,
         #display_numbers = Cann_Clock_DEG,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)


#########################################################################################################################################
####################################  Phyto  #############################################################################################
#########################################################################################################################################

Phyto_Ath <- read.table("Funcional Groups/Ath_Phyto.txt", header = TRUE)
Phyto_Ath <- merge(Phyto_Ath, Ath_Gene_Models, by.x="ID", by.y=1)
Phyto_Ath_fasta <- Ath11_fasta[c(which(names(Ath11_fasta) %in% Phyto_Ath$V2))]
Phyto_Ath_2 <- Phyto_Ath %>% select(V2, Name)
write.fasta(Phyto_Ath_fasta, as.string=TRUE, names= names(Phyto_Ath_fasta),
            file.out = "Funcional Groups/Ath_Phyto.fasta")
rename.fasta(infile = "Funcional Groups/Ath_Phyto.fasta", Phyto_Ath_2, outfile = "Funcional Groups/Ath_Phyto_2.fasta")

Phyto_Niben_Candidates <- read.table("Funcional Groups/Niben_Candidates_Phyto.txt")
Phyto_Niben_Candidates <- Phyto_Niben_Candidates[Phyto_Niben_Candidates$V3>=35,] %>% select(V2)
Phyto_Niben_Candidates <- unique(Phyto_Niben_Candidates)
Phyto_Niben_Candidates_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Phyto_Niben_Candidates$V2))]
write.fasta(Phyto_Niben_Candidates_fasta, as.string=TRUE, names= names(Phyto_Niben_Candidates_fasta),
            file.out = "Funcional Groups/Phyto_Niben_Candidates_35.fasta")

Phyto_Sly_Candidates <- read.table("Funcional Groups/Sly_Candidates_Phyto.txt")
Phyto_Sly_Candidates <- Phyto_Sly_Candidates[Phyto_Sly_Candidates$V3>=35,] %>% select(V2)
Phyto_Sly_Candidates <- unique(Phyto_Sly_Candidates)
Phyto_Sly_Candidates_fasta <- Tomato_4.0_fasta[c(which(names(Tomato_4.0_fasta) %in% Phyto_Sly_Candidates$V2))]
write.fasta(Phyto_Sly_Candidates_fasta, as.string=TRUE, names= names(Phyto_Sly_Candidates_fasta),
            file.out = "Funcional Groups/Phyto_Sly_Candidates_35.fasta")

Phyto_Cann_Candidates <- read.table("Funcional Groups/Cann_Candidates_Phyto.txt")
Phyto_Cann_Candidates <- Phyto_Cann_Candidates[Phyto_Cann_Candidates$V3>=35,] %>% select(V2)
Phyto_Cann_Candidates <- unique(Phyto_Cann_Candidates)
Phyto_Cann_Candidates_fasta <- Cannuum_ASM51225v2_fasta[c(which(names(Cannuum_ASM51225v2_fasta) %in% Phyto_Cann_Candidates$V2))]
write.fasta(Phyto_Cann_Candidates_fasta, as.string=TRUE, names= names(Phyto_Cann_Candidates_fasta),
            file.out = "Funcional Groups/Phyto_Cann_Candidates_35.fasta")

Niben_Phyto_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_Phyto_Ann$Name))]
write.fasta(Niben_Phyto_fasta, as.string=TRUE, names= names(Niben_Phyto_fasta),
            file.out = "Funcional Groups/Niben_Phyto.fasta")
Niben_Phyto_Ann_2 <- Niben_Phyto_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_Phyto.fasta", Niben_Phyto_Ann, outfile = "Funcional Groups/Niben_Phyto_2.fasta")

save_heatmaps("Phyto", height = 5)

## Representing Heatmap

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Niben_Phyto_Ann <- read.table("Funcional Groups/FigTree/Niben_Phyto.txt",
                             sep = "\t", header = TRUE)

colnames(Niben_Phyto_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_Phyto <- merge(Niben_Phyto_Ann, TPM_All_FC_Annotated, by="Name")

## We create the annotation for Modules

Niben_Phyto_Modules <- which_Module_pwd30(Niben_Phyto_Ann)

## We select only the FC data from the different comparisons
Niben_Phyto_TPM <- Niben_Phyto %>% select(Protein, 4:51, 61:66)
Niben_Phyto_TPM <- Niben_Phyto_TPM[!duplicated(Niben_Phyto_TPM$Protein),]
Niben_Phyto_TPM <- Niben_Phyto_TPM[order(match(Niben_Phyto_TPM[,1], Niben_Phyto_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_Phyto_FC <- Niben_Phyto %>% select(Protein, 71, 73:87)
Niben_Phyto_FC <- Niben_Phyto_FC[!duplicated(Niben_Phyto_FC$Protein),]
Niben_Phyto_FC <- Niben_Phyto_FC[order(match(Niben_Phyto_FC[,1], Niben_Phyto_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_Phyto_DEG <- Niben_Phyto %>% select(Protein, 89, 91:105)
Niben_Phyto_DEG <- Niben_Phyto_DEG[!duplicated(Niben_Phyto_DEG$Protein),]
Niben_Phyto_DEG <- Niben_Phyto_DEG[order(match(Niben_Phyto_DEG[,1], Niben_Phyto_Ann[,2])),]

Niben_Phyto_DEG$pcrtB22h_vs_GFP22h_filtered <- ifelse(is.na(Niben_Phyto_DEG$pcrtB22h_vs_GFP22h_filtered) == TRUE, " ", "*")
Niben_Phyto_DEG$pcrtB25h_vs_GFP25h_filtered <- ifelse(is.na(Niben_Phyto_DEG$pcrtB25h_vs_GFP25h_filtered) == TRUE, " ", "*")
Niben_Phyto_DEG$pcrtBAA_vs_GFPAA_filtered <- ifelse(is.na(Niben_Phyto_DEG$pcrtBAA_vs_GFPAA_filtered) == TRUE, " ", "*")
Niben_Phyto_DEG$pcrtBBB_vs_GFPBB_filtered <- ifelse(is.na(Niben_Phyto_DEG$pcrtBBB_vs_GFPBB_filtered) == TRUE, " ", "*")
Niben_Phyto_DEG$pcrtBYY_vs_GFPYY_filtered <- ifelse(is.na(Niben_Phyto_DEG$pcrtBYY_vs_GFPYY_filtered) == TRUE, " ", "*")
Niben_Phyto_DEG$pcrtB28h_vs_GFP28h_filtered <- ifelse(is.na(Niben_Phyto_DEG$pcrtB28h_vs_GFP28h_filtered) == TRUE, " ", "*")
Niben_Phyto_DEG$pcrtB34h_vs_GFP34h_filtered <- ifelse(is.na(Niben_Phyto_DEG$pcrtB34h_vs_GFP34h_filtered) == TRUE, " ", "*")
Niben_Phyto_DEG$pcrtB37h_vs_GFP37h_filtered <- ifelse(is.na(Niben_Phyto_DEG$pcrtB37h_vs_GFP37h_filtered) == TRUE, " ", "*")
Niben_Phyto_DEG$pcrtBZZ_vs_GFPZZ_filtered <- ifelse(is.na(Niben_Phyto_DEG$pcrtBZZ_vs_GFPZZ_filtered) == TRUE, " ", "*")
Niben_Phyto_DEG$pcrtBCC_vs_GFPCC_filtered <- ifelse(is.na(Niben_Phyto_DEG$pcrtBCC_vs_GFPCC_filtered) == TRUE, " ", "*")
Niben_Phyto_DEG$pcrtB40h_vs_GFP40h_filtered <- ifelse(is.na(Niben_Phyto_DEG$pcrtB40h_vs_GFP40h_filtered) == TRUE, " ", "*")
Niben_Phyto_DEG$pcrtBDD_vs_GFPDD_filtered <- ifelse(is.na(Niben_Phyto_DEG$pcrtBDD_vs_GFPDD_filtered) == TRUE, " ", "*")
Niben_Phyto_DEG$pcrtB46h_vs_GFP46h_filtered <- ifelse(is.na(Niben_Phyto_DEG$pcrtB46h_vs_GFP46h_filtered) == TRUE, " ", "*")
Niben_Phyto_DEG$pcrtB56h_vs_GFP56h_filtered <- ifelse(is.na(Niben_Phyto_DEG$pcrtB56h_vs_GFP56h_filtered) == TRUE, " ", "*")
Niben_Phyto_DEG$pcrtBEE_vs_GFPEE_filtered <- ifelse(is.na(Niben_Phyto_DEG$pcrtBEE_vs_GFPEE_filtered) == TRUE, " ", "*")
Niben_Phyto_DEG$pcrtB96h_vs_GFP96h_filtered <- ifelse(is.na(Niben_Phyto_DEG$pcrtB96h_vs_GFP96h_filtered) == TRUE, " ", "*")
row.names(Niben_Phyto_DEG) <- Niben_Phyto_DEG$Protein
Niben_Phyto_DEG[1] <- NULL

Niben_Phyto_DEG <- Niben_Phyto_DEG %>%
  select(pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

row.names(Niben_Phyto_FC) <- Niben_Phyto_FC$Protein
Niben_Phyto_FC[1] <- NULL

Niben_Phyto_FC <- Niben_Phyto_FC %>%
  select(pcrtB22h_vs_GFP22h_All, pcrtBAA_vs_GFPAA_All, pcrtB25h_vs_GFP25h_All, 
         pcrtBYY_vs_GFPYY_All, pcrtBBB_vs_GFPBB_All, pcrtB28h_vs_GFP28h_All, pcrtB34h_vs_GFP34h_All, pcrtBZZ_vs_GFPZZ_All, 
         pcrtB37h_vs_GFP37h_All, pcrtBCC_vs_GFPCC_All, pcrtB40h_vs_GFP40h_All, pcrtBDD_vs_GFPDD_All, pcrtB46h_vs_GFP46h_All, 
         pcrtBEE_vs_GFPEE_All, pcrtB56h_vs_GFP56h_All, pcrtB96h_vs_GFP96h_All)


min(Niben_Phyto_FC, na.rm = TRUE)
max(Niben_Phyto_FC, na.rm = TRUE)
breaks_Phyto_FC=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Niben_Phyto_FC, # %>% select(1:15), 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Phyto_FC,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_Phyto_Annotation, annotation_colors=color_Phyto, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h", "96h"),
         cex = 1,
         angle_col=45,
         main = "Phyto",
         annotation_legend = TRUE,
         display_numbers = Niben_Phyto_DEG, # %>% select(1:15),
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(84, 104),
         annotation_row = Niben_Phyto_Modules,
         annotation_colors = module_colors
         #gaps_col = c(15)
)


rownames(Niben_Phyto_TPM) <- Niben_Phyto_TPM$Protein
Niben_Phyto_TPM[1] <- NULL

min(Niben_Phyto_TPM, na.rm = TRUE)
max(Niben_Phyto_TPM, na.rm = TRUE)
breaks_Niben_Phyto_TPM=c(seq(0,100, length.out = 100))

Niben_Phyto_TPM_norm <- normalize_df(Niben_Phyto_TPM)
Niben_Phyto_TPM_norm_Cornell <- Niben_Phyto_TPM %>%
  select(1:48) 
Niben_Phyto_TPM_norm_Cornell <- normalize_df(Niben_Phyto_TPM_norm_Cornell)


pheatmap(Niben_Phyto_TPM_norm_Cornell, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Niben_Phyto_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_Phyto_Annotation, annotation_colors=color_Phyto, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Phyto",
         annotation_legend = TRUE,
         #display_numbers = Niben_Phyto_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         annotation_row = Niben_Phyto_Modules,
         annotation_colors = module_colors,
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(24)#, 48, 51)
)



#####Representing Heatmap for Sly

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Sly_Phyto_Ann <- read.table("Funcional Groups/FigTree/Sly_Phyto.txt",
                           sep = "\t", header = TRUE)

colnames(Sly_Phyto_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Sly_Phyto <- merge(Sly_Phyto_Ann, Summary_Yazdani_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Sly_Phyto_TPM <- Sly_Phyto %>% select(Protein, 4:66)
Sly_Phyto_TPM <- Sly_Phyto_TPM[!duplicated(Sly_Phyto_TPM$Protein),]
Sly_Phyto_TPM <- Sly_Phyto_TPM[order(match(Sly_Phyto_TPM[,1], Sly_Phyto_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_Phyto_FC <- Sly_Phyto %>% select(Protein, 67:71)
Sly_Phyto_FC <- Sly_Phyto_FC[!duplicated(Sly_Phyto_FC$Protein),]
Sly_Phyto_FC <- Sly_Phyto_FC[order(match(Sly_Phyto_FC[,1], Sly_Phyto_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_Phyto_DEG <- Sly_Phyto %>% select(Protein, 72:76)
Sly_Phyto_DEG <- Sly_Phyto_DEG[!duplicated(Sly_Phyto_DEG$Protein),]
Sly_Phyto_DEG <- Sly_Phyto_DEG[order(match(Sly_Phyto_DEG[,1], Sly_Phyto_Ann[,2])),]

Sly_Phyto_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Sly_Phyto_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Sly_Phyto_DEG$Ripening_Orange_DEG <- ifelse(is.na(Sly_Phyto_DEG$Ripening_Orange_DEG) == TRUE, " ", "*")
Sly_Phyto_DEG$Ripening_Red_DEG <- ifelse(is.na(Sly_Phyto_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
Sly_Phyto_DEG$ORHis_MG_DEG <- ifelse(is.na(Sly_Phyto_DEG$ORHis_MG_DEG) == TRUE, " ", "*")
Sly_Phyto_DEG$ORHis_Breaker_DEG <- ifelse(is.na(Sly_Phyto_DEG$ORHis_Breaker_DEG) == TRUE, " ", "*")
row.names(Sly_Phyto_DEG) <- Sly_Phyto_DEG$Protein
Sly_Phyto_DEG[1] <- NULL

rownames(Sly_Phyto_FC) <- Sly_Phyto_FC$Protein
Sly_Phyto_FC[1] <- NULL

min(Sly_Phyto_FC, na.rm = TRUE)
max(Sly_Phyto_FC, na.rm = TRUE)
breaks_Sly_Phyto=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Sly_Phyto_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Sly_Phyto,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_Phyto_Annotation, annotation_colors=color_Phyto, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Phyto",
         annotation_legend = TRUE,
         display_numbers = Sly_Phyto_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(39, 48),
         gaps_col = c(3)
)


rownames(Sly_Phyto_TPM) <- Sly_Phyto_TPM$Protein
Sly_Phyto_TPM[1] <- NULL

min(Sly_Phyto_TPM, na.rm = TRUE)
max(Sly_Phyto_TPM, na.rm = TRUE)
breaks_Sly_Phyto_TPM=c(seq(0,100, length.out = 200))

Sly_Phyto_TPM_norm <- normalize_df(Sly_Phyto_TPM)

pheatmap(Sly_Phyto_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Sly_Phyto_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_Phyto_Annotation, annotation_colors=color_Phyto, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Phyto",
         annotation_legend = TRUE,
         #display_numbers = Sly_Phyto_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(21,42)
)


#####Representing Heatmap for Cann

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Cann_Phyto_Ann <- read.table("Funcional Groups/FigTree/Cann_Phyto.txt",
                            sep = "\t", header = TRUE)

colnames(Cann_Phyto_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Cann_Phyto <- merge(Cann_Phyto_Ann, Summary_Pepper_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Cann_Phyto_TPM <- Cann_Phyto %>% select(Protein, 4:17)
Cann_Phyto_TPM <- Cann_Phyto_TPM[!duplicated(Cann_Phyto_TPM$Protein),]
Cann_Phyto_TPM <- Cann_Phyto_TPM[order(match(Cann_Phyto_TPM[,1], Cann_Phyto_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_Phyto_FC <- Cann_Phyto %>% select(Protein, 18:19)
Cann_Phyto_FC <- Cann_Phyto_FC[!duplicated(Cann_Phyto_FC$Protein),]
Cann_Phyto_FC <- Cann_Phyto_FC[order(match(Cann_Phyto_FC[,1], Cann_Phyto_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_Phyto_DEG <- Cann_Phyto %>% select(Protein, 20:21)
Cann_Phyto_DEG <- Cann_Phyto_DEG[!duplicated(Cann_Phyto_DEG$Protein),]
Cann_Phyto_DEG <- Cann_Phyto_DEG[order(match(Cann_Phyto_DEG[,1], Cann_Phyto_Ann[,2])),]

Cann_Phyto_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Cann_Phyto_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Cann_Phyto_DEG$Ripening_Red_DEG <- ifelse(is.na(Cann_Phyto_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
row.names(Cann_Phyto_DEG) <- Cann_Phyto_DEG$Protein
Cann_Phyto_DEG[1] <- NULL

rownames(Cann_Phyto_FC) <- Cann_Phyto_FC$Protein
Cann_Phyto_FC[1] <- NULL

min(Cann_Phyto_FC, na.rm = TRUE)
max(Cann_Phyto_FC, na.rm = TRUE)
breaks_Cann_Phyto=c(seq(-4.5,0, length.out = 100), seq(0.01,6,length.out = 100))

pheatmap(Cann_Phyto_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Cann_Phyto,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_Phyto_Annotation, annotation_colors=color_Phyto, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Red"),
         cex = 1,
         angle_col=45,
         main = "Phyto",
         annotation_legend = TRUE,
         display_numbers = Cann_Phyto_DEG,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)


rownames(Cann_Phyto_TPM) <- Cann_Phyto_TPM$Protein
Cann_Phyto_TPM[1] <- NULL

min(Cann_Phyto_TPM, na.rm = TRUE)
max(Cann_Phyto_TPM, na.rm = TRUE)
breaks_Cann_Phyto_TPM=c(seq(0,100, length.out = 200))

Cann_Phyto_TPM_norm <- normalize_df(Cann_Phyto_TPM)

pheatmap(Cann_Phyto_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Cann_Phyto_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_Phyto_Annotation, annotation_colors=color_Phyto, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Phyto",
         annotation_legend = TRUE,
         #display_numbers = Cann_Phyto_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(4,9)
)

####################################  Primers  ###########################################################################################
#########################################################################################################################################

Niben_Primers_list <- list("Cal_BS"= "Niben101Scf05341g03002.1",
                           "Chap"="Niben101Scf10400Ctg002.1",
                           "Chl_ab"="Niben101Scf02262g01007.1",
                           "EL-IP1"="Niben101Scf01372g09018.1",
                           "EL-IP2"="Niben101Scf05962g04010.1",
                           "FIB1"="Niben101Scf09217g01001.1",
                           "FIB2"="Niben101Scf02953g00007.1",
                           "G6Pase"="Niben101Scf04011g11005.1",
                           "GLUT3"="Niben101Scf08196g01001.1",
                           "Hsp3"="Niben101Scf09262g01017.1",
                           "RAD"="Niben101Scf08718g00004.1",
                           "UNKN"="Niben101Scf01847g01001.1",
                           "Hsp21"="Niben101Scf08206g01019.1",
                           "Gluc6"="Niben101Scf04011g11005.1",
                           "MYC2"="Niben101Scf06822g04004.1",
                           "CHAP60A"="Niben101Scf01890g01005.1",
                           "Phenyl"="Niben101Scf04090g02003.1",
                           "HsfB1b"="Niben101Scf04313g01012.1",
                           "WRKY"="Niben101Scf04944g05002.1",
                           "Chap60B"="Niben101Scf10400g00005.1",
                           "Hsp18"="Niben101Scf01475g00019.1",
                           "AllOxCy"="Niben101Scf13816g00005.1",
                           "DXS2"="Niben101Scf02399g01011.1",
                           "GGDS8"="Niben101Scf01269Ctg111",
                           "HDS_2"="Niben101Scf03422Ctg027")

Niben_Primers_df <- as.data.frame(Niben_Primers_list)
Niben_Primers_df <- Niben_Primers_df %>% gather(key= "Protein", value = "Name")

Niben_Primers_261 <- read.table("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan/Funcional Groups/Niben_Primers_261.txt")
Niben_Primers_261 <- Niben_Primers_261[order(-Niben_Primers_261[, "V3"]),]
Niben_Primers_261 <- Niben_Primers_261[!duplicated(Niben_Primers_261$V1),]

## Representing Heatmap

Niben_Primers_Ann <- Niben_Primers_df %>% select(Name, Protein)
  
## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_Primers <- merge(Niben_Primers_Ann, TPM_All_FC_Annotated, by="Name")

## We create the annotation for Modules

Niben_Primers_Modules <- which_Module_pwd30(Niben_Primers_Ann)

## We select only the FC data from the different comparisons
Niben_Primers_TPM <- Niben_Primers %>% select(Protein, 4:51, 61:66)
Niben_Primers_TPM <- Niben_Primers_TPM[!duplicated(Niben_Primers_TPM$Protein),]
Niben_Primers_TPM <- Niben_Primers_TPM[order(match(Niben_Primers_TPM[,1], Niben_Primers_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_Primers_FC <- Niben_Primers %>% select(Protein, 71, 73:87)
Niben_Primers_FC <- Niben_Primers_FC[!duplicated(Niben_Primers_FC$Protein),]
Niben_Primers_FC <- Niben_Primers_FC[order(match(Niben_Primers_FC[,1], Niben_Primers_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_Primers_DEG <- Niben_Primers %>% select(Protein, 89, 91:105)
Niben_Primers_DEG <- Niben_Primers_DEG[!duplicated(Niben_Primers_DEG$Protein),]
Niben_Primers_DEG <- Niben_Primers_DEG[order(match(Niben_Primers_DEG[,1], Niben_Primers_Ann[,2])),]

Niben_Primers_DEG$pcrtB22h_vs_GFP22h_filtered <- ifelse(is.na(Niben_Primers_DEG$pcrtB22h_vs_GFP22h_filtered) == TRUE, " ", "*")
Niben_Primers_DEG$pcrtB25h_vs_GFP25h_filtered <- ifelse(is.na(Niben_Primers_DEG$pcrtB25h_vs_GFP25h_filtered) == TRUE, " ", "*")
Niben_Primers_DEG$pcrtBAA_vs_GFPAA_filtered <- ifelse(is.na(Niben_Primers_DEG$pcrtBAA_vs_GFPAA_filtered) == TRUE, " ", "*")
Niben_Primers_DEG$pcrtBBB_vs_GFPBB_filtered <- ifelse(is.na(Niben_Primers_DEG$pcrtBBB_vs_GFPBB_filtered) == TRUE, " ", "*")
Niben_Primers_DEG$pcrtBYY_vs_GFPYY_filtered <- ifelse(is.na(Niben_Primers_DEG$pcrtBYY_vs_GFPYY_filtered) == TRUE, " ", "*")
Niben_Primers_DEG$pcrtB28h_vs_GFP28h_filtered <- ifelse(is.na(Niben_Primers_DEG$pcrtB28h_vs_GFP28h_filtered) == TRUE, " ", "*")
Niben_Primers_DEG$pcrtB34h_vs_GFP34h_filtered <- ifelse(is.na(Niben_Primers_DEG$pcrtB34h_vs_GFP34h_filtered) == TRUE, " ", "*")
Niben_Primers_DEG$pcrtB37h_vs_GFP37h_filtered <- ifelse(is.na(Niben_Primers_DEG$pcrtB37h_vs_GFP37h_filtered) == TRUE, " ", "*")
Niben_Primers_DEG$pcrtBZZ_vs_GFPZZ_filtered <- ifelse(is.na(Niben_Primers_DEG$pcrtBZZ_vs_GFPZZ_filtered) == TRUE, " ", "*")
Niben_Primers_DEG$pcrtBCC_vs_GFPCC_filtered <- ifelse(is.na(Niben_Primers_DEG$pcrtBCC_vs_GFPCC_filtered) == TRUE, " ", "*")
Niben_Primers_DEG$pcrtB40h_vs_GFP40h_filtered <- ifelse(is.na(Niben_Primers_DEG$pcrtB40h_vs_GFP40h_filtered) == TRUE, " ", "*")
Niben_Primers_DEG$pcrtBDD_vs_GFPDD_filtered <- ifelse(is.na(Niben_Primers_DEG$pcrtBDD_vs_GFPDD_filtered) == TRUE, " ", "*")
Niben_Primers_DEG$pcrtB46h_vs_GFP46h_filtered <- ifelse(is.na(Niben_Primers_DEG$pcrtB46h_vs_GFP46h_filtered) == TRUE, " ", "*")
Niben_Primers_DEG$pcrtB56h_vs_GFP56h_filtered <- ifelse(is.na(Niben_Primers_DEG$pcrtB56h_vs_GFP56h_filtered) == TRUE, " ", "*")
Niben_Primers_DEG$pcrtBEE_vs_GFPEE_filtered <- ifelse(is.na(Niben_Primers_DEG$pcrtBEE_vs_GFPEE_filtered) == TRUE, " ", "*")
Niben_Primers_DEG$pcrtB96h_vs_GFP96h_filtered <- ifelse(is.na(Niben_Primers_DEG$pcrtB96h_vs_GFP96h_filtered) == TRUE, " ", "*")
row.names(Niben_Primers_DEG) <- Niben_Primers_DEG$Protein
Niben_Primers_DEG[1] <- NULL

Niben_Primers_DEG <- Niben_Primers_DEG %>%
  select(pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

row.names(Niben_Primers_FC) <- Niben_Primers_FC$Protein
Niben_Primers_FC[1] <- NULL

Niben_Primers_FC <- Niben_Primers_FC %>%
  select(pcrtB22h_vs_GFP22h_All, pcrtBAA_vs_GFPAA_All, pcrtB25h_vs_GFP25h_All, 
         pcrtBYY_vs_GFPYY_All, pcrtBBB_vs_GFPBB_All, pcrtB28h_vs_GFP28h_All, pcrtB34h_vs_GFP34h_All, pcrtBZZ_vs_GFPZZ_All, 
         pcrtB37h_vs_GFP37h_All, pcrtBCC_vs_GFPCC_All, pcrtB40h_vs_GFP40h_All, pcrtBDD_vs_GFPDD_All, pcrtB46h_vs_GFP46h_All, 
         pcrtBEE_vs_GFPEE_All, pcrtB56h_vs_GFP56h_All, pcrtB96h_vs_GFP96h_All)


min(Niben_Primers_FC, na.rm = TRUE)
max(Niben_Primers_FC, na.rm = TRUE)
breaks_Primers_FC=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Niben_Primers_FC, # %>% select(1:15), 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Primers_FC,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_Primers_Annotation, annotation_colors=color_Primers, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h", "96h"),
         cex = 1,
         angle_col=45,
         main = "Primers",
         annotation_legend = TRUE,
         display_numbers = Niben_Primers_DEG, # %>% select(1:15),
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(84, 104),
         annotation_row = Niben_Primers_Modules,
         annotation_colors = module_colors
         #gaps_col = c(15)
)


rownames(Niben_Primers_TPM) <- Niben_Primers_TPM$Protein
Niben_Primers_TPM[1] <- NULL

min(Niben_Primers_TPM, na.rm = TRUE)
max(Niben_Primers_TPM, na.rm = TRUE)
breaks_Niben_Primers_TPM=c(seq(0,100, length.out = 100))

Niben_Primers_TPM_norm <- normalize_df(Niben_Primers_TPM)
Niben_Primers_TPM_norm_Cornell <- Niben_Primers_TPM %>%
  select(1:48) 
Niben_Primers_TPM_norm_Cornell <- normalize_df(Niben_Primers_TPM_norm_Cornell)


pheatmap(Niben_Primers_TPM_norm_Cornell, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Niben_Primers_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_Primers_Annotation, annotation_colors=color_Primers, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Primers",
         annotation_legend = TRUE,
         #display_numbers = Niben_Primers_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         annotation_row = Niben_Primers_Modules,
         annotation_colors = module_colors,
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(24)#, 48, 51)
)



Primers_qPCR <- c("CHAP60A", "Phenyl", "Gluc6", "MYC2", "Chap60B", "Hsp18", "HsfB1b", "WRKY")

Niben_Primers_TPM_ggplot <- Niben_Primers_TPM
Niben_Primers_TPM_ggplot$Gene <- rownames(Niben_Primers_TPM_ggplot)
Niben_Primers_TPM_ggplot <- Niben_Primers_TPM_ggplot %>% gather(key= "Sample", value= "value", -Gene)
Niben_Primers_TPM_ggplot$Gene <- factor(Niben_Primers_TPM_ggplot$Gene, levels = c("CHAP60A", "Phenyl", "Gluc6", "MYC2", "Chap60B", "Hsp18", "HsfB1b", "WRKY", "AllOxCy", "Cal_BS", "Chl_ab", "DXS2", "EL.IP1", "EL.IP2", "FIB1", "FIB2", "GLUT3", "Hsp21", "Hsp3", "RAD", "UNKN"))
Niben_Primers_TPM_ggplot$Construct <- str_sub(Niben_Primers_TPM_ggplot$Sample, 1, -5)
Niben_Primers_TPM_ggplot$Construct[Niben_Primers_TPM_ggplot$Construct=="S19_GFP4d_1"] <- "GFP"
Niben_Primers_TPM_ggplot$Construct[Niben_Primers_TPM_ggplot$Construct=="S20_GFP4d_2"] <- "GFP"
Niben_Primers_TPM_ggplot$Construct[Niben_Primers_TPM_ggplot$Construct=="S21_GFP4d_4"] <- "GFP"
Niben_Primers_TPM_ggplot$Construct[Niben_Primers_TPM_ggplot$Construct=="S22_pcrtB4d_2"] <- "pcrtB"
Niben_Primers_TPM_ggplot$Construct[Niben_Primers_TPM_ggplot$Construct=="S23_pcrtB4d_3"] <- "pcrtB"
Niben_Primers_TPM_ggplot$Construct[Niben_Primers_TPM_ggplot$Construct=="S24_pcrtB4d_4"] <- "pcrtB"
Niben_Primers_TPM_ggplot$Construct <- as.factor(Niben_Primers_TPM_ggplot$Construct)
Niben_Primers_TPM_ggplot$TimePoint <- str_sub(Niben_Primers_TPM_ggplot$Sample, -4, -2)
Niben_Primers_TPM_ggplot$TimePoint[Niben_Primers_TPM_ggplot$TimePoint=="_S1"] <- "96h"
Niben_Primers_TPM_ggplot$TimePoint[Niben_Primers_TPM_ggplot$TimePoint=="_S2"] <- "96h"
Niben_Primers_TPM_ggplot$TimePoint <- as.factor(Niben_Primers_TPM_ggplot$TimePoint)
Niben_Primers_TPM_ggplot$Replicate <- str_sub(Niben_Primers_TPM_ggplot$Sample, -1)
Niben_Primers_TPM_ggplot$Replicate <- as.factor(Niben_Primers_TPM_ggplot$Replicate)
Niben_Primers_TPM_ggplot_Yes <- Niben_Primers_TPM_ggplot[Niben_Primers_TPM_ggplot$Gene %in% Primers_qPCR,]

'%ni%' <- Negate('%in%')

Niben_Primers_TPM_ggplot_No <- Niben_Primers_TPM_ggplot[Niben_Primers_TPM_ggplot$Gene %ni% Primers_qPCR,]

ggplot(Niben_Primers_TPM_ggplot[Niben_Primers_TPM_ggplot$Gene %ni% Primers_qPCR,], aes(x = TimePoint, y = value, group = Construct))+
  geom_point(shape = 20, size = 3, aes(col = Construct), alpha = 0.5)+
  #stat_summary(fun.data = mean_sdl, geom = "errorbar",fun.args = list(mult = 1), width = 0.4)+
  geom_line(data = Niben_Primers_TPM_ggplot[Niben_Primers_TPM_ggplot$Gene %ni% Primers_qPCR,] %>%
  select(Gene, Construct, TimePoint, value) %>%
  group_by(Gene, Construct, TimePoint)%>%
  summarise(AVG = mean(value)),
  aes(x = TimePoint, y = AVG, col = Construct),
  size = 1)+
  geom_crossbar(data = Niben_Primers_TPM_ggplot[Niben_Primers_TPM_ggplot$Gene %ni% Primers_qPCR,] %>%
  select(Gene, Construct, TimePoint, value) %>%
  group_by(Gene, Construct, TimePoint)%>%
  summarise(AVG = mean(value)),
  aes(x = TimePoint, y = AVG, ymin = AVG, ymax = AVG, col = Construct))+
  scale_fill_manual(values = c("GFP"= "#b2df8a",
  "pcrtB" = "#fdbf6f"))+
  scale_color_manual(values = c("GFP"= "#33a02c",
  "pcrtB" = "#ff7f00"))+
  scale_y_continuous("Normalized values")+
  labs(title = "Expression RNAseq", x = "Time Points")+
  facet_wrap(~Gene, scales = "free_y", ncol = 2)+
  theme_Publication()+
  theme(legend.position = "top")



####################################  HSP20  #############################################################################################
#########################################################################################################################################

HSP20_Ath <- read.delim("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan/Funcional Groups/Ath_HSP20.txt")
HSP20_Ath <- merge(HSP20_Ath, Ath_Gene_Models, by.x="GeneID", by.y=1)
HSP20_Ath_fasta <- Ath11_fasta[c(which(names(Ath11_fasta) %in% HSP20_Ath$V2))]
HSP20_Ath_2 <- HSP20_Ath %>% select(V2, Name)
write.fasta(HSP20_Ath_fasta, as.string=TRUE, names= names(HSP20_Ath_fasta),
            file.out = "Funcional Groups/Ath_HSP20.fasta")
rename.fasta(infile = "Funcional Groups/Ath_HSP20.fasta", HSP20_Ath_2, outfile = "Funcional Groups/Ath_HSP20_2.fasta")

HSP20_Niben_Candidates <- read.table("Funcional Groups/Niben_Candidates_HSP20.txt")
HSP20_Niben_Candidates_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% HSP20_Niben_Candidates$V1))]
write.fasta(HSP20_Niben_Candidates_fasta, as.string=TRUE, names= names(HSP20_Niben_Candidates_fasta),
            file.out = "Funcional Groups/HSP20_Niben_Candidates_35.fasta")

HSP20_Sly_Candidates <- read.table("Funcional Groups/Sly_Candidates_HSP20.txt")
HSP20_Sly_Candidates_fasta <- Tomato_4.0_fasta[c(which(names(Tomato_4.0_fasta) %in% HSP20_Sly_Candidates$V1))]
write.fasta(HSP20_Sly_Candidates_fasta, as.string=TRUE, names= names(HSP20_Sly_Candidates_fasta),
            file.out = "Funcional Groups/HSP20_Sly_Candidates_35.fasta")

HSP20_Cann_Candidates <- read.table("Funcional Groups/Cann_Candidates_HSP20.txt")
HSP20_Cann_Candidates_fasta <- Cannuum_ASM51225v2_fasta[c(which(names(Cannuum_ASM51225v2_fasta) %in% HSP20_Cann_Candidates$V1))]
write.fasta(HSP20_Cann_Candidates_fasta, as.string=TRUE, names= names(HSP20_Cann_Candidates_fasta),
            file.out = "Funcional Groups/HSP20_Cann_Candidates_35.fasta")

Niben_HSP20_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_HSP20_Ann$Name))]
write.fasta(Niben_HSP20_fasta, as.string=TRUE, names= names(Niben_HSP20_fasta),
            file.out = "Funcional Groups/Niben_HSP20.fasta")
Niben_HSP20_Ann_2 <- Niben_HSP20_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_HSP20.fasta", Niben_HSP20_Ann, outfile = "Funcional Groups/Niben_HSP20_2.fasta")

save_heatmaps("HSP20", height = 10)


## Representing Heatmap

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Niben_HSP20_Ann <- read.table("Funcional Groups/FigTree/Niben_HSP20.txt",
                              sep = "\t", header = TRUE)

colnames(Niben_HSP20_Ann) <- c("Name", "Protein")

Niben_HSP20_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_HSP20_Ann$Name))]
write.fasta(Niben_HSP20_fasta, as.string=TRUE, names= names(Niben_HSP20_fasta),
            file.out = "Funcional Groups/Niben_HSP20.fasta")
Niben_HSP20_Ann_2 <- Niben_HSP20_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_HSP20.fasta", Niben_HSP20_Ann, outfile = "Funcional Groups/Niben_HSP20_2.fasta")

Niben_HSP20_Chloro <- read.delim("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan/Funcional Groups/Niben_HSP20_2_summary.targetp2", 
                                 header=FALSE, comment.char="#")
colnames(Niben_HSP20_Chloro) <- c("ID",	"Prediction",	"noTP",	"SP",	"mTP",	"cTP",	"luTP",	"CS-Position")
Niben_HSP20_Chloro$Prediction <- factor(Niben_HSP20_Chloro$Prediction, levels = c("cTP","mTP","noTP"))
Niben_HSP20_Chloro <- merge(Niben_HSP20_Ann, Niben_HSP20_Chloro, by.x = "Protein", by.y = "ID")

Niben_HSP20_Chloro <- Niben_HSP20_Chloro[order(match(Niben_HSP20_Chloro$Protein, Niben_HSP20_Ann$Protein)),]
rownames(Niben_HSP20_Chloro) <- Niben_HSP20_Chloro$Protein
Niben_HSP20_Chloro <- Niben_HSP20_Chloro %>% select(Prediction)


## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_HSP20 <- merge(Niben_HSP20_Ann, TPM_All_FC_Annotated, by="Name")

## We create the annotation for Modules

Niben_HSP20_Modules <- which_Module_pwd30(Niben_HSP20_Ann)

## We merge this annotation of Modules with the annotation of Chloroplast TP prediction

Niben_HSP20_Modules_Chl <- merge(Niben_HSP20_Modules, Niben_HSP20_Chloro, by = 0)
Niben_HSP20_Modules_Chl <- Niben_HSP20_Modules_Chl[order(match(Niben_HSP20_Modules_Chl$Row.names, Niben_HSP20_Ann$Protein)),]
rownames(Niben_HSP20_Modules_Chl) <- Niben_HSP20_Modules_Chl$Row.names
Niben_HSP20_Modules_Chl$Row.names <- NULL

## We select only the FC data from the different comparisons
Niben_HSP20_TPM <- Niben_HSP20 %>% select(Protein, 4:51, 61:66)
Niben_HSP20_TPM <- Niben_HSP20_TPM[!duplicated(Niben_HSP20_TPM$Protein),]
Niben_HSP20_TPM <- Niben_HSP20_TPM[order(match(Niben_HSP20_TPM[,1], Niben_HSP20_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_HSP20_FC <- Niben_HSP20 %>% select(Protein, 71, 73:87)
Niben_HSP20_FC <- Niben_HSP20_FC[!duplicated(Niben_HSP20_FC$Protein),]
Niben_HSP20_FC <- Niben_HSP20_FC[order(match(Niben_HSP20_FC[,1], Niben_HSP20_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_HSP20_DEG <- Niben_HSP20 %>% select(Protein, 89, 91:105)
Niben_HSP20_DEG <- Niben_HSP20_DEG[!duplicated(Niben_HSP20_DEG$Protein),]
Niben_HSP20_DEG <- Niben_HSP20_DEG[order(match(Niben_HSP20_DEG[,1], Niben_HSP20_Ann[,2])),]

Niben_HSP20_DEG$pcrtB22h_vs_GFP22h_filtered <- ifelse(is.na(Niben_HSP20_DEG$pcrtB22h_vs_GFP22h_filtered) == TRUE, " ", "*")
Niben_HSP20_DEG$pcrtB25h_vs_GFP25h_filtered <- ifelse(is.na(Niben_HSP20_DEG$pcrtB25h_vs_GFP25h_filtered) == TRUE, " ", "*")
Niben_HSP20_DEG$pcrtBAA_vs_GFPAA_filtered <- ifelse(is.na(Niben_HSP20_DEG$pcrtBAA_vs_GFPAA_filtered) == TRUE, " ", "*")
Niben_HSP20_DEG$pcrtBBB_vs_GFPBB_filtered <- ifelse(is.na(Niben_HSP20_DEG$pcrtBBB_vs_GFPBB_filtered) == TRUE, " ", "*")
Niben_HSP20_DEG$pcrtBYY_vs_GFPYY_filtered <- ifelse(is.na(Niben_HSP20_DEG$pcrtBYY_vs_GFPYY_filtered) == TRUE, " ", "*")
Niben_HSP20_DEG$pcrtB28h_vs_GFP28h_filtered <- ifelse(is.na(Niben_HSP20_DEG$pcrtB28h_vs_GFP28h_filtered) == TRUE, " ", "*")
Niben_HSP20_DEG$pcrtB34h_vs_GFP34h_filtered <- ifelse(is.na(Niben_HSP20_DEG$pcrtB34h_vs_GFP34h_filtered) == TRUE, " ", "*")
Niben_HSP20_DEG$pcrtB37h_vs_GFP37h_filtered <- ifelse(is.na(Niben_HSP20_DEG$pcrtB37h_vs_GFP37h_filtered) == TRUE, " ", "*")
Niben_HSP20_DEG$pcrtBZZ_vs_GFPZZ_filtered <- ifelse(is.na(Niben_HSP20_DEG$pcrtBZZ_vs_GFPZZ_filtered) == TRUE, " ", "*")
Niben_HSP20_DEG$pcrtBCC_vs_GFPCC_filtered <- ifelse(is.na(Niben_HSP20_DEG$pcrtBCC_vs_GFPCC_filtered) == TRUE, " ", "*")
Niben_HSP20_DEG$pcrtB40h_vs_GFP40h_filtered <- ifelse(is.na(Niben_HSP20_DEG$pcrtB40h_vs_GFP40h_filtered) == TRUE, " ", "*")
Niben_HSP20_DEG$pcrtBDD_vs_GFPDD_filtered <- ifelse(is.na(Niben_HSP20_DEG$pcrtBDD_vs_GFPDD_filtered) == TRUE, " ", "*")
Niben_HSP20_DEG$pcrtB46h_vs_GFP46h_filtered <- ifelse(is.na(Niben_HSP20_DEG$pcrtB46h_vs_GFP46h_filtered) == TRUE, " ", "*")
Niben_HSP20_DEG$pcrtB56h_vs_GFP56h_filtered <- ifelse(is.na(Niben_HSP20_DEG$pcrtB56h_vs_GFP56h_filtered) == TRUE, " ", "*")
Niben_HSP20_DEG$pcrtBEE_vs_GFPEE_filtered <- ifelse(is.na(Niben_HSP20_DEG$pcrtBEE_vs_GFPEE_filtered) == TRUE, " ", "*")
Niben_HSP20_DEG$pcrtB96h_vs_GFP96h_filtered <- ifelse(is.na(Niben_HSP20_DEG$pcrtB96h_vs_GFP96h_filtered) == TRUE, " ", "*")
row.names(Niben_HSP20_DEG) <- Niben_HSP20_DEG$Protein
Niben_HSP20_DEG[1] <- NULL

Niben_HSP20_DEG <- Niben_HSP20_DEG %>%
  select(pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

row.names(Niben_HSP20_FC) <- Niben_HSP20_FC$Protein
Niben_HSP20_FC[1] <- NULL

Niben_HSP20_FC <- Niben_HSP20_FC %>%
  select(pcrtB22h_vs_GFP22h_All, pcrtBAA_vs_GFPAA_All, pcrtB25h_vs_GFP25h_All, 
         pcrtBYY_vs_GFPYY_All, pcrtBBB_vs_GFPBB_All, pcrtB28h_vs_GFP28h_All, pcrtB34h_vs_GFP34h_All, pcrtBZZ_vs_GFPZZ_All, 
         pcrtB37h_vs_GFP37h_All, pcrtBCC_vs_GFPCC_All, pcrtB40h_vs_GFP40h_All, pcrtBDD_vs_GFPDD_All, pcrtB46h_vs_GFP46h_All, 
         pcrtBEE_vs_GFPEE_All, pcrtB56h_vs_GFP56h_All, pcrtB96h_vs_GFP96h_All)


min(Niben_HSP20_FC, na.rm = TRUE)
max(Niben_HSP20_FC, na.rm = TRUE)
breaks_HSP20_FC=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Niben_HSP20_FC, # %>% select(1:15), 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_HSP20_FC,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_HSP20_Annotation, annotation_colors=color_HSP20, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h", "96h"),
         cex = 1,
         angle_col=45,
         main = "HSP20",
         annotation_legend = TRUE,
         display_numbers = Niben_HSP20_DEG, # %>% select(1:15),
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(84, 104),
         annotation_row = Niben_HSP20_Modules_Chl,
         annotation_colors = module_Chl_colors
         #gaps_col = c(15)
)


rownames(Niben_HSP20_TPM) <- Niben_HSP20_TPM$Protein
Niben_HSP20_TPM[1] <- NULL

min(Niben_HSP20_TPM, na.rm = TRUE)
max(Niben_HSP20_TPM, na.rm = TRUE)
breaks_Niben_HSP20_TPM=c(seq(0,100, length.out = 100))

Niben_HSP20_TPM_norm <- normalize_df(Niben_HSP20_TPM)
Niben_HSP20_TPM_norm_Cornell <- Niben_HSP20_TPM %>%
  select(1:48) 
Niben_HSP20_TPM_norm_Cornell <- normalize_df(Niben_HSP20_TPM_norm_Cornell)


pheatmap(Niben_HSP20_TPM, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         breaks = breaks_Niben_HSP20_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_HSP20_Annotation, annotation_colors=color_HSP20, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "HSP20",
         annotation_legend = TRUE,
         #display_numbers = Niben_HSP20_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         annotation_row = Niben_HSP20_Modules_Chl,
         annotation_colors = module_Chl_colors,
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(24)#, 48, 51)
)



#####Representing Heatmap for Sly

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Sly_HSP20_Ann <- read.table("Funcional Groups/FigTree/Sly_HSP20.txt",
                            sep = "\t", header = TRUE)

colnames(Sly_HSP20_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Sly_HSP20 <- merge(Sly_HSP20_Ann, Summary_Yazdani_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Sly_HSP20_TPM <- Sly_HSP20 %>% select(Protein, 4:66)
Sly_HSP20_TPM <- Sly_HSP20_TPM[!duplicated(Sly_HSP20_TPM$Protein),]
Sly_HSP20_TPM <- Sly_HSP20_TPM[order(match(Sly_HSP20_TPM[,1], Sly_HSP20_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_HSP20_FC <- Sly_HSP20 %>% select(Protein, 67:71)
Sly_HSP20_FC <- Sly_HSP20_FC[!duplicated(Sly_HSP20_FC$Protein),]
Sly_HSP20_FC <- Sly_HSP20_FC[order(match(Sly_HSP20_FC[,1], Sly_HSP20_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_HSP20_DEG <- Sly_HSP20 %>% select(Protein, 72:76)
Sly_HSP20_DEG <- Sly_HSP20_DEG[!duplicated(Sly_HSP20_DEG$Protein),]
Sly_HSP20_DEG <- Sly_HSP20_DEG[order(match(Sly_HSP20_DEG[,1], Sly_HSP20_Ann[,2])),]

Sly_HSP20_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Sly_HSP20_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Sly_HSP20_DEG$Ripening_Orange_DEG <- ifelse(is.na(Sly_HSP20_DEG$Ripening_Orange_DEG) == TRUE, " ", "*")
Sly_HSP20_DEG$Ripening_Red_DEG <- ifelse(is.na(Sly_HSP20_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
Sly_HSP20_DEG$ORHis_MG_DEG <- ifelse(is.na(Sly_HSP20_DEG$ORHis_MG_DEG) == TRUE, " ", "*")
Sly_HSP20_DEG$ORHis_Breaker_DEG <- ifelse(is.na(Sly_HSP20_DEG$ORHis_Breaker_DEG) == TRUE, " ", "*")
row.names(Sly_HSP20_DEG) <- Sly_HSP20_DEG$Protein
Sly_HSP20_DEG[1] <- NULL

rownames(Sly_HSP20_FC) <- Sly_HSP20_FC$Protein
Sly_HSP20_FC[1] <- NULL

min(Sly_HSP20_FC, na.rm = TRUE)
max(Sly_HSP20_FC, na.rm = TRUE)
breaks_Sly_HSP20=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Sly_HSP20_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Sly_HSP20,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_HSP20_Annotation, annotation_colors=color_HSP20, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "HSP20",
         annotation_legend = TRUE,
         display_numbers = Sly_HSP20_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(39, 48),
         gaps_col = c(3)
)


rownames(Sly_HSP20_TPM) <- Sly_HSP20_TPM$Protein
Sly_HSP20_TPM[1] <- NULL

min(Sly_HSP20_TPM, na.rm = TRUE)
max(Sly_HSP20_TPM, na.rm = TRUE)
breaks_Sly_HSP20_TPM=c(seq(0,100, length.out = 200))

Sly_HSP20_TPM_norm <- normalize_df(Sly_HSP20_TPM)

pheatmap(Sly_HSP20_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Sly_HSP20_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_HSP20_Annotation, annotation_colors=color_HSP20, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "HSP20",
         annotation_legend = TRUE,
         #display_numbers = Sly_HSP20_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(21,42)
)


#####Representing Heatmap for Cann

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Cann_HSP20_Ann <- read.table("Funcional Groups/FigTree/Cann_HSP20.txt",
                             sep = "\t", header = TRUE)

colnames(Cann_HSP20_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Cann_HSP20 <- merge(Cann_HSP20_Ann, Summary_Pepper_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Cann_HSP20_TPM <- Cann_HSP20 %>% select(Protein, 4:17)
Cann_HSP20_TPM <- Cann_HSP20_TPM[!duplicated(Cann_HSP20_TPM$Protein),]
Cann_HSP20_TPM <- Cann_HSP20_TPM[order(match(Cann_HSP20_TPM[,1], Cann_HSP20_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_HSP20_FC <- Cann_HSP20 %>% select(Protein, 18:19)
Cann_HSP20_FC <- Cann_HSP20_FC[!duplicated(Cann_HSP20_FC$Protein),]
Cann_HSP20_FC <- Cann_HSP20_FC[order(match(Cann_HSP20_FC[,1], Cann_HSP20_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_HSP20_DEG <- Cann_HSP20 %>% select(Protein, 20:21)
Cann_HSP20_DEG <- Cann_HSP20_DEG[!duplicated(Cann_HSP20_DEG$Protein),]
Cann_HSP20_DEG <- Cann_HSP20_DEG[order(match(Cann_HSP20_DEG[,1], Cann_HSP20_Ann[,2])),]

Cann_HSP20_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Cann_HSP20_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Cann_HSP20_DEG$Ripening_Red_DEG <- ifelse(is.na(Cann_HSP20_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
row.names(Cann_HSP20_DEG) <- Cann_HSP20_DEG$Protein
Cann_HSP20_DEG[1] <- NULL

rownames(Cann_HSP20_FC) <- Cann_HSP20_FC$Protein
Cann_HSP20_FC[1] <- NULL

min(Cann_HSP20_FC, na.rm = TRUE)
max(Cann_HSP20_FC, na.rm = TRUE)
breaks_Cann_HSP20=c(seq(-4.5,0, length.out = 100), seq(0.01,6,length.out = 100))

pheatmap(Cann_HSP20_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Cann_HSP20,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_HSP20_Annotation, annotation_colors=color_HSP20, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Red"),
         cex = 1,
         angle_col=45,
         main = "HSP20",
         annotation_legend = TRUE,
         display_numbers = Cann_HSP20_DEG,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)


rownames(Cann_HSP20_TPM) <- Cann_HSP20_TPM$Protein
Cann_HSP20_TPM[1] <- NULL

min(Cann_HSP20_TPM, na.rm = TRUE)
max(Cann_HSP20_TPM, na.rm = TRUE)
breaks_Cann_HSP20_TPM=c(seq(0,100, length.out = 200))

Cann_HSP20_TPM_norm <- normalize_df(Cann_HSP20_TPM)

pheatmap(Cann_HSP20_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Cann_HSP20_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_HSP20_Annotation, annotation_colors=color_HSP20, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "HSP20",
         annotation_legend = TRUE,
         #display_numbers = Cann_HSP20_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(4,9)
)


## Comparativa 96h_SlyYaz_Ath

Ath_HSP20_Ann <- read.table("Funcional Groups/Ath_HSP20.txt", header = TRUE, sep = "\t")

Ath_HSP20_FC <- merge(Ath_HSP20_Ann, Summary_Ath_FC, by.x=1, by.y = 0, all.x = TRUE) %>%
  dplyr::select(Name, log2FoldChange)
Ath_HSP20_FC <- Ath_HSP20_FC[order(match(Ath_HSP20_FC[,1], Ath_HSP20_Ann[,2])),]

Ath_HSP20_DEG <- merge(Ath_HSP20_Ann, Summary_Ath_DEGs, by.x=1, by.y = 1, all.x = TRUE) %>%
  dplyr::select(Name, log2FC)
Ath_HSP20_DEG <- Ath_HSP20_DEG[order(match(Ath_HSP20_DEG[,1], Ath_HSP20_Ann[,2])),]

Ath_HSP20_DEG$log2FC <- ifelse(is.na(Ath_HSP20_DEG$log2FC) == TRUE, " ", "*")
row.names(Ath_HSP20_DEG) <- Ath_HSP20_DEG$Name
Ath_HSP20_DEG[1] <- NULL

rownames(Ath_HSP20_FC) <- Ath_HSP20_FC$Name
Ath_HSP20_FC[1] <- NULL

min(Ath_HSP20_FC, na.rm = TRUE)
max(Ath_HSP20_FC, na.rm = TRUE)
breaks_Ath_HSP20=c(seq(-4.5,0, length.out = 100), seq(0.01,4.5,length.out = 100))

pheatmap(Ath_HSP20_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Stndr,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_HSP20_Annotation, annotation_colors=color_HSP20, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = "Senescence",
         cex = 1,
         angle_col=45,
         main = "HSP20",
         annotation_legend = TRUE,
         display_numbers = Ath_HSP20_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         gaps_row = c(3, 4, 5, 6, 7, 8, 9, 10)
         #gaps_col = c(2,5)
)

pheatmap(Niben_HSP20_FC %>% select(16), 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         #breaks = breaks_HSP20_FC,
         breaks = breaks_Stndr,
         #legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         #legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_HSP20_Annotation, annotation_colors=color_HSP20, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = "96h",
         cex = 1,
         angle_col=45,
         main = "HSP20",
         annotation_legend = TRUE,
         display_numbers = Niben_HSP20_DEG %>% select(16),
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(84, 104),
         annotation_row = Niben_HSP20_Modules_Chl,
         annotation_colors = module_Chl_colors_2,
         gaps_row = c(3, 5, 9, 12, 15, 17)
)


pheatmap(Sly_HSP20_FC %>% select(2,3), 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Stndr,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         #legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_HSP20_Annotation, annotation_colors=color_HSP20, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Orange", "Ripening_Red"),
         cex = 1,
         angle_col=45,
         main = "HSP20",
         annotation_legend = TRUE,
         display_numbers = Sly_HSP20_DEG %>% select(2,3),
         fontsize_number = 15,
         na_col = "darkgrey",
         gaps_row = c(1, 3,  4, 7, 8)
)


#########################################################################################################################################
####################################  Cpn60  #############################################################################################
#########################################################################################################################################

Cpn60_Ath <- read.delim("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan/Funcional Groups/Ath_Cpn60.txt")
Cpn60_Ath <- merge(Cpn60_Ath, Ath_Gene_Models, by.x="GeneID", by.y=1)
Cpn60_Ath_fasta <- Ath11_fasta[c(which(names(Ath11_fasta) %in% Cpn60_Ath$V2))]
Cpn60_Ath_2 <- Cpn60_Ath %>% select(V2, Name)
write.fasta(Cpn60_Ath_fasta, as.string=TRUE, names= names(Cpn60_Ath_fasta),
            file.out = "Funcional Groups/Ath_Cpn60.fasta")
rename.fasta(infile = "Funcional Groups/Ath_Cpn60.fasta", Cpn60_Ath_2, outfile = "Funcional Groups/Ath_Cpn60_2.fasta")

Cpn60_Niben_Candidates <- read.table("Funcional Groups/Niben_Candidates_Cpn60.txt")
Cpn60_Niben_Candidates_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Cpn60_Niben_Candidates$V1))]
write.fasta(Cpn60_Niben_Candidates_fasta, as.string=TRUE, names= names(Cpn60_Niben_Candidates_fasta),
            file.out = "Funcional Groups/Cpn60_Niben_Candidates_35.fasta")

Cpn60_Sly_Candidates <- read.table("Funcional Groups/Sly_Candidates_Cpn60.txt")
Cpn60_Sly_Candidates_fasta <- Tomato_4.0_fasta[c(which(names(Tomato_4.0_fasta) %in% Cpn60_Sly_Candidates$V1))]
write.fasta(Cpn60_Sly_Candidates_fasta, as.string=TRUE, names= names(Cpn60_Sly_Candidates_fasta),
            file.out = "Funcional Groups/Cpn60_Sly_Candidates_35.fasta")

Cpn60_Cann_Candidates <- read.table("Funcional Groups/Cann_Candidates_Cpn60.txt")
Cpn60_Cann_Candidates_fasta <- Cannuum_ASM51225v2_fasta[c(which(names(Cannuum_ASM51225v2_fasta) %in% Cpn60_Cann_Candidates$V1))]
write.fasta(Cpn60_Cann_Candidates_fasta, as.string=TRUE, names= names(Cpn60_Cann_Candidates_fasta),
            file.out = "Funcional Groups/Cpn60_Cann_Candidates_35.fasta")

Niben_Cpn60_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_Cpn60_Ann$Name))]
write.fasta(Niben_Cpn60_fasta, as.string=TRUE, names= names(Niben_Cpn60_fasta),
            file.out = "Funcional Groups/Niben_Cpn60.fasta")
Niben_Cpn60_Ann_2 <- Niben_Cpn60_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_Cpn60.fasta", Niben_Cpn60_Ann, outfile = "Funcional Groups/Niben_Cpn60_2.fasta")

save_heatmaps("Cpn60", height = 10)


## Representing Heatmap

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Niben_Cpn60_Ann <- read.table("Funcional Groups/FigTree/Niben_Cpn60.txt",
                              sep = "\t", header = TRUE)

colnames(Niben_Cpn60_Ann) <- c("Name", "Protein")

Niben_Cpn60_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_Cpn60_Ann$Name))]
write.fasta(Niben_Cpn60_fasta, as.string=TRUE, names= names(Niben_Cpn60_fasta),
            file.out = "Funcional Groups/Niben_Cpn60.fasta")
Niben_Cpn60_Ann_2 <- Niben_Cpn60_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_Cpn60.fasta", Niben_Cpn60_Ann, outfile = "Funcional Groups/Niben_Cpn60_2.fasta")

Niben_Cpn60_Chloro <- read.delim("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan/Funcional Groups/Niben_Cpn60_2_summary.targetp2", 
                                 header=FALSE, comment.char="#")
colnames(Niben_Cpn60_Chloro) <- c("ID",	"Prediction",	"noTP",	"SP",	"mTP",	"cTP",	"luTP",	"CS-Position")
Niben_Cpn60_Chloro$Prediction <- factor(Niben_Cpn60_Chloro$Prediction, levels = c("cTP","mTP","noTP"))
Niben_Cpn60_Chloro <- merge(Niben_Cpn60_Ann, Niben_Cpn60_Chloro, by.x = "Protein", by.y = "ID")

Niben_Cpn60_Chloro <- Niben_Cpn60_Chloro[order(match(Niben_Cpn60_Chloro$Protein, Niben_Cpn60_Ann$Protein)),]
rownames(Niben_Cpn60_Chloro) <- Niben_Cpn60_Chloro$Protein
Niben_Cpn60_Chloro <- Niben_Cpn60_Chloro %>% select(Prediction)


## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_Cpn60 <- merge(Niben_Cpn60_Ann, TPM_All_FC_Annotated, by="Name")

## We create the annotation for Modules

Niben_Cpn60_Modules <- which_Module_pwd30(Niben_Cpn60_Ann)

## We merge this annotation of Modules with the annotation of Chloroplast TP prediction

Niben_Cpn60_Modules_Chl <- merge(Niben_Cpn60_Modules, Niben_Cpn60_Chloro, by = 0)
Niben_Cpn60_Modules_Chl <- Niben_Cpn60_Modules_Chl[order(match(Niben_Cpn60_Modules_Chl$Row.names, Niben_Cpn60_Ann$Protein)),]
rownames(Niben_Cpn60_Modules_Chl) <- Niben_Cpn60_Modules_Chl$Row.names
Niben_Cpn60_Modules_Chl$Row.names <- NULL

## We select only the FC data from the different comparisons
Niben_Cpn60_TPM <- Niben_Cpn60 %>% select(Protein, 4:51, 61:66)
Niben_Cpn60_TPM <- Niben_Cpn60_TPM[!duplicated(Niben_Cpn60_TPM$Protein),]
Niben_Cpn60_TPM <- Niben_Cpn60_TPM[order(match(Niben_Cpn60_TPM[,1], Niben_Cpn60_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_Cpn60_FC <- Niben_Cpn60 %>% select(Protein, 71, 73:87)
Niben_Cpn60_FC <- Niben_Cpn60_FC[!duplicated(Niben_Cpn60_FC$Protein),]
Niben_Cpn60_FC <- Niben_Cpn60_FC[order(match(Niben_Cpn60_FC[,1], Niben_Cpn60_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_Cpn60_DEG <- Niben_Cpn60 %>% select(Protein, 89, 91:105)
Niben_Cpn60_DEG <- Niben_Cpn60_DEG[!duplicated(Niben_Cpn60_DEG$Protein),]
Niben_Cpn60_DEG <- Niben_Cpn60_DEG[order(match(Niben_Cpn60_DEG[,1], Niben_Cpn60_Ann[,2])),]

Niben_Cpn60_DEG$pcrtB22h_vs_GFP22h_filtered <- ifelse(is.na(Niben_Cpn60_DEG$pcrtB22h_vs_GFP22h_filtered) == TRUE, " ", "*")
Niben_Cpn60_DEG$pcrtB25h_vs_GFP25h_filtered <- ifelse(is.na(Niben_Cpn60_DEG$pcrtB25h_vs_GFP25h_filtered) == TRUE, " ", "*")
Niben_Cpn60_DEG$pcrtBAA_vs_GFPAA_filtered <- ifelse(is.na(Niben_Cpn60_DEG$pcrtBAA_vs_GFPAA_filtered) == TRUE, " ", "*")
Niben_Cpn60_DEG$pcrtBBB_vs_GFPBB_filtered <- ifelse(is.na(Niben_Cpn60_DEG$pcrtBBB_vs_GFPBB_filtered) == TRUE, " ", "*")
Niben_Cpn60_DEG$pcrtBYY_vs_GFPYY_filtered <- ifelse(is.na(Niben_Cpn60_DEG$pcrtBYY_vs_GFPYY_filtered) == TRUE, " ", "*")
Niben_Cpn60_DEG$pcrtB28h_vs_GFP28h_filtered <- ifelse(is.na(Niben_Cpn60_DEG$pcrtB28h_vs_GFP28h_filtered) == TRUE, " ", "*")
Niben_Cpn60_DEG$pcrtB34h_vs_GFP34h_filtered <- ifelse(is.na(Niben_Cpn60_DEG$pcrtB34h_vs_GFP34h_filtered) == TRUE, " ", "*")
Niben_Cpn60_DEG$pcrtB37h_vs_GFP37h_filtered <- ifelse(is.na(Niben_Cpn60_DEG$pcrtB37h_vs_GFP37h_filtered) == TRUE, " ", "*")
Niben_Cpn60_DEG$pcrtBZZ_vs_GFPZZ_filtered <- ifelse(is.na(Niben_Cpn60_DEG$pcrtBZZ_vs_GFPZZ_filtered) == TRUE, " ", "*")
Niben_Cpn60_DEG$pcrtBCC_vs_GFPCC_filtered <- ifelse(is.na(Niben_Cpn60_DEG$pcrtBCC_vs_GFPCC_filtered) == TRUE, " ", "*")
Niben_Cpn60_DEG$pcrtB40h_vs_GFP40h_filtered <- ifelse(is.na(Niben_Cpn60_DEG$pcrtB40h_vs_GFP40h_filtered) == TRUE, " ", "*")
Niben_Cpn60_DEG$pcrtBDD_vs_GFPDD_filtered <- ifelse(is.na(Niben_Cpn60_DEG$pcrtBDD_vs_GFPDD_filtered) == TRUE, " ", "*")
Niben_Cpn60_DEG$pcrtB46h_vs_GFP46h_filtered <- ifelse(is.na(Niben_Cpn60_DEG$pcrtB46h_vs_GFP46h_filtered) == TRUE, " ", "*")
Niben_Cpn60_DEG$pcrtB56h_vs_GFP56h_filtered <- ifelse(is.na(Niben_Cpn60_DEG$pcrtB56h_vs_GFP56h_filtered) == TRUE, " ", "*")
Niben_Cpn60_DEG$pcrtBEE_vs_GFPEE_filtered <- ifelse(is.na(Niben_Cpn60_DEG$pcrtBEE_vs_GFPEE_filtered) == TRUE, " ", "*")
Niben_Cpn60_DEG$pcrtB96h_vs_GFP96h_filtered <- ifelse(is.na(Niben_Cpn60_DEG$pcrtB96h_vs_GFP96h_filtered) == TRUE, " ", "*")
row.names(Niben_Cpn60_DEG) <- Niben_Cpn60_DEG$Protein
Niben_Cpn60_DEG[1] <- NULL

Niben_Cpn60_DEG <- Niben_Cpn60_DEG %>%
  select(pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

row.names(Niben_Cpn60_FC) <- Niben_Cpn60_FC$Protein
Niben_Cpn60_FC[1] <- NULL

Niben_Cpn60_FC <- Niben_Cpn60_FC %>%
  select(pcrtB22h_vs_GFP22h_All, pcrtBAA_vs_GFPAA_All, pcrtB25h_vs_GFP25h_All, 
         pcrtBYY_vs_GFPYY_All, pcrtBBB_vs_GFPBB_All, pcrtB28h_vs_GFP28h_All, pcrtB34h_vs_GFP34h_All, pcrtBZZ_vs_GFPZZ_All, 
         pcrtB37h_vs_GFP37h_All, pcrtBCC_vs_GFPCC_All, pcrtB40h_vs_GFP40h_All, pcrtBDD_vs_GFPDD_All, pcrtB46h_vs_GFP46h_All, 
         pcrtBEE_vs_GFPEE_All, pcrtB56h_vs_GFP56h_All, pcrtB96h_vs_GFP96h_All)


min(Niben_Cpn60_FC, na.rm = TRUE)
max(Niben_Cpn60_FC, na.rm = TRUE)
breaks_Cpn60_FC=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Niben_Cpn60_FC, # %>% select(1:15), 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Cpn60_FC,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_Cpn60_Annotation, annotation_colors=color_Cpn60, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h", "96h"),
         cex = 1,
         angle_col=45,
         main = "Cpn60",
         annotation_legend = TRUE,
         display_numbers = Niben_Cpn60_DEG, # %>% select(1:15),
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(84, 104),
         annotation_row = Niben_Cpn60_Modules_Chl,
         annotation_colors = module_Chl_colors
         #gaps_col = c(15)
)


rownames(Niben_Cpn60_TPM) <- Niben_Cpn60_TPM$Protein
Niben_Cpn60_TPM[1] <- NULL

min(Niben_Cpn60_TPM, na.rm = TRUE)
max(Niben_Cpn60_TPM, na.rm = TRUE)
breaks_Niben_Cpn60_TPM=c(seq(0,100, length.out = 100))

Niben_Cpn60_TPM_norm <- normalize_df(Niben_Cpn60_TPM)
Niben_Cpn60_TPM_norm_Cornell <- Niben_Cpn60_TPM %>%
  select(1:48) 
Niben_Cpn60_TPM_norm_Cornell <- normalize_df(Niben_Cpn60_TPM_norm_Cornell)


pheatmap(Niben_Cpn60_TPM, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         breaks = breaks_Niben_Cpn60_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_Cpn60_Annotation, annotation_colors=color_Cpn60, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Cpn60",
         annotation_legend = TRUE,
         #display_numbers = Niben_Cpn60_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         annotation_row = Niben_Cpn60_Modules_Chl,
         annotation_colors = module_Chl_colors,
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(24)#, 48, 51)
)



#####Representing Heatmap for Sly

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Sly_Cpn60_Ann <- read.table("Funcional Groups/FigTree/Sly_Cpn60.txt",
                            sep = "\t", header = TRUE)

colnames(Sly_Cpn60_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Sly_Cpn60 <- merge(Sly_Cpn60_Ann, Summary_Yazdani_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Sly_Cpn60_TPM <- Sly_Cpn60 %>% select(Protein, 4:66)
Sly_Cpn60_TPM <- Sly_Cpn60_TPM[!duplicated(Sly_Cpn60_TPM$Protein),]
Sly_Cpn60_TPM <- Sly_Cpn60_TPM[order(match(Sly_Cpn60_TPM[,1], Sly_Cpn60_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_Cpn60_FC <- Sly_Cpn60 %>% select(Protein, 67:71)
Sly_Cpn60_FC <- Sly_Cpn60_FC[!duplicated(Sly_Cpn60_FC$Protein),]
Sly_Cpn60_FC <- Sly_Cpn60_FC[order(match(Sly_Cpn60_FC[,1], Sly_Cpn60_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_Cpn60_DEG <- Sly_Cpn60 %>% select(Protein, 72:76)
Sly_Cpn60_DEG <- Sly_Cpn60_DEG[!duplicated(Sly_Cpn60_DEG$Protein),]
Sly_Cpn60_DEG <- Sly_Cpn60_DEG[order(match(Sly_Cpn60_DEG[,1], Sly_Cpn60_Ann[,2])),]

Sly_Cpn60_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Sly_Cpn60_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Sly_Cpn60_DEG$Ripening_Orange_DEG <- ifelse(is.na(Sly_Cpn60_DEG$Ripening_Orange_DEG) == TRUE, " ", "*")
Sly_Cpn60_DEG$Ripening_Red_DEG <- ifelse(is.na(Sly_Cpn60_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
Sly_Cpn60_DEG$ORHis_MG_DEG <- ifelse(is.na(Sly_Cpn60_DEG$ORHis_MG_DEG) == TRUE, " ", "*")
Sly_Cpn60_DEG$ORHis_Breaker_DEG <- ifelse(is.na(Sly_Cpn60_DEG$ORHis_Breaker_DEG) == TRUE, " ", "*")
row.names(Sly_Cpn60_DEG) <- Sly_Cpn60_DEG$Protein
Sly_Cpn60_DEG[1] <- NULL

rownames(Sly_Cpn60_FC) <- Sly_Cpn60_FC$Protein
Sly_Cpn60_FC[1] <- NULL

min(Sly_Cpn60_FC, na.rm = TRUE)
max(Sly_Cpn60_FC, na.rm = TRUE)
breaks_Sly_Cpn60=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Sly_Cpn60_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Sly_Cpn60,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_Cpn60_Annotation, annotation_colors=color_Cpn60, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Cpn60",
         annotation_legend = TRUE,
         display_numbers = Sly_Cpn60_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(39, 48),
         gaps_col = c(3)
)


rownames(Sly_Cpn60_TPM) <- Sly_Cpn60_TPM$Protein
Sly_Cpn60_TPM[1] <- NULL

min(Sly_Cpn60_TPM, na.rm = TRUE)
max(Sly_Cpn60_TPM, na.rm = TRUE)
breaks_Sly_Cpn60_TPM=c(seq(0,100, length.out = 200))

Sly_Cpn60_TPM_norm <- normalize_df(Sly_Cpn60_TPM)

pheatmap(Sly_Cpn60_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Sly_Cpn60_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_Cpn60_Annotation, annotation_colors=color_Cpn60, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Cpn60",
         annotation_legend = TRUE,
         #display_numbers = Sly_Cpn60_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(21,42)
)


#####Representing Heatmap for Cann

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Cann_Cpn60_Ann <- read.table("Funcional Groups/FigTree/Cann_Cpn60.txt",
                             sep = "\t", header = TRUE)

colnames(Cann_Cpn60_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Cann_Cpn60 <- merge(Cann_Cpn60_Ann, Summary_Pepper_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Cann_Cpn60_TPM <- Cann_Cpn60 %>% select(Protein, 4:17)
Cann_Cpn60_TPM <- Cann_Cpn60_TPM[!duplicated(Cann_Cpn60_TPM$Protein),]
Cann_Cpn60_TPM <- Cann_Cpn60_TPM[order(match(Cann_Cpn60_TPM[,1], Cann_Cpn60_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_Cpn60_FC <- Cann_Cpn60 %>% select(Protein, 18:19)
Cann_Cpn60_FC <- Cann_Cpn60_FC[!duplicated(Cann_Cpn60_FC$Protein),]
Cann_Cpn60_FC <- Cann_Cpn60_FC[order(match(Cann_Cpn60_FC[,1], Cann_Cpn60_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_Cpn60_DEG <- Cann_Cpn60 %>% select(Protein, 20:21)
Cann_Cpn60_DEG <- Cann_Cpn60_DEG[!duplicated(Cann_Cpn60_DEG$Protein),]
Cann_Cpn60_DEG <- Cann_Cpn60_DEG[order(match(Cann_Cpn60_DEG[,1], Cann_Cpn60_Ann[,2])),]

Cann_Cpn60_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Cann_Cpn60_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Cann_Cpn60_DEG$Ripening_Red_DEG <- ifelse(is.na(Cann_Cpn60_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
row.names(Cann_Cpn60_DEG) <- Cann_Cpn60_DEG$Protein
Cann_Cpn60_DEG[1] <- NULL

rownames(Cann_Cpn60_FC) <- Cann_Cpn60_FC$Protein
Cann_Cpn60_FC[1] <- NULL

min(Cann_Cpn60_FC, na.rm = TRUE)
max(Cann_Cpn60_FC, na.rm = TRUE)
breaks_Cann_Cpn60=c(seq(-4.5,0, length.out = 100), seq(0.01,6,length.out = 100))

pheatmap(Cann_Cpn60_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Cann_Cpn60,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_Cpn60_Annotation, annotation_colors=color_Cpn60, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Red"),
         cex = 1,
         angle_col=45,
         main = "Cpn60",
         annotation_legend = TRUE,
         display_numbers = Cann_Cpn60_DEG,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)


rownames(Cann_Cpn60_TPM) <- Cann_Cpn60_TPM$Protein
Cann_Cpn60_TPM[1] <- NULL

min(Cann_Cpn60_TPM, na.rm = TRUE)
max(Cann_Cpn60_TPM, na.rm = TRUE)
breaks_Cann_Cpn60_TPM=c(seq(0,100, length.out = 200))

Cann_Cpn60_TPM_norm <- normalize_df(Cann_Cpn60_TPM)

pheatmap(Cann_Cpn60_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Cann_Cpn60_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_Cpn60_Annotation, annotation_colors=color_Cpn60, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Cpn60",
         annotation_legend = TRUE,
         #display_numbers = Cann_Cpn60_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(4,9)
)

#########################################################################################################################################
####################################  Cpn10  #############################################################################################
#########################################################################################################################################

Cpn10_Ath <- read.delim("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan/Funcional Groups/Ath_Cpn10.txt")
Cpn10_Ath <- merge(Cpn10_Ath, Ath_Gene_Models, by.x="GeneID", by.y=1)
Cpn10_Ath_fasta <- Ath11_fasta[c(which(names(Ath11_fasta) %in% Cpn10_Ath$V2))]
Cpn10_Ath_2 <- Cpn10_Ath %>% select(V2, Name)
write.fasta(Cpn10_Ath_fasta, as.string=TRUE, names= names(Cpn10_Ath_fasta),
            file.out = "Funcional Groups/Ath_Cpn10.fasta")
rename.fasta(infile = "Funcional Groups/Ath_Cpn10.fasta", Cpn10_Ath_2, outfile = "Funcional Groups/Ath_Cpn10_2.fasta")

Cpn10_Niben_Candidates <- read.table("Funcional Groups/Niben_Candidates_Cpn10.txt")
Cpn10_Niben_Candidates_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Cpn10_Niben_Candidates$V1))]
write.fasta(Cpn10_Niben_Candidates_fasta, as.string=TRUE, names= names(Cpn10_Niben_Candidates_fasta),
            file.out = "Funcional Groups/Cpn10_Niben_Candidates_35.fasta")

Cpn10_Sly_Candidates <- read.table("Funcional Groups/Sly_Candidates_Cpn10.txt")
Cpn10_Sly_Candidates_fasta <- Tomato_4.0_fasta[c(which(names(Tomato_4.0_fasta) %in% Cpn10_Sly_Candidates$V1))]
write.fasta(Cpn10_Sly_Candidates_fasta, as.string=TRUE, names= names(Cpn10_Sly_Candidates_fasta),
            file.out = "Funcional Groups/Cpn10_Sly_Candidates_35.fasta")

Cpn10_Cann_Candidates <- read.table("Funcional Groups/Cann_Candidates_Cpn10.txt")
Cpn10_Cann_Candidates_fasta <- Cannuum_ASM51225v2_fasta[c(which(names(Cannuum_ASM51225v2_fasta) %in% Cpn10_Cann_Candidates$V1))]
write.fasta(Cpn10_Cann_Candidates_fasta, as.string=TRUE, names= names(Cpn10_Cann_Candidates_fasta),
            file.out = "Funcional Groups/Cpn10_Cann_Candidates_35.fasta")

Niben_Cpn10_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_Cpn10_Ann$Name))]
write.fasta(Niben_Cpn10_fasta, as.string=TRUE, names= names(Niben_Cpn10_fasta),
            file.out = "Funcional Groups/Niben_Cpn10.fasta")
Niben_Cpn10_Ann_2 <- Niben_Cpn10_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_Cpn10.fasta", Niben_Cpn10_Ann, outfile = "Funcional Groups/Niben_Cpn10_2.fasta")

save_heatmaps("Cpn10", height = 4)

## Representing Heatmap

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Niben_Cpn10_Ann <- read.table("Funcional Groups/FigTree/Niben_Cpn10.txt",
                              sep = "\t", header = TRUE)

colnames(Niben_Cpn10_Ann) <- c("Name", "Protein")

Niben_Cpn10_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_Cpn10_Ann$Name))]
write.fasta(Niben_Cpn10_fasta, as.string=TRUE, names= names(Niben_Cpn10_fasta),
            file.out = "Funcional Groups/Niben_Cpn10.fasta")
Niben_Cpn10_Ann_2 <- Niben_Cpn10_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_Cpn10.fasta", Niben_Cpn10_Ann, outfile = "Funcional Groups/Niben_Cpn10_2.fasta")

Niben_Cpn10_Chloro <- read.delim("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan/Funcional Groups/Niben_Cpn10_2_summary.targetp2", 
                                 header=FALSE, comment.char="#")
colnames(Niben_Cpn10_Chloro) <- c("ID",	"Prediction",	"noTP",	"SP",	"mTP",	"cTP",	"luTP",	"CS-Position")
Niben_Cpn10_Chloro$Prediction <- factor(Niben_Cpn10_Chloro$Prediction, levels = c("cTP","mTP","noTP"))
Niben_Cpn10_Chloro <- merge(Niben_Cpn10_Ann, Niben_Cpn10_Chloro, by.x = "Protein", by.y = "ID")

Niben_Cpn10_Chloro <- Niben_Cpn10_Chloro[order(match(Niben_Cpn10_Chloro$Protein, Niben_Cpn10_Ann$Protein)),]
rownames(Niben_Cpn10_Chloro) <- Niben_Cpn10_Chloro$Protein
Niben_Cpn10_Chloro <- Niben_Cpn10_Chloro %>% select(Prediction)


## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_Cpn10 <- merge(Niben_Cpn10_Ann, TPM_All_FC_Annotated, by="Name")

## We create the annotation for Modules

Niben_Cpn10_Modules <- which_Module_pwd30(Niben_Cpn10_Ann)

## We merge this annotation of Modules with the annotation of Chloroplast TP prediction

Niben_Cpn10_Modules_Chl <- merge(Niben_Cpn10_Modules, Niben_Cpn10_Chloro, by = 0)
Niben_Cpn10_Modules_Chl <- Niben_Cpn10_Modules_Chl[order(match(Niben_Cpn10_Modules_Chl$Row.names, Niben_Cpn10_Ann$Protein)),]
rownames(Niben_Cpn10_Modules_Chl) <- Niben_Cpn10_Modules_Chl$Row.names
Niben_Cpn10_Modules_Chl$Row.names <- NULL

## We select only the FC data from the different comparisons
Niben_Cpn10_TPM <- Niben_Cpn10 %>% select(Protein, 4:51, 61:66)
Niben_Cpn10_TPM <- Niben_Cpn10_TPM[!duplicated(Niben_Cpn10_TPM$Protein),]
Niben_Cpn10_TPM <- Niben_Cpn10_TPM[order(match(Niben_Cpn10_TPM[,1], Niben_Cpn10_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_Cpn10_FC <- Niben_Cpn10 %>% select(Protein, 71, 73:87)
Niben_Cpn10_FC <- Niben_Cpn10_FC[!duplicated(Niben_Cpn10_FC$Protein),]
Niben_Cpn10_FC <- Niben_Cpn10_FC[order(match(Niben_Cpn10_FC[,1], Niben_Cpn10_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_Cpn10_DEG <- Niben_Cpn10 %>% select(Protein, 89, 91:105)
Niben_Cpn10_DEG <- Niben_Cpn10_DEG[!duplicated(Niben_Cpn10_DEG$Protein),]
Niben_Cpn10_DEG <- Niben_Cpn10_DEG[order(match(Niben_Cpn10_DEG[,1], Niben_Cpn10_Ann[,2])),]

Niben_Cpn10_DEG$pcrtB22h_vs_GFP22h_filtered <- ifelse(is.na(Niben_Cpn10_DEG$pcrtB22h_vs_GFP22h_filtered) == TRUE, " ", "*")
Niben_Cpn10_DEG$pcrtB25h_vs_GFP25h_filtered <- ifelse(is.na(Niben_Cpn10_DEG$pcrtB25h_vs_GFP25h_filtered) == TRUE, " ", "*")
Niben_Cpn10_DEG$pcrtBAA_vs_GFPAA_filtered <- ifelse(is.na(Niben_Cpn10_DEG$pcrtBAA_vs_GFPAA_filtered) == TRUE, " ", "*")
Niben_Cpn10_DEG$pcrtBBB_vs_GFPBB_filtered <- ifelse(is.na(Niben_Cpn10_DEG$pcrtBBB_vs_GFPBB_filtered) == TRUE, " ", "*")
Niben_Cpn10_DEG$pcrtBYY_vs_GFPYY_filtered <- ifelse(is.na(Niben_Cpn10_DEG$pcrtBYY_vs_GFPYY_filtered) == TRUE, " ", "*")
Niben_Cpn10_DEG$pcrtB28h_vs_GFP28h_filtered <- ifelse(is.na(Niben_Cpn10_DEG$pcrtB28h_vs_GFP28h_filtered) == TRUE, " ", "*")
Niben_Cpn10_DEG$pcrtB34h_vs_GFP34h_filtered <- ifelse(is.na(Niben_Cpn10_DEG$pcrtB34h_vs_GFP34h_filtered) == TRUE, " ", "*")
Niben_Cpn10_DEG$pcrtB37h_vs_GFP37h_filtered <- ifelse(is.na(Niben_Cpn10_DEG$pcrtB37h_vs_GFP37h_filtered) == TRUE, " ", "*")
Niben_Cpn10_DEG$pcrtBZZ_vs_GFPZZ_filtered <- ifelse(is.na(Niben_Cpn10_DEG$pcrtBZZ_vs_GFPZZ_filtered) == TRUE, " ", "*")
Niben_Cpn10_DEG$pcrtBCC_vs_GFPCC_filtered <- ifelse(is.na(Niben_Cpn10_DEG$pcrtBCC_vs_GFPCC_filtered) == TRUE, " ", "*")
Niben_Cpn10_DEG$pcrtB40h_vs_GFP40h_filtered <- ifelse(is.na(Niben_Cpn10_DEG$pcrtB40h_vs_GFP40h_filtered) == TRUE, " ", "*")
Niben_Cpn10_DEG$pcrtBDD_vs_GFPDD_filtered <- ifelse(is.na(Niben_Cpn10_DEG$pcrtBDD_vs_GFPDD_filtered) == TRUE, " ", "*")
Niben_Cpn10_DEG$pcrtB46h_vs_GFP46h_filtered <- ifelse(is.na(Niben_Cpn10_DEG$pcrtB46h_vs_GFP46h_filtered) == TRUE, " ", "*")
Niben_Cpn10_DEG$pcrtB56h_vs_GFP56h_filtered <- ifelse(is.na(Niben_Cpn10_DEG$pcrtB56h_vs_GFP56h_filtered) == TRUE, " ", "*")
Niben_Cpn10_DEG$pcrtBEE_vs_GFPEE_filtered <- ifelse(is.na(Niben_Cpn10_DEG$pcrtBEE_vs_GFPEE_filtered) == TRUE, " ", "*")
Niben_Cpn10_DEG$pcrtB96h_vs_GFP96h_filtered <- ifelse(is.na(Niben_Cpn10_DEG$pcrtB96h_vs_GFP96h_filtered) == TRUE, " ", "*")
row.names(Niben_Cpn10_DEG) <- Niben_Cpn10_DEG$Protein
Niben_Cpn10_DEG[1] <- NULL

Niben_Cpn10_DEG <- Niben_Cpn10_DEG %>%
  select(pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

row.names(Niben_Cpn10_FC) <- Niben_Cpn10_FC$Protein
Niben_Cpn10_FC[1] <- NULL

Niben_Cpn10_FC <- Niben_Cpn10_FC %>%
  select(pcrtB22h_vs_GFP22h_All, pcrtBAA_vs_GFPAA_All, pcrtB25h_vs_GFP25h_All, 
         pcrtBYY_vs_GFPYY_All, pcrtBBB_vs_GFPBB_All, pcrtB28h_vs_GFP28h_All, pcrtB34h_vs_GFP34h_All, pcrtBZZ_vs_GFPZZ_All, 
         pcrtB37h_vs_GFP37h_All, pcrtBCC_vs_GFPCC_All, pcrtB40h_vs_GFP40h_All, pcrtBDD_vs_GFPDD_All, pcrtB46h_vs_GFP46h_All, 
         pcrtBEE_vs_GFPEE_All, pcrtB56h_vs_GFP56h_All, pcrtB96h_vs_GFP96h_All)


min(Niben_Cpn10_FC, na.rm = TRUE)
max(Niben_Cpn10_FC, na.rm = TRUE)
breaks_Cpn10_FC=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Niben_Cpn10_FC, # %>% select(1:15), 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Cpn10_FC,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_Cpn10_Annotation, annotation_colors=color_Cpn10, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h", "96h"),
         cex = 1,
         angle_col=45,
         main = "Cpn10",
         annotation_legend = TRUE,
         display_numbers = Niben_Cpn10_DEG, # %>% select(1:15),
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(84, 104),
         annotation_row = Niben_Cpn10_Modules_Chl,
         annotation_colors = module_Chl_colors
         #gaps_col = c(15)
)


rownames(Niben_Cpn10_TPM) <- Niben_Cpn10_TPM$Protein
Niben_Cpn10_TPM[1] <- NULL

min(Niben_Cpn10_TPM, na.rm = TRUE)
max(Niben_Cpn10_TPM, na.rm = TRUE)
breaks_Niben_Cpn10_TPM=c(seq(0,100, length.out = 100))

Niben_Cpn10_TPM_norm <- normalize_df(Niben_Cpn10_TPM)
Niben_Cpn10_TPM_norm_Cornell <- Niben_Cpn10_TPM %>%
  select(1:48) 
Niben_Cpn10_TPM_norm_Cornell <- normalize_df(Niben_Cpn10_TPM_norm_Cornell)


pheatmap(Niben_Cpn10_TPM, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         breaks = breaks_Niben_Cpn10_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_Cpn10_Annotation, annotation_colors=color_Cpn10, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Cpn10",
         annotation_legend = TRUE,
         #display_numbers = Niben_Cpn10_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         annotation_row = Niben_Cpn10_Modules_Chl,
         annotation_colors = module_Chl_colors,
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(24)#, 48, 51)
)



#####Representing Heatmap for Sly

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Sly_Cpn10_Ann <- read.table("Funcional Groups/FigTree/Sly_Cpn10.txt",
                            sep = "\t", header = TRUE)

colnames(Sly_Cpn10_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Sly_Cpn10 <- merge(Sly_Cpn10_Ann, Summary_Yazdani_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Sly_Cpn10_TPM <- Sly_Cpn10 %>% select(Protein, 4:66)
Sly_Cpn10_TPM <- Sly_Cpn10_TPM[!duplicated(Sly_Cpn10_TPM$Protein),]
Sly_Cpn10_TPM <- Sly_Cpn10_TPM[order(match(Sly_Cpn10_TPM[,1], Sly_Cpn10_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_Cpn10_FC <- Sly_Cpn10 %>% select(Protein, 67:71)
Sly_Cpn10_FC <- Sly_Cpn10_FC[!duplicated(Sly_Cpn10_FC$Protein),]
Sly_Cpn10_FC <- Sly_Cpn10_FC[order(match(Sly_Cpn10_FC[,1], Sly_Cpn10_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_Cpn10_DEG <- Sly_Cpn10 %>% select(Protein, 72:76)
Sly_Cpn10_DEG <- Sly_Cpn10_DEG[!duplicated(Sly_Cpn10_DEG$Protein),]
Sly_Cpn10_DEG <- Sly_Cpn10_DEG[order(match(Sly_Cpn10_DEG[,1], Sly_Cpn10_Ann[,2])),]

Sly_Cpn10_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Sly_Cpn10_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Sly_Cpn10_DEG$Ripening_Orange_DEG <- ifelse(is.na(Sly_Cpn10_DEG$Ripening_Orange_DEG) == TRUE, " ", "*")
Sly_Cpn10_DEG$Ripening_Red_DEG <- ifelse(is.na(Sly_Cpn10_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
Sly_Cpn10_DEG$ORHis_MG_DEG <- ifelse(is.na(Sly_Cpn10_DEG$ORHis_MG_DEG) == TRUE, " ", "*")
Sly_Cpn10_DEG$ORHis_Breaker_DEG <- ifelse(is.na(Sly_Cpn10_DEG$ORHis_Breaker_DEG) == TRUE, " ", "*")
row.names(Sly_Cpn10_DEG) <- Sly_Cpn10_DEG$Protein
Sly_Cpn10_DEG[1] <- NULL

rownames(Sly_Cpn10_FC) <- Sly_Cpn10_FC$Protein
Sly_Cpn10_FC[1] <- NULL

min(Sly_Cpn10_FC, na.rm = TRUE)
max(Sly_Cpn10_FC, na.rm = TRUE)
breaks_Sly_Cpn10=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Sly_Cpn10_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Sly_Cpn10,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_Cpn10_Annotation, annotation_colors=color_Cpn10, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Cpn10",
         annotation_legend = TRUE,
         display_numbers = Sly_Cpn10_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(39, 48),
         gaps_col = c(3)
)


rownames(Sly_Cpn10_TPM) <- Sly_Cpn10_TPM$Protein
Sly_Cpn10_TPM[1] <- NULL

min(Sly_Cpn10_TPM, na.rm = TRUE)
max(Sly_Cpn10_TPM, na.rm = TRUE)
breaks_Sly_Cpn10_TPM=c(seq(0,100, length.out = 200))

Sly_Cpn10_TPM_norm <- normalize_df(Sly_Cpn10_TPM)

pheatmap(Sly_Cpn10_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Sly_Cpn10_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_Cpn10_Annotation, annotation_colors=color_Cpn10, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Cpn10",
         annotation_legend = TRUE,
         #display_numbers = Sly_Cpn10_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(21,42)
)


#####Representing Heatmap for Cann

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Cann_Cpn10_Ann <- read.table("Funcional Groups/FigTree/Cann_Cpn10.txt",
                             sep = "\t", header = TRUE)

colnames(Cann_Cpn10_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Cann_Cpn10 <- merge(Cann_Cpn10_Ann, Summary_Pepper_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Cann_Cpn10_TPM <- Cann_Cpn10 %>% select(Protein, 4:17)
Cann_Cpn10_TPM <- Cann_Cpn10_TPM[!duplicated(Cann_Cpn10_TPM$Protein),]
Cann_Cpn10_TPM <- Cann_Cpn10_TPM[order(match(Cann_Cpn10_TPM[,1], Cann_Cpn10_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_Cpn10_FC <- Cann_Cpn10 %>% select(Protein, 18:19)
Cann_Cpn10_FC <- Cann_Cpn10_FC[!duplicated(Cann_Cpn10_FC$Protein),]
Cann_Cpn10_FC <- Cann_Cpn10_FC[order(match(Cann_Cpn10_FC[,1], Cann_Cpn10_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_Cpn10_DEG <- Cann_Cpn10 %>% select(Protein, 20:21)
Cann_Cpn10_DEG <- Cann_Cpn10_DEG[!duplicated(Cann_Cpn10_DEG$Protein),]
Cann_Cpn10_DEG <- Cann_Cpn10_DEG[order(match(Cann_Cpn10_DEG[,1], Cann_Cpn10_Ann[,2])),]

Cann_Cpn10_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Cann_Cpn10_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Cann_Cpn10_DEG$Ripening_Red_DEG <- ifelse(is.na(Cann_Cpn10_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
row.names(Cann_Cpn10_DEG) <- Cann_Cpn10_DEG$Protein
Cann_Cpn10_DEG[1] <- NULL

rownames(Cann_Cpn10_FC) <- Cann_Cpn10_FC$Protein
Cann_Cpn10_FC[1] <- NULL

min(Cann_Cpn10_FC, na.rm = TRUE)
max(Cann_Cpn10_FC, na.rm = TRUE)
breaks_Cann_Cpn10=c(seq(-4.5,0, length.out = 100), seq(0.01,6,length.out = 100))

pheatmap(Cann_Cpn10_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Cann_Cpn10,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_Cpn10_Annotation, annotation_colors=color_Cpn10, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Red"),
         cex = 1,
         angle_col=45,
         main = "Cpn10",
         annotation_legend = TRUE,
         display_numbers = Cann_Cpn10_DEG,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)


rownames(Cann_Cpn10_TPM) <- Cann_Cpn10_TPM$Protein
Cann_Cpn10_TPM[1] <- NULL

min(Cann_Cpn10_TPM, na.rm = TRUE)
max(Cann_Cpn10_TPM, na.rm = TRUE)
breaks_Cann_Cpn10_TPM=c(seq(0,100, length.out = 200))

Cann_Cpn10_TPM_norm <- normalize_df(Cann_Cpn10_TPM)

pheatmap(Cann_Cpn10_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Cann_Cpn10_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_Cpn10_Annotation, annotation_colors=color_Cpn10, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Cpn10",
         annotation_legend = TRUE,
         #display_numbers = Cann_Cpn10_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(4,9)
)

#########################################################################################################################################
####################################  Prefoldin  #############################################################################################
#########################################################################################################################################

Prefoldin_Ath <- read.delim("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan/Funcional Groups/Ath_Prefoldin.txt")
Prefoldin_Ath <- merge(Prefoldin_Ath, Ath_Gene_Models, by.x="GeneID", by.y=1)
Prefoldin_Ath_fasta <- Ath11_fasta[c(which(names(Ath11_fasta) %in% Prefoldin_Ath$V2))]
Prefoldin_Ath_2 <- Prefoldin_Ath %>% select(V2, Name)
write.fasta(Prefoldin_Ath_fasta, as.string=TRUE, names= names(Prefoldin_Ath_fasta),
            file.out = "Funcional Groups/Ath_Prefoldin.fasta")
rename.fasta(infile = "Funcional Groups/Ath_Prefoldin.fasta", Prefoldin_Ath_2, outfile = "Funcional Groups/Ath_Prefoldin_2.fasta")

Prefoldin_Niben_Candidates <- read.table("Funcional Groups/Niben_Candidates_Prefoldin.txt")
Prefoldin_Niben_Candidates_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Prefoldin_Niben_Candidates$V1))]
write.fasta(Prefoldin_Niben_Candidates_fasta, as.string=TRUE, names= names(Prefoldin_Niben_Candidates_fasta),
            file.out = "Funcional Groups/Prefoldin_Niben_Candidates_35.fasta")

Prefoldin_Sly_Candidates <- read.table("Funcional Groups/Sly_Candidates_Prefoldin.txt")
Prefoldin_Sly_Candidates_fasta <- Tomato_4.0_fasta[c(which(names(Tomato_4.0_fasta) %in% Prefoldin_Sly_Candidates$V1))]
write.fasta(Prefoldin_Sly_Candidates_fasta, as.string=TRUE, names= names(Prefoldin_Sly_Candidates_fasta),
            file.out = "Funcional Groups/Prefoldin_Sly_Candidates_35.fasta")

Prefoldin_Cann_Candidates <- read.table("Funcional Groups/Cann_Candidates_Prefoldin.txt")
Prefoldin_Cann_Candidates_fasta <- Cannuum_ASM51225v2_fasta[c(which(names(Cannuum_ASM51225v2_fasta) %in% Prefoldin_Cann_Candidates$V1))]
write.fasta(Prefoldin_Cann_Candidates_fasta, as.string=TRUE, names= names(Prefoldin_Cann_Candidates_fasta),
            file.out = "Funcional Groups/Prefoldin_Cann_Candidates_35.fasta")

Niben_Prefoldin_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_Prefoldin_Ann$Name))]
write.fasta(Niben_Prefoldin_fasta, as.string=TRUE, names= names(Niben_Prefoldin_fasta),
            file.out = "Funcional Groups/Niben_Prefoldin.fasta")
Niben_Prefoldin_Ann_2 <- Niben_Prefoldin_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_Prefoldin.fasta", Niben_Prefoldin_Ann, outfile = "Funcional Groups/Niben_Prefoldin_2.fasta")

save_heatmaps("Prefoldin", height = 6)

## Representing Heatmap

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Niben_Prefoldin_Ann <- read.table("Funcional Groups/FigTree/Niben_Prefoldin.txt",
                              sep = "\t", header = TRUE)

colnames(Niben_Prefoldin_Ann) <- c("Name", "Protein")

Niben_Prefoldin_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_Prefoldin_Ann$Name))]
write.fasta(Niben_Prefoldin_fasta, as.string=TRUE, names= names(Niben_Prefoldin_fasta),
            file.out = "Funcional Groups/Niben_Prefoldin.fasta")
Niben_Prefoldin_Ann_2 <- Niben_Prefoldin_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_Prefoldin.fasta", Niben_Prefoldin_Ann, outfile = "Funcional Groups/Niben_Prefoldin_2.fasta")

Niben_Prefoldin_Chloro <- read.delim("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan/Funcional Groups/Niben_Prefoldin_2_summary.targetp2", 
                                 header=FALSE, comment.char="#")
colnames(Niben_Prefoldin_Chloro) <- c("ID",	"Prediction",	"noTP",	"SP",	"mTP",	"cTP",	"luTP",	"CS-Position")
Niben_Prefoldin_Chloro$Prediction <- factor(Niben_Prefoldin_Chloro$Prediction, levels = c("cTP","mTP","noTP"))
Niben_Prefoldin_Chloro <- merge(Niben_Prefoldin_Ann, Niben_Prefoldin_Chloro, by.x = "Protein", by.y = "ID")

Niben_Prefoldin_Chloro <- Niben_Prefoldin_Chloro[order(match(Niben_Prefoldin_Chloro$Protein, Niben_Prefoldin_Ann$Protein)),]
rownames(Niben_Prefoldin_Chloro) <- Niben_Prefoldin_Chloro$Protein
Niben_Prefoldin_Chloro <- Niben_Prefoldin_Chloro %>% select(Prediction)


## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_Prefoldin <- merge(Niben_Prefoldin_Ann, TPM_All_FC_Annotated, by="Name")

## We create the annotation for Modules

Niben_Prefoldin_Modules <- which_Module_pwd30(Niben_Prefoldin_Ann)

## We merge this annotation of Modules with the annotation of Chloroplast TP prediction

Niben_Prefoldin_Modules_Chl <- merge(Niben_Prefoldin_Modules, Niben_Prefoldin_Chloro, by = 0)
Niben_Prefoldin_Modules_Chl <- Niben_Prefoldin_Modules_Chl[order(match(Niben_Prefoldin_Modules_Chl$Row.names, Niben_Prefoldin_Ann$Protein)),]
rownames(Niben_Prefoldin_Modules_Chl) <- Niben_Prefoldin_Modules_Chl$Row.names
Niben_Prefoldin_Modules_Chl$Row.names <- NULL

## We select only the FC data from the different comparisons
Niben_Prefoldin_TPM <- Niben_Prefoldin %>% select(Protein, 4:51, 61:66)
Niben_Prefoldin_TPM <- Niben_Prefoldin_TPM[!duplicated(Niben_Prefoldin_TPM$Protein),]
Niben_Prefoldin_TPM <- Niben_Prefoldin_TPM[order(match(Niben_Prefoldin_TPM[,1], Niben_Prefoldin_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_Prefoldin_FC <- Niben_Prefoldin %>% select(Protein, 71, 73:87)
Niben_Prefoldin_FC <- Niben_Prefoldin_FC[!duplicated(Niben_Prefoldin_FC$Protein),]
Niben_Prefoldin_FC <- Niben_Prefoldin_FC[order(match(Niben_Prefoldin_FC[,1], Niben_Prefoldin_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_Prefoldin_DEG <- Niben_Prefoldin %>% select(Protein, 89, 91:105)
Niben_Prefoldin_DEG <- Niben_Prefoldin_DEG[!duplicated(Niben_Prefoldin_DEG$Protein),]
Niben_Prefoldin_DEG <- Niben_Prefoldin_DEG[order(match(Niben_Prefoldin_DEG[,1], Niben_Prefoldin_Ann[,2])),]

Niben_Prefoldin_DEG$pcrtB22h_vs_GFP22h_filtered <- ifelse(is.na(Niben_Prefoldin_DEG$pcrtB22h_vs_GFP22h_filtered) == TRUE, " ", "*")
Niben_Prefoldin_DEG$pcrtB25h_vs_GFP25h_filtered <- ifelse(is.na(Niben_Prefoldin_DEG$pcrtB25h_vs_GFP25h_filtered) == TRUE, " ", "*")
Niben_Prefoldin_DEG$pcrtBAA_vs_GFPAA_filtered <- ifelse(is.na(Niben_Prefoldin_DEG$pcrtBAA_vs_GFPAA_filtered) == TRUE, " ", "*")
Niben_Prefoldin_DEG$pcrtBBB_vs_GFPBB_filtered <- ifelse(is.na(Niben_Prefoldin_DEG$pcrtBBB_vs_GFPBB_filtered) == TRUE, " ", "*")
Niben_Prefoldin_DEG$pcrtBYY_vs_GFPYY_filtered <- ifelse(is.na(Niben_Prefoldin_DEG$pcrtBYY_vs_GFPYY_filtered) == TRUE, " ", "*")
Niben_Prefoldin_DEG$pcrtB28h_vs_GFP28h_filtered <- ifelse(is.na(Niben_Prefoldin_DEG$pcrtB28h_vs_GFP28h_filtered) == TRUE, " ", "*")
Niben_Prefoldin_DEG$pcrtB34h_vs_GFP34h_filtered <- ifelse(is.na(Niben_Prefoldin_DEG$pcrtB34h_vs_GFP34h_filtered) == TRUE, " ", "*")
Niben_Prefoldin_DEG$pcrtB37h_vs_GFP37h_filtered <- ifelse(is.na(Niben_Prefoldin_DEG$pcrtB37h_vs_GFP37h_filtered) == TRUE, " ", "*")
Niben_Prefoldin_DEG$pcrtBZZ_vs_GFPZZ_filtered <- ifelse(is.na(Niben_Prefoldin_DEG$pcrtBZZ_vs_GFPZZ_filtered) == TRUE, " ", "*")
Niben_Prefoldin_DEG$pcrtBCC_vs_GFPCC_filtered <- ifelse(is.na(Niben_Prefoldin_DEG$pcrtBCC_vs_GFPCC_filtered) == TRUE, " ", "*")
Niben_Prefoldin_DEG$pcrtB40h_vs_GFP40h_filtered <- ifelse(is.na(Niben_Prefoldin_DEG$pcrtB40h_vs_GFP40h_filtered) == TRUE, " ", "*")
Niben_Prefoldin_DEG$pcrtBDD_vs_GFPDD_filtered <- ifelse(is.na(Niben_Prefoldin_DEG$pcrtBDD_vs_GFPDD_filtered) == TRUE, " ", "*")
Niben_Prefoldin_DEG$pcrtB46h_vs_GFP46h_filtered <- ifelse(is.na(Niben_Prefoldin_DEG$pcrtB46h_vs_GFP46h_filtered) == TRUE, " ", "*")
Niben_Prefoldin_DEG$pcrtB56h_vs_GFP56h_filtered <- ifelse(is.na(Niben_Prefoldin_DEG$pcrtB56h_vs_GFP56h_filtered) == TRUE, " ", "*")
Niben_Prefoldin_DEG$pcrtBEE_vs_GFPEE_filtered <- ifelse(is.na(Niben_Prefoldin_DEG$pcrtBEE_vs_GFPEE_filtered) == TRUE, " ", "*")
Niben_Prefoldin_DEG$pcrtB96h_vs_GFP96h_filtered <- ifelse(is.na(Niben_Prefoldin_DEG$pcrtB96h_vs_GFP96h_filtered) == TRUE, " ", "*")
row.names(Niben_Prefoldin_DEG) <- Niben_Prefoldin_DEG$Protein
Niben_Prefoldin_DEG[1] <- NULL

Niben_Prefoldin_DEG <- Niben_Prefoldin_DEG %>%
  select(pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

row.names(Niben_Prefoldin_FC) <- Niben_Prefoldin_FC$Protein
Niben_Prefoldin_FC[1] <- NULL

Niben_Prefoldin_FC <- Niben_Prefoldin_FC %>%
  select(pcrtB22h_vs_GFP22h_All, pcrtBAA_vs_GFPAA_All, pcrtB25h_vs_GFP25h_All, 
         pcrtBYY_vs_GFPYY_All, pcrtBBB_vs_GFPBB_All, pcrtB28h_vs_GFP28h_All, pcrtB34h_vs_GFP34h_All, pcrtBZZ_vs_GFPZZ_All, 
         pcrtB37h_vs_GFP37h_All, pcrtBCC_vs_GFPCC_All, pcrtB40h_vs_GFP40h_All, pcrtBDD_vs_GFPDD_All, pcrtB46h_vs_GFP46h_All, 
         pcrtBEE_vs_GFPEE_All, pcrtB56h_vs_GFP56h_All, pcrtB96h_vs_GFP96h_All)


min(Niben_Prefoldin_FC, na.rm = TRUE)
max(Niben_Prefoldin_FC, na.rm = TRUE)
breaks_Prefoldin_FC=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Niben_Prefoldin_FC, # %>% select(1:15), 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Prefoldin_FC,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_Prefoldin_Annotation, annotation_colors=color_Prefoldin, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h", "96h"),
         cex = 1,
         angle_col=45,
         main = "Prefoldin",
         annotation_legend = TRUE,
         display_numbers = Niben_Prefoldin_DEG, # %>% select(1:15),
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(84, 104),
         annotation_row = Niben_Prefoldin_Modules_Chl,
         annotation_colors = module_Chl_colors
         #gaps_col = c(15)
)


rownames(Niben_Prefoldin_TPM) <- Niben_Prefoldin_TPM$Protein
Niben_Prefoldin_TPM[1] <- NULL

min(Niben_Prefoldin_TPM, na.rm = TRUE)
max(Niben_Prefoldin_TPM, na.rm = TRUE)
breaks_Niben_Prefoldin_TPM=c(seq(0,100, length.out = 100))

Niben_Prefoldin_TPM_norm <- normalize_df(Niben_Prefoldin_TPM)
Niben_Prefoldin_TPM_norm_Cornell <- Niben_Prefoldin_TPM %>%
  select(1:48) 
Niben_Prefoldin_TPM_norm_Cornell <- normalize_df(Niben_Prefoldin_TPM_norm_Cornell)


pheatmap(Niben_Prefoldin_TPM, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         breaks = breaks_Niben_Prefoldin_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_Prefoldin_Annotation, annotation_colors=color_Prefoldin, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Prefoldin",
         annotation_legend = TRUE,
         #display_numbers = Niben_Prefoldin_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         annotation_row = Niben_Prefoldin_Modules_Chl,
         annotation_colors = module_Chl_colors,
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(24)#, 48, 51)
)



#####Representing Heatmap for Sly

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Sly_Prefoldin_Ann <- read.table("Funcional Groups/FigTree/Sly_Prefoldin.txt",
                            sep = "\t", header = TRUE)

colnames(Sly_Prefoldin_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Sly_Prefoldin <- merge(Sly_Prefoldin_Ann, Summary_Yazdani_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Sly_Prefoldin_TPM <- Sly_Prefoldin %>% select(Protein, 4:66)
Sly_Prefoldin_TPM <- Sly_Prefoldin_TPM[!duplicated(Sly_Prefoldin_TPM$Protein),]
Sly_Prefoldin_TPM <- Sly_Prefoldin_TPM[order(match(Sly_Prefoldin_TPM[,1], Sly_Prefoldin_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_Prefoldin_FC <- Sly_Prefoldin %>% select(Protein, 67:71)
Sly_Prefoldin_FC <- Sly_Prefoldin_FC[!duplicated(Sly_Prefoldin_FC$Protein),]
Sly_Prefoldin_FC <- Sly_Prefoldin_FC[order(match(Sly_Prefoldin_FC[,1], Sly_Prefoldin_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_Prefoldin_DEG <- Sly_Prefoldin %>% select(Protein, 72:76)
Sly_Prefoldin_DEG <- Sly_Prefoldin_DEG[!duplicated(Sly_Prefoldin_DEG$Protein),]
Sly_Prefoldin_DEG <- Sly_Prefoldin_DEG[order(match(Sly_Prefoldin_DEG[,1], Sly_Prefoldin_Ann[,2])),]

Sly_Prefoldin_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Sly_Prefoldin_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Sly_Prefoldin_DEG$Ripening_Orange_DEG <- ifelse(is.na(Sly_Prefoldin_DEG$Ripening_Orange_DEG) == TRUE, " ", "*")
Sly_Prefoldin_DEG$Ripening_Red_DEG <- ifelse(is.na(Sly_Prefoldin_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
Sly_Prefoldin_DEG$ORHis_MG_DEG <- ifelse(is.na(Sly_Prefoldin_DEG$ORHis_MG_DEG) == TRUE, " ", "*")
Sly_Prefoldin_DEG$ORHis_Breaker_DEG <- ifelse(is.na(Sly_Prefoldin_DEG$ORHis_Breaker_DEG) == TRUE, " ", "*")
row.names(Sly_Prefoldin_DEG) <- Sly_Prefoldin_DEG$Protein
Sly_Prefoldin_DEG[1] <- NULL

rownames(Sly_Prefoldin_FC) <- Sly_Prefoldin_FC$Protein
Sly_Prefoldin_FC[1] <- NULL

min(Sly_Prefoldin_FC, na.rm = TRUE)
max(Sly_Prefoldin_FC, na.rm = TRUE)
breaks_Sly_Prefoldin=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Sly_Prefoldin_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Sly_Prefoldin,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_Prefoldin_Annotation, annotation_colors=color_Prefoldin, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Prefoldin",
         annotation_legend = TRUE,
         display_numbers = Sly_Prefoldin_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(39, 48),
         gaps_col = c(3)
)


rownames(Sly_Prefoldin_TPM) <- Sly_Prefoldin_TPM$Protein
Sly_Prefoldin_TPM[1] <- NULL

min(Sly_Prefoldin_TPM, na.rm = TRUE)
max(Sly_Prefoldin_TPM, na.rm = TRUE)
breaks_Sly_Prefoldin_TPM=c(seq(0,100, length.out = 200))

Sly_Prefoldin_TPM_norm <- normalize_df(Sly_Prefoldin_TPM)

pheatmap(Sly_Prefoldin_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Sly_Prefoldin_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_Prefoldin_Annotation, annotation_colors=color_Prefoldin, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Prefoldin",
         annotation_legend = TRUE,
         #display_numbers = Sly_Prefoldin_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(21,42)
)


#####Representing Heatmap for Cann

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Cann_Prefoldin_Ann <- read.table("Funcional Groups/FigTree/Cann_Prefoldin.txt",
                             sep = "\t", header = TRUE)

colnames(Cann_Prefoldin_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Cann_Prefoldin <- merge(Cann_Prefoldin_Ann, Summary_Pepper_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Cann_Prefoldin_TPM <- Cann_Prefoldin %>% select(Protein, 4:17)
Cann_Prefoldin_TPM <- Cann_Prefoldin_TPM[!duplicated(Cann_Prefoldin_TPM$Protein),]
Cann_Prefoldin_TPM <- Cann_Prefoldin_TPM[order(match(Cann_Prefoldin_TPM[,1], Cann_Prefoldin_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_Prefoldin_FC <- Cann_Prefoldin %>% select(Protein, 18:19)
Cann_Prefoldin_FC <- Cann_Prefoldin_FC[!duplicated(Cann_Prefoldin_FC$Protein),]
Cann_Prefoldin_FC <- Cann_Prefoldin_FC[order(match(Cann_Prefoldin_FC[,1], Cann_Prefoldin_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_Prefoldin_DEG <- Cann_Prefoldin %>% select(Protein, 20:21)
Cann_Prefoldin_DEG <- Cann_Prefoldin_DEG[!duplicated(Cann_Prefoldin_DEG$Protein),]
Cann_Prefoldin_DEG <- Cann_Prefoldin_DEG[order(match(Cann_Prefoldin_DEG[,1], Cann_Prefoldin_Ann[,2])),]

Cann_Prefoldin_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Cann_Prefoldin_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Cann_Prefoldin_DEG$Ripening_Red_DEG <- ifelse(is.na(Cann_Prefoldin_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
row.names(Cann_Prefoldin_DEG) <- Cann_Prefoldin_DEG$Protein
Cann_Prefoldin_DEG[1] <- NULL

rownames(Cann_Prefoldin_FC) <- Cann_Prefoldin_FC$Protein
Cann_Prefoldin_FC[1] <- NULL

min(Cann_Prefoldin_FC, na.rm = TRUE)
max(Cann_Prefoldin_FC, na.rm = TRUE)
breaks_Cann_Prefoldin=c(seq(-4.5,0, length.out = 100), seq(0.01,6,length.out = 100))

pheatmap(Cann_Prefoldin_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Cann_Prefoldin,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_Prefoldin_Annotation, annotation_colors=color_Prefoldin, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Red"),
         cex = 1,
         angle_col=45,
         main = "Prefoldin",
         annotation_legend = TRUE,
         display_numbers = Cann_Prefoldin_DEG,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)


rownames(Cann_Prefoldin_TPM) <- Cann_Prefoldin_TPM$Protein
Cann_Prefoldin_TPM[1] <- NULL

min(Cann_Prefoldin_TPM, na.rm = TRUE)
max(Cann_Prefoldin_TPM, na.rm = TRUE)
breaks_Cann_Prefoldin_TPM=c(seq(0,100, length.out = 200))

Cann_Prefoldin_TPM_norm <- normalize_df(Cann_Prefoldin_TPM)

pheatmap(Cann_Prefoldin_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Cann_Prefoldin_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_Prefoldin_Annotation, annotation_colors=color_Prefoldin, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Prefoldin",
         annotation_legend = TRUE,
         #display_numbers = Cann_Prefoldin_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(4,9)
)

#########################################################################################################################################
####################################  DNAJ  #############################################################################################
#########################################################################################################################################

DNAJ_Ath <- read.delim("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan/Funcional Groups/Ath_DNAJ.txt")
DNAJ_Ath <- merge(DNAJ_Ath, Ath_Gene_Models, by.x="GeneID", by.y=1)
DNAJ_Ath_fasta <- Ath11_fasta[c(which(names(Ath11_fasta) %in% DNAJ_Ath$V2))]
DNAJ_Ath_2 <- DNAJ_Ath %>% select(V2, Name)
write.fasta(DNAJ_Ath_fasta, as.string=TRUE, names= names(DNAJ_Ath_fasta),
            file.out = "Funcional Groups/Ath_DNAJ.fasta")
rename.fasta(infile = "Funcional Groups/Ath_DNAJ.fasta", DNAJ_Ath_2, outfile = "Funcional Groups/Ath_DNAJ_2.fasta")

DNAJ_Niben_Candidates <- read.table("Funcional Groups/Niben_Candidates_DNAJ.txt")
DNAJ_Niben_Candidates_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% DNAJ_Niben_Candidates$V1))]
write.fasta(DNAJ_Niben_Candidates_fasta, as.string=TRUE, names= names(DNAJ_Niben_Candidates_fasta),
            file.out = "Funcional Groups/DNAJ_Niben_Candidates_35.fasta")

DNAJ_Sly_Candidates <- read.table("Funcional Groups/Sly_Candidates_DNAJ.txt")
DNAJ_Sly_Candidates_fasta <- Tomato_4.0_fasta[c(which(names(Tomato_4.0_fasta) %in% DNAJ_Sly_Candidates$V1))]
write.fasta(DNAJ_Sly_Candidates_fasta, as.string=TRUE, names= names(DNAJ_Sly_Candidates_fasta),
            file.out = "Funcional Groups/DNAJ_Sly_Candidates_35.fasta")

DNAJ_Cann_Candidates <- read.table("Funcional Groups/Cann_Candidates_DNAJ.txt")
DNAJ_Cann_Candidates_fasta <- Cannuum_ASM51225v2_fasta[c(which(names(Cannuum_ASM51225v2_fasta) %in% DNAJ_Cann_Candidates$V1))]
write.fasta(DNAJ_Cann_Candidates_fasta, as.string=TRUE, names= names(DNAJ_Cann_Candidates_fasta),
            file.out = "Funcional Groups/DNAJ_Cann_Candidates_35.fasta")

Niben_DNAJ_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_DNAJ_Ann$Name))]
write.fasta(Niben_DNAJ_fasta, as.string=TRUE, names= names(Niben_DNAJ_fasta),
            file.out = "Funcional Groups/Niben_DNAJ.fasta")
Niben_DNAJ_Ann_2 <- Niben_DNAJ_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_DNAJ.fasta", Niben_DNAJ_Ann, outfile = "Funcional Groups/Niben_DNAJ_2.fasta")

save_heatmaps("DNAJ", height = 20)


## Representing Heatmap

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Niben_DNAJ_Ann <- read.table("Funcional Groups/FigTree/Niben_DNAJ.txt",
                              sep = "\t", header = TRUE)

colnames(Niben_DNAJ_Ann) <- c("Name", "Protein")

Niben_DNAJ_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_DNAJ_Ann$Name))]
write.fasta(Niben_DNAJ_fasta, as.string=TRUE, names= names(Niben_DNAJ_fasta),
            file.out = "Funcional Groups/Niben_DNAJ.fasta")
Niben_DNAJ_Ann_2 <- Niben_DNAJ_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_DNAJ.fasta", Niben_DNAJ_Ann, outfile = "Funcional Groups/Niben_DNAJ_2.fasta")

Niben_DNAJ_Chloro <- read.delim("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan/Funcional Groups/Niben_DNAJ_2_summary.targetp2", 
                                 header=FALSE, comment.char="#")
colnames(Niben_DNAJ_Chloro) <- c("ID",	"Prediction",	"noTP",	"SP",	"mTP",	"cTP",	"luTP",	"CS-Position")
Niben_DNAJ_Chloro$Prediction <- factor(Niben_DNAJ_Chloro$Prediction, levels = c("cTP","mTP","noTP"))
Niben_DNAJ_Chloro <- merge(Niben_DNAJ_Ann, Niben_DNAJ_Chloro, by.x = "Protein", by.y = "ID")

Niben_DNAJ_Chloro <- Niben_DNAJ_Chloro[order(match(Niben_DNAJ_Chloro$Protein, Niben_DNAJ_Ann$Protein)),]
rownames(Niben_DNAJ_Chloro) <- Niben_DNAJ_Chloro$Protein
Niben_DNAJ_Chloro <- Niben_DNAJ_Chloro %>% select(Prediction)


## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_DNAJ <- merge(Niben_DNAJ_Ann, TPM_All_FC_Annotated, by="Name")

## We create the annotation for Modules

Niben_DNAJ_Modules <- which_Module_pwd30(Niben_DNAJ_Ann)

## We merge this annotation of Modules with the annotation of Chloroplast TP prediction

Niben_DNAJ_Modules_Chl <- merge(Niben_DNAJ_Modules, Niben_DNAJ_Chloro, by = 0)
Niben_DNAJ_Modules_Chl <- Niben_DNAJ_Modules_Chl[order(match(Niben_DNAJ_Modules_Chl$Row.names, Niben_DNAJ_Ann$Protein)),]
rownames(Niben_DNAJ_Modules_Chl) <- Niben_DNAJ_Modules_Chl$Row.names
Niben_DNAJ_Modules_Chl$Row.names <- NULL

## We select only the FC data from the different comparisons
Niben_DNAJ_TPM <- Niben_DNAJ %>% select(Protein, 4:51, 61:66)
Niben_DNAJ_TPM <- Niben_DNAJ_TPM[!duplicated(Niben_DNAJ_TPM$Protein),]
Niben_DNAJ_TPM <- Niben_DNAJ_TPM[order(match(Niben_DNAJ_TPM[,1], Niben_DNAJ_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_DNAJ_FC <- Niben_DNAJ %>% select(Protein, 71, 73:87)
Niben_DNAJ_FC <- Niben_DNAJ_FC[!duplicated(Niben_DNAJ_FC$Protein),]
Niben_DNAJ_FC <- Niben_DNAJ_FC[order(match(Niben_DNAJ_FC[,1], Niben_DNAJ_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_DNAJ_DEG <- Niben_DNAJ %>% select(Protein, 89, 91:105)
Niben_DNAJ_DEG <- Niben_DNAJ_DEG[!duplicated(Niben_DNAJ_DEG$Protein),]
Niben_DNAJ_DEG <- Niben_DNAJ_DEG[order(match(Niben_DNAJ_DEG[,1], Niben_DNAJ_Ann[,2])),]

Niben_DNAJ_DEG$pcrtB22h_vs_GFP22h_filtered <- ifelse(is.na(Niben_DNAJ_DEG$pcrtB22h_vs_GFP22h_filtered) == TRUE, " ", "*")
Niben_DNAJ_DEG$pcrtB25h_vs_GFP25h_filtered <- ifelse(is.na(Niben_DNAJ_DEG$pcrtB25h_vs_GFP25h_filtered) == TRUE, " ", "*")
Niben_DNAJ_DEG$pcrtBAA_vs_GFPAA_filtered <- ifelse(is.na(Niben_DNAJ_DEG$pcrtBAA_vs_GFPAA_filtered) == TRUE, " ", "*")
Niben_DNAJ_DEG$pcrtBBB_vs_GFPBB_filtered <- ifelse(is.na(Niben_DNAJ_DEG$pcrtBBB_vs_GFPBB_filtered) == TRUE, " ", "*")
Niben_DNAJ_DEG$pcrtBYY_vs_GFPYY_filtered <- ifelse(is.na(Niben_DNAJ_DEG$pcrtBYY_vs_GFPYY_filtered) == TRUE, " ", "*")
Niben_DNAJ_DEG$pcrtB28h_vs_GFP28h_filtered <- ifelse(is.na(Niben_DNAJ_DEG$pcrtB28h_vs_GFP28h_filtered) == TRUE, " ", "*")
Niben_DNAJ_DEG$pcrtB34h_vs_GFP34h_filtered <- ifelse(is.na(Niben_DNAJ_DEG$pcrtB34h_vs_GFP34h_filtered) == TRUE, " ", "*")
Niben_DNAJ_DEG$pcrtB37h_vs_GFP37h_filtered <- ifelse(is.na(Niben_DNAJ_DEG$pcrtB37h_vs_GFP37h_filtered) == TRUE, " ", "*")
Niben_DNAJ_DEG$pcrtBZZ_vs_GFPZZ_filtered <- ifelse(is.na(Niben_DNAJ_DEG$pcrtBZZ_vs_GFPZZ_filtered) == TRUE, " ", "*")
Niben_DNAJ_DEG$pcrtBCC_vs_GFPCC_filtered <- ifelse(is.na(Niben_DNAJ_DEG$pcrtBCC_vs_GFPCC_filtered) == TRUE, " ", "*")
Niben_DNAJ_DEG$pcrtB40h_vs_GFP40h_filtered <- ifelse(is.na(Niben_DNAJ_DEG$pcrtB40h_vs_GFP40h_filtered) == TRUE, " ", "*")
Niben_DNAJ_DEG$pcrtBDD_vs_GFPDD_filtered <- ifelse(is.na(Niben_DNAJ_DEG$pcrtBDD_vs_GFPDD_filtered) == TRUE, " ", "*")
Niben_DNAJ_DEG$pcrtB46h_vs_GFP46h_filtered <- ifelse(is.na(Niben_DNAJ_DEG$pcrtB46h_vs_GFP46h_filtered) == TRUE, " ", "*")
Niben_DNAJ_DEG$pcrtB56h_vs_GFP56h_filtered <- ifelse(is.na(Niben_DNAJ_DEG$pcrtB56h_vs_GFP56h_filtered) == TRUE, " ", "*")
Niben_DNAJ_DEG$pcrtBEE_vs_GFPEE_filtered <- ifelse(is.na(Niben_DNAJ_DEG$pcrtBEE_vs_GFPEE_filtered) == TRUE, " ", "*")
Niben_DNAJ_DEG$pcrtB96h_vs_GFP96h_filtered <- ifelse(is.na(Niben_DNAJ_DEG$pcrtB96h_vs_GFP96h_filtered) == TRUE, " ", "*")
row.names(Niben_DNAJ_DEG) <- Niben_DNAJ_DEG$Protein
Niben_DNAJ_DEG[1] <- NULL

Niben_DNAJ_DEG <- Niben_DNAJ_DEG %>%
  select(pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

row.names(Niben_DNAJ_FC) <- Niben_DNAJ_FC$Protein
Niben_DNAJ_FC[1] <- NULL

Niben_DNAJ_FC <- Niben_DNAJ_FC %>%
  select(pcrtB22h_vs_GFP22h_All, pcrtBAA_vs_GFPAA_All, pcrtB25h_vs_GFP25h_All, 
         pcrtBYY_vs_GFPYY_All, pcrtBBB_vs_GFPBB_All, pcrtB28h_vs_GFP28h_All, pcrtB34h_vs_GFP34h_All, pcrtBZZ_vs_GFPZZ_All, 
         pcrtB37h_vs_GFP37h_All, pcrtBCC_vs_GFPCC_All, pcrtB40h_vs_GFP40h_All, pcrtBDD_vs_GFPDD_All, pcrtB46h_vs_GFP46h_All, 
         pcrtBEE_vs_GFPEE_All, pcrtB56h_vs_GFP56h_All, pcrtB96h_vs_GFP96h_All)


min(Niben_DNAJ_FC, na.rm = TRUE)
max(Niben_DNAJ_FC, na.rm = TRUE)
breaks_DNAJ_FC=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Niben_DNAJ_FC, # %>% select(1:15), 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_DNAJ_FC,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_DNAJ_Annotation, annotation_colors=color_DNAJ, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h", "96h"),
         cex = 1,
         angle_col=45,
         main = "DNAJ",
         annotation_legend = TRUE,
         display_numbers = Niben_DNAJ_DEG, # %>% select(1:15),
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(84, 104),
         annotation_row = Niben_DNAJ_Modules_Chl,
         annotation_colors = module_Chl_colors
         #gaps_col = c(15)
)


rownames(Niben_DNAJ_TPM) <- Niben_DNAJ_TPM$Protein
Niben_DNAJ_TPM[1] <- NULL

min(Niben_DNAJ_TPM, na.rm = TRUE)
max(Niben_DNAJ_TPM, na.rm = TRUE)
breaks_Niben_DNAJ_TPM=c(seq(0,100, length.out = 100))

Niben_DNAJ_TPM_norm <- normalize_df(Niben_DNAJ_TPM)
Niben_DNAJ_TPM_norm_Cornell <- Niben_DNAJ_TPM %>%
  select(1:48) 
Niben_DNAJ_TPM_norm_Cornell <- normalize_df(Niben_DNAJ_TPM_norm_Cornell)


pheatmap(Niben_DNAJ_TPM, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         breaks = breaks_Niben_DNAJ_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_DNAJ_Annotation, annotation_colors=color_DNAJ, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "DNAJ",
         annotation_legend = TRUE,
         #display_numbers = Niben_DNAJ_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         annotation_row = Niben_DNAJ_Modules_Chl,
         annotation_colors = module_Chl_colors,
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(24)#, 48, 51)
)



#####Representing Heatmap for Sly

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Sly_DNAJ_Ann <- read.table("Funcional Groups/FigTree/Sly_DNAJ.txt",
                            sep = "\t", header = TRUE)

colnames(Sly_DNAJ_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Sly_DNAJ <- merge(Sly_DNAJ_Ann, Summary_Yazdani_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Sly_DNAJ_TPM <- Sly_DNAJ %>% select(Protein, 4:66)
Sly_DNAJ_TPM <- Sly_DNAJ_TPM[!duplicated(Sly_DNAJ_TPM$Protein),]
Sly_DNAJ_TPM <- Sly_DNAJ_TPM[order(match(Sly_DNAJ_TPM[,1], Sly_DNAJ_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_DNAJ_FC <- Sly_DNAJ %>% select(Protein, 67:71)
Sly_DNAJ_FC <- Sly_DNAJ_FC[!duplicated(Sly_DNAJ_FC$Protein),]
Sly_DNAJ_FC <- Sly_DNAJ_FC[order(match(Sly_DNAJ_FC[,1], Sly_DNAJ_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_DNAJ_DEG <- Sly_DNAJ %>% select(Protein, 72:76)
Sly_DNAJ_DEG <- Sly_DNAJ_DEG[!duplicated(Sly_DNAJ_DEG$Protein),]
Sly_DNAJ_DEG <- Sly_DNAJ_DEG[order(match(Sly_DNAJ_DEG[,1], Sly_DNAJ_Ann[,2])),]

Sly_DNAJ_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Sly_DNAJ_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Sly_DNAJ_DEG$Ripening_Orange_DEG <- ifelse(is.na(Sly_DNAJ_DEG$Ripening_Orange_DEG) == TRUE, " ", "*")
Sly_DNAJ_DEG$Ripening_Red_DEG <- ifelse(is.na(Sly_DNAJ_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
Sly_DNAJ_DEG$ORHis_MG_DEG <- ifelse(is.na(Sly_DNAJ_DEG$ORHis_MG_DEG) == TRUE, " ", "*")
Sly_DNAJ_DEG$ORHis_Breaker_DEG <- ifelse(is.na(Sly_DNAJ_DEG$ORHis_Breaker_DEG) == TRUE, " ", "*")
row.names(Sly_DNAJ_DEG) <- Sly_DNAJ_DEG$Protein
Sly_DNAJ_DEG[1] <- NULL

rownames(Sly_DNAJ_FC) <- Sly_DNAJ_FC$Protein
Sly_DNAJ_FC[1] <- NULL

min(Sly_DNAJ_FC, na.rm = TRUE)
max(Sly_DNAJ_FC, na.rm = TRUE)
breaks_Sly_DNAJ=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Sly_DNAJ_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Sly_DNAJ,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_DNAJ_Annotation, annotation_colors=color_DNAJ, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "DNAJ",
         annotation_legend = TRUE,
         display_numbers = Sly_DNAJ_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(39, 48),
         gaps_col = c(3)
)


rownames(Sly_DNAJ_TPM) <- Sly_DNAJ_TPM$Protein
Sly_DNAJ_TPM[1] <- NULL

min(Sly_DNAJ_TPM, na.rm = TRUE)
max(Sly_DNAJ_TPM, na.rm = TRUE)
breaks_Sly_DNAJ_TPM=c(seq(0,100, length.out = 200))

Sly_DNAJ_TPM_norm <- normalize_df(Sly_DNAJ_TPM)

pheatmap(Sly_DNAJ_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Sly_DNAJ_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_DNAJ_Annotation, annotation_colors=color_DNAJ, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "DNAJ",
         annotation_legend = TRUE,
         #display_numbers = Sly_DNAJ_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(21,42)
)


#####Representing Heatmap for Cann

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Cann_DNAJ_Ann <- read.table("Funcional Groups/FigTree/Cann_DNAJ.txt",
                             sep = "\t", header = TRUE)

colnames(Cann_DNAJ_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Cann_DNAJ <- merge(Cann_DNAJ_Ann, Summary_Pepper_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Cann_DNAJ_TPM <- Cann_DNAJ %>% select(Protein, 4:17)
Cann_DNAJ_TPM <- Cann_DNAJ_TPM[!duplicated(Cann_DNAJ_TPM$Protein),]
Cann_DNAJ_TPM <- Cann_DNAJ_TPM[order(match(Cann_DNAJ_TPM[,1], Cann_DNAJ_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_DNAJ_FC <- Cann_DNAJ %>% select(Protein, 18:19)
Cann_DNAJ_FC <- Cann_DNAJ_FC[!duplicated(Cann_DNAJ_FC$Protein),]
Cann_DNAJ_FC <- Cann_DNAJ_FC[order(match(Cann_DNAJ_FC[,1], Cann_DNAJ_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_DNAJ_DEG <- Cann_DNAJ %>% select(Protein, 20:21)
Cann_DNAJ_DEG <- Cann_DNAJ_DEG[!duplicated(Cann_DNAJ_DEG$Protein),]
Cann_DNAJ_DEG <- Cann_DNAJ_DEG[order(match(Cann_DNAJ_DEG[,1], Cann_DNAJ_Ann[,2])),]

Cann_DNAJ_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Cann_DNAJ_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Cann_DNAJ_DEG$Ripening_Red_DEG <- ifelse(is.na(Cann_DNAJ_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
row.names(Cann_DNAJ_DEG) <- Cann_DNAJ_DEG$Protein
Cann_DNAJ_DEG[1] <- NULL

rownames(Cann_DNAJ_FC) <- Cann_DNAJ_FC$Protein
Cann_DNAJ_FC[1] <- NULL

min(Cann_DNAJ_FC, na.rm = TRUE)
max(Cann_DNAJ_FC, na.rm = TRUE)
breaks_Cann_DNAJ=c(seq(-4.5,0, length.out = 100), seq(0.01,6,length.out = 100))

pheatmap(Cann_DNAJ_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Cann_DNAJ,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_DNAJ_Annotation, annotation_colors=color_DNAJ, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Red"),
         cex = 1,
         angle_col=45,
         main = "DNAJ",
         annotation_legend = TRUE,
         display_numbers = Cann_DNAJ_DEG,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)


rownames(Cann_DNAJ_TPM) <- Cann_DNAJ_TPM$Protein
Cann_DNAJ_TPM[1] <- NULL

min(Cann_DNAJ_TPM, na.rm = TRUE)
max(Cann_DNAJ_TPM, na.rm = TRUE)
breaks_Cann_DNAJ_TPM=c(seq(0,100, length.out = 200))

Cann_DNAJ_TPM_norm <- normalize_df(Cann_DNAJ_TPM)

pheatmap(Cann_DNAJ_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Cann_DNAJ_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_DNAJ_Annotation, annotation_colors=color_DNAJ, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "DNAJ",
         annotation_legend = TRUE,
         #display_numbers = Cann_DNAJ_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(4,9)
)


#########################################################################################################################################
####################################  HSP70  #############################################################################################
#########################################################################################################################################

HSP70_Ath <- read.delim("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan/Funcional Groups/Ath_HSP70.txt")
HSP70_Ath <- merge(HSP70_Ath, Ath_Gene_Models, by.x="GeneID", by.y=1)
HSP70_Ath_fasta <- Ath11_fasta[c(which(names(Ath11_fasta) %in% HSP70_Ath$V2))]
HSP70_Ath_2 <- HSP70_Ath %>% select(V2, Name)
write.fasta(HSP70_Ath_fasta, as.string=TRUE, names= names(HSP70_Ath_fasta),
            file.out = "Funcional Groups/Ath_HSP70.fasta")
rename.fasta(infile = "Funcional Groups/Ath_HSP70.fasta", HSP70_Ath_2, outfile = "Funcional Groups/Ath_HSP70_2.fasta")

HSP70_Niben_Candidates <- read.table("Funcional Groups/Niben_Candidates_HSP70.txt")
HSP70_Niben_Candidates_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% HSP70_Niben_Candidates$V1))]
write.fasta(HSP70_Niben_Candidates_fasta, as.string=TRUE, names= names(HSP70_Niben_Candidates_fasta),
            file.out = "Funcional Groups/HSP70_Niben_Candidates_35.fasta")

HSP70_Sly_Candidates <- read.table("Funcional Groups/Sly_Candidates_HSP70.txt")
HSP70_Sly_Candidates_fasta <- Tomato_4.0_fasta[c(which(names(Tomato_4.0_fasta) %in% HSP70_Sly_Candidates$V1))]
write.fasta(HSP70_Sly_Candidates_fasta, as.string=TRUE, names= names(HSP70_Sly_Candidates_fasta),
            file.out = "Funcional Groups/HSP70_Sly_Candidates_35.fasta")

HSP70_Cann_Candidates <- read.table("Funcional Groups/Cann_Candidates_HSP70.txt")
HSP70_Cann_Candidates_fasta <- Cannuum_ASM51225v2_fasta[c(which(names(Cannuum_ASM51225v2_fasta) %in% HSP70_Cann_Candidates$V1))]
write.fasta(HSP70_Cann_Candidates_fasta, as.string=TRUE, names= names(HSP70_Cann_Candidates_fasta),
            file.out = "Funcional Groups/HSP70_Cann_Candidates_35.fasta")

Niben_HSP70_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_HSP70_Ann$Name))]
write.fasta(Niben_HSP70_fasta, as.string=TRUE, names= names(Niben_HSP70_fasta),
            file.out = "Funcional Groups/Niben_HSP70.fasta")
Niben_HSP70_Ann_2 <- Niben_HSP70_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_HSP70.fasta", Niben_HSP70_Ann, outfile = "Funcional Groups/Niben_HSP70_2.fasta")

save_heatmaps("HSP70", height = 10)

## Representing Heatmap

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Niben_HSP70_Ann <- read.table("Funcional Groups/FigTree/Niben_HSP70.txt",
                              sep = "\t", header = TRUE)

colnames(Niben_HSP70_Ann) <- c("Name", "Protein")

Niben_HSP70_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_HSP70_Ann$Name))]
write.fasta(Niben_HSP70_fasta, as.string=TRUE, names= names(Niben_HSP70_fasta),
            file.out = "Funcional Groups/Niben_HSP70.fasta")
Niben_HSP70_Ann_2 <- Niben_HSP70_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_HSP70.fasta", Niben_HSP70_Ann, outfile = "Funcional Groups/Niben_HSP70_2.fasta")

Niben_HSP70_Chloro <- read.delim("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan/Funcional Groups/Niben_HSP70_2_summary.targetp2", 
                                 header=FALSE, comment.char="#")
colnames(Niben_HSP70_Chloro) <- c("ID",	"Prediction",	"noTP",	"SP",	"mTP",	"cTP",	"luTP",	"CS-Position")
Niben_HSP70_Chloro$Prediction <- factor(Niben_HSP70_Chloro$Prediction, levels = c("cTP","mTP","noTP"))
Niben_HSP70_Chloro <- merge(Niben_HSP70_Ann, Niben_HSP70_Chloro, by.x = "Protein", by.y = "ID")

Niben_HSP70_Chloro <- Niben_HSP70_Chloro[order(match(Niben_HSP70_Chloro$Protein, Niben_HSP70_Ann$Protein)),]
rownames(Niben_HSP70_Chloro) <- Niben_HSP70_Chloro$Protein
Niben_HSP70_Chloro <- Niben_HSP70_Chloro %>% select(Prediction)


## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_HSP70 <- merge(Niben_HSP70_Ann, TPM_All_FC_Annotated, by="Name")

## We create the annotation for Modules

Niben_HSP70_Modules <- which_Module_pwd30(Niben_HSP70_Ann)

## We merge this annotation of Modules with the annotation of Chloroplast TP prediction

Niben_HSP70_Modules_Chl <- merge(Niben_HSP70_Modules, Niben_HSP70_Chloro, by = 0)
Niben_HSP70_Modules_Chl <- Niben_HSP70_Modules_Chl[order(match(Niben_HSP70_Modules_Chl$Row.names, Niben_HSP70_Ann$Protein)),]
rownames(Niben_HSP70_Modules_Chl) <- Niben_HSP70_Modules_Chl$Row.names
Niben_HSP70_Modules_Chl$Row.names <- NULL

## We select only the FC data from the different comparisons
Niben_HSP70_TPM <- Niben_HSP70 %>% select(Protein, 4:51, 61:66)
Niben_HSP70_TPM <- Niben_HSP70_TPM[!duplicated(Niben_HSP70_TPM$Protein),]
Niben_HSP70_TPM <- Niben_HSP70_TPM[order(match(Niben_HSP70_TPM[,1], Niben_HSP70_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_HSP70_FC <- Niben_HSP70 %>% select(Protein, 71, 73:87)
Niben_HSP70_FC <- Niben_HSP70_FC[!duplicated(Niben_HSP70_FC$Protein),]
Niben_HSP70_FC <- Niben_HSP70_FC[order(match(Niben_HSP70_FC[,1], Niben_HSP70_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_HSP70_DEG <- Niben_HSP70 %>% select(Protein, 89, 91:105)
Niben_HSP70_DEG <- Niben_HSP70_DEG[!duplicated(Niben_HSP70_DEG$Protein),]
Niben_HSP70_DEG <- Niben_HSP70_DEG[order(match(Niben_HSP70_DEG[,1], Niben_HSP70_Ann[,2])),]

Niben_HSP70_DEG$pcrtB22h_vs_GFP22h_filtered <- ifelse(is.na(Niben_HSP70_DEG$pcrtB22h_vs_GFP22h_filtered) == TRUE, " ", "*")
Niben_HSP70_DEG$pcrtB25h_vs_GFP25h_filtered <- ifelse(is.na(Niben_HSP70_DEG$pcrtB25h_vs_GFP25h_filtered) == TRUE, " ", "*")
Niben_HSP70_DEG$pcrtBAA_vs_GFPAA_filtered <- ifelse(is.na(Niben_HSP70_DEG$pcrtBAA_vs_GFPAA_filtered) == TRUE, " ", "*")
Niben_HSP70_DEG$pcrtBBB_vs_GFPBB_filtered <- ifelse(is.na(Niben_HSP70_DEG$pcrtBBB_vs_GFPBB_filtered) == TRUE, " ", "*")
Niben_HSP70_DEG$pcrtBYY_vs_GFPYY_filtered <- ifelse(is.na(Niben_HSP70_DEG$pcrtBYY_vs_GFPYY_filtered) == TRUE, " ", "*")
Niben_HSP70_DEG$pcrtB28h_vs_GFP28h_filtered <- ifelse(is.na(Niben_HSP70_DEG$pcrtB28h_vs_GFP28h_filtered) == TRUE, " ", "*")
Niben_HSP70_DEG$pcrtB34h_vs_GFP34h_filtered <- ifelse(is.na(Niben_HSP70_DEG$pcrtB34h_vs_GFP34h_filtered) == TRUE, " ", "*")
Niben_HSP70_DEG$pcrtB37h_vs_GFP37h_filtered <- ifelse(is.na(Niben_HSP70_DEG$pcrtB37h_vs_GFP37h_filtered) == TRUE, " ", "*")
Niben_HSP70_DEG$pcrtBZZ_vs_GFPZZ_filtered <- ifelse(is.na(Niben_HSP70_DEG$pcrtBZZ_vs_GFPZZ_filtered) == TRUE, " ", "*")
Niben_HSP70_DEG$pcrtBCC_vs_GFPCC_filtered <- ifelse(is.na(Niben_HSP70_DEG$pcrtBCC_vs_GFPCC_filtered) == TRUE, " ", "*")
Niben_HSP70_DEG$pcrtB40h_vs_GFP40h_filtered <- ifelse(is.na(Niben_HSP70_DEG$pcrtB40h_vs_GFP40h_filtered) == TRUE, " ", "*")
Niben_HSP70_DEG$pcrtBDD_vs_GFPDD_filtered <- ifelse(is.na(Niben_HSP70_DEG$pcrtBDD_vs_GFPDD_filtered) == TRUE, " ", "*")
Niben_HSP70_DEG$pcrtB46h_vs_GFP46h_filtered <- ifelse(is.na(Niben_HSP70_DEG$pcrtB46h_vs_GFP46h_filtered) == TRUE, " ", "*")
Niben_HSP70_DEG$pcrtB56h_vs_GFP56h_filtered <- ifelse(is.na(Niben_HSP70_DEG$pcrtB56h_vs_GFP56h_filtered) == TRUE, " ", "*")
Niben_HSP70_DEG$pcrtBEE_vs_GFPEE_filtered <- ifelse(is.na(Niben_HSP70_DEG$pcrtBEE_vs_GFPEE_filtered) == TRUE, " ", "*")
Niben_HSP70_DEG$pcrtB96h_vs_GFP96h_filtered <- ifelse(is.na(Niben_HSP70_DEG$pcrtB96h_vs_GFP96h_filtered) == TRUE, " ", "*")
row.names(Niben_HSP70_DEG) <- Niben_HSP70_DEG$Protein
Niben_HSP70_DEG[1] <- NULL

Niben_HSP70_DEG <- Niben_HSP70_DEG %>%
  select(pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

row.names(Niben_HSP70_FC) <- Niben_HSP70_FC$Protein
Niben_HSP70_FC[1] <- NULL

Niben_HSP70_FC <- Niben_HSP70_FC %>%
  select(pcrtB22h_vs_GFP22h_All, pcrtBAA_vs_GFPAA_All, pcrtB25h_vs_GFP25h_All, 
         pcrtBYY_vs_GFPYY_All, pcrtBBB_vs_GFPBB_All, pcrtB28h_vs_GFP28h_All, pcrtB34h_vs_GFP34h_All, pcrtBZZ_vs_GFPZZ_All, 
         pcrtB37h_vs_GFP37h_All, pcrtBCC_vs_GFPCC_All, pcrtB40h_vs_GFP40h_All, pcrtBDD_vs_GFPDD_All, pcrtB46h_vs_GFP46h_All, 
         pcrtBEE_vs_GFPEE_All, pcrtB56h_vs_GFP56h_All, pcrtB96h_vs_GFP96h_All)


min(Niben_HSP70_FC, na.rm = TRUE)
max(Niben_HSP70_FC, na.rm = TRUE)
breaks_HSP70_FC=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Niben_HSP70_FC, # %>% select(1:15), 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_HSP70_FC,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_HSP70_Annotation, annotation_colors=color_HSP70, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h", "96h"),
         cex = 1,
         angle_col=45,
         main = "HSP70",
         annotation_legend = TRUE,
         display_numbers = Niben_HSP70_DEG, # %>% select(1:15),
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(84, 104),
         annotation_row = Niben_HSP70_Modules_Chl,
         annotation_colors = module_Chl_colors
         #gaps_col = c(15)
)


rownames(Niben_HSP70_TPM) <- Niben_HSP70_TPM$Protein
Niben_HSP70_TPM[1] <- NULL

min(Niben_HSP70_TPM, na.rm = TRUE)
max(Niben_HSP70_TPM, na.rm = TRUE)
breaks_Niben_HSP70_TPM=c(seq(0,100, length.out = 100))

Niben_HSP70_TPM_norm <- normalize_df(Niben_HSP70_TPM)
Niben_HSP70_TPM_norm_Cornell <- Niben_HSP70_TPM %>%
  select(1:48) 
Niben_HSP70_TPM_norm_Cornell <- normalize_df(Niben_HSP70_TPM_norm_Cornell)


pheatmap(Niben_HSP70_TPM, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         breaks = breaks_Niben_HSP70_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_HSP70_Annotation, annotation_colors=color_HSP70, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "HSP70",
         annotation_legend = TRUE,
         #display_numbers = Niben_HSP70_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         annotation_row = Niben_HSP70_Modules_Chl,
         annotation_colors = module_Chl_colors,
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(24)#, 48, 51)
)



#####Representing Heatmap for Sly

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Sly_HSP70_Ann <- read.table("Funcional Groups/FigTree/Sly_HSP70.txt",
                            sep = "\t", header = TRUE)

colnames(Sly_HSP70_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Sly_HSP70 <- merge(Sly_HSP70_Ann, Summary_Yazdani_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Sly_HSP70_TPM <- Sly_HSP70 %>% select(Protein, 4:66)
Sly_HSP70_TPM <- Sly_HSP70_TPM[!duplicated(Sly_HSP70_TPM$Protein),]
Sly_HSP70_TPM <- Sly_HSP70_TPM[order(match(Sly_HSP70_TPM[,1], Sly_HSP70_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_HSP70_FC <- Sly_HSP70 %>% select(Protein, 67:71)
Sly_HSP70_FC <- Sly_HSP70_FC[!duplicated(Sly_HSP70_FC$Protein),]
Sly_HSP70_FC <- Sly_HSP70_FC[order(match(Sly_HSP70_FC[,1], Sly_HSP70_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_HSP70_DEG <- Sly_HSP70 %>% select(Protein, 72:76)
Sly_HSP70_DEG <- Sly_HSP70_DEG[!duplicated(Sly_HSP70_DEG$Protein),]
Sly_HSP70_DEG <- Sly_HSP70_DEG[order(match(Sly_HSP70_DEG[,1], Sly_HSP70_Ann[,2])),]

Sly_HSP70_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Sly_HSP70_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Sly_HSP70_DEG$Ripening_Orange_DEG <- ifelse(is.na(Sly_HSP70_DEG$Ripening_Orange_DEG) == TRUE, " ", "*")
Sly_HSP70_DEG$Ripening_Red_DEG <- ifelse(is.na(Sly_HSP70_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
Sly_HSP70_DEG$ORHis_MG_DEG <- ifelse(is.na(Sly_HSP70_DEG$ORHis_MG_DEG) == TRUE, " ", "*")
Sly_HSP70_DEG$ORHis_Breaker_DEG <- ifelse(is.na(Sly_HSP70_DEG$ORHis_Breaker_DEG) == TRUE, " ", "*")
row.names(Sly_HSP70_DEG) <- Sly_HSP70_DEG$Protein
Sly_HSP70_DEG[1] <- NULL

rownames(Sly_HSP70_FC) <- Sly_HSP70_FC$Protein
Sly_HSP70_FC[1] <- NULL

min(Sly_HSP70_FC, na.rm = TRUE)
max(Sly_HSP70_FC, na.rm = TRUE)
breaks_Sly_HSP70=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Sly_HSP70_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Sly_HSP70,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_HSP70_Annotation, annotation_colors=color_HSP70, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "HSP70",
         annotation_legend = TRUE,
         display_numbers = Sly_HSP70_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(39, 48),
         gaps_col = c(3)
)


rownames(Sly_HSP70_TPM) <- Sly_HSP70_TPM$Protein
Sly_HSP70_TPM[1] <- NULL

min(Sly_HSP70_TPM, na.rm = TRUE)
max(Sly_HSP70_TPM, na.rm = TRUE)
breaks_Sly_HSP70_TPM=c(seq(0,100, length.out = 200))

Sly_HSP70_TPM_norm <- normalize_df(Sly_HSP70_TPM)

pheatmap(Sly_HSP70_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Sly_HSP70_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_HSP70_Annotation, annotation_colors=color_HSP70, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "HSP70",
         annotation_legend = TRUE,
         #display_numbers = Sly_HSP70_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(21,42)
)


#####Representing Heatmap for Cann

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Cann_HSP70_Ann <- read.table("Funcional Groups/FigTree/Cann_HSP70.txt",
                             sep = "\t", header = TRUE)

colnames(Cann_HSP70_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Cann_HSP70 <- merge(Cann_HSP70_Ann, Summary_Pepper_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Cann_HSP70_TPM <- Cann_HSP70 %>% select(Protein, 4:17)
Cann_HSP70_TPM <- Cann_HSP70_TPM[!duplicated(Cann_HSP70_TPM$Protein),]
Cann_HSP70_TPM <- Cann_HSP70_TPM[order(match(Cann_HSP70_TPM[,1], Cann_HSP70_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_HSP70_FC <- Cann_HSP70 %>% select(Protein, 18:19)
Cann_HSP70_FC <- Cann_HSP70_FC[!duplicated(Cann_HSP70_FC$Protein),]
Cann_HSP70_FC <- Cann_HSP70_FC[order(match(Cann_HSP70_FC[,1], Cann_HSP70_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_HSP70_DEG <- Cann_HSP70 %>% select(Protein, 20:21)
Cann_HSP70_DEG <- Cann_HSP70_DEG[!duplicated(Cann_HSP70_DEG$Protein),]
Cann_HSP70_DEG <- Cann_HSP70_DEG[order(match(Cann_HSP70_DEG[,1], Cann_HSP70_Ann[,2])),]

Cann_HSP70_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Cann_HSP70_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Cann_HSP70_DEG$Ripening_Red_DEG <- ifelse(is.na(Cann_HSP70_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
row.names(Cann_HSP70_DEG) <- Cann_HSP70_DEG$Protein
Cann_HSP70_DEG[1] <- NULL

rownames(Cann_HSP70_FC) <- Cann_HSP70_FC$Protein
Cann_HSP70_FC[1] <- NULL

min(Cann_HSP70_FC, na.rm = TRUE)
max(Cann_HSP70_FC, na.rm = TRUE)
breaks_Cann_HSP70=c(seq(-4.5,0, length.out = 100), seq(0.01,6,length.out = 100))

pheatmap(Cann_HSP70_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Cann_HSP70,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_HSP70_Annotation, annotation_colors=color_HSP70, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Red"),
         cex = 1,
         angle_col=45,
         main = "HSP70",
         annotation_legend = TRUE,
         display_numbers = Cann_HSP70_DEG,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)


rownames(Cann_HSP70_TPM) <- Cann_HSP70_TPM$Protein
Cann_HSP70_TPM[1] <- NULL

min(Cann_HSP70_TPM, na.rm = TRUE)
max(Cann_HSP70_TPM, na.rm = TRUE)
breaks_Cann_HSP70_TPM=c(seq(0,100, length.out = 200))

Cann_HSP70_TPM_norm <- normalize_df(Cann_HSP70_TPM)

pheatmap(Cann_HSP70_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Cann_HSP70_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_HSP70_Annotation, annotation_colors=color_HSP70, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "HSP70",
         annotation_legend = TRUE,
         #display_numbers = Cann_HSP70_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(4,9)
)


## Comparativa 96h_SlyYaz_Ath

Ath_HSP70_Ann <- read.table("Funcional Groups/Ath_HSP70.txt", header = TRUE)

Ath_HSP70_FC <- merge(Ath_HSP70_Ann, Summary_Ath_FC, by.x=1, by.y = 0, all.x = TRUE) %>%
  dplyr::select(Name, log2FoldChange)
Ath_HSP70_FC <- Ath_HSP70_FC[order(match(Ath_HSP70_FC[,1], Ath_HSP70_Ann[,2])),]

Ath_HSP70_DEG <- merge(Ath_HSP70_Ann, Summary_Ath_DEGs, by.x=1, by.y = 1, all.x = TRUE) %>%
  dplyr::select(Name, log2FC)
Ath_HSP70_DEG <- Ath_HSP70_DEG[order(match(Ath_HSP70_DEG[,1], Ath_HSP70_Ann[,2])),]

Ath_HSP70_DEG$log2FC <- ifelse(is.na(Ath_HSP70_DEG$log2FC) == TRUE, " ", "*")
row.names(Ath_HSP70_DEG) <- Ath_HSP70_DEG$Name
Ath_HSP70_DEG[1] <- NULL

rownames(Ath_HSP70_FC) <- Ath_HSP70_FC$Name
Ath_HSP70_FC[1] <- NULL

min(Ath_HSP70_FC, na.rm = TRUE)
max(Ath_HSP70_FC, na.rm = TRUE)
breaks_Ath_HSP70=c(seq(-4.5,0, length.out = 100), seq(0.01,4.5,length.out = 100))

pheatmap(Ath_HSP70_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Stndr,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_HSP70_Annotation, annotation_colors=color_HSP70, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = "Senescence",
         cex = 1,
         angle_col=45,
         main = "HSP70",
         annotation_legend = TRUE,
         display_numbers = Ath_HSP70_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         gaps_row = c(3, 4, 5, 6, 7, 8, 9, 10)
         #gaps_col = c(2,5)
)

pheatmap(Niben_HSP70_FC %>% select(16), 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         #breaks = breaks_HSP70_FC,
         breaks = breaks_Stndr,
         #legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         #legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_HSP70_Annotation, annotation_colors=color_HSP70, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = "96h",
         cex = 1,
         angle_col=45,
         main = "HSP70",
         annotation_legend = TRUE,
         display_numbers = Niben_HSP70_DEG %>% select(16),
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(84, 104),
         annotation_row = Niben_HSP70_Modules_Chl,
         annotation_colors = module_Chl_colors_2,
         gaps_row = c(3, 5, 9, 12, 15, 17)
)


pheatmap(Sly_HSP70_FC %>% select(2,3), 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Stndr,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         #legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_HSP70_Annotation, annotation_colors=color_HSP70, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Orange", "Ripening_Red"),
         cex = 1,
         angle_col=45,
         main = "HSP70",
         annotation_legend = TRUE,
         display_numbers = Sly_HSP70_DEG %>% select(2,3),
         fontsize_number = 15,
         na_col = "darkgrey",
         gaps_row = c(1, 3,  4, 7, 8)
)


#########################################################################################################################################
####################################  BAG  #############################################################################################
#########################################################################################################################################

BAG_Ath <- read.delim("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan/Funcional Groups/Ath_BAG.txt")
BAG_Ath <- merge(BAG_Ath, Ath_Gene_Models, by.x="GeneID", by.y=1)
BAG_Ath_fasta <- Ath11_fasta[c(which(names(Ath11_fasta) %in% BAG_Ath$V2))]
BAG_Ath_2 <- BAG_Ath %>% select(V2, Name)
write.fasta(BAG_Ath_fasta, as.string=TRUE, names= names(BAG_Ath_fasta),
            file.out = "Funcional Groups/Ath_BAG.fasta")
rename.fasta(infile = "Funcional Groups/Ath_BAG.fasta", BAG_Ath_2, outfile = "Funcional Groups/Ath_BAG_2.fasta")

BAG_Niben_Candidates <- read.table("Funcional Groups/Niben_Candidates_BAG.txt")
BAG_Niben_Candidates_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% BAG_Niben_Candidates$V1))]
write.fasta(BAG_Niben_Candidates_fasta, as.string=TRUE, names= names(BAG_Niben_Candidates_fasta),
            file.out = "Funcional Groups/BAG_Niben_Candidates_35.fasta")

BAG_Sly_Candidates <- read.table("Funcional Groups/Sly_Candidates_BAG.txt")
BAG_Sly_Candidates_fasta <- Tomato_4.0_fasta[c(which(names(Tomato_4.0_fasta) %in% BAG_Sly_Candidates$V1))]
write.fasta(BAG_Sly_Candidates_fasta, as.string=TRUE, names= names(BAG_Sly_Candidates_fasta),
            file.out = "Funcional Groups/BAG_Sly_Candidates_35.fasta")

BAG_Cann_Candidates <- read.table("Funcional Groups/Cann_Candidates_BAG.txt")
BAG_Cann_Candidates_fasta <- Cannuum_ASM51225v2_fasta[c(which(names(Cannuum_ASM51225v2_fasta) %in% BAG_Cann_Candidates$V1))]
write.fasta(BAG_Cann_Candidates_fasta, as.string=TRUE, names= names(BAG_Cann_Candidates_fasta),
            file.out = "Funcional Groups/BAG_Cann_Candidates_35.fasta")

Niben_BAG_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_BAG_Ann$Name))]
write.fasta(Niben_BAG_fasta, as.string=TRUE, names= names(Niben_BAG_fasta),
            file.out = "Funcional Groups/Niben_BAG.fasta")
Niben_BAG_Ann_2 <- Niben_BAG_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_BAG.fasta", Niben_BAG_Ann, outfile = "Funcional Groups/Niben_BAG_2.fasta")

save_heatmaps("BAG", height = 5)


## Representing Heatmap

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Niben_BAG_Ann <- read.table("Funcional Groups/FigTree/Niben_BAG.txt",
                              sep = "\t", header = TRUE)

colnames(Niben_BAG_Ann) <- c("Name", "Protein")

Niben_BAG_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_BAG_Ann$Name))]
write.fasta(Niben_BAG_fasta, as.string=TRUE, names= names(Niben_BAG_fasta),
            file.out = "Funcional Groups/Niben_BAG.fasta")
Niben_BAG_Ann_2 <- Niben_BAG_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_BAG.fasta", Niben_BAG_Ann, outfile = "Funcional Groups/Niben_BAG_2.fasta")

Niben_BAG_Chloro <- read.delim("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan/Funcional Groups/Niben_BAG_2_summary.targetp2", 
                                 header=FALSE, comment.char="#")
colnames(Niben_BAG_Chloro) <- c("ID",	"Prediction",	"noTP",	"SP",	"mTP",	"cTP",	"luTP",	"CS-Position")
Niben_BAG_Chloro$Prediction <- factor(Niben_BAG_Chloro$Prediction, levels = c("cTP","mTP","noTP"))
Niben_BAG_Chloro <- merge(Niben_BAG_Ann, Niben_BAG_Chloro, by.x = "Protein", by.y = "ID")

Niben_BAG_Chloro <- Niben_BAG_Chloro[order(match(Niben_BAG_Chloro$Protein, Niben_BAG_Ann$Protein)),]
rownames(Niben_BAG_Chloro) <- Niben_BAG_Chloro$Protein
Niben_BAG_Chloro <- Niben_BAG_Chloro %>% select(Prediction)


## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_BAG <- merge(Niben_BAG_Ann, TPM_All_FC_Annotated, by="Name")

## We create the annotation for Modules

Niben_BAG_Modules <- which_Module_pwd30(Niben_BAG_Ann)

## We merge this annotation of Modules with the annotation of Chloroplast TP prediction

Niben_BAG_Modules_Chl <- merge(Niben_BAG_Modules, Niben_BAG_Chloro, by = 0)
Niben_BAG_Modules_Chl <- Niben_BAG_Modules_Chl[order(match(Niben_BAG_Modules_Chl$Row.names, Niben_BAG_Ann$Protein)),]
rownames(Niben_BAG_Modules_Chl) <- Niben_BAG_Modules_Chl$Row.names
Niben_BAG_Modules_Chl$Row.names <- NULL

## We select only the FC data from the different comparisons
Niben_BAG_TPM <- Niben_BAG %>% select(Protein, 4:51, 61:66)
Niben_BAG_TPM <- Niben_BAG_TPM[!duplicated(Niben_BAG_TPM$Protein),]
Niben_BAG_TPM <- Niben_BAG_TPM[order(match(Niben_BAG_TPM[,1], Niben_BAG_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_BAG_FC <- Niben_BAG %>% select(Protein, 71, 73:87)
Niben_BAG_FC <- Niben_BAG_FC[!duplicated(Niben_BAG_FC$Protein),]
Niben_BAG_FC <- Niben_BAG_FC[order(match(Niben_BAG_FC[,1], Niben_BAG_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_BAG_DEG <- Niben_BAG %>% select(Protein, 89, 91:105)
Niben_BAG_DEG <- Niben_BAG_DEG[!duplicated(Niben_BAG_DEG$Protein),]
Niben_BAG_DEG <- Niben_BAG_DEG[order(match(Niben_BAG_DEG[,1], Niben_BAG_Ann[,2])),]

Niben_BAG_DEG$pcrtB22h_vs_GFP22h_filtered <- ifelse(is.na(Niben_BAG_DEG$pcrtB22h_vs_GFP22h_filtered) == TRUE, " ", "*")
Niben_BAG_DEG$pcrtB25h_vs_GFP25h_filtered <- ifelse(is.na(Niben_BAG_DEG$pcrtB25h_vs_GFP25h_filtered) == TRUE, " ", "*")
Niben_BAG_DEG$pcrtBAA_vs_GFPAA_filtered <- ifelse(is.na(Niben_BAG_DEG$pcrtBAA_vs_GFPAA_filtered) == TRUE, " ", "*")
Niben_BAG_DEG$pcrtBBB_vs_GFPBB_filtered <- ifelse(is.na(Niben_BAG_DEG$pcrtBBB_vs_GFPBB_filtered) == TRUE, " ", "*")
Niben_BAG_DEG$pcrtBYY_vs_GFPYY_filtered <- ifelse(is.na(Niben_BAG_DEG$pcrtBYY_vs_GFPYY_filtered) == TRUE, " ", "*")
Niben_BAG_DEG$pcrtB28h_vs_GFP28h_filtered <- ifelse(is.na(Niben_BAG_DEG$pcrtB28h_vs_GFP28h_filtered) == TRUE, " ", "*")
Niben_BAG_DEG$pcrtB34h_vs_GFP34h_filtered <- ifelse(is.na(Niben_BAG_DEG$pcrtB34h_vs_GFP34h_filtered) == TRUE, " ", "*")
Niben_BAG_DEG$pcrtB37h_vs_GFP37h_filtered <- ifelse(is.na(Niben_BAG_DEG$pcrtB37h_vs_GFP37h_filtered) == TRUE, " ", "*")
Niben_BAG_DEG$pcrtBZZ_vs_GFPZZ_filtered <- ifelse(is.na(Niben_BAG_DEG$pcrtBZZ_vs_GFPZZ_filtered) == TRUE, " ", "*")
Niben_BAG_DEG$pcrtBCC_vs_GFPCC_filtered <- ifelse(is.na(Niben_BAG_DEG$pcrtBCC_vs_GFPCC_filtered) == TRUE, " ", "*")
Niben_BAG_DEG$pcrtB40h_vs_GFP40h_filtered <- ifelse(is.na(Niben_BAG_DEG$pcrtB40h_vs_GFP40h_filtered) == TRUE, " ", "*")
Niben_BAG_DEG$pcrtBDD_vs_GFPDD_filtered <- ifelse(is.na(Niben_BAG_DEG$pcrtBDD_vs_GFPDD_filtered) == TRUE, " ", "*")
Niben_BAG_DEG$pcrtB46h_vs_GFP46h_filtered <- ifelse(is.na(Niben_BAG_DEG$pcrtB46h_vs_GFP46h_filtered) == TRUE, " ", "*")
Niben_BAG_DEG$pcrtB56h_vs_GFP56h_filtered <- ifelse(is.na(Niben_BAG_DEG$pcrtB56h_vs_GFP56h_filtered) == TRUE, " ", "*")
Niben_BAG_DEG$pcrtBEE_vs_GFPEE_filtered <- ifelse(is.na(Niben_BAG_DEG$pcrtBEE_vs_GFPEE_filtered) == TRUE, " ", "*")
Niben_BAG_DEG$pcrtB96h_vs_GFP96h_filtered <- ifelse(is.na(Niben_BAG_DEG$pcrtB96h_vs_GFP96h_filtered) == TRUE, " ", "*")
row.names(Niben_BAG_DEG) <- Niben_BAG_DEG$Protein
Niben_BAG_DEG[1] <- NULL

Niben_BAG_DEG <- Niben_BAG_DEG %>%
  select(pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

row.names(Niben_BAG_FC) <- Niben_BAG_FC$Protein
Niben_BAG_FC[1] <- NULL

Niben_BAG_FC <- Niben_BAG_FC %>%
  select(pcrtB22h_vs_GFP22h_All, pcrtBAA_vs_GFPAA_All, pcrtB25h_vs_GFP25h_All, 
         pcrtBYY_vs_GFPYY_All, pcrtBBB_vs_GFPBB_All, pcrtB28h_vs_GFP28h_All, pcrtB34h_vs_GFP34h_All, pcrtBZZ_vs_GFPZZ_All, 
         pcrtB37h_vs_GFP37h_All, pcrtBCC_vs_GFPCC_All, pcrtB40h_vs_GFP40h_All, pcrtBDD_vs_GFPDD_All, pcrtB46h_vs_GFP46h_All, 
         pcrtBEE_vs_GFPEE_All, pcrtB56h_vs_GFP56h_All, pcrtB96h_vs_GFP96h_All)


min(Niben_BAG_FC, na.rm = TRUE)
max(Niben_BAG_FC, na.rm = TRUE)
breaks_BAG_FC=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Niben_BAG_FC, # %>% select(1:15), 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_BAG_FC,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_BAG_Annotation, annotation_colors=color_BAG, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h", "96h"),
         cex = 1,
         angle_col=45,
         main = "BAG",
         annotation_legend = TRUE,
         display_numbers = Niben_BAG_DEG, # %>% select(1:15),
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(84, 104),
         annotation_row = Niben_BAG_Modules_Chl,
         annotation_colors = module_Chl_colors
         #gaps_col = c(15)
)


rownames(Niben_BAG_TPM) <- Niben_BAG_TPM$Protein
Niben_BAG_TPM[1] <- NULL

min(Niben_BAG_TPM, na.rm = TRUE)
max(Niben_BAG_TPM, na.rm = TRUE)
breaks_Niben_BAG_TPM=c(seq(0,100, length.out = 100))

Niben_BAG_TPM_norm <- normalize_df(Niben_BAG_TPM)
Niben_BAG_TPM_norm_Cornell <- Niben_BAG_TPM %>%
  select(1:48) 
Niben_BAG_TPM_norm_Cornell <- normalize_df(Niben_BAG_TPM_norm_Cornell)


pheatmap(Niben_BAG_TPM, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         breaks = breaks_Niben_BAG_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_BAG_Annotation, annotation_colors=color_BAG, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "BAG",
         annotation_legend = TRUE,
         #display_numbers = Niben_BAG_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         annotation_row = Niben_BAG_Modules_Chl,
         annotation_colors = module_Chl_colors,
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(24)#, 48, 51)
)



#####Representing Heatmap for Sly

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Sly_BAG_Ann <- read.table("Funcional Groups/FigTree/Sly_BAG.txt",
                            sep = "\t", header = TRUE)

colnames(Sly_BAG_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Sly_BAG <- merge(Sly_BAG_Ann, Summary_Yazdani_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Sly_BAG_TPM <- Sly_BAG %>% select(Protein, 4:66)
Sly_BAG_TPM <- Sly_BAG_TPM[!duplicated(Sly_BAG_TPM$Protein),]
Sly_BAG_TPM <- Sly_BAG_TPM[order(match(Sly_BAG_TPM[,1], Sly_BAG_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_BAG_FC <- Sly_BAG %>% select(Protein, 67:71)
Sly_BAG_FC <- Sly_BAG_FC[!duplicated(Sly_BAG_FC$Protein),]
Sly_BAG_FC <- Sly_BAG_FC[order(match(Sly_BAG_FC[,1], Sly_BAG_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_BAG_DEG <- Sly_BAG %>% select(Protein, 72:76)
Sly_BAG_DEG <- Sly_BAG_DEG[!duplicated(Sly_BAG_DEG$Protein),]
Sly_BAG_DEG <- Sly_BAG_DEG[order(match(Sly_BAG_DEG[,1], Sly_BAG_Ann[,2])),]

Sly_BAG_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Sly_BAG_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Sly_BAG_DEG$Ripening_Orange_DEG <- ifelse(is.na(Sly_BAG_DEG$Ripening_Orange_DEG) == TRUE, " ", "*")
Sly_BAG_DEG$Ripening_Red_DEG <- ifelse(is.na(Sly_BAG_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
Sly_BAG_DEG$ORHis_MG_DEG <- ifelse(is.na(Sly_BAG_DEG$ORHis_MG_DEG) == TRUE, " ", "*")
Sly_BAG_DEG$ORHis_Breaker_DEG <- ifelse(is.na(Sly_BAG_DEG$ORHis_Breaker_DEG) == TRUE, " ", "*")
row.names(Sly_BAG_DEG) <- Sly_BAG_DEG$Protein
Sly_BAG_DEG[1] <- NULL

rownames(Sly_BAG_FC) <- Sly_BAG_FC$Protein
Sly_BAG_FC[1] <- NULL

min(Sly_BAG_FC, na.rm = TRUE)
max(Sly_BAG_FC, na.rm = TRUE)
breaks_Sly_BAG=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Sly_BAG_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Sly_BAG,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_BAG_Annotation, annotation_colors=color_BAG, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "BAG",
         annotation_legend = TRUE,
         display_numbers = Sly_BAG_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(39, 48),
         gaps_col = c(3)
)


rownames(Sly_BAG_TPM) <- Sly_BAG_TPM$Protein
Sly_BAG_TPM[1] <- NULL

min(Sly_BAG_TPM, na.rm = TRUE)
max(Sly_BAG_TPM, na.rm = TRUE)
breaks_Sly_BAG_TPM=c(seq(0,100, length.out = 200))

Sly_BAG_TPM_norm <- normalize_df(Sly_BAG_TPM)

pheatmap(Sly_BAG_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Sly_BAG_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_BAG_Annotation, annotation_colors=color_BAG, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "BAG",
         annotation_legend = TRUE,
         #display_numbers = Sly_BAG_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(21,42)
)


#####Representing Heatmap for Cann

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Cann_BAG_Ann <- read.table("Funcional Groups/FigTree/Cann_BAG.txt",
                             sep = "\t", header = TRUE)

colnames(Cann_BAG_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Cann_BAG <- merge(Cann_BAG_Ann, Summary_Pepper_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Cann_BAG_TPM <- Cann_BAG %>% select(Protein, 4:17)
Cann_BAG_TPM <- Cann_BAG_TPM[!duplicated(Cann_BAG_TPM$Protein),]
Cann_BAG_TPM <- Cann_BAG_TPM[order(match(Cann_BAG_TPM[,1], Cann_BAG_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_BAG_FC <- Cann_BAG %>% select(Protein, 18:19)
Cann_BAG_FC <- Cann_BAG_FC[!duplicated(Cann_BAG_FC$Protein),]
Cann_BAG_FC <- Cann_BAG_FC[order(match(Cann_BAG_FC[,1], Cann_BAG_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_BAG_DEG <- Cann_BAG %>% select(Protein, 20:21)
Cann_BAG_DEG <- Cann_BAG_DEG[!duplicated(Cann_BAG_DEG$Protein),]
Cann_BAG_DEG <- Cann_BAG_DEG[order(match(Cann_BAG_DEG[,1], Cann_BAG_Ann[,2])),]

Cann_BAG_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Cann_BAG_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Cann_BAG_DEG$Ripening_Red_DEG <- ifelse(is.na(Cann_BAG_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
row.names(Cann_BAG_DEG) <- Cann_BAG_DEG$Protein
Cann_BAG_DEG[1] <- NULL

rownames(Cann_BAG_FC) <- Cann_BAG_FC$Protein
Cann_BAG_FC[1] <- NULL

min(Cann_BAG_FC, na.rm = TRUE)
max(Cann_BAG_FC, na.rm = TRUE)
breaks_Cann_BAG=c(seq(-4.5,0, length.out = 100), seq(0.01,6,length.out = 100))

pheatmap(Cann_BAG_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Cann_BAG,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_BAG_Annotation, annotation_colors=color_BAG, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Red"),
         cex = 1,
         angle_col=45,
         main = "BAG",
         annotation_legend = TRUE,
         display_numbers = Cann_BAG_DEG,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)


rownames(Cann_BAG_TPM) <- Cann_BAG_TPM$Protein
Cann_BAG_TPM[1] <- NULL

min(Cann_BAG_TPM, na.rm = TRUE)
max(Cann_BAG_TPM, na.rm = TRUE)
breaks_Cann_BAG_TPM=c(seq(0,100, length.out = 200))

Cann_BAG_TPM_norm <- normalize_df(Cann_BAG_TPM)

pheatmap(Cann_BAG_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Cann_BAG_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_BAG_Annotation, annotation_colors=color_BAG, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "BAG",
         annotation_legend = TRUE,
         #display_numbers = Cann_BAG_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(4,9)
)


#########################################################################################################################################
####################################  HSP90  #############################################################################################
#########################################################################################################################################

HSP90_Ath <- read.delim("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan/Funcional Groups/Ath_HSP90.txt")
HSP90_Ath <- merge(HSP90_Ath, Ath_Gene_Models, by.x="GeneID", by.y=1)
HSP90_Ath_fasta <- Ath11_fasta[c(which(names(Ath11_fasta) %in% HSP90_Ath$V2))]
HSP90_Ath_2 <- HSP90_Ath %>% select(V2, Name)
write.fasta(HSP90_Ath_fasta, as.string=TRUE, names= names(HSP90_Ath_fasta),
            file.out = "Funcional Groups/Ath_HSP90.fasta")
rename.fasta(infile = "Funcional Groups/Ath_HSP90.fasta", HSP90_Ath_2, outfile = "Funcional Groups/Ath_HSP90_2.fasta")

HSP90_Niben_Candidates <- read.table("Funcional Groups/Niben_Candidates_HSP90.txt")
HSP90_Niben_Candidates_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% HSP90_Niben_Candidates$V1))]
write.fasta(HSP90_Niben_Candidates_fasta, as.string=TRUE, names= names(HSP90_Niben_Candidates_fasta),
            file.out = "Funcional Groups/HSP90_Niben_Candidates_35.fasta")

HSP90_Sly_Candidates <- read.table("Funcional Groups/Sly_Candidates_HSP90.txt")
HSP90_Sly_Candidates_fasta <- Tomato_4.0_fasta[c(which(names(Tomato_4.0_fasta) %in% HSP90_Sly_Candidates$V1))]
write.fasta(HSP90_Sly_Candidates_fasta, as.string=TRUE, names= names(HSP90_Sly_Candidates_fasta),
            file.out = "Funcional Groups/HSP90_Sly_Candidates_35.fasta")

HSP90_Cann_Candidates <- read.table("Funcional Groups/Cann_Candidates_HSP90.txt")
HSP90_Cann_Candidates_fasta <- Cannuum_ASM51225v2_fasta[c(which(names(Cannuum_ASM51225v2_fasta) %in% HSP90_Cann_Candidates$V1))]
write.fasta(HSP90_Cann_Candidates_fasta, as.string=TRUE, names= names(HSP90_Cann_Candidates_fasta),
            file.out = "Funcional Groups/HSP90_Cann_Candidates_35.fasta")

Niben_HSP90_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_HSP90_Ann$Name))]
write.fasta(Niben_HSP90_fasta, as.string=TRUE, names= names(Niben_HSP90_fasta),
            file.out = "Funcional Groups/Niben_HSP90.fasta")
Niben_HSP90_Ann_2 <- Niben_HSP90_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_HSP90.fasta", Niben_HSP90_Ann, outfile = "Funcional Groups/Niben_HSP90_2.fasta")

save_heatmaps("HSP90", height = 7)


## Representing Heatmap

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Niben_HSP90_Ann <- read.table("Funcional Groups/FigTree/Niben_HSP90.txt",
                              sep = "\t", header = TRUE)

colnames(Niben_HSP90_Ann) <- c("Name", "Protein")

Niben_HSP90_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_HSP90_Ann$Name))]
write.fasta(Niben_HSP90_fasta, as.string=TRUE, names= names(Niben_HSP90_fasta),
            file.out = "Funcional Groups/Niben_HSP90.fasta")
Niben_HSP90_Ann_2 <- Niben_HSP90_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_HSP90.fasta", Niben_HSP90_Ann, outfile = "Funcional Groups/Niben_HSP90_2.fasta")

Niben_HSP90_Chloro <- read.delim("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan/Funcional Groups/Niben_HSP90_2_summary.targetp2", 
                                 header=FALSE, comment.char="#")
colnames(Niben_HSP90_Chloro) <- c("ID",	"Prediction",	"noTP",	"SP",	"mTP",	"cTP",	"luTP",	"CS-Position")
Niben_HSP90_Chloro$Prediction <- factor(Niben_HSP90_Chloro$Prediction, levels = c("cTP","mTP","noTP"))
Niben_HSP90_Chloro <- merge(Niben_HSP90_Ann, Niben_HSP90_Chloro, by.x = "Protein", by.y = "ID")

Niben_HSP90_Chloro <- Niben_HSP90_Chloro[order(match(Niben_HSP90_Chloro$Protein, Niben_HSP90_Ann$Protein)),]
rownames(Niben_HSP90_Chloro) <- Niben_HSP90_Chloro$Protein
Niben_HSP90_Chloro <- Niben_HSP90_Chloro %>% select(Prediction)


## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_HSP90 <- merge(Niben_HSP90_Ann, TPM_All_FC_Annotated, by="Name")

## We create the annotation for Modules

Niben_HSP90_Modules <- which_Module_pwd30(Niben_HSP90_Ann)

## We merge this annotation of Modules with the annotation of Chloroplast TP prediction

Niben_HSP90_Modules_Chl <- merge(Niben_HSP90_Modules, Niben_HSP90_Chloro, by = 0)
Niben_HSP90_Modules_Chl <- Niben_HSP90_Modules_Chl[order(match(Niben_HSP90_Modules_Chl$Row.names, Niben_HSP90_Ann$Protein)),]
rownames(Niben_HSP90_Modules_Chl) <- Niben_HSP90_Modules_Chl$Row.names
Niben_HSP90_Modules_Chl$Row.names <- NULL

## We select only the FC data from the different comparisons
Niben_HSP90_TPM <- Niben_HSP90 %>% select(Protein, 4:51, 61:66)
Niben_HSP90_TPM <- Niben_HSP90_TPM[!duplicated(Niben_HSP90_TPM$Protein),]
Niben_HSP90_TPM <- Niben_HSP90_TPM[order(match(Niben_HSP90_TPM[,1], Niben_HSP90_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_HSP90_FC <- Niben_HSP90 %>% select(Protein, 71, 73:87)
Niben_HSP90_FC <- Niben_HSP90_FC[!duplicated(Niben_HSP90_FC$Protein),]
Niben_HSP90_FC <- Niben_HSP90_FC[order(match(Niben_HSP90_FC[,1], Niben_HSP90_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_HSP90_DEG <- Niben_HSP90 %>% select(Protein, 89, 91:105)
Niben_HSP90_DEG <- Niben_HSP90_DEG[!duplicated(Niben_HSP90_DEG$Protein),]
Niben_HSP90_DEG <- Niben_HSP90_DEG[order(match(Niben_HSP90_DEG[,1], Niben_HSP90_Ann[,2])),]

Niben_HSP90_DEG$pcrtB22h_vs_GFP22h_filtered <- ifelse(is.na(Niben_HSP90_DEG$pcrtB22h_vs_GFP22h_filtered) == TRUE, " ", "*")
Niben_HSP90_DEG$pcrtB25h_vs_GFP25h_filtered <- ifelse(is.na(Niben_HSP90_DEG$pcrtB25h_vs_GFP25h_filtered) == TRUE, " ", "*")
Niben_HSP90_DEG$pcrtBAA_vs_GFPAA_filtered <- ifelse(is.na(Niben_HSP90_DEG$pcrtBAA_vs_GFPAA_filtered) == TRUE, " ", "*")
Niben_HSP90_DEG$pcrtBBB_vs_GFPBB_filtered <- ifelse(is.na(Niben_HSP90_DEG$pcrtBBB_vs_GFPBB_filtered) == TRUE, " ", "*")
Niben_HSP90_DEG$pcrtBYY_vs_GFPYY_filtered <- ifelse(is.na(Niben_HSP90_DEG$pcrtBYY_vs_GFPYY_filtered) == TRUE, " ", "*")
Niben_HSP90_DEG$pcrtB28h_vs_GFP28h_filtered <- ifelse(is.na(Niben_HSP90_DEG$pcrtB28h_vs_GFP28h_filtered) == TRUE, " ", "*")
Niben_HSP90_DEG$pcrtB34h_vs_GFP34h_filtered <- ifelse(is.na(Niben_HSP90_DEG$pcrtB34h_vs_GFP34h_filtered) == TRUE, " ", "*")
Niben_HSP90_DEG$pcrtB37h_vs_GFP37h_filtered <- ifelse(is.na(Niben_HSP90_DEG$pcrtB37h_vs_GFP37h_filtered) == TRUE, " ", "*")
Niben_HSP90_DEG$pcrtBZZ_vs_GFPZZ_filtered <- ifelse(is.na(Niben_HSP90_DEG$pcrtBZZ_vs_GFPZZ_filtered) == TRUE, " ", "*")
Niben_HSP90_DEG$pcrtBCC_vs_GFPCC_filtered <- ifelse(is.na(Niben_HSP90_DEG$pcrtBCC_vs_GFPCC_filtered) == TRUE, " ", "*")
Niben_HSP90_DEG$pcrtB40h_vs_GFP40h_filtered <- ifelse(is.na(Niben_HSP90_DEG$pcrtB40h_vs_GFP40h_filtered) == TRUE, " ", "*")
Niben_HSP90_DEG$pcrtBDD_vs_GFPDD_filtered <- ifelse(is.na(Niben_HSP90_DEG$pcrtBDD_vs_GFPDD_filtered) == TRUE, " ", "*")
Niben_HSP90_DEG$pcrtB46h_vs_GFP46h_filtered <- ifelse(is.na(Niben_HSP90_DEG$pcrtB46h_vs_GFP46h_filtered) == TRUE, " ", "*")
Niben_HSP90_DEG$pcrtB56h_vs_GFP56h_filtered <- ifelse(is.na(Niben_HSP90_DEG$pcrtB56h_vs_GFP56h_filtered) == TRUE, " ", "*")
Niben_HSP90_DEG$pcrtBEE_vs_GFPEE_filtered <- ifelse(is.na(Niben_HSP90_DEG$pcrtBEE_vs_GFPEE_filtered) == TRUE, " ", "*")
Niben_HSP90_DEG$pcrtB96h_vs_GFP96h_filtered <- ifelse(is.na(Niben_HSP90_DEG$pcrtB96h_vs_GFP96h_filtered) == TRUE, " ", "*")
row.names(Niben_HSP90_DEG) <- Niben_HSP90_DEG$Protein
Niben_HSP90_DEG[1] <- NULL

Niben_HSP90_DEG <- Niben_HSP90_DEG %>%
  select(pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

row.names(Niben_HSP90_FC) <- Niben_HSP90_FC$Protein
Niben_HSP90_FC[1] <- NULL

Niben_HSP90_FC <- Niben_HSP90_FC %>%
  select(pcrtB22h_vs_GFP22h_All, pcrtBAA_vs_GFPAA_All, pcrtB25h_vs_GFP25h_All, 
         pcrtBYY_vs_GFPYY_All, pcrtBBB_vs_GFPBB_All, pcrtB28h_vs_GFP28h_All, pcrtB34h_vs_GFP34h_All, pcrtBZZ_vs_GFPZZ_All, 
         pcrtB37h_vs_GFP37h_All, pcrtBCC_vs_GFPCC_All, pcrtB40h_vs_GFP40h_All, pcrtBDD_vs_GFPDD_All, pcrtB46h_vs_GFP46h_All, 
         pcrtBEE_vs_GFPEE_All, pcrtB56h_vs_GFP56h_All, pcrtB96h_vs_GFP96h_All)


min(Niben_HSP90_FC, na.rm = TRUE)
max(Niben_HSP90_FC, na.rm = TRUE)
breaks_HSP90_FC=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Niben_HSP90_FC, # %>% select(1:15), 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_HSP90_FC,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_HSP90_Annotation, annotation_colors=color_HSP90, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h", "96h"),
         cex = 1,
         angle_col=45,
         main = "HSP90",
         annotation_legend = TRUE,
         display_numbers = Niben_HSP90_DEG, # %>% select(1:15),
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(84, 104),
         annotation_row = Niben_HSP90_Modules_Chl,
         annotation_colors = module_Chl_colors
         #gaps_col = c(15)
)


rownames(Niben_HSP90_TPM) <- Niben_HSP90_TPM$Protein
Niben_HSP90_TPM[1] <- NULL

min(Niben_HSP90_TPM, na.rm = TRUE)
max(Niben_HSP90_TPM, na.rm = TRUE)
breaks_Niben_HSP90_TPM=c(seq(0,100, length.out = 100))

Niben_HSP90_TPM_norm <- normalize_df(Niben_HSP90_TPM)
Niben_HSP90_TPM_norm_Cornell <- Niben_HSP90_TPM %>%
  select(1:48) 
Niben_HSP90_TPM_norm_Cornell <- normalize_df(Niben_HSP90_TPM_norm_Cornell)


pheatmap(Niben_HSP90_TPM, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         breaks = breaks_Niben_HSP90_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_HSP90_Annotation, annotation_colors=color_HSP90, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "HSP90",
         annotation_legend = TRUE,
         #display_numbers = Niben_HSP90_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         annotation_row = Niben_HSP90_Modules_Chl,
         annotation_colors = module_Chl_colors,
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(24)#, 48, 51)
)



#####Representing Heatmap for Sly

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Sly_HSP90_Ann <- read.table("Funcional Groups/FigTree/Sly_HSP90.txt",
                            sep = "\t", header = TRUE)

colnames(Sly_HSP90_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Sly_HSP90 <- merge(Sly_HSP90_Ann, Summary_Yazdani_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Sly_HSP90_TPM <- Sly_HSP90 %>% select(Protein, 4:66)
Sly_HSP90_TPM <- Sly_HSP90_TPM[!duplicated(Sly_HSP90_TPM$Protein),]
Sly_HSP90_TPM <- Sly_HSP90_TPM[order(match(Sly_HSP90_TPM[,1], Sly_HSP90_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_HSP90_FC <- Sly_HSP90 %>% select(Protein, 67:71)
Sly_HSP90_FC <- Sly_HSP90_FC[!duplicated(Sly_HSP90_FC$Protein),]
Sly_HSP90_FC <- Sly_HSP90_FC[order(match(Sly_HSP90_FC[,1], Sly_HSP90_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_HSP90_DEG <- Sly_HSP90 %>% select(Protein, 72:76)
Sly_HSP90_DEG <- Sly_HSP90_DEG[!duplicated(Sly_HSP90_DEG$Protein),]
Sly_HSP90_DEG <- Sly_HSP90_DEG[order(match(Sly_HSP90_DEG[,1], Sly_HSP90_Ann[,2])),]

Sly_HSP90_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Sly_HSP90_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Sly_HSP90_DEG$Ripening_Orange_DEG <- ifelse(is.na(Sly_HSP90_DEG$Ripening_Orange_DEG) == TRUE, " ", "*")
Sly_HSP90_DEG$Ripening_Red_DEG <- ifelse(is.na(Sly_HSP90_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
Sly_HSP90_DEG$ORHis_MG_DEG <- ifelse(is.na(Sly_HSP90_DEG$ORHis_MG_DEG) == TRUE, " ", "*")
Sly_HSP90_DEG$ORHis_Breaker_DEG <- ifelse(is.na(Sly_HSP90_DEG$ORHis_Breaker_DEG) == TRUE, " ", "*")
row.names(Sly_HSP90_DEG) <- Sly_HSP90_DEG$Protein
Sly_HSP90_DEG[1] <- NULL

rownames(Sly_HSP90_FC) <- Sly_HSP90_FC$Protein
Sly_HSP90_FC[1] <- NULL

min(Sly_HSP90_FC, na.rm = TRUE)
max(Sly_HSP90_FC, na.rm = TRUE)
breaks_Sly_HSP90=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Sly_HSP90_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Sly_HSP90,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_HSP90_Annotation, annotation_colors=color_HSP90, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "HSP90",
         annotation_legend = TRUE,
         display_numbers = Sly_HSP90_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(39, 48),
         gaps_col = c(3)
)


rownames(Sly_HSP90_TPM) <- Sly_HSP90_TPM$Protein
Sly_HSP90_TPM[1] <- NULL

min(Sly_HSP90_TPM, na.rm = TRUE)
max(Sly_HSP90_TPM, na.rm = TRUE)
breaks_Sly_HSP90_TPM=c(seq(0,100, length.out = 200))

Sly_HSP90_TPM_norm <- normalize_df(Sly_HSP90_TPM)

pheatmap(Sly_HSP90_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Sly_HSP90_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_HSP90_Annotation, annotation_colors=color_HSP90, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "HSP90",
         annotation_legend = TRUE,
         #display_numbers = Sly_HSP90_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(21,42)
)


#####Representing Heatmap for Cann

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Cann_HSP90_Ann <- read.table("Funcional Groups/FigTree/Cann_HSP90.txt",
                             sep = "\t", header = TRUE)

colnames(Cann_HSP90_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Cann_HSP90 <- merge(Cann_HSP90_Ann, Summary_Pepper_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Cann_HSP90_TPM <- Cann_HSP90 %>% select(Protein, 4:17)
Cann_HSP90_TPM <- Cann_HSP90_TPM[!duplicated(Cann_HSP90_TPM$Protein),]
Cann_HSP90_TPM <- Cann_HSP90_TPM[order(match(Cann_HSP90_TPM[,1], Cann_HSP90_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_HSP90_FC <- Cann_HSP90 %>% select(Protein, 18:19)
Cann_HSP90_FC <- Cann_HSP90_FC[!duplicated(Cann_HSP90_FC$Protein),]
Cann_HSP90_FC <- Cann_HSP90_FC[order(match(Cann_HSP90_FC[,1], Cann_HSP90_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_HSP90_DEG <- Cann_HSP90 %>% select(Protein, 20:21)
Cann_HSP90_DEG <- Cann_HSP90_DEG[!duplicated(Cann_HSP90_DEG$Protein),]
Cann_HSP90_DEG <- Cann_HSP90_DEG[order(match(Cann_HSP90_DEG[,1], Cann_HSP90_Ann[,2])),]

Cann_HSP90_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Cann_HSP90_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Cann_HSP90_DEG$Ripening_Red_DEG <- ifelse(is.na(Cann_HSP90_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
row.names(Cann_HSP90_DEG) <- Cann_HSP90_DEG$Protein
Cann_HSP90_DEG[1] <- NULL

rownames(Cann_HSP90_FC) <- Cann_HSP90_FC$Protein
Cann_HSP90_FC[1] <- NULL

min(Cann_HSP90_FC, na.rm = TRUE)
max(Cann_HSP90_FC, na.rm = TRUE)
breaks_Cann_HSP90=c(seq(-4.5,0, length.out = 100), seq(0.01,6,length.out = 100))

pheatmap(Cann_HSP90_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Cann_HSP90,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_HSP90_Annotation, annotation_colors=color_HSP90, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Red"),
         cex = 1,
         angle_col=45,
         main = "HSP90",
         annotation_legend = TRUE,
         display_numbers = Cann_HSP90_DEG,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)


rownames(Cann_HSP90_TPM) <- Cann_HSP90_TPM$Protein
Cann_HSP90_TPM[1] <- NULL

min(Cann_HSP90_TPM, na.rm = TRUE)
max(Cann_HSP90_TPM, na.rm = TRUE)
breaks_Cann_HSP90_TPM=c(seq(0,100, length.out = 200))

Cann_HSP90_TPM_norm <- normalize_df(Cann_HSP90_TPM)

pheatmap(Cann_HSP90_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Cann_HSP90_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_HSP90_Annotation, annotation_colors=color_HSP90, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "HSP90",
         annotation_legend = TRUE,
         #display_numbers = Cann_HSP90_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(4,9)
)


####################################  HSP100  #############################################################################################
#########################################################################################################################################

HSP100_Ath <- read.delim("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan/Funcional Groups/Ath_HSP100.txt")
HSP100_Ath <- merge(HSP100_Ath, Ath_Gene_Models, by.x="GeneID", by.y=1)
HSP100_Ath_fasta <- Ath11_fasta[c(which(names(Ath11_fasta) %in% HSP100_Ath$V2))]
HSP100_Ath_2 <- HSP100_Ath %>% select(V2, Name)
write.fasta(HSP100_Ath_fasta, as.string=TRUE, names= names(HSP100_Ath_fasta),
            file.out = "Funcional Groups/Ath_HSP100.fasta")
rename.fasta(infile = "Funcional Groups/Ath_HSP100.fasta", HSP100_Ath_2, outfile = "Funcional Groups/Ath_HSP100_2.fasta")

HSP100_Niben_Candidates <- read.table("Funcional Groups/Niben_Candidates_HSP100.txt")
HSP100_Niben_Candidates_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% HSP100_Niben_Candidates$V1))]
write.fasta(HSP100_Niben_Candidates_fasta, as.string=TRUE, names= names(HSP100_Niben_Candidates_fasta),
            file.out = "Funcional Groups/HSP100_Niben_Candidates_35.fasta")

HSP100_Sly_Candidates <- read.table("Funcional Groups/Sly_Candidates_HSP100.txt")
HSP100_Sly_Candidates_fasta <- Tomato_4.0_fasta[c(which(names(Tomato_4.0_fasta) %in% HSP100_Sly_Candidates$V1))]
write.fasta(HSP100_Sly_Candidates_fasta, as.string=TRUE, names= names(HSP100_Sly_Candidates_fasta),
            file.out = "Funcional Groups/HSP100_Sly_Candidates_35.fasta")

HSP100_Cann_Candidates <- read.table("Funcional Groups/Cann_Candidates_HSP100.txt")
HSP100_Cann_Candidates_fasta <- Cannuum_ASM51225v2_fasta[c(which(names(Cannuum_ASM51225v2_fasta) %in% HSP100_Cann_Candidates$V1))]
write.fasta(HSP100_Cann_Candidates_fasta, as.string=TRUE, names= names(HSP100_Cann_Candidates_fasta),
            file.out = "Funcional Groups/HSP100_Cann_Candidates_35.fasta")

Niben_HSP100_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_HSP100_Ann$Name))]
write.fasta(Niben_HSP100_fasta, as.string=TRUE, names= names(Niben_HSP100_fasta),
            file.out = "Funcional Groups/Niben_HSP100.fasta")
Niben_HSP100_Ann_2 <- Niben_HSP100_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_HSP100.fasta", Niben_HSP100_Ann, outfile = "Funcional Groups/Niben_HSP100_2.fasta")

save_heatmaps("HSP100", height = 7)


## Representing Heatmap

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Niben_HSP100_Ann <- read.table("Funcional Groups/FigTree/Niben_HSP100.txt",
                              sep = "\t", header = TRUE)

colnames(Niben_HSP100_Ann) <- c("Name", "Protein")

Niben_HSP100_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_HSP100_Ann$Name))]
write.fasta(Niben_HSP100_fasta, as.string=TRUE, names= names(Niben_HSP100_fasta),
            file.out = "Funcional Groups/Niben_HSP100.fasta")
Niben_HSP100_Ann_2 <- Niben_HSP100_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_HSP100.fasta", Niben_HSP100_Ann, outfile = "Funcional Groups/Niben_HSP100_2.fasta")

Niben_HSP100_Chloro <- read.delim("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan/Funcional Groups/Niben_HSP100_2_summary.targetp2", 
                                 header=FALSE, comment.char="#")
colnames(Niben_HSP100_Chloro) <- c("ID",	"Prediction",	"noTP",	"SP",	"mTP",	"cTP",	"luTP",	"CS-Position")
Niben_HSP100_Chloro$Prediction <- factor(Niben_HSP100_Chloro$Prediction, levels = c("cTP","mTP","noTP"))
Niben_HSP100_Chloro <- merge(Niben_HSP100_Ann, Niben_HSP100_Chloro, by.x = "Protein", by.y = "ID")

Niben_HSP100_Chloro <- Niben_HSP100_Chloro[order(match(Niben_HSP100_Chloro$Protein, Niben_HSP100_Ann$Protein)),]
rownames(Niben_HSP100_Chloro) <- Niben_HSP100_Chloro$Protein
Niben_HSP100_Chloro <- Niben_HSP100_Chloro %>% select(Prediction)


## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_HSP100 <- merge(Niben_HSP100_Ann, TPM_All_FC_Annotated, by="Name")

## We create the annotation for Modules

Niben_HSP100_Modules <- which_Module_pwd30(Niben_HSP100_Ann)

## We merge this annotation of Modules with the annotation of Chloroplast TP prediction

Niben_HSP100_Modules_Chl <- merge(Niben_HSP100_Modules, Niben_HSP100_Chloro, by = 0)
Niben_HSP100_Modules_Chl <- Niben_HSP100_Modules_Chl[order(match(Niben_HSP100_Modules_Chl$Row.names, Niben_HSP100_Ann$Protein)),]
rownames(Niben_HSP100_Modules_Chl) <- Niben_HSP100_Modules_Chl$Row.names
Niben_HSP100_Modules_Chl$Row.names <- NULL

## We select only the FC data from the different comparisons
Niben_HSP100_TPM <- Niben_HSP100 %>% select(Protein, 4:51, 61:66)
Niben_HSP100_TPM <- Niben_HSP100_TPM[!duplicated(Niben_HSP100_TPM$Protein),]
Niben_HSP100_TPM <- Niben_HSP100_TPM[order(match(Niben_HSP100_TPM[,1], Niben_HSP100_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_HSP100_FC <- Niben_HSP100 %>% select(Protein, 71, 73:87)
Niben_HSP100_FC <- Niben_HSP100_FC[!duplicated(Niben_HSP100_FC$Protein),]
Niben_HSP100_FC <- Niben_HSP100_FC[order(match(Niben_HSP100_FC[,1], Niben_HSP100_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_HSP100_DEG <- Niben_HSP100 %>% select(Protein, 89, 91:105)
Niben_HSP100_DEG <- Niben_HSP100_DEG[!duplicated(Niben_HSP100_DEG$Protein),]
Niben_HSP100_DEG <- Niben_HSP100_DEG[order(match(Niben_HSP100_DEG[,1], Niben_HSP100_Ann[,2])),]

Niben_HSP100_DEG$pcrtB22h_vs_GFP22h_filtered <- ifelse(is.na(Niben_HSP100_DEG$pcrtB22h_vs_GFP22h_filtered) == TRUE, " ", "*")
Niben_HSP100_DEG$pcrtB25h_vs_GFP25h_filtered <- ifelse(is.na(Niben_HSP100_DEG$pcrtB25h_vs_GFP25h_filtered) == TRUE, " ", "*")
Niben_HSP100_DEG$pcrtBAA_vs_GFPAA_filtered <- ifelse(is.na(Niben_HSP100_DEG$pcrtBAA_vs_GFPAA_filtered) == TRUE, " ", "*")
Niben_HSP100_DEG$pcrtBBB_vs_GFPBB_filtered <- ifelse(is.na(Niben_HSP100_DEG$pcrtBBB_vs_GFPBB_filtered) == TRUE, " ", "*")
Niben_HSP100_DEG$pcrtBYY_vs_GFPYY_filtered <- ifelse(is.na(Niben_HSP100_DEG$pcrtBYY_vs_GFPYY_filtered) == TRUE, " ", "*")
Niben_HSP100_DEG$pcrtB28h_vs_GFP28h_filtered <- ifelse(is.na(Niben_HSP100_DEG$pcrtB28h_vs_GFP28h_filtered) == TRUE, " ", "*")
Niben_HSP100_DEG$pcrtB34h_vs_GFP34h_filtered <- ifelse(is.na(Niben_HSP100_DEG$pcrtB34h_vs_GFP34h_filtered) == TRUE, " ", "*")
Niben_HSP100_DEG$pcrtB37h_vs_GFP37h_filtered <- ifelse(is.na(Niben_HSP100_DEG$pcrtB37h_vs_GFP37h_filtered) == TRUE, " ", "*")
Niben_HSP100_DEG$pcrtBZZ_vs_GFPZZ_filtered <- ifelse(is.na(Niben_HSP100_DEG$pcrtBZZ_vs_GFPZZ_filtered) == TRUE, " ", "*")
Niben_HSP100_DEG$pcrtBCC_vs_GFPCC_filtered <- ifelse(is.na(Niben_HSP100_DEG$pcrtBCC_vs_GFPCC_filtered) == TRUE, " ", "*")
Niben_HSP100_DEG$pcrtB40h_vs_GFP40h_filtered <- ifelse(is.na(Niben_HSP100_DEG$pcrtB40h_vs_GFP40h_filtered) == TRUE, " ", "*")
Niben_HSP100_DEG$pcrtBDD_vs_GFPDD_filtered <- ifelse(is.na(Niben_HSP100_DEG$pcrtBDD_vs_GFPDD_filtered) == TRUE, " ", "*")
Niben_HSP100_DEG$pcrtB46h_vs_GFP46h_filtered <- ifelse(is.na(Niben_HSP100_DEG$pcrtB46h_vs_GFP46h_filtered) == TRUE, " ", "*")
Niben_HSP100_DEG$pcrtB56h_vs_GFP56h_filtered <- ifelse(is.na(Niben_HSP100_DEG$pcrtB56h_vs_GFP56h_filtered) == TRUE, " ", "*")
Niben_HSP100_DEG$pcrtBEE_vs_GFPEE_filtered <- ifelse(is.na(Niben_HSP100_DEG$pcrtBEE_vs_GFPEE_filtered) == TRUE, " ", "*")
Niben_HSP100_DEG$pcrtB96h_vs_GFP96h_filtered <- ifelse(is.na(Niben_HSP100_DEG$pcrtB96h_vs_GFP96h_filtered) == TRUE, " ", "*")
row.names(Niben_HSP100_DEG) <- Niben_HSP100_DEG$Protein
Niben_HSP100_DEG[1] <- NULL

Niben_HSP100_DEG <- Niben_HSP100_DEG %>%
  select(pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

row.names(Niben_HSP100_FC) <- Niben_HSP100_FC$Protein
Niben_HSP100_FC[1] <- NULL

Niben_HSP100_FC <- Niben_HSP100_FC %>%
  select(pcrtB22h_vs_GFP22h_All, pcrtBAA_vs_GFPAA_All, pcrtB25h_vs_GFP25h_All, 
         pcrtBYY_vs_GFPYY_All, pcrtBBB_vs_GFPBB_All, pcrtB28h_vs_GFP28h_All, pcrtB34h_vs_GFP34h_All, pcrtBZZ_vs_GFPZZ_All, 
         pcrtB37h_vs_GFP37h_All, pcrtBCC_vs_GFPCC_All, pcrtB40h_vs_GFP40h_All, pcrtBDD_vs_GFPDD_All, pcrtB46h_vs_GFP46h_All, 
         pcrtBEE_vs_GFPEE_All, pcrtB56h_vs_GFP56h_All, pcrtB96h_vs_GFP96h_All)


min(Niben_HSP100_FC, na.rm = TRUE)
max(Niben_HSP100_FC, na.rm = TRUE)
breaks_HSP100_FC=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Niben_HSP100_FC, # %>% select(1:15), 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_HSP100_FC,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_HSP100_Annotation, annotation_colors=color_HSP100, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h", "96h"),
         cex = 1,
         angle_col=45,
         main = "HSP100",
         annotation_legend = TRUE,
         display_numbers = Niben_HSP100_DEG, # %>% select(1:15),
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(84, 104),
         annotation_row = Niben_HSP100_Modules_Chl,
         annotation_colors = module_Chl_colors
         #gaps_col = c(15)
)


rownames(Niben_HSP100_TPM) <- Niben_HSP100_TPM$Protein
Niben_HSP100_TPM[1] <- NULL

min(Niben_HSP100_TPM, na.rm = TRUE)
max(Niben_HSP100_TPM, na.rm = TRUE)
breaks_Niben_HSP100_TPM=c(seq(0,100, length.out = 100))

Niben_HSP100_TPM_norm <- normalize_df(Niben_HSP100_TPM)
Niben_HSP100_TPM_norm_Cornell <- Niben_HSP100_TPM %>%
  select(1:48) 
Niben_HSP100_TPM_norm_Cornell <- normalize_df(Niben_HSP100_TPM_norm_Cornell)


pheatmap(Niben_HSP100_TPM, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         breaks = breaks_Niben_HSP100_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_HSP100_Annotation, annotation_colors=color_HSP100, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "HSP100",
         annotation_legend = TRUE,
         #display_numbers = Niben_HSP100_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         annotation_row = Niben_HSP100_Modules_Chl,
         annotation_colors = module_Chl_colors,
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(24)#, 48, 51)
)



#####Representing Heatmap for Sly

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Sly_HSP100_Ann <- read.table("Funcional Groups/FigTree/Sly_HSP100.txt",
                            sep = "\t", header = TRUE)

colnames(Sly_HSP100_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Sly_HSP100 <- merge(Sly_HSP100_Ann, Summary_Yazdani_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Sly_HSP100_TPM <- Sly_HSP100 %>% select(Protein, 4:66)
Sly_HSP100_TPM <- Sly_HSP100_TPM[!duplicated(Sly_HSP100_TPM$Protein),]
Sly_HSP100_TPM <- Sly_HSP100_TPM[order(match(Sly_HSP100_TPM[,1], Sly_HSP100_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_HSP100_FC <- Sly_HSP100 %>% select(Protein, 67:71)
Sly_HSP100_FC <- Sly_HSP100_FC[!duplicated(Sly_HSP100_FC$Protein),]
Sly_HSP100_FC <- Sly_HSP100_FC[order(match(Sly_HSP100_FC[,1], Sly_HSP100_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_HSP100_DEG <- Sly_HSP100 %>% select(Protein, 72:76)
Sly_HSP100_DEG <- Sly_HSP100_DEG[!duplicated(Sly_HSP100_DEG$Protein),]
Sly_HSP100_DEG <- Sly_HSP100_DEG[order(match(Sly_HSP100_DEG[,1], Sly_HSP100_Ann[,2])),]

Sly_HSP100_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Sly_HSP100_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Sly_HSP100_DEG$Ripening_Orange_DEG <- ifelse(is.na(Sly_HSP100_DEG$Ripening_Orange_DEG) == TRUE, " ", "*")
Sly_HSP100_DEG$Ripening_Red_DEG <- ifelse(is.na(Sly_HSP100_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
Sly_HSP100_DEG$ORHis_MG_DEG <- ifelse(is.na(Sly_HSP100_DEG$ORHis_MG_DEG) == TRUE, " ", "*")
Sly_HSP100_DEG$ORHis_Breaker_DEG <- ifelse(is.na(Sly_HSP100_DEG$ORHis_Breaker_DEG) == TRUE, " ", "*")
row.names(Sly_HSP100_DEG) <- Sly_HSP100_DEG$Protein
Sly_HSP100_DEG[1] <- NULL

rownames(Sly_HSP100_FC) <- Sly_HSP100_FC$Protein
Sly_HSP100_FC[1] <- NULL

min(Sly_HSP100_FC, na.rm = TRUE)
max(Sly_HSP100_FC, na.rm = TRUE)
breaks_Sly_HSP100=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Sly_HSP100_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Sly_HSP100,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_HSP100_Annotation, annotation_colors=color_HSP100, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "HSP100",
         annotation_legend = TRUE,
         display_numbers = Sly_HSP100_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(39, 48),
         gaps_col = c(3)
)


rownames(Sly_HSP100_TPM) <- Sly_HSP100_TPM$Protein
Sly_HSP100_TPM[1] <- NULL

min(Sly_HSP100_TPM, na.rm = TRUE)
max(Sly_HSP100_TPM, na.rm = TRUE)
breaks_Sly_HSP100_TPM=c(seq(0,100, length.out = 200))

Sly_HSP100_TPM_norm <- normalize_df(Sly_HSP100_TPM)

pheatmap(Sly_HSP100_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Sly_HSP100_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_HSP100_Annotation, annotation_colors=color_HSP100, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "HSP100",
         annotation_legend = TRUE,
         #display_numbers = Sly_HSP100_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(21,42)
)


#####Representing Heatmap for Cann

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Cann_HSP100_Ann <- read.table("Funcional Groups/FigTree/Cann_HSP100.txt",
                             sep = "\t", header = TRUE)

colnames(Cann_HSP100_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Cann_HSP100 <- merge(Cann_HSP100_Ann, Summary_Pepper_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Cann_HSP100_TPM <- Cann_HSP100 %>% select(Protein, 4:17)
Cann_HSP100_TPM <- Cann_HSP100_TPM[!duplicated(Cann_HSP100_TPM$Protein),]
Cann_HSP100_TPM <- Cann_HSP100_TPM[order(match(Cann_HSP100_TPM[,1], Cann_HSP100_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_HSP100_FC <- Cann_HSP100 %>% select(Protein, 18:19)
Cann_HSP100_FC <- Cann_HSP100_FC[!duplicated(Cann_HSP100_FC$Protein),]
Cann_HSP100_FC <- Cann_HSP100_FC[order(match(Cann_HSP100_FC[,1], Cann_HSP100_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_HSP100_DEG <- Cann_HSP100 %>% select(Protein, 20:21)
Cann_HSP100_DEG <- Cann_HSP100_DEG[!duplicated(Cann_HSP100_DEG$Protein),]
Cann_HSP100_DEG <- Cann_HSP100_DEG[order(match(Cann_HSP100_DEG[,1], Cann_HSP100_Ann[,2])),]

Cann_HSP100_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Cann_HSP100_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Cann_HSP100_DEG$Ripening_Red_DEG <- ifelse(is.na(Cann_HSP100_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
row.names(Cann_HSP100_DEG) <- Cann_HSP100_DEG$Protein
Cann_HSP100_DEG[1] <- NULL

rownames(Cann_HSP100_FC) <- Cann_HSP100_FC$Protein
Cann_HSP100_FC[1] <- NULL

min(Cann_HSP100_FC, na.rm = TRUE)
max(Cann_HSP100_FC, na.rm = TRUE)
breaks_Cann_HSP100=c(seq(-4.5,0, length.out = 100), seq(0.01,6,length.out = 100))

pheatmap(Cann_HSP100_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Cann_HSP100,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_HSP100_Annotation, annotation_colors=color_HSP100, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Red"),
         cex = 1,
         angle_col=45,
         main = "HSP100",
         annotation_legend = TRUE,
         display_numbers = Cann_HSP100_DEG,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)


rownames(Cann_HSP100_TPM) <- Cann_HSP100_TPM$Protein
Cann_HSP100_TPM[1] <- NULL

min(Cann_HSP100_TPM, na.rm = TRUE)
max(Cann_HSP100_TPM, na.rm = TRUE)
breaks_Cann_HSP100_TPM=c(seq(0,100, length.out = 200))

Cann_HSP100_TPM_norm <- normalize_df(Cann_HSP100_TPM)

pheatmap(Cann_HSP100_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Cann_HSP100_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_HSP100_Annotation, annotation_colors=color_HSP100, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "HSP100",
         annotation_legend = TRUE,
         #display_numbers = Cann_HSP100_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(4,9)
)


## Comparativa 96h_SlyYaz_Ath

Ath_HSP100_Ann <- read.table("Funcional Groups/Ath_HSP100.txt", header = TRUE)

Ath_HSP100_FC <- merge(Ath_HSP100_Ann, Summary_Ath_FC, by.x=1, by.y = 0, all.x = TRUE) %>%
  dplyr::select(Name, log2FoldChange)
Ath_HSP100_FC <- Ath_HSP100_FC[order(match(Ath_HSP100_FC[,1], Ath_HSP100_Ann[,2])),]

Ath_HSP100_DEG <- merge(Ath_HSP100_Ann, Summary_Ath_DEGs, by.x=1, by.y = 1, all.x = TRUE) %>%
  dplyr::select(Name, log2FC)
Ath_HSP100_DEG <- Ath_HSP100_DEG[order(match(Ath_HSP100_DEG[,1], Ath_HSP100_Ann[,2])),]

Ath_HSP100_DEG$log2FC <- ifelse(is.na(Ath_HSP100_DEG$log2FC) == TRUE, " ", "*")
row.names(Ath_HSP100_DEG) <- Ath_HSP100_DEG$Name
Ath_HSP100_DEG[1] <- NULL

rownames(Ath_HSP100_FC) <- Ath_HSP100_FC$Name
Ath_HSP100_FC[1] <- NULL

min(Ath_HSP100_FC, na.rm = TRUE)
max(Ath_HSP100_FC, na.rm = TRUE)
breaks_Ath_HSP100=c(seq(-4.5,0, length.out = 100), seq(0.01,4.5,length.out = 100))

pheatmap(Ath_HSP100_FC, 
         border_color = "grey60",legend= TRUE,
         color=color3, 
         cluster_rows =FALSE,
         breaks = breaks_Stndr,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_HSP100_Annotation, annotation_colors=color_HSP100, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = "Senescence",
         cex = 1,
         angle_col=45,
         main = "HSP100",
         annotation_legend = TRUE,
         display_numbers = Ath_HSP100_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         gaps_row = c(3, 4, 5, 6, 7, 8, 9, 10)
         #gaps_col = c(2,5)
)

pheatmap(Niben_HSP100_FC %>% select(16), 
         border_color = "grey60",legend= TRUE,
         color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_HSP100_FC,
         breaks = breaks_Stndr,
         #legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         #legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_HSP100_Annotation, annotation_colors=color_HSP100, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = "96h",
         cex = 1,
         angle_col=45,
         main = "HSP100",
         annotation_legend = TRUE,
         display_numbers = Niben_HSP100_DEG %>% select(16),
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(84, 104),
         #annotation_row = Niben_HSP100_Modules_Chl,
         #annotation_colors = module_Chl_colors_2,
         gaps_row = c(3, 5, 9, 12, 15, 17)
)


pheatmap(Sly_HSP100_FC %>% select(2,3), 
         border_color = "grey60",legend= TRUE,
         color=color3, 
         cluster_rows =FALSE,
         breaks = breaks_Stndr,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         #legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_HSP100_Annotation, annotation_colors=color_HSP100, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Orange", "Ripening_Red"),
         cex = 1,
         angle_col=45,
         main = "HSP100",
         annotation_legend = TRUE,
         display_numbers = Sly_HSP100_DEG %>% select(2,3),
         fontsize_number = 15,
         na_col = "darkgrey",
         gaps_row = c(1, 3,  4, 7, 8)
)

#########################################################################################################################################
####################################  Cyclophilin  #############################################################################################
#########################################################################################################################################

Cyclophilin_Ath <- read.delim("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan/Funcional Groups/Ath_Cyclophilin.txt")
Cyclophilin_Ath <- merge(Cyclophilin_Ath, Ath_Gene_Models, by.x="GeneID", by.y=1)
Cyclophilin_Ath_fasta <- Ath11_fasta[c(which(names(Ath11_fasta) %in% Cyclophilin_Ath$V2))]
Cyclophilin_Ath_2 <- Cyclophilin_Ath %>% select(V2, Name)
write.fasta(Cyclophilin_Ath_fasta, as.string=TRUE, names= names(Cyclophilin_Ath_fasta),
            file.out = "Funcional Groups/Ath_Cyclophilin.fasta")
rename.fasta(infile = "Funcional Groups/Ath_Cyclophilin.fasta", Cyclophilin_Ath_2, outfile = "Funcional Groups/Ath_Cyclophilin_2.fasta")

Cyclophilin_Niben_Candidates <- read.table("Funcional Groups/Niben_Candidates_Cyclophilin.txt")
Cyclophilin_Niben_Candidates_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Cyclophilin_Niben_Candidates$V1))]
write.fasta(Cyclophilin_Niben_Candidates_fasta, as.string=TRUE, names= names(Cyclophilin_Niben_Candidates_fasta),
            file.out = "Funcional Groups/Cyclophilin_Niben_Candidates_35.fasta")

Cyclophilin_Sly_Candidates <- read.table("Funcional Groups/Sly_Candidates_Cyclophilin.txt")
Cyclophilin_Sly_Candidates_fasta <- Tomato_4.0_fasta[c(which(names(Tomato_4.0_fasta) %in% Cyclophilin_Sly_Candidates$V1))]
write.fasta(Cyclophilin_Sly_Candidates_fasta, as.string=TRUE, names= names(Cyclophilin_Sly_Candidates_fasta),
            file.out = "Funcional Groups/Cyclophilin_Sly_Candidates_35.fasta")

Cyclophilin_Cann_Candidates <- read.table("Funcional Groups/Cann_Candidates_Cyclophilin.txt")
Cyclophilin_Cann_Candidates_fasta <- Cannuum_ASM51225v2_fasta[c(which(names(Cannuum_ASM51225v2_fasta) %in% Cyclophilin_Cann_Candidates$V1))]
write.fasta(Cyclophilin_Cann_Candidates_fasta, as.string=TRUE, names= names(Cyclophilin_Cann_Candidates_fasta),
            file.out = "Funcional Groups/Cyclophilin_Cann_Candidates_35.fasta")

Niben_Cyclophilin_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_Cyclophilin_Ann$Name))]
write.fasta(Niben_Cyclophilin_fasta, as.string=TRUE, names= names(Niben_Cyclophilin_fasta),
            file.out = "Funcional Groups/Niben_Cyclophilin.fasta")
Niben_Cyclophilin_Ann_2 <- Niben_Cyclophilin_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_Cyclophilin.fasta", Niben_Cyclophilin_Ann, outfile = "Funcional Groups/Niben_Cyclophilin_2.fasta")

save_heatmaps("Cyclophilin", height = 10)

## Representing Heatmap

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Niben_Cyclophilin_Ann <- read.table("Funcional Groups/FigTree/Niben_Cyclophilin.txt",
                              sep = "\t", header = TRUE)

colnames(Niben_Cyclophilin_Ann) <- c("Name", "Protein")

Niben_Cyclophilin_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_Cyclophilin_Ann$Name))]
write.fasta(Niben_Cyclophilin_fasta, as.string=TRUE, names= names(Niben_Cyclophilin_fasta),
            file.out = "Funcional Groups/Niben_Cyclophilin.fasta")
Niben_Cyclophilin_Ann_2 <- Niben_Cyclophilin_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_Cyclophilin.fasta", Niben_Cyclophilin_Ann, outfile = "Funcional Groups/Niben_Cyclophilin_2.fasta")

Niben_Cyclophilin_Chloro <- read.delim("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan/Funcional Groups/Niben_Cyclophilin_2_summary.targetp2", 
                                 header=FALSE, comment.char="#")
colnames(Niben_Cyclophilin_Chloro) <- c("ID",	"Prediction",	"noTP",	"SP",	"mTP",	"cTP",	"luTP",	"CS-Position")
Niben_Cyclophilin_Chloro$Prediction <- factor(Niben_Cyclophilin_Chloro$Prediction, levels = c("cTP","mTP","noTP"))
Niben_Cyclophilin_Chloro <- merge(Niben_Cyclophilin_Ann, Niben_Cyclophilin_Chloro, by.x = "Protein", by.y = "ID")

Niben_Cyclophilin_Chloro <- Niben_Cyclophilin_Chloro[order(match(Niben_Cyclophilin_Chloro$Protein, Niben_Cyclophilin_Ann$Protein)),]
rownames(Niben_Cyclophilin_Chloro) <- Niben_Cyclophilin_Chloro$Protein
Niben_Cyclophilin_Chloro <- Niben_Cyclophilin_Chloro %>% select(Prediction)


## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_Cyclophilin <- merge(Niben_Cyclophilin_Ann, TPM_All_FC_Annotated, by="Name")

## We create the annotation for Modules

Niben_Cyclophilin_Modules <- which_Module_pwd30(Niben_Cyclophilin_Ann)

## We merge this annotation of Modules with the annotation of Chloroplast TP prediction

Niben_Cyclophilin_Modules_Chl <- merge(Niben_Cyclophilin_Modules, Niben_Cyclophilin_Chloro, by = 0)
Niben_Cyclophilin_Modules_Chl <- Niben_Cyclophilin_Modules_Chl[order(match(Niben_Cyclophilin_Modules_Chl$Row.names, Niben_Cyclophilin_Ann$Protein)),]
rownames(Niben_Cyclophilin_Modules_Chl) <- Niben_Cyclophilin_Modules_Chl$Row.names
Niben_Cyclophilin_Modules_Chl$Row.names <- NULL

## We select only the FC data from the different comparisons
Niben_Cyclophilin_TPM <- Niben_Cyclophilin %>% select(Protein, 4:51, 61:66)
Niben_Cyclophilin_TPM <- Niben_Cyclophilin_TPM[!duplicated(Niben_Cyclophilin_TPM$Protein),]
Niben_Cyclophilin_TPM <- Niben_Cyclophilin_TPM[order(match(Niben_Cyclophilin_TPM[,1], Niben_Cyclophilin_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_Cyclophilin_FC <- Niben_Cyclophilin %>% select(Protein, 71, 73:87)
Niben_Cyclophilin_FC <- Niben_Cyclophilin_FC[!duplicated(Niben_Cyclophilin_FC$Protein),]
Niben_Cyclophilin_FC <- Niben_Cyclophilin_FC[order(match(Niben_Cyclophilin_FC[,1], Niben_Cyclophilin_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_Cyclophilin_DEG <- Niben_Cyclophilin %>% select(Protein, 89, 91:105)
Niben_Cyclophilin_DEG <- Niben_Cyclophilin_DEG[!duplicated(Niben_Cyclophilin_DEG$Protein),]
Niben_Cyclophilin_DEG <- Niben_Cyclophilin_DEG[order(match(Niben_Cyclophilin_DEG[,1], Niben_Cyclophilin_Ann[,2])),]

Niben_Cyclophilin_DEG$pcrtB22h_vs_GFP22h_filtered <- ifelse(is.na(Niben_Cyclophilin_DEG$pcrtB22h_vs_GFP22h_filtered) == TRUE, " ", "*")
Niben_Cyclophilin_DEG$pcrtB25h_vs_GFP25h_filtered <- ifelse(is.na(Niben_Cyclophilin_DEG$pcrtB25h_vs_GFP25h_filtered) == TRUE, " ", "*")
Niben_Cyclophilin_DEG$pcrtBAA_vs_GFPAA_filtered <- ifelse(is.na(Niben_Cyclophilin_DEG$pcrtBAA_vs_GFPAA_filtered) == TRUE, " ", "*")
Niben_Cyclophilin_DEG$pcrtBBB_vs_GFPBB_filtered <- ifelse(is.na(Niben_Cyclophilin_DEG$pcrtBBB_vs_GFPBB_filtered) == TRUE, " ", "*")
Niben_Cyclophilin_DEG$pcrtBYY_vs_GFPYY_filtered <- ifelse(is.na(Niben_Cyclophilin_DEG$pcrtBYY_vs_GFPYY_filtered) == TRUE, " ", "*")
Niben_Cyclophilin_DEG$pcrtB28h_vs_GFP28h_filtered <- ifelse(is.na(Niben_Cyclophilin_DEG$pcrtB28h_vs_GFP28h_filtered) == TRUE, " ", "*")
Niben_Cyclophilin_DEG$pcrtB34h_vs_GFP34h_filtered <- ifelse(is.na(Niben_Cyclophilin_DEG$pcrtB34h_vs_GFP34h_filtered) == TRUE, " ", "*")
Niben_Cyclophilin_DEG$pcrtB37h_vs_GFP37h_filtered <- ifelse(is.na(Niben_Cyclophilin_DEG$pcrtB37h_vs_GFP37h_filtered) == TRUE, " ", "*")
Niben_Cyclophilin_DEG$pcrtBZZ_vs_GFPZZ_filtered <- ifelse(is.na(Niben_Cyclophilin_DEG$pcrtBZZ_vs_GFPZZ_filtered) == TRUE, " ", "*")
Niben_Cyclophilin_DEG$pcrtBCC_vs_GFPCC_filtered <- ifelse(is.na(Niben_Cyclophilin_DEG$pcrtBCC_vs_GFPCC_filtered) == TRUE, " ", "*")
Niben_Cyclophilin_DEG$pcrtB40h_vs_GFP40h_filtered <- ifelse(is.na(Niben_Cyclophilin_DEG$pcrtB40h_vs_GFP40h_filtered) == TRUE, " ", "*")
Niben_Cyclophilin_DEG$pcrtBDD_vs_GFPDD_filtered <- ifelse(is.na(Niben_Cyclophilin_DEG$pcrtBDD_vs_GFPDD_filtered) == TRUE, " ", "*")
Niben_Cyclophilin_DEG$pcrtB46h_vs_GFP46h_filtered <- ifelse(is.na(Niben_Cyclophilin_DEG$pcrtB46h_vs_GFP46h_filtered) == TRUE, " ", "*")
Niben_Cyclophilin_DEG$pcrtB56h_vs_GFP56h_filtered <- ifelse(is.na(Niben_Cyclophilin_DEG$pcrtB56h_vs_GFP56h_filtered) == TRUE, " ", "*")
Niben_Cyclophilin_DEG$pcrtBEE_vs_GFPEE_filtered <- ifelse(is.na(Niben_Cyclophilin_DEG$pcrtBEE_vs_GFPEE_filtered) == TRUE, " ", "*")
Niben_Cyclophilin_DEG$pcrtB96h_vs_GFP96h_filtered <- ifelse(is.na(Niben_Cyclophilin_DEG$pcrtB96h_vs_GFP96h_filtered) == TRUE, " ", "*")
row.names(Niben_Cyclophilin_DEG) <- Niben_Cyclophilin_DEG$Protein
Niben_Cyclophilin_DEG[1] <- NULL

Niben_Cyclophilin_DEG <- Niben_Cyclophilin_DEG %>%
  select(pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

row.names(Niben_Cyclophilin_FC) <- Niben_Cyclophilin_FC$Protein
Niben_Cyclophilin_FC[1] <- NULL

Niben_Cyclophilin_FC <- Niben_Cyclophilin_FC %>%
  select(pcrtB22h_vs_GFP22h_All, pcrtBAA_vs_GFPAA_All, pcrtB25h_vs_GFP25h_All, 
         pcrtBYY_vs_GFPYY_All, pcrtBBB_vs_GFPBB_All, pcrtB28h_vs_GFP28h_All, pcrtB34h_vs_GFP34h_All, pcrtBZZ_vs_GFPZZ_All, 
         pcrtB37h_vs_GFP37h_All, pcrtBCC_vs_GFPCC_All, pcrtB40h_vs_GFP40h_All, pcrtBDD_vs_GFPDD_All, pcrtB46h_vs_GFP46h_All, 
         pcrtBEE_vs_GFPEE_All, pcrtB56h_vs_GFP56h_All, pcrtB96h_vs_GFP96h_All)


min(Niben_Cyclophilin_FC, na.rm = TRUE)
max(Niben_Cyclophilin_FC, na.rm = TRUE)
breaks_Cyclophilin_FC=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Niben_Cyclophilin_FC, # %>% select(1:15), 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Cyclophilin_FC,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_Cyclophilin_Annotation, annotation_colors=color_Cyclophilin, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h", "96h"),
         cex = 1,
         angle_col=45,
         main = "Cyclophilin",
         annotation_legend = TRUE,
         display_numbers = Niben_Cyclophilin_DEG, # %>% select(1:15),
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(84, 104),
         annotation_row = Niben_Cyclophilin_Modules_Chl,
         annotation_colors = module_Chl_colors
         #gaps_col = c(15)
)


rownames(Niben_Cyclophilin_TPM) <- Niben_Cyclophilin_TPM$Protein
Niben_Cyclophilin_TPM[1] <- NULL

min(Niben_Cyclophilin_TPM, na.rm = TRUE)
max(Niben_Cyclophilin_TPM, na.rm = TRUE)
breaks_Niben_Cyclophilin_TPM=c(seq(0,100, length.out = 100))

Niben_Cyclophilin_TPM_norm <- normalize_df(Niben_Cyclophilin_TPM)
Niben_Cyclophilin_TPM_norm_Cornell <- Niben_Cyclophilin_TPM %>%
  select(1:48) 
Niben_Cyclophilin_TPM_norm_Cornell <- normalize_df(Niben_Cyclophilin_TPM_norm_Cornell)


pheatmap(Niben_Cyclophilin_TPM, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         breaks = breaks_Niben_Cyclophilin_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_Cyclophilin_Annotation, annotation_colors=color_Cyclophilin, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Cyclophilin",
         annotation_legend = TRUE,
         #display_numbers = Niben_Cyclophilin_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         annotation_row = Niben_Cyclophilin_Modules_Chl,
         annotation_colors = module_Chl_colors,
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(24)#, 48, 51)
)



#####Representing Heatmap for Sly

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Sly_Cyclophilin_Ann <- read.table("Funcional Groups/FigTree/Sly_Cyclophilin.txt",
                            sep = "\t", header = TRUE)

colnames(Sly_Cyclophilin_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Sly_Cyclophilin <- merge(Sly_Cyclophilin_Ann, Summary_Yazdani_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Sly_Cyclophilin_TPM <- Sly_Cyclophilin %>% select(Protein, 4:66)
Sly_Cyclophilin_TPM <- Sly_Cyclophilin_TPM[!duplicated(Sly_Cyclophilin_TPM$Protein),]
Sly_Cyclophilin_TPM <- Sly_Cyclophilin_TPM[order(match(Sly_Cyclophilin_TPM[,1], Sly_Cyclophilin_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_Cyclophilin_FC <- Sly_Cyclophilin %>% select(Protein, 67:71)
Sly_Cyclophilin_FC <- Sly_Cyclophilin_FC[!duplicated(Sly_Cyclophilin_FC$Protein),]
Sly_Cyclophilin_FC <- Sly_Cyclophilin_FC[order(match(Sly_Cyclophilin_FC[,1], Sly_Cyclophilin_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_Cyclophilin_DEG <- Sly_Cyclophilin %>% select(Protein, 72:76)
Sly_Cyclophilin_DEG <- Sly_Cyclophilin_DEG[!duplicated(Sly_Cyclophilin_DEG$Protein),]
Sly_Cyclophilin_DEG <- Sly_Cyclophilin_DEG[order(match(Sly_Cyclophilin_DEG[,1], Sly_Cyclophilin_Ann[,2])),]

Sly_Cyclophilin_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Sly_Cyclophilin_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Sly_Cyclophilin_DEG$Ripening_Orange_DEG <- ifelse(is.na(Sly_Cyclophilin_DEG$Ripening_Orange_DEG) == TRUE, " ", "*")
Sly_Cyclophilin_DEG$Ripening_Red_DEG <- ifelse(is.na(Sly_Cyclophilin_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
Sly_Cyclophilin_DEG$ORHis_MG_DEG <- ifelse(is.na(Sly_Cyclophilin_DEG$ORHis_MG_DEG) == TRUE, " ", "*")
Sly_Cyclophilin_DEG$ORHis_Breaker_DEG <- ifelse(is.na(Sly_Cyclophilin_DEG$ORHis_Breaker_DEG) == TRUE, " ", "*")
row.names(Sly_Cyclophilin_DEG) <- Sly_Cyclophilin_DEG$Protein
Sly_Cyclophilin_DEG[1] <- NULL

rownames(Sly_Cyclophilin_FC) <- Sly_Cyclophilin_FC$Protein
Sly_Cyclophilin_FC[1] <- NULL

min(Sly_Cyclophilin_FC, na.rm = TRUE)
max(Sly_Cyclophilin_FC, na.rm = TRUE)
breaks_Sly_Cyclophilin=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Sly_Cyclophilin_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Sly_Cyclophilin,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_Cyclophilin_Annotation, annotation_colors=color_Cyclophilin, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Cyclophilin",
         annotation_legend = TRUE,
         display_numbers = Sly_Cyclophilin_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(39, 48),
         gaps_col = c(3)
)


rownames(Sly_Cyclophilin_TPM) <- Sly_Cyclophilin_TPM$Protein
Sly_Cyclophilin_TPM[1] <- NULL

min(Sly_Cyclophilin_TPM, na.rm = TRUE)
max(Sly_Cyclophilin_TPM, na.rm = TRUE)
breaks_Sly_Cyclophilin_TPM=c(seq(0,100, length.out = 200))

Sly_Cyclophilin_TPM_norm <- normalize_df(Sly_Cyclophilin_TPM)

pheatmap(Sly_Cyclophilin_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Sly_Cyclophilin_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_Cyclophilin_Annotation, annotation_colors=color_Cyclophilin, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Cyclophilin",
         annotation_legend = TRUE,
         #display_numbers = Sly_Cyclophilin_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(21,42)
)


#####Representing Heatmap for Cann

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Cann_Cyclophilin_Ann <- read.table("Funcional Groups/FigTree/Cann_Cyclophilin.txt",
                             sep = "\t", header = TRUE)

colnames(Cann_Cyclophilin_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Cann_Cyclophilin <- merge(Cann_Cyclophilin_Ann, Summary_Pepper_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Cann_Cyclophilin_TPM <- Cann_Cyclophilin %>% select(Protein, 4:17)
Cann_Cyclophilin_TPM <- Cann_Cyclophilin_TPM[!duplicated(Cann_Cyclophilin_TPM$Protein),]
Cann_Cyclophilin_TPM <- Cann_Cyclophilin_TPM[order(match(Cann_Cyclophilin_TPM[,1], Cann_Cyclophilin_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_Cyclophilin_FC <- Cann_Cyclophilin %>% select(Protein, 18:19)
Cann_Cyclophilin_FC <- Cann_Cyclophilin_FC[!duplicated(Cann_Cyclophilin_FC$Protein),]
Cann_Cyclophilin_FC <- Cann_Cyclophilin_FC[order(match(Cann_Cyclophilin_FC[,1], Cann_Cyclophilin_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_Cyclophilin_DEG <- Cann_Cyclophilin %>% select(Protein, 20:21)
Cann_Cyclophilin_DEG <- Cann_Cyclophilin_DEG[!duplicated(Cann_Cyclophilin_DEG$Protein),]
Cann_Cyclophilin_DEG <- Cann_Cyclophilin_DEG[order(match(Cann_Cyclophilin_DEG[,1], Cann_Cyclophilin_Ann[,2])),]

Cann_Cyclophilin_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Cann_Cyclophilin_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Cann_Cyclophilin_DEG$Ripening_Red_DEG <- ifelse(is.na(Cann_Cyclophilin_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
row.names(Cann_Cyclophilin_DEG) <- Cann_Cyclophilin_DEG$Protein
Cann_Cyclophilin_DEG[1] <- NULL

rownames(Cann_Cyclophilin_FC) <- Cann_Cyclophilin_FC$Protein
Cann_Cyclophilin_FC[1] <- NULL

min(Cann_Cyclophilin_FC, na.rm = TRUE)
max(Cann_Cyclophilin_FC, na.rm = TRUE)
breaks_Cann_Cyclophilin=c(seq(-4.5,0, length.out = 100), seq(0.01,6,length.out = 100))

pheatmap(Cann_Cyclophilin_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Cann_Cyclophilin,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_Cyclophilin_Annotation, annotation_colors=color_Cyclophilin, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Red"),
         cex = 1,
         angle_col=45,
         main = "Cyclophilin",
         annotation_legend = TRUE,
         display_numbers = Cann_Cyclophilin_DEG,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)


rownames(Cann_Cyclophilin_TPM) <- Cann_Cyclophilin_TPM$Protein
Cann_Cyclophilin_TPM[1] <- NULL

min(Cann_Cyclophilin_TPM, na.rm = TRUE)
max(Cann_Cyclophilin_TPM, na.rm = TRUE)
breaks_Cann_Cyclophilin_TPM=c(seq(0,100, length.out = 200))

Cann_Cyclophilin_TPM_norm <- normalize_df(Cann_Cyclophilin_TPM)

pheatmap(Cann_Cyclophilin_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Cann_Cyclophilin_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_Cyclophilin_Annotation, annotation_colors=color_Cyclophilin, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Cyclophilin",
         annotation_legend = TRUE,
         #display_numbers = Cann_Cyclophilin_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(4,9)
)

#########################################################################################################################################
####################################  Rotamase  #############################################################################################
#########################################################################################################################################

Rotamase_Ath <- read.delim("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan/Funcional Groups/Ath_Rotamase.txt")
Rotamase_Ath <- merge(Rotamase_Ath, Ath_Gene_Models, by.x="GeneID", by.y=1)
Rotamase_Ath_fasta <- Ath11_fasta[c(which(names(Ath11_fasta) %in% Rotamase_Ath$V2))]
Rotamase_Ath_2 <- Rotamase_Ath %>% select(V2, Name)
write.fasta(Rotamase_Ath_fasta, as.string=TRUE, names= names(Rotamase_Ath_fasta),
            file.out = "Funcional Groups/Ath_Rotamase.fasta")
rename.fasta(infile = "Funcional Groups/Ath_Rotamase.fasta", Rotamase_Ath_2, outfile = "Funcional Groups/Ath_Rotamase_2.fasta")

Rotamase_Niben_Candidates <- read.table("Funcional Groups/Niben_Candidates_Rotamase.txt")
Rotamase_Niben_Candidates_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Rotamase_Niben_Candidates$V1))]
write.fasta(Rotamase_Niben_Candidates_fasta, as.string=TRUE, names= names(Rotamase_Niben_Candidates_fasta),
            file.out = "Funcional Groups/Rotamase_Niben_Candidates_35.fasta")

Rotamase_Sly_Candidates <- read.table("Funcional Groups/Sly_Candidates_Rotamase.txt")
Rotamase_Sly_Candidates_fasta <- Tomato_4.0_fasta[c(which(names(Tomato_4.0_fasta) %in% Rotamase_Sly_Candidates$V1))]
write.fasta(Rotamase_Sly_Candidates_fasta, as.string=TRUE, names= names(Rotamase_Sly_Candidates_fasta),
            file.out = "Funcional Groups/Rotamase_Sly_Candidates_35.fasta")

Rotamase_Cann_Candidates <- read.table("Funcional Groups/Cann_Candidates_Rotamase.txt")
Rotamase_Cann_Candidates_fasta <- Cannuum_ASM51225v2_fasta[c(which(names(Cannuum_ASM51225v2_fasta) %in% Rotamase_Cann_Candidates$V1))]
write.fasta(Rotamase_Cann_Candidates_fasta, as.string=TRUE, names= names(Rotamase_Cann_Candidates_fasta),
            file.out = "Funcional Groups/Rotamase_Cann_Candidates_35.fasta")

Niben_Rotamase_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_Rotamase_Ann$Name))]
write.fasta(Niben_Rotamase_fasta, as.string=TRUE, names= names(Niben_Rotamase_fasta),
            file.out = "Funcional Groups/Niben_Rotamase.fasta")
Niben_Rotamase_Ann_2 <- Niben_Rotamase_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_Rotamase.fasta", Niben_Rotamase_Ann, outfile = "Funcional Groups/Niben_Rotamase_2.fasta")

save_heatmaps("Rotamase", height = 10)


## Representing Heatmap

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Niben_Rotamase_Ann <- read.table("Funcional Groups/FigTree/Niben_Rotamase.txt",
                              sep = "\t", header = TRUE)

colnames(Niben_Rotamase_Ann) <- c("Name", "Protein")

Niben_Rotamase_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_Rotamase_Ann$Name))]
write.fasta(Niben_Rotamase_fasta, as.string=TRUE, names= names(Niben_Rotamase_fasta),
            file.out = "Funcional Groups/Niben_Rotamase.fasta")
Niben_Rotamase_Ann_2 <- Niben_Rotamase_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_Rotamase.fasta", Niben_Rotamase_Ann, outfile = "Funcional Groups/Niben_Rotamase_2.fasta")

Niben_Rotamase_Chloro <- read.delim("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan/Funcional Groups/Niben_Rotamase_2_summary.targetp2", 
                                 header=FALSE, comment.char="#")
colnames(Niben_Rotamase_Chloro) <- c("ID",	"Prediction",	"noTP",	"SP",	"mTP",	"cTP",	"luTP",	"CS-Position")
Niben_Rotamase_Chloro$Prediction <- factor(Niben_Rotamase_Chloro$Prediction, levels = c("cTP","mTP","noTP"))
Niben_Rotamase_Chloro <- merge(Niben_Rotamase_Ann, Niben_Rotamase_Chloro, by.x = "Protein", by.y = "ID")

Niben_Rotamase_Chloro <- Niben_Rotamase_Chloro[order(match(Niben_Rotamase_Chloro$Protein, Niben_Rotamase_Ann$Protein)),]
rownames(Niben_Rotamase_Chloro) <- Niben_Rotamase_Chloro$Protein
Niben_Rotamase_Chloro <- Niben_Rotamase_Chloro %>% select(Prediction)


## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_Rotamase <- merge(Niben_Rotamase_Ann, TPM_All_FC_Annotated, by="Name")

## We create the annotation for Modules

Niben_Rotamase_Modules <- which_Module_pwd30(Niben_Rotamase_Ann)

## We merge this annotation of Modules with the annotation of Chloroplast TP prediction

Niben_Rotamase_Modules_Chl <- merge(Niben_Rotamase_Modules, Niben_Rotamase_Chloro, by = 0)
Niben_Rotamase_Modules_Chl <- Niben_Rotamase_Modules_Chl[order(match(Niben_Rotamase_Modules_Chl$Row.names, Niben_Rotamase_Ann$Protein)),]
rownames(Niben_Rotamase_Modules_Chl) <- Niben_Rotamase_Modules_Chl$Row.names
Niben_Rotamase_Modules_Chl$Row.names <- NULL

## We select only the FC data from the different comparisons
Niben_Rotamase_TPM <- Niben_Rotamase %>% select(Protein, 4:51, 61:66)
Niben_Rotamase_TPM <- Niben_Rotamase_TPM[!duplicated(Niben_Rotamase_TPM$Protein),]
Niben_Rotamase_TPM <- Niben_Rotamase_TPM[order(match(Niben_Rotamase_TPM[,1], Niben_Rotamase_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_Rotamase_FC <- Niben_Rotamase %>% select(Protein, 71, 73:87)
Niben_Rotamase_FC <- Niben_Rotamase_FC[!duplicated(Niben_Rotamase_FC$Protein),]
Niben_Rotamase_FC <- Niben_Rotamase_FC[order(match(Niben_Rotamase_FC[,1], Niben_Rotamase_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_Rotamase_DEG <- Niben_Rotamase %>% select(Protein, 89, 91:105)
Niben_Rotamase_DEG <- Niben_Rotamase_DEG[!duplicated(Niben_Rotamase_DEG$Protein),]
Niben_Rotamase_DEG <- Niben_Rotamase_DEG[order(match(Niben_Rotamase_DEG[,1], Niben_Rotamase_Ann[,2])),]

Niben_Rotamase_DEG$pcrtB22h_vs_GFP22h_filtered <- ifelse(is.na(Niben_Rotamase_DEG$pcrtB22h_vs_GFP22h_filtered) == TRUE, " ", "*")
Niben_Rotamase_DEG$pcrtB25h_vs_GFP25h_filtered <- ifelse(is.na(Niben_Rotamase_DEG$pcrtB25h_vs_GFP25h_filtered) == TRUE, " ", "*")
Niben_Rotamase_DEG$pcrtBAA_vs_GFPAA_filtered <- ifelse(is.na(Niben_Rotamase_DEG$pcrtBAA_vs_GFPAA_filtered) == TRUE, " ", "*")
Niben_Rotamase_DEG$pcrtBBB_vs_GFPBB_filtered <- ifelse(is.na(Niben_Rotamase_DEG$pcrtBBB_vs_GFPBB_filtered) == TRUE, " ", "*")
Niben_Rotamase_DEG$pcrtBYY_vs_GFPYY_filtered <- ifelse(is.na(Niben_Rotamase_DEG$pcrtBYY_vs_GFPYY_filtered) == TRUE, " ", "*")
Niben_Rotamase_DEG$pcrtB28h_vs_GFP28h_filtered <- ifelse(is.na(Niben_Rotamase_DEG$pcrtB28h_vs_GFP28h_filtered) == TRUE, " ", "*")
Niben_Rotamase_DEG$pcrtB34h_vs_GFP34h_filtered <- ifelse(is.na(Niben_Rotamase_DEG$pcrtB34h_vs_GFP34h_filtered) == TRUE, " ", "*")
Niben_Rotamase_DEG$pcrtB37h_vs_GFP37h_filtered <- ifelse(is.na(Niben_Rotamase_DEG$pcrtB37h_vs_GFP37h_filtered) == TRUE, " ", "*")
Niben_Rotamase_DEG$pcrtBZZ_vs_GFPZZ_filtered <- ifelse(is.na(Niben_Rotamase_DEG$pcrtBZZ_vs_GFPZZ_filtered) == TRUE, " ", "*")
Niben_Rotamase_DEG$pcrtBCC_vs_GFPCC_filtered <- ifelse(is.na(Niben_Rotamase_DEG$pcrtBCC_vs_GFPCC_filtered) == TRUE, " ", "*")
Niben_Rotamase_DEG$pcrtB40h_vs_GFP40h_filtered <- ifelse(is.na(Niben_Rotamase_DEG$pcrtB40h_vs_GFP40h_filtered) == TRUE, " ", "*")
Niben_Rotamase_DEG$pcrtBDD_vs_GFPDD_filtered <- ifelse(is.na(Niben_Rotamase_DEG$pcrtBDD_vs_GFPDD_filtered) == TRUE, " ", "*")
Niben_Rotamase_DEG$pcrtB46h_vs_GFP46h_filtered <- ifelse(is.na(Niben_Rotamase_DEG$pcrtB46h_vs_GFP46h_filtered) == TRUE, " ", "*")
Niben_Rotamase_DEG$pcrtB56h_vs_GFP56h_filtered <- ifelse(is.na(Niben_Rotamase_DEG$pcrtB56h_vs_GFP56h_filtered) == TRUE, " ", "*")
Niben_Rotamase_DEG$pcrtBEE_vs_GFPEE_filtered <- ifelse(is.na(Niben_Rotamase_DEG$pcrtBEE_vs_GFPEE_filtered) == TRUE, " ", "*")
Niben_Rotamase_DEG$pcrtB96h_vs_GFP96h_filtered <- ifelse(is.na(Niben_Rotamase_DEG$pcrtB96h_vs_GFP96h_filtered) == TRUE, " ", "*")
row.names(Niben_Rotamase_DEG) <- Niben_Rotamase_DEG$Protein
Niben_Rotamase_DEG[1] <- NULL

Niben_Rotamase_DEG <- Niben_Rotamase_DEG %>%
  select(pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

row.names(Niben_Rotamase_FC) <- Niben_Rotamase_FC$Protein
Niben_Rotamase_FC[1] <- NULL

Niben_Rotamase_FC <- Niben_Rotamase_FC %>%
  select(pcrtB22h_vs_GFP22h_All, pcrtBAA_vs_GFPAA_All, pcrtB25h_vs_GFP25h_All, 
         pcrtBYY_vs_GFPYY_All, pcrtBBB_vs_GFPBB_All, pcrtB28h_vs_GFP28h_All, pcrtB34h_vs_GFP34h_All, pcrtBZZ_vs_GFPZZ_All, 
         pcrtB37h_vs_GFP37h_All, pcrtBCC_vs_GFPCC_All, pcrtB40h_vs_GFP40h_All, pcrtBDD_vs_GFPDD_All, pcrtB46h_vs_GFP46h_All, 
         pcrtBEE_vs_GFPEE_All, pcrtB56h_vs_GFP56h_All, pcrtB96h_vs_GFP96h_All)


min(Niben_Rotamase_FC, na.rm = TRUE)
max(Niben_Rotamase_FC, na.rm = TRUE)
breaks_Rotamase_FC=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Niben_Rotamase_FC, # %>% select(1:15), 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Rotamase_FC,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_Rotamase_Annotation, annotation_colors=color_Rotamase, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h", "96h"),
         cex = 1,
         angle_col=45,
         main = "Rotamase",
         annotation_legend = TRUE,
         display_numbers = Niben_Rotamase_DEG, # %>% select(1:15),
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(84, 104),
         annotation_row = Niben_Rotamase_Modules_Chl,
         annotation_colors = module_Chl_colors
         #gaps_col = c(15)
)


rownames(Niben_Rotamase_TPM) <- Niben_Rotamase_TPM$Protein
Niben_Rotamase_TPM[1] <- NULL

min(Niben_Rotamase_TPM, na.rm = TRUE)
max(Niben_Rotamase_TPM, na.rm = TRUE)
breaks_Niben_Rotamase_TPM=c(seq(0,100, length.out = 100))

Niben_Rotamase_TPM_norm <- normalize_df(Niben_Rotamase_TPM)
Niben_Rotamase_TPM_norm_Cornell <- Niben_Rotamase_TPM %>%
  select(1:48) 
Niben_Rotamase_TPM_norm_Cornell <- normalize_df(Niben_Rotamase_TPM_norm_Cornell)


pheatmap(Niben_Rotamase_TPM, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         breaks = breaks_Niben_Rotamase_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_Rotamase_Annotation, annotation_colors=color_Rotamase, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Rotamase",
         annotation_legend = TRUE,
         #display_numbers = Niben_Rotamase_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         annotation_row = Niben_Rotamase_Modules_Chl,
         annotation_colors = module_Chl_colors,
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(24)#, 48, 51)
)



#####Representing Heatmap for Sly

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Sly_Rotamase_Ann <- read.table("Funcional Groups/FigTree/Sly_Rotamase.txt",
                            sep = "\t", header = TRUE)

colnames(Sly_Rotamase_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Sly_Rotamase <- merge(Sly_Rotamase_Ann, Summary_Yazdani_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Sly_Rotamase_TPM <- Sly_Rotamase %>% select(Protein, 4:66)
Sly_Rotamase_TPM <- Sly_Rotamase_TPM[!duplicated(Sly_Rotamase_TPM$Protein),]
Sly_Rotamase_TPM <- Sly_Rotamase_TPM[order(match(Sly_Rotamase_TPM[,1], Sly_Rotamase_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_Rotamase_FC <- Sly_Rotamase %>% select(Protein, 67:71)
Sly_Rotamase_FC <- Sly_Rotamase_FC[!duplicated(Sly_Rotamase_FC$Protein),]
Sly_Rotamase_FC <- Sly_Rotamase_FC[order(match(Sly_Rotamase_FC[,1], Sly_Rotamase_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_Rotamase_DEG <- Sly_Rotamase %>% select(Protein, 72:76)
Sly_Rotamase_DEG <- Sly_Rotamase_DEG[!duplicated(Sly_Rotamase_DEG$Protein),]
Sly_Rotamase_DEG <- Sly_Rotamase_DEG[order(match(Sly_Rotamase_DEG[,1], Sly_Rotamase_Ann[,2])),]

Sly_Rotamase_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Sly_Rotamase_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Sly_Rotamase_DEG$Ripening_Orange_DEG <- ifelse(is.na(Sly_Rotamase_DEG$Ripening_Orange_DEG) == TRUE, " ", "*")
Sly_Rotamase_DEG$Ripening_Red_DEG <- ifelse(is.na(Sly_Rotamase_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
Sly_Rotamase_DEG$ORHis_MG_DEG <- ifelse(is.na(Sly_Rotamase_DEG$ORHis_MG_DEG) == TRUE, " ", "*")
Sly_Rotamase_DEG$ORHis_Breaker_DEG <- ifelse(is.na(Sly_Rotamase_DEG$ORHis_Breaker_DEG) == TRUE, " ", "*")
row.names(Sly_Rotamase_DEG) <- Sly_Rotamase_DEG$Protein
Sly_Rotamase_DEG[1] <- NULL

rownames(Sly_Rotamase_FC) <- Sly_Rotamase_FC$Protein
Sly_Rotamase_FC[1] <- NULL

min(Sly_Rotamase_FC, na.rm = TRUE)
max(Sly_Rotamase_FC, na.rm = TRUE)
breaks_Sly_Rotamase=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Sly_Rotamase_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Sly_Rotamase,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_Rotamase_Annotation, annotation_colors=color_Rotamase, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Rotamase",
         annotation_legend = TRUE,
         display_numbers = Sly_Rotamase_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(39, 48),
         gaps_col = c(3)
)


rownames(Sly_Rotamase_TPM) <- Sly_Rotamase_TPM$Protein
Sly_Rotamase_TPM[1] <- NULL

min(Sly_Rotamase_TPM, na.rm = TRUE)
max(Sly_Rotamase_TPM, na.rm = TRUE)
breaks_Sly_Rotamase_TPM=c(seq(0,100, length.out = 200))

Sly_Rotamase_TPM_norm <- normalize_df(Sly_Rotamase_TPM)

pheatmap(Sly_Rotamase_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Sly_Rotamase_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_Rotamase_Annotation, annotation_colors=color_Rotamase, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Rotamase",
         annotation_legend = TRUE,
         #display_numbers = Sly_Rotamase_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(21,42)
)


#####Representing Heatmap for Cann

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Cann_Rotamase_Ann <- read.table("Funcional Groups/FigTree/Cann_Rotamase.txt",
                             sep = "\t", header = TRUE)

colnames(Cann_Rotamase_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Cann_Rotamase <- merge(Cann_Rotamase_Ann, Summary_Pepper_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Cann_Rotamase_TPM <- Cann_Rotamase %>% select(Protein, 4:17)
Cann_Rotamase_TPM <- Cann_Rotamase_TPM[!duplicated(Cann_Rotamase_TPM$Protein),]
Cann_Rotamase_TPM <- Cann_Rotamase_TPM[order(match(Cann_Rotamase_TPM[,1], Cann_Rotamase_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_Rotamase_FC <- Cann_Rotamase %>% select(Protein, 18:19)
Cann_Rotamase_FC <- Cann_Rotamase_FC[!duplicated(Cann_Rotamase_FC$Protein),]
Cann_Rotamase_FC <- Cann_Rotamase_FC[order(match(Cann_Rotamase_FC[,1], Cann_Rotamase_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_Rotamase_DEG <- Cann_Rotamase %>% select(Protein, 20:21)
Cann_Rotamase_DEG <- Cann_Rotamase_DEG[!duplicated(Cann_Rotamase_DEG$Protein),]
Cann_Rotamase_DEG <- Cann_Rotamase_DEG[order(match(Cann_Rotamase_DEG[,1], Cann_Rotamase_Ann[,2])),]

Cann_Rotamase_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Cann_Rotamase_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Cann_Rotamase_DEG$Ripening_Red_DEG <- ifelse(is.na(Cann_Rotamase_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
row.names(Cann_Rotamase_DEG) <- Cann_Rotamase_DEG$Protein
Cann_Rotamase_DEG[1] <- NULL

rownames(Cann_Rotamase_FC) <- Cann_Rotamase_FC$Protein
Cann_Rotamase_FC[1] <- NULL

min(Cann_Rotamase_FC, na.rm = TRUE)
max(Cann_Rotamase_FC, na.rm = TRUE)
breaks_Cann_Rotamase=c(seq(-4.5,0, length.out = 100), seq(0.01,6,length.out = 100))

pheatmap(Cann_Rotamase_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Cann_Rotamase,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_Rotamase_Annotation, annotation_colors=color_Rotamase, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Red"),
         cex = 1,
         angle_col=45,
         main = "Rotamase",
         annotation_legend = TRUE,
         display_numbers = Cann_Rotamase_DEG,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)


rownames(Cann_Rotamase_TPM) <- Cann_Rotamase_TPM$Protein
Cann_Rotamase_TPM[1] <- NULL

min(Cann_Rotamase_TPM, na.rm = TRUE)
max(Cann_Rotamase_TPM, na.rm = TRUE)
breaks_Cann_Rotamase_TPM=c(seq(0,100, length.out = 200))

Cann_Rotamase_TPM_norm <- normalize_df(Cann_Rotamase_TPM)

pheatmap(Cann_Rotamase_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Cann_Rotamase_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_Rotamase_Annotation, annotation_colors=color_Rotamase, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Rotamase",
         annotation_legend = TRUE,
         #display_numbers = Cann_Rotamase_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(4,9)
)

#########################################################################################################################################
####################################  PDI  #############################################################################################
#########################################################################################################################################

PDI_Ath <- read.delim("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan/Funcional Groups/Ath_PDI.txt")
PDI_Ath <- merge(PDI_Ath, Ath_Gene_Models, by.x="GeneID", by.y=1)
PDI_Ath_fasta <- Ath11_fasta[c(which(names(Ath11_fasta) %in% PDI_Ath$V2))]
PDI_Ath_2 <- PDI_Ath %>% select(V2, Name)
write.fasta(PDI_Ath_fasta, as.string=TRUE, names= names(PDI_Ath_fasta),
            file.out = "Funcional Groups/Ath_PDI.fasta")
rename.fasta(infile = "Funcional Groups/Ath_PDI.fasta", PDI_Ath_2, outfile = "Funcional Groups/Ath_PDI_2.fasta")

PDI_Niben_Candidates <- read.table("Funcional Groups/Niben_Candidates_PDI.txt")
PDI_Niben_Candidates_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% PDI_Niben_Candidates$V1))]
write.fasta(PDI_Niben_Candidates_fasta, as.string=TRUE, names= names(PDI_Niben_Candidates_fasta),
            file.out = "Funcional Groups/PDI_Niben_Candidates_35.fasta")

PDI_Sly_Candidates <- read.table("Funcional Groups/Sly_Candidates_PDI.txt")
PDI_Sly_Candidates_fasta <- Tomato_4.0_fasta[c(which(names(Tomato_4.0_fasta) %in% PDI_Sly_Candidates$V1))]
write.fasta(PDI_Sly_Candidates_fasta, as.string=TRUE, names= names(PDI_Sly_Candidates_fasta),
            file.out = "Funcional Groups/PDI_Sly_Candidates_35.fasta")

PDI_Cann_Candidates <- read.table("Funcional Groups/Cann_Candidates_PDI.txt")
PDI_Cann_Candidates_fasta <- Cannuum_ASM51225v2_fasta[c(which(names(Cannuum_ASM51225v2_fasta) %in% PDI_Cann_Candidates$V1))]
write.fasta(PDI_Cann_Candidates_fasta, as.string=TRUE, names= names(PDI_Cann_Candidates_fasta),
            file.out = "Funcional Groups/PDI_Cann_Candidates_35.fasta")

Niben_PDI_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_PDI_Ann$Name))]
write.fasta(Niben_PDI_fasta, as.string=TRUE, names= names(Niben_PDI_fasta),
            file.out = "Funcional Groups/Niben_PDI.fasta")
Niben_PDI_Ann_2 <- Niben_PDI_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_PDI.fasta", Niben_PDI_Ann, outfile = "Funcional Groups/Niben_PDI_2.fasta")

save_heatmaps("PDI", height = 10)


## Representing Heatmap

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Niben_PDI_Ann <- read.table("Funcional Groups/FigTree/Niben_PDI.txt",
                              sep = "\t", header = TRUE)

colnames(Niben_PDI_Ann) <- c("Name", "Protein")

Niben_PDI_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_PDI_Ann$Name))]
write.fasta(Niben_PDI_fasta, as.string=TRUE, names= names(Niben_PDI_fasta),
            file.out = "Funcional Groups/Niben_PDI.fasta")
Niben_PDI_Ann_2 <- Niben_PDI_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_PDI.fasta", Niben_PDI_Ann, outfile = "Funcional Groups/Niben_PDI_2.fasta")

Niben_PDI_Chloro <- read.delim("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan/Funcional Groups/Niben_PDI_2_summary.targetp2", 
                                 header=FALSE, comment.char="#")
colnames(Niben_PDI_Chloro) <- c("ID",	"Prediction",	"noTP",	"SP",	"mTP",	"cTP",	"luTP",	"CS-Position")
Niben_PDI_Chloro$Prediction <- factor(Niben_PDI_Chloro$Prediction, levels = c("cTP","mTP","noTP"))
Niben_PDI_Chloro <- merge(Niben_PDI_Ann, Niben_PDI_Chloro, by.x = "Protein", by.y = "ID")

Niben_PDI_Chloro <- Niben_PDI_Chloro[order(match(Niben_PDI_Chloro$Protein, Niben_PDI_Ann$Protein)),]
rownames(Niben_PDI_Chloro) <- Niben_PDI_Chloro$Protein
Niben_PDI_Chloro <- Niben_PDI_Chloro %>% select(Prediction)


## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_PDI <- merge(Niben_PDI_Ann, TPM_All_FC_Annotated, by="Name")

## We create the annotation for Modules

Niben_PDI_Modules <- which_Module_pwd30(Niben_PDI_Ann)

## We merge this annotation of Modules with the annotation of Chloroplast TP prediction

Niben_PDI_Modules_Chl <- merge(Niben_PDI_Modules, Niben_PDI_Chloro, by = 0)
Niben_PDI_Modules_Chl <- Niben_PDI_Modules_Chl[order(match(Niben_PDI_Modules_Chl$Row.names, Niben_PDI_Ann$Protein)),]
rownames(Niben_PDI_Modules_Chl) <- Niben_PDI_Modules_Chl$Row.names
Niben_PDI_Modules_Chl$Row.names <- NULL

## We select only the FC data from the different comparisons
Niben_PDI_TPM <- Niben_PDI %>% select(Protein, 4:51, 61:66)
Niben_PDI_TPM <- Niben_PDI_TPM[!duplicated(Niben_PDI_TPM$Protein),]
Niben_PDI_TPM <- Niben_PDI_TPM[order(match(Niben_PDI_TPM[,1], Niben_PDI_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_PDI_FC <- Niben_PDI %>% select(Protein, 71, 73:87)
Niben_PDI_FC <- Niben_PDI_FC[!duplicated(Niben_PDI_FC$Protein),]
Niben_PDI_FC <- Niben_PDI_FC[order(match(Niben_PDI_FC[,1], Niben_PDI_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_PDI_DEG <- Niben_PDI %>% select(Protein, 89, 91:105)
Niben_PDI_DEG <- Niben_PDI_DEG[!duplicated(Niben_PDI_DEG$Protein),]
Niben_PDI_DEG <- Niben_PDI_DEG[order(match(Niben_PDI_DEG[,1], Niben_PDI_Ann[,2])),]

Niben_PDI_DEG$pcrtB22h_vs_GFP22h_filtered <- ifelse(is.na(Niben_PDI_DEG$pcrtB22h_vs_GFP22h_filtered) == TRUE, " ", "*")
Niben_PDI_DEG$pcrtB25h_vs_GFP25h_filtered <- ifelse(is.na(Niben_PDI_DEG$pcrtB25h_vs_GFP25h_filtered) == TRUE, " ", "*")
Niben_PDI_DEG$pcrtBAA_vs_GFPAA_filtered <- ifelse(is.na(Niben_PDI_DEG$pcrtBAA_vs_GFPAA_filtered) == TRUE, " ", "*")
Niben_PDI_DEG$pcrtBBB_vs_GFPBB_filtered <- ifelse(is.na(Niben_PDI_DEG$pcrtBBB_vs_GFPBB_filtered) == TRUE, " ", "*")
Niben_PDI_DEG$pcrtBYY_vs_GFPYY_filtered <- ifelse(is.na(Niben_PDI_DEG$pcrtBYY_vs_GFPYY_filtered) == TRUE, " ", "*")
Niben_PDI_DEG$pcrtB28h_vs_GFP28h_filtered <- ifelse(is.na(Niben_PDI_DEG$pcrtB28h_vs_GFP28h_filtered) == TRUE, " ", "*")
Niben_PDI_DEG$pcrtB34h_vs_GFP34h_filtered <- ifelse(is.na(Niben_PDI_DEG$pcrtB34h_vs_GFP34h_filtered) == TRUE, " ", "*")
Niben_PDI_DEG$pcrtB37h_vs_GFP37h_filtered <- ifelse(is.na(Niben_PDI_DEG$pcrtB37h_vs_GFP37h_filtered) == TRUE, " ", "*")
Niben_PDI_DEG$pcrtBZZ_vs_GFPZZ_filtered <- ifelse(is.na(Niben_PDI_DEG$pcrtBZZ_vs_GFPZZ_filtered) == TRUE, " ", "*")
Niben_PDI_DEG$pcrtBCC_vs_GFPCC_filtered <- ifelse(is.na(Niben_PDI_DEG$pcrtBCC_vs_GFPCC_filtered) == TRUE, " ", "*")
Niben_PDI_DEG$pcrtB40h_vs_GFP40h_filtered <- ifelse(is.na(Niben_PDI_DEG$pcrtB40h_vs_GFP40h_filtered) == TRUE, " ", "*")
Niben_PDI_DEG$pcrtBDD_vs_GFPDD_filtered <- ifelse(is.na(Niben_PDI_DEG$pcrtBDD_vs_GFPDD_filtered) == TRUE, " ", "*")
Niben_PDI_DEG$pcrtB46h_vs_GFP46h_filtered <- ifelse(is.na(Niben_PDI_DEG$pcrtB46h_vs_GFP46h_filtered) == TRUE, " ", "*")
Niben_PDI_DEG$pcrtB56h_vs_GFP56h_filtered <- ifelse(is.na(Niben_PDI_DEG$pcrtB56h_vs_GFP56h_filtered) == TRUE, " ", "*")
Niben_PDI_DEG$pcrtBEE_vs_GFPEE_filtered <- ifelse(is.na(Niben_PDI_DEG$pcrtBEE_vs_GFPEE_filtered) == TRUE, " ", "*")
Niben_PDI_DEG$pcrtB96h_vs_GFP96h_filtered <- ifelse(is.na(Niben_PDI_DEG$pcrtB96h_vs_GFP96h_filtered) == TRUE, " ", "*")
row.names(Niben_PDI_DEG) <- Niben_PDI_DEG$Protein
Niben_PDI_DEG[1] <- NULL

Niben_PDI_DEG <- Niben_PDI_DEG %>%
  select(pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

row.names(Niben_PDI_FC) <- Niben_PDI_FC$Protein
Niben_PDI_FC[1] <- NULL

Niben_PDI_FC <- Niben_PDI_FC %>%
  select(pcrtB22h_vs_GFP22h_All, pcrtBAA_vs_GFPAA_All, pcrtB25h_vs_GFP25h_All, 
         pcrtBYY_vs_GFPYY_All, pcrtBBB_vs_GFPBB_All, pcrtB28h_vs_GFP28h_All, pcrtB34h_vs_GFP34h_All, pcrtBZZ_vs_GFPZZ_All, 
         pcrtB37h_vs_GFP37h_All, pcrtBCC_vs_GFPCC_All, pcrtB40h_vs_GFP40h_All, pcrtBDD_vs_GFPDD_All, pcrtB46h_vs_GFP46h_All, 
         pcrtBEE_vs_GFPEE_All, pcrtB56h_vs_GFP56h_All, pcrtB96h_vs_GFP96h_All)


min(Niben_PDI_FC, na.rm = TRUE)
max(Niben_PDI_FC, na.rm = TRUE)
breaks_PDI_FC=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Niben_PDI_FC, # %>% select(1:15), 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_PDI_FC,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_PDI_Annotation, annotation_colors=color_PDI, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h", "96h"),
         cex = 1,
         angle_col=45,
         main = "PDI",
         annotation_legend = TRUE,
         display_numbers = Niben_PDI_DEG, # %>% select(1:15),
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(84, 104),
         annotation_row = Niben_PDI_Modules_Chl,
         annotation_colors = module_Chl_colors
         #gaps_col = c(15)
)


rownames(Niben_PDI_TPM) <- Niben_PDI_TPM$Protein
Niben_PDI_TPM[1] <- NULL

min(Niben_PDI_TPM, na.rm = TRUE)
max(Niben_PDI_TPM, na.rm = TRUE)
breaks_Niben_PDI_TPM=c(seq(0,100, length.out = 100))

Niben_PDI_TPM_norm <- normalize_df(Niben_PDI_TPM)
Niben_PDI_TPM_norm_Cornell <- Niben_PDI_TPM %>%
  select(1:48) 
Niben_PDI_TPM_norm_Cornell <- normalize_df(Niben_PDI_TPM_norm_Cornell)


pheatmap(Niben_PDI_TPM, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         breaks = breaks_Niben_PDI_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_PDI_Annotation, annotation_colors=color_PDI, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "PDI",
         annotation_legend = TRUE,
         #display_numbers = Niben_PDI_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         annotation_row = Niben_PDI_Modules_Chl,
         annotation_colors = module_Chl_colors,
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(24)#, 48, 51)
)



#####Representing Heatmap for Sly

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Sly_PDI_Ann <- read.table("Funcional Groups/FigTree/Sly_PDI.txt",
                            sep = "\t", header = TRUE)

colnames(Sly_PDI_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Sly_PDI <- merge(Sly_PDI_Ann, Summary_Yazdani_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Sly_PDI_TPM <- Sly_PDI %>% select(Protein, 4:66)
Sly_PDI_TPM <- Sly_PDI_TPM[!duplicated(Sly_PDI_TPM$Protein),]
Sly_PDI_TPM <- Sly_PDI_TPM[order(match(Sly_PDI_TPM[,1], Sly_PDI_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_PDI_FC <- Sly_PDI %>% select(Protein, 67:71)
Sly_PDI_FC <- Sly_PDI_FC[!duplicated(Sly_PDI_FC$Protein),]
Sly_PDI_FC <- Sly_PDI_FC[order(match(Sly_PDI_FC[,1], Sly_PDI_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_PDI_DEG <- Sly_PDI %>% select(Protein, 72:76)
Sly_PDI_DEG <- Sly_PDI_DEG[!duplicated(Sly_PDI_DEG$Protein),]
Sly_PDI_DEG <- Sly_PDI_DEG[order(match(Sly_PDI_DEG[,1], Sly_PDI_Ann[,2])),]

Sly_PDI_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Sly_PDI_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Sly_PDI_DEG$Ripening_Orange_DEG <- ifelse(is.na(Sly_PDI_DEG$Ripening_Orange_DEG) == TRUE, " ", "*")
Sly_PDI_DEG$Ripening_Red_DEG <- ifelse(is.na(Sly_PDI_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
Sly_PDI_DEG$ORHis_MG_DEG <- ifelse(is.na(Sly_PDI_DEG$ORHis_MG_DEG) == TRUE, " ", "*")
Sly_PDI_DEG$ORHis_Breaker_DEG <- ifelse(is.na(Sly_PDI_DEG$ORHis_Breaker_DEG) == TRUE, " ", "*")
row.names(Sly_PDI_DEG) <- Sly_PDI_DEG$Protein
Sly_PDI_DEG[1] <- NULL

rownames(Sly_PDI_FC) <- Sly_PDI_FC$Protein
Sly_PDI_FC[1] <- NULL

min(Sly_PDI_FC, na.rm = TRUE)
max(Sly_PDI_FC, na.rm = TRUE)
breaks_Sly_PDI=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Sly_PDI_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Sly_PDI,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_PDI_Annotation, annotation_colors=color_PDI, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "PDI",
         annotation_legend = TRUE,
         display_numbers = Sly_PDI_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(39, 48),
         gaps_col = c(3)
)


rownames(Sly_PDI_TPM) <- Sly_PDI_TPM$Protein
Sly_PDI_TPM[1] <- NULL

min(Sly_PDI_TPM, na.rm = TRUE)
max(Sly_PDI_TPM, na.rm = TRUE)
breaks_Sly_PDI_TPM=c(seq(0,100, length.out = 200))

Sly_PDI_TPM_norm <- normalize_df(Sly_PDI_TPM)

pheatmap(Sly_PDI_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Sly_PDI_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_PDI_Annotation, annotation_colors=color_PDI, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "PDI",
         annotation_legend = TRUE,
         #display_numbers = Sly_PDI_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(21,42)
)


#####Representing Heatmap for Cann

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Cann_PDI_Ann <- read.table("Funcional Groups/FigTree/Cann_PDI.txt",
                             sep = "\t", header = TRUE)

colnames(Cann_PDI_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Cann_PDI <- merge(Cann_PDI_Ann, Summary_Pepper_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Cann_PDI_TPM <- Cann_PDI %>% select(Protein, 4:17)
Cann_PDI_TPM <- Cann_PDI_TPM[!duplicated(Cann_PDI_TPM$Protein),]
Cann_PDI_TPM <- Cann_PDI_TPM[order(match(Cann_PDI_TPM[,1], Cann_PDI_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_PDI_FC <- Cann_PDI %>% select(Protein, 18:19)
Cann_PDI_FC <- Cann_PDI_FC[!duplicated(Cann_PDI_FC$Protein),]
Cann_PDI_FC <- Cann_PDI_FC[order(match(Cann_PDI_FC[,1], Cann_PDI_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_PDI_DEG <- Cann_PDI %>% select(Protein, 20:21)
Cann_PDI_DEG <- Cann_PDI_DEG[!duplicated(Cann_PDI_DEG$Protein),]
Cann_PDI_DEG <- Cann_PDI_DEG[order(match(Cann_PDI_DEG[,1], Cann_PDI_Ann[,2])),]

Cann_PDI_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Cann_PDI_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Cann_PDI_DEG$Ripening_Red_DEG <- ifelse(is.na(Cann_PDI_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
row.names(Cann_PDI_DEG) <- Cann_PDI_DEG$Protein
Cann_PDI_DEG[1] <- NULL

rownames(Cann_PDI_FC) <- Cann_PDI_FC$Protein
Cann_PDI_FC[1] <- NULL

min(Cann_PDI_FC, na.rm = TRUE)
max(Cann_PDI_FC, na.rm = TRUE)
breaks_Cann_PDI=c(seq(-4.5,0, length.out = 100), seq(0.01,6,length.out = 100))

pheatmap(Cann_PDI_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Cann_PDI,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_PDI_Annotation, annotation_colors=color_PDI, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Red"),
         cex = 1,
         angle_col=45,
         main = "PDI",
         annotation_legend = TRUE,
         display_numbers = Cann_PDI_DEG,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)


rownames(Cann_PDI_TPM) <- Cann_PDI_TPM$Protein
Cann_PDI_TPM[1] <- NULL

min(Cann_PDI_TPM, na.rm = TRUE)
max(Cann_PDI_TPM, na.rm = TRUE)
breaks_Cann_PDI_TPM=c(seq(0,100, length.out = 200))

Cann_PDI_TPM_norm <- normalize_df(Cann_PDI_TPM)

pheatmap(Cann_PDI_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Cann_PDI_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_PDI_Annotation, annotation_colors=color_PDI, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "PDI",
         annotation_legend = TRUE,
         #display_numbers = Cann_PDI_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(4,9)
)

#########################################################################################################################################
####################################  HSP90_CoChap  #############################################################################################
#########################################################################################################################################

HSP90_CoChap_Ath <- read.delim("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan/Funcional Groups/Ath_HSP90_CoChap.txt")
HSP90_CoChap_Ath <- merge(HSP90_CoChap_Ath, Ath_Gene_Models, by.x="GeneID", by.y=1)
HSP90_CoChap_Ath_fasta <- Ath11_fasta[c(which(names(Ath11_fasta) %in% HSP90_CoChap_Ath$V2))]
HSP90_CoChap_Ath_2 <- HSP90_CoChap_Ath %>% select(V2, Name)
write.fasta(HSP90_CoChap_Ath_fasta, as.string=TRUE, names= names(HSP90_CoChap_Ath_fasta),
            file.out = "Funcional Groups/Ath_HSP90_CoChap.fasta")
rename.fasta(infile = "Funcional Groups/Ath_HSP90_CoChap.fasta", HSP90_CoChap_Ath_2, outfile = "Funcional Groups/Ath_HSP90_CoChap_2.fasta")

HSP90_CoChap_Niben_Candidates <- read.table("Funcional Groups/Niben_Candidates_HSP90_CoChap.txt")
HSP90_CoChap_Niben_Candidates_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% HSP90_CoChap_Niben_Candidates$V1))]
write.fasta(HSP90_CoChap_Niben_Candidates_fasta, as.string=TRUE, names= names(HSP90_CoChap_Niben_Candidates_fasta),
            file.out = "Funcional Groups/HSP90_CoChap_Niben_Candidates_35.fasta")

HSP90_CoChap_Sly_Candidates <- read.table("Funcional Groups/Sly_Candidates_HSP90_CoChap.txt")
HSP90_CoChap_Sly_Candidates_fasta <- Tomato_4.0_fasta[c(which(names(Tomato_4.0_fasta) %in% HSP90_CoChap_Sly_Candidates$V1))]
write.fasta(HSP90_CoChap_Sly_Candidates_fasta, as.string=TRUE, names= names(HSP90_CoChap_Sly_Candidates_fasta),
            file.out = "Funcional Groups/HSP90_CoChap_Sly_Candidates_35.fasta")

HSP90_CoChap_Cann_Candidates <- read.table("Funcional Groups/Cann_Candidates_HSP90_CoChap.txt")
HSP90_CoChap_Cann_Candidates_fasta <- Cannuum_ASM51225v2_fasta[c(which(names(Cannuum_ASM51225v2_fasta) %in% HSP90_CoChap_Cann_Candidates$V1))]
write.fasta(HSP90_CoChap_Cann_Candidates_fasta, as.string=TRUE, names= names(HSP90_CoChap_Cann_Candidates_fasta),
            file.out = "Funcional Groups/HSP90_CoChap_Cann_Candidates_35.fasta")

Niben_HSP90_CoChap_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_HSP90_CoChap_Ann$Name))]
write.fasta(Niben_HSP90_CoChap_fasta, as.string=TRUE, names= names(Niben_HSP90_CoChap_fasta),
            file.out = "Funcional Groups/Niben_HSP90_CoChap.fasta")
Niben_HSP90_CoChap_Ann_2 <- Niben_HSP90_CoChap_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_HSP90_CoChap.fasta", Niben_HSP90_CoChap_Ann, outfile = "Funcional Groups/Niben_HSP90_CoChap_2.fasta")

save_heatmaps("HSP90_CoChap", height = 13)


## Representing Heatmap

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Niben_HSP90_CoChap_Ann <- read.table("Funcional Groups/FigTree/Niben_HSP90_CoChap.txt",
                              sep = "\t", header = TRUE)

colnames(Niben_HSP90_CoChap_Ann) <- c("Name", "Protein")

Niben_HSP90_CoChap_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_HSP90_CoChap_Ann$Name))]
write.fasta(Niben_HSP90_CoChap_fasta, as.string=TRUE, names= names(Niben_HSP90_CoChap_fasta),
            file.out = "Funcional Groups/Niben_HSP90_CoChap.fasta")
Niben_HSP90_CoChap_Ann_2 <- Niben_HSP90_CoChap_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_HSP90_CoChap.fasta", Niben_HSP90_CoChap_Ann, outfile = "Funcional Groups/Niben_HSP90_CoChap_2.fasta")

Niben_HSP90_CoChap_Chloro <- read.delim("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan/Funcional Groups/Niben_HSP90_CoChap_2_summary.targetp2", 
                                 header=FALSE, comment.char="#")
colnames(Niben_HSP90_CoChap_Chloro) <- c("ID",	"Prediction",	"noTP",	"SP",	"mTP",	"cTP",	"luTP",	"CS-Position")
Niben_HSP90_CoChap_Chloro$Prediction <- factor(Niben_HSP90_CoChap_Chloro$Prediction, levels = c("cTP","mTP","noTP"))
Niben_HSP90_CoChap_Chloro <- merge(Niben_HSP90_CoChap_Ann, Niben_HSP90_CoChap_Chloro, by.x = "Protein", by.y = "ID")

Niben_HSP90_CoChap_Chloro <- Niben_HSP90_CoChap_Chloro[order(match(Niben_HSP90_CoChap_Chloro$Protein, Niben_HSP90_CoChap_Ann$Protein)),]
rownames(Niben_HSP90_CoChap_Chloro) <- Niben_HSP90_CoChap_Chloro$Protein
Niben_HSP90_CoChap_Chloro <- Niben_HSP90_CoChap_Chloro %>% select(Prediction)


## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_HSP90_CoChap <- merge(Niben_HSP90_CoChap_Ann, TPM_All_FC_Annotated, by="Name")

## We create the annotation for Modules

Niben_HSP90_CoChap_Modules <- which_Module_pwd30(Niben_HSP90_CoChap_Ann)

## We merge this annotation of Modules with the annotation of Chloroplast TP prediction

Niben_HSP90_CoChap_Modules_Chl <- merge(Niben_HSP90_CoChap_Modules, Niben_HSP90_CoChap_Chloro, by = 0)
Niben_HSP90_CoChap_Modules_Chl <- Niben_HSP90_CoChap_Modules_Chl[order(match(Niben_HSP90_CoChap_Modules_Chl$Row.names, Niben_HSP90_CoChap_Ann$Protein)),]
rownames(Niben_HSP90_CoChap_Modules_Chl) <- Niben_HSP90_CoChap_Modules_Chl$Row.names
Niben_HSP90_CoChap_Modules_Chl$Row.names <- NULL

## We select only the FC data from the different comparisons
Niben_HSP90_CoChap_TPM <- Niben_HSP90_CoChap %>% select(Protein, 4:51, 61:66)
Niben_HSP90_CoChap_TPM <- Niben_HSP90_CoChap_TPM[!duplicated(Niben_HSP90_CoChap_TPM$Protein),]
Niben_HSP90_CoChap_TPM <- Niben_HSP90_CoChap_TPM[order(match(Niben_HSP90_CoChap_TPM[,1], Niben_HSP90_CoChap_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_HSP90_CoChap_FC <- Niben_HSP90_CoChap %>% select(Protein, 71, 73:87)
Niben_HSP90_CoChap_FC <- Niben_HSP90_CoChap_FC[!duplicated(Niben_HSP90_CoChap_FC$Protein),]
Niben_HSP90_CoChap_FC <- Niben_HSP90_CoChap_FC[order(match(Niben_HSP90_CoChap_FC[,1], Niben_HSP90_CoChap_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_HSP90_CoChap_DEG <- Niben_HSP90_CoChap %>% select(Protein, 89, 91:105)
Niben_HSP90_CoChap_DEG <- Niben_HSP90_CoChap_DEG[!duplicated(Niben_HSP90_CoChap_DEG$Protein),]
Niben_HSP90_CoChap_DEG <- Niben_HSP90_CoChap_DEG[order(match(Niben_HSP90_CoChap_DEG[,1], Niben_HSP90_CoChap_Ann[,2])),]

Niben_HSP90_CoChap_DEG$pcrtB22h_vs_GFP22h_filtered <- ifelse(is.na(Niben_HSP90_CoChap_DEG$pcrtB22h_vs_GFP22h_filtered) == TRUE, " ", "*")
Niben_HSP90_CoChap_DEG$pcrtB25h_vs_GFP25h_filtered <- ifelse(is.na(Niben_HSP90_CoChap_DEG$pcrtB25h_vs_GFP25h_filtered) == TRUE, " ", "*")
Niben_HSP90_CoChap_DEG$pcrtBAA_vs_GFPAA_filtered <- ifelse(is.na(Niben_HSP90_CoChap_DEG$pcrtBAA_vs_GFPAA_filtered) == TRUE, " ", "*")
Niben_HSP90_CoChap_DEG$pcrtBBB_vs_GFPBB_filtered <- ifelse(is.na(Niben_HSP90_CoChap_DEG$pcrtBBB_vs_GFPBB_filtered) == TRUE, " ", "*")
Niben_HSP90_CoChap_DEG$pcrtBYY_vs_GFPYY_filtered <- ifelse(is.na(Niben_HSP90_CoChap_DEG$pcrtBYY_vs_GFPYY_filtered) == TRUE, " ", "*")
Niben_HSP90_CoChap_DEG$pcrtB28h_vs_GFP28h_filtered <- ifelse(is.na(Niben_HSP90_CoChap_DEG$pcrtB28h_vs_GFP28h_filtered) == TRUE, " ", "*")
Niben_HSP90_CoChap_DEG$pcrtB34h_vs_GFP34h_filtered <- ifelse(is.na(Niben_HSP90_CoChap_DEG$pcrtB34h_vs_GFP34h_filtered) == TRUE, " ", "*")
Niben_HSP90_CoChap_DEG$pcrtB37h_vs_GFP37h_filtered <- ifelse(is.na(Niben_HSP90_CoChap_DEG$pcrtB37h_vs_GFP37h_filtered) == TRUE, " ", "*")
Niben_HSP90_CoChap_DEG$pcrtBZZ_vs_GFPZZ_filtered <- ifelse(is.na(Niben_HSP90_CoChap_DEG$pcrtBZZ_vs_GFPZZ_filtered) == TRUE, " ", "*")
Niben_HSP90_CoChap_DEG$pcrtBCC_vs_GFPCC_filtered <- ifelse(is.na(Niben_HSP90_CoChap_DEG$pcrtBCC_vs_GFPCC_filtered) == TRUE, " ", "*")
Niben_HSP90_CoChap_DEG$pcrtB40h_vs_GFP40h_filtered <- ifelse(is.na(Niben_HSP90_CoChap_DEG$pcrtB40h_vs_GFP40h_filtered) == TRUE, " ", "*")
Niben_HSP90_CoChap_DEG$pcrtBDD_vs_GFPDD_filtered <- ifelse(is.na(Niben_HSP90_CoChap_DEG$pcrtBDD_vs_GFPDD_filtered) == TRUE, " ", "*")
Niben_HSP90_CoChap_DEG$pcrtB46h_vs_GFP46h_filtered <- ifelse(is.na(Niben_HSP90_CoChap_DEG$pcrtB46h_vs_GFP46h_filtered) == TRUE, " ", "*")
Niben_HSP90_CoChap_DEG$pcrtB56h_vs_GFP56h_filtered <- ifelse(is.na(Niben_HSP90_CoChap_DEG$pcrtB56h_vs_GFP56h_filtered) == TRUE, " ", "*")
Niben_HSP90_CoChap_DEG$pcrtBEE_vs_GFPEE_filtered <- ifelse(is.na(Niben_HSP90_CoChap_DEG$pcrtBEE_vs_GFPEE_filtered) == TRUE, " ", "*")
Niben_HSP90_CoChap_DEG$pcrtB96h_vs_GFP96h_filtered <- ifelse(is.na(Niben_HSP90_CoChap_DEG$pcrtB96h_vs_GFP96h_filtered) == TRUE, " ", "*")
row.names(Niben_HSP90_CoChap_DEG) <- Niben_HSP90_CoChap_DEG$Protein
Niben_HSP90_CoChap_DEG[1] <- NULL

Niben_HSP90_CoChap_DEG <- Niben_HSP90_CoChap_DEG %>%
  select(pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

row.names(Niben_HSP90_CoChap_FC) <- Niben_HSP90_CoChap_FC$Protein
Niben_HSP90_CoChap_FC[1] <- NULL

Niben_HSP90_CoChap_FC <- Niben_HSP90_CoChap_FC %>%
  select(pcrtB22h_vs_GFP22h_All, pcrtBAA_vs_GFPAA_All, pcrtB25h_vs_GFP25h_All, 
         pcrtBYY_vs_GFPYY_All, pcrtBBB_vs_GFPBB_All, pcrtB28h_vs_GFP28h_All, pcrtB34h_vs_GFP34h_All, pcrtBZZ_vs_GFPZZ_All, 
         pcrtB37h_vs_GFP37h_All, pcrtBCC_vs_GFPCC_All, pcrtB40h_vs_GFP40h_All, pcrtBDD_vs_GFPDD_All, pcrtB46h_vs_GFP46h_All, 
         pcrtBEE_vs_GFPEE_All, pcrtB56h_vs_GFP56h_All, pcrtB96h_vs_GFP96h_All)


min(Niben_HSP90_CoChap_FC, na.rm = TRUE)
max(Niben_HSP90_CoChap_FC, na.rm = TRUE)
breaks_HSP90_CoChap_FC=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Niben_HSP90_CoChap_FC, # %>% select(1:15), 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_HSP90_CoChap_FC,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_HSP90_CoChap_Annotation, annotation_colors=color_HSP90_CoChap, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h", "96h"),
         cex = 1,
         angle_col=45,
         main = "HSP90_CoChap",
         annotation_legend = TRUE,
         display_numbers = Niben_HSP90_CoChap_DEG, # %>% select(1:15),
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(84, 104),
         annotation_row = Niben_HSP90_CoChap_Modules_Chl,
         annotation_colors = module_Chl_colors
         #gaps_col = c(15)
)


rownames(Niben_HSP90_CoChap_TPM) <- Niben_HSP90_CoChap_TPM$Protein
Niben_HSP90_CoChap_TPM[1] <- NULL

min(Niben_HSP90_CoChap_TPM, na.rm = TRUE)
max(Niben_HSP90_CoChap_TPM, na.rm = TRUE)
breaks_Niben_HSP90_CoChap_TPM=c(seq(0,100, length.out = 100))

Niben_HSP90_CoChap_TPM_norm <- normalize_df(Niben_HSP90_CoChap_TPM)
Niben_HSP90_CoChap_TPM_norm_Cornell <- Niben_HSP90_CoChap_TPM %>%
  select(1:48) 
Niben_HSP90_CoChap_TPM_norm_Cornell <- normalize_df(Niben_HSP90_CoChap_TPM_norm_Cornell)


pheatmap(Niben_HSP90_CoChap_TPM, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         breaks = breaks_Niben_HSP90_CoChap_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_HSP90_CoChap_Annotation, annotation_colors=color_HSP90_CoChap, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "HSP90_CoChap",
         annotation_legend = TRUE,
         #display_numbers = Niben_HSP90_CoChap_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         annotation_row = Niben_HSP90_CoChap_Modules_Chl,
         annotation_colors = module_Chl_colors,
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(24)#, 48, 51)
)



#####Representing Heatmap for Sly

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Sly_HSP90_CoChap_Ann <- read.table("Funcional Groups/FigTree/Sly_HSP90_CoChap.txt",
                            sep = "\t", header = TRUE)

colnames(Sly_HSP90_CoChap_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Sly_HSP90_CoChap <- merge(Sly_HSP90_CoChap_Ann, Summary_Yazdani_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Sly_HSP90_CoChap_TPM <- Sly_HSP90_CoChap %>% select(Protein, 4:66)
Sly_HSP90_CoChap_TPM <- Sly_HSP90_CoChap_TPM[!duplicated(Sly_HSP90_CoChap_TPM$Protein),]
Sly_HSP90_CoChap_TPM <- Sly_HSP90_CoChap_TPM[order(match(Sly_HSP90_CoChap_TPM[,1], Sly_HSP90_CoChap_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_HSP90_CoChap_FC <- Sly_HSP90_CoChap %>% select(Protein, 67:71)
Sly_HSP90_CoChap_FC <- Sly_HSP90_CoChap_FC[!duplicated(Sly_HSP90_CoChap_FC$Protein),]
Sly_HSP90_CoChap_FC <- Sly_HSP90_CoChap_FC[order(match(Sly_HSP90_CoChap_FC[,1], Sly_HSP90_CoChap_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_HSP90_CoChap_DEG <- Sly_HSP90_CoChap %>% select(Protein, 72:76)
Sly_HSP90_CoChap_DEG <- Sly_HSP90_CoChap_DEG[!duplicated(Sly_HSP90_CoChap_DEG$Protein),]
Sly_HSP90_CoChap_DEG <- Sly_HSP90_CoChap_DEG[order(match(Sly_HSP90_CoChap_DEG[,1], Sly_HSP90_CoChap_Ann[,2])),]

Sly_HSP90_CoChap_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Sly_HSP90_CoChap_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Sly_HSP90_CoChap_DEG$Ripening_Orange_DEG <- ifelse(is.na(Sly_HSP90_CoChap_DEG$Ripening_Orange_DEG) == TRUE, " ", "*")
Sly_HSP90_CoChap_DEG$Ripening_Red_DEG <- ifelse(is.na(Sly_HSP90_CoChap_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
Sly_HSP90_CoChap_DEG$ORHis_MG_DEG <- ifelse(is.na(Sly_HSP90_CoChap_DEG$ORHis_MG_DEG) == TRUE, " ", "*")
Sly_HSP90_CoChap_DEG$ORHis_Breaker_DEG <- ifelse(is.na(Sly_HSP90_CoChap_DEG$ORHis_Breaker_DEG) == TRUE, " ", "*")
row.names(Sly_HSP90_CoChap_DEG) <- Sly_HSP90_CoChap_DEG$Protein
Sly_HSP90_CoChap_DEG[1] <- NULL

rownames(Sly_HSP90_CoChap_FC) <- Sly_HSP90_CoChap_FC$Protein
Sly_HSP90_CoChap_FC[1] <- NULL

min(Sly_HSP90_CoChap_FC, na.rm = TRUE)
max(Sly_HSP90_CoChap_FC, na.rm = TRUE)
breaks_Sly_HSP90_CoChap=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Sly_HSP90_CoChap_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Sly_HSP90_CoChap,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_HSP90_CoChap_Annotation, annotation_colors=color_HSP90_CoChap, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "HSP90_CoChap",
         annotation_legend = TRUE,
         display_numbers = Sly_HSP90_CoChap_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(39, 48),
         gaps_col = c(3)
)


rownames(Sly_HSP90_CoChap_TPM) <- Sly_HSP90_CoChap_TPM$Protein
Sly_HSP90_CoChap_TPM[1] <- NULL

min(Sly_HSP90_CoChap_TPM, na.rm = TRUE)
max(Sly_HSP90_CoChap_TPM, na.rm = TRUE)
breaks_Sly_HSP90_CoChap_TPM=c(seq(0,100, length.out = 200))

Sly_HSP90_CoChap_TPM_norm <- normalize_df(Sly_HSP90_CoChap_TPM)

pheatmap(Sly_HSP90_CoChap_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Sly_HSP90_CoChap_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_HSP90_CoChap_Annotation, annotation_colors=color_HSP90_CoChap, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "HSP90_CoChap",
         annotation_legend = TRUE,
         #display_numbers = Sly_HSP90_CoChap_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(21,42)
)


#####Representing Heatmap for Cann

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Cann_HSP90_CoChap_Ann <- read.table("Funcional Groups/FigTree/Cann_HSP90_CoChap.txt",
                             sep = "\t", header = TRUE)

colnames(Cann_HSP90_CoChap_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Cann_HSP90_CoChap <- merge(Cann_HSP90_CoChap_Ann, Summary_Pepper_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Cann_HSP90_CoChap_TPM <- Cann_HSP90_CoChap %>% select(Protein, 4:17)
Cann_HSP90_CoChap_TPM <- Cann_HSP90_CoChap_TPM[!duplicated(Cann_HSP90_CoChap_TPM$Protein),]
Cann_HSP90_CoChap_TPM <- Cann_HSP90_CoChap_TPM[order(match(Cann_HSP90_CoChap_TPM[,1], Cann_HSP90_CoChap_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_HSP90_CoChap_FC <- Cann_HSP90_CoChap %>% select(Protein, 18:19)
Cann_HSP90_CoChap_FC <- Cann_HSP90_CoChap_FC[!duplicated(Cann_HSP90_CoChap_FC$Protein),]
Cann_HSP90_CoChap_FC <- Cann_HSP90_CoChap_FC[order(match(Cann_HSP90_CoChap_FC[,1], Cann_HSP90_CoChap_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_HSP90_CoChap_DEG <- Cann_HSP90_CoChap %>% select(Protein, 20:21)
Cann_HSP90_CoChap_DEG <- Cann_HSP90_CoChap_DEG[!duplicated(Cann_HSP90_CoChap_DEG$Protein),]
Cann_HSP90_CoChap_DEG <- Cann_HSP90_CoChap_DEG[order(match(Cann_HSP90_CoChap_DEG[,1], Cann_HSP90_CoChap_Ann[,2])),]

Cann_HSP90_CoChap_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Cann_HSP90_CoChap_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Cann_HSP90_CoChap_DEG$Ripening_Red_DEG <- ifelse(is.na(Cann_HSP90_CoChap_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
row.names(Cann_HSP90_CoChap_DEG) <- Cann_HSP90_CoChap_DEG$Protein
Cann_HSP90_CoChap_DEG[1] <- NULL

rownames(Cann_HSP90_CoChap_FC) <- Cann_HSP90_CoChap_FC$Protein
Cann_HSP90_CoChap_FC[1] <- NULL

min(Cann_HSP90_CoChap_FC, na.rm = TRUE)
max(Cann_HSP90_CoChap_FC, na.rm = TRUE)
breaks_Cann_HSP90_CoChap=c(seq(-4.5,0, length.out = 100), seq(0.01,6,length.out = 100))

pheatmap(Cann_HSP90_CoChap_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Cann_HSP90_CoChap,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_HSP90_CoChap_Annotation, annotation_colors=color_HSP90_CoChap, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Red"),
         cex = 1,
         angle_col=45,
         main = "HSP90_CoChap",
         annotation_legend = TRUE,
         display_numbers = Cann_HSP90_CoChap_DEG,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)


rownames(Cann_HSP90_CoChap_TPM) <- Cann_HSP90_CoChap_TPM$Protein
Cann_HSP90_CoChap_TPM[1] <- NULL

min(Cann_HSP90_CoChap_TPM, na.rm = TRUE)
max(Cann_HSP90_CoChap_TPM, na.rm = TRUE)
breaks_Cann_HSP90_CoChap_TPM=c(seq(0,100, length.out = 200))

Cann_HSP90_CoChap_TPM_norm <- normalize_df(Cann_HSP90_CoChap_TPM)

pheatmap(Cann_HSP90_CoChap_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Cann_HSP90_CoChap_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_HSP90_CoChap_Annotation, annotation_colors=color_HSP90_CoChap, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "HSP90_CoChap",
         annotation_legend = TRUE,
         #display_numbers = Cann_HSP90_CoChap_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(4,9)
)


#########################################################################################################################################
####################################  Phyllo  #############################################################################################
#########################################################################################################################################

Phyllo_Ath <- read.table("Funcional Groups/Ath_Phyllo.txt", header = TRUE)
Phyllo_Ath <- merge(Phyllo_Ath, Ath_Gene_Models, by.x="ID", by.y=1)
Phyllo_Ath_fasta <- Ath11_fasta[c(which(names(Ath11_fasta) %in% Phyllo_Ath$V2))]
Phyllo_Ath_2 <- Phyllo_Ath %>% select(V2, Name)
write.fasta(Phyllo_Ath_fasta, as.string=TRUE, names= names(Phyllo_Ath_fasta),
            file.out = "Funcional Groups/Ath_Phyllo.fasta")
rename.fasta(infile = "Funcional Groups/Ath_Phyllo.fasta", Phyllo_Ath_2, outfile = "Funcional Groups/Ath_Phyllo_2.fasta")

Phyllo_Niben_Candidates <- read.table("Funcional Groups/Niben_Candidates_Phyllo.txt")
Phyllo_Niben_Candidates <- Phyllo_Niben_Candidates[Phyllo_Niben_Candidates$V3>=35,] %>% select(V2)
Phyllo_Niben_Candidates <- unique(Phyllo_Niben_Candidates)
Phyllo_Niben_Candidates_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Phyllo_Niben_Candidates$V2))]
write.fasta(Phyllo_Niben_Candidates_fasta, as.string=TRUE, names= names(Phyllo_Niben_Candidates_fasta),
            file.out = "Funcional Groups/Phyllo_Niben_Candidates_35.fasta")

Phyllo_Sly_Candidates <- read.table("Funcional Groups/Sly_Candidates_Phyllo.txt")
Phyllo_Sly_Candidates <- Phyllo_Sly_Candidates[Phyllo_Sly_Candidates$V3>=35,] %>% select(V2)
Phyllo_Sly_Candidates <- unique(Phyllo_Sly_Candidates)
Phyllo_Sly_Candidates_fasta <- Tomato_4.0_fasta[c(which(names(Tomato_4.0_fasta) %in% Phyllo_Sly_Candidates$V2))]
write.fasta(Phyllo_Sly_Candidates_fasta, as.string=TRUE, names= names(Phyllo_Sly_Candidates_fasta),
            file.out = "Funcional Groups/Phyllo_Sly_Candidates_35.fasta")

Phyllo_Cann_Candidates <- read.table("Funcional Groups/Cann_Candidates_Phyllo.txt")
Phyllo_Cann_Candidates <- Phyllo_Cann_Candidates[Phyllo_Cann_Candidates$V3>=35,] %>% select(V2)
Phyllo_Cann_Candidates <- unique(Phyllo_Cann_Candidates)
Phyllo_Cann_Candidates_fasta <- Cannuum_ASM51225v2_fasta[c(which(names(Cannuum_ASM51225v2_fasta) %in% Phyllo_Cann_Candidates$V2))]
write.fasta(Phyllo_Cann_Candidates_fasta, as.string=TRUE, names= names(Phyllo_Cann_Candidates_fasta),
            file.out = "Funcional Groups/Phyllo_Cann_Candidates_35.fasta")

Niben_Phyllo_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_Phyllo_Ann$Name))]
write.fasta(Niben_Phyllo_fasta, as.string=TRUE, names= names(Niben_Phyllo_fasta),
            file.out = "Funcional Groups/Niben_Phyllo.fasta")
Niben_Phyllo_Ann_2 <- Niben_Phyllo_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_Phyllo.fasta", Niben_Phyllo_Ann, outfile = "Funcional Groups/Niben_Phyllo_2.fasta")

save_heatmaps("Phyllo", height = 6)

## Representing Heatmap

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Niben_Phyllo_Ann <- read.table("Funcional Groups/FigTree/Niben_Phyllo.txt",
                            sep = "\t", header = TRUE)

colnames(Niben_Phyllo_Ann) <- c("Name", "Protein")

Niben_Phyllo_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_Phyllo_Ann$Name))]
write.fasta(Niben_Phyllo_fasta, as.string=TRUE, names= names(Niben_Phyllo_fasta),
            file.out = "Funcional Groups/Niben_Phyllo.fasta")
Niben_Phyllo_Ann_2 <- Niben_Phyllo_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_Phyllo.fasta", Niben_Phyllo_Ann, outfile = "Funcional Groups/Niben_Phyllo_2.fasta")

Niben_Phyllo_Chloro <- read.delim("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan/Funcional Groups/Niben_Phyllo_2_summary.targetp2", 
                               header=FALSE, comment.char="#")
colnames(Niben_Phyllo_Chloro) <- c("ID",	"Prediction",	"noTP",	"SP",	"mTP",	"cTP",	"luTP",	"CS-Position")
Niben_Phyllo_Chloro$Prediction <- factor(Niben_Phyllo_Chloro$Prediction, levels = c("cTP","mTP","noTP"))
Niben_Phyllo_Chloro <- merge(Niben_Phyllo_Ann, Niben_Phyllo_Chloro, by.x = "Protein", by.y = "ID")

Niben_Phyllo_Chloro <- Niben_Phyllo_Chloro[order(match(Niben_Phyllo_Chloro$Protein, Niben_Phyllo_Ann$Protein)),]
rownames(Niben_Phyllo_Chloro) <- Niben_Phyllo_Chloro$Protein
Niben_Phyllo_Chloro <- Niben_Phyllo_Chloro %>% select(Prediction)


## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_Phyllo <- merge(Niben_Phyllo_Ann, TPM_All_FC_Annotated, by="Name")

## We create the annotation for Modules

Niben_Phyllo_Modules <- which_Module_pwd12(Niben_Phyllo_Ann)

## We merge this annotation of Modules with the annotation of Chloroplast TP prediction

Niben_Phyllo_Modules_Chl <- merge(Niben_Phyllo_Modules, Niben_Phyllo_Chloro, by = 0)
Niben_Phyllo_Modules_Chl <- Niben_Phyllo_Modules_Chl[order(match(Niben_Phyllo_Modules_Chl$Row.names, Niben_Phyllo_Ann$Protein)),]
rownames(Niben_Phyllo_Modules_Chl) <- Niben_Phyllo_Modules_Chl$Row.names
Niben_Phyllo_Modules_Chl$Row.names <- NULL

## We select only the FC data from the different comparisons
Niben_Phyllo_TPM <- Niben_Phyllo %>% select(Protein, 4:51, 61:66)
Niben_Phyllo_TPM <- Niben_Phyllo_TPM[!duplicated(Niben_Phyllo_TPM$Protein),]
Niben_Phyllo_TPM <- Niben_Phyllo_TPM[order(match(Niben_Phyllo_TPM[,1], Niben_Phyllo_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_Phyllo_FC <- Niben_Phyllo %>% select(Protein, 71, 73:87)
Niben_Phyllo_FC <- Niben_Phyllo_FC[!duplicated(Niben_Phyllo_FC$Protein),]
Niben_Phyllo_FC <- Niben_Phyllo_FC[order(match(Niben_Phyllo_FC[,1], Niben_Phyllo_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_Phyllo_DEG <- Niben_Phyllo %>% select(Protein, 89, 91:105)
Niben_Phyllo_DEG <- Niben_Phyllo_DEG[!duplicated(Niben_Phyllo_DEG$Protein),]
Niben_Phyllo_DEG <- Niben_Phyllo_DEG[order(match(Niben_Phyllo_DEG[,1], Niben_Phyllo_Ann[,2])),]

Niben_Phyllo_DEG$pcrtB22h_vs_GFP22h_filtered <- ifelse(is.na(Niben_Phyllo_DEG$pcrtB22h_vs_GFP22h_filtered) == TRUE, " ", "*")
Niben_Phyllo_DEG$pcrtB25h_vs_GFP25h_filtered <- ifelse(is.na(Niben_Phyllo_DEG$pcrtB25h_vs_GFP25h_filtered) == TRUE, " ", "*")
Niben_Phyllo_DEG$pcrtBAA_vs_GFPAA_filtered <- ifelse(is.na(Niben_Phyllo_DEG$pcrtBAA_vs_GFPAA_filtered) == TRUE, " ", "*")
Niben_Phyllo_DEG$pcrtBBB_vs_GFPBB_filtered <- ifelse(is.na(Niben_Phyllo_DEG$pcrtBBB_vs_GFPBB_filtered) == TRUE, " ", "*")
Niben_Phyllo_DEG$pcrtBYY_vs_GFPYY_filtered <- ifelse(is.na(Niben_Phyllo_DEG$pcrtBYY_vs_GFPYY_filtered) == TRUE, " ", "*")
Niben_Phyllo_DEG$pcrtB28h_vs_GFP28h_filtered <- ifelse(is.na(Niben_Phyllo_DEG$pcrtB28h_vs_GFP28h_filtered) == TRUE, " ", "*")
Niben_Phyllo_DEG$pcrtB34h_vs_GFP34h_filtered <- ifelse(is.na(Niben_Phyllo_DEG$pcrtB34h_vs_GFP34h_filtered) == TRUE, " ", "*")
Niben_Phyllo_DEG$pcrtB37h_vs_GFP37h_filtered <- ifelse(is.na(Niben_Phyllo_DEG$pcrtB37h_vs_GFP37h_filtered) == TRUE, " ", "*")
Niben_Phyllo_DEG$pcrtBZZ_vs_GFPZZ_filtered <- ifelse(is.na(Niben_Phyllo_DEG$pcrtBZZ_vs_GFPZZ_filtered) == TRUE, " ", "*")
Niben_Phyllo_DEG$pcrtBCC_vs_GFPCC_filtered <- ifelse(is.na(Niben_Phyllo_DEG$pcrtBCC_vs_GFPCC_filtered) == TRUE, " ", "*")
Niben_Phyllo_DEG$pcrtB40h_vs_GFP40h_filtered <- ifelse(is.na(Niben_Phyllo_DEG$pcrtB40h_vs_GFP40h_filtered) == TRUE, " ", "*")
Niben_Phyllo_DEG$pcrtBDD_vs_GFPDD_filtered <- ifelse(is.na(Niben_Phyllo_DEG$pcrtBDD_vs_GFPDD_filtered) == TRUE, " ", "*")
Niben_Phyllo_DEG$pcrtB46h_vs_GFP46h_filtered <- ifelse(is.na(Niben_Phyllo_DEG$pcrtB46h_vs_GFP46h_filtered) == TRUE, " ", "*")
Niben_Phyllo_DEG$pcrtB56h_vs_GFP56h_filtered <- ifelse(is.na(Niben_Phyllo_DEG$pcrtB56h_vs_GFP56h_filtered) == TRUE, " ", "*")
Niben_Phyllo_DEG$pcrtBEE_vs_GFPEE_filtered <- ifelse(is.na(Niben_Phyllo_DEG$pcrtBEE_vs_GFPEE_filtered) == TRUE, " ", "*")
Niben_Phyllo_DEG$pcrtB96h_vs_GFP96h_filtered <- ifelse(is.na(Niben_Phyllo_DEG$pcrtB96h_vs_GFP96h_filtered) == TRUE, " ", "*")
row.names(Niben_Phyllo_DEG) <- Niben_Phyllo_DEG$Protein
Niben_Phyllo_DEG[1] <- NULL

Niben_Phyllo_DEG <- Niben_Phyllo_DEG %>%
  select(pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

row.names(Niben_Phyllo_FC) <- Niben_Phyllo_FC$Protein
Niben_Phyllo_FC[1] <- NULL

Niben_Phyllo_FC <- Niben_Phyllo_FC %>%
  select(pcrtB22h_vs_GFP22h_All, pcrtBAA_vs_GFPAA_All, pcrtB25h_vs_GFP25h_All, 
         pcrtBYY_vs_GFPYY_All, pcrtBBB_vs_GFPBB_All, pcrtB28h_vs_GFP28h_All, pcrtB34h_vs_GFP34h_All, pcrtBZZ_vs_GFPZZ_All, 
         pcrtB37h_vs_GFP37h_All, pcrtBCC_vs_GFPCC_All, pcrtB40h_vs_GFP40h_All, pcrtBDD_vs_GFPDD_All, pcrtB46h_vs_GFP46h_All, 
         pcrtBEE_vs_GFPEE_All, pcrtB56h_vs_GFP56h_All, pcrtB96h_vs_GFP96h_All)


min(Niben_Phyllo_FC, na.rm = TRUE)
max(Niben_Phyllo_FC, na.rm = TRUE)
breaks_Phyllo_FC=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Niben_Phyllo_FC %>% select(1:15), 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Phyllo_FC,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_Phyllo_Annotation, annotation_colors=color_Phyllo, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h"),#, "96h"),
         cex = 1,
         angle_col=45,
         main = "Phyllo",
         annotation_legend = TRUE,
         display_numbers = Niben_Phyllo_DEG %>% select(1:15),
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(84, 104),
         annotation_row = Niben_Phyllo_Modules_Chl,
         annotation_colors = module_Chl_colors_2
         #gaps_col = c(15)
)


rownames(Niben_Phyllo_TPM) <- Niben_Phyllo_TPM$Protein
Niben_Phyllo_TPM[1] <- NULL

min(Niben_Phyllo_TPM, na.rm = TRUE)
max(Niben_Phyllo_TPM, na.rm = TRUE)
breaks_Niben_Phyllo_TPM=c(seq(0,100, length.out = 100))

Niben_Phyllo_TPM_norm <- normalize_df(Niben_Phyllo_TPM)
Niben_Phyllo_TPM_norm_Cornell <- Niben_Phyllo_TPM %>%
  select(1:48) 
Niben_Phyllo_TPM_norm_Cornell <- normalize_df(Niben_Phyllo_TPM_norm_Cornell)


pheatmap(Niben_Phyllo_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Niben_Phyllo_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_Phyllo_Annotation, annotation_colors=color_Phyllo, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Phyllo",
         annotation_legend = TRUE,
         #display_numbers = Niben_Phyllo_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         annotation_row = Niben_Phyllo_Modules,
         annotation_colors = module_colors,
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(24, 48, 51)
)



#####Representing Heatmap for Sly

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Sly_Phyllo_Ann <- read.table("Funcional Groups/FigTree/Sly_Phyllo.txt",
                          sep = "\t", header = TRUE)

colnames(Sly_Phyllo_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Sly_Phyllo <- merge(Sly_Phyllo_Ann, Summary_Yazdani_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Sly_Phyllo_TPM <- Sly_Phyllo %>% select(Protein, 4:66)
Sly_Phyllo_TPM <- Sly_Phyllo_TPM[!duplicated(Sly_Phyllo_TPM$Protein),]
Sly_Phyllo_TPM <- Sly_Phyllo_TPM[order(match(Sly_Phyllo_TPM[,1], Sly_Phyllo_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_Phyllo_FC <- Sly_Phyllo %>% select(Protein, 67:71)
Sly_Phyllo_FC <- Sly_Phyllo_FC[!duplicated(Sly_Phyllo_FC$Protein),]
Sly_Phyllo_FC <- Sly_Phyllo_FC[order(match(Sly_Phyllo_FC[,1], Sly_Phyllo_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_Phyllo_DEG <- Sly_Phyllo %>% select(Protein, 72:76)
Sly_Phyllo_DEG <- Sly_Phyllo_DEG[!duplicated(Sly_Phyllo_DEG$Protein),]
Sly_Phyllo_DEG <- Sly_Phyllo_DEG[order(match(Sly_Phyllo_DEG[,1], Sly_Phyllo_Ann[,2])),]

Sly_Phyllo_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Sly_Phyllo_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Sly_Phyllo_DEG$Ripening_Orange_DEG <- ifelse(is.na(Sly_Phyllo_DEG$Ripening_Orange_DEG) == TRUE, " ", "*")
Sly_Phyllo_DEG$Ripening_Red_DEG <- ifelse(is.na(Sly_Phyllo_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
Sly_Phyllo_DEG$ORHis_MG_DEG <- ifelse(is.na(Sly_Phyllo_DEG$ORHis_MG_DEG) == TRUE, " ", "*")
Sly_Phyllo_DEG$ORHis_Breaker_DEG <- ifelse(is.na(Sly_Phyllo_DEG$ORHis_Breaker_DEG) == TRUE, " ", "*")
row.names(Sly_Phyllo_DEG) <- Sly_Phyllo_DEG$Protein
Sly_Phyllo_DEG[1] <- NULL

rownames(Sly_Phyllo_FC) <- Sly_Phyllo_FC$Protein
Sly_Phyllo_FC[1] <- NULL

min(Sly_Phyllo_FC, na.rm = TRUE)
max(Sly_Phyllo_FC, na.rm = TRUE)
breaks_Sly_Phyllo=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Sly_Phyllo_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Sly_Phyllo,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_Phyllo_Annotation, annotation_colors=color_Phyllo, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Phyllo",
         annotation_legend = TRUE,
         display_numbers = Sly_Phyllo_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(39, 48),
         gaps_col = c(3)
)


rownames(Sly_Phyllo_TPM) <- Sly_Phyllo_TPM$Protein
Sly_Phyllo_TPM[1] <- NULL

min(Sly_Phyllo_TPM, na.rm = TRUE)
max(Sly_Phyllo_TPM, na.rm = TRUE)
breaks_Sly_Phyllo_TPM=c(seq(0,100, length.out = 200))

Sly_Phyllo_TPM_norm <- normalize_df(Sly_Phyllo_TPM)

pheatmap(Sly_Phyllo_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Sly_Phyllo_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_Phyllo_Annotation, annotation_colors=color_Phyllo, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Phyllo",
         annotation_legend = TRUE,
         #display_numbers = Sly_Phyllo_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(21,42)
)


#####Representing Heatmap for Cann

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Cann_Phyllo_Ann <- read.table("Funcional Groups/FigTree/Cann_Phyllo.txt",
                           sep = "\t", header = TRUE)

colnames(Cann_Phyllo_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Cann_Phyllo <- merge(Cann_Phyllo_Ann, Summary_Pepper_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Cann_Phyllo_TPM <- Cann_Phyllo %>% select(Protein, 4:17)
Cann_Phyllo_TPM <- Cann_Phyllo_TPM[!duplicated(Cann_Phyllo_TPM$Protein),]
Cann_Phyllo_TPM <- Cann_Phyllo_TPM[order(match(Cann_Phyllo_TPM[,1], Cann_Phyllo_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_Phyllo_FC <- Cann_Phyllo %>% select(Protein, 18:19)
Cann_Phyllo_FC <- Cann_Phyllo_FC[!duplicated(Cann_Phyllo_FC$Protein),]
Cann_Phyllo_FC <- Cann_Phyllo_FC[order(match(Cann_Phyllo_FC[,1], Cann_Phyllo_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_Phyllo_DEG <- Cann_Phyllo %>% select(Protein, 20:21)
Cann_Phyllo_DEG <- Cann_Phyllo_DEG[!duplicated(Cann_Phyllo_DEG$Protein),]
Cann_Phyllo_DEG <- Cann_Phyllo_DEG[order(match(Cann_Phyllo_DEG[,1], Cann_Phyllo_Ann[,2])),]

Cann_Phyllo_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Cann_Phyllo_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Cann_Phyllo_DEG$Ripening_Red_DEG <- ifelse(is.na(Cann_Phyllo_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
row.names(Cann_Phyllo_DEG) <- Cann_Phyllo_DEG$Protein
Cann_Phyllo_DEG[1] <- NULL

rownames(Cann_Phyllo_FC) <- Cann_Phyllo_FC$Protein
Cann_Phyllo_FC[1] <- NULL

min(Cann_Phyllo_FC, na.rm = TRUE)
max(Cann_Phyllo_FC, na.rm = TRUE)
breaks_Cann_Phyllo=c(seq(-4.5,0, length.out = 100), seq(0.01,6,length.out = 100))

pheatmap(Cann_Phyllo_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Cann_Phyllo,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_Phyllo_Annotation, annotation_colors=color_Phyllo, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Red"),
         cex = 1,
         angle_col=45,
         main = "Phyllo",
         annotation_legend = TRUE,
         display_numbers = Cann_Phyllo_DEG,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)


rownames(Cann_Phyllo_TPM) <- Cann_Phyllo_TPM$Protein
Cann_Phyllo_TPM[1] <- NULL

min(Cann_Phyllo_TPM, na.rm = TRUE)
max(Cann_Phyllo_TPM, na.rm = TRUE)
breaks_Cann_Phyllo_TPM=c(seq(0,100, length.out = 200))

Cann_Phyllo_TPM_norm <- normalize_df(Cann_Phyllo_TPM)

pheatmap(Cann_Phyllo_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Cann_Phyllo_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_Phyllo_Annotation, annotation_colors=color_Phyllo, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Phyllo",
         annotation_legend = TRUE,
         #display_numbers = Cann_Phyllo_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(4,9)
)


#########################################################################################################################################
####################################  Plasto  #############################################################################################
#########################################################################################################################################

Plasto_Ath <- read.table("Funcional Groups/Ath_Plasto.txt", header = TRUE)
Plasto_Ath <- merge(Plasto_Ath, Ath_Gene_Models, by.x="ID", by.y=1)
Plasto_Ath_fasta <- Ath11_fasta[c(which(names(Ath11_fasta) %in% Plasto_Ath$V2))]
Plasto_Ath_2 <- Plasto_Ath %>% select(V2, Name)
write.fasta(Plasto_Ath_fasta, as.string=TRUE, names= names(Plasto_Ath_fasta),
            file.out = "Funcional Groups/Ath_Plasto.fasta")
rename.fasta(infile = "Funcional Groups/Ath_Plasto.fasta", Plasto_Ath_2, outfile = "Funcional Groups/Ath_Plasto_2.fasta")

Plasto_Niben_Candidates <- read.table("Funcional Groups/Niben_Candidates_Plasto.txt")
Plasto_Niben_Candidates <- Plasto_Niben_Candidates[Plasto_Niben_Candidates$V3>=35,] %>% select(V2)
Plasto_Niben_Candidates <- unique(Plasto_Niben_Candidates)
Plasto_Niben_Candidates_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Plasto_Niben_Candidates$V2))]
write.fasta(Plasto_Niben_Candidates_fasta, as.string=TRUE, names= names(Plasto_Niben_Candidates_fasta),
            file.out = "Funcional Groups/Plasto_Niben_Candidates_35.fasta")

Plasto_Sly_Candidates <- read.table("Funcional Groups/Sly_Candidates_Plasto.txt")
Plasto_Sly_Candidates <- Plasto_Sly_Candidates[Plasto_Sly_Candidates$V3>=35,] %>% select(V2)
Plasto_Sly_Candidates <- unique(Plasto_Sly_Candidates)
Plasto_Sly_Candidates_fasta <- Tomato_4.0_fasta[c(which(names(Tomato_4.0_fasta) %in% Plasto_Sly_Candidates$V2))]
write.fasta(Plasto_Sly_Candidates_fasta, as.string=TRUE, names= names(Plasto_Sly_Candidates_fasta),
            file.out = "Funcional Groups/Plasto_Sly_Candidates_35.fasta")

Plasto_Cann_Candidates <- read.table("Funcional Groups/Cann_Candidates_Plasto.txt")
Plasto_Cann_Candidates <- Plasto_Cann_Candidates[Plasto_Cann_Candidates$V3>=35,] %>% select(V2)
Plasto_Cann_Candidates <- unique(Plasto_Cann_Candidates)
Plasto_Cann_Candidates_fasta <- Cannuum_ASM51225v2_fasta[c(which(names(Cannuum_ASM51225v2_fasta) %in% Plasto_Cann_Candidates$V2))]
write.fasta(Plasto_Cann_Candidates_fasta, as.string=TRUE, names= names(Plasto_Cann_Candidates_fasta),
            file.out = "Funcional Groups/Plasto_Cann_Candidates_35.fasta")

Niben_Plasto_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_Plasto_Ann$Name))]
write.fasta(Niben_Plasto_fasta, as.string=TRUE, names= names(Niben_Plasto_fasta),
            file.out = "Funcional Groups/Niben_Plasto.fasta")
Niben_Plasto_Ann_2 <- Niben_Plasto_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_Plasto.fasta", Niben_Plasto_Ann, outfile = "Funcional Groups/Niben_Plasto_2.fasta")

save_heatmaps("Plasto", height = 6)

## Representing Heatmap

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Niben_Plasto_Ann <- read.table("Funcional Groups/FigTree/Niben_Plasto.txt",
                               sep = "\t", header = TRUE)

colnames(Niben_Plasto_Ann) <- c("Name", "Protein")

Niben_Plasto_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_Plasto_Ann$Name))]
write.fasta(Niben_Plasto_fasta, as.string=TRUE, names= names(Niben_Plasto_fasta),
            file.out = "Funcional Groups/Niben_Plasto.fasta")
Niben_Plasto_Ann_2 <- Niben_Plasto_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_Plasto.fasta", Niben_Plasto_Ann, outfile = "Funcional Groups/Niben_Plasto_2.fasta")

Niben_Plasto_Chloro <- read.delim("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan/Funcional Groups/Niben_Plasto_2_summary.targetp2", 
                                  header=FALSE, comment.char="#")
colnames(Niben_Plasto_Chloro) <- c("ID",	"Prediction",	"noTP",	"SP",	"mTP",	"cTP",	"luTP",	"CS-Position")
Niben_Plasto_Chloro$Prediction <- factor(Niben_Plasto_Chloro$Prediction, levels = c("cTP","mTP","noTP"))
Niben_Plasto_Chloro <- merge(Niben_Plasto_Ann, Niben_Plasto_Chloro, by.x = "Protein", by.y = "ID")

Niben_Plasto_Chloro <- Niben_Plasto_Chloro[order(match(Niben_Plasto_Chloro$Protein, Niben_Plasto_Ann$Protein)),]
rownames(Niben_Plasto_Chloro) <- Niben_Plasto_Chloro$Protein
Niben_Plasto_Chloro <- Niben_Plasto_Chloro %>% select(Prediction)


## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_Plasto <- merge(Niben_Plasto_Ann, TPM_All_FC_Annotated, by="Name")

## We create the annotation for Modules

Niben_Plasto_Modules <- which_Module_pwd12(Niben_Plasto_Ann)

## We merge this annotation of Modules with the annotation of Chloroplast TP prediction

Niben_Plasto_Modules_Chl <- merge(Niben_Plasto_Modules, Niben_Plasto_Chloro, by = 0)
Niben_Plasto_Modules_Chl <- Niben_Plasto_Modules_Chl[order(match(Niben_Plasto_Modules_Chl$Row.names, Niben_Plasto_Ann$Protein)),]
rownames(Niben_Plasto_Modules_Chl) <- Niben_Plasto_Modules_Chl$Row.names
Niben_Plasto_Modules_Chl$Row.names <- NULL

## We select only the FC data from the different comparisons
Niben_Plasto_TPM <- Niben_Plasto %>% select(Protein, 4:51, 61:66)
Niben_Plasto_TPM <- Niben_Plasto_TPM[!duplicated(Niben_Plasto_TPM$Protein),]
Niben_Plasto_TPM <- Niben_Plasto_TPM[order(match(Niben_Plasto_TPM[,1], Niben_Plasto_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_Plasto_FC <- Niben_Plasto %>% select(Protein, 71, 73:87)
Niben_Plasto_FC <- Niben_Plasto_FC[!duplicated(Niben_Plasto_FC$Protein),]
Niben_Plasto_FC <- Niben_Plasto_FC[order(match(Niben_Plasto_FC[,1], Niben_Plasto_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_Plasto_DEG <- Niben_Plasto %>% select(Protein, 89, 91:105)
Niben_Plasto_DEG <- Niben_Plasto_DEG[!duplicated(Niben_Plasto_DEG$Protein),]
Niben_Plasto_DEG <- Niben_Plasto_DEG[order(match(Niben_Plasto_DEG[,1], Niben_Plasto_Ann[,2])),]

Niben_Plasto_DEG$pcrtB22h_vs_GFP22h_filtered <- ifelse(is.na(Niben_Plasto_DEG$pcrtB22h_vs_GFP22h_filtered) == TRUE, " ", "*")
Niben_Plasto_DEG$pcrtB25h_vs_GFP25h_filtered <- ifelse(is.na(Niben_Plasto_DEG$pcrtB25h_vs_GFP25h_filtered) == TRUE, " ", "*")
Niben_Plasto_DEG$pcrtBAA_vs_GFPAA_filtered <- ifelse(is.na(Niben_Plasto_DEG$pcrtBAA_vs_GFPAA_filtered) == TRUE, " ", "*")
Niben_Plasto_DEG$pcrtBBB_vs_GFPBB_filtered <- ifelse(is.na(Niben_Plasto_DEG$pcrtBBB_vs_GFPBB_filtered) == TRUE, " ", "*")
Niben_Plasto_DEG$pcrtBYY_vs_GFPYY_filtered <- ifelse(is.na(Niben_Plasto_DEG$pcrtBYY_vs_GFPYY_filtered) == TRUE, " ", "*")
Niben_Plasto_DEG$pcrtB28h_vs_GFP28h_filtered <- ifelse(is.na(Niben_Plasto_DEG$pcrtB28h_vs_GFP28h_filtered) == TRUE, " ", "*")
Niben_Plasto_DEG$pcrtB34h_vs_GFP34h_filtered <- ifelse(is.na(Niben_Plasto_DEG$pcrtB34h_vs_GFP34h_filtered) == TRUE, " ", "*")
Niben_Plasto_DEG$pcrtB37h_vs_GFP37h_filtered <- ifelse(is.na(Niben_Plasto_DEG$pcrtB37h_vs_GFP37h_filtered) == TRUE, " ", "*")
Niben_Plasto_DEG$pcrtBZZ_vs_GFPZZ_filtered <- ifelse(is.na(Niben_Plasto_DEG$pcrtBZZ_vs_GFPZZ_filtered) == TRUE, " ", "*")
Niben_Plasto_DEG$pcrtBCC_vs_GFPCC_filtered <- ifelse(is.na(Niben_Plasto_DEG$pcrtBCC_vs_GFPCC_filtered) == TRUE, " ", "*")
Niben_Plasto_DEG$pcrtB40h_vs_GFP40h_filtered <- ifelse(is.na(Niben_Plasto_DEG$pcrtB40h_vs_GFP40h_filtered) == TRUE, " ", "*")
Niben_Plasto_DEG$pcrtBDD_vs_GFPDD_filtered <- ifelse(is.na(Niben_Plasto_DEG$pcrtBDD_vs_GFPDD_filtered) == TRUE, " ", "*")
Niben_Plasto_DEG$pcrtB46h_vs_GFP46h_filtered <- ifelse(is.na(Niben_Plasto_DEG$pcrtB46h_vs_GFP46h_filtered) == TRUE, " ", "*")
Niben_Plasto_DEG$pcrtB56h_vs_GFP56h_filtered <- ifelse(is.na(Niben_Plasto_DEG$pcrtB56h_vs_GFP56h_filtered) == TRUE, " ", "*")
Niben_Plasto_DEG$pcrtBEE_vs_GFPEE_filtered <- ifelse(is.na(Niben_Plasto_DEG$pcrtBEE_vs_GFPEE_filtered) == TRUE, " ", "*")
Niben_Plasto_DEG$pcrtB96h_vs_GFP96h_filtered <- ifelse(is.na(Niben_Plasto_DEG$pcrtB96h_vs_GFP96h_filtered) == TRUE, " ", "*")
row.names(Niben_Plasto_DEG) <- Niben_Plasto_DEG$Protein
Niben_Plasto_DEG[1] <- NULL

Niben_Plasto_DEG <- Niben_Plasto_DEG %>%
  select(pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

row.names(Niben_Plasto_FC) <- Niben_Plasto_FC$Protein
Niben_Plasto_FC[1] <- NULL

Niben_Plasto_FC <- Niben_Plasto_FC %>%
  select(pcrtB22h_vs_GFP22h_All, pcrtBAA_vs_GFPAA_All, pcrtB25h_vs_GFP25h_All, 
         pcrtBYY_vs_GFPYY_All, pcrtBBB_vs_GFPBB_All, pcrtB28h_vs_GFP28h_All, pcrtB34h_vs_GFP34h_All, pcrtBZZ_vs_GFPZZ_All, 
         pcrtB37h_vs_GFP37h_All, pcrtBCC_vs_GFPCC_All, pcrtB40h_vs_GFP40h_All, pcrtBDD_vs_GFPDD_All, pcrtB46h_vs_GFP46h_All, 
         pcrtBEE_vs_GFPEE_All, pcrtB56h_vs_GFP56h_All, pcrtB96h_vs_GFP96h_All)


min(Niben_Plasto_FC, na.rm = TRUE)
max(Niben_Plasto_FC, na.rm = TRUE)
breaks_Plasto_FC=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Niben_Plasto_FC %>% select(1:15), 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Plasto_FC,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_Plasto_Annotation, annotation_colors=color_Plasto, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h"),#, "96h"),
         cex = 1,
         angle_col=45,
         main = "Plasto",
         annotation_legend = TRUE,
         display_numbers = Niben_Plasto_DEG %>% select(1:15),
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(84, 104),
         annotation_row = Niben_Plasto_Modules_Chl,
         annotation_colors = module_Chl_colors_2
         #gaps_col = c(15)
)


rownames(Niben_Plasto_TPM) <- Niben_Plasto_TPM$Protein
Niben_Plasto_TPM[1] <- NULL

min(Niben_Plasto_TPM, na.rm = TRUE)
max(Niben_Plasto_TPM, na.rm = TRUE)
breaks_Niben_Plasto_TPM=c(seq(0,100, length.out = 100))

Niben_Plasto_TPM_norm <- normalize_df(Niben_Plasto_TPM)
Niben_Plasto_TPM_norm_Cornell <- Niben_Plasto_TPM %>%
  select(1:48) 
Niben_Plasto_TPM_norm_Cornell <- normalize_df(Niben_Plasto_TPM_norm_Cornell)


pheatmap(Niben_Plasto_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Niben_Plasto_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_Plasto_Annotation, annotation_colors=color_Plasto, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Plasto",
         annotation_legend = TRUE,
         #display_numbers = Niben_Plasto_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         annotation_row = Niben_Plasto_Modules,
         annotation_colors = module_colors,
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(24, 48, 51)
)



#####Representing Heatmap for Sly

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Sly_Plasto_Ann <- read.table("Funcional Groups/FigTree/Sly_Plasto.txt",
                             sep = "\t", header = TRUE)

colnames(Sly_Plasto_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Sly_Plasto <- merge(Sly_Plasto_Ann, Summary_Yazdani_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Sly_Plasto_TPM <- Sly_Plasto %>% select(Protein, 4:66)
Sly_Plasto_TPM <- Sly_Plasto_TPM[!duplicated(Sly_Plasto_TPM$Protein),]
Sly_Plasto_TPM <- Sly_Plasto_TPM[order(match(Sly_Plasto_TPM[,1], Sly_Plasto_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_Plasto_FC <- Sly_Plasto %>% select(Protein, 67:71)
Sly_Plasto_FC <- Sly_Plasto_FC[!duplicated(Sly_Plasto_FC$Protein),]
Sly_Plasto_FC <- Sly_Plasto_FC[order(match(Sly_Plasto_FC[,1], Sly_Plasto_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_Plasto_DEG <- Sly_Plasto %>% select(Protein, 72:76)
Sly_Plasto_DEG <- Sly_Plasto_DEG[!duplicated(Sly_Plasto_DEG$Protein),]
Sly_Plasto_DEG <- Sly_Plasto_DEG[order(match(Sly_Plasto_DEG[,1], Sly_Plasto_Ann[,2])),]

Sly_Plasto_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Sly_Plasto_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Sly_Plasto_DEG$Ripening_Orange_DEG <- ifelse(is.na(Sly_Plasto_DEG$Ripening_Orange_DEG) == TRUE, " ", "*")
Sly_Plasto_DEG$Ripening_Red_DEG <- ifelse(is.na(Sly_Plasto_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
Sly_Plasto_DEG$ORHis_MG_DEG <- ifelse(is.na(Sly_Plasto_DEG$ORHis_MG_DEG) == TRUE, " ", "*")
Sly_Plasto_DEG$ORHis_Breaker_DEG <- ifelse(is.na(Sly_Plasto_DEG$ORHis_Breaker_DEG) == TRUE, " ", "*")
row.names(Sly_Plasto_DEG) <- Sly_Plasto_DEG$Protein
Sly_Plasto_DEG[1] <- NULL

rownames(Sly_Plasto_FC) <- Sly_Plasto_FC$Protein
Sly_Plasto_FC[1] <- NULL

min(Sly_Plasto_FC, na.rm = TRUE)
max(Sly_Plasto_FC, na.rm = TRUE)
breaks_Sly_Plasto=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Sly_Plasto_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Sly_Plasto,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_Plasto_Annotation, annotation_colors=color_Plasto, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Plasto",
         annotation_legend = TRUE,
         display_numbers = Sly_Plasto_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(39, 48),
         gaps_col = c(3)
)


rownames(Sly_Plasto_TPM) <- Sly_Plasto_TPM$Protein
Sly_Plasto_TPM[1] <- NULL

min(Sly_Plasto_TPM, na.rm = TRUE)
max(Sly_Plasto_TPM, na.rm = TRUE)
breaks_Sly_Plasto_TPM=c(seq(0,100, length.out = 200))

Sly_Plasto_TPM_norm <- normalize_df(Sly_Plasto_TPM)

pheatmap(Sly_Plasto_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Sly_Plasto_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_Plasto_Annotation, annotation_colors=color_Plasto, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Plasto",
         annotation_legend = TRUE,
         #display_numbers = Sly_Plasto_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(21,42)
)


#####Representing Heatmap for Cann

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Cann_Plasto_Ann <- read.table("Funcional Groups/FigTree/Cann_Plasto.txt",
                              sep = "\t", header = TRUE)

colnames(Cann_Plasto_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Cann_Plasto <- merge(Cann_Plasto_Ann, Summary_Pepper_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Cann_Plasto_TPM <- Cann_Plasto %>% select(Protein, 4:17)
Cann_Plasto_TPM <- Cann_Plasto_TPM[!duplicated(Cann_Plasto_TPM$Protein),]
Cann_Plasto_TPM <- Cann_Plasto_TPM[order(match(Cann_Plasto_TPM[,1], Cann_Plasto_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_Plasto_FC <- Cann_Plasto %>% select(Protein, 18:19)
Cann_Plasto_FC <- Cann_Plasto_FC[!duplicated(Cann_Plasto_FC$Protein),]
Cann_Plasto_FC <- Cann_Plasto_FC[order(match(Cann_Plasto_FC[,1], Cann_Plasto_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_Plasto_DEG <- Cann_Plasto %>% select(Protein, 20:21)
Cann_Plasto_DEG <- Cann_Plasto_DEG[!duplicated(Cann_Plasto_DEG$Protein),]
Cann_Plasto_DEG <- Cann_Plasto_DEG[order(match(Cann_Plasto_DEG[,1], Cann_Plasto_Ann[,2])),]

Cann_Plasto_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Cann_Plasto_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Cann_Plasto_DEG$Ripening_Red_DEG <- ifelse(is.na(Cann_Plasto_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
row.names(Cann_Plasto_DEG) <- Cann_Plasto_DEG$Protein
Cann_Plasto_DEG[1] <- NULL

rownames(Cann_Plasto_FC) <- Cann_Plasto_FC$Protein
Cann_Plasto_FC[1] <- NULL

min(Cann_Plasto_FC, na.rm = TRUE)
max(Cann_Plasto_FC, na.rm = TRUE)
breaks_Cann_Plasto=c(seq(-4.5,0, length.out = 100), seq(0.01,6,length.out = 100))

pheatmap(Cann_Plasto_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Cann_Plasto,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_Plasto_Annotation, annotation_colors=color_Plasto, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Red"),
         cex = 1,
         angle_col=45,
         main = "Plasto",
         annotation_legend = TRUE,
         display_numbers = Cann_Plasto_DEG,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)


rownames(Cann_Plasto_TPM) <- Cann_Plasto_TPM$Protein
Cann_Plasto_TPM[1] <- NULL

min(Cann_Plasto_TPM, na.rm = TRUE)
max(Cann_Plasto_TPM, na.rm = TRUE)
breaks_Cann_Plasto_TPM=c(seq(0,100, length.out = 200))

Cann_Plasto_TPM_norm <- normalize_df(Cann_Plasto_TPM)

pheatmap(Cann_Plasto_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Cann_Plasto_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_Plasto_Annotation, annotation_colors=color_Plasto, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Plasto",
         annotation_legend = TRUE,
         #display_numbers = Cann_Plasto_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(4,9)
)


#########################################################################################################################################
####################################  Light  #############################################################################################
#########################################################################################################################################

Light_Ath <- read.table("Funcional Groups/Ath_Light.txt", header = TRUE)
Light_Ath <- merge(Light_Ath, Ath_Gene_Models, by.x="ID", by.y=1)
Light_Ath_fasta <- Ath11_fasta[c(which(names(Ath11_fasta) %in% Light_Ath$V2))]
Light_Ath_2 <- Light_Ath %>% select(V2, Name)
write.fasta(Light_Ath_fasta, as.string=TRUE, names= names(Light_Ath_fasta),
            file.out = "Funcional Groups/Ath_Light.fasta")
rename.fasta(infile = "Funcional Groups/Ath_Light.fasta", Light_Ath_2, outfile = "Funcional Groups/Ath_Light_2.fasta")

Light_Niben_Candidates <- read.table("Funcional Groups/Niben_Candidates_Light.txt")
Light_Niben_Candidates <- Light_Niben_Candidates[Light_Niben_Candidates$V3>=35,] %>% select(V2)
Light_Niben_Candidates <- unique(Light_Niben_Candidates)
Light_Niben_Candidates_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Light_Niben_Candidates$V2))]
write.fasta(Light_Niben_Candidates_fasta, as.string=TRUE, names= names(Light_Niben_Candidates_fasta),
            file.out = "Funcional Groups/Light_Niben_Candidates_35.fasta")

Light_Sly_Candidates <- read.table("Funcional Groups/Sly_Candidates_Light.txt")
Light_Sly_Candidates <- Light_Sly_Candidates[Light_Sly_Candidates$V3>=35,] %>% select(V2)
Light_Sly_Candidates <- unique(Light_Sly_Candidates)
Light_Sly_Candidates_fasta <- Tomato_4.0_fasta[c(which(names(Tomato_4.0_fasta) %in% Light_Sly_Candidates$V2))]
write.fasta(Light_Sly_Candidates_fasta, as.string=TRUE, names= names(Light_Sly_Candidates_fasta),
            file.out = "Funcional Groups/Light_Sly_Candidates_35.fasta")

Light_Cann_Candidates <- read.table("Funcional Groups/Cann_Candidates_Light.txt")
Light_Cann_Candidates <- Light_Cann_Candidates[Light_Cann_Candidates$V3>=35,] %>% select(V2)
Light_Cann_Candidates <- unique(Light_Cann_Candidates)
Light_Cann_Candidates_fasta <- Cannuum_ASM51225v2_fasta[c(which(names(Cannuum_ASM51225v2_fasta) %in% Light_Cann_Candidates$V2))]
write.fasta(Light_Cann_Candidates_fasta, as.string=TRUE, names= names(Light_Cann_Candidates_fasta),
            file.out = "Funcional Groups/Light_Cann_Candidates_35.fasta")

Niben_Light_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_Light_Ann$Name))]
write.fasta(Niben_Light_fasta, as.string=TRUE, names= names(Niben_Light_fasta),
            file.out = "Funcional Groups/Niben_Light.fasta")
Niben_Light_Ann_2 <- Niben_Light_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_Light.fasta", Niben_Light_Ann, outfile = "Funcional Groups/Niben_Light_2.fasta")

save_heatmaps("Light", height = 6)

## Representing Heatmap

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Niben_Light_Ann <- read.table("Funcional Groups/FigTree/Niben_Light.txt",
                               sep = "\t", header = TRUE)

colnames(Niben_Light_Ann) <- c("Name", "Protein")

Niben_Light_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_Light_Ann$Name))]
write.fasta(Niben_Light_fasta, as.string=TRUE, names= names(Niben_Light_fasta),
            file.out = "Funcional Groups/Niben_Light.fasta")
Niben_Light_Ann_2 <- Niben_Light_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_Light.fasta", Niben_Light_Ann, outfile = "Funcional Groups/Niben_Light_2.fasta")

Niben_Light_Chloro <- read.delim("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan/Funcional Groups/Niben_Light_2_summary.targetp2", 
                                  header=FALSE, comment.char="#")
colnames(Niben_Light_Chloro) <- c("ID",	"Prediction",	"noTP",	"SP",	"mTP",	"cTP",	"luTP",	"CS-Position")
Niben_Light_Chloro$Prediction <- factor(Niben_Light_Chloro$Prediction, levels = c("cTP","mTP","noTP"))
Niben_Light_Chloro <- merge(Niben_Light_Ann, Niben_Light_Chloro, by.x = "Protein", by.y = "ID")

Niben_Light_Chloro <- Niben_Light_Chloro[order(match(Niben_Light_Chloro$Protein, Niben_Light_Ann$Protein)),]
rownames(Niben_Light_Chloro) <- Niben_Light_Chloro$Protein
Niben_Light_Chloro <- Niben_Light_Chloro %>% select(Prediction)


## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_Light <- merge(Niben_Light_Ann, TPM_All_FC_Annotated, by="Name")

## We create the annotation for Modules

Niben_Light_Modules <- which_Module_pwd12(Niben_Light_Ann)

## We merge this annotation of Modules with the annotation of Chloroplast TP prediction

Niben_Light_Modules_Chl <- merge(Niben_Light_Modules, Niben_Light_Chloro, by = 0)
Niben_Light_Modules_Chl <- Niben_Light_Modules_Chl[order(match(Niben_Light_Modules_Chl$Row.names, Niben_Light_Ann$Protein)),]
rownames(Niben_Light_Modules_Chl) <- Niben_Light_Modules_Chl$Row.names
Niben_Light_Modules_Chl$Row.names <- NULL

## We select only the FC data from the different comparisons
Niben_Light_TPM <- Niben_Light %>% select(Protein, 4:51, 61:66)
Niben_Light_TPM <- Niben_Light_TPM[!duplicated(Niben_Light_TPM$Protein),]
Niben_Light_TPM <- Niben_Light_TPM[order(match(Niben_Light_TPM[,1], Niben_Light_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_Light_FC <- Niben_Light %>% select(Protein, 71, 73:87)
Niben_Light_FC <- Niben_Light_FC[!duplicated(Niben_Light_FC$Protein),]
Niben_Light_FC <- Niben_Light_FC[order(match(Niben_Light_FC[,1], Niben_Light_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_Light_DEG <- Niben_Light %>% select(Protein, 89, 91:105)
Niben_Light_DEG <- Niben_Light_DEG[!duplicated(Niben_Light_DEG$Protein),]
Niben_Light_DEG <- Niben_Light_DEG[order(match(Niben_Light_DEG[,1], Niben_Light_Ann[,2])),]

Niben_Light_DEG$pcrtB22h_vs_GFP22h_filtered <- ifelse(is.na(Niben_Light_DEG$pcrtB22h_vs_GFP22h_filtered) == TRUE, " ", "*")
Niben_Light_DEG$pcrtB25h_vs_GFP25h_filtered <- ifelse(is.na(Niben_Light_DEG$pcrtB25h_vs_GFP25h_filtered) == TRUE, " ", "*")
Niben_Light_DEG$pcrtBAA_vs_GFPAA_filtered <- ifelse(is.na(Niben_Light_DEG$pcrtBAA_vs_GFPAA_filtered) == TRUE, " ", "*")
Niben_Light_DEG$pcrtBBB_vs_GFPBB_filtered <- ifelse(is.na(Niben_Light_DEG$pcrtBBB_vs_GFPBB_filtered) == TRUE, " ", "*")
Niben_Light_DEG$pcrtBYY_vs_GFPYY_filtered <- ifelse(is.na(Niben_Light_DEG$pcrtBYY_vs_GFPYY_filtered) == TRUE, " ", "*")
Niben_Light_DEG$pcrtB28h_vs_GFP28h_filtered <- ifelse(is.na(Niben_Light_DEG$pcrtB28h_vs_GFP28h_filtered) == TRUE, " ", "*")
Niben_Light_DEG$pcrtB34h_vs_GFP34h_filtered <- ifelse(is.na(Niben_Light_DEG$pcrtB34h_vs_GFP34h_filtered) == TRUE, " ", "*")
Niben_Light_DEG$pcrtB37h_vs_GFP37h_filtered <- ifelse(is.na(Niben_Light_DEG$pcrtB37h_vs_GFP37h_filtered) == TRUE, " ", "*")
Niben_Light_DEG$pcrtBZZ_vs_GFPZZ_filtered <- ifelse(is.na(Niben_Light_DEG$pcrtBZZ_vs_GFPZZ_filtered) == TRUE, " ", "*")
Niben_Light_DEG$pcrtBCC_vs_GFPCC_filtered <- ifelse(is.na(Niben_Light_DEG$pcrtBCC_vs_GFPCC_filtered) == TRUE, " ", "*")
Niben_Light_DEG$pcrtB40h_vs_GFP40h_filtered <- ifelse(is.na(Niben_Light_DEG$pcrtB40h_vs_GFP40h_filtered) == TRUE, " ", "*")
Niben_Light_DEG$pcrtBDD_vs_GFPDD_filtered <- ifelse(is.na(Niben_Light_DEG$pcrtBDD_vs_GFPDD_filtered) == TRUE, " ", "*")
Niben_Light_DEG$pcrtB46h_vs_GFP46h_filtered <- ifelse(is.na(Niben_Light_DEG$pcrtB46h_vs_GFP46h_filtered) == TRUE, " ", "*")
Niben_Light_DEG$pcrtB56h_vs_GFP56h_filtered <- ifelse(is.na(Niben_Light_DEG$pcrtB56h_vs_GFP56h_filtered) == TRUE, " ", "*")
Niben_Light_DEG$pcrtBEE_vs_GFPEE_filtered <- ifelse(is.na(Niben_Light_DEG$pcrtBEE_vs_GFPEE_filtered) == TRUE, " ", "*")
Niben_Light_DEG$pcrtB96h_vs_GFP96h_filtered <- ifelse(is.na(Niben_Light_DEG$pcrtB96h_vs_GFP96h_filtered) == TRUE, " ", "*")
row.names(Niben_Light_DEG) <- Niben_Light_DEG$Protein
Niben_Light_DEG[1] <- NULL

Niben_Light_DEG <- Niben_Light_DEG %>%
  select(pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

row.names(Niben_Light_FC) <- Niben_Light_FC$Protein
Niben_Light_FC[1] <- NULL

Niben_Light_FC <- Niben_Light_FC %>%
  select(pcrtB22h_vs_GFP22h_All, pcrtBAA_vs_GFPAA_All, pcrtB25h_vs_GFP25h_All, 
         pcrtBYY_vs_GFPYY_All, pcrtBBB_vs_GFPBB_All, pcrtB28h_vs_GFP28h_All, pcrtB34h_vs_GFP34h_All, pcrtBZZ_vs_GFPZZ_All, 
         pcrtB37h_vs_GFP37h_All, pcrtBCC_vs_GFPCC_All, pcrtB40h_vs_GFP40h_All, pcrtBDD_vs_GFPDD_All, pcrtB46h_vs_GFP46h_All, 
         pcrtBEE_vs_GFPEE_All, pcrtB56h_vs_GFP56h_All, pcrtB96h_vs_GFP96h_All)


min(Niben_Light_FC, na.rm = TRUE)
max(Niben_Light_FC, na.rm = TRUE)
breaks_Light_FC=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Niben_Light_FC %>% select(1:15), 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Light_FC,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_Light_Annotation, annotation_colors=color_Light, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h"),#, "96h"),
         cex = 1,
         angle_col=45,
         main = "Light",
         annotation_legend = TRUE,
         display_numbers = Niben_Light_DEG %>% select(1:15),
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(84, 104),
         annotation_row = Niben_Light_Modules_Chl,
         annotation_colors = module_Chl_colors_2
         #gaps_col = c(15)
)


rownames(Niben_Light_TPM) <- Niben_Light_TPM$Protein
Niben_Light_TPM[1] <- NULL

min(Niben_Light_TPM, na.rm = TRUE)
max(Niben_Light_TPM, na.rm = TRUE)
breaks_Niben_Light_TPM=c(seq(0,100, length.out = 100))

Niben_Light_TPM_norm <- normalize_df(Niben_Light_TPM)
Niben_Light_TPM_norm_Cornell <- Niben_Light_TPM %>%
  select(1:48) 
Niben_Light_TPM_norm_Cornell <- normalize_df(Niben_Light_TPM_norm_Cornell)


pheatmap(Niben_Light_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Niben_Light_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_Light_Annotation, annotation_colors=color_Light, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Light",
         annotation_legend = TRUE,
         #display_numbers = Niben_Light_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         annotation_row = Niben_Light_Modules,
         annotation_colors = module_colors,
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(24, 48, 51)
)



#####Representing Heatmap for Sly

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Sly_Light_Ann <- read.table("Funcional Groups/FigTree/Sly_Light.txt",
                             sep = "\t", header = TRUE)

colnames(Sly_Light_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Sly_Light <- merge(Sly_Light_Ann, Summary_Yazdani_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Sly_Light_TPM <- Sly_Light %>% select(Protein, 4:66)
Sly_Light_TPM <- Sly_Light_TPM[!duplicated(Sly_Light_TPM$Protein),]
Sly_Light_TPM <- Sly_Light_TPM[order(match(Sly_Light_TPM[,1], Sly_Light_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_Light_FC <- Sly_Light %>% select(Protein, 67:71)
Sly_Light_FC <- Sly_Light_FC[!duplicated(Sly_Light_FC$Protein),]
Sly_Light_FC <- Sly_Light_FC[order(match(Sly_Light_FC[,1], Sly_Light_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_Light_DEG <- Sly_Light %>% select(Protein, 72:76)
Sly_Light_DEG <- Sly_Light_DEG[!duplicated(Sly_Light_DEG$Protein),]
Sly_Light_DEG <- Sly_Light_DEG[order(match(Sly_Light_DEG[,1], Sly_Light_Ann[,2])),]

Sly_Light_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Sly_Light_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Sly_Light_DEG$Ripening_Orange_DEG <- ifelse(is.na(Sly_Light_DEG$Ripening_Orange_DEG) == TRUE, " ", "*")
Sly_Light_DEG$Ripening_Red_DEG <- ifelse(is.na(Sly_Light_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
Sly_Light_DEG$ORHis_MG_DEG <- ifelse(is.na(Sly_Light_DEG$ORHis_MG_DEG) == TRUE, " ", "*")
Sly_Light_DEG$ORHis_Breaker_DEG <- ifelse(is.na(Sly_Light_DEG$ORHis_Breaker_DEG) == TRUE, " ", "*")
row.names(Sly_Light_DEG) <- Sly_Light_DEG$Protein
Sly_Light_DEG[1] <- NULL

rownames(Sly_Light_FC) <- Sly_Light_FC$Protein
Sly_Light_FC[1] <- NULL

min(Sly_Light_FC, na.rm = TRUE)
max(Sly_Light_FC, na.rm = TRUE)
breaks_Sly_Light=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Sly_Light_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Sly_Light,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_Light_Annotation, annotation_colors=color_Light, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Light",
         annotation_legend = TRUE,
         display_numbers = Sly_Light_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(39, 48),
         gaps_col = c(3)
)


rownames(Sly_Light_TPM) <- Sly_Light_TPM$Protein
Sly_Light_TPM[1] <- NULL

min(Sly_Light_TPM, na.rm = TRUE)
max(Sly_Light_TPM, na.rm = TRUE)
breaks_Sly_Light_TPM=c(seq(0,100, length.out = 200))

Sly_Light_TPM_norm <- normalize_df(Sly_Light_TPM)

pheatmap(Sly_Light_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Sly_Light_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_Light_Annotation, annotation_colors=color_Light, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Light",
         annotation_legend = TRUE,
         #display_numbers = Sly_Light_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(21,42)
)


#####Representing Heatmap for Cann

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Cann_Light_Ann <- read.table("Funcional Groups/FigTree/Cann_Light.txt",
                              sep = "\t", header = TRUE)

colnames(Cann_Light_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Cann_Light <- merge(Cann_Light_Ann, Summary_Pepper_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Cann_Light_TPM <- Cann_Light %>% select(Protein, 4:17)
Cann_Light_TPM <- Cann_Light_TPM[!duplicated(Cann_Light_TPM$Protein),]
Cann_Light_TPM <- Cann_Light_TPM[order(match(Cann_Light_TPM[,1], Cann_Light_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_Light_FC <- Cann_Light %>% select(Protein, 18:19)
Cann_Light_FC <- Cann_Light_FC[!duplicated(Cann_Light_FC$Protein),]
Cann_Light_FC <- Cann_Light_FC[order(match(Cann_Light_FC[,1], Cann_Light_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_Light_DEG <- Cann_Light %>% select(Protein, 20:21)
Cann_Light_DEG <- Cann_Light_DEG[!duplicated(Cann_Light_DEG$Protein),]
Cann_Light_DEG <- Cann_Light_DEG[order(match(Cann_Light_DEG[,1], Cann_Light_Ann[,2])),]

Cann_Light_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Cann_Light_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Cann_Light_DEG$Ripening_Red_DEG <- ifelse(is.na(Cann_Light_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
row.names(Cann_Light_DEG) <- Cann_Light_DEG$Protein
Cann_Light_DEG[1] <- NULL

rownames(Cann_Light_FC) <- Cann_Light_FC$Protein
Cann_Light_FC[1] <- NULL

min(Cann_Light_FC, na.rm = TRUE)
max(Cann_Light_FC, na.rm = TRUE)
breaks_Cann_Light=c(seq(-4.5,0, length.out = 100), seq(0.01,6,length.out = 100))

pheatmap(Cann_Light_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Cann_Light,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_Light_Annotation, annotation_colors=color_Light, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Red"),
         cex = 1,
         angle_col=45,
         main = "Light",
         annotation_legend = TRUE,
         display_numbers = Cann_Light_DEG,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)


rownames(Cann_Light_TPM) <- Cann_Light_TPM$Protein
Cann_Light_TPM[1] <- NULL

min(Cann_Light_TPM, na.rm = TRUE)
max(Cann_Light_TPM, na.rm = TRUE)
breaks_Cann_Light_TPM=c(seq(0,100, length.out = 200))

Cann_Light_TPM_norm <- normalize_df(Cann_Light_TPM)

pheatmap(Cann_Light_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Cann_Light_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_Light_Annotation, annotation_colors=color_Light, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Light",
         annotation_legend = TRUE,
         #display_numbers = Cann_Light_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(4,9)
)


#########################################################################################################################################
####################################  CellWall  #############################################################################################
#########################################################################################################################################

CellWall_Ath <- read.table("Funcional Groups/Ath_CellWall.txt", header = TRUE)
CellWall_Ath <- merge(CellWall_Ath, Ath_Gene_Models, by.x="ID", by.y=1)
CellWall_Ath_fasta <- Ath11_fasta[c(which(names(Ath11_fasta) %in% CellWall_Ath$V2))]
CellWall_Ath_2 <- CellWall_Ath %>% select(V2, Name)
write.fasta(CellWall_Ath_fasta, as.string=TRUE, names= names(CellWall_Ath_fasta),
            file.out = "Funcional Groups/Ath_CellWall.fasta")
rename.fasta(infile = "Funcional Groups/Ath_CellWall.fasta", CellWall_Ath_2, outfile = "Funcional Groups/Ath_CellWall_2.fasta")

CellWall_Niben_Candidates <- read.table("Funcional Groups/Niben_Candidates_CellWall.txt")
CellWall_Niben_Candidates <- CellWall_Niben_Candidates[CellWall_Niben_Candidates$V3>=35,] %>% select(V2)
CellWall_Niben_Candidates <- unique(CellWall_Niben_Candidates)
CellWall_Niben_Candidates_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% CellWall_Niben_Candidates$V2))]
write.fasta(CellWall_Niben_Candidates_fasta, as.string=TRUE, names= names(CellWall_Niben_Candidates_fasta),
            file.out = "Funcional Groups/CellWall_Niben_Candidates_35.fasta")

CellWall_Sly_Candidates <- read.table("Funcional Groups/Sly_Candidates_CellWall.txt")
CellWall_Sly_Candidates <- CellWall_Sly_Candidates[CellWall_Sly_Candidates$V3>=35,] %>% select(V2)
CellWall_Sly_Candidates <- unique(CellWall_Sly_Candidates)
CellWall_Sly_Candidates_fasta <- Tomato_4.0_fasta[c(which(names(Tomato_4.0_fasta) %in% CellWall_Sly_Candidates$V2))]
write.fasta(CellWall_Sly_Candidates_fasta, as.string=TRUE, names= names(CellWall_Sly_Candidates_fasta),
            file.out = "Funcional Groups/CellWall_Sly_Candidates_35.fasta")

CellWall_Cann_Candidates <- read.table("Funcional Groups/Cann_Candidates_CellWall.txt")
CellWall_Cann_Candidates <- CellWall_Cann_Candidates[CellWall_Cann_Candidates$V3>=35,] %>% select(V2)
CellWall_Cann_Candidates <- unique(CellWall_Cann_Candidates)
CellWall_Cann_Candidates_fasta <- Cannuum_ASM51225v2_fasta[c(which(names(Cannuum_ASM51225v2_fasta) %in% CellWall_Cann_Candidates$V2))]
write.fasta(CellWall_Cann_Candidates_fasta, as.string=TRUE, names= names(CellWall_Cann_Candidates_fasta),
            file.out = "Funcional Groups/CellWall_Cann_Candidates_35.fasta")

Niben_CellWall_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_CellWall_Ann$Name))]
write.fasta(Niben_CellWall_fasta, as.string=TRUE, names= names(Niben_CellWall_fasta),
            file.out = "Funcional Groups/Niben_CellWall.fasta")
Niben_CellWall_Ann_2 <- Niben_CellWall_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_CellWall.fasta", Niben_CellWall_Ann, outfile = "Funcional Groups/Niben_CellWall_2.fasta")

save_heatmaps("CellWall", height = 6)

## Representing Heatmap

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Niben_CellWall_Ann <- read.table("Funcional Groups/FigTree/Niben_CellWall.txt",
                               sep = "\t", header = TRUE)

colnames(Niben_CellWall_Ann) <- c("Name", "Protein")

Niben_CellWall_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_CellWall_Ann$Name))]
write.fasta(Niben_CellWall_fasta, as.string=TRUE, names= names(Niben_CellWall_fasta),
            file.out = "Funcional Groups/Niben_CellWall.fasta")
Niben_CellWall_Ann_2 <- Niben_CellWall_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_CellWall.fasta", Niben_CellWall_Ann, outfile = "Funcional Groups/Niben_CellWall_2.fasta")

Niben_CellWall_Chloro <- read.delim("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan/Funcional Groups/Niben_CellWall_2_summary.targetp2", 
                                  header=FALSE, comment.char="#")
colnames(Niben_CellWall_Chloro) <- c("ID",	"Prediction",	"noTP",	"SP",	"mTP",	"cTP",	"luTP",	"CS-Position")
Niben_CellWall_Chloro$Prediction <- factor(Niben_CellWall_Chloro$Prediction, levels = c("cTP","mTP","noTP"))
Niben_CellWall_Chloro <- merge(Niben_CellWall_Ann, Niben_CellWall_Chloro, by.x = "Protein", by.y = "ID")

Niben_CellWall_Chloro <- Niben_CellWall_Chloro[order(match(Niben_CellWall_Chloro$Protein, Niben_CellWall_Ann$Protein)),]
rownames(Niben_CellWall_Chloro) <- Niben_CellWall_Chloro$Protein
Niben_CellWall_Chloro <- Niben_CellWall_Chloro %>% select(Prediction)


## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_CellWall <- merge(Niben_CellWall_Ann, TPM_All_FC_Annotated, by="Name")

## We create the annotation for Modules

Niben_CellWall_Modules <- which_Module_pwd12(Niben_CellWall_Ann)

## We merge this annotation of Modules with the annotation of Chloroplast TP prediction

Niben_CellWall_Modules_Chl <- merge(Niben_CellWall_Modules, Niben_CellWall_Chloro, by = 0)
Niben_CellWall_Modules_Chl <- Niben_CellWall_Modules_Chl[order(match(Niben_CellWall_Modules_Chl$Row.names, Niben_CellWall_Ann$Protein)),]
rownames(Niben_CellWall_Modules_Chl) <- Niben_CellWall_Modules_Chl$Row.names
Niben_CellWall_Modules_Chl$Row.names <- NULL

## We select only the FC data from the different comparisons
Niben_CellWall_TPM <- Niben_CellWall %>% select(Protein, 4:51, 61:66)
Niben_CellWall_TPM <- Niben_CellWall_TPM[!duplicated(Niben_CellWall_TPM$Protein),]
Niben_CellWall_TPM <- Niben_CellWall_TPM[order(match(Niben_CellWall_TPM[,1], Niben_CellWall_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_CellWall_FC <- Niben_CellWall %>% select(Protein, 71, 73:87)
Niben_CellWall_FC <- Niben_CellWall_FC[!duplicated(Niben_CellWall_FC$Protein),]
Niben_CellWall_FC <- Niben_CellWall_FC[order(match(Niben_CellWall_FC[,1], Niben_CellWall_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_CellWall_DEG <- Niben_CellWall %>% select(Protein, 89, 91:105)
Niben_CellWall_DEG <- Niben_CellWall_DEG[!duplicated(Niben_CellWall_DEG$Protein),]
Niben_CellWall_DEG <- Niben_CellWall_DEG[order(match(Niben_CellWall_DEG[,1], Niben_CellWall_Ann[,2])),]

Niben_CellWall_DEG$pcrtB22h_vs_GFP22h_filtered <- ifelse(is.na(Niben_CellWall_DEG$pcrtB22h_vs_GFP22h_filtered) == TRUE, " ", "*")
Niben_CellWall_DEG$pcrtB25h_vs_GFP25h_filtered <- ifelse(is.na(Niben_CellWall_DEG$pcrtB25h_vs_GFP25h_filtered) == TRUE, " ", "*")
Niben_CellWall_DEG$pcrtBAA_vs_GFPAA_filtered <- ifelse(is.na(Niben_CellWall_DEG$pcrtBAA_vs_GFPAA_filtered) == TRUE, " ", "*")
Niben_CellWall_DEG$pcrtBBB_vs_GFPBB_filtered <- ifelse(is.na(Niben_CellWall_DEG$pcrtBBB_vs_GFPBB_filtered) == TRUE, " ", "*")
Niben_CellWall_DEG$pcrtBYY_vs_GFPYY_filtered <- ifelse(is.na(Niben_CellWall_DEG$pcrtBYY_vs_GFPYY_filtered) == TRUE, " ", "*")
Niben_CellWall_DEG$pcrtB28h_vs_GFP28h_filtered <- ifelse(is.na(Niben_CellWall_DEG$pcrtB28h_vs_GFP28h_filtered) == TRUE, " ", "*")
Niben_CellWall_DEG$pcrtB34h_vs_GFP34h_filtered <- ifelse(is.na(Niben_CellWall_DEG$pcrtB34h_vs_GFP34h_filtered) == TRUE, " ", "*")
Niben_CellWall_DEG$pcrtB37h_vs_GFP37h_filtered <- ifelse(is.na(Niben_CellWall_DEG$pcrtB37h_vs_GFP37h_filtered) == TRUE, " ", "*")
Niben_CellWall_DEG$pcrtBZZ_vs_GFPZZ_filtered <- ifelse(is.na(Niben_CellWall_DEG$pcrtBZZ_vs_GFPZZ_filtered) == TRUE, " ", "*")
Niben_CellWall_DEG$pcrtBCC_vs_GFPCC_filtered <- ifelse(is.na(Niben_CellWall_DEG$pcrtBCC_vs_GFPCC_filtered) == TRUE, " ", "*")
Niben_CellWall_DEG$pcrtB40h_vs_GFP40h_filtered <- ifelse(is.na(Niben_CellWall_DEG$pcrtB40h_vs_GFP40h_filtered) == TRUE, " ", "*")
Niben_CellWall_DEG$pcrtBDD_vs_GFPDD_filtered <- ifelse(is.na(Niben_CellWall_DEG$pcrtBDD_vs_GFPDD_filtered) == TRUE, " ", "*")
Niben_CellWall_DEG$pcrtB46h_vs_GFP46h_filtered <- ifelse(is.na(Niben_CellWall_DEG$pcrtB46h_vs_GFP46h_filtered) == TRUE, " ", "*")
Niben_CellWall_DEG$pcrtB56h_vs_GFP56h_filtered <- ifelse(is.na(Niben_CellWall_DEG$pcrtB56h_vs_GFP56h_filtered) == TRUE, " ", "*")
Niben_CellWall_DEG$pcrtBEE_vs_GFPEE_filtered <- ifelse(is.na(Niben_CellWall_DEG$pcrtBEE_vs_GFPEE_filtered) == TRUE, " ", "*")
Niben_CellWall_DEG$pcrtB96h_vs_GFP96h_filtered <- ifelse(is.na(Niben_CellWall_DEG$pcrtB96h_vs_GFP96h_filtered) == TRUE, " ", "*")
row.names(Niben_CellWall_DEG) <- Niben_CellWall_DEG$Protein
Niben_CellWall_DEG[1] <- NULL

Niben_CellWall_DEG <- Niben_CellWall_DEG %>%
  select(pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

row.names(Niben_CellWall_FC) <- Niben_CellWall_FC$Protein
Niben_CellWall_FC[1] <- NULL

Niben_CellWall_FC <- Niben_CellWall_FC %>%
  select(pcrtB22h_vs_GFP22h_All, pcrtBAA_vs_GFPAA_All, pcrtB25h_vs_GFP25h_All, 
         pcrtBYY_vs_GFPYY_All, pcrtBBB_vs_GFPBB_All, pcrtB28h_vs_GFP28h_All, pcrtB34h_vs_GFP34h_All, pcrtBZZ_vs_GFPZZ_All, 
         pcrtB37h_vs_GFP37h_All, pcrtBCC_vs_GFPCC_All, pcrtB40h_vs_GFP40h_All, pcrtBDD_vs_GFPDD_All, pcrtB46h_vs_GFP46h_All, 
         pcrtBEE_vs_GFPEE_All, pcrtB56h_vs_GFP56h_All, pcrtB96h_vs_GFP96h_All)


min(Niben_CellWall_FC, na.rm = TRUE)
max(Niben_CellWall_FC, na.rm = TRUE)
breaks_CellWall_FC=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Niben_CellWall_FC %>% select(1:15), 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_CellWall_FC,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_CellWall_Annotation, annotation_colors=color_CellWall, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h"),#, "96h"),
         cex = 1,
         angle_col=45,
         main = "CellWall",
         annotation_legend = TRUE,
         display_numbers = Niben_CellWall_DEG %>% select(1:15),
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(84, 104),
         annotation_row = Niben_CellWall_Modules_Chl,
         annotation_colors = module_Chl_colors_2
         #gaps_col = c(15)
)


rownames(Niben_CellWall_TPM) <- Niben_CellWall_TPM$Protein
Niben_CellWall_TPM[1] <- NULL

min(Niben_CellWall_TPM, na.rm = TRUE)
max(Niben_CellWall_TPM, na.rm = TRUE)
breaks_Niben_CellWall_TPM=c(seq(0,100, length.out = 100))

Niben_CellWall_TPM_norm <- normalize_df(Niben_CellWall_TPM)
Niben_CellWall_TPM_norm_Cornell <- Niben_CellWall_TPM %>%
  select(1:48) 
Niben_CellWall_TPM_norm_Cornell <- normalize_df(Niben_CellWall_TPM_norm_Cornell)


pheatmap(Niben_CellWall_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Niben_CellWall_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_CellWall_Annotation, annotation_colors=color_CellWall, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "CellWall",
         annotation_legend = TRUE,
         #display_numbers = Niben_CellWall_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         annotation_row = Niben_CellWall_Modules,
         annotation_colors = module_colors,
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(24, 48, 51)
)



#####Representing Heatmap for Sly

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Sly_CellWall_Ann <- read.table("Funcional Groups/FigTree/Sly_CellWall.txt",
                             sep = "\t", header = TRUE)

colnames(Sly_CellWall_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Sly_CellWall <- merge(Sly_CellWall_Ann, Summary_Yazdani_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Sly_CellWall_TPM <- Sly_CellWall %>% select(Protein, 4:66)
Sly_CellWall_TPM <- Sly_CellWall_TPM[!duplicated(Sly_CellWall_TPM$Protein),]
Sly_CellWall_TPM <- Sly_CellWall_TPM[order(match(Sly_CellWall_TPM[,1], Sly_CellWall_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_CellWall_FC <- Sly_CellWall %>% select(Protein, 67:71)
Sly_CellWall_FC <- Sly_CellWall_FC[!duplicated(Sly_CellWall_FC$Protein),]
Sly_CellWall_FC <- Sly_CellWall_FC[order(match(Sly_CellWall_FC[,1], Sly_CellWall_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_CellWall_DEG <- Sly_CellWall %>% select(Protein, 72:76)
Sly_CellWall_DEG <- Sly_CellWall_DEG[!duplicated(Sly_CellWall_DEG$Protein),]
Sly_CellWall_DEG <- Sly_CellWall_DEG[order(match(Sly_CellWall_DEG[,1], Sly_CellWall_Ann[,2])),]

Sly_CellWall_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Sly_CellWall_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Sly_CellWall_DEG$Ripening_Orange_DEG <- ifelse(is.na(Sly_CellWall_DEG$Ripening_Orange_DEG) == TRUE, " ", "*")
Sly_CellWall_DEG$Ripening_Red_DEG <- ifelse(is.na(Sly_CellWall_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
Sly_CellWall_DEG$ORHis_MG_DEG <- ifelse(is.na(Sly_CellWall_DEG$ORHis_MG_DEG) == TRUE, " ", "*")
Sly_CellWall_DEG$ORHis_Breaker_DEG <- ifelse(is.na(Sly_CellWall_DEG$ORHis_Breaker_DEG) == TRUE, " ", "*")
row.names(Sly_CellWall_DEG) <- Sly_CellWall_DEG$Protein
Sly_CellWall_DEG[1] <- NULL

rownames(Sly_CellWall_FC) <- Sly_CellWall_FC$Protein
Sly_CellWall_FC[1] <- NULL

min(Sly_CellWall_FC, na.rm = TRUE)
max(Sly_CellWall_FC, na.rm = TRUE)
breaks_Sly_CellWall=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Sly_CellWall_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Sly_CellWall,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_CellWall_Annotation, annotation_colors=color_CellWall, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "CellWall",
         annotation_legend = TRUE,
         display_numbers = Sly_CellWall_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(39, 48),
         gaps_col = c(3)
)


rownames(Sly_CellWall_TPM) <- Sly_CellWall_TPM$Protein
Sly_CellWall_TPM[1] <- NULL

min(Sly_CellWall_TPM, na.rm = TRUE)
max(Sly_CellWall_TPM, na.rm = TRUE)
breaks_Sly_CellWall_TPM=c(seq(0,100, length.out = 200))

Sly_CellWall_TPM_norm <- normalize_df(Sly_CellWall_TPM)

pheatmap(Sly_CellWall_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Sly_CellWall_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_CellWall_Annotation, annotation_colors=color_CellWall, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "CellWall",
         annotation_legend = TRUE,
         #display_numbers = Sly_CellWall_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(21,42)
)


#####Representing Heatmap for Cann

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Cann_CellWall_Ann <- read.table("Funcional Groups/FigTree/Cann_CellWall.txt",
                              sep = "\t", header = TRUE)

colnames(Cann_CellWall_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Cann_CellWall <- merge(Cann_CellWall_Ann, Summary_Pepper_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Cann_CellWall_TPM <- Cann_CellWall %>% select(Protein, 4:17)
Cann_CellWall_TPM <- Cann_CellWall_TPM[!duplicated(Cann_CellWall_TPM$Protein),]
Cann_CellWall_TPM <- Cann_CellWall_TPM[order(match(Cann_CellWall_TPM[,1], Cann_CellWall_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_CellWall_FC <- Cann_CellWall %>% select(Protein, 18:19)
Cann_CellWall_FC <- Cann_CellWall_FC[!duplicated(Cann_CellWall_FC$Protein),]
Cann_CellWall_FC <- Cann_CellWall_FC[order(match(Cann_CellWall_FC[,1], Cann_CellWall_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_CellWall_DEG <- Cann_CellWall %>% select(Protein, 20:21)
Cann_CellWall_DEG <- Cann_CellWall_DEG[!duplicated(Cann_CellWall_DEG$Protein),]
Cann_CellWall_DEG <- Cann_CellWall_DEG[order(match(Cann_CellWall_DEG[,1], Cann_CellWall_Ann[,2])),]

Cann_CellWall_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Cann_CellWall_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Cann_CellWall_DEG$Ripening_Red_DEG <- ifelse(is.na(Cann_CellWall_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
row.names(Cann_CellWall_DEG) <- Cann_CellWall_DEG$Protein
Cann_CellWall_DEG[1] <- NULL

rownames(Cann_CellWall_FC) <- Cann_CellWall_FC$Protein
Cann_CellWall_FC[1] <- NULL

min(Cann_CellWall_FC, na.rm = TRUE)
max(Cann_CellWall_FC, na.rm = TRUE)
breaks_Cann_CellWall=c(seq(-4.5,0, length.out = 100), seq(0.01,6,length.out = 100))

pheatmap(Cann_CellWall_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Cann_CellWall,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_CellWall_Annotation, annotation_colors=color_CellWall, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Red"),
         cex = 1,
         angle_col=45,
         main = "CellWall",
         annotation_legend = TRUE,
         display_numbers = Cann_CellWall_DEG,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)


rownames(Cann_CellWall_TPM) <- Cann_CellWall_TPM$Protein
Cann_CellWall_TPM[1] <- NULL

min(Cann_CellWall_TPM, na.rm = TRUE)
max(Cann_CellWall_TPM, na.rm = TRUE)
breaks_Cann_CellWall_TPM=c(seq(0,100, length.out = 200))

Cann_CellWall_TPM_norm <- normalize_df(Cann_CellWall_TPM)

pheatmap(Cann_CellWall_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Cann_CellWall_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_CellWall_Annotation, annotation_colors=color_CellWall, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "CellWall",
         annotation_legend = TRUE,
         #display_numbers = Cann_CellWall_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(4,9)
)



#########################################################################################################################################
####################################  AbStress  #############################################################################################
#########################################################################################################################################

AbStress_Ath <- read.table("Funcional Groups/Ath_AbStress.txt", header = TRUE)
AbStress_Ath <- merge(AbStress_Ath, Ath_Gene_Models, by.x="ID", by.y=1)
AbStress_Ath_fasta <- Ath11_fasta[c(which(names(Ath11_fasta) %in% AbStress_Ath$V2))]
AbStress_Ath_2 <- AbStress_Ath %>% select(V2, Name)
write.fasta(AbStress_Ath_fasta, as.string=TRUE, names= names(AbStress_Ath_fasta),
            file.out = "Funcional Groups/Ath_AbStress.fasta")
rename.fasta(infile = "Funcional Groups/Ath_AbStress.fasta", AbStress_Ath_2, outfile = "Funcional Groups/Ath_AbStress_2.fasta")

AbStress_Niben_Candidates <- read.table("Funcional Groups/Niben_Candidates_AbStress.txt")
AbStress_Niben_Candidates <- AbStress_Niben_Candidates[AbStress_Niben_Candidates$V3>=35,] %>% select(V2)
AbStress_Niben_Candidates <- unique(AbStress_Niben_Candidates)
AbStress_Niben_Candidates_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% AbStress_Niben_Candidates$V2))]
write.fasta(AbStress_Niben_Candidates_fasta, as.string=TRUE, names= names(AbStress_Niben_Candidates_fasta),
            file.out = "Funcional Groups/AbStress_Niben_Candidates_35.fasta")

AbStress_Sly_Candidates <- read.table("Funcional Groups/Sly_Candidates_AbStress.txt")
AbStress_Sly_Candidates <- AbStress_Sly_Candidates[AbStress_Sly_Candidates$V3>=35,] %>% select(V2)
AbStress_Sly_Candidates <- unique(AbStress_Sly_Candidates)
AbStress_Sly_Candidates_fasta <- Tomato_4.0_fasta[c(which(names(Tomato_4.0_fasta) %in% AbStress_Sly_Candidates$V2))]
write.fasta(AbStress_Sly_Candidates_fasta, as.string=TRUE, names= names(AbStress_Sly_Candidates_fasta),
            file.out = "Funcional Groups/AbStress_Sly_Candidates_35.fasta")

AbStress_Cann_Candidates <- read.table("Funcional Groups/Cann_Candidates_AbStress.txt")
AbStress_Cann_Candidates <- AbStress_Cann_Candidates[AbStress_Cann_Candidates$V3>=35,] %>% select(V2)
AbStress_Cann_Candidates <- unique(AbStress_Cann_Candidates)
AbStress_Cann_Candidates_fasta <- Cannuum_ASM51225v2_fasta[c(which(names(Cannuum_ASM51225v2_fasta) %in% AbStress_Cann_Candidates$V2))]
write.fasta(AbStress_Cann_Candidates_fasta, as.string=TRUE, names= names(AbStress_Cann_Candidates_fasta),
            file.out = "Funcional Groups/AbStress_Cann_Candidates_35.fasta")

Niben_AbStress_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_AbStress_Ann$Name))]
write.fasta(Niben_AbStress_fasta, as.string=TRUE, names= names(Niben_AbStress_fasta),
            file.out = "Funcional Groups/Niben_AbStress.fasta")
Niben_AbStress_Ann_2 <- Niben_AbStress_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_AbStress.fasta", Niben_AbStress_Ann, outfile = "Funcional Groups/Niben_AbStress_2.fasta")

save_heatmaps("AbStress", height = 6)

## Representing Heatmap

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Niben_AbStress_Ann <- read.table("Funcional Groups/FigTree/Niben_AbStress.txt",
                               sep = "\t", header = TRUE)

colnames(Niben_AbStress_Ann) <- c("Name", "Protein")

Niben_AbStress_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_AbStress_Ann$Name))]
write.fasta(Niben_AbStress_fasta, as.string=TRUE, names= names(Niben_AbStress_fasta),
            file.out = "Funcional Groups/Niben_AbStress.fasta")
Niben_AbStress_Ann_2 <- Niben_AbStress_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_AbStress.fasta", Niben_AbStress_Ann, outfile = "Funcional Groups/Niben_AbStress_2.fasta")

Niben_AbStress_Chloro <- read.delim("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan/Funcional Groups/Niben_AbStress_2_summary.targetp2", 
                                  header=FALSE, comment.char="#")
colnames(Niben_AbStress_Chloro) <- c("ID",	"Prediction",	"noTP",	"SP",	"mTP",	"cTP",	"luTP",	"CS-Position")
Niben_AbStress_Chloro$Prediction <- factor(Niben_AbStress_Chloro$Prediction, levels = c("cTP","mTP","noTP"))
Niben_AbStress_Chloro <- merge(Niben_AbStress_Ann, Niben_AbStress_Chloro, by.x = "Protein", by.y = "ID")

Niben_AbStress_Chloro <- Niben_AbStress_Chloro[order(match(Niben_AbStress_Chloro$Protein, Niben_AbStress_Ann$Protein)),]
rownames(Niben_AbStress_Chloro) <- Niben_AbStress_Chloro$Protein
Niben_AbStress_Chloro <- Niben_AbStress_Chloro %>% select(Prediction)


## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_AbStress <- merge(Niben_AbStress_Ann, TPM_All_FC_Annotated, by="Name")

## We create the annotation for Modules

Niben_AbStress_Modules <- which_Module_pwd12(Niben_AbStress_Ann)

## We merge this annotation of Modules with the annotation of Chloroplast TP prediction

Niben_AbStress_Modules_Chl <- merge(Niben_AbStress_Modules, Niben_AbStress_Chloro, by = 0)
Niben_AbStress_Modules_Chl <- Niben_AbStress_Modules_Chl[order(match(Niben_AbStress_Modules_Chl$Row.names, Niben_AbStress_Ann$Protein)),]
rownames(Niben_AbStress_Modules_Chl) <- Niben_AbStress_Modules_Chl$Row.names
Niben_AbStress_Modules_Chl$Row.names <- NULL

## We select only the FC data from the different comparisons
Niben_AbStress_TPM <- Niben_AbStress %>% select(Protein, 4:51, 61:66)
Niben_AbStress_TPM <- Niben_AbStress_TPM[!duplicated(Niben_AbStress_TPM$Protein),]
Niben_AbStress_TPM <- Niben_AbStress_TPM[order(match(Niben_AbStress_TPM[,1], Niben_AbStress_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_AbStress_FC <- Niben_AbStress %>% select(Protein, 71, 73:87)
Niben_AbStress_FC <- Niben_AbStress_FC[!duplicated(Niben_AbStress_FC$Protein),]
Niben_AbStress_FC <- Niben_AbStress_FC[order(match(Niben_AbStress_FC[,1], Niben_AbStress_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_AbStress_DEG <- Niben_AbStress %>% select(Protein, 89, 91:105)
Niben_AbStress_DEG <- Niben_AbStress_DEG[!duplicated(Niben_AbStress_DEG$Protein),]
Niben_AbStress_DEG <- Niben_AbStress_DEG[order(match(Niben_AbStress_DEG[,1], Niben_AbStress_Ann[,2])),]

Niben_AbStress_DEG$pcrtB22h_vs_GFP22h_filtered <- ifelse(is.na(Niben_AbStress_DEG$pcrtB22h_vs_GFP22h_filtered) == TRUE, " ", "*")
Niben_AbStress_DEG$pcrtB25h_vs_GFP25h_filtered <- ifelse(is.na(Niben_AbStress_DEG$pcrtB25h_vs_GFP25h_filtered) == TRUE, " ", "*")
Niben_AbStress_DEG$pcrtBAA_vs_GFPAA_filtered <- ifelse(is.na(Niben_AbStress_DEG$pcrtBAA_vs_GFPAA_filtered) == TRUE, " ", "*")
Niben_AbStress_DEG$pcrtBBB_vs_GFPBB_filtered <- ifelse(is.na(Niben_AbStress_DEG$pcrtBBB_vs_GFPBB_filtered) == TRUE, " ", "*")
Niben_AbStress_DEG$pcrtBYY_vs_GFPYY_filtered <- ifelse(is.na(Niben_AbStress_DEG$pcrtBYY_vs_GFPYY_filtered) == TRUE, " ", "*")
Niben_AbStress_DEG$pcrtB28h_vs_GFP28h_filtered <- ifelse(is.na(Niben_AbStress_DEG$pcrtB28h_vs_GFP28h_filtered) == TRUE, " ", "*")
Niben_AbStress_DEG$pcrtB34h_vs_GFP34h_filtered <- ifelse(is.na(Niben_AbStress_DEG$pcrtB34h_vs_GFP34h_filtered) == TRUE, " ", "*")
Niben_AbStress_DEG$pcrtB37h_vs_GFP37h_filtered <- ifelse(is.na(Niben_AbStress_DEG$pcrtB37h_vs_GFP37h_filtered) == TRUE, " ", "*")
Niben_AbStress_DEG$pcrtBZZ_vs_GFPZZ_filtered <- ifelse(is.na(Niben_AbStress_DEG$pcrtBZZ_vs_GFPZZ_filtered) == TRUE, " ", "*")
Niben_AbStress_DEG$pcrtBCC_vs_GFPCC_filtered <- ifelse(is.na(Niben_AbStress_DEG$pcrtBCC_vs_GFPCC_filtered) == TRUE, " ", "*")
Niben_AbStress_DEG$pcrtB40h_vs_GFP40h_filtered <- ifelse(is.na(Niben_AbStress_DEG$pcrtB40h_vs_GFP40h_filtered) == TRUE, " ", "*")
Niben_AbStress_DEG$pcrtBDD_vs_GFPDD_filtered <- ifelse(is.na(Niben_AbStress_DEG$pcrtBDD_vs_GFPDD_filtered) == TRUE, " ", "*")
Niben_AbStress_DEG$pcrtB46h_vs_GFP46h_filtered <- ifelse(is.na(Niben_AbStress_DEG$pcrtB46h_vs_GFP46h_filtered) == TRUE, " ", "*")
Niben_AbStress_DEG$pcrtB56h_vs_GFP56h_filtered <- ifelse(is.na(Niben_AbStress_DEG$pcrtB56h_vs_GFP56h_filtered) == TRUE, " ", "*")
Niben_AbStress_DEG$pcrtBEE_vs_GFPEE_filtered <- ifelse(is.na(Niben_AbStress_DEG$pcrtBEE_vs_GFPEE_filtered) == TRUE, " ", "*")
Niben_AbStress_DEG$pcrtB96h_vs_GFP96h_filtered <- ifelse(is.na(Niben_AbStress_DEG$pcrtB96h_vs_GFP96h_filtered) == TRUE, " ", "*")
row.names(Niben_AbStress_DEG) <- Niben_AbStress_DEG$Protein
Niben_AbStress_DEG[1] <- NULL

Niben_AbStress_DEG <- Niben_AbStress_DEG %>%
  select(pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

row.names(Niben_AbStress_FC) <- Niben_AbStress_FC$Protein
Niben_AbStress_FC[1] <- NULL

Niben_AbStress_FC <- Niben_AbStress_FC %>%
  select(pcrtB22h_vs_GFP22h_All, pcrtBAA_vs_GFPAA_All, pcrtB25h_vs_GFP25h_All, 
         pcrtBYY_vs_GFPYY_All, pcrtBBB_vs_GFPBB_All, pcrtB28h_vs_GFP28h_All, pcrtB34h_vs_GFP34h_All, pcrtBZZ_vs_GFPZZ_All, 
         pcrtB37h_vs_GFP37h_All, pcrtBCC_vs_GFPCC_All, pcrtB40h_vs_GFP40h_All, pcrtBDD_vs_GFPDD_All, pcrtB46h_vs_GFP46h_All, 
         pcrtBEE_vs_GFPEE_All, pcrtB56h_vs_GFP56h_All, pcrtB96h_vs_GFP96h_All)


min(Niben_AbStress_FC, na.rm = TRUE)
max(Niben_AbStress_FC, na.rm = TRUE)
breaks_AbStress_FC=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Niben_AbStress_FC %>% select(1:15), 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_AbStress_FC,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_AbStress_Annotation, annotation_colors=color_AbStress, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h"),#, "96h"),
         cex = 1,
         angle_col=45,
         main = "AbStress",
         annotation_legend = TRUE,
         display_numbers = Niben_AbStress_DEG %>% select(1:15),
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(84, 104),
         annotation_row = Niben_AbStress_Modules_Chl,
         annotation_colors = module_Chl_colors_2
         #gaps_col = c(15)
)


rownames(Niben_AbStress_TPM) <- Niben_AbStress_TPM$Protein
Niben_AbStress_TPM[1] <- NULL

min(Niben_AbStress_TPM, na.rm = TRUE)
max(Niben_AbStress_TPM, na.rm = TRUE)
breaks_Niben_AbStress_TPM=c(seq(0,100, length.out = 100))

Niben_AbStress_TPM_norm <- normalize_df(Niben_AbStress_TPM)
Niben_AbStress_TPM_norm_Cornell <- Niben_AbStress_TPM %>%
  select(1:48) 
Niben_AbStress_TPM_norm_Cornell <- normalize_df(Niben_AbStress_TPM_norm_Cornell)


pheatmap(Niben_AbStress_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Niben_AbStress_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_AbStress_Annotation, annotation_colors=color_AbStress, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "AbStress",
         annotation_legend = TRUE,
         #display_numbers = Niben_AbStress_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         annotation_row = Niben_AbStress_Modules,
         annotation_colors = module_colors,
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(24, 48, 51)
)



#####Representing Heatmap for Sly

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Sly_AbStress_Ann <- read.table("Funcional Groups/FigTree/Sly_AbStress.txt",
                             sep = "\t", header = TRUE)

colnames(Sly_AbStress_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Sly_AbStress <- merge(Sly_AbStress_Ann, Summary_Yazdani_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Sly_AbStress_TPM <- Sly_AbStress %>% select(Protein, 4:66)
Sly_AbStress_TPM <- Sly_AbStress_TPM[!duplicated(Sly_AbStress_TPM$Protein),]
Sly_AbStress_TPM <- Sly_AbStress_TPM[order(match(Sly_AbStress_TPM[,1], Sly_AbStress_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_AbStress_FC <- Sly_AbStress %>% select(Protein, 67:71)
Sly_AbStress_FC <- Sly_AbStress_FC[!duplicated(Sly_AbStress_FC$Protein),]
Sly_AbStress_FC <- Sly_AbStress_FC[order(match(Sly_AbStress_FC[,1], Sly_AbStress_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_AbStress_DEG <- Sly_AbStress %>% select(Protein, 72:76)
Sly_AbStress_DEG <- Sly_AbStress_DEG[!duplicated(Sly_AbStress_DEG$Protein),]
Sly_AbStress_DEG <- Sly_AbStress_DEG[order(match(Sly_AbStress_DEG[,1], Sly_AbStress_Ann[,2])),]

Sly_AbStress_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Sly_AbStress_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Sly_AbStress_DEG$Ripening_Orange_DEG <- ifelse(is.na(Sly_AbStress_DEG$Ripening_Orange_DEG) == TRUE, " ", "*")
Sly_AbStress_DEG$Ripening_Red_DEG <- ifelse(is.na(Sly_AbStress_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
Sly_AbStress_DEG$ORHis_MG_DEG <- ifelse(is.na(Sly_AbStress_DEG$ORHis_MG_DEG) == TRUE, " ", "*")
Sly_AbStress_DEG$ORHis_Breaker_DEG <- ifelse(is.na(Sly_AbStress_DEG$ORHis_Breaker_DEG) == TRUE, " ", "*")
row.names(Sly_AbStress_DEG) <- Sly_AbStress_DEG$Protein
Sly_AbStress_DEG[1] <- NULL

rownames(Sly_AbStress_FC) <- Sly_AbStress_FC$Protein
Sly_AbStress_FC[1] <- NULL

min(Sly_AbStress_FC, na.rm = TRUE)
max(Sly_AbStress_FC, na.rm = TRUE)
breaks_Sly_AbStress=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Sly_AbStress_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Sly_AbStress,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_AbStress_Annotation, annotation_colors=color_AbStress, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "AbStress",
         annotation_legend = TRUE,
         display_numbers = Sly_AbStress_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(39, 48),
         gaps_col = c(3)
)


rownames(Sly_AbStress_TPM) <- Sly_AbStress_TPM$Protein
Sly_AbStress_TPM[1] <- NULL

min(Sly_AbStress_TPM, na.rm = TRUE)
max(Sly_AbStress_TPM, na.rm = TRUE)
breaks_Sly_AbStress_TPM=c(seq(0,100, length.out = 200))

Sly_AbStress_TPM_norm <- normalize_df(Sly_AbStress_TPM)

pheatmap(Sly_AbStress_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Sly_AbStress_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_AbStress_Annotation, annotation_colors=color_AbStress, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "AbStress",
         annotation_legend = TRUE,
         #display_numbers = Sly_AbStress_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(21,42)
)


#####Representing Heatmap for Cann

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Cann_AbStress_Ann <- read.table("Funcional Groups/FigTree/Cann_AbStress.txt",
                              sep = "\t", header = TRUE)

colnames(Cann_AbStress_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Cann_AbStress <- merge(Cann_AbStress_Ann, Summary_Pepper_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Cann_AbStress_TPM <- Cann_AbStress %>% select(Protein, 4:17)
Cann_AbStress_TPM <- Cann_AbStress_TPM[!duplicated(Cann_AbStress_TPM$Protein),]
Cann_AbStress_TPM <- Cann_AbStress_TPM[order(match(Cann_AbStress_TPM[,1], Cann_AbStress_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_AbStress_FC <- Cann_AbStress %>% select(Protein, 18:19)
Cann_AbStress_FC <- Cann_AbStress_FC[!duplicated(Cann_AbStress_FC$Protein),]
Cann_AbStress_FC <- Cann_AbStress_FC[order(match(Cann_AbStress_FC[,1], Cann_AbStress_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_AbStress_DEG <- Cann_AbStress %>% select(Protein, 20:21)
Cann_AbStress_DEG <- Cann_AbStress_DEG[!duplicated(Cann_AbStress_DEG$Protein),]
Cann_AbStress_DEG <- Cann_AbStress_DEG[order(match(Cann_AbStress_DEG[,1], Cann_AbStress_Ann[,2])),]

Cann_AbStress_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Cann_AbStress_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Cann_AbStress_DEG$Ripening_Red_DEG <- ifelse(is.na(Cann_AbStress_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
row.names(Cann_AbStress_DEG) <- Cann_AbStress_DEG$Protein
Cann_AbStress_DEG[1] <- NULL

rownames(Cann_AbStress_FC) <- Cann_AbStress_FC$Protein
Cann_AbStress_FC[1] <- NULL

min(Cann_AbStress_FC, na.rm = TRUE)
max(Cann_AbStress_FC, na.rm = TRUE)
breaks_Cann_AbStress=c(seq(-4.5,0, length.out = 100), seq(0.01,6,length.out = 100))

pheatmap(Cann_AbStress_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Cann_AbStress,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_AbStress_Annotation, annotation_colors=color_AbStress, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Red"),
         cex = 1,
         angle_col=45,
         main = "AbStress",
         annotation_legend = TRUE,
         display_numbers = Cann_AbStress_DEG,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)


rownames(Cann_AbStress_TPM) <- Cann_AbStress_TPM$Protein
Cann_AbStress_TPM[1] <- NULL

min(Cann_AbStress_TPM, na.rm = TRUE)
max(Cann_AbStress_TPM, na.rm = TRUE)
breaks_Cann_AbStress_TPM=c(seq(0,100, length.out = 200))

Cann_AbStress_TPM_norm <- normalize_df(Cann_AbStress_TPM)

pheatmap(Cann_AbStress_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Cann_AbStress_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_AbStress_Annotation, annotation_colors=color_AbStress, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "AbStress",
         annotation_legend = TRUE,
         #display_numbers = Cann_AbStress_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(4,9)
)


#########################################################################################################################################
####################################  Stromules  #############################################################################################
#########################################################################################################################################

Stromules_Ath <- read.table("Funcional Groups/Ath_Stromules.txt", header = TRUE)
Stromules_Ath <- merge(Stromules_Ath, Ath_Gene_Models, by.x="ID", by.y=1)
Stromules_Ath_fasta <- Ath11_fasta[c(which(names(Ath11_fasta) %in% Stromules_Ath$V2))]
Stromules_Ath_2 <- Stromules_Ath %>% select(V2, Name)
write.fasta(Stromules_Ath_fasta, as.string=TRUE, names= names(Stromules_Ath_fasta),
            file.out = "Funcional Groups/Ath_Stromules.fasta")
rename.fasta(infile = "Funcional Groups/Ath_Stromules.fasta", Stromules_Ath_2, outfile = "Funcional Groups/Ath_Stromules_2.fasta")

Stromules_Niben_Candidates <- read.table("Funcional Groups/Niben_Candidates_Stromules.txt")
Stromules_Niben_Candidates <- Stromules_Niben_Candidates[Stromules_Niben_Candidates$V3>=35,] %>% select(V2)
Stromules_Niben_Candidates <- unique(Stromules_Niben_Candidates)
Stromules_Niben_Candidates_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Stromules_Niben_Candidates$V2))]
write.fasta(Stromules_Niben_Candidates_fasta, as.string=TRUE, names= names(Stromules_Niben_Candidates_fasta),
            file.out = "Funcional Groups/Stromules_Niben_Candidates_35.fasta")

Stromules_Sly_Candidates <- read.table("Funcional Groups/Sly_Candidates_Stromules.txt")
Stromules_Sly_Candidates <- Stromules_Sly_Candidates[Stromules_Sly_Candidates$V3>=35,] %>% select(V2)
Stromules_Sly_Candidates <- unique(Stromules_Sly_Candidates)
Stromules_Sly_Candidates_fasta <- Tomato_4.0_fasta[c(which(names(Tomato_4.0_fasta) %in% Stromules_Sly_Candidates$V2))]
write.fasta(Stromules_Sly_Candidates_fasta, as.string=TRUE, names= names(Stromules_Sly_Candidates_fasta),
            file.out = "Funcional Groups/Stromules_Sly_Candidates_35.fasta")

Stromules_Cann_Candidates <- read.table("Funcional Groups/Cann_Candidates_Stromules.txt")
Stromules_Cann_Candidates <- Stromules_Cann_Candidates[Stromules_Cann_Candidates$V3>=35,] %>% select(V2)
Stromules_Cann_Candidates <- unique(Stromules_Cann_Candidates)
Stromules_Cann_Candidates_fasta <- Cannuum_ASM51225v2_fasta[c(which(names(Cannuum_ASM51225v2_fasta) %in% Stromules_Cann_Candidates$V2))]
write.fasta(Stromules_Cann_Candidates_fasta, as.string=TRUE, names= names(Stromules_Cann_Candidates_fasta),
            file.out = "Funcional Groups/Stromules_Cann_Candidates_35.fasta")

Niben_Stromules_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_Stromules_Ann$Name))]
write.fasta(Niben_Stromules_fasta, as.string=TRUE, names= names(Niben_Stromules_fasta),
            file.out = "Funcional Groups/Niben_Stromules.fasta")
Niben_Stromules_Ann_2 <- Niben_Stromules_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_Stromules.fasta", Niben_Stromules_Ann, outfile = "Funcional Groups/Niben_Stromules_2.fasta")

save_heatmaps("Stromules", height = 6)

## Representing Heatmap

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Niben_Stromules_Ann <- read.table("Funcional Groups/FigTree/Niben_Stromules.txt",
                               sep = "\t", header = TRUE)

colnames(Niben_Stromules_Ann) <- c("Name", "Protein")

Niben_Stromules_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_Stromules_Ann$Name))]
write.fasta(Niben_Stromules_fasta, as.string=TRUE, names= names(Niben_Stromules_fasta),
            file.out = "Funcional Groups/Niben_Stromules.fasta")
Niben_Stromules_Ann_2 <- Niben_Stromules_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_Stromules.fasta", Niben_Stromules_Ann, outfile = "Funcional Groups/Niben_Stromules_2.fasta")

Niben_Stromules_Chloro <- read.delim("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan/Funcional Groups/Niben_Stromules_2_summary.targetp2", 
                                  header=FALSE, comment.char="#")
colnames(Niben_Stromules_Chloro) <- c("ID",	"Prediction",	"noTP",	"SP",	"mTP",	"cTP",	"luTP",	"CS-Position")
Niben_Stromules_Chloro$Prediction <- factor(Niben_Stromules_Chloro$Prediction, levels = c("cTP","mTP","noTP"))
Niben_Stromules_Chloro <- merge(Niben_Stromules_Ann, Niben_Stromules_Chloro, by.x = "Protein", by.y = "ID")

Niben_Stromules_Chloro <- Niben_Stromules_Chloro[order(match(Niben_Stromules_Chloro$Protein, Niben_Stromules_Ann$Protein)),]
rownames(Niben_Stromules_Chloro) <- Niben_Stromules_Chloro$Protein
Niben_Stromules_Chloro <- Niben_Stromules_Chloro %>% select(Prediction)


## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_Stromules <- merge(Niben_Stromules_Ann, TPM_All_FC_Annotated, by="Name")

## We create the annotation for Modules

Niben_Stromules_Modules <- which_Module_pwd12(Niben_Stromules_Ann)

## We merge this annotation of Modules with the annotation of Chloroplast TP prediction

Niben_Stromules_Modules_Chl <- merge(Niben_Stromules_Modules, Niben_Stromules_Chloro, by = 0)
Niben_Stromules_Modules_Chl <- Niben_Stromules_Modules_Chl[order(match(Niben_Stromules_Modules_Chl$Row.names, Niben_Stromules_Ann$Protein)),]
rownames(Niben_Stromules_Modules_Chl) <- Niben_Stromules_Modules_Chl$Row.names
Niben_Stromules_Modules_Chl$Row.names <- NULL

## We select only the FC data from the different comparisons
Niben_Stromules_TPM <- Niben_Stromules %>% select(Protein, 4:51, 61:66)
Niben_Stromules_TPM <- Niben_Stromules_TPM[!duplicated(Niben_Stromules_TPM$Protein),]
Niben_Stromules_TPM <- Niben_Stromules_TPM[order(match(Niben_Stromules_TPM[,1], Niben_Stromules_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_Stromules_FC <- Niben_Stromules %>% select(Protein, 71, 73:87)
Niben_Stromules_FC <- Niben_Stromules_FC[!duplicated(Niben_Stromules_FC$Protein),]
Niben_Stromules_FC <- Niben_Stromules_FC[order(match(Niben_Stromules_FC[,1], Niben_Stromules_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_Stromules_DEG <- Niben_Stromules %>% select(Protein, 89, 91:105)
Niben_Stromules_DEG <- Niben_Stromules_DEG[!duplicated(Niben_Stromules_DEG$Protein),]
Niben_Stromules_DEG <- Niben_Stromules_DEG[order(match(Niben_Stromules_DEG[,1], Niben_Stromules_Ann[,2])),]

Niben_Stromules_DEG$pcrtB22h_vs_GFP22h_filtered <- ifelse(is.na(Niben_Stromules_DEG$pcrtB22h_vs_GFP22h_filtered) == TRUE, " ", "*")
Niben_Stromules_DEG$pcrtB25h_vs_GFP25h_filtered <- ifelse(is.na(Niben_Stromules_DEG$pcrtB25h_vs_GFP25h_filtered) == TRUE, " ", "*")
Niben_Stromules_DEG$pcrtBAA_vs_GFPAA_filtered <- ifelse(is.na(Niben_Stromules_DEG$pcrtBAA_vs_GFPAA_filtered) == TRUE, " ", "*")
Niben_Stromules_DEG$pcrtBBB_vs_GFPBB_filtered <- ifelse(is.na(Niben_Stromules_DEG$pcrtBBB_vs_GFPBB_filtered) == TRUE, " ", "*")
Niben_Stromules_DEG$pcrtBYY_vs_GFPYY_filtered <- ifelse(is.na(Niben_Stromules_DEG$pcrtBYY_vs_GFPYY_filtered) == TRUE, " ", "*")
Niben_Stromules_DEG$pcrtB28h_vs_GFP28h_filtered <- ifelse(is.na(Niben_Stromules_DEG$pcrtB28h_vs_GFP28h_filtered) == TRUE, " ", "*")
Niben_Stromules_DEG$pcrtB34h_vs_GFP34h_filtered <- ifelse(is.na(Niben_Stromules_DEG$pcrtB34h_vs_GFP34h_filtered) == TRUE, " ", "*")
Niben_Stromules_DEG$pcrtB37h_vs_GFP37h_filtered <- ifelse(is.na(Niben_Stromules_DEG$pcrtB37h_vs_GFP37h_filtered) == TRUE, " ", "*")
Niben_Stromules_DEG$pcrtBZZ_vs_GFPZZ_filtered <- ifelse(is.na(Niben_Stromules_DEG$pcrtBZZ_vs_GFPZZ_filtered) == TRUE, " ", "*")
Niben_Stromules_DEG$pcrtBCC_vs_GFPCC_filtered <- ifelse(is.na(Niben_Stromules_DEG$pcrtBCC_vs_GFPCC_filtered) == TRUE, " ", "*")
Niben_Stromules_DEG$pcrtB40h_vs_GFP40h_filtered <- ifelse(is.na(Niben_Stromules_DEG$pcrtB40h_vs_GFP40h_filtered) == TRUE, " ", "*")
Niben_Stromules_DEG$pcrtBDD_vs_GFPDD_filtered <- ifelse(is.na(Niben_Stromules_DEG$pcrtBDD_vs_GFPDD_filtered) == TRUE, " ", "*")
Niben_Stromules_DEG$pcrtB46h_vs_GFP46h_filtered <- ifelse(is.na(Niben_Stromules_DEG$pcrtB46h_vs_GFP46h_filtered) == TRUE, " ", "*")
Niben_Stromules_DEG$pcrtB56h_vs_GFP56h_filtered <- ifelse(is.na(Niben_Stromules_DEG$pcrtB56h_vs_GFP56h_filtered) == TRUE, " ", "*")
Niben_Stromules_DEG$pcrtBEE_vs_GFPEE_filtered <- ifelse(is.na(Niben_Stromules_DEG$pcrtBEE_vs_GFPEE_filtered) == TRUE, " ", "*")
Niben_Stromules_DEG$pcrtB96h_vs_GFP96h_filtered <- ifelse(is.na(Niben_Stromules_DEG$pcrtB96h_vs_GFP96h_filtered) == TRUE, " ", "*")
row.names(Niben_Stromules_DEG) <- Niben_Stromules_DEG$Protein
Niben_Stromules_DEG[1] <- NULL

Niben_Stromules_DEG <- Niben_Stromules_DEG %>%
  select(pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

row.names(Niben_Stromules_FC) <- Niben_Stromules_FC$Protein
Niben_Stromules_FC[1] <- NULL

Niben_Stromules_FC <- Niben_Stromules_FC %>%
  select(pcrtB22h_vs_GFP22h_All, pcrtBAA_vs_GFPAA_All, pcrtB25h_vs_GFP25h_All, 
         pcrtBYY_vs_GFPYY_All, pcrtBBB_vs_GFPBB_All, pcrtB28h_vs_GFP28h_All, pcrtB34h_vs_GFP34h_All, pcrtBZZ_vs_GFPZZ_All, 
         pcrtB37h_vs_GFP37h_All, pcrtBCC_vs_GFPCC_All, pcrtB40h_vs_GFP40h_All, pcrtBDD_vs_GFPDD_All, pcrtB46h_vs_GFP46h_All, 
         pcrtBEE_vs_GFPEE_All, pcrtB56h_vs_GFP56h_All, pcrtB96h_vs_GFP96h_All)


min(Niben_Stromules_FC, na.rm = TRUE)
max(Niben_Stromules_FC, na.rm = TRUE)
breaks_Stromules_FC=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Niben_Stromules_FC %>% select(1:15), 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Stromules_FC,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_Stromules_Annotation, annotation_colors=color_Stromules, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h"),#, "96h"),
         cex = 1,
         angle_col=45,
         main = "Stromules",
         annotation_legend = TRUE,
         display_numbers = Niben_Stromules_DEG %>% select(1:15),
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(84, 104),
         annotation_row = Niben_Stromules_Modules_Chl,
         annotation_colors = module_Chl_colors_2
         #gaps_col = c(15)
)


rownames(Niben_Stromules_TPM) <- Niben_Stromules_TPM$Protein
Niben_Stromules_TPM[1] <- NULL

min(Niben_Stromules_TPM, na.rm = TRUE)
max(Niben_Stromules_TPM, na.rm = TRUE)
breaks_Niben_Stromules_TPM=c(seq(0,100, length.out = 100))

Niben_Stromules_TPM_norm <- normalize_df(Niben_Stromules_TPM)
Niben_Stromules_TPM_norm_Cornell <- Niben_Stromules_TPM %>%
  select(1:48) 
Niben_Stromules_TPM_norm_Cornell <- normalize_df(Niben_Stromules_TPM_norm_Cornell)


pheatmap(Niben_Stromules_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Niben_Stromules_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_Stromules_Annotation, annotation_colors=color_Stromules, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Stromules",
         annotation_legend = TRUE,
         #display_numbers = Niben_Stromules_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         annotation_row = Niben_Stromules_Modules,
         annotation_colors = module_colors,
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(24, 48, 51)
)



#####Representing Heatmap for Sly

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Sly_Stromules_Ann <- read.table("Funcional Groups/FigTree/Sly_Stromules.txt",
                             sep = "\t", header = TRUE)

colnames(Sly_Stromules_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Sly_Stromules <- merge(Sly_Stromules_Ann, Summary_Yazdani_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Sly_Stromules_TPM <- Sly_Stromules %>% select(Protein, 4:66)
Sly_Stromules_TPM <- Sly_Stromules_TPM[!duplicated(Sly_Stromules_TPM$Protein),]
Sly_Stromules_TPM <- Sly_Stromules_TPM[order(match(Sly_Stromules_TPM[,1], Sly_Stromules_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_Stromules_FC <- Sly_Stromules %>% select(Protein, 67:71)
Sly_Stromules_FC <- Sly_Stromules_FC[!duplicated(Sly_Stromules_FC$Protein),]
Sly_Stromules_FC <- Sly_Stromules_FC[order(match(Sly_Stromules_FC[,1], Sly_Stromules_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_Stromules_DEG <- Sly_Stromules %>% select(Protein, 72:76)
Sly_Stromules_DEG <- Sly_Stromules_DEG[!duplicated(Sly_Stromules_DEG$Protein),]
Sly_Stromules_DEG <- Sly_Stromules_DEG[order(match(Sly_Stromules_DEG[,1], Sly_Stromules_Ann[,2])),]

Sly_Stromules_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Sly_Stromules_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Sly_Stromules_DEG$Ripening_Orange_DEG <- ifelse(is.na(Sly_Stromules_DEG$Ripening_Orange_DEG) == TRUE, " ", "*")
Sly_Stromules_DEG$Ripening_Red_DEG <- ifelse(is.na(Sly_Stromules_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
Sly_Stromules_DEG$ORHis_MG_DEG <- ifelse(is.na(Sly_Stromules_DEG$ORHis_MG_DEG) == TRUE, " ", "*")
Sly_Stromules_DEG$ORHis_Breaker_DEG <- ifelse(is.na(Sly_Stromules_DEG$ORHis_Breaker_DEG) == TRUE, " ", "*")
row.names(Sly_Stromules_DEG) <- Sly_Stromules_DEG$Protein
Sly_Stromules_DEG[1] <- NULL

rownames(Sly_Stromules_FC) <- Sly_Stromules_FC$Protein
Sly_Stromules_FC[1] <- NULL

min(Sly_Stromules_FC, na.rm = TRUE)
max(Sly_Stromules_FC, na.rm = TRUE)
breaks_Sly_Stromules=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Sly_Stromules_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Sly_Stromules,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_Stromules_Annotation, annotation_colors=color_Stromules, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Stromules",
         annotation_legend = TRUE,
         display_numbers = Sly_Stromules_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(39, 48),
         gaps_col = c(3)
)


rownames(Sly_Stromules_TPM) <- Sly_Stromules_TPM$Protein
Sly_Stromules_TPM[1] <- NULL

min(Sly_Stromules_TPM, na.rm = TRUE)
max(Sly_Stromules_TPM, na.rm = TRUE)
breaks_Sly_Stromules_TPM=c(seq(0,100, length.out = 200))

Sly_Stromules_TPM_norm <- normalize_df(Sly_Stromules_TPM)

pheatmap(Sly_Stromules_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Sly_Stromules_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_Stromules_Annotation, annotation_colors=color_Stromules, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Stromules",
         annotation_legend = TRUE,
         #display_numbers = Sly_Stromules_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(21,42)
)


#####Representing Heatmap for Cann

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Cann_Stromules_Ann <- read.table("Funcional Groups/FigTree/Cann_Stromules.txt",
                              sep = "\t", header = TRUE)

colnames(Cann_Stromules_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Cann_Stromules <- merge(Cann_Stromules_Ann, Summary_Pepper_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Cann_Stromules_TPM <- Cann_Stromules %>% select(Protein, 4:17)
Cann_Stromules_TPM <- Cann_Stromules_TPM[!duplicated(Cann_Stromules_TPM$Protein),]
Cann_Stromules_TPM <- Cann_Stromules_TPM[order(match(Cann_Stromules_TPM[,1], Cann_Stromules_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_Stromules_FC <- Cann_Stromules %>% select(Protein, 18:19)
Cann_Stromules_FC <- Cann_Stromules_FC[!duplicated(Cann_Stromules_FC$Protein),]
Cann_Stromules_FC <- Cann_Stromules_FC[order(match(Cann_Stromules_FC[,1], Cann_Stromules_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_Stromules_DEG <- Cann_Stromules %>% select(Protein, 20:21)
Cann_Stromules_DEG <- Cann_Stromules_DEG[!duplicated(Cann_Stromules_DEG$Protein),]
Cann_Stromules_DEG <- Cann_Stromules_DEG[order(match(Cann_Stromules_DEG[,1], Cann_Stromules_Ann[,2])),]

Cann_Stromules_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Cann_Stromules_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Cann_Stromules_DEG$Ripening_Red_DEG <- ifelse(is.na(Cann_Stromules_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
row.names(Cann_Stromules_DEG) <- Cann_Stromules_DEG$Protein
Cann_Stromules_DEG[1] <- NULL

rownames(Cann_Stromules_FC) <- Cann_Stromules_FC$Protein
Cann_Stromules_FC[1] <- NULL

min(Cann_Stromules_FC, na.rm = TRUE)
max(Cann_Stromules_FC, na.rm = TRUE)
breaks_Cann_Stromules=c(seq(-4.5,0, length.out = 100), seq(0.01,6,length.out = 100))

pheatmap(Cann_Stromules_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Cann_Stromules,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_Stromules_Annotation, annotation_colors=color_Stromules, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Red"),
         cex = 1,
         angle_col=45,
         main = "Stromules",
         annotation_legend = TRUE,
         display_numbers = Cann_Stromules_DEG,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)


rownames(Cann_Stromules_TPM) <- Cann_Stromules_TPM$Protein
Cann_Stromules_TPM[1] <- NULL

min(Cann_Stromules_TPM, na.rm = TRUE)
max(Cann_Stromules_TPM, na.rm = TRUE)
breaks_Cann_Stromules_TPM=c(seq(0,100, length.out = 200))

Cann_Stromules_TPM_norm <- normalize_df(Cann_Stromules_TPM)

pheatmap(Cann_Stromules_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Cann_Stromules_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_Stromules_Annotation, annotation_colors=color_Stromules, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Stromules",
         annotation_legend = TRUE,
         #display_numbers = Cann_Stromules_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(4,9)
)

## Comparativa 96h_SlyYaz_Ath

Ath_Stromules_Ann <- read.table("Funcional Groups/Ath_Stromules.txt", header = TRUE)

Ath_Stromules_FC <- merge(Ath_Stromules_Ann, Summary_Ath_FC, by.x=1, by.y = 0, all.x = TRUE) %>%
  dplyr::select(Name, log2FoldChange)
Ath_Stromules_FC <- Ath_Stromules_FC[order(match(Ath_Stromules_FC[,1], Ath_Stromules_Ann[,2])),]

Ath_Stromules_DEG <- merge(Ath_Stromules_Ann, Summary_Ath_DEGs, by.x=1, by.y = 1, all.x = TRUE) %>%
  dplyr::select(Name, log2FC)
Ath_Stromules_DEG <- Ath_Stromules_DEG[order(match(Ath_Stromules_DEG[,1], Ath_Stromules_Ann[,2])),]

Ath_Stromules_DEG$log2FC <- ifelse(is.na(Ath_Stromules_DEG$log2FC) == TRUE, " ", "*")
row.names(Ath_Stromules_DEG) <- Ath_Stromules_DEG$Name
Ath_Stromules_DEG[1] <- NULL

rownames(Ath_Stromules_FC) <- Ath_Stromules_FC$Name
Ath_Stromules_FC[1] <- NULL

min(Ath_Stromules_FC, na.rm = TRUE)
max(Ath_Stromules_FC, na.rm = TRUE)
breaks_Ath_Stromules=c(seq(-4.5,0, length.out = 100), seq(0.01,4.5,length.out = 100))

pheatmap(Ath_Stromules_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Stndr,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_Stromules_Annotation, annotation_colors=color_Stromules, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = "Senescence",
         cex = 1,
         angle_col=45,
         main = "Stromules",
         annotation_legend = TRUE,
         display_numbers = Ath_Stromules_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         gaps_row = c(3, 4)
         #gaps_col = c(2,5)
)

pheatmap(Niben_Stromules_FC %>% select(16), 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         #breaks = breaks_Stromules_FC,
         breaks = breaks_Stndr,
         #legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         #legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_Stromules_Annotation, annotation_colors=color_Stromules, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = "96h",
         cex = 1,
         angle_col=45,
         main = "Stromules",
         annotation_legend = TRUE,
         display_numbers = Niben_Stromules_DEG %>% select(16),
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(84, 104),
         annotation_row = Niben_Stromules_Modules_Chl,
         annotation_colors = module_Chl_colors_2,
         gaps_row = c(3, 5, 9, 12, 15, 17)
)


pheatmap(Sly_Stromules_FC %>% select(2,3), 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Stndr,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         #legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_Stromules_Annotation, annotation_colors=color_Stromules, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Orange", "Ripening_Red"),
         cex = 1,
         angle_col=45,
         main = "Stromules",
         annotation_legend = TRUE,
         display_numbers = Sly_Stromules_DEG %>% select(2,3),
         fontsize_number = 15,
         na_col = "darkgrey",
         gaps_row = c(1, 3,  4, 7, 8)
)


#########################################################################################################################################
####################################  JAZ  #############################################################################################
#########################################################################################################################################

JAZ_Ath <- read.table("Funcional Groups/HMMER/Jas/Ath_JAZ.txt", header = TRUE)
JAZ_Ath <- merge(JAZ_Ath, Ath_Gene_Models, by.x="ID", by.y=1)
JAZ_Ath_fasta <- Ath11_fasta[c(which(names(Ath11_fasta) %in% JAZ_Ath$V2))]
JAZ_Ath_2 <- JAZ_Ath %>% select(V2, Name)
write.fasta(JAZ_Ath_fasta, as.string=TRUE, names= names(JAZ_Ath_fasta),
            file.out = "Funcional Groups/Ath_JAZ.fasta")
rename.fasta(infile = "Funcional Groups/Ath_JAZ.fasta", JAZ_Ath_2, outfile = "Funcional Groups/Ath_JAZ_2.fasta")

JAZ_Niben_Candidates <- read.table("Funcional Groups/HMMER/Jas/Niben_JAZ_HMMER.txt", header = TRUE)
JAZ_Niben_Candidates <- unique(JAZ_Niben_Candidates)
JAZ_Niben_Candidates_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% JAZ_Niben_Candidates$ID))]
write.fasta(JAZ_Niben_Candidates_fasta, as.string=TRUE, names= names(JAZ_Niben_Candidates_fasta),
            file.out = "Funcional Groups/JAZ_Niben_Candidates_35.fasta")

JAZ_Sly_Candidates <- read.table("Funcional Groups/HMMER/Jas/Sly_JAZ_HMMER.txt", header = TRUE)
JAZ_Sly_Candidates <- unique(JAZ_Sly_Candidates)
JAZ_Sly_Candidates_fasta <- Tomato_4.0_fasta[c(which(names(Tomato_4.0_fasta) %in% JAZ_Sly_Candidates$ID))]
write.fasta(JAZ_Sly_Candidates_fasta, as.string=TRUE, names= names(JAZ_Sly_Candidates_fasta),
            file.out = "Funcional Groups/JAZ_Sly_Candidates_35.fasta")

JAZ_Cann_Candidates <- read.table("Funcional Groups/HMMER/Jas/Cann_JAZ_HMMER.txt", header = TRUE)
JAZ_Cann_Candidates <- unique(JAZ_Cann_Candidates)
JAZ_Cann_Candidates_fasta <- Cannuum_ASM51225v2_fasta[c(which(names(Cannuum_ASM51225v2_fasta) %in% JAZ_Cann_Candidates$ID))]
write.fasta(JAZ_Cann_Candidates_fasta, as.string=TRUE, names= names(JAZ_Cann_Candidates_fasta),
            file.out = "Funcional Groups/JAZ_Cann_Candidates_35.fasta")

Niben_JAZ_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_JAZ_Ann$Name))]
write.fasta(Niben_JAZ_fasta, as.string=TRUE, names= names(Niben_JAZ_fasta),
            file.out = "Funcional Groups/Niben_JAZ.fasta")
Niben_JAZ_Ann_2 <- Niben_JAZ_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_JAZ.fasta", Niben_JAZ_Ann, outfile = "Funcional Groups/Niben_JAZ_2.fasta")

save_heatmaps("JAZ", height = 6)

## Representing Heatmap

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Niben_JAZ_Ann <- read.table("Funcional Groups/FigTree/Niben_JAZ.txt",
                                  sep = "\t", header = TRUE)

colnames(Niben_JAZ_Ann) <- c("Name", "Protein")

Niben_JAZ_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_JAZ_Ann$Name))]
write.fasta(Niben_JAZ_fasta, as.string=TRUE, names= names(Niben_JAZ_fasta),
            file.out = "Funcional Groups/Niben_JAZ.fasta")
Niben_JAZ_Ann_2 <- Niben_JAZ_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_JAZ.fasta", Niben_JAZ_Ann, outfile = "Funcional Groups/Niben_JAZ_2.fasta")

Niben_JAZ_Chloro <- read.delim("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan/Funcional Groups/Niben_JAZ_2_summary.targetp2", 
                                     header=FALSE, comment.char="#")
colnames(Niben_JAZ_Chloro) <- c("ID",	"Prediction",	"noTP",	"SP",	"mTP",	"cTP",	"luTP",	"CS-Position")
Niben_JAZ_Chloro$Prediction <- factor(Niben_JAZ_Chloro$Prediction, levels = c("cTP","mTP","noTP"))
Niben_JAZ_Chloro <- merge(Niben_JAZ_Ann, Niben_JAZ_Chloro, by.x = "Protein", by.y = "ID")

Niben_JAZ_Chloro <- Niben_JAZ_Chloro[order(match(Niben_JAZ_Chloro$Protein, Niben_JAZ_Ann$Protein)),]
rownames(Niben_JAZ_Chloro) <- Niben_JAZ_Chloro$Protein
Niben_JAZ_Chloro <- Niben_JAZ_Chloro %>% select(Prediction)


## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_JAZ <- merge(Niben_JAZ_Ann, TPM_All_FC_Annotated, by="Name")

## We create the annotation for Modules

Niben_JAZ_Modules <- which_Module_pwd12(Niben_JAZ_Ann)

## We merge this annotation of Modules with the annotation of Chloroplast TP prediction

Niben_JAZ_Modules_Chl <- merge(Niben_JAZ_Modules, Niben_JAZ_Chloro, by = 0)
Niben_JAZ_Modules_Chl <- Niben_JAZ_Modules_Chl[order(match(Niben_JAZ_Modules_Chl$Row.names, Niben_JAZ_Ann$Protein)),]
rownames(Niben_JAZ_Modules_Chl) <- Niben_JAZ_Modules_Chl$Row.names
Niben_JAZ_Modules_Chl$Row.names <- NULL

## We select only the FC data from the different comparisons
Niben_JAZ_TPM <- Niben_JAZ %>% select(Protein, 4:51, 61:66)
Niben_JAZ_TPM <- Niben_JAZ_TPM[!duplicated(Niben_JAZ_TPM$Protein),]
Niben_JAZ_TPM <- Niben_JAZ_TPM[order(match(Niben_JAZ_TPM[,1], Niben_JAZ_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_JAZ_FC <- Niben_JAZ %>% select(Protein, 71, 73:87)
Niben_JAZ_FC <- Niben_JAZ_FC[!duplicated(Niben_JAZ_FC$Protein),]
Niben_JAZ_FC <- Niben_JAZ_FC[order(match(Niben_JAZ_FC[,1], Niben_JAZ_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_JAZ_DEG <- Niben_JAZ %>% select(Protein, 89, 91:105)
Niben_JAZ_DEG <- Niben_JAZ_DEG[!duplicated(Niben_JAZ_DEG$Protein),]
Niben_JAZ_DEG <- Niben_JAZ_DEG[order(match(Niben_JAZ_DEG[,1], Niben_JAZ_Ann[,2])),]

Niben_JAZ_DEG$pcrtB22h_vs_GFP22h_filtered <- ifelse(is.na(Niben_JAZ_DEG$pcrtB22h_vs_GFP22h_filtered) == TRUE, " ", "*")
Niben_JAZ_DEG$pcrtB25h_vs_GFP25h_filtered <- ifelse(is.na(Niben_JAZ_DEG$pcrtB25h_vs_GFP25h_filtered) == TRUE, " ", "*")
Niben_JAZ_DEG$pcrtBAA_vs_GFPAA_filtered <- ifelse(is.na(Niben_JAZ_DEG$pcrtBAA_vs_GFPAA_filtered) == TRUE, " ", "*")
Niben_JAZ_DEG$pcrtBBB_vs_GFPBB_filtered <- ifelse(is.na(Niben_JAZ_DEG$pcrtBBB_vs_GFPBB_filtered) == TRUE, " ", "*")
Niben_JAZ_DEG$pcrtBYY_vs_GFPYY_filtered <- ifelse(is.na(Niben_JAZ_DEG$pcrtBYY_vs_GFPYY_filtered) == TRUE, " ", "*")
Niben_JAZ_DEG$pcrtB28h_vs_GFP28h_filtered <- ifelse(is.na(Niben_JAZ_DEG$pcrtB28h_vs_GFP28h_filtered) == TRUE, " ", "*")
Niben_JAZ_DEG$pcrtB34h_vs_GFP34h_filtered <- ifelse(is.na(Niben_JAZ_DEG$pcrtB34h_vs_GFP34h_filtered) == TRUE, " ", "*")
Niben_JAZ_DEG$pcrtB37h_vs_GFP37h_filtered <- ifelse(is.na(Niben_JAZ_DEG$pcrtB37h_vs_GFP37h_filtered) == TRUE, " ", "*")
Niben_JAZ_DEG$pcrtBZZ_vs_GFPZZ_filtered <- ifelse(is.na(Niben_JAZ_DEG$pcrtBZZ_vs_GFPZZ_filtered) == TRUE, " ", "*")
Niben_JAZ_DEG$pcrtBCC_vs_GFPCC_filtered <- ifelse(is.na(Niben_JAZ_DEG$pcrtBCC_vs_GFPCC_filtered) == TRUE, " ", "*")
Niben_JAZ_DEG$pcrtB40h_vs_GFP40h_filtered <- ifelse(is.na(Niben_JAZ_DEG$pcrtB40h_vs_GFP40h_filtered) == TRUE, " ", "*")
Niben_JAZ_DEG$pcrtBDD_vs_GFPDD_filtered <- ifelse(is.na(Niben_JAZ_DEG$pcrtBDD_vs_GFPDD_filtered) == TRUE, " ", "*")
Niben_JAZ_DEG$pcrtB46h_vs_GFP46h_filtered <- ifelse(is.na(Niben_JAZ_DEG$pcrtB46h_vs_GFP46h_filtered) == TRUE, " ", "*")
Niben_JAZ_DEG$pcrtB56h_vs_GFP56h_filtered <- ifelse(is.na(Niben_JAZ_DEG$pcrtB56h_vs_GFP56h_filtered) == TRUE, " ", "*")
Niben_JAZ_DEG$pcrtBEE_vs_GFPEE_filtered <- ifelse(is.na(Niben_JAZ_DEG$pcrtBEE_vs_GFPEE_filtered) == TRUE, " ", "*")
Niben_JAZ_DEG$pcrtB96h_vs_GFP96h_filtered <- ifelse(is.na(Niben_JAZ_DEG$pcrtB96h_vs_GFP96h_filtered) == TRUE, " ", "*")
row.names(Niben_JAZ_DEG) <- Niben_JAZ_DEG$Protein
Niben_JAZ_DEG[1] <- NULL

Niben_JAZ_DEG <- Niben_JAZ_DEG %>%
  select(pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

row.names(Niben_JAZ_FC) <- Niben_JAZ_FC$Protein
Niben_JAZ_FC[1] <- NULL

Niben_JAZ_FC <- Niben_JAZ_FC %>%
  select(pcrtB22h_vs_GFP22h_All, pcrtBAA_vs_GFPAA_All, pcrtB25h_vs_GFP25h_All, 
         pcrtBYY_vs_GFPYY_All, pcrtBBB_vs_GFPBB_All, pcrtB28h_vs_GFP28h_All, pcrtB34h_vs_GFP34h_All, pcrtBZZ_vs_GFPZZ_All, 
         pcrtB37h_vs_GFP37h_All, pcrtBCC_vs_GFPCC_All, pcrtB40h_vs_GFP40h_All, pcrtBDD_vs_GFPDD_All, pcrtB46h_vs_GFP46h_All, 
         pcrtBEE_vs_GFPEE_All, pcrtB56h_vs_GFP56h_All, pcrtB96h_vs_GFP96h_All)


min(Niben_JAZ_FC, na.rm = TRUE)
max(Niben_JAZ_FC, na.rm = TRUE)
breaks_JAZ_FC=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Niben_JAZ_FC %>% select(1:15), 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_JAZ_FC,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_JAZ_Annotation, annotation_colors=color_JAZ, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h"),#, "96h"),
         cex = 1,
         angle_col=45,
         main = "JAZ",
         annotation_legend = TRUE,
         display_numbers = Niben_JAZ_DEG %>% select(1:15),
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(84, 104),
         annotation_row = Niben_JAZ_Modules_Chl,
         annotation_colors = module_Chl_colors_2
         #gaps_col = c(15)
)


rownames(Niben_JAZ_TPM) <- Niben_JAZ_TPM$Protein
Niben_JAZ_TPM[1] <- NULL

min(Niben_JAZ_TPM, na.rm = TRUE)
max(Niben_JAZ_TPM, na.rm = TRUE)
breaks_Niben_JAZ_TPM=c(seq(0,100, length.out = 100))

Niben_JAZ_TPM_norm <- normalize_df(Niben_JAZ_TPM)
Niben_JAZ_TPM_norm_Cornell <- Niben_JAZ_TPM %>%
  select(1:48) 
Niben_JAZ_TPM_norm_Cornell <- normalize_df(Niben_JAZ_TPM_norm_Cornell)


pheatmap(Niben_JAZ_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Niben_JAZ_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_JAZ_Annotation, annotation_colors=color_JAZ, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "JAZ",
         annotation_legend = TRUE,
         #display_numbers = Niben_JAZ_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         annotation_row = Niben_JAZ_Modules,
         annotation_colors = module_colors,
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(24, 48, 51)
)



#####Representing Heatmap for Sly

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Sly_JAZ_Ann <- read.table("Funcional Groups/FigTree/Sly_JAZ.txt",
                                sep = "\t", header = TRUE)

colnames(Sly_JAZ_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Sly_JAZ <- merge(Sly_JAZ_Ann, Summary_Yazdani_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Sly_JAZ_TPM <- Sly_JAZ %>% select(Protein, 4:66)
Sly_JAZ_TPM <- Sly_JAZ_TPM[!duplicated(Sly_JAZ_TPM$Protein),]
Sly_JAZ_TPM <- Sly_JAZ_TPM[order(match(Sly_JAZ_TPM[,1], Sly_JAZ_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_JAZ_FC <- Sly_JAZ %>% select(Protein, 67:71)
Sly_JAZ_FC <- Sly_JAZ_FC[!duplicated(Sly_JAZ_FC$Protein),]
Sly_JAZ_FC <- Sly_JAZ_FC[order(match(Sly_JAZ_FC[,1], Sly_JAZ_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_JAZ_DEG <- Sly_JAZ %>% select(Protein, 72:76)
Sly_JAZ_DEG <- Sly_JAZ_DEG[!duplicated(Sly_JAZ_DEG$Protein),]
Sly_JAZ_DEG <- Sly_JAZ_DEG[order(match(Sly_JAZ_DEG[,1], Sly_JAZ_Ann[,2])),]

Sly_JAZ_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Sly_JAZ_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Sly_JAZ_DEG$Ripening_Orange_DEG <- ifelse(is.na(Sly_JAZ_DEG$Ripening_Orange_DEG) == TRUE, " ", "*")
Sly_JAZ_DEG$Ripening_Red_DEG <- ifelse(is.na(Sly_JAZ_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
Sly_JAZ_DEG$ORHis_MG_DEG <- ifelse(is.na(Sly_JAZ_DEG$ORHis_MG_DEG) == TRUE, " ", "*")
Sly_JAZ_DEG$ORHis_Breaker_DEG <- ifelse(is.na(Sly_JAZ_DEG$ORHis_Breaker_DEG) == TRUE, " ", "*")
row.names(Sly_JAZ_DEG) <- Sly_JAZ_DEG$Protein
Sly_JAZ_DEG[1] <- NULL

rownames(Sly_JAZ_FC) <- Sly_JAZ_FC$Protein
Sly_JAZ_FC[1] <- NULL

min(Sly_JAZ_FC, na.rm = TRUE)
max(Sly_JAZ_FC, na.rm = TRUE)
breaks_Sly_JAZ=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Sly_JAZ_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Sly_JAZ,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_JAZ_Annotation, annotation_colors=color_JAZ, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "JAZ",
         annotation_legend = TRUE,
         display_numbers = Sly_JAZ_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(39, 48),
         gaps_col = c(3)
)


rownames(Sly_JAZ_TPM) <- Sly_JAZ_TPM$Protein
Sly_JAZ_TPM[1] <- NULL

min(Sly_JAZ_TPM, na.rm = TRUE)
max(Sly_JAZ_TPM, na.rm = TRUE)
breaks_Sly_JAZ_TPM=c(seq(0,100, length.out = 200))

Sly_JAZ_TPM_norm <- normalize_df(Sly_JAZ_TPM)

pheatmap(Sly_JAZ_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Sly_JAZ_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_JAZ_Annotation, annotation_colors=color_JAZ, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "JAZ",
         annotation_legend = TRUE,
         #display_numbers = Sly_JAZ_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(21,42)
)


#####Representing Heatmap for Cann

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Cann_JAZ_Ann <- read.table("Funcional Groups/FigTree/Cann_JAZ.txt",
                                 sep = "\t", header = TRUE)

colnames(Cann_JAZ_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Cann_JAZ <- merge(Cann_JAZ_Ann, Summary_Pepper_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Cann_JAZ_TPM <- Cann_JAZ %>% select(Protein, 4:17)
Cann_JAZ_TPM <- Cann_JAZ_TPM[!duplicated(Cann_JAZ_TPM$Protein),]
Cann_JAZ_TPM <- Cann_JAZ_TPM[order(match(Cann_JAZ_TPM[,1], Cann_JAZ_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_JAZ_FC <- Cann_JAZ %>% select(Protein, 18:19)
Cann_JAZ_FC <- Cann_JAZ_FC[!duplicated(Cann_JAZ_FC$Protein),]
Cann_JAZ_FC <- Cann_JAZ_FC[order(match(Cann_JAZ_FC[,1], Cann_JAZ_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_JAZ_DEG <- Cann_JAZ %>% select(Protein, 20:21)
Cann_JAZ_DEG <- Cann_JAZ_DEG[!duplicated(Cann_JAZ_DEG$Protein),]
Cann_JAZ_DEG <- Cann_JAZ_DEG[order(match(Cann_JAZ_DEG[,1], Cann_JAZ_Ann[,2])),]

Cann_JAZ_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Cann_JAZ_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Cann_JAZ_DEG$Ripening_Red_DEG <- ifelse(is.na(Cann_JAZ_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
row.names(Cann_JAZ_DEG) <- Cann_JAZ_DEG$Protein
Cann_JAZ_DEG[1] <- NULL

rownames(Cann_JAZ_FC) <- Cann_JAZ_FC$Protein
Cann_JAZ_FC[1] <- NULL

min(Cann_JAZ_FC, na.rm = TRUE)
max(Cann_JAZ_FC, na.rm = TRUE)
breaks_Cann_JAZ=c(seq(-4.5,0, length.out = 100), seq(0.01,6,length.out = 100))

pheatmap(Cann_JAZ_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Cann_JAZ,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_JAZ_Annotation, annotation_colors=color_JAZ, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Red"),
         cex = 1,
         angle_col=45,
         main = "JAZ",
         annotation_legend = TRUE,
         display_numbers = Cann_JAZ_DEG,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)


rownames(Cann_JAZ_TPM) <- Cann_JAZ_TPM$Protein
Cann_JAZ_TPM[1] <- NULL

min(Cann_JAZ_TPM, na.rm = TRUE)
max(Cann_JAZ_TPM, na.rm = TRUE)
breaks_Cann_JAZ_TPM=c(seq(0,100, length.out = 200))

Cann_JAZ_TPM_norm <- normalize_df(Cann_JAZ_TPM)

pheatmap(Cann_JAZ_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Cann_JAZ_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_JAZ_Annotation, annotation_colors=color_JAZ, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "JAZ",
         annotation_legend = TRUE,
         #display_numbers = Cann_JAZ_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(4,9)
)

#########################################################################################################################################
####################################  JasResp  #############################################################################################
#########################################################################################################################################

JasResp_Ath <- read.table("Funcional Groups/Ath_JasResp.txt", header = TRUE)
JasResp_Ath <- merge(JasResp_Ath, Ath_Gene_Models, by.x="ID", by.y=1)
JasResp_Ath_fasta <- Ath11_fasta[c(which(names(Ath11_fasta) %in% JasResp_Ath$V2))]
JasResp_Ath_2 <- JasResp_Ath %>% select(V2, Name)
write.fasta(JasResp_Ath_fasta, as.string=TRUE, names= names(JasResp_Ath_fasta),
            file.out = "Funcional Groups/Ath_JasResp.fasta")
rename.fasta(infile = "Funcional Groups/Ath_JasResp.fasta", JasResp_Ath_2, outfile = "Funcional Groups/Ath_JasResp_2.fasta")

JasResp_Niben_Candidates <- read.table("Funcional Groups/Niben_Candidates_JasResp.txt")
JasResp_Niben_Candidates <- JasResp_Niben_Candidates[JasResp_Niben_Candidates$V3>=35,] %>% select(V2)
JasResp_Niben_Candidates <- unique(JasResp_Niben_Candidates)
JasResp_Niben_Candidates_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% JasResp_Niben_Candidates$V2))]
write.fasta(JasResp_Niben_Candidates_fasta, as.string=TRUE, names= names(JasResp_Niben_Candidates_fasta),
            file.out = "Funcional Groups/JasResp_Niben_Candidates_35.fasta")

JasResp_Sly_Candidates <- read.table("Funcional Groups/Sly_Candidates_JasResp.txt")
JasResp_Sly_Candidates <- JasResp_Sly_Candidates[JasResp_Sly_Candidates$V3>=35,] %>% select(V2)
JasResp_Sly_Candidates <- unique(JasResp_Sly_Candidates)
JasResp_Sly_Candidates_fasta <- Tomato_4.0_fasta[c(which(names(Tomato_4.0_fasta) %in% JasResp_Sly_Candidates$V2))]
write.fasta(JasResp_Sly_Candidates_fasta, as.string=TRUE, names= names(JasResp_Sly_Candidates_fasta),
            file.out = "Funcional Groups/JasResp_Sly_Candidates_35.fasta")

JasResp_Cann_Candidates <- read.table("Funcional Groups/Cann_Candidates_JasResp.txt")
JasResp_Cann_Candidates <- JasResp_Cann_Candidates[JasResp_Cann_Candidates$V3>=35,] %>% select(V2)
JasResp_Cann_Candidates <- unique(JasResp_Cann_Candidates)
JasResp_Cann_Candidates_fasta <- Cannuum_ASM51225v2_fasta[c(which(names(Cannuum_ASM51225v2_fasta) %in% JasResp_Cann_Candidates$V2))]
write.fasta(JasResp_Cann_Candidates_fasta, as.string=TRUE, names= names(JasResp_Cann_Candidates_fasta),
            file.out = "Funcional Groups/JasResp_Cann_Candidates_35.fasta")

Niben_JasResp_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_JasResp_Ann$Name))]
write.fasta(Niben_JasResp_fasta, as.string=TRUE, names= names(Niben_JasResp_fasta),
            file.out = "Funcional Groups/Niben_JasResp.fasta")
Niben_JasResp_Ann_2 <- Niben_JasResp_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_JasResp.fasta", Niben_JasResp_Ann, outfile = "Funcional Groups/Niben_JasResp_2.fasta")

save_heatmaps("JasResp", height = 6)

## Representing Heatmap

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Niben_JasResp_Ann <- read.table("Funcional Groups/FigTree/Niben_JasResp.txt",
                                  sep = "\t", header = TRUE)

colnames(Niben_JasResp_Ann) <- c("Name", "Protein")

Niben_JasResp_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_JasResp_Ann$Name))]
write.fasta(Niben_JasResp_fasta, as.string=TRUE, names= names(Niben_JasResp_fasta),
            file.out = "Funcional Groups/Niben_JasResp.fasta")
Niben_JasResp_Ann_2 <- Niben_JasResp_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_JasResp.fasta", Niben_JasResp_Ann, outfile = "Funcional Groups/Niben_JasResp_2.fasta")

Niben_JasResp_Chloro <- read.delim("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan/Funcional Groups/Niben_JasResp_2_summary.targetp2", 
                                     header=FALSE, comment.char="#")
colnames(Niben_JasResp_Chloro) <- c("ID",	"Prediction",	"noTP",	"SP",	"mTP",	"cTP",	"luTP",	"CS-Position")
Niben_JasResp_Chloro$Prediction <- factor(Niben_JasResp_Chloro$Prediction, levels = c("cTP","mTP","noTP"))
Niben_JasResp_Chloro <- merge(Niben_JasResp_Ann, Niben_JasResp_Chloro, by.x = "Protein", by.y = "ID")

Niben_JasResp_Chloro <- Niben_JasResp_Chloro[order(match(Niben_JasResp_Chloro$Protein, Niben_JasResp_Ann$Protein)),]
rownames(Niben_JasResp_Chloro) <- Niben_JasResp_Chloro$Protein
Niben_JasResp_Chloro <- Niben_JasResp_Chloro %>% select(Prediction)


## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_JasResp <- merge(Niben_JasResp_Ann, TPM_All_FC_Annotated, by="Name")

## We create the annotation for Modules

Niben_JasResp_Modules <- which_Module_pwd12(Niben_JasResp_Ann)

## We merge this annotation of Modules with the annotation of Chloroplast TP prediction

Niben_JasResp_Modules_Chl <- merge(Niben_JasResp_Modules, Niben_JasResp_Chloro, by = 0)
Niben_JasResp_Modules_Chl <- Niben_JasResp_Modules_Chl[order(match(Niben_JasResp_Modules_Chl$Row.names, Niben_JasResp_Ann$Protein)),]
rownames(Niben_JasResp_Modules_Chl) <- Niben_JasResp_Modules_Chl$Row.names
Niben_JasResp_Modules_Chl$Row.names <- NULL

## We select only the FC data from the different comparisons
Niben_JasResp_TPM <- Niben_JasResp %>% select(Protein, 4:51, 61:66)
Niben_JasResp_TPM <- Niben_JasResp_TPM[!duplicated(Niben_JasResp_TPM$Protein),]
Niben_JasResp_TPM <- Niben_JasResp_TPM[order(match(Niben_JasResp_TPM[,1], Niben_JasResp_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_JasResp_FC <- Niben_JasResp %>% select(Protein, 71, 73:87)
Niben_JasResp_FC <- Niben_JasResp_FC[!duplicated(Niben_JasResp_FC$Protein),]
Niben_JasResp_FC <- Niben_JasResp_FC[order(match(Niben_JasResp_FC[,1], Niben_JasResp_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_JasResp_DEG <- Niben_JasResp %>% select(Protein, 89, 91:105)
Niben_JasResp_DEG <- Niben_JasResp_DEG[!duplicated(Niben_JasResp_DEG$Protein),]
Niben_JasResp_DEG <- Niben_JasResp_DEG[order(match(Niben_JasResp_DEG[,1], Niben_JasResp_Ann[,2])),]

Niben_JasResp_DEG$pcrtB22h_vs_GFP22h_filtered <- ifelse(is.na(Niben_JasResp_DEG$pcrtB22h_vs_GFP22h_filtered) == TRUE, " ", "*")
Niben_JasResp_DEG$pcrtB25h_vs_GFP25h_filtered <- ifelse(is.na(Niben_JasResp_DEG$pcrtB25h_vs_GFP25h_filtered) == TRUE, " ", "*")
Niben_JasResp_DEG$pcrtBAA_vs_GFPAA_filtered <- ifelse(is.na(Niben_JasResp_DEG$pcrtBAA_vs_GFPAA_filtered) == TRUE, " ", "*")
Niben_JasResp_DEG$pcrtBBB_vs_GFPBB_filtered <- ifelse(is.na(Niben_JasResp_DEG$pcrtBBB_vs_GFPBB_filtered) == TRUE, " ", "*")
Niben_JasResp_DEG$pcrtBYY_vs_GFPYY_filtered <- ifelse(is.na(Niben_JasResp_DEG$pcrtBYY_vs_GFPYY_filtered) == TRUE, " ", "*")
Niben_JasResp_DEG$pcrtB28h_vs_GFP28h_filtered <- ifelse(is.na(Niben_JasResp_DEG$pcrtB28h_vs_GFP28h_filtered) == TRUE, " ", "*")
Niben_JasResp_DEG$pcrtB34h_vs_GFP34h_filtered <- ifelse(is.na(Niben_JasResp_DEG$pcrtB34h_vs_GFP34h_filtered) == TRUE, " ", "*")
Niben_JasResp_DEG$pcrtB37h_vs_GFP37h_filtered <- ifelse(is.na(Niben_JasResp_DEG$pcrtB37h_vs_GFP37h_filtered) == TRUE, " ", "*")
Niben_JasResp_DEG$pcrtBZZ_vs_GFPZZ_filtered <- ifelse(is.na(Niben_JasResp_DEG$pcrtBZZ_vs_GFPZZ_filtered) == TRUE, " ", "*")
Niben_JasResp_DEG$pcrtBCC_vs_GFPCC_filtered <- ifelse(is.na(Niben_JasResp_DEG$pcrtBCC_vs_GFPCC_filtered) == TRUE, " ", "*")
Niben_JasResp_DEG$pcrtB40h_vs_GFP40h_filtered <- ifelse(is.na(Niben_JasResp_DEG$pcrtB40h_vs_GFP40h_filtered) == TRUE, " ", "*")
Niben_JasResp_DEG$pcrtBDD_vs_GFPDD_filtered <- ifelse(is.na(Niben_JasResp_DEG$pcrtBDD_vs_GFPDD_filtered) == TRUE, " ", "*")
Niben_JasResp_DEG$pcrtB46h_vs_GFP46h_filtered <- ifelse(is.na(Niben_JasResp_DEG$pcrtB46h_vs_GFP46h_filtered) == TRUE, " ", "*")
Niben_JasResp_DEG$pcrtB56h_vs_GFP56h_filtered <- ifelse(is.na(Niben_JasResp_DEG$pcrtB56h_vs_GFP56h_filtered) == TRUE, " ", "*")
Niben_JasResp_DEG$pcrtBEE_vs_GFPEE_filtered <- ifelse(is.na(Niben_JasResp_DEG$pcrtBEE_vs_GFPEE_filtered) == TRUE, " ", "*")
Niben_JasResp_DEG$pcrtB96h_vs_GFP96h_filtered <- ifelse(is.na(Niben_JasResp_DEG$pcrtB96h_vs_GFP96h_filtered) == TRUE, " ", "*")
row.names(Niben_JasResp_DEG) <- Niben_JasResp_DEG$Protein
Niben_JasResp_DEG[1] <- NULL

Niben_JasResp_DEG <- Niben_JasResp_DEG %>%
  select(pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

row.names(Niben_JasResp_FC) <- Niben_JasResp_FC$Protein
Niben_JasResp_FC[1] <- NULL

Niben_JasResp_FC <- Niben_JasResp_FC %>%
  select(pcrtB22h_vs_GFP22h_All, pcrtBAA_vs_GFPAA_All, pcrtB25h_vs_GFP25h_All, 
         pcrtBYY_vs_GFPYY_All, pcrtBBB_vs_GFPBB_All, pcrtB28h_vs_GFP28h_All, pcrtB34h_vs_GFP34h_All, pcrtBZZ_vs_GFPZZ_All, 
         pcrtB37h_vs_GFP37h_All, pcrtBCC_vs_GFPCC_All, pcrtB40h_vs_GFP40h_All, pcrtBDD_vs_GFPDD_All, pcrtB46h_vs_GFP46h_All, 
         pcrtBEE_vs_GFPEE_All, pcrtB56h_vs_GFP56h_All, pcrtB96h_vs_GFP96h_All)


min(Niben_JasResp_FC, na.rm = TRUE)
max(Niben_JasResp_FC, na.rm = TRUE)
breaks_JasResp_FC=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Niben_JasResp_FC %>% select(1:15), 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_JasResp_FC,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_JasResp_Annotation, annotation_colors=color_JasResp, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h"),#, "96h"),
         cex = 1,
         angle_col=45,
         main = "JasResp",
         annotation_legend = TRUE,
         display_numbers = Niben_JasResp_DEG %>% select(1:15),
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(84, 104),
         annotation_row = Niben_JasResp_Modules_Chl,
         annotation_colors = module_Chl_colors_2
         #gaps_col = c(15)
)


rownames(Niben_JasResp_TPM) <- Niben_JasResp_TPM$Protein
Niben_JasResp_TPM[1] <- NULL

min(Niben_JasResp_TPM, na.rm = TRUE)
max(Niben_JasResp_TPM, na.rm = TRUE)
breaks_Niben_JasResp_TPM=c(seq(0,100, length.out = 100))

Niben_JasResp_TPM_norm <- normalize_df(Niben_JasResp_TPM)
Niben_JasResp_TPM_norm_Cornell <- Niben_JasResp_TPM %>%
  select(1:48) 
Niben_JasResp_TPM_norm_Cornell <- normalize_df(Niben_JasResp_TPM_norm_Cornell)


pheatmap(Niben_JasResp_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Niben_JasResp_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_JasResp_Annotation, annotation_colors=color_JasResp, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "JasResp",
         annotation_legend = TRUE,
         #display_numbers = Niben_JasResp_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         annotation_row = Niben_JasResp_Modules,
         annotation_colors = module_colors,
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(24, 48, 51)
)



#####Representing Heatmap for Sly

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Sly_JasResp_Ann <- read.table("Funcional Groups/FigTree/Sly_JasResp.txt",
                                sep = "\t", header = TRUE)

colnames(Sly_JasResp_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Sly_JasResp <- merge(Sly_JasResp_Ann, Summary_Yazdani_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Sly_JasResp_TPM <- Sly_JasResp %>% select(Protein, 4:66)
Sly_JasResp_TPM <- Sly_JasResp_TPM[!duplicated(Sly_JasResp_TPM$Protein),]
Sly_JasResp_TPM <- Sly_JasResp_TPM[order(match(Sly_JasResp_TPM[,1], Sly_JasResp_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_JasResp_FC <- Sly_JasResp %>% select(Protein, 67:71)
Sly_JasResp_FC <- Sly_JasResp_FC[!duplicated(Sly_JasResp_FC$Protein),]
Sly_JasResp_FC <- Sly_JasResp_FC[order(match(Sly_JasResp_FC[,1], Sly_JasResp_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_JasResp_DEG <- Sly_JasResp %>% select(Protein, 72:76)
Sly_JasResp_DEG <- Sly_JasResp_DEG[!duplicated(Sly_JasResp_DEG$Protein),]
Sly_JasResp_DEG <- Sly_JasResp_DEG[order(match(Sly_JasResp_DEG[,1], Sly_JasResp_Ann[,2])),]

Sly_JasResp_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Sly_JasResp_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Sly_JasResp_DEG$Ripening_Orange_DEG <- ifelse(is.na(Sly_JasResp_DEG$Ripening_Orange_DEG) == TRUE, " ", "*")
Sly_JasResp_DEG$Ripening_Red_DEG <- ifelse(is.na(Sly_JasResp_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
Sly_JasResp_DEG$ORHis_MG_DEG <- ifelse(is.na(Sly_JasResp_DEG$ORHis_MG_DEG) == TRUE, " ", "*")
Sly_JasResp_DEG$ORHis_Breaker_DEG <- ifelse(is.na(Sly_JasResp_DEG$ORHis_Breaker_DEG) == TRUE, " ", "*")
row.names(Sly_JasResp_DEG) <- Sly_JasResp_DEG$Protein
Sly_JasResp_DEG[1] <- NULL

rownames(Sly_JasResp_FC) <- Sly_JasResp_FC$Protein
Sly_JasResp_FC[1] <- NULL

min(Sly_JasResp_FC, na.rm = TRUE)
max(Sly_JasResp_FC, na.rm = TRUE)
breaks_Sly_JasResp=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Sly_JasResp_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Sly_JasResp,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_JasResp_Annotation, annotation_colors=color_JasResp, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "JasResp",
         annotation_legend = TRUE,
         display_numbers = Sly_JasResp_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(39, 48),
         gaps_col = c(3)
)


rownames(Sly_JasResp_TPM) <- Sly_JasResp_TPM$Protein
Sly_JasResp_TPM[1] <- NULL

min(Sly_JasResp_TPM, na.rm = TRUE)
max(Sly_JasResp_TPM, na.rm = TRUE)
breaks_Sly_JasResp_TPM=c(seq(0,100, length.out = 200))

Sly_JasResp_TPM_norm <- normalize_df(Sly_JasResp_TPM)

pheatmap(Sly_JasResp_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Sly_JasResp_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_JasResp_Annotation, annotation_colors=color_JasResp, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "JasResp",
         annotation_legend = TRUE,
         #display_numbers = Sly_JasResp_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(21,42)
)


#####Representing Heatmap for Cann

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Cann_JasResp_Ann <- read.table("Funcional Groups/FigTree/Cann_JasResp.txt",
                                 sep = "\t", header = TRUE)

colnames(Cann_JasResp_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Cann_JasResp <- merge(Cann_JasResp_Ann, Summary_Pepper_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Cann_JasResp_TPM <- Cann_JasResp %>% select(Protein, 4:17)
Cann_JasResp_TPM <- Cann_JasResp_TPM[!duplicated(Cann_JasResp_TPM$Protein),]
Cann_JasResp_TPM <- Cann_JasResp_TPM[order(match(Cann_JasResp_TPM[,1], Cann_JasResp_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_JasResp_FC <- Cann_JasResp %>% select(Protein, 18:19)
Cann_JasResp_FC <- Cann_JasResp_FC[!duplicated(Cann_JasResp_FC$Protein),]
Cann_JasResp_FC <- Cann_JasResp_FC[order(match(Cann_JasResp_FC[,1], Cann_JasResp_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_JasResp_DEG <- Cann_JasResp %>% select(Protein, 20:21)
Cann_JasResp_DEG <- Cann_JasResp_DEG[!duplicated(Cann_JasResp_DEG$Protein),]
Cann_JasResp_DEG <- Cann_JasResp_DEG[order(match(Cann_JasResp_DEG[,1], Cann_JasResp_Ann[,2])),]

Cann_JasResp_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Cann_JasResp_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Cann_JasResp_DEG$Ripening_Red_DEG <- ifelse(is.na(Cann_JasResp_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
row.names(Cann_JasResp_DEG) <- Cann_JasResp_DEG$Protein
Cann_JasResp_DEG[1] <- NULL

rownames(Cann_JasResp_FC) <- Cann_JasResp_FC$Protein
Cann_JasResp_FC[1] <- NULL

min(Cann_JasResp_FC, na.rm = TRUE)
max(Cann_JasResp_FC, na.rm = TRUE)
breaks_Cann_JasResp=c(seq(-4.5,0, length.out = 100), seq(0.01,6,length.out = 100))

pheatmap(Cann_JasResp_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Cann_JasResp,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_JasResp_Annotation, annotation_colors=color_JasResp, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Red"),
         cex = 1,
         angle_col=45,
         main = "JasResp",
         annotation_legend = TRUE,
         display_numbers = Cann_JasResp_DEG,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)


rownames(Cann_JasResp_TPM) <- Cann_JasResp_TPM$Protein
Cann_JasResp_TPM[1] <- NULL

min(Cann_JasResp_TPM, na.rm = TRUE)
max(Cann_JasResp_TPM, na.rm = TRUE)
breaks_Cann_JasResp_TPM=c(seq(0,100, length.out = 200))

Cann_JasResp_TPM_norm <- normalize_df(Cann_JasResp_TPM)

pheatmap(Cann_JasResp_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Cann_JasResp_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_JasResp_Annotation, annotation_colors=color_JasResp, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "JasResp",
         annotation_legend = TRUE,
         #display_numbers = Cann_JasResp_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(4,9)
)


#########################################################################################################################################
####################################  Carbon  #############################################################################################
#########################################################################################################################################

Carbon_Ath <- read.table("Funcional Groups/Ath_Carbon.txt", header = TRUE)
Carbon_Ath <- merge(Carbon_Ath, Ath_Gene_Models, by.x="ID", by.y=1)
Carbon_Ath_fasta <- Ath11_fasta[c(which(names(Ath11_fasta) %in% Carbon_Ath$V2))]
Carbon_Ath_2 <- Carbon_Ath %>% select(V2, Name)
write.fasta(Carbon_Ath_fasta, as.string=TRUE, names= names(Carbon_Ath_fasta),
            file.out = "Funcional Groups/Ath_Carbon.fasta")
rename.fasta(infile = "Funcional Groups/Ath_Carbon.fasta", Carbon_Ath_2, outfile = "Funcional Groups/Ath_Carbon_2.fasta")

Carbon_Niben_Candidates <- read.table("Funcional Groups/Niben_Candidates_Carbon.txt")
Carbon_Niben_Candidates <- Carbon_Niben_Candidates[Carbon_Niben_Candidates$V3>=35,] %>% select(V2)
Carbon_Niben_Candidates <- unique(Carbon_Niben_Candidates)
Carbon_Niben_Candidates_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Carbon_Niben_Candidates$V2))]
write.fasta(Carbon_Niben_Candidates_fasta, as.string=TRUE, names= names(Carbon_Niben_Candidates_fasta),
            file.out = "Funcional Groups/Carbon_Niben_Candidates_35.fasta")

Carbon_Sly_Candidates <- read.table("Funcional Groups/Sly_Candidates_Carbon.txt")
Carbon_Sly_Candidates <- Carbon_Sly_Candidates[Carbon_Sly_Candidates$V3>=35,] %>% select(V2)
Carbon_Sly_Candidates <- unique(Carbon_Sly_Candidates)
Carbon_Sly_Candidates_fasta <- Tomato_4.0_fasta[c(which(names(Tomato_4.0_fasta) %in% Carbon_Sly_Candidates$V2))]
write.fasta(Carbon_Sly_Candidates_fasta, as.string=TRUE, names= names(Carbon_Sly_Candidates_fasta),
            file.out = "Funcional Groups/Carbon_Sly_Candidates_35.fasta")

Carbon_Cann_Candidates <- read.table("Funcional Groups/Cann_Candidates_Carbon.txt")
Carbon_Cann_Candidates <- Carbon_Cann_Candidates[Carbon_Cann_Candidates$V3>=35,] %>% select(V2)
Carbon_Cann_Candidates <- unique(Carbon_Cann_Candidates)
Carbon_Cann_Candidates_fasta <- Cannuum_ASM51225v2_fasta[c(which(names(Cannuum_ASM51225v2_fasta) %in% Carbon_Cann_Candidates$V2))]
write.fasta(Carbon_Cann_Candidates_fasta, as.string=TRUE, names= names(Carbon_Cann_Candidates_fasta),
            file.out = "Funcional Groups/Carbon_Cann_Candidates_35.fasta")

Niben_Carbon_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_Carbon_Ann$Name))]
write.fasta(Niben_Carbon_fasta, as.string=TRUE, names= names(Niben_Carbon_fasta),
            file.out = "Funcional Groups/Niben_Carbon.fasta")
Niben_Carbon_Ann_2 <- Niben_Carbon_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_Carbon.fasta", Niben_Carbon_Ann, outfile = "Funcional Groups/Niben_Carbon_2.fasta")

save_heatmaps("Carbon", height = 6)

## Representing Heatmap

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
#Niben_Carbon_Ann <- read.table("Funcional Groups/FigTree/Niben_Carbon.txt",
#                                sep = "\t", header = TRUE)
Niben_Carbon_Ann <- Carbon_Niben_Candidates
Niben_Carbon_Ann$Protein <- paste("TCArel", seq(1, nrow(Niben_Carbon_Ann)), sep = "_")
Niben_Carbon_Descr <- merge(Niben_Carbon_Ann, TPM_All_Annotated, by= 1) %>% select(1, Description)

colnames(Niben_Carbon_Ann) <- c("Name", "Protein")
write.table(Niben_Carbon_Ann, file= "Funcional Groups/FigTree/Niben_Carbon.txt",
            sep = "\t", row.names = FALSE, quote = FALSE)

Niben_Carbon_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_Carbon_Ann$Name))]
write.fasta(Niben_Carbon_fasta, as.string=TRUE, names= names(Niben_Carbon_fasta),
            file.out = "Funcional Groups/Niben_Carbon.fasta")
Niben_Carbon_Ann_2 <- Niben_Carbon_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_Carbon.fasta", Niben_Carbon_Ann, outfile = "Funcional Groups/Niben_Carbon_2.fasta")

Niben_Carbon_Chloro <- read.delim("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan/Funcional Groups/Niben_Carbon_2_summary.targetp2", 
                                   header=FALSE, comment.char="#")
colnames(Niben_Carbon_Chloro) <- c("ID",	"Prediction",	"noTP",	"SP",	"mTP",	"cTP",	"luTP",	"CS-Position")
Niben_Carbon_Chloro$Prediction <- factor(Niben_Carbon_Chloro$Prediction, levels = c("cTP","mTP","noTP"))
Niben_Carbon_Chloro <- merge(Niben_Carbon_Ann, Niben_Carbon_Chloro, by.x = "Protein", by.y = "ID")

Niben_Carbon_Chloro <- Niben_Carbon_Chloro[order(match(Niben_Carbon_Chloro$Protein, Niben_Carbon_Ann$Protein)),]
rownames(Niben_Carbon_Chloro) <- Niben_Carbon_Chloro$Protein
Niben_Carbon_Chloro <- Niben_Carbon_Chloro %>% select(Prediction)


## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_Carbon <- merge(Niben_Carbon_Ann, TPM_All_FC_Annotated, by="Name")

## We create the annotation for Modules

Niben_Carbon_Modules <- which_Module_pwd12(Niben_Carbon_Ann)

## We merge this annotation of Modules with the annotation of Chloroplast TP prediction

Niben_Carbon_Modules_Chl <- merge(Niben_Carbon_Modules, Niben_Carbon_Chloro, by = 0)
Niben_Carbon_Modules_Chl <- Niben_Carbon_Modules_Chl[order(match(Niben_Carbon_Modules_Chl$Row.names, Niben_Carbon_Ann$Protein)),]
rownames(Niben_Carbon_Modules_Chl) <- Niben_Carbon_Modules_Chl$Row.names
Niben_Carbon_Modules_Chl$Row.names <- NULL

## We select only the FC data from the different comparisons
Niben_Carbon_TPM <- Niben_Carbon %>% select(Protein, 4:51, 61:66)
Niben_Carbon_TPM <- Niben_Carbon_TPM[!duplicated(Niben_Carbon_TPM$Protein),]
Niben_Carbon_TPM <- Niben_Carbon_TPM[order(match(Niben_Carbon_TPM[,1], Niben_Carbon_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_Carbon_FC <- Niben_Carbon %>% select(Protein, 71, 73:87)
Niben_Carbon_FC <- Niben_Carbon_FC[!duplicated(Niben_Carbon_FC$Protein),]
Niben_Carbon_FC <- Niben_Carbon_FC[order(match(Niben_Carbon_FC[,1], Niben_Carbon_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_Carbon_DEG <- Niben_Carbon %>% select(Protein, 89, 91:105)
Niben_Carbon_DEG <- Niben_Carbon_DEG[!duplicated(Niben_Carbon_DEG$Protein),]
Niben_Carbon_DEG <- Niben_Carbon_DEG[order(match(Niben_Carbon_DEG[,1], Niben_Carbon_Ann[,2])),]

Niben_Carbon_DEG$pcrtB22h_vs_GFP22h_filtered <- ifelse(is.na(Niben_Carbon_DEG$pcrtB22h_vs_GFP22h_filtered) == TRUE, " ", "*")
Niben_Carbon_DEG$pcrtB25h_vs_GFP25h_filtered <- ifelse(is.na(Niben_Carbon_DEG$pcrtB25h_vs_GFP25h_filtered) == TRUE, " ", "*")
Niben_Carbon_DEG$pcrtBAA_vs_GFPAA_filtered <- ifelse(is.na(Niben_Carbon_DEG$pcrtBAA_vs_GFPAA_filtered) == TRUE, " ", "*")
Niben_Carbon_DEG$pcrtBBB_vs_GFPBB_filtered <- ifelse(is.na(Niben_Carbon_DEG$pcrtBBB_vs_GFPBB_filtered) == TRUE, " ", "*")
Niben_Carbon_DEG$pcrtBYY_vs_GFPYY_filtered <- ifelse(is.na(Niben_Carbon_DEG$pcrtBYY_vs_GFPYY_filtered) == TRUE, " ", "*")
Niben_Carbon_DEG$pcrtB28h_vs_GFP28h_filtered <- ifelse(is.na(Niben_Carbon_DEG$pcrtB28h_vs_GFP28h_filtered) == TRUE, " ", "*")
Niben_Carbon_DEG$pcrtB34h_vs_GFP34h_filtered <- ifelse(is.na(Niben_Carbon_DEG$pcrtB34h_vs_GFP34h_filtered) == TRUE, " ", "*")
Niben_Carbon_DEG$pcrtB37h_vs_GFP37h_filtered <- ifelse(is.na(Niben_Carbon_DEG$pcrtB37h_vs_GFP37h_filtered) == TRUE, " ", "*")
Niben_Carbon_DEG$pcrtBZZ_vs_GFPZZ_filtered <- ifelse(is.na(Niben_Carbon_DEG$pcrtBZZ_vs_GFPZZ_filtered) == TRUE, " ", "*")
Niben_Carbon_DEG$pcrtBCC_vs_GFPCC_filtered <- ifelse(is.na(Niben_Carbon_DEG$pcrtBCC_vs_GFPCC_filtered) == TRUE, " ", "*")
Niben_Carbon_DEG$pcrtB40h_vs_GFP40h_filtered <- ifelse(is.na(Niben_Carbon_DEG$pcrtB40h_vs_GFP40h_filtered) == TRUE, " ", "*")
Niben_Carbon_DEG$pcrtBDD_vs_GFPDD_filtered <- ifelse(is.na(Niben_Carbon_DEG$pcrtBDD_vs_GFPDD_filtered) == TRUE, " ", "*")
Niben_Carbon_DEG$pcrtB46h_vs_GFP46h_filtered <- ifelse(is.na(Niben_Carbon_DEG$pcrtB46h_vs_GFP46h_filtered) == TRUE, " ", "*")
Niben_Carbon_DEG$pcrtB56h_vs_GFP56h_filtered <- ifelse(is.na(Niben_Carbon_DEG$pcrtB56h_vs_GFP56h_filtered) == TRUE, " ", "*")
Niben_Carbon_DEG$pcrtBEE_vs_GFPEE_filtered <- ifelse(is.na(Niben_Carbon_DEG$pcrtBEE_vs_GFPEE_filtered) == TRUE, " ", "*")
Niben_Carbon_DEG$pcrtB96h_vs_GFP96h_filtered <- ifelse(is.na(Niben_Carbon_DEG$pcrtB96h_vs_GFP96h_filtered) == TRUE, " ", "*")
row.names(Niben_Carbon_DEG) <- Niben_Carbon_DEG$Protein
Niben_Carbon_DEG[1] <- NULL

Niben_Carbon_DEG <- Niben_Carbon_DEG %>%
  select(pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

row.names(Niben_Carbon_FC) <- Niben_Carbon_FC$Protein
Niben_Carbon_FC[1] <- NULL

Niben_Carbon_FC <- Niben_Carbon_FC %>%
  select(pcrtB22h_vs_GFP22h_All, pcrtBAA_vs_GFPAA_All, pcrtB25h_vs_GFP25h_All, 
         pcrtBYY_vs_GFPYY_All, pcrtBBB_vs_GFPBB_All, pcrtB28h_vs_GFP28h_All, pcrtB34h_vs_GFP34h_All, pcrtBZZ_vs_GFPZZ_All, 
         pcrtB37h_vs_GFP37h_All, pcrtBCC_vs_GFPCC_All, pcrtB40h_vs_GFP40h_All, pcrtBDD_vs_GFPDD_All, pcrtB46h_vs_GFP46h_All, 
         pcrtBEE_vs_GFPEE_All, pcrtB56h_vs_GFP56h_All, pcrtB96h_vs_GFP96h_All)


min(Niben_Carbon_FC, na.rm = TRUE)
max(Niben_Carbon_FC, na.rm = TRUE)
breaks_Carbon_FC=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Niben_Carbon_FC %>% select(1:15), 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =TRUE,
         breaks = breaks_Carbon_FC,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_Carbon_Annotation, annotation_colors=color_Carbon, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h"),#, "96h"),
         cex = 1,
         angle_col=45,
         main = "Carbon",
         annotation_legend = TRUE,
         display_numbers = Niben_Carbon_DEG %>% select(1:15),
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(84, 104),
         annotation_row = Niben_Carbon_Modules_Chl,
         annotation_colors = module_Chl_colors_2,
         cutree_rows = 7
         #gaps_col = c(15)
)


rownames(Niben_Carbon_TPM) <- Niben_Carbon_TPM$Protein
Niben_Carbon_TPM[1] <- NULL

min(Niben_Carbon_TPM, na.rm = TRUE)
max(Niben_Carbon_TPM, na.rm = TRUE)
breaks_Niben_Carbon_TPM=c(seq(0,100, length.out = 100))

Niben_Carbon_TPM_norm <- normalize_df(Niben_Carbon_TPM)
Niben_Carbon_TPM_norm_Cornell <- Niben_Carbon_TPM %>%
  select(1:48) 
Niben_Carbon_TPM_norm_Cornell <- normalize_df(Niben_Carbon_TPM_norm_Cornell)


pheatmap(Niben_Carbon_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Niben_Carbon_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_Carbon_Annotation, annotation_colors=color_Carbon, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Carbon",
         annotation_legend = TRUE,
         #display_numbers = Niben_Carbon_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         annotation_row = Niben_Carbon_Modules,
         annotation_colors = module_colors,
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(24, 48, 51)
)



#####Representing Heatmap for Sly

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Sly_Carbon_Ann <- read.table("Funcional Groups/FigTree/Sly_Carbon.txt",
                              sep = "\t", header = TRUE)

colnames(Sly_Carbon_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Sly_Carbon <- merge(Sly_Carbon_Ann, Summary_Yazdani_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Sly_Carbon_TPM <- Sly_Carbon %>% select(Protein, 4:66)
Sly_Carbon_TPM <- Sly_Carbon_TPM[!duplicated(Sly_Carbon_TPM$Protein),]
Sly_Carbon_TPM <- Sly_Carbon_TPM[order(match(Sly_Carbon_TPM[,1], Sly_Carbon_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_Carbon_FC <- Sly_Carbon %>% select(Protein, 67:71)
Sly_Carbon_FC <- Sly_Carbon_FC[!duplicated(Sly_Carbon_FC$Protein),]
Sly_Carbon_FC <- Sly_Carbon_FC[order(match(Sly_Carbon_FC[,1], Sly_Carbon_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_Carbon_DEG <- Sly_Carbon %>% select(Protein, 72:76)
Sly_Carbon_DEG <- Sly_Carbon_DEG[!duplicated(Sly_Carbon_DEG$Protein),]
Sly_Carbon_DEG <- Sly_Carbon_DEG[order(match(Sly_Carbon_DEG[,1], Sly_Carbon_Ann[,2])),]

Sly_Carbon_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Sly_Carbon_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Sly_Carbon_DEG$Ripening_Orange_DEG <- ifelse(is.na(Sly_Carbon_DEG$Ripening_Orange_DEG) == TRUE, " ", "*")
Sly_Carbon_DEG$Ripening_Red_DEG <- ifelse(is.na(Sly_Carbon_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
Sly_Carbon_DEG$ORHis_MG_DEG <- ifelse(is.na(Sly_Carbon_DEG$ORHis_MG_DEG) == TRUE, " ", "*")
Sly_Carbon_DEG$ORHis_Breaker_DEG <- ifelse(is.na(Sly_Carbon_DEG$ORHis_Breaker_DEG) == TRUE, " ", "*")
row.names(Sly_Carbon_DEG) <- Sly_Carbon_DEG$Protein
Sly_Carbon_DEG[1] <- NULL

rownames(Sly_Carbon_FC) <- Sly_Carbon_FC$Protein
Sly_Carbon_FC[1] <- NULL

min(Sly_Carbon_FC, na.rm = TRUE)
max(Sly_Carbon_FC, na.rm = TRUE)
breaks_Sly_Carbon=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Sly_Carbon_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Sly_Carbon,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_Carbon_Annotation, annotation_colors=color_Carbon, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Carbon",
         annotation_legend = TRUE,
         display_numbers = Sly_Carbon_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(39, 48),
         gaps_col = c(3)
)


rownames(Sly_Carbon_TPM) <- Sly_Carbon_TPM$Protein
Sly_Carbon_TPM[1] <- NULL

min(Sly_Carbon_TPM, na.rm = TRUE)
max(Sly_Carbon_TPM, na.rm = TRUE)
breaks_Sly_Carbon_TPM=c(seq(0,100, length.out = 200))

Sly_Carbon_TPM_norm <- normalize_df(Sly_Carbon_TPM)

pheatmap(Sly_Carbon_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Sly_Carbon_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_Carbon_Annotation, annotation_colors=color_Carbon, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Carbon",
         annotation_legend = TRUE,
         #display_numbers = Sly_Carbon_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(21,42)
)


#####Representing Heatmap for Cann

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Cann_Carbon_Ann <- read.table("Funcional Groups/FigTree/Cann_Carbon.txt",
                               sep = "\t", header = TRUE)

colnames(Cann_Carbon_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Cann_Carbon <- merge(Cann_Carbon_Ann, Summary_Pepper_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Cann_Carbon_TPM <- Cann_Carbon %>% select(Protein, 4:17)
Cann_Carbon_TPM <- Cann_Carbon_TPM[!duplicated(Cann_Carbon_TPM$Protein),]
Cann_Carbon_TPM <- Cann_Carbon_TPM[order(match(Cann_Carbon_TPM[,1], Cann_Carbon_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_Carbon_FC <- Cann_Carbon %>% select(Protein, 18:19)
Cann_Carbon_FC <- Cann_Carbon_FC[!duplicated(Cann_Carbon_FC$Protein),]
Cann_Carbon_FC <- Cann_Carbon_FC[order(match(Cann_Carbon_FC[,1], Cann_Carbon_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_Carbon_DEG <- Cann_Carbon %>% select(Protein, 20:21)
Cann_Carbon_DEG <- Cann_Carbon_DEG[!duplicated(Cann_Carbon_DEG$Protein),]
Cann_Carbon_DEG <- Cann_Carbon_DEG[order(match(Cann_Carbon_DEG[,1], Cann_Carbon_Ann[,2])),]

Cann_Carbon_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Cann_Carbon_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Cann_Carbon_DEG$Ripening_Red_DEG <- ifelse(is.na(Cann_Carbon_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
row.names(Cann_Carbon_DEG) <- Cann_Carbon_DEG$Protein
Cann_Carbon_DEG[1] <- NULL

rownames(Cann_Carbon_FC) <- Cann_Carbon_FC$Protein
Cann_Carbon_FC[1] <- NULL

min(Cann_Carbon_FC, na.rm = TRUE)
max(Cann_Carbon_FC, na.rm = TRUE)
breaks_Cann_Carbon=c(seq(-4.5,0, length.out = 100), seq(0.01,6,length.out = 100))

pheatmap(Cann_Carbon_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Cann_Carbon,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_Carbon_Annotation, annotation_colors=color_Carbon, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Red"),
         cex = 1,
         angle_col=45,
         main = "Carbon",
         annotation_legend = TRUE,
         display_numbers = Cann_Carbon_DEG,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)


rownames(Cann_Carbon_TPM) <- Cann_Carbon_TPM$Protein
Cann_Carbon_TPM[1] <- NULL

min(Cann_Carbon_TPM, na.rm = TRUE)
max(Cann_Carbon_TPM, na.rm = TRUE)
breaks_Cann_Carbon_TPM=c(seq(0,100, length.out = 200))

Cann_Carbon_TPM_norm <- normalize_df(Cann_Carbon_TPM)

pheatmap(Cann_Carbon_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Cann_Carbon_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_Carbon_Annotation, annotation_colors=color_Carbon, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Carbon",
         annotation_legend = TRUE,
         #display_numbers = Cann_Carbon_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(4,9)
)

#########################################################################################################################################
####################################  Calvin  #############################################################################################
#########################################################################################################################################

Calvin_Ath <- read.table("Funcional Groups/Ath_Calvin.txt", header = TRUE)
Calvin_Ath <- merge(Calvin_Ath, Ath_Gene_Models, by.x="ID", by.y=1)
Calvin_Ath_fasta <- Ath11_fasta[c(which(names(Ath11_fasta) %in% Calvin_Ath$V2))]
Calvin_Ath_2 <- Calvin_Ath %>% select(V2, Name)
write.fasta(Calvin_Ath_fasta, as.string=TRUE, names= names(Calvin_Ath_fasta),
            file.out = "Funcional Groups/Ath_Calvin.fasta")
rename.fasta(infile = "Funcional Groups/Ath_Calvin.fasta", Calvin_Ath_2, outfile = "Funcional Groups/Ath_Calvin_2.fasta")

Calvin_Niben_Candidates <- read.table("Funcional Groups/Niben_Candidates_Calvin.txt")
Calvin_Niben_Candidates <- Calvin_Niben_Candidates[Calvin_Niben_Candidates$V3>=35,] %>% select(V2)
Calvin_Niben_Candidates <- unique(Calvin_Niben_Candidates)
Calvin_Niben_Candidates_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Calvin_Niben_Candidates$V2))]
write.fasta(Calvin_Niben_Candidates_fasta, as.string=TRUE, names= names(Calvin_Niben_Candidates_fasta),
            file.out = "Funcional Groups/Calvin_Niben_Candidates_35.fasta")

Calvin_Sly_Candidates <- read.table("Funcional Groups/Sly_Candidates_Calvin.txt")
Calvin_Sly_Candidates <- Calvin_Sly_Candidates[Calvin_Sly_Candidates$V3>=35,] %>% select(V2)
Calvin_Sly_Candidates <- unique(Calvin_Sly_Candidates)
Calvin_Sly_Candidates_fasta <- Tomato_4.0_fasta[c(which(names(Tomato_4.0_fasta) %in% Calvin_Sly_Candidates$V2))]
write.fasta(Calvin_Sly_Candidates_fasta, as.string=TRUE, names= names(Calvin_Sly_Candidates_fasta),
            file.out = "Funcional Groups/Calvin_Sly_Candidates_35.fasta")

Calvin_Cann_Candidates <- read.table("Funcional Groups/Cann_Candidates_Calvin.txt")
Calvin_Cann_Candidates <- Calvin_Cann_Candidates[Calvin_Cann_Candidates$V3>=35,] %>% select(V2)
Calvin_Cann_Candidates <- unique(Calvin_Cann_Candidates)
Calvin_Cann_Candidates_fasta <- Cannuum_ASM51225v2_fasta[c(which(names(Cannuum_ASM51225v2_fasta) %in% Calvin_Cann_Candidates$V2))]
write.fasta(Calvin_Cann_Candidates_fasta, as.string=TRUE, names= names(Calvin_Cann_Candidates_fasta),
            file.out = "Funcional Groups/Calvin_Cann_Candidates_35.fasta")

Niben_Calvin_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_Calvin_Ann$Name))]
write.fasta(Niben_Calvin_fasta, as.string=TRUE, names= names(Niben_Calvin_fasta),
            file.out = "Funcional Groups/Niben_Calvin.fasta")
Niben_Calvin_Ann_2 <- Niben_Calvin_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_Calvin.fasta", Niben_Calvin_Ann, outfile = "Funcional Groups/Niben_Calvin_2.fasta")

save_heatmaps("Calvin", height = 6)

## Representing Heatmap

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
#Niben_Calvin_Ann <- read.table("Funcional Groups/FigTree/Niben_Calvin.txt",
#                                sep = "\t", header = TRUE)
Niben_Calvin_Ann <- Calvin_Niben_Candidates
Niben_Calvin_Ann$Protein <- paste("Calvin", seq(1, nrow(Niben_Calvin_Ann)), sep = "_")
Niben_Calvin_Descr <- merge(Niben_Calvin_Ann, TPM_All_Annotated, by= 1) %>% select(1, Description)

colnames(Niben_Calvin_Ann) <- c("Name", "Protein")
write.table(Niben_Calvin_Ann, file= "Funcional Groups/FigTree/Niben_Calvin.txt",
            sep = "\t", row.names = FALSE, quote = FALSE)

Niben_Calvin_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_Calvin_Ann$Name))]
write.fasta(Niben_Calvin_fasta, as.string=TRUE, names= names(Niben_Calvin_fasta),
            file.out = "Funcional Groups/Niben_Calvin.fasta")
Niben_Calvin_Ann_2 <- Niben_Calvin_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_Calvin.fasta", Niben_Calvin_Ann, outfile = "Funcional Groups/Niben_Calvin_2.fasta")

Niben_Calvin_Chloro <- read.delim("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan/Funcional Groups/Niben_Calvin_2_summary.targetp2", 
                                  header=FALSE, comment.char="#")
colnames(Niben_Calvin_Chloro) <- c("ID",	"Prediction",	"noTP",	"SP",	"mTP",	"cTP",	"luTP",	"CS-Position")
Niben_Calvin_Chloro$Prediction <- factor(Niben_Calvin_Chloro$Prediction, levels = c("cTP","mTP","noTP"))
Niben_Calvin_Chloro <- merge(Niben_Calvin_Ann, Niben_Calvin_Chloro, by.x = "Protein", by.y = "ID")

Niben_Calvin_Chloro <- Niben_Calvin_Chloro[order(match(Niben_Calvin_Chloro$Protein, Niben_Calvin_Ann$Protein)),]
rownames(Niben_Calvin_Chloro) <- Niben_Calvin_Chloro$Protein
Niben_Calvin_Chloro <- Niben_Calvin_Chloro %>% select(Prediction)


## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_Calvin <- merge(Niben_Calvin_Ann, TPM_All_FC_Annotated, by="Name")

## We create the annotation for Modules

Niben_Calvin_Modules <- which_Module_pwd12(Niben_Calvin_Ann)

## We merge this annotation of Modules with the annotation of Chloroplast TP prediction

Niben_Calvin_Modules_Chl <- merge(Niben_Calvin_Modules, Niben_Calvin_Chloro, by = 0)
Niben_Calvin_Modules_Chl <- Niben_Calvin_Modules_Chl[order(match(Niben_Calvin_Modules_Chl$Row.names, Niben_Calvin_Ann$Protein)),]
rownames(Niben_Calvin_Modules_Chl) <- Niben_Calvin_Modules_Chl$Row.names
Niben_Calvin_Modules_Chl$Row.names <- NULL

## We select only the FC data from the different comparisons
Niben_Calvin_TPM <- Niben_Calvin %>% select(Protein, 4:51, 61:66)
Niben_Calvin_TPM <- Niben_Calvin_TPM[!duplicated(Niben_Calvin_TPM$Protein),]
Niben_Calvin_TPM <- Niben_Calvin_TPM[order(match(Niben_Calvin_TPM[,1], Niben_Calvin_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_Calvin_FC <- Niben_Calvin %>% select(Protein, 71, 73:87)
Niben_Calvin_FC <- Niben_Calvin_FC[!duplicated(Niben_Calvin_FC$Protein),]
Niben_Calvin_FC <- Niben_Calvin_FC[order(match(Niben_Calvin_FC[,1], Niben_Calvin_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_Calvin_DEG <- Niben_Calvin %>% select(Protein, 89, 91:105)
Niben_Calvin_DEG <- Niben_Calvin_DEG[!duplicated(Niben_Calvin_DEG$Protein),]
Niben_Calvin_DEG <- Niben_Calvin_DEG[order(match(Niben_Calvin_DEG[,1], Niben_Calvin_Ann[,2])),]

Niben_Calvin_DEG$pcrtB22h_vs_GFP22h_filtered <- ifelse(is.na(Niben_Calvin_DEG$pcrtB22h_vs_GFP22h_filtered) == TRUE, " ", "*")
Niben_Calvin_DEG$pcrtB25h_vs_GFP25h_filtered <- ifelse(is.na(Niben_Calvin_DEG$pcrtB25h_vs_GFP25h_filtered) == TRUE, " ", "*")
Niben_Calvin_DEG$pcrtBAA_vs_GFPAA_filtered <- ifelse(is.na(Niben_Calvin_DEG$pcrtBAA_vs_GFPAA_filtered) == TRUE, " ", "*")
Niben_Calvin_DEG$pcrtBBB_vs_GFPBB_filtered <- ifelse(is.na(Niben_Calvin_DEG$pcrtBBB_vs_GFPBB_filtered) == TRUE, " ", "*")
Niben_Calvin_DEG$pcrtBYY_vs_GFPYY_filtered <- ifelse(is.na(Niben_Calvin_DEG$pcrtBYY_vs_GFPYY_filtered) == TRUE, " ", "*")
Niben_Calvin_DEG$pcrtB28h_vs_GFP28h_filtered <- ifelse(is.na(Niben_Calvin_DEG$pcrtB28h_vs_GFP28h_filtered) == TRUE, " ", "*")
Niben_Calvin_DEG$pcrtB34h_vs_GFP34h_filtered <- ifelse(is.na(Niben_Calvin_DEG$pcrtB34h_vs_GFP34h_filtered) == TRUE, " ", "*")
Niben_Calvin_DEG$pcrtB37h_vs_GFP37h_filtered <- ifelse(is.na(Niben_Calvin_DEG$pcrtB37h_vs_GFP37h_filtered) == TRUE, " ", "*")
Niben_Calvin_DEG$pcrtBZZ_vs_GFPZZ_filtered <- ifelse(is.na(Niben_Calvin_DEG$pcrtBZZ_vs_GFPZZ_filtered) == TRUE, " ", "*")
Niben_Calvin_DEG$pcrtBCC_vs_GFPCC_filtered <- ifelse(is.na(Niben_Calvin_DEG$pcrtBCC_vs_GFPCC_filtered) == TRUE, " ", "*")
Niben_Calvin_DEG$pcrtB40h_vs_GFP40h_filtered <- ifelse(is.na(Niben_Calvin_DEG$pcrtB40h_vs_GFP40h_filtered) == TRUE, " ", "*")
Niben_Calvin_DEG$pcrtBDD_vs_GFPDD_filtered <- ifelse(is.na(Niben_Calvin_DEG$pcrtBDD_vs_GFPDD_filtered) == TRUE, " ", "*")
Niben_Calvin_DEG$pcrtB46h_vs_GFP46h_filtered <- ifelse(is.na(Niben_Calvin_DEG$pcrtB46h_vs_GFP46h_filtered) == TRUE, " ", "*")
Niben_Calvin_DEG$pcrtB56h_vs_GFP56h_filtered <- ifelse(is.na(Niben_Calvin_DEG$pcrtB56h_vs_GFP56h_filtered) == TRUE, " ", "*")
Niben_Calvin_DEG$pcrtBEE_vs_GFPEE_filtered <- ifelse(is.na(Niben_Calvin_DEG$pcrtBEE_vs_GFPEE_filtered) == TRUE, " ", "*")
Niben_Calvin_DEG$pcrtB96h_vs_GFP96h_filtered <- ifelse(is.na(Niben_Calvin_DEG$pcrtB96h_vs_GFP96h_filtered) == TRUE, " ", "*")
row.names(Niben_Calvin_DEG) <- Niben_Calvin_DEG$Protein
Niben_Calvin_DEG[1] <- NULL

Niben_Calvin_DEG <- Niben_Calvin_DEG %>%
  select(pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

row.names(Niben_Calvin_FC) <- Niben_Calvin_FC$Protein
Niben_Calvin_FC[1] <- NULL

Niben_Calvin_FC <- Niben_Calvin_FC %>%
  select(pcrtB22h_vs_GFP22h_All, pcrtBAA_vs_GFPAA_All, pcrtB25h_vs_GFP25h_All, 
         pcrtBYY_vs_GFPYY_All, pcrtBBB_vs_GFPBB_All, pcrtB28h_vs_GFP28h_All, pcrtB34h_vs_GFP34h_All, pcrtBZZ_vs_GFPZZ_All, 
         pcrtB37h_vs_GFP37h_All, pcrtBCC_vs_GFPCC_All, pcrtB40h_vs_GFP40h_All, pcrtBDD_vs_GFPDD_All, pcrtB46h_vs_GFP46h_All, 
         pcrtBEE_vs_GFPEE_All, pcrtB56h_vs_GFP56h_All, pcrtB96h_vs_GFP96h_All)


min(Niben_Calvin_FC, na.rm = TRUE)
max(Niben_Calvin_FC, na.rm = TRUE)
breaks_Calvin_FC=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Niben_Calvin_FC %>% select(1:15), 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =TRUE,
         breaks = breaks_Calvin_FC,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_Calvin_Annotation, annotation_colors=color_Calvin, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h"),#, "96h"),
         cex = 1,
         angle_col=45,
         main = "Calvin",
         annotation_legend = TRUE,
         display_numbers = Niben_Calvin_DEG %>% select(1:15),
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(84, 104),
         annotation_row = Niben_Calvin_Modules_Chl,
         annotation_colors = module_Chl_colors_2,
         cutree_rows = 5
         #labels_row = Niben_Calvin_Descr$Description
         #gaps_col = c(15)
)


rownames(Niben_Calvin_TPM) <- Niben_Calvin_TPM$Protein
Niben_Calvin_TPM[1] <- NULL

min(Niben_Calvin_TPM, na.rm = TRUE)
max(Niben_Calvin_TPM, na.rm = TRUE)
breaks_Niben_Calvin_TPM=c(seq(0,100, length.out = 100))

Niben_Calvin_TPM_norm <- normalize_df(Niben_Calvin_TPM)
Niben_Calvin_TPM_norm_Cornell <- Niben_Calvin_TPM %>%
  select(1:48) 
Niben_Calvin_TPM_norm_Cornell <- normalize_df(Niben_Calvin_TPM_norm_Cornell)


pheatmap(Niben_Calvin_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Niben_Calvin_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_Calvin_Annotation, annotation_colors=color_Calvin, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Calvin",
         annotation_legend = TRUE,
         #display_numbers = Niben_Calvin_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         annotation_row = Niben_Calvin_Modules,
         annotation_colors = module_colors,
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(24, 48, 51)
)



#####Representing Heatmap for Sly

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Sly_Calvin_Ann <- read.table("Funcional Groups/FigTree/Sly_Calvin.txt",
                             sep = "\t", header = TRUE)

colnames(Sly_Calvin_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Sly_Calvin <- merge(Sly_Calvin_Ann, Summary_Yazdani_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Sly_Calvin_TPM <- Sly_Calvin %>% select(Protein, 4:66)
Sly_Calvin_TPM <- Sly_Calvin_TPM[!duplicated(Sly_Calvin_TPM$Protein),]
Sly_Calvin_TPM <- Sly_Calvin_TPM[order(match(Sly_Calvin_TPM[,1], Sly_Calvin_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_Calvin_FC <- Sly_Calvin %>% select(Protein, 67:71)
Sly_Calvin_FC <- Sly_Calvin_FC[!duplicated(Sly_Calvin_FC$Protein),]
Sly_Calvin_FC <- Sly_Calvin_FC[order(match(Sly_Calvin_FC[,1], Sly_Calvin_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_Calvin_DEG <- Sly_Calvin %>% select(Protein, 72:76)
Sly_Calvin_DEG <- Sly_Calvin_DEG[!duplicated(Sly_Calvin_DEG$Protein),]
Sly_Calvin_DEG <- Sly_Calvin_DEG[order(match(Sly_Calvin_DEG[,1], Sly_Calvin_Ann[,2])),]

Sly_Calvin_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Sly_Calvin_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Sly_Calvin_DEG$Ripening_Orange_DEG <- ifelse(is.na(Sly_Calvin_DEG$Ripening_Orange_DEG) == TRUE, " ", "*")
Sly_Calvin_DEG$Ripening_Red_DEG <- ifelse(is.na(Sly_Calvin_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
Sly_Calvin_DEG$ORHis_MG_DEG <- ifelse(is.na(Sly_Calvin_DEG$ORHis_MG_DEG) == TRUE, " ", "*")
Sly_Calvin_DEG$ORHis_Breaker_DEG <- ifelse(is.na(Sly_Calvin_DEG$ORHis_Breaker_DEG) == TRUE, " ", "*")
row.names(Sly_Calvin_DEG) <- Sly_Calvin_DEG$Protein
Sly_Calvin_DEG[1] <- NULL

rownames(Sly_Calvin_FC) <- Sly_Calvin_FC$Protein
Sly_Calvin_FC[1] <- NULL

min(Sly_Calvin_FC, na.rm = TRUE)
max(Sly_Calvin_FC, na.rm = TRUE)
breaks_Sly_Calvin=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Sly_Calvin_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Sly_Calvin,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_Calvin_Annotation, annotation_colors=color_Calvin, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Calvin",
         annotation_legend = TRUE,
         display_numbers = Sly_Calvin_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(39, 48),
         gaps_col = c(3)
)


rownames(Sly_Calvin_TPM) <- Sly_Calvin_TPM$Protein
Sly_Calvin_TPM[1] <- NULL

min(Sly_Calvin_TPM, na.rm = TRUE)
max(Sly_Calvin_TPM, na.rm = TRUE)
breaks_Sly_Calvin_TPM=c(seq(0,100, length.out = 200))

Sly_Calvin_TPM_norm <- normalize_df(Sly_Calvin_TPM)

pheatmap(Sly_Calvin_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Sly_Calvin_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_Calvin_Annotation, annotation_colors=color_Calvin, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Calvin",
         annotation_legend = TRUE,
         #display_numbers = Sly_Calvin_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(21,42)
)


#####Representing Heatmap for Cann

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Cann_Calvin_Ann <- read.table("Funcional Groups/FigTree/Cann_Calvin.txt",
                              sep = "\t", header = TRUE)

colnames(Cann_Calvin_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Cann_Calvin <- merge(Cann_Calvin_Ann, Summary_Pepper_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Cann_Calvin_TPM <- Cann_Calvin %>% select(Protein, 4:17)
Cann_Calvin_TPM <- Cann_Calvin_TPM[!duplicated(Cann_Calvin_TPM$Protein),]
Cann_Calvin_TPM <- Cann_Calvin_TPM[order(match(Cann_Calvin_TPM[,1], Cann_Calvin_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_Calvin_FC <- Cann_Calvin %>% select(Protein, 18:19)
Cann_Calvin_FC <- Cann_Calvin_FC[!duplicated(Cann_Calvin_FC$Protein),]
Cann_Calvin_FC <- Cann_Calvin_FC[order(match(Cann_Calvin_FC[,1], Cann_Calvin_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_Calvin_DEG <- Cann_Calvin %>% select(Protein, 20:21)
Cann_Calvin_DEG <- Cann_Calvin_DEG[!duplicated(Cann_Calvin_DEG$Protein),]
Cann_Calvin_DEG <- Cann_Calvin_DEG[order(match(Cann_Calvin_DEG[,1], Cann_Calvin_Ann[,2])),]

Cann_Calvin_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Cann_Calvin_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Cann_Calvin_DEG$Ripening_Red_DEG <- ifelse(is.na(Cann_Calvin_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
row.names(Cann_Calvin_DEG) <- Cann_Calvin_DEG$Protein
Cann_Calvin_DEG[1] <- NULL

rownames(Cann_Calvin_FC) <- Cann_Calvin_FC$Protein
Cann_Calvin_FC[1] <- NULL

min(Cann_Calvin_FC, na.rm = TRUE)
max(Cann_Calvin_FC, na.rm = TRUE)
breaks_Cann_Calvin=c(seq(-4.5,0, length.out = 100), seq(0.01,6,length.out = 100))

pheatmap(Cann_Calvin_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Cann_Calvin,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_Calvin_Annotation, annotation_colors=color_Calvin, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Red"),
         cex = 1,
         angle_col=45,
         main = "Calvin",
         annotation_legend = TRUE,
         display_numbers = Cann_Calvin_DEG,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)


rownames(Cann_Calvin_TPM) <- Cann_Calvin_TPM$Protein
Cann_Calvin_TPM[1] <- NULL

min(Cann_Calvin_TPM, na.rm = TRUE)
max(Cann_Calvin_TPM, na.rm = TRUE)
breaks_Cann_Calvin_TPM=c(seq(0,100, length.out = 200))

Cann_Calvin_TPM_norm <- normalize_df(Cann_Calvin_TPM)

pheatmap(Cann_Calvin_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Cann_Calvin_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_Calvin_Annotation, annotation_colors=color_Calvin, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Calvin",
         annotation_legend = TRUE,
         #display_numbers = Cann_Calvin_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(4,9)
)

#########################################################################################################################################
####################################  Photosynthesis  #############################################################################################
#########################################################################################################################################

Photosynthesis_Ath <- read.table("Funcional Groups/Ath_Photosynthesis.txt", header = TRUE)
Photosynthesis_Ath <- merge(Photosynthesis_Ath, Ath_Gene_Models, by.x="ID", by.y=1)
Photosynthesis_Ath_fasta <- Ath11_fasta[c(which(names(Ath11_fasta) %in% Photosynthesis_Ath$V2))]
Photosynthesis_Ath_2 <- Photosynthesis_Ath %>% select(V2, Name)
write.fasta(Photosynthesis_Ath_fasta, as.string=TRUE, names= names(Photosynthesis_Ath_fasta),
            file.out = "Funcional Groups/Ath_Photosynthesis.fasta")
rename.fasta(infile = "Funcional Groups/Ath_Photosynthesis.fasta", Photosynthesis_Ath_2, outfile = "Funcional Groups/Ath_Photosynthesis_2.fasta")

Photosynthesis_Niben_Candidates <- read.table("Funcional Groups/Niben_Candidates_Photosynthesis.txt")
Photosynthesis_Niben_Candidates <- Photosynthesis_Niben_Candidates[Photosynthesis_Niben_Candidates$V3>=35,] %>% select(V2)
Photosynthesis_Niben_Candidates <- unique(Photosynthesis_Niben_Candidates)
Photosynthesis_Niben_Candidates_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Photosynthesis_Niben_Candidates$V2))]
write.fasta(Photosynthesis_Niben_Candidates_fasta, as.string=TRUE, names= names(Photosynthesis_Niben_Candidates_fasta),
            file.out = "Funcional Groups/Photosynthesis_Niben_Candidates_35.fasta")

Photosynthesis_Sly_Candidates <- read.table("Funcional Groups/Sly_Candidates_Photosynthesis.txt")
Photosynthesis_Sly_Candidates <- Photosynthesis_Sly_Candidates[Photosynthesis_Sly_Candidates$V3>=35,] %>% select(V2)
Photosynthesis_Sly_Candidates <- unique(Photosynthesis_Sly_Candidates)
Photosynthesis_Sly_Candidates_fasta <- Tomato_4.0_fasta[c(which(names(Tomato_4.0_fasta) %in% Photosynthesis_Sly_Candidates$V2))]
write.fasta(Photosynthesis_Sly_Candidates_fasta, as.string=TRUE, names= names(Photosynthesis_Sly_Candidates_fasta),
            file.out = "Funcional Groups/Photosynthesis_Sly_Candidates_35.fasta")

Photosynthesis_Cann_Candidates <- read.table("Funcional Groups/Cann_Candidates_Photosynthesis.txt")
Photosynthesis_Cann_Candidates <- Photosynthesis_Cann_Candidates[Photosynthesis_Cann_Candidates$V3>=35,] %>% select(V2)
Photosynthesis_Cann_Candidates <- unique(Photosynthesis_Cann_Candidates)
Photosynthesis_Cann_Candidates_fasta <- Cannuum_ASM51225v2_fasta[c(which(names(Cannuum_ASM51225v2_fasta) %in% Photosynthesis_Cann_Candidates$V2))]
write.fasta(Photosynthesis_Cann_Candidates_fasta, as.string=TRUE, names= names(Photosynthesis_Cann_Candidates_fasta),
            file.out = "Funcional Groups/Photosynthesis_Cann_Candidates_35.fasta")

Niben_Photosynthesis_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_Photosynthesis_Ann$Name))]
write.fasta(Niben_Photosynthesis_fasta, as.string=TRUE, names= names(Niben_Photosynthesis_fasta),
            file.out = "Funcional Groups/Niben_Photosynthesis.fasta")
Niben_Photosynthesis_Ann_2 <- Niben_Photosynthesis_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_Photosynthesis.fasta", Niben_Photosynthesis_Ann, outfile = "Funcional Groups/Niben_Photosynthesis_2.fasta")

save_heatmaps("Photosynthesis", height = 6)

## Representing Heatmap

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
#Niben_Photosynthesis_Ann <- read.table("Funcional Groups/FigTree/Niben_Photosynthesis.txt",
#                                sep = "\t", header = TRUE)
Niben_Photosynthesis_Ann <- Photosynthesis_Niben_Candidates
Niben_Photosynthesis_Ann$Protein <- paste("Photosynthesis", seq(1, nrow(Niben_Photosynthesis_Ann)), sep = "_")
Niben_Photosynthesis_Descr <- merge(Niben_Photosynthesis_Ann, TPM_All_Annotated, by= 1) %>% select(1, Description)

colnames(Niben_Photosynthesis_Ann) <- c("Name", "Protein")
write.table(Niben_Photosynthesis_Ann, file= "Funcional Groups/FigTree/Niben_Photosynthesis.txt",
            sep = "\t", row.names = FALSE, quote = FALSE)

Niben_Photosynthesis_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_Photosynthesis_Ann$Name))]
write.fasta(Niben_Photosynthesis_fasta, as.string=TRUE, names= names(Niben_Photosynthesis_fasta),
            file.out = "Funcional Groups/Niben_Photosynthesis.fasta")
Niben_Photosynthesis_Ann_2 <- Niben_Photosynthesis_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_Photosynthesis.fasta", Niben_Photosynthesis_Ann, outfile = "Funcional Groups/Niben_Photosynthesis_2.fasta")

Niben_Photosynthesis_Chloro <- read.delim("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan/Funcional Groups/Niben_Photosynthesis_2_summary.targetp2", 
                                  header=FALSE, comment.char="#")
colnames(Niben_Photosynthesis_Chloro) <- c("ID",	"Prediction",	"noTP",	"SP",	"mTP",	"cTP",	"luTP",	"CS-Position")
Niben_Photosynthesis_Chloro$Prediction <- factor(Niben_Photosynthesis_Chloro$Prediction, levels = c("cTP","mTP","noTP"))
Niben_Photosynthesis_Chloro <- merge(Niben_Photosynthesis_Ann, Niben_Photosynthesis_Chloro, by.x = "Protein", by.y = "ID")

Niben_Photosynthesis_Chloro <- Niben_Photosynthesis_Chloro[order(match(Niben_Photosynthesis_Chloro$Protein, Niben_Photosynthesis_Ann$Protein)),]
rownames(Niben_Photosynthesis_Chloro) <- Niben_Photosynthesis_Chloro$Protein
Niben_Photosynthesis_Chloro <- Niben_Photosynthesis_Chloro %>% select(Prediction)


## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_Photosynthesis <- merge(Niben_Photosynthesis_Ann, TPM_All_FC_Annotated, by="Name")

## We create the annotation for Modules

Niben_Photosynthesis_Modules <- which_Module_pwd12(Niben_Photosynthesis_Ann)

## We merge this annotation of Modules with the annotation of Chloroplast TP prediction

Niben_Photosynthesis_Modules_Chl <- merge(Niben_Photosynthesis_Modules, Niben_Photosynthesis_Chloro, by = 0)
Niben_Photosynthesis_Modules_Chl <- Niben_Photosynthesis_Modules_Chl[order(match(Niben_Photosynthesis_Modules_Chl$Row.names, Niben_Photosynthesis_Ann$Protein)),]
rownames(Niben_Photosynthesis_Modules_Chl) <- Niben_Photosynthesis_Modules_Chl$Row.names
Niben_Photosynthesis_Modules_Chl$Row.names <- NULL

## We select only the FC data from the different comparisons
Niben_Photosynthesis_TPM <- Niben_Photosynthesis %>% select(Protein, 4:51, 61:66)
Niben_Photosynthesis_TPM <- Niben_Photosynthesis_TPM[!duplicated(Niben_Photosynthesis_TPM$Protein),]
Niben_Photosynthesis_TPM <- Niben_Photosynthesis_TPM[order(match(Niben_Photosynthesis_TPM[,1], Niben_Photosynthesis_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_Photosynthesis_FC <- Niben_Photosynthesis %>% select(Protein, 71, 73:87)
Niben_Photosynthesis_FC <- Niben_Photosynthesis_FC[!duplicated(Niben_Photosynthesis_FC$Protein),]
Niben_Photosynthesis_FC <- Niben_Photosynthesis_FC[order(match(Niben_Photosynthesis_FC[,1], Niben_Photosynthesis_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_Photosynthesis_DEG <- Niben_Photosynthesis %>% select(Protein, 89, 91:105)
Niben_Photosynthesis_DEG <- Niben_Photosynthesis_DEG[!duplicated(Niben_Photosynthesis_DEG$Protein),]
Niben_Photosynthesis_DEG <- Niben_Photosynthesis_DEG[order(match(Niben_Photosynthesis_DEG[,1], Niben_Photosynthesis_Ann[,2])),]

Niben_Photosynthesis_DEG$pcrtB22h_vs_GFP22h_filtered <- ifelse(is.na(Niben_Photosynthesis_DEG$pcrtB22h_vs_GFP22h_filtered) == TRUE, " ", "*")
Niben_Photosynthesis_DEG$pcrtB25h_vs_GFP25h_filtered <- ifelse(is.na(Niben_Photosynthesis_DEG$pcrtB25h_vs_GFP25h_filtered) == TRUE, " ", "*")
Niben_Photosynthesis_DEG$pcrtBAA_vs_GFPAA_filtered <- ifelse(is.na(Niben_Photosynthesis_DEG$pcrtBAA_vs_GFPAA_filtered) == TRUE, " ", "*")
Niben_Photosynthesis_DEG$pcrtBBB_vs_GFPBB_filtered <- ifelse(is.na(Niben_Photosynthesis_DEG$pcrtBBB_vs_GFPBB_filtered) == TRUE, " ", "*")
Niben_Photosynthesis_DEG$pcrtBYY_vs_GFPYY_filtered <- ifelse(is.na(Niben_Photosynthesis_DEG$pcrtBYY_vs_GFPYY_filtered) == TRUE, " ", "*")
Niben_Photosynthesis_DEG$pcrtB28h_vs_GFP28h_filtered <- ifelse(is.na(Niben_Photosynthesis_DEG$pcrtB28h_vs_GFP28h_filtered) == TRUE, " ", "*")
Niben_Photosynthesis_DEG$pcrtB34h_vs_GFP34h_filtered <- ifelse(is.na(Niben_Photosynthesis_DEG$pcrtB34h_vs_GFP34h_filtered) == TRUE, " ", "*")
Niben_Photosynthesis_DEG$pcrtB37h_vs_GFP37h_filtered <- ifelse(is.na(Niben_Photosynthesis_DEG$pcrtB37h_vs_GFP37h_filtered) == TRUE, " ", "*")
Niben_Photosynthesis_DEG$pcrtBZZ_vs_GFPZZ_filtered <- ifelse(is.na(Niben_Photosynthesis_DEG$pcrtBZZ_vs_GFPZZ_filtered) == TRUE, " ", "*")
Niben_Photosynthesis_DEG$pcrtBCC_vs_GFPCC_filtered <- ifelse(is.na(Niben_Photosynthesis_DEG$pcrtBCC_vs_GFPCC_filtered) == TRUE, " ", "*")
Niben_Photosynthesis_DEG$pcrtB40h_vs_GFP40h_filtered <- ifelse(is.na(Niben_Photosynthesis_DEG$pcrtB40h_vs_GFP40h_filtered) == TRUE, " ", "*")
Niben_Photosynthesis_DEG$pcrtBDD_vs_GFPDD_filtered <- ifelse(is.na(Niben_Photosynthesis_DEG$pcrtBDD_vs_GFPDD_filtered) == TRUE, " ", "*")
Niben_Photosynthesis_DEG$pcrtB46h_vs_GFP46h_filtered <- ifelse(is.na(Niben_Photosynthesis_DEG$pcrtB46h_vs_GFP46h_filtered) == TRUE, " ", "*")
Niben_Photosynthesis_DEG$pcrtB56h_vs_GFP56h_filtered <- ifelse(is.na(Niben_Photosynthesis_DEG$pcrtB56h_vs_GFP56h_filtered) == TRUE, " ", "*")
Niben_Photosynthesis_DEG$pcrtBEE_vs_GFPEE_filtered <- ifelse(is.na(Niben_Photosynthesis_DEG$pcrtBEE_vs_GFPEE_filtered) == TRUE, " ", "*")
Niben_Photosynthesis_DEG$pcrtB96h_vs_GFP96h_filtered <- ifelse(is.na(Niben_Photosynthesis_DEG$pcrtB96h_vs_GFP96h_filtered) == TRUE, " ", "*")
row.names(Niben_Photosynthesis_DEG) <- Niben_Photosynthesis_DEG$Protein
Niben_Photosynthesis_DEG[1] <- NULL

Niben_Photosynthesis_DEG <- Niben_Photosynthesis_DEG %>%
  select(pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

row.names(Niben_Photosynthesis_FC) <- Niben_Photosynthesis_FC$Protein
Niben_Photosynthesis_FC[1] <- NULL

Niben_Photosynthesis_FC <- Niben_Photosynthesis_FC %>%
  select(pcrtB22h_vs_GFP22h_All, pcrtBAA_vs_GFPAA_All, pcrtB25h_vs_GFP25h_All, 
         pcrtBYY_vs_GFPYY_All, pcrtBBB_vs_GFPBB_All, pcrtB28h_vs_GFP28h_All, pcrtB34h_vs_GFP34h_All, pcrtBZZ_vs_GFPZZ_All, 
         pcrtB37h_vs_GFP37h_All, pcrtBCC_vs_GFPCC_All, pcrtB40h_vs_GFP40h_All, pcrtBDD_vs_GFPDD_All, pcrtB46h_vs_GFP46h_All, 
         pcrtBEE_vs_GFPEE_All, pcrtB56h_vs_GFP56h_All, pcrtB96h_vs_GFP96h_All)


min(Niben_Photosynthesis_FC, na.rm = TRUE)
max(Niben_Photosynthesis_FC, na.rm = TRUE)
breaks_Photosynthesis_FC=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Niben_Photosynthesis_FC %>% select(1:15), 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =TRUE,
         breaks = breaks_Photosynthesis_FC,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_Photosynthesis_Annotation, annotation_colors=color_Photosynthesis, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h"),#, "96h"),
         cex = 1,
         angle_col=45,
         main = "Photosynthesis",
         annotation_legend = TRUE,
         display_numbers = Niben_Photosynthesis_DEG %>% select(1:15),
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(84, 104),
         annotation_row = Niben_Photosynthesis_Modules_Chl,
         annotation_colors = module_Chl_colors_2,
         labels_row = Niben_Photosynthesis_Descr$Description
         #gaps_col = c(15)
)


rownames(Niben_Photosynthesis_TPM) <- Niben_Photosynthesis_TPM$Protein
Niben_Photosynthesis_TPM[1] <- NULL

min(Niben_Photosynthesis_TPM, na.rm = TRUE)
max(Niben_Photosynthesis_TPM, na.rm = TRUE)
breaks_Niben_Photosynthesis_TPM=c(seq(0,100, length.out = 100))

Niben_Photosynthesis_TPM_norm <- normalize_df(Niben_Photosynthesis_TPM)
Niben_Photosynthesis_TPM_norm_Cornell <- Niben_Photosynthesis_TPM %>%
  select(1:48) 
Niben_Photosynthesis_TPM_norm_Cornell <- normalize_df(Niben_Photosynthesis_TPM_norm_Cornell)


pheatmap(Niben_Photosynthesis_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Niben_Photosynthesis_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_Photosynthesis_Annotation, annotation_colors=color_Photosynthesis, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Photosynthesis",
         annotation_legend = TRUE,
         #display_numbers = Niben_Photosynthesis_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         annotation_row = Niben_Photosynthesis_Modules,
         annotation_colors = module_colors,
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(24, 48, 51)
)



#####Representing Heatmap for Sly

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Sly_Photosynthesis_Ann <- read.table("Funcional Groups/FigTree/Sly_Photosynthesis.txt",
                             sep = "\t", header = TRUE)

colnames(Sly_Photosynthesis_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Sly_Photosynthesis <- merge(Sly_Photosynthesis_Ann, Summary_Yazdani_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Sly_Photosynthesis_TPM <- Sly_Photosynthesis %>% select(Protein, 4:66)
Sly_Photosynthesis_TPM <- Sly_Photosynthesis_TPM[!duplicated(Sly_Photosynthesis_TPM$Protein),]
Sly_Photosynthesis_TPM <- Sly_Photosynthesis_TPM[order(match(Sly_Photosynthesis_TPM[,1], Sly_Photosynthesis_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_Photosynthesis_FC <- Sly_Photosynthesis %>% select(Protein, 67:71)
Sly_Photosynthesis_FC <- Sly_Photosynthesis_FC[!duplicated(Sly_Photosynthesis_FC$Protein),]
Sly_Photosynthesis_FC <- Sly_Photosynthesis_FC[order(match(Sly_Photosynthesis_FC[,1], Sly_Photosynthesis_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_Photosynthesis_DEG <- Sly_Photosynthesis %>% select(Protein, 72:76)
Sly_Photosynthesis_DEG <- Sly_Photosynthesis_DEG[!duplicated(Sly_Photosynthesis_DEG$Protein),]
Sly_Photosynthesis_DEG <- Sly_Photosynthesis_DEG[order(match(Sly_Photosynthesis_DEG[,1], Sly_Photosynthesis_Ann[,2])),]

Sly_Photosynthesis_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Sly_Photosynthesis_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Sly_Photosynthesis_DEG$Ripening_Orange_DEG <- ifelse(is.na(Sly_Photosynthesis_DEG$Ripening_Orange_DEG) == TRUE, " ", "*")
Sly_Photosynthesis_DEG$Ripening_Red_DEG <- ifelse(is.na(Sly_Photosynthesis_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
Sly_Photosynthesis_DEG$ORHis_MG_DEG <- ifelse(is.na(Sly_Photosynthesis_DEG$ORHis_MG_DEG) == TRUE, " ", "*")
Sly_Photosynthesis_DEG$ORHis_Breaker_DEG <- ifelse(is.na(Sly_Photosynthesis_DEG$ORHis_Breaker_DEG) == TRUE, " ", "*")
row.names(Sly_Photosynthesis_DEG) <- Sly_Photosynthesis_DEG$Protein
Sly_Photosynthesis_DEG[1] <- NULL

rownames(Sly_Photosynthesis_FC) <- Sly_Photosynthesis_FC$Protein
Sly_Photosynthesis_FC[1] <- NULL

min(Sly_Photosynthesis_FC, na.rm = TRUE)
max(Sly_Photosynthesis_FC, na.rm = TRUE)
breaks_Sly_Photosynthesis=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Sly_Photosynthesis_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Sly_Photosynthesis,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_Photosynthesis_Annotation, annotation_colors=color_Photosynthesis, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Photosynthesis",
         annotation_legend = TRUE,
         display_numbers = Sly_Photosynthesis_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(39, 48),
         gaps_col = c(3)
)


rownames(Sly_Photosynthesis_TPM) <- Sly_Photosynthesis_TPM$Protein
Sly_Photosynthesis_TPM[1] <- NULL

min(Sly_Photosynthesis_TPM, na.rm = TRUE)
max(Sly_Photosynthesis_TPM, na.rm = TRUE)
breaks_Sly_Photosynthesis_TPM=c(seq(0,100, length.out = 200))

Sly_Photosynthesis_TPM_norm <- normalize_df(Sly_Photosynthesis_TPM)

pheatmap(Sly_Photosynthesis_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Sly_Photosynthesis_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_Photosynthesis_Annotation, annotation_colors=color_Photosynthesis, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Photosynthesis",
         annotation_legend = TRUE,
         #display_numbers = Sly_Photosynthesis_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(21,42)
)


#####Representing Heatmap for Cann

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Cann_Photosynthesis_Ann <- read.table("Funcional Groups/FigTree/Cann_Photosynthesis.txt",
                              sep = "\t", header = TRUE)

colnames(Cann_Photosynthesis_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Cann_Photosynthesis <- merge(Cann_Photosynthesis_Ann, Summary_Pepper_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Cann_Photosynthesis_TPM <- Cann_Photosynthesis %>% select(Protein, 4:17)
Cann_Photosynthesis_TPM <- Cann_Photosynthesis_TPM[!duplicated(Cann_Photosynthesis_TPM$Protein),]
Cann_Photosynthesis_TPM <- Cann_Photosynthesis_TPM[order(match(Cann_Photosynthesis_TPM[,1], Cann_Photosynthesis_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_Photosynthesis_FC <- Cann_Photosynthesis %>% select(Protein, 18:19)
Cann_Photosynthesis_FC <- Cann_Photosynthesis_FC[!duplicated(Cann_Photosynthesis_FC$Protein),]
Cann_Photosynthesis_FC <- Cann_Photosynthesis_FC[order(match(Cann_Photosynthesis_FC[,1], Cann_Photosynthesis_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_Photosynthesis_DEG <- Cann_Photosynthesis %>% select(Protein, 20:21)
Cann_Photosynthesis_DEG <- Cann_Photosynthesis_DEG[!duplicated(Cann_Photosynthesis_DEG$Protein),]
Cann_Photosynthesis_DEG <- Cann_Photosynthesis_DEG[order(match(Cann_Photosynthesis_DEG[,1], Cann_Photosynthesis_Ann[,2])),]

Cann_Photosynthesis_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Cann_Photosynthesis_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Cann_Photosynthesis_DEG$Ripening_Red_DEG <- ifelse(is.na(Cann_Photosynthesis_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
row.names(Cann_Photosynthesis_DEG) <- Cann_Photosynthesis_DEG$Protein
Cann_Photosynthesis_DEG[1] <- NULL

rownames(Cann_Photosynthesis_FC) <- Cann_Photosynthesis_FC$Protein
Cann_Photosynthesis_FC[1] <- NULL

min(Cann_Photosynthesis_FC, na.rm = TRUE)
max(Cann_Photosynthesis_FC, na.rm = TRUE)
breaks_Cann_Photosynthesis=c(seq(-4.5,0, length.out = 100), seq(0.01,6,length.out = 100))

pheatmap(Cann_Photosynthesis_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Cann_Photosynthesis,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_Photosynthesis_Annotation, annotation_colors=color_Photosynthesis, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Red"),
         cex = 1,
         angle_col=45,
         main = "Photosynthesis",
         annotation_legend = TRUE,
         display_numbers = Cann_Photosynthesis_DEG,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)


rownames(Cann_Photosynthesis_TPM) <- Cann_Photosynthesis_TPM$Protein
Cann_Photosynthesis_TPM[1] <- NULL

min(Cann_Photosynthesis_TPM, na.rm = TRUE)
max(Cann_Photosynthesis_TPM, na.rm = TRUE)
breaks_Cann_Photosynthesis_TPM=c(seq(0,100, length.out = 200))

Cann_Photosynthesis_TPM_norm <- normalize_df(Cann_Photosynthesis_TPM)

pheatmap(Cann_Photosynthesis_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Cann_Photosynthesis_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_Photosynthesis_Annotation, annotation_colors=color_Photosynthesis, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Photosynthesis",
         annotation_legend = TRUE,
         #display_numbers = Cann_Photosynthesis_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(4,9)
)

#########################################################################################################################################
####################################  Phenyl  #############################################################################################
#########################################################################################################################################

Phenyl_Ath <- read.table("Funcional Groups/Ath_Phenyl.txt", header = TRUE)
Phenyl_Ath <- merge(Phenyl_Ath, Ath_Gene_Models, by.x="ID", by.y=1)
Phenyl_Ath_fasta <- Ath11_fasta[c(which(names(Ath11_fasta) %in% Phenyl_Ath$V2))]
Phenyl_Ath_2 <- Phenyl_Ath %>% select(V2, Name)
write.fasta(Phenyl_Ath_fasta, as.string=TRUE, names= names(Phenyl_Ath_fasta),
            file.out = "Funcional Groups/Ath_Phenyl.fasta")
rename.fasta(infile = "Funcional Groups/Ath_Phenyl.fasta", Phenyl_Ath_2, outfile = "Funcional Groups/Ath_Phenyl_2.fasta")

Phenyl_Niben_Candidates <- read.table("Funcional Groups/Niben_Candidates_Phenyl.txt")
Phenyl_Niben_Candidates <- Phenyl_Niben_Candidates[Phenyl_Niben_Candidates$V3>=35,] %>% select(V2)
Phenyl_Niben_Candidates <- unique(Phenyl_Niben_Candidates)
Phenyl_Niben_Candidates_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Phenyl_Niben_Candidates$V2))]
write.fasta(Phenyl_Niben_Candidates_fasta, as.string=TRUE, names= names(Phenyl_Niben_Candidates_fasta),
            file.out = "Funcional Groups/Phenyl_Niben_Candidates_35.fasta")

Phenyl_Sly_Candidates <- read.table("Funcional Groups/Sly_Candidates_Phenyl.txt")
Phenyl_Sly_Candidates <- Phenyl_Sly_Candidates[Phenyl_Sly_Candidates$V3>=35,] %>% select(V2)
Phenyl_Sly_Candidates <- unique(Phenyl_Sly_Candidates)
Phenyl_Sly_Candidates_fasta <- Tomato_4.0_fasta[c(which(names(Tomato_4.0_fasta) %in% Phenyl_Sly_Candidates$V2))]
write.fasta(Phenyl_Sly_Candidates_fasta, as.string=TRUE, names= names(Phenyl_Sly_Candidates_fasta),
            file.out = "Funcional Groups/Phenyl_Sly_Candidates_35.fasta")

Phenyl_Cann_Candidates <- read.table("Funcional Groups/Cann_Candidates_Phenyl.txt")
Phenyl_Cann_Candidates <- Phenyl_Cann_Candidates[Phenyl_Cann_Candidates$V3>=35,] %>% select(V2)
Phenyl_Cann_Candidates <- unique(Phenyl_Cann_Candidates)
Phenyl_Cann_Candidates_fasta <- Cannuum_ASM51225v2_fasta[c(which(names(Cannuum_ASM51225v2_fasta) %in% Phenyl_Cann_Candidates$V2))]
write.fasta(Phenyl_Cann_Candidates_fasta, as.string=TRUE, names= names(Phenyl_Cann_Candidates_fasta),
            file.out = "Funcional Groups/Phenyl_Cann_Candidates_35.fasta")

Niben_Phenyl_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_Phenyl_Ann$Name))]
write.fasta(Niben_Phenyl_fasta, as.string=TRUE, names= names(Niben_Phenyl_fasta),
            file.out = "Funcional Groups/Niben_Phenyl.fasta")
Niben_Phenyl_Ann_2 <- Niben_Phenyl_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_Phenyl.fasta", Niben_Phenyl_Ann, outfile = "Funcional Groups/Niben_Phenyl_2.fasta")

save_heatmaps("Phenyl", height = 6)

## Representing Heatmap

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
#Niben_Phenyl_Ann <- read.table("Funcional Groups/FigTree/Niben_Phenyl.txt",
#                                sep = "\t", header = TRUE)
Niben_Phenyl_Ann <- Phenyl_Niben_Candidates
Niben_Phenyl_Ann$Protein <- paste("Phenyl", seq(1, nrow(Niben_Phenyl_Ann)), sep = "_")
Niben_Phenyl_Descr <- merge(Niben_Phenyl_Ann, TPM_All_Annotated, by= 1) %>% select(1, Description)

colnames(Niben_Phenyl_Ann) <- c("Name", "Protein")
write.table(Niben_Phenyl_Ann, file= "Funcional Groups/FigTree/Niben_Phenyl.txt",
            sep = "\t", row.names = FALSE, quote = FALSE)

Niben_Phenyl_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_Phenyl_Ann$Name))]
write.fasta(Niben_Phenyl_fasta, as.string=TRUE, names= names(Niben_Phenyl_fasta),
            file.out = "Funcional Groups/Niben_Phenyl.fasta")
Niben_Phenyl_Ann_2 <- Niben_Phenyl_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_Phenyl.fasta", Niben_Phenyl_Ann, outfile = "Funcional Groups/Niben_Phenyl_2.fasta")

Niben_Phenyl_Chloro <- read.delim("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan/Funcional Groups/Niben_Phenyl_2_summary.targetp2", 
                                          header=FALSE, comment.char="#")
colnames(Niben_Phenyl_Chloro) <- c("ID",	"Prediction",	"noTP",	"SP",	"mTP",	"cTP",	"luTP",	"CS-Position")
Niben_Phenyl_Chloro$Prediction <- factor(Niben_Phenyl_Chloro$Prediction, levels = c("cTP","mTP","noTP"))
Niben_Phenyl_Chloro <- merge(Niben_Phenyl_Ann, Niben_Phenyl_Chloro, by.x = "Protein", by.y = "ID")

Niben_Phenyl_Chloro <- Niben_Phenyl_Chloro[order(match(Niben_Phenyl_Chloro$Protein, Niben_Phenyl_Ann$Protein)),]
rownames(Niben_Phenyl_Chloro) <- Niben_Phenyl_Chloro$Protein
Niben_Phenyl_Chloro <- Niben_Phenyl_Chloro %>% select(Prediction)


## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_Phenyl <- merge(Niben_Phenyl_Ann, TPM_All_FC_Annotated, by="Name")

## We create the annotation for Modules

Niben_Phenyl_Modules <- which_Module_pwd12(Niben_Phenyl_Ann)

## We merge this annotation of Modules with the annotation of Chloroplast TP prediction

Niben_Phenyl_Modules_Chl <- merge(Niben_Phenyl_Modules, Niben_Phenyl_Chloro, by = 0)
Niben_Phenyl_Modules_Chl <- Niben_Phenyl_Modules_Chl[order(match(Niben_Phenyl_Modules_Chl$Row.names, Niben_Phenyl_Ann$Protein)),]
rownames(Niben_Phenyl_Modules_Chl) <- Niben_Phenyl_Modules_Chl$Row.names
Niben_Phenyl_Modules_Chl$Row.names <- NULL

## We select only the FC data from the different comparisons
Niben_Phenyl_TPM <- Niben_Phenyl %>% select(Protein, 4:51, 61:66)
Niben_Phenyl_TPM <- Niben_Phenyl_TPM[!duplicated(Niben_Phenyl_TPM$Protein),]
Niben_Phenyl_TPM <- Niben_Phenyl_TPM[order(match(Niben_Phenyl_TPM[,1], Niben_Phenyl_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_Phenyl_FC <- Niben_Phenyl %>% select(Protein, 71, 73:87)
Niben_Phenyl_FC <- Niben_Phenyl_FC[!duplicated(Niben_Phenyl_FC$Protein),]
Niben_Phenyl_FC <- Niben_Phenyl_FC[order(match(Niben_Phenyl_FC[,1], Niben_Phenyl_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_Phenyl_DEG <- Niben_Phenyl %>% select(Protein, 89, 91:105)
Niben_Phenyl_DEG <- Niben_Phenyl_DEG[!duplicated(Niben_Phenyl_DEG$Protein),]
Niben_Phenyl_DEG <- Niben_Phenyl_DEG[order(match(Niben_Phenyl_DEG[,1], Niben_Phenyl_Ann[,2])),]

Niben_Phenyl_DEG$pcrtB22h_vs_GFP22h_filtered <- ifelse(is.na(Niben_Phenyl_DEG$pcrtB22h_vs_GFP22h_filtered) == TRUE, " ", "*")
Niben_Phenyl_DEG$pcrtB25h_vs_GFP25h_filtered <- ifelse(is.na(Niben_Phenyl_DEG$pcrtB25h_vs_GFP25h_filtered) == TRUE, " ", "*")
Niben_Phenyl_DEG$pcrtBAA_vs_GFPAA_filtered <- ifelse(is.na(Niben_Phenyl_DEG$pcrtBAA_vs_GFPAA_filtered) == TRUE, " ", "*")
Niben_Phenyl_DEG$pcrtBBB_vs_GFPBB_filtered <- ifelse(is.na(Niben_Phenyl_DEG$pcrtBBB_vs_GFPBB_filtered) == TRUE, " ", "*")
Niben_Phenyl_DEG$pcrtBYY_vs_GFPYY_filtered <- ifelse(is.na(Niben_Phenyl_DEG$pcrtBYY_vs_GFPYY_filtered) == TRUE, " ", "*")
Niben_Phenyl_DEG$pcrtB28h_vs_GFP28h_filtered <- ifelse(is.na(Niben_Phenyl_DEG$pcrtB28h_vs_GFP28h_filtered) == TRUE, " ", "*")
Niben_Phenyl_DEG$pcrtB34h_vs_GFP34h_filtered <- ifelse(is.na(Niben_Phenyl_DEG$pcrtB34h_vs_GFP34h_filtered) == TRUE, " ", "*")
Niben_Phenyl_DEG$pcrtB37h_vs_GFP37h_filtered <- ifelse(is.na(Niben_Phenyl_DEG$pcrtB37h_vs_GFP37h_filtered) == TRUE, " ", "*")
Niben_Phenyl_DEG$pcrtBZZ_vs_GFPZZ_filtered <- ifelse(is.na(Niben_Phenyl_DEG$pcrtBZZ_vs_GFPZZ_filtered) == TRUE, " ", "*")
Niben_Phenyl_DEG$pcrtBCC_vs_GFPCC_filtered <- ifelse(is.na(Niben_Phenyl_DEG$pcrtBCC_vs_GFPCC_filtered) == TRUE, " ", "*")
Niben_Phenyl_DEG$pcrtB40h_vs_GFP40h_filtered <- ifelse(is.na(Niben_Phenyl_DEG$pcrtB40h_vs_GFP40h_filtered) == TRUE, " ", "*")
Niben_Phenyl_DEG$pcrtBDD_vs_GFPDD_filtered <- ifelse(is.na(Niben_Phenyl_DEG$pcrtBDD_vs_GFPDD_filtered) == TRUE, " ", "*")
Niben_Phenyl_DEG$pcrtB46h_vs_GFP46h_filtered <- ifelse(is.na(Niben_Phenyl_DEG$pcrtB46h_vs_GFP46h_filtered) == TRUE, " ", "*")
Niben_Phenyl_DEG$pcrtB56h_vs_GFP56h_filtered <- ifelse(is.na(Niben_Phenyl_DEG$pcrtB56h_vs_GFP56h_filtered) == TRUE, " ", "*")
Niben_Phenyl_DEG$pcrtBEE_vs_GFPEE_filtered <- ifelse(is.na(Niben_Phenyl_DEG$pcrtBEE_vs_GFPEE_filtered) == TRUE, " ", "*")
Niben_Phenyl_DEG$pcrtB96h_vs_GFP96h_filtered <- ifelse(is.na(Niben_Phenyl_DEG$pcrtB96h_vs_GFP96h_filtered) == TRUE, " ", "*")
row.names(Niben_Phenyl_DEG) <- Niben_Phenyl_DEG$Protein
Niben_Phenyl_DEG[1] <- NULL

Niben_Phenyl_DEG <- Niben_Phenyl_DEG %>%
  select(pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

row.names(Niben_Phenyl_FC) <- Niben_Phenyl_FC$Protein
Niben_Phenyl_FC[1] <- NULL

Niben_Phenyl_FC <- Niben_Phenyl_FC %>%
  select(pcrtB22h_vs_GFP22h_All, pcrtBAA_vs_GFPAA_All, pcrtB25h_vs_GFP25h_All, 
         pcrtBYY_vs_GFPYY_All, pcrtBBB_vs_GFPBB_All, pcrtB28h_vs_GFP28h_All, pcrtB34h_vs_GFP34h_All, pcrtBZZ_vs_GFPZZ_All, 
         pcrtB37h_vs_GFP37h_All, pcrtBCC_vs_GFPCC_All, pcrtB40h_vs_GFP40h_All, pcrtBDD_vs_GFPDD_All, pcrtB46h_vs_GFP46h_All, 
         pcrtBEE_vs_GFPEE_All, pcrtB56h_vs_GFP56h_All, pcrtB96h_vs_GFP96h_All)


min(Niben_Phenyl_FC, na.rm = TRUE)
max(Niben_Phenyl_FC, na.rm = TRUE)
breaks_Phenyl_FC=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Niben_Phenyl_FC %>% select(1:15), 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =TRUE,
         breaks = breaks_Phenyl_FC,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_Phenyl_Annotation, annotation_colors=color_Phenyl, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h"),#, "96h"),
         cex = 1,
         angle_col=45,
         main = "Phenyl",
         annotation_legend = TRUE,
         display_numbers = Niben_Phenyl_DEG %>% select(1:15),
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(84, 104),
         annotation_row = Niben_Phenyl_Modules_Chl,
         annotation_colors = module_Chl_colors_2,
         #labels_row = Niben_Phenyl_Descr$Description
         #gaps_col = c(15)
)


rownames(Niben_Phenyl_TPM) <- Niben_Phenyl_TPM$Protein
Niben_Phenyl_TPM[1] <- NULL

min(Niben_Phenyl_TPM, na.rm = TRUE)
max(Niben_Phenyl_TPM, na.rm = TRUE)
breaks_Niben_Phenyl_TPM=c(seq(0,100, length.out = 100))

Niben_Phenyl_TPM_norm <- normalize_df(Niben_Phenyl_TPM)
Niben_Phenyl_TPM_norm_Cornell <- Niben_Phenyl_TPM %>%
  select(1:48) 
Niben_Phenyl_TPM_norm_Cornell <- normalize_df(Niben_Phenyl_TPM_norm_Cornell)


pheatmap(Niben_Phenyl_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Niben_Phenyl_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_Phenyl_Annotation, annotation_colors=color_Phenyl, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Phenyl",
         annotation_legend = TRUE,
         #display_numbers = Niben_Phenyl_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         annotation_row = Niben_Phenyl_Modules,
         annotation_colors = module_colors,
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(24, 48, 51)
)



#####Representing Heatmap for Sly

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Sly_Phenyl_Ann <- read.table("Funcional Groups/FigTree/Sly_Phenyl.txt",
                                     sep = "\t", header = TRUE)

colnames(Sly_Phenyl_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Sly_Phenyl <- merge(Sly_Phenyl_Ann, Summary_Yazdani_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Sly_Phenyl_TPM <- Sly_Phenyl %>% select(Protein, 4:66)
Sly_Phenyl_TPM <- Sly_Phenyl_TPM[!duplicated(Sly_Phenyl_TPM$Protein),]
Sly_Phenyl_TPM <- Sly_Phenyl_TPM[order(match(Sly_Phenyl_TPM[,1], Sly_Phenyl_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_Phenyl_FC <- Sly_Phenyl %>% select(Protein, 67:71)
Sly_Phenyl_FC <- Sly_Phenyl_FC[!duplicated(Sly_Phenyl_FC$Protein),]
Sly_Phenyl_FC <- Sly_Phenyl_FC[order(match(Sly_Phenyl_FC[,1], Sly_Phenyl_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_Phenyl_DEG <- Sly_Phenyl %>% select(Protein, 72:76)
Sly_Phenyl_DEG <- Sly_Phenyl_DEG[!duplicated(Sly_Phenyl_DEG$Protein),]
Sly_Phenyl_DEG <- Sly_Phenyl_DEG[order(match(Sly_Phenyl_DEG[,1], Sly_Phenyl_Ann[,2])),]

Sly_Phenyl_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Sly_Phenyl_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Sly_Phenyl_DEG$Ripening_Orange_DEG <- ifelse(is.na(Sly_Phenyl_DEG$Ripening_Orange_DEG) == TRUE, " ", "*")
Sly_Phenyl_DEG$Ripening_Red_DEG <- ifelse(is.na(Sly_Phenyl_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
Sly_Phenyl_DEG$ORHis_MG_DEG <- ifelse(is.na(Sly_Phenyl_DEG$ORHis_MG_DEG) == TRUE, " ", "*")
Sly_Phenyl_DEG$ORHis_Breaker_DEG <- ifelse(is.na(Sly_Phenyl_DEG$ORHis_Breaker_DEG) == TRUE, " ", "*")
row.names(Sly_Phenyl_DEG) <- Sly_Phenyl_DEG$Protein
Sly_Phenyl_DEG[1] <- NULL

rownames(Sly_Phenyl_FC) <- Sly_Phenyl_FC$Protein
Sly_Phenyl_FC[1] <- NULL

min(Sly_Phenyl_FC, na.rm = TRUE)
max(Sly_Phenyl_FC, na.rm = TRUE)
breaks_Sly_Phenyl=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Sly_Phenyl_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Sly_Phenyl,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_Phenyl_Annotation, annotation_colors=color_Phenyl, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Phenyl",
         annotation_legend = TRUE,
         display_numbers = Sly_Phenyl_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(39, 48),
         gaps_col = c(3)
)


rownames(Sly_Phenyl_TPM) <- Sly_Phenyl_TPM$Protein
Sly_Phenyl_TPM[1] <- NULL

min(Sly_Phenyl_TPM, na.rm = TRUE)
max(Sly_Phenyl_TPM, na.rm = TRUE)
breaks_Sly_Phenyl_TPM=c(seq(0,100, length.out = 200))

Sly_Phenyl_TPM_norm <- normalize_df(Sly_Phenyl_TPM)

pheatmap(Sly_Phenyl_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Sly_Phenyl_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_Phenyl_Annotation, annotation_colors=color_Phenyl, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Phenyl",
         annotation_legend = TRUE,
         #display_numbers = Sly_Phenyl_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(21,42)
)


#####Representing Heatmap for Cann

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Cann_Phenyl_Ann <- read.table("Funcional Groups/FigTree/Cann_Phenyl.txt",
                                      sep = "\t", header = TRUE)

colnames(Cann_Phenyl_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Cann_Phenyl <- merge(Cann_Phenyl_Ann, Summary_Pepper_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Cann_Phenyl_TPM <- Cann_Phenyl %>% select(Protein, 4:17)
Cann_Phenyl_TPM <- Cann_Phenyl_TPM[!duplicated(Cann_Phenyl_TPM$Protein),]
Cann_Phenyl_TPM <- Cann_Phenyl_TPM[order(match(Cann_Phenyl_TPM[,1], Cann_Phenyl_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_Phenyl_FC <- Cann_Phenyl %>% select(Protein, 18:19)
Cann_Phenyl_FC <- Cann_Phenyl_FC[!duplicated(Cann_Phenyl_FC$Protein),]
Cann_Phenyl_FC <- Cann_Phenyl_FC[order(match(Cann_Phenyl_FC[,1], Cann_Phenyl_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_Phenyl_DEG <- Cann_Phenyl %>% select(Protein, 20:21)
Cann_Phenyl_DEG <- Cann_Phenyl_DEG[!duplicated(Cann_Phenyl_DEG$Protein),]
Cann_Phenyl_DEG <- Cann_Phenyl_DEG[order(match(Cann_Phenyl_DEG[,1], Cann_Phenyl_Ann[,2])),]

Cann_Phenyl_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Cann_Phenyl_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Cann_Phenyl_DEG$Ripening_Red_DEG <- ifelse(is.na(Cann_Phenyl_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
row.names(Cann_Phenyl_DEG) <- Cann_Phenyl_DEG$Protein
Cann_Phenyl_DEG[1] <- NULL

rownames(Cann_Phenyl_FC) <- Cann_Phenyl_FC$Protein
Cann_Phenyl_FC[1] <- NULL

min(Cann_Phenyl_FC, na.rm = TRUE)
max(Cann_Phenyl_FC, na.rm = TRUE)
breaks_Cann_Phenyl=c(seq(-4.5,0, length.out = 100), seq(0.01,6,length.out = 100))

pheatmap(Cann_Phenyl_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Cann_Phenyl,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_Phenyl_Annotation, annotation_colors=color_Phenyl, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Red"),
         cex = 1,
         angle_col=45,
         main = "Phenyl",
         annotation_legend = TRUE,
         display_numbers = Cann_Phenyl_DEG,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)


rownames(Cann_Phenyl_TPM) <- Cann_Phenyl_TPM$Protein
Cann_Phenyl_TPM[1] <- NULL

min(Cann_Phenyl_TPM, na.rm = TRUE)
max(Cann_Phenyl_TPM, na.rm = TRUE)
breaks_Cann_Phenyl_TPM=c(seq(0,100, length.out = 200))

Cann_Phenyl_TPM_norm <- normalize_df(Cann_Phenyl_TPM)

pheatmap(Cann_Phenyl_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Cann_Phenyl_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_Phenyl_Annotation, annotation_colors=color_Phenyl, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Phenyl",
         annotation_legend = TRUE,
         #display_numbers = Cann_Phenyl_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(4,9)
)



#########################################################################################################################################
####################################  Isoprenoids  #############################################################################################
#########################################################################################################################################


## Representing Heatmap

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Niben_Isoprenoids_Ann <- read.table("Funcional Groups/FigTree/Niben_Isoprenoids.txt",
                            sep = "\t", header = TRUE)

colnames(Niben_Isoprenoids_Ann) <- c("Name", "Protein", "Family")

Niben_Isoprenoids_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_Isoprenoids_Ann$Name))]
write.fasta(Niben_Isoprenoids_fasta, as.string=TRUE, names= names(Niben_Isoprenoids_fasta),
            file.out = "Funcional Groups/Niben_Isoprenoids.fasta")
Niben_Isoprenoids_Ann_2 <- Niben_Isoprenoids_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_Isoprenoids.fasta", Niben_Isoprenoids_Ann_2, outfile = "Funcional Groups/Niben_Isoprenoids_2.fasta")

Niben_Isoprenoids_Chloro <- read.table("Funcional Groups/Isoprenoids_TransitPep.txt",
                                       sep = "\t", header = TRUE)
Niben_Isoprenoids_Chloro$Prediction <- factor(Niben_Isoprenoids_Chloro$Prediction, levels = c("cTP", "OTHER", "mTP"), labels = c("Yes", "No", "No"))
colnames(Niben_Isoprenoids_Chloro)[2] <- "Chloroplast"
Niben_Isoprenoids_Chloro <- merge(Niben_Isoprenoids_Ann, Niben_Isoprenoids_Chloro, by.x = "Protein", by.y = "Name")

Niben_Isoprenoids_Ann <- Niben_Isoprenoids_Ann %>% select(Name, Protein)

Niben_Isoprenoids_Chloro$Family <- factor(Niben_Isoprenoids_Chloro$Family, levels = c("MEP", "GGPPS", "MVA", "IPPI", "FPPS"))
Niben_Isoprenoids_Chloro <- Niben_Isoprenoids_Chloro[order(match(Niben_Isoprenoids_Chloro$Protein, Niben_Isoprenoids_Ann$Protein)),]
rownames(Niben_Isoprenoids_Chloro) <- Niben_Isoprenoids_Chloro$Protein
Niben_Isoprenoids_Chloro$Name <- NULL
Niben_Isoprenoids_Chloro$Protein <- NULL


Isoprenoid_colors <- list(
  Family= c(MEP = "#f46d43",
            GGPPS = "#fdae61",
            MVA="#4575b4",
            IPPI = "#74add1",
            FPPS = "#abd9e9"),
  Chloroplast= c(Yes= "#a6d96a",
                 No= "#ffffbf")
)


## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_Isoprenoids <- merge(Niben_Isoprenoids_Ann, TPM_All_FC_Annotated, by="Name")

## We create the annotation for Modules

Niben_Isoprenoids_Modules <- which_Module_pwd30(Niben_Isoprenoids_Ann)

## We select only the FC data from the different comparisons
Niben_Isoprenoids_TPM <- Niben_Isoprenoids %>% select(Protein, 4:51, 61:66)
Niben_Isoprenoids_TPM <- Niben_Isoprenoids_TPM[!duplicated(Niben_Isoprenoids_TPM$Protein),]
Niben_Isoprenoids_TPM <- Niben_Isoprenoids_TPM[order(match(Niben_Isoprenoids_TPM[,1], Niben_Isoprenoids_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_Isoprenoids_FC <- Niben_Isoprenoids %>% select(Protein, 71, 73:87)
Niben_Isoprenoids_FC <- Niben_Isoprenoids_FC[!duplicated(Niben_Isoprenoids_FC$Protein),]
Niben_Isoprenoids_FC <- Niben_Isoprenoids_FC[order(match(Niben_Isoprenoids_FC[,1], Niben_Isoprenoids_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_Isoprenoids_DEG <- Niben_Isoprenoids %>% select(Protein, 89, 91:105)
Niben_Isoprenoids_DEG <- Niben_Isoprenoids_DEG[!duplicated(Niben_Isoprenoids_DEG$Protein),]
Niben_Isoprenoids_DEG <- Niben_Isoprenoids_DEG[order(match(Niben_Isoprenoids_DEG[,1], Niben_Isoprenoids_Ann[,2])),]

Niben_Isoprenoids_DEG$pcrtB22h_vs_GFP22h_filtered <- ifelse(is.na(Niben_Isoprenoids_DEG$pcrtB22h_vs_GFP22h_filtered) == TRUE, " ", "*")
Niben_Isoprenoids_DEG$pcrtB25h_vs_GFP25h_filtered <- ifelse(is.na(Niben_Isoprenoids_DEG$pcrtB25h_vs_GFP25h_filtered) == TRUE, " ", "*")
Niben_Isoprenoids_DEG$pcrtBAA_vs_GFPAA_filtered <- ifelse(is.na(Niben_Isoprenoids_DEG$pcrtBAA_vs_GFPAA_filtered) == TRUE, " ", "*")
Niben_Isoprenoids_DEG$pcrtBBB_vs_GFPBB_filtered <- ifelse(is.na(Niben_Isoprenoids_DEG$pcrtBBB_vs_GFPBB_filtered) == TRUE, " ", "*")
Niben_Isoprenoids_DEG$pcrtBYY_vs_GFPYY_filtered <- ifelse(is.na(Niben_Isoprenoids_DEG$pcrtBYY_vs_GFPYY_filtered) == TRUE, " ", "*")
Niben_Isoprenoids_DEG$pcrtB28h_vs_GFP28h_filtered <- ifelse(is.na(Niben_Isoprenoids_DEG$pcrtB28h_vs_GFP28h_filtered) == TRUE, " ", "*")
Niben_Isoprenoids_DEG$pcrtB34h_vs_GFP34h_filtered <- ifelse(is.na(Niben_Isoprenoids_DEG$pcrtB34h_vs_GFP34h_filtered) == TRUE, " ", "*")
Niben_Isoprenoids_DEG$pcrtB37h_vs_GFP37h_filtered <- ifelse(is.na(Niben_Isoprenoids_DEG$pcrtB37h_vs_GFP37h_filtered) == TRUE, " ", "*")
Niben_Isoprenoids_DEG$pcrtBZZ_vs_GFPZZ_filtered <- ifelse(is.na(Niben_Isoprenoids_DEG$pcrtBZZ_vs_GFPZZ_filtered) == TRUE, " ", "*")
Niben_Isoprenoids_DEG$pcrtBCC_vs_GFPCC_filtered <- ifelse(is.na(Niben_Isoprenoids_DEG$pcrtBCC_vs_GFPCC_filtered) == TRUE, " ", "*")
Niben_Isoprenoids_DEG$pcrtB40h_vs_GFP40h_filtered <- ifelse(is.na(Niben_Isoprenoids_DEG$pcrtB40h_vs_GFP40h_filtered) == TRUE, " ", "*")
Niben_Isoprenoids_DEG$pcrtBDD_vs_GFPDD_filtered <- ifelse(is.na(Niben_Isoprenoids_DEG$pcrtBDD_vs_GFPDD_filtered) == TRUE, " ", "*")
Niben_Isoprenoids_DEG$pcrtB46h_vs_GFP46h_filtered <- ifelse(is.na(Niben_Isoprenoids_DEG$pcrtB46h_vs_GFP46h_filtered) == TRUE, " ", "*")
Niben_Isoprenoids_DEG$pcrtB56h_vs_GFP56h_filtered <- ifelse(is.na(Niben_Isoprenoids_DEG$pcrtB56h_vs_GFP56h_filtered) == TRUE, " ", "*")
Niben_Isoprenoids_DEG$pcrtBEE_vs_GFPEE_filtered <- ifelse(is.na(Niben_Isoprenoids_DEG$pcrtBEE_vs_GFPEE_filtered) == TRUE, " ", "*")
Niben_Isoprenoids_DEG$pcrtB96h_vs_GFP96h_filtered <- ifelse(is.na(Niben_Isoprenoids_DEG$pcrtB96h_vs_GFP96h_filtered) == TRUE, " ", "*")
row.names(Niben_Isoprenoids_DEG) <- Niben_Isoprenoids_DEG$Protein
Niben_Isoprenoids_DEG[1] <- NULL

Niben_Isoprenoids_DEG <- Niben_Isoprenoids_DEG %>%
  select(pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

row.names(Niben_Isoprenoids_FC) <- Niben_Isoprenoids_FC$Protein
Niben_Isoprenoids_FC[1] <- NULL

Niben_Isoprenoids_FC <- Niben_Isoprenoids_FC %>%
  select(pcrtB22h_vs_GFP22h_All, pcrtBAA_vs_GFPAA_All, pcrtB25h_vs_GFP25h_All, 
         pcrtBYY_vs_GFPYY_All, pcrtBBB_vs_GFPBB_All, pcrtB28h_vs_GFP28h_All, pcrtB34h_vs_GFP34h_All, pcrtBZZ_vs_GFPZZ_All, 
         pcrtB37h_vs_GFP37h_All, pcrtBCC_vs_GFPCC_All, pcrtB40h_vs_GFP40h_All, pcrtBDD_vs_GFPDD_All, pcrtB46h_vs_GFP46h_All, 
         pcrtBEE_vs_GFPEE_All, pcrtB56h_vs_GFP56h_All, pcrtB96h_vs_GFP96h_All)


min(Niben_Isoprenoids_FC, na.rm = TRUE)
max(Niben_Isoprenoids_FC, na.rm = TRUE)
breaks_Isoprenoids_FC=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Niben_Isoprenoids_FC %>% select(1, 3, 5, 7, 9, 11, 13, 15), 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Isoprenoids_FC,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_Isoprenoids_Annotation, annotation_colors=color_Isoprenoids, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("22h", "25h",  "28h", "34h","37h", "40h","46h", "56h"),# "96h"),
         cex = 1,
         angle_col=45,
         main = "Isoprenoid-Precursors Biosynthesis",
         annotation_legend = TRUE,
         display_numbers = Niben_Isoprenoids_DEG %>% select(1, 3, 5, 7, 9, 11, 13, 15),
         fontsize_number = 15,
         na_col = "darkgrey",
         gaps_row = c(20, 26, 48, 52),
         annotation_row = Niben_Isoprenoids_Chloro,
         annotation_colors = Isoprenoid_colors
         #gaps_col = c(15)
)


rownames(Niben_Isoprenoids_TPM) <- Niben_Isoprenoids_TPM$Protein
Niben_Isoprenoids_TPM[1] <- NULL

min(Niben_Isoprenoids_TPM, na.rm = TRUE)
max(Niben_Isoprenoids_TPM, na.rm = TRUE)
breaks_Niben_Isoprenoids_TPM=c(seq(0,100, length.out = 100))

Niben_Isoprenoids_TPM_norm <- normalize_df(Niben_Isoprenoids_TPM)
Niben_Isoprenoids_TPM_norm_Cornell <- Niben_Isoprenoids_TPM %>%
  select(1:48) 
Niben_Isoprenoids_TPM_norm_Cornell <- normalize_df(Niben_Isoprenoids_TPM_norm_Cornell)


pheatmap(Niben_Isoprenoids_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Niben_Isoprenoids_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_Isoprenoids_Annotation, annotation_colors=color_Isoprenoids, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Isoprenoids",
         annotation_legend = TRUE,
         #display_numbers = Niben_Isoprenoids_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         annotation_row = Niben_Isoprenoids_Modules,
         annotation_colors = module_colors,
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(24, 48, 51)
)



#########################################################################################
###### ARGONATUTES


setwd("/Users/salva/GoogleDrive/My Drive/CRAG/Bologna")

Argo_Rodriguez_Proteins <- seqinr::read.fasta("AGO_Protein-seqs.fasta", seqtype = "AA", as.string = TRUE)
ARGO1_ortologs <- read.table("AGO1_RodriguezLeal.txt", sep= "\t", header = TRUE)

ARGO1_ortologs_fasta <- Argo_Rodriguez_Proteins[c(which(names(Argo_Rodriguez_Proteins) %in% ARGO1_ortologs$AGO1))]


Niben_PSY_Ann <- Niben_Carotenoids_Ann %>% filter(Protein %in% grep("PSY", Niben_Carotenoids_Ann$Protein, value= TRUE))
crtB_Ann <- data.frame(Name=grep("crtB", names(Niben261_fasta), value = TRUE),
                       Protein= "crtB")
Niben_PSY_Ann <- rbind(Niben_PSY_Ann, crtB_Ann)
Niben_PSY_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_PSY_Ann$Name))]
write.fasta(Niben_PSY_fasta, as.string=TRUE, names= names(Niben_PSY_fasta),
            file.out = "Funcional Groups/Niben_PSY.fasta")
rename.fasta(infile = "Funcional Groups/Niben_PSY.fasta", Niben_PSY_Ann, outfile = "Funcional Groups/Niben_PSY_2.fasta")

Sly_PSY_Ann <- Sly_Carotenoids_Ann %>% filter(Protein %in% grep("PSY", Sly_Carotenoids_Ann$Protein, value= TRUE))
Sly_PSY_fasta <- Tomato_4.0_fasta[c(which(names(Tomato_4.0_fasta) %in% Sly_PSY_Ann$Name))]
write.fasta(Sly_PSY_fasta, as.string=TRUE, names= names(Sly_PSY_fasta),
            file.out = "Funcional Groups/Sly_PSY.fasta")
rename.fasta(infile = "Funcional Groups/Sly_PSY.fasta", Sly_PSY_Ann, outfile = "Funcional Groups/Sly_PSY_2.fasta")

Cann_PSY_Ann <- Cann_Carotenoids_Ann %>% filter(Protein %in% grep("PSY", Cann_Carotenoids_Ann$Protein, value= TRUE))
Cann_PSY_fasta <- Cannuum_ASM51225v2_fasta[c(which(names(Cannuum_ASM51225v2_fasta) %in% Cann_PSY_Ann$Name))]
write.fasta(Cann_PSY_fasta, as.string=TRUE, names= names(Cann_PSY_fasta),
            file.out = "Funcional Groups/Cann_PSY.fasta")
rename.fasta(infile = "Funcional Groups/Cann_PSY.fasta", Cann_PSY_Ann, outfile = "Funcional Groups/Cann_PSY_2.fasta")

Ath_PSY_Ann <- Ath_Carotenoids_Ann %>% filter(Name %in% grep("PSY", Ath_Carotenoids_Ann$Name, value= TRUE))
Ath_PSY_Ann <- merge(Ath_PSY_Ann, Ath_Gene_Models, by.x="ID", by.y=1)
Ath_PSY_fasta <- Ath11_fasta[c(which(names(Ath11_fasta) %in% Ath_PSY_Ann$V2))]
write.fasta(Ath_PSY_fasta, as.string=TRUE, names= names(Ath_PSY_fasta),
            file.out = "Funcional Groups/Ath_PSY.fasta")
rename.fasta(infile = "Funcional Groups/Ath_PSY.fasta", Ath_PSY_Ann %>% select(3,2), outfile = "Funcional Groups/Ath_PSY_2.fasta")

Niben_PSY_fasta <- seqinr::read.fasta("Funcional Groups/Niben_PSY_2.fasta", as.string = TRUE, seqtype = "AA")
Sly_PSY_fasta <- seqinr::read.fasta("Funcional Groups/Sly_PSY_2.fasta", as.string = TRUE, seqtype = "AA")
Cann_PSY_fasta <- seqinr::read.fasta("Funcional Groups/Cann_PSY_2.fasta", as.string = TRUE, seqtype = "AA")
Ath_PSY_fasta <- seqinr::read.fasta("Funcional Groups/Ath_PSY_2.fasta", as.string = TRUE, seqtype = "AA")

PSYs_fasta <- append(Niben_PSY_fasta, append(Sly_PSY_fasta, append(Cann_PSY_fasta, Ath_PSY_fasta)))
write.fasta(PSYs_fasta, as.string=TRUE, names= names(PSYs_fasta),
            file.out = "Funcional Groups/PSYs.fasta")


#########################################################################################################################################
####################################  Retrograde  #############################################################################################
#########################################################################################################################################

Retrograde_Ath <- read.table("Funcional Groups/Ath_Retrograde.txt", header = TRUE) %>% select(ID, Name)
Retrograde_Ath <- merge(Retrograde_Ath, Ath_Gene_Models, by.x="ID", by.y=1)
Retrograde_Ath_fasta <- Ath11_fasta[c(which(names(Ath11_fasta) %in% Retrograde_Ath$V2))]
Retrograde_Ath_2 <- Retrograde_Ath %>% select(V2, Name)
write.fasta(Retrograde_Ath_fasta, as.string=TRUE, names= names(Retrograde_Ath_fasta),
            file.out = "Funcional Groups/Ath_Retrograde.fasta")
rename.fasta(infile = "Funcional Groups/Ath_Retrograde.fasta", Retrograde_Ath_2, outfile = "Funcional Groups/Ath_Retrograde_2.fasta")

Retrograde_Niben_Candidates <- read.table("Funcional Groups/Niben_Candidates_Retrograde.txt")
Retrograde_Niben_Candidates <- Retrograde_Niben_Candidates[Retrograde_Niben_Candidates$V3>=35,] %>% select(V2)
Retrograde_Niben_Candidates <- unique(Retrograde_Niben_Candidates)
Retrograde_Niben_Candidates_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Retrograde_Niben_Candidates$V2))]
write.fasta(Retrograde_Niben_Candidates_fasta, as.string=TRUE, names= names(Retrograde_Niben_Candidates_fasta),
            file.out = "Funcional Groups/Retrograde_Niben_Candidates_35.fasta")

Retrograde_Sly_Candidates <- read.table("Funcional Groups/Sly_Candidates_Retrograde.txt")
Retrograde_Sly_Candidates <- Retrograde_Sly_Candidates[Retrograde_Sly_Candidates$V3>=35,] %>% select(V2)
Retrograde_Sly_Candidates <- unique(Retrograde_Sly_Candidates)
Retrograde_Sly_Candidates_fasta <- Tomato_4.0_fasta[c(which(names(Tomato_4.0_fasta) %in% Retrograde_Sly_Candidates$V2))]
write.fasta(Retrograde_Sly_Candidates_fasta, as.string=TRUE, names= names(Retrograde_Sly_Candidates_fasta),
            file.out = "Funcional Groups/Retrograde_Sly_Candidates_35.fasta")

Retrograde_Cann_Candidates <- read.table("Funcional Groups/Cann_Candidates_Retrograde.txt")
Retrograde_Cann_Candidates <- Retrograde_Cann_Candidates[Retrograde_Cann_Candidates$V3>=35,] %>% select(V2)
Retrograde_Cann_Candidates <- unique(Retrograde_Cann_Candidates)
Retrograde_Cann_Candidates_fasta <- Cannuum_ASM51225v2_fasta[c(which(names(Cannuum_ASM51225v2_fasta) %in% Retrograde_Cann_Candidates$V2))]
write.fasta(Retrograde_Cann_Candidates_fasta, as.string=TRUE, names= names(Retrograde_Cann_Candidates_fasta),
            file.out = "Funcional Groups/Retrograde_Cann_Candidates_35.fasta")


Niben_Retrograde_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_Retrograde_Ann$Name))]
write.fasta(Niben_Retrograde_fasta, as.string=TRUE, names= names(Niben_Retrograde_fasta),
            file.out = "Funcional Groups/Niben_Retrograde.fasta")
Niben_Retrograde_Ann_2 <- Niben_Retrograde_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_Retrograde.fasta", Niben_Retrograde_Ann, outfile = "Funcional Groups/Niben_Retrograde_2.fasta")

save_heatmaps("Retrograde", height = 6)

## Representing Heatmap

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Niben_Retrograde_Ann <- read.table("Funcional Groups/FigTree/Niben_Retrograde.txt",
                            sep = "\t", header = TRUE)

colnames(Niben_Retrograde_Ann) <- c("Name", "Protein")

Niben_Retrograde_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_Retrograde_Ann$Name))]
write.fasta(Niben_Retrograde_fasta, as.string=TRUE, names= names(Niben_Retrograde_fasta),
            file.out = "Funcional Groups/Niben_Retrograde.fasta")
Niben_Retrograde_Ann_2 <- Niben_Retrograde_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_Retrograde.fasta", Niben_Retrograde_Ann, outfile = "Funcional Groups/Niben_Retrograde_2.fasta")

Niben_Retrograde_Chloro <- read.delim("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan/Funcional Groups/Niben_Retrograde_2_summary.targetp2", 
                               header=FALSE, comment.char="#")
colnames(Niben_Retrograde_Chloro) <- c("ID",	"Prediction",	"noTP",	"SP",	"mTP",	"cTP",	"luTP",	"CS-Position")
Niben_Retrograde_Chloro$Prediction <- factor(Niben_Retrograde_Chloro$Prediction, levels = c("cTP","mTP","noTP"))
Niben_Retrograde_Chloro <- merge(Niben_Retrograde_Ann, Niben_Retrograde_Chloro, by.x = "Protein", by.y = "ID")

Niben_Retrograde_Chloro <- Niben_Retrograde_Chloro[order(match(Niben_Retrograde_Chloro$Protein, Niben_Retrograde_Ann$Protein)),]
rownames(Niben_Retrograde_Chloro) <- Niben_Retrograde_Chloro$Protein
Niben_Retrograde_Chloro <- Niben_Retrograde_Chloro %>% select(Prediction)


## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_Retrograde <- merge(Niben_Retrograde_Ann, TPM_All_FC_Annotated, by="Name")

## We create the annotation for Modules

Niben_Retrograde_Modules <- which_Module_pwd12(Niben_Retrograde_Ann)

## We merge this annotation of Modules with the annotation of Chloroplast TP prediction

Niben_Retrograde_Modules_Chl <- merge(Niben_Retrograde_Modules, Niben_Retrograde_Chloro, by = 0)
Niben_Retrograde_Modules_Chl <- Niben_Retrograde_Modules_Chl[order(match(Niben_Retrograde_Modules_Chl$Row.names, Niben_Retrograde_Ann$Protein)),]
rownames(Niben_Retrograde_Modules_Chl) <- Niben_Retrograde_Modules_Chl$Row.names
Niben_Retrograde_Modules_Chl$Row.names <- NULL

## We select only the FC data from the different comparisons
Niben_Retrograde_TPM <- Niben_Retrograde %>% select(Protein, 4:51, 61:66)
Niben_Retrograde_TPM <- Niben_Retrograde_TPM[!duplicated(Niben_Retrograde_TPM$Protein),]
Niben_Retrograde_TPM <- Niben_Retrograde_TPM[order(match(Niben_Retrograde_TPM[,1], Niben_Retrograde_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_Retrograde_FC <- Niben_Retrograde %>% select(Protein, 71, 73:87)
Niben_Retrograde_FC <- Niben_Retrograde_FC[!duplicated(Niben_Retrograde_FC$Protein),]
Niben_Retrograde_FC <- Niben_Retrograde_FC[order(match(Niben_Retrograde_FC[,1], Niben_Retrograde_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_Retrograde_DEG <- Niben_Retrograde %>% select(Protein, 89, 91:105)
Niben_Retrograde_DEG <- Niben_Retrograde_DEG[!duplicated(Niben_Retrograde_DEG$Protein),]
Niben_Retrograde_DEG <- Niben_Retrograde_DEG[order(match(Niben_Retrograde_DEG[,1], Niben_Retrograde_Ann[,2])),]

Niben_Retrograde_DEG$pcrtB22h_vs_GFP22h_filtered <- ifelse(is.na(Niben_Retrograde_DEG$pcrtB22h_vs_GFP22h_filtered) == TRUE, " ", "*")
Niben_Retrograde_DEG$pcrtB25h_vs_GFP25h_filtered <- ifelse(is.na(Niben_Retrograde_DEG$pcrtB25h_vs_GFP25h_filtered) == TRUE, " ", "*")
Niben_Retrograde_DEG$pcrtBAA_vs_GFPAA_filtered <- ifelse(is.na(Niben_Retrograde_DEG$pcrtBAA_vs_GFPAA_filtered) == TRUE, " ", "*")
Niben_Retrograde_DEG$pcrtBBB_vs_GFPBB_filtered <- ifelse(is.na(Niben_Retrograde_DEG$pcrtBBB_vs_GFPBB_filtered) == TRUE, " ", "*")
Niben_Retrograde_DEG$pcrtBYY_vs_GFPYY_filtered <- ifelse(is.na(Niben_Retrograde_DEG$pcrtBYY_vs_GFPYY_filtered) == TRUE, " ", "*")
Niben_Retrograde_DEG$pcrtB28h_vs_GFP28h_filtered <- ifelse(is.na(Niben_Retrograde_DEG$pcrtB28h_vs_GFP28h_filtered) == TRUE, " ", "*")
Niben_Retrograde_DEG$pcrtB34h_vs_GFP34h_filtered <- ifelse(is.na(Niben_Retrograde_DEG$pcrtB34h_vs_GFP34h_filtered) == TRUE, " ", "*")
Niben_Retrograde_DEG$pcrtB37h_vs_GFP37h_filtered <- ifelse(is.na(Niben_Retrograde_DEG$pcrtB37h_vs_GFP37h_filtered) == TRUE, " ", "*")
Niben_Retrograde_DEG$pcrtBZZ_vs_GFPZZ_filtered <- ifelse(is.na(Niben_Retrograde_DEG$pcrtBZZ_vs_GFPZZ_filtered) == TRUE, " ", "*")
Niben_Retrograde_DEG$pcrtBCC_vs_GFPCC_filtered <- ifelse(is.na(Niben_Retrograde_DEG$pcrtBCC_vs_GFPCC_filtered) == TRUE, " ", "*")
Niben_Retrograde_DEG$pcrtB40h_vs_GFP40h_filtered <- ifelse(is.na(Niben_Retrograde_DEG$pcrtB40h_vs_GFP40h_filtered) == TRUE, " ", "*")
Niben_Retrograde_DEG$pcrtBDD_vs_GFPDD_filtered <- ifelse(is.na(Niben_Retrograde_DEG$pcrtBDD_vs_GFPDD_filtered) == TRUE, " ", "*")
Niben_Retrograde_DEG$pcrtB46h_vs_GFP46h_filtered <- ifelse(is.na(Niben_Retrograde_DEG$pcrtB46h_vs_GFP46h_filtered) == TRUE, " ", "*")
Niben_Retrograde_DEG$pcrtB56h_vs_GFP56h_filtered <- ifelse(is.na(Niben_Retrograde_DEG$pcrtB56h_vs_GFP56h_filtered) == TRUE, " ", "*")
Niben_Retrograde_DEG$pcrtBEE_vs_GFPEE_filtered <- ifelse(is.na(Niben_Retrograde_DEG$pcrtBEE_vs_GFPEE_filtered) == TRUE, " ", "*")
Niben_Retrograde_DEG$pcrtB96h_vs_GFP96h_filtered <- ifelse(is.na(Niben_Retrograde_DEG$pcrtB96h_vs_GFP96h_filtered) == TRUE, " ", "*")
row.names(Niben_Retrograde_DEG) <- Niben_Retrograde_DEG$Protein
Niben_Retrograde_DEG[1] <- NULL

Niben_Retrograde_DEG <- Niben_Retrograde_DEG %>%
  select(pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

row.names(Niben_Retrograde_FC) <- Niben_Retrograde_FC$Protein
Niben_Retrograde_FC[1] <- NULL

Niben_Retrograde_FC <- Niben_Retrograde_FC %>%
  select(pcrtB22h_vs_GFP22h_All, pcrtBAA_vs_GFPAA_All, pcrtB25h_vs_GFP25h_All, 
         pcrtBYY_vs_GFPYY_All, pcrtBBB_vs_GFPBB_All, pcrtB28h_vs_GFP28h_All, pcrtB34h_vs_GFP34h_All, pcrtBZZ_vs_GFPZZ_All, 
         pcrtB37h_vs_GFP37h_All, pcrtBCC_vs_GFPCC_All, pcrtB40h_vs_GFP40h_All, pcrtBDD_vs_GFPDD_All, pcrtB46h_vs_GFP46h_All, 
         pcrtBEE_vs_GFPEE_All, pcrtB56h_vs_GFP56h_All, pcrtB96h_vs_GFP96h_All)


min(Niben_Retrograde_FC, na.rm = TRUE)
max(Niben_Retrograde_FC, na.rm = TRUE)
breaks_Retrograde_FC=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Niben_Retrograde_FC %>% select(1:15), 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Retrograde_FC,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_Retrograde_Annotation, annotation_colors=color_Retrograde, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h"),#, "96h"),
         cex = 1,
         angle_col=45,
         main = "Retrograde",
         annotation_legend = TRUE,
         display_numbers = Niben_Retrograde_DEG %>% select(1:15),
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(84, 104),
         annotation_row = Niben_Retrograde_Modules_Chl,
         annotation_colors = module_Chl_colors_2
         #gaps_col = c(15)
)


rownames(Niben_Retrograde_TPM) <- Niben_Retrograde_TPM$Protein
Niben_Retrograde_TPM[1] <- NULL

min(Niben_Retrograde_TPM, na.rm = TRUE)
max(Niben_Retrograde_TPM, na.rm = TRUE)
breaks_Niben_Retrograde_TPM=c(seq(0,100, length.out = 100))

Niben_Retrograde_TPM_norm <- normalize_df(Niben_Retrograde_TPM)
Niben_Retrograde_TPM_norm_Cornell <- Niben_Retrograde_TPM %>%
  select(1:48) 
Niben_Retrograde_TPM_norm_Cornell <- normalize_df(Niben_Retrograde_TPM_norm_Cornell)


pheatmap(Niben_Retrograde_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Niben_Retrograde_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_Retrograde_Annotation, annotation_colors=color_Retrograde, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Retrograde",
         annotation_legend = TRUE,
         #display_numbers = Niben_Retrograde_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         annotation_row = Niben_Retrograde_Modules,
         annotation_colors = module_colors,
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(24, 48, 51)
)



#####Representing Heatmap for Sly

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Sly_Retrograde_Ann <- read.table("Funcional Groups/FigTree/Sly_Retrograde.txt",
                          sep = "\t", header = TRUE)

colnames(Sly_Retrograde_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Sly_Retrograde <- merge(Sly_Retrograde_Ann, Summary_Yazdani_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Sly_Retrograde_TPM <- Sly_Retrograde %>% select(Protein, 4:66)
Sly_Retrograde_TPM <- Sly_Retrograde_TPM[!duplicated(Sly_Retrograde_TPM$Protein),]
Sly_Retrograde_TPM <- Sly_Retrograde_TPM[order(match(Sly_Retrograde_TPM[,1], Sly_Retrograde_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_Retrograde_FC <- Sly_Retrograde %>% select(Protein, 67:71)
Sly_Retrograde_FC <- Sly_Retrograde_FC[!duplicated(Sly_Retrograde_FC$Protein),]
Sly_Retrograde_FC <- Sly_Retrograde_FC[order(match(Sly_Retrograde_FC[,1], Sly_Retrograde_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_Retrograde_DEG <- Sly_Retrograde %>% select(Protein, 72:76)
Sly_Retrograde_DEG <- Sly_Retrograde_DEG[!duplicated(Sly_Retrograde_DEG$Protein),]
Sly_Retrograde_DEG <- Sly_Retrograde_DEG[order(match(Sly_Retrograde_DEG[,1], Sly_Retrograde_Ann[,2])),]

Sly_Retrograde_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Sly_Retrograde_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Sly_Retrograde_DEG$Ripening_Orange_DEG <- ifelse(is.na(Sly_Retrograde_DEG$Ripening_Orange_DEG) == TRUE, " ", "*")
Sly_Retrograde_DEG$Ripening_Red_DEG <- ifelse(is.na(Sly_Retrograde_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
Sly_Retrograde_DEG$ORHis_MG_DEG <- ifelse(is.na(Sly_Retrograde_DEG$ORHis_MG_DEG) == TRUE, " ", "*")
Sly_Retrograde_DEG$ORHis_Breaker_DEG <- ifelse(is.na(Sly_Retrograde_DEG$ORHis_Breaker_DEG) == TRUE, " ", "*")
row.names(Sly_Retrograde_DEG) <- Sly_Retrograde_DEG$Protein
Sly_Retrograde_DEG[1] <- NULL

rownames(Sly_Retrograde_FC) <- Sly_Retrograde_FC$Protein
Sly_Retrograde_FC[1] <- NULL

min(Sly_Retrograde_FC, na.rm = TRUE)
max(Sly_Retrograde_FC, na.rm = TRUE)
breaks_Sly_Retrograde=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Sly_Retrograde_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Sly_Retrograde,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_Retrograde_Annotation, annotation_colors=color_Retrograde, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Retrograde",
         annotation_legend = TRUE,
         display_numbers = Sly_Retrograde_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(39, 48),
         gaps_col = c(3)
)


rownames(Sly_Retrograde_TPM) <- Sly_Retrograde_TPM$Protein
Sly_Retrograde_TPM[1] <- NULL

min(Sly_Retrograde_TPM, na.rm = TRUE)
max(Sly_Retrograde_TPM, na.rm = TRUE)
breaks_Sly_Retrograde_TPM=c(seq(0,100, length.out = 200))

Sly_Retrograde_TPM_norm <- normalize_df(Sly_Retrograde_TPM)

pheatmap(Sly_Retrograde_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Sly_Retrograde_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_Retrograde_Annotation, annotation_colors=color_Retrograde, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Retrograde",
         annotation_legend = TRUE,
         #display_numbers = Sly_Retrograde_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(21,42)
)


#####Representing Heatmap for Cann

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Cann_Retrograde_Ann <- read.table("Funcional Groups/FigTree/Cann_Retrograde.txt",
                           sep = "\t", header = TRUE)

colnames(Cann_Retrograde_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Cann_Retrograde <- merge(Cann_Retrograde_Ann, Summary_Pepper_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Cann_Retrograde_TPM <- Cann_Retrograde %>% select(Protein, 4:17)
Cann_Retrograde_TPM <- Cann_Retrograde_TPM[!duplicated(Cann_Retrograde_TPM$Protein),]
Cann_Retrograde_TPM <- Cann_Retrograde_TPM[order(match(Cann_Retrograde_TPM[,1], Cann_Retrograde_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_Retrograde_FC <- Cann_Retrograde %>% select(Protein, 18:19)
Cann_Retrograde_FC <- Cann_Retrograde_FC[!duplicated(Cann_Retrograde_FC$Protein),]
Cann_Retrograde_FC <- Cann_Retrograde_FC[order(match(Cann_Retrograde_FC[,1], Cann_Retrograde_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_Retrograde_DEG <- Cann_Retrograde %>% select(Protein, 20:21)
Cann_Retrograde_DEG <- Cann_Retrograde_DEG[!duplicated(Cann_Retrograde_DEG$Protein),]
Cann_Retrograde_DEG <- Cann_Retrograde_DEG[order(match(Cann_Retrograde_DEG[,1], Cann_Retrograde_Ann[,2])),]

Cann_Retrograde_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Cann_Retrograde_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Cann_Retrograde_DEG$Ripening_Red_DEG <- ifelse(is.na(Cann_Retrograde_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
row.names(Cann_Retrograde_DEG) <- Cann_Retrograde_DEG$Protein
Cann_Retrograde_DEG[1] <- NULL

rownames(Cann_Retrograde_FC) <- Cann_Retrograde_FC$Protein
Cann_Retrograde_FC[1] <- NULL

min(Cann_Retrograde_FC, na.rm = TRUE)
max(Cann_Retrograde_FC, na.rm = TRUE)
breaks_Cann_Retrograde=c(seq(-4.5,0, length.out = 100), seq(0.01,6,length.out = 100))

pheatmap(Cann_Retrograde_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Cann_Retrograde,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_Retrograde_Annotation, annotation_colors=color_Retrograde, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Red"),
         cex = 1,
         angle_col=45,
         main = "Retrograde",
         annotation_legend = TRUE,
         display_numbers = Cann_Retrograde_DEG,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)


rownames(Cann_Retrograde_TPM) <- Cann_Retrograde_TPM$Protein
Cann_Retrograde_TPM[1] <- NULL

min(Cann_Retrograde_TPM, na.rm = TRUE)
max(Cann_Retrograde_TPM, na.rm = TRUE)
breaks_Cann_Retrograde_TPM=c(seq(0,100, length.out = 200))

Cann_Retrograde_TPM_norm <- normalize_df(Cann_Retrograde_TPM)

pheatmap(Cann_Retrograde_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Cann_Retrograde_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_Retrograde_Annotation, annotation_colors=color_Retrograde, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "Retrograde",
         annotation_legend = TRUE,
         #display_numbers = Cann_Retrograde_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(4,9)
)


## Comparativa 96h_SlyYaz_Ath

Ath_Retrograde_Ann <- read.table("Funcional Groups/Ath_Retrograde.txt", header = TRUE)

Ath_Retrograde_FC <- merge(Ath_Retrograde_Ann, Summary_Ath_FC, by.x=1, by.y = 0) %>%
  dplyr::select(Name, log2FoldChange)
Ath_Retrograde_FC <- Ath_Retrograde_FC[order(match(Ath_Retrograde_FC[,1], Ath_Retrograde_Ann[,2])),]

Ath_Retrograde_DEG <- merge(Ath_Retrograde_Ann, Summary_Ath_DEGs, by.x=1, by.y = 1, all.x = TRUE) %>%
  dplyr::select(Name, log2FC)
Ath_Retrograde_DEG <- Ath_Retrograde_DEG[order(match(Ath_Retrograde_DEG[,1], Ath_Retrograde_Ann[,2])),]

Ath_Retrograde_DEG$log2FC <- ifelse(is.na(Ath_Retrograde_DEG$log2FC) == TRUE, " ", "*")
row.names(Ath_Retrograde_DEG) <- Ath_Retrograde_DEG$Name
Ath_Retrograde_DEG[1] <- NULL

rownames(Ath_Retrograde_FC) <- Ath_Retrograde_FC$Name
Ath_Retrograde_FC[1] <- NULL

min(Ath_Retrograde_FC, na.rm = TRUE)
max(Ath_Retrograde_FC, na.rm = TRUE)
breaks_Ath_Retrograde=c(seq(-4.5,0, length.out = 100), seq(0.01,4.5,length.out = 100))

pheatmap(Ath_Retrograde_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Stndr,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_Retrograde_Annotation, annotation_colors=color_Retrograde, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = "Senescence",
         cex = 1,
         angle_col=45,
         main = "Retrograde",
         annotation_legend = TRUE,
         display_numbers = Ath_Retrograde_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         gaps_row = c(1, 2, 3, 4, 5, 6, 7),
         #gaps_col = c(2,5)
)

pheatmap(Niben_Retrograde_FC %>% select(16), 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         #breaks = breaks_Retrograde_FC,
         breaks = breaks_Stndr,
         #legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         #legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_Retrograde_Annotation, annotation_colors=color_Retrograde, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = "96h",
         cex = 1,
         angle_col=45,
         main = "Retrograde",
         annotation_legend = TRUE,
         display_numbers = Niben_Retrograde_DEG %>% select(16),
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(84, 104),
         annotation_row = Niben_Retrograde_Modules_Chl,
         annotation_colors = module_Chl_colors_2,
         gaps_row = c(10, 12,14,15,17,21)
)


pheatmap(Sly_Retrograde_FC %>% select(2,3), 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Stndr,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         #legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_Retrograde_Annotation, annotation_colors=color_Retrograde, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Orange", "Ripening_Red"),
         cex = 1,
         angle_col=45,
         main = "Retrograde",
         annotation_legend = TRUE,
         display_numbers = Sly_Retrograde_DEG %>% select(2,3),
         fontsize_number = 15,
         na_col = "darkgrey",
         gaps_row = c(3, 4, 5, 6, 7, 8)
)


#########################################################################################################################################
####################################  HSFs  #############################################################################################
#########################################################################################################################################

HSFs_Ath <- read.table("Funcional Groups/Ath_HSFs.txt", header = TRUE) %>% select(ID, Name)
HSFs_Ath <- merge(HSFs_Ath, Ath_Gene_Models, by.x="ID", by.y=1)
HSFs_Ath_fasta <- Ath11_fasta[c(which(names(Ath11_fasta) %in% HSFs_Ath$V2))]
HSFs_Ath_2 <- HSFs_Ath %>% select(V2, Name)
write.fasta(HSFs_Ath_fasta, as.string=TRUE, names= names(HSFs_Ath_fasta),
            file.out = "Funcional Groups/Ath_HSFs.fasta")
rename.fasta(infile = "Funcional Groups/Ath_HSFs.fasta", HSFs_Ath_2, outfile = "Funcional Groups/Ath_HSFs_2.fasta")

HSFs_Niben_Candidates <- read.table("Funcional Groups/Niben_Candidates_HSFs.txt")
HSFs_Niben_Candidates <- HSFs_Niben_Candidates[HSFs_Niben_Candidates$V3>=90,] %>% select(V1, V2, V3, V12)
HSFs_Niben_Candidates <- HSFs_Niben_Candidates[order(HSFs_Niben_Candidates$V3, decreasing = TRUE),]
HSFs_Niben_Candidates <- HSFs_Niben_Candidates[!duplicated(HSFs_Niben_Candidates$V2),]
HSFs_Niben_Candidates <- HSFs_Niben_Candidates[order(HSFs_Niben_Candidates$V1),]
HSFs_Niben_Candidates <- HSFs_Niben_Candidates %>% separate(V1, into = c("Protein", "Old"), sep = "\\|")
HSFs_Niben_Candidates <- HSFs_Niben_Candidates %>% select(V2, Protein)
colnames(HSFs_Niben_Candidates) <- c("Name", "Protein")
HSFs_Niben_Candidates$Protein <- str_sub(HSFs_Niben_Candidates$Protein, 6, -1)
HSFs_Niben_Candidates$Protein <- paste0("Nb", HSFs_Niben_Candidates$Protein)
HSFs_Niben_Candidates$Protein[HSFs_Niben_Candidates$Name=="Niben261Chr17g0082001.1"] <- "NbHsfA5a"
HSFs_Niben_Candidates$Protein[HSFs_Niben_Candidates$Name=="Niben261Chr12g0687011.1"] <- "NbHsfDis1"
HSFs_Niben_Candidates$Protein[HSFs_Niben_Candidates$Name=="Niben261Chr16g0192021.1"] <- "NbHsfDis2"
HSFs_Niben_Candidates$Protein[HSFs_Niben_Candidates$Name=="Niben261Chr10g0965009.1"] <- "NbHsfDis3"
HSFs_Niben_Candidates$Protein[HSFs_Niben_Candidates$Name=="Niben261Chr01g0150003.1"] <- "NbHsfDis4"
HSFs_Niben_Candidates$Protein[HSFs_Niben_Candidates$Name=="Niben261Chr07g1326008.1"] <- "NbHsfDis5"


Niben_HSFs_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_HSFs_Ann$Name))]
write.fasta(Niben_HSFs_fasta, as.string=TRUE, names= names(Niben_HSFs_fasta),
            file.out = "Funcional Groups/Niben_HSFs.fasta")
Niben_HSFs_Ann_2 <- Niben_HSFs_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_HSFs.fasta", Niben_HSFs_Ann, outfile = "Funcional Groups/Niben_HSFs_2.fasta")

save_heatmaps("HSFs", height = 6)

## Representing Heatmap

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Niben_HSFs_Ann <- HSFs_Niben_Candidates
write.table(Niben_HSFs_Ann, "/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan/Funcional Groups/FigTree/Niben_HSFs.txt",
            row.names = FALSE, sep = "\t", quote = FALSE)

Niben_HSFs_fasta <- Niben261_fasta[c(which(names(Niben261_fasta) %in% Niben_HSFs_Ann$Name))]
write.fasta(Niben_HSFs_fasta, as.string=TRUE, names= names(Niben_HSFs_fasta),
            file.out = "Funcional Groups/Niben_HSFs.fasta")
Niben_HSFs_Ann_2 <- Niben_HSFs_Ann %>% select(Name, Protein)
rename.fasta(infile = "Funcional Groups/Niben_HSFs.fasta", Niben_HSFs_Ann, outfile = "Funcional Groups/Niben_HSFs_2.fasta")

Niben_HSFs_Chloro <- read.delim("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan/Funcional Groups/Niben_HSFs_2_summary.targetp2", 
                                      header=FALSE, comment.char="#")
colnames(Niben_HSFs_Chloro) <- c("ID",	"Prediction",	"noTP",	"SP",	"mTP",	"cTP",	"luTP",	"CS-Position")
Niben_HSFs_Chloro$Prediction <- factor(Niben_HSFs_Chloro$Prediction, levels = c("cTP","mTP","noTP"))
Niben_HSFs_Chloro <- merge(Niben_HSFs_Ann, Niben_HSFs_Chloro, by.x = "Protein", by.y = "ID")

Niben_HSFs_Chloro <- Niben_HSFs_Chloro[order(match(Niben_HSFs_Chloro$Protein, Niben_HSFs_Ann$Protein)),]
rownames(Niben_HSFs_Chloro) <- Niben_HSFs_Chloro$Protein
Niben_HSFs_Chloro <- Niben_HSFs_Chloro %>% select(Prediction)


## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Niben_HSFs <- merge(Niben_HSFs_Ann, TPM_All_FC_Annotated, by="Name")

## We create the annotation for Modules

Niben_HSFs_Modules <- which_Module_pwd12(Niben_HSFs_Ann)

## We merge this annotation of Modules with the annotation of Chloroplast TP prediction

Niben_HSFs_Modules_Chl <- merge(Niben_HSFs_Modules, Niben_HSFs_Chloro, by = 0)
Niben_HSFs_Modules_Chl <- Niben_HSFs_Modules_Chl[order(match(Niben_HSFs_Modules_Chl$Row.names, Niben_HSFs_Ann$Protein)),]
rownames(Niben_HSFs_Modules_Chl) <- Niben_HSFs_Modules_Chl$Row.names
Niben_HSFs_Modules_Chl$Row.names <- NULL

## We select only the FC data from the different comparisons
Niben_HSFs_TPM <- Niben_HSFs %>% select(Protein, 4:51, 61:66)
Niben_HSFs_TPM <- Niben_HSFs_TPM[!duplicated(Niben_HSFs_TPM$Protein),]
Niben_HSFs_TPM <- Niben_HSFs_TPM[order(match(Niben_HSFs_TPM[,1], Niben_HSFs_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_HSFs_FC <- Niben_HSFs %>% select(Protein, 71, 73:87)
Niben_HSFs_FC <- Niben_HSFs_FC[!duplicated(Niben_HSFs_FC$Protein),]
Niben_HSFs_FC <- Niben_HSFs_FC[order(match(Niben_HSFs_FC[,1], Niben_HSFs_Ann[,2])),]


## We select only the FC data from the different comparisons
Niben_HSFs_DEG <- Niben_HSFs %>% select(Protein, 89, 91:105)
Niben_HSFs_DEG <- Niben_HSFs_DEG[!duplicated(Niben_HSFs_DEG$Protein),]
Niben_HSFs_DEG <- Niben_HSFs_DEG[order(match(Niben_HSFs_DEG[,1], Niben_HSFs_Ann[,2])),]

Niben_HSFs_DEG$pcrtB22h_vs_GFP22h_filtered <- ifelse(is.na(Niben_HSFs_DEG$pcrtB22h_vs_GFP22h_filtered) == TRUE, " ", "*")
Niben_HSFs_DEG$pcrtB25h_vs_GFP25h_filtered <- ifelse(is.na(Niben_HSFs_DEG$pcrtB25h_vs_GFP25h_filtered) == TRUE, " ", "*")
Niben_HSFs_DEG$pcrtBAA_vs_GFPAA_filtered <- ifelse(is.na(Niben_HSFs_DEG$pcrtBAA_vs_GFPAA_filtered) == TRUE, " ", "*")
Niben_HSFs_DEG$pcrtBBB_vs_GFPBB_filtered <- ifelse(is.na(Niben_HSFs_DEG$pcrtBBB_vs_GFPBB_filtered) == TRUE, " ", "*")
Niben_HSFs_DEG$pcrtBYY_vs_GFPYY_filtered <- ifelse(is.na(Niben_HSFs_DEG$pcrtBYY_vs_GFPYY_filtered) == TRUE, " ", "*")
Niben_HSFs_DEG$pcrtB28h_vs_GFP28h_filtered <- ifelse(is.na(Niben_HSFs_DEG$pcrtB28h_vs_GFP28h_filtered) == TRUE, " ", "*")
Niben_HSFs_DEG$pcrtB34h_vs_GFP34h_filtered <- ifelse(is.na(Niben_HSFs_DEG$pcrtB34h_vs_GFP34h_filtered) == TRUE, " ", "*")
Niben_HSFs_DEG$pcrtB37h_vs_GFP37h_filtered <- ifelse(is.na(Niben_HSFs_DEG$pcrtB37h_vs_GFP37h_filtered) == TRUE, " ", "*")
Niben_HSFs_DEG$pcrtBZZ_vs_GFPZZ_filtered <- ifelse(is.na(Niben_HSFs_DEG$pcrtBZZ_vs_GFPZZ_filtered) == TRUE, " ", "*")
Niben_HSFs_DEG$pcrtBCC_vs_GFPCC_filtered <- ifelse(is.na(Niben_HSFs_DEG$pcrtBCC_vs_GFPCC_filtered) == TRUE, " ", "*")
Niben_HSFs_DEG$pcrtB40h_vs_GFP40h_filtered <- ifelse(is.na(Niben_HSFs_DEG$pcrtB40h_vs_GFP40h_filtered) == TRUE, " ", "*")
Niben_HSFs_DEG$pcrtBDD_vs_GFPDD_filtered <- ifelse(is.na(Niben_HSFs_DEG$pcrtBDD_vs_GFPDD_filtered) == TRUE, " ", "*")
Niben_HSFs_DEG$pcrtB46h_vs_GFP46h_filtered <- ifelse(is.na(Niben_HSFs_DEG$pcrtB46h_vs_GFP46h_filtered) == TRUE, " ", "*")
Niben_HSFs_DEG$pcrtB56h_vs_GFP56h_filtered <- ifelse(is.na(Niben_HSFs_DEG$pcrtB56h_vs_GFP56h_filtered) == TRUE, " ", "*")
Niben_HSFs_DEG$pcrtBEE_vs_GFPEE_filtered <- ifelse(is.na(Niben_HSFs_DEG$pcrtBEE_vs_GFPEE_filtered) == TRUE, " ", "*")
Niben_HSFs_DEG$pcrtB96h_vs_GFP96h_filtered <- ifelse(is.na(Niben_HSFs_DEG$pcrtB96h_vs_GFP96h_filtered) == TRUE, " ", "*")
row.names(Niben_HSFs_DEG) <- Niben_HSFs_DEG$Protein
Niben_HSFs_DEG[1] <- NULL

Niben_HSFs_DEG <- Niben_HSFs_DEG %>%
  select(pcrtB22h_vs_GFP22h_filtered, pcrtBAA_vs_GFPAA_filtered, pcrtB25h_vs_GFP25h_filtered, 
         pcrtBYY_vs_GFPYY_filtered, pcrtBBB_vs_GFPBB_filtered, pcrtB28h_vs_GFP28h_filtered, pcrtB34h_vs_GFP34h_filtered, pcrtBZZ_vs_GFPZZ_filtered, 
         pcrtB37h_vs_GFP37h_filtered, pcrtBCC_vs_GFPCC_filtered, pcrtB40h_vs_GFP40h_filtered, pcrtBDD_vs_GFPDD_filtered, pcrtB46h_vs_GFP46h_filtered, 
         pcrtBEE_vs_GFPEE_filtered, pcrtB56h_vs_GFP56h_filtered, pcrtB96h_vs_GFP96h_filtered)

row.names(Niben_HSFs_FC) <- Niben_HSFs_FC$Protein
Niben_HSFs_FC[1] <- NULL

Niben_HSFs_FC <- Niben_HSFs_FC %>%
  select(pcrtB22h_vs_GFP22h_All, pcrtBAA_vs_GFPAA_All, pcrtB25h_vs_GFP25h_All, 
         pcrtBYY_vs_GFPYY_All, pcrtBBB_vs_GFPBB_All, pcrtB28h_vs_GFP28h_All, pcrtB34h_vs_GFP34h_All, pcrtBZZ_vs_GFPZZ_All, 
         pcrtB37h_vs_GFP37h_All, pcrtBCC_vs_GFPCC_All, pcrtB40h_vs_GFP40h_All, pcrtBDD_vs_GFPDD_All, pcrtB46h_vs_GFP46h_All, 
         pcrtBEE_vs_GFPEE_All, pcrtB56h_vs_GFP56h_All, pcrtB96h_vs_GFP96h_All)


min(Niben_HSFs_FC, na.rm = TRUE)
max(Niben_HSFs_FC, na.rm = TRUE)
breaks_HSFs_FC=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Niben_HSFs_FC %>% select(1:15), 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_HSFs_FC,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_HSFs_Annotation, annotation_colors=color_HSFs, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("22h", "22-25h", "25h",  "25-28h", "25-28h", "28h", "34h","34-37h", "37h", "34-37-40h","40h","40-46h","46h", "46-56h","56h"),#, "96h"),
         cex = 1,
         angle_col=45,
         main = "HSFs",
         annotation_legend = TRUE,
         display_numbers = Niben_HSFs_DEG %>% select(1:15),
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(84, 104),
         annotation_row = Niben_HSFs_Modules_Chl,
         annotation_colors = module_Chl_colors_2
         #gaps_col = c(15)
)


rownames(Niben_HSFs_TPM) <- Niben_HSFs_TPM$Protein
Niben_HSFs_TPM[1] <- NULL

min(Niben_HSFs_TPM, na.rm = TRUE)
max(Niben_HSFs_TPM, na.rm = TRUE)
breaks_Niben_HSFs_TPM=c(seq(0,100, length.out = 100))

Niben_HSFs_TPM_norm <- normalize_df(Niben_HSFs_TPM)
Niben_HSFs_TPM_norm_Cornell <- Niben_HSFs_TPM %>%
  select(1:48) 
Niben_HSFs_TPM_norm_Cornell <- normalize_df(Niben_HSFs_TPM_norm_Cornell)


pheatmap(Niben_HSFs_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Niben_HSFs_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_HSFs_Annotation, annotation_colors=color_HSFs, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "HSFs",
         annotation_legend = TRUE,
         #display_numbers = Niben_HSFs_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         annotation_row = Niben_HSFs_Modules,
         annotation_colors = module_colors,
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(24, 48, 51)
)



#####Representing Heatmap for Sly

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Sly_HSFs_Ann <- read.table("Funcional Groups/FigTree/Sly_HSFs.txt",
                                 sep = "\t", header = TRUE)

colnames(Sly_HSFs_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Sly_HSFs <- merge(Sly_HSFs_Ann, Summary_Yazdani_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Sly_HSFs_TPM <- Sly_HSFs %>% select(Protein, 4:66)
Sly_HSFs_TPM <- Sly_HSFs_TPM[!duplicated(Sly_HSFs_TPM$Protein),]
Sly_HSFs_TPM <- Sly_HSFs_TPM[order(match(Sly_HSFs_TPM[,1], Sly_HSFs_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_HSFs_FC <- Sly_HSFs %>% select(Protein, 67:71)
Sly_HSFs_FC <- Sly_HSFs_FC[!duplicated(Sly_HSFs_FC$Protein),]
Sly_HSFs_FC <- Sly_HSFs_FC[order(match(Sly_HSFs_FC[,1], Sly_HSFs_Ann[,2])),]


## We select only the FC data from the different comparisons
Sly_HSFs_DEG <- Sly_HSFs %>% select(Protein, 72:76)
Sly_HSFs_DEG <- Sly_HSFs_DEG[!duplicated(Sly_HSFs_DEG$Protein),]
Sly_HSFs_DEG <- Sly_HSFs_DEG[order(match(Sly_HSFs_DEG[,1], Sly_HSFs_Ann[,2])),]

Sly_HSFs_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Sly_HSFs_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Sly_HSFs_DEG$Ripening_Orange_DEG <- ifelse(is.na(Sly_HSFs_DEG$Ripening_Orange_DEG) == TRUE, " ", "*")
Sly_HSFs_DEG$Ripening_Red_DEG <- ifelse(is.na(Sly_HSFs_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
Sly_HSFs_DEG$ORHis_MG_DEG <- ifelse(is.na(Sly_HSFs_DEG$ORHis_MG_DEG) == TRUE, " ", "*")
Sly_HSFs_DEG$ORHis_Breaker_DEG <- ifelse(is.na(Sly_HSFs_DEG$ORHis_Breaker_DEG) == TRUE, " ", "*")
row.names(Sly_HSFs_DEG) <- Sly_HSFs_DEG$Protein
Sly_HSFs_DEG[1] <- NULL

rownames(Sly_HSFs_FC) <- Sly_HSFs_FC$Protein
Sly_HSFs_FC[1] <- NULL

min(Sly_HSFs_FC, na.rm = TRUE)
max(Sly_HSFs_FC, na.rm = TRUE)
breaks_Sly_HSFs=c(seq(-3,0, length.out = 100), seq(0.01,3,length.out = 100))

pheatmap(Sly_HSFs_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Sly_HSFs,
         #breaks = breaks_Stndr,
         legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_HSFs_Annotation, annotation_colors=color_HSFs, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "HSFs",
         annotation_legend = TRUE,
         display_numbers = Sly_HSFs_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(39, 48),
         gaps_col = c(3)
)


rownames(Sly_HSFs_TPM) <- Sly_HSFs_TPM$Protein
Sly_HSFs_TPM[1] <- NULL

min(Sly_HSFs_TPM, na.rm = TRUE)
max(Sly_HSFs_TPM, na.rm = TRUE)
breaks_Sly_HSFs_TPM=c(seq(0,100, length.out = 200))

Sly_HSFs_TPM_norm <- normalize_df(Sly_HSFs_TPM)

pheatmap(Sly_HSFs_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Sly_HSFs_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_HSFs_Annotation, annotation_colors=color_HSFs, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "HSFs",
         annotation_legend = TRUE,
         #display_numbers = Sly_HSFs_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(21,42)
)


#####Representing Heatmap for Cann

setwd("/Users/salva/GoogleDrive/My Drive/CRAG/RNAseq/Milan/RNAseq_Milan")
Cann_HSFs_Ann <- read.table("Funcional Groups/FigTree/Cann_HSFs.txt",
                                  sep = "\t", header = TRUE)

colnames(Cann_HSFs_Ann) <- c("Name", "Protein")

## We merge it with the big df created in the "Summary_RNAseq_Milan.R script with all the information

Cann_HSFs <- merge(Cann_HSFs_Ann, Summary_Pepper_TPM_FC_Ann, by.x=1, by.y = 1)

## We select only the FC data from the different comparisons
Cann_HSFs_TPM <- Cann_HSFs %>% select(Protein, 4:17)
Cann_HSFs_TPM <- Cann_HSFs_TPM[!duplicated(Cann_HSFs_TPM$Protein),]
Cann_HSFs_TPM <- Cann_HSFs_TPM[order(match(Cann_HSFs_TPM[,1], Cann_HSFs_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_HSFs_FC <- Cann_HSFs %>% select(Protein, 18:19)
Cann_HSFs_FC <- Cann_HSFs_FC[!duplicated(Cann_HSFs_FC$Protein),]
Cann_HSFs_FC <- Cann_HSFs_FC[order(match(Cann_HSFs_FC[,1], Cann_HSFs_Ann[,2])),]


## We select only the FC data from the different comparisons
Cann_HSFs_DEG <- Cann_HSFs %>% select(Protein, 20:21)
Cann_HSFs_DEG <- Cann_HSFs_DEG[!duplicated(Cann_HSFs_DEG$Protein),]
Cann_HSFs_DEG <- Cann_HSFs_DEG[order(match(Cann_HSFs_DEG[,1], Cann_HSFs_Ann[,2])),]

Cann_HSFs_DEG$Ripening_Breaker_DEG <- ifelse(is.na(Cann_HSFs_DEG$Ripening_Breaker_DEG) == TRUE, " ", "*")
Cann_HSFs_DEG$Ripening_Red_DEG <- ifelse(is.na(Cann_HSFs_DEG$Ripening_Red_DEG) == TRUE, " ", "*")
row.names(Cann_HSFs_DEG) <- Cann_HSFs_DEG$Protein
Cann_HSFs_DEG[1] <- NULL

rownames(Cann_HSFs_FC) <- Cann_HSFs_FC$Protein
Cann_HSFs_FC[1] <- NULL

min(Cann_HSFs_FC, na.rm = TRUE)
max(Cann_HSFs_FC, na.rm = TRUE)
breaks_Cann_HSFs=c(seq(-4.5,0, length.out = 100), seq(0.01,6,length.out = 100))

pheatmap(Cann_HSFs_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Cann_HSFs,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_HSFs_Annotation, annotation_colors=color_HSFs, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Breaker", "Ripening_Red"),
         cex = 1,
         angle_col=45,
         main = "HSFs",
         annotation_legend = TRUE,
         display_numbers = Cann_HSFs_DEG,
         fontsize_number = 15,
         na_col = "darkgrey"
         #gaps_row = c(22, 29, 49, 77),
         #gaps_col = c(2,5)
)


rownames(Cann_HSFs_TPM) <- Cann_HSFs_TPM$Protein
Cann_HSFs_TPM[1] <- NULL

min(Cann_HSFs_TPM, na.rm = TRUE)
max(Cann_HSFs_TPM, na.rm = TRUE)
breaks_Cann_HSFs_TPM=c(seq(0,100, length.out = 200))

Cann_HSFs_TPM_norm <- normalize_df(Cann_HSFs_TPM)

pheatmap(Cann_HSFs_TPM_norm, 
         border_color = "grey60",legend= TRUE,
         #color=color3, 
         cluster_rows =FALSE,
         #breaks = breaks_Cann_HSFs_TPM,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_HSFs_Annotation, annotation_colors=color_HSFs, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         #labels_col = c("Ripening_Breaker", "Ripening_Orange", "Ripening_Red",  "ORHis_MG", "ORHis_Breaker"),
         cex = 1,
         angle_col=45,
         main = "HSFs",
         annotation_legend = TRUE,
         #display_numbers = Cann_HSFs_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(22, 29, 49, 77),
         gaps_col = c(4,9)
)


## Comparativa 96h_SlyYaz_Ath

Ath_HSFs_Ann <- read.table("Funcional Groups/Ath_HSFs.txt", header = TRUE)

Ath_HSFs_FC <- merge(Ath_HSFs_Ann, Summary_Ath_FC, by.x=1, by.y = 0) %>%
  dplyr::select(Name, log2FoldChange)
Ath_HSFs_FC <- Ath_HSFs_FC[order(match(Ath_HSFs_FC[,1], Ath_HSFs_Ann[,2])),]

Ath_HSFs_DEG <- merge(Ath_HSFs_Ann, Summary_Ath_DEGs, by.x=1, by.y = 1, all.x = TRUE) %>%
  dplyr::select(Name, log2FC)
Ath_HSFs_DEG <- Ath_HSFs_DEG[order(match(Ath_HSFs_DEG[,1], Ath_HSFs_Ann[,2])),]

Ath_HSFs_DEG$log2FC <- ifelse(is.na(Ath_HSFs_DEG$log2FC) == TRUE, " ", "*")
row.names(Ath_HSFs_DEG) <- Ath_HSFs_DEG$Name
Ath_HSFs_DEG[1] <- NULL

rownames(Ath_HSFs_FC) <- Ath_HSFs_FC$Name
Ath_HSFs_FC[1] <- NULL

min(Ath_HSFs_FC, na.rm = TRUE)
max(Ath_HSFs_FC, na.rm = TRUE)
breaks_Ath_HSFs=c(seq(-4.5,0, length.out = 100), seq(0.01,4.5,length.out = 100))

pheatmap(Ath_HSFs_FC, 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Stndr,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-4, -2, 0, 2, 4),
         #legend_labels = c("<-4", "-2", "0", "2", ">4"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Cann_HSFs_Annotation, annotation_colors=color_HSFs, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = "Senescence",
         cex = 1,
         angle_col=45,
         main = "HSFs",
         annotation_legend = TRUE,
         display_numbers = Ath_HSFs_DEG,
         fontsize_number = 15,
         na_col = "darkgrey",
         gaps_row = c(1, 2, 3, 4, 5, 6, 7),
         #gaps_col = c(2,5)
)

pheatmap(Niben_HSFs_FC %>% select(16), 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         #breaks = breaks_HSFs_FC,
         breaks = breaks_Stndr,
         #legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         #legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Niben_HSFs_Annotation, annotation_colors=color_HSFs, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = "96h",
         cex = 1,
         angle_col=45,
         main = "HSFs",
         annotation_legend = TRUE,
         display_numbers = Niben_HSFs_DEG %>% select(16),
         fontsize_number = 15,
         na_col = "darkgrey",
         #gaps_row = c(84, 104),
         annotation_row = Niben_HSFs_Modules_Chl,
         annotation_colors = module_Chl_colors_2,
         gaps_row = c(10, 12,14,15,17,21)
)


pheatmap(Sly_HSFs_FC %>% select(2,3), 
         border_color = "grey60",legend= TRUE,
         color=color2, 
         cluster_rows =FALSE,
         breaks = breaks_Stndr,
         #breaks = breaks_Stndr,
         #legend_breaks = c(-3, -1.5, 0, 1.5, 3),
         #legend_labels = c("<-3", "-1.5", "0", "1.5", ">3"),
         #legend_breaks = c(0, 40, 80, 120, 160, 200),
         #legend_labels = c("0", "40", "80", "120", "160", ">200"),
         cluster_cols = FALSE,
         show_colnames=T, treeheight_col = 35, 
         fontsize=12, 
         scale = "none",
         #annotation_row=Sly_HSFs_Annotation, annotation_colors=color_HSFs, 
         #annotation_names_row = TRUE, 
         show_rownames = TRUE,  
         #cellwidth = 15, cellheight = 15,
         fontsize_row = 7,
         fontsize_col = 8,
         labels_col = c("Ripening_Orange", "Ripening_Red"),
         cex = 1,
         angle_col=45,
         main = "HSFs",
         annotation_legend = TRUE,
         display_numbers = Sly_HSFs_DEG %>% select(2,3),
         fontsize_number = 15,
         na_col = "darkgrey",
         gaps_row = c(3, 4, 5, 6, 7, 8)
)




